<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.32">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>3&nbsp; 注释数据库 – Bioinformatics</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./ExpressionSet.html" rel="next">
<link href="./biodata_structure.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-2486e1f0a3ee9ee1fc393803a1361cdb.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-f90a1b115807ea56b20e50efb3bb25e2.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="site_libs/datatables-binding-0.33/datatables.js"></script><script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
</head>
<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./biodata_structure.html">数据结构与数据库</a></li><li class="breadcrumb-item"><a href="./annotation_database.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">注释数据库</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Bioinformatics</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/Wanganlin00/Bioinformatics" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">前言</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./upstream_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">上游分析</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">数据结构与数据库</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./biodata_structure.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">生物数据结构</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./annotation_database.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">注释数据库</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ExpressionSet.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title"><code>ExpressionSet</code></span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./SummarizedExperiment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title"><code>SummarizedExperiment</code></span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./SingleCellExperiment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title"><code>SingleCellExperiment</code></span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">下游分析</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./DEGs_Analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">差异表达基因分析</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./EnrichmentAnalysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">富集分析</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PseudotimeAnalysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">伪时序分析</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">单细胞分析</span></span>
  </li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li>
<a href="#biomart" id="toc-biomart" class="nav-link active" data-scroll-target="#biomart"><span class="header-section-number">3.1</span> BioMart</a>
  <ul>
<li><a href="#%E6%98%A0%E5%B0%84" id="toc-映射" class="nav-link" data-scroll-target="#%E6%98%A0%E5%B0%84"><span class="header-section-number">3.1.1</span> 映射</a></li>
  <li><a href="#%E4%BD%BF%E7%94%A8-ensembl-%E7%9A%84%E5%AD%98%E6%A1%A3%E7%89%88%E6%9C%AC" id="toc-使用-ensembl-的存档版本" class="nav-link" data-scroll-target="#%E4%BD%BF%E7%94%A8-ensembl-%E7%9A%84%E5%AD%98%E6%A1%A3%E7%89%88%E6%9C%AC"><span class="header-section-number">3.1.2</span> <strong>使用 Ensembl 的存档版本</strong></a></li>
  <li><a href="#%E4%BD%BF%E7%94%A8-ensembl-%E5%9F%BA%E5%9B%A0%E7%BB%84" id="toc-使用-ensembl-基因组" class="nav-link" data-scroll-target="#%E4%BD%BF%E7%94%A8-ensembl-%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="header-section-number">3.1.3</span> <strong>使用 Ensembl 基因组</strong></a></li>
  </ul>
</li>
  <li>
<a href="#annotationhub" id="toc-annotationhub" class="nav-link" data-scroll-target="#annotationhub"><span class="header-section-number">3.2</span> AnnotationHub</a>
  <ul>
<li><a href="#%E5%AD%90%E9%9B%86" id="toc-子集" class="nav-link" data-scroll-target="#%E5%AD%90%E9%9B%86"><span class="header-section-number">3.2.1</span> 子集</a></li>
  <li><a href="#%E6%9F%A5%E8%AF%A2" id="toc-查询" class="nav-link" data-scroll-target="#%E6%9F%A5%E8%AF%A2"><span class="header-section-number">3.2.2</span> 查询</a></li>
  </ul>
</li>
  <li><a href="#bsgenome" id="toc-bsgenome" class="nav-link" data-scroll-target="#bsgenome"><span class="header-section-number">3.3</span> BSgenome</a></li>
  <li>
<a href="#annotationdbi" id="toc-annotationdbi" class="nav-link" data-scroll-target="#annotationdbi"><span class="header-section-number">3.4</span> AnnotationDbi</a>
  <ul>
<li><a href="#orgdb" id="toc-orgdb" class="nav-link" data-scroll-target="#orgdb"><span class="header-section-number">3.4.1</span> OrgDb</a></li>
  </ul>
</li>
  <li><a href="#experimenthub" id="toc-experimenthub" class="nav-link" data-scroll-target="#experimenthub"><span class="header-section-number">3.5</span> ExperimentHub</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/Wanganlin00/Bioinformatics/edit/main/annotation_database.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/Wanganlin00/Bioinformatics/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./biodata_structure.html">数据结构与数据库</a></li><li class="breadcrumb-item"><a href="./annotation_database.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">注释数据库</span></a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">
<span class="chapter-number">3</span>&nbsp; <span class="chapter-title">注释数据库</span>
</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><p><a href="https://bioconductor.org/packages/release/workflows/html/annotation.html" class="uri">https://bioconductor.org/packages/release/workflows/html/annotation.html</a></p>
<section id="biomart" class="level2" data-number="3.1"><h2 data-number="3.1" class="anchored" data-anchor-id="biomart">
<span class="header-section-number">3.1</span> <a href="https://bioconductor.org/packages/release/bioc/html/biomaRt.html">BioMart</a>
</h2>
<p>即Ensembl ID</p>
<p>查找智人（homo sapien）基因组</p>
<p><a href="https://bioconductor.org/packages/release/bioc/vignettes/biomaRt/inst/doc/accessing_ensembl.html#introduction" class="uri">https://bioconductor.org/packages/release/bioc/vignettes/biomaRt/inst/doc/accessing_ensembl.html#introduction</a></p>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># BiocManager::install("biomaRt")</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/grimbough/biomaRt">biomaRt</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listEnsembl.html">listEnsembl</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="kable-table">
<table class="caption-top table table-sm table-striped small">
<thead><tr class="header">
<th style="text-align: left;">biomart</th>
<th style="text-align: left;">version</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">genes</td>
<td style="text-align: left;">Ensembl Genes 113</td>
</tr>
<tr class="even">
<td style="text-align: left;">mouse_strains</td>
<td style="text-align: left;">Mouse strains 113</td>
</tr>
<tr class="odd">
<td style="text-align: left;">snps</td>
<td style="text-align: left;">Ensembl Variation 113</td>
</tr>
<tr class="even">
<td style="text-align: left;">regulation</td>
<td style="text-align: left;">Ensembl Regulation 113</td>
</tr>
</tbody>
</table>
</div>
</div>
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span></span>
<span><span class="co"># 连接到Ensembl BioMart</span></span>
<span><span class="va">ensembl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useEnsembl.html">useEnsembl</a></span><span class="op">(</span>biomart <span class="op">=</span> <span class="st">"genes"</span><span class="op">)</span></span>
<span><span class="va">ensembl</span></span>
<span><span class="co">#&gt; Object of class 'Mart':</span></span>
<span><span class="co">#&gt;   Using the ENSEMBL_MART_ENSEMBL BioMart database</span></span>
<span><span class="co">#&gt;   No dataset selected.</span></span>
<span></span>
<span><span class="co"># 列出数据集</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listDatasets.html">listDatasets</a></span><span class="op">(</span><span class="va">ensembl</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="kable-table">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 25%">
<col style="width: 50%">
<col style="width: 23%">
</colgroup>
<thead><tr class="header">
<th style="text-align: left;">dataset</th>
<th style="text-align: left;">description</th>
<th style="text-align: left;">version</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">abrachyrhynchus_gene_ensembl</td>
<td style="text-align: left;">Pink-footed goose genes (ASM259213v1)</td>
<td style="text-align: left;">ASM259213v1</td>
</tr>
<tr class="even">
<td style="text-align: left;">acalliptera_gene_ensembl</td>
<td style="text-align: left;">Eastern happy genes (fAstCal1.3)</td>
<td style="text-align: left;">fAstCal1.3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">acarolinensis_gene_ensembl</td>
<td style="text-align: left;">Green anole genes (AnoCar2.0v2)</td>
<td style="text-align: left;">AnoCar2.0v2</td>
</tr>
<tr class="even">
<td style="text-align: left;">acchrysaetos_gene_ensembl</td>
<td style="text-align: left;">Golden eagle genes (bAquChr1.2)</td>
<td style="text-align: left;">bAquChr1.2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">acitrinellus_gene_ensembl</td>
<td style="text-align: left;">Midas cichlid genes (Midas_v5)</td>
<td style="text-align: left;">Midas_v5</td>
</tr>
<tr class="even">
<td style="text-align: left;">amelanoleuca_gene_ensembl</td>
<td style="text-align: left;">Giant panda genes (ASM200744v2)</td>
<td style="text-align: left;">ASM200744v2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">amexicanus_gene_ensembl</td>
<td style="text-align: left;">Mexican tetra genes (Astyanax_mexicanus-2.0)</td>
<td style="text-align: left;">Astyanax_mexicanus-2.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">anancymaae_gene_ensembl</td>
<td style="text-align: left;">Ma’s night monkey genes (Anan_2.0)</td>
<td style="text-align: left;">Anan_2.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">aocellaris_gene_ensembl</td>
<td style="text-align: left;">Clown anemonefish genes (ASM2253959v1)</td>
<td style="text-align: left;">ASM2253959v1</td>
</tr>
<tr class="even">
<td style="text-align: left;">apercula_gene_ensembl</td>
<td style="text-align: left;">Orange clownfish genes (Nemo_v1)</td>
<td style="text-align: left;">Nemo_v1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">aplatyrhynchos_gene_ensembl</td>
<td style="text-align: left;">Mallard genes (ASM874695v1)</td>
<td style="text-align: left;">ASM874695v1</td>
</tr>
<tr class="even">
<td style="text-align: left;">apolyacanthus_gene_ensembl</td>
<td style="text-align: left;">Spiny chromis genes (ASM210954v1)</td>
<td style="text-align: left;">ASM210954v1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">applatyrhynchos_gene_ensembl</td>
<td style="text-align: left;">Duck genes (CAU_duck1.0)</td>
<td style="text-align: left;">CAU_duck1.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">atestudineus_gene_ensembl</td>
<td style="text-align: left;">Climbing perch genes (fAnaTes1.3)</td>
<td style="text-align: left;">fAnaTes1.3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">bbbison_gene_ensembl</td>
<td style="text-align: left;">American bison genes (Bison_UMD1.0)</td>
<td style="text-align: left;">Bison_UMD1.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">bgrunniens_gene_ensembl</td>
<td style="text-align: left;">Domestic yak genes (LU_Bosgru_v3.0)</td>
<td style="text-align: left;">LU_Bosgru_v3.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">bihybrid_gene_ensembl</td>
<td style="text-align: left;">Hybrid - Bos Indicus genes (UOA_Brahman_1)</td>
<td style="text-align: left;">UOA_Brahman_1</td>
</tr>
<tr class="even">
<td style="text-align: left;">bmusculus_gene_ensembl</td>
<td style="text-align: left;">Blue whale genes (mBalMus1.v2)</td>
<td style="text-align: left;">mBalMus1.v2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">bmutus_gene_ensembl</td>
<td style="text-align: left;">Wild yak genes (BosGru_v2.0)</td>
<td style="text-align: left;">BosGru_v2.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">bsplendens_gene_ensembl</td>
<td style="text-align: left;">Siamese fighting fish genes (fBetSpl5.2)</td>
<td style="text-align: left;">fBetSpl5.2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">btaurus_gene_ensembl</td>
<td style="text-align: left;">Cow genes (ARS-UCD1.3)</td>
<td style="text-align: left;">ARS-UCD1.3</td>
</tr>
<tr class="even">
<td style="text-align: left;">cabingdonii_gene_ensembl</td>
<td style="text-align: left;">Abingdon island giant tortoise genes (ASM359739v1)</td>
<td style="text-align: left;">ASM359739v1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">catys_gene_ensembl</td>
<td style="text-align: left;">Sooty mangabey genes (Caty_1.0)</td>
<td style="text-align: left;">Caty_1.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">cauratus_gene_ensembl</td>
<td style="text-align: left;">Goldfish genes (ASM336829v1)</td>
<td style="text-align: left;">ASM336829v1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">cccarpio_gene_ensembl</td>
<td style="text-align: left;">Common carp genes (Cypcar_WagV4.0)</td>
<td style="text-align: left;">Cypcar_WagV4.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">cdromedarius_gene_ensembl</td>
<td style="text-align: left;">Arabian camel genes (CamDro2)</td>
<td style="text-align: left;">CamDro2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">celegans_gene_ensembl</td>
<td style="text-align: left;">Caenorhabditis elegans (Nematode, N2) genes (WBcel235)</td>
<td style="text-align: left;">WBcel235</td>
</tr>
<tr class="even">
<td style="text-align: left;">cgchok1gshd_gene_ensembl</td>
<td style="text-align: left;">Chinese hamster CHOK1GS genes (CHOK1GS_HDv1)</td>
<td style="text-align: left;">CHOK1GS_HDv1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">cgobio_gene_ensembl</td>
<td style="text-align: left;">Channel bull blenny genes (fCotGob3.1)</td>
<td style="text-align: left;">fCotGob3.1</td>
</tr>
<tr class="even">
<td style="text-align: left;">charengus_gene_ensembl</td>
<td style="text-align: left;">Atlantic herring genes (Ch_v2.0.2v2)</td>
<td style="text-align: left;">Ch_v2.0.2v2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">chircus_gene_ensembl</td>
<td style="text-align: left;">Goat genes (ARS1)</td>
<td style="text-align: left;">ARS1</td>
</tr>
<tr class="even">
<td style="text-align: left;">choffmanni_gene_ensembl</td>
<td style="text-align: left;">Sloth genes (choHof1)</td>
<td style="text-align: left;">choHof1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">chyarkandensis_gene_ensembl</td>
<td style="text-align: left;">Yarkand deer genes (CEY_v1)</td>
<td style="text-align: left;">CEY_v1</td>
</tr>
<tr class="even">
<td style="text-align: left;">cimitator_gene_ensembl</td>
<td style="text-align: left;">Panamanian white-faced capuchin genes (Cebus_imitator-1.0)</td>
<td style="text-align: left;">Cebus_imitator-1.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">cintestinalis_gene_ensembl</td>
<td style="text-align: left;">C.intestinalis genes (KH)</td>
<td style="text-align: left;">KH</td>
</tr>
<tr class="even">
<td style="text-align: left;">cjacchus_gene_ensembl</td>
<td style="text-align: left;">White-tufted-ear marmoset genes (mCalJac1.pat.X)</td>
<td style="text-align: left;">mCalJac1.pat.X</td>
</tr>
<tr class="odd">
<td style="text-align: left;">cjaponica_gene_ensembl</td>
<td style="text-align: left;">Japanese quail genes (Coturnix_japonica_2.0)</td>
<td style="text-align: left;">Coturnix_japonica_2.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">clanigera_gene_ensembl</td>
<td style="text-align: left;">Long-tailed chinchilla genes (ChiLan1.0)</td>
<td style="text-align: left;">ChiLan1.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">cldingo_gene_ensembl</td>
<td style="text-align: left;">Dingo genes (ASM325472v1)</td>
<td style="text-align: left;">ASM325472v1</td>
</tr>
<tr class="even">
<td style="text-align: left;">clfamiliaris_gene_ensembl</td>
<td style="text-align: left;">Dog genes (ROS_Cfam_1.0)</td>
<td style="text-align: left;">ROS_Cfam_1.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">clumpus_gene_ensembl</td>
<td style="text-align: left;">Lumpfish genes (fCycLum1.pri)</td>
<td style="text-align: left;">fCycLum1.pri</td>
</tr>
<tr class="even">
<td style="text-align: left;">cmilii_gene_ensembl</td>
<td style="text-align: left;">Elephant shark genes (Callorhinchus_milii-6.1.3)</td>
<td style="text-align: left;">Callorhinchus_milii-6.1.3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">cpbellii_gene_ensembl</td>
<td style="text-align: left;">Painted turtle genes (Chrysemys_picta_bellii-3.0.3)</td>
<td style="text-align: left;">Chrysemys_picta_bellii-3.0.3</td>
</tr>
<tr class="even">
<td style="text-align: left;">cporcellus_gene_ensembl</td>
<td style="text-align: left;">Guinea Pig genes (Cavpor3.0)</td>
<td style="text-align: left;">Cavpor3.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">cporosus_gene_ensembl</td>
<td style="text-align: left;">Australian saltwater crocodile genes (CroPor_comp1)</td>
<td style="text-align: left;">CroPor_comp1</td>
</tr>
<tr class="even">
<td style="text-align: left;">csabaeus_gene_ensembl</td>
<td style="text-align: left;">Vervet-AGM genes (ChlSab1.1)</td>
<td style="text-align: left;">ChlSab1.1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">csavignyi_gene_ensembl</td>
<td style="text-align: left;">C.savignyi genes (CSAV 2.0)</td>
<td style="text-align: left;">CSAV 2.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">csemilaevis_gene_ensembl</td>
<td style="text-align: left;">Tongue sole genes (Cse_v1.0)</td>
<td style="text-align: left;">Cse_v1.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">csyrichta_gene_ensembl</td>
<td style="text-align: left;">Tarsier genes (Tarsius_syrichta-2.0.1)</td>
<td style="text-align: left;">Tarsius_syrichta-2.0.1</td>
</tr>
<tr class="even">
<td style="text-align: left;">cvariegatus_gene_ensembl</td>
<td style="text-align: left;">Sheepshead minnow genes (C_variegatus-1.0)</td>
<td style="text-align: left;">C_variegatus-1.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">cwagneri_gene_ensembl</td>
<td style="text-align: left;">Chacoan peccary genes (CatWag_v2_BIUU_UCD)</td>
<td style="text-align: left;">CatWag_v2_BIUU_UCD</td>
</tr>
<tr class="even">
<td style="text-align: left;">dclupeoides_gene_ensembl</td>
<td style="text-align: left;">Denticle herring genes (fDenClu1.2)</td>
<td style="text-align: left;">fDenClu1.2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">dlabrax_gene_ensembl</td>
<td style="text-align: left;">European seabass genes (dlabrax2021)</td>
<td style="text-align: left;">dlabrax2021</td>
</tr>
<tr class="even">
<td style="text-align: left;">dleucas_gene_ensembl</td>
<td style="text-align: left;">Beluga whale genes (ASM228892v3)</td>
<td style="text-align: left;">ASM228892v3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">dmelanogaster_gene_ensembl</td>
<td style="text-align: left;">Drosophila melanogaster (Fruit fly) genes (BDGP6.46)</td>
<td style="text-align: left;">BDGP6.46</td>
</tr>
<tr class="even">
<td style="text-align: left;">dnovemcinctus_gene_ensembl</td>
<td style="text-align: left;">Armadillo genes (Dasnov3.0)</td>
<td style="text-align: left;">Dasnov3.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">dordii_gene_ensembl</td>
<td style="text-align: left;">Kangaroo rat genes (Dord_2.0)</td>
<td style="text-align: left;">Dord_2.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">drerio_gene_ensembl</td>
<td style="text-align: left;">Zebrafish genes (GRCz11)</td>
<td style="text-align: left;">GRCz11</td>
</tr>
<tr class="odd">
<td style="text-align: left;">easinus_gene_ensembl</td>
<td style="text-align: left;">Donkey genes (ASM1607732v2)</td>
<td style="text-align: left;">ASM1607732v2</td>
</tr>
<tr class="even">
<td style="text-align: left;">eburgeri_gene_ensembl</td>
<td style="text-align: left;">Hagfish genes (Eburgeri_3.2)</td>
<td style="text-align: left;">Eburgeri_3.2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ecaballus_gene_ensembl</td>
<td style="text-align: left;">Horse genes (EquCab3.0)</td>
<td style="text-align: left;">EquCab3.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">ecalabaricus_gene_ensembl</td>
<td style="text-align: left;">Reedfish genes (fErpCal1.1)</td>
<td style="text-align: left;">fErpCal1.1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">eelectricus_gene_ensembl</td>
<td style="text-align: left;">Electric eel genes (fEleEle1.pri)</td>
<td style="text-align: left;">fEleEle1.pri</td>
</tr>
<tr class="even">
<td style="text-align: left;">eeuropaeus_gene_ensembl</td>
<td style="text-align: left;">Hedgehog genes (eriEur1)</td>
<td style="text-align: left;">eriEur1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">elucius_gene_ensembl</td>
<td style="text-align: left;">Northern pike genes (fEsoLuc1.pri)</td>
<td style="text-align: left;">fEsoLuc1.pri</td>
</tr>
<tr class="even">
<td style="text-align: left;">etelfairi_gene_ensembl</td>
<td style="text-align: left;">Lesser hedgehog tenrec genes (TENREC)</td>
<td style="text-align: left;">TENREC</td>
</tr>
<tr class="odd">
<td style="text-align: left;">falbicollis_gene_ensembl</td>
<td style="text-align: left;">Collared flycatcher genes (FicAlb1.5)</td>
<td style="text-align: left;">FicAlb1.5</td>
</tr>
<tr class="even">
<td style="text-align: left;">fcatus_gene_ensembl</td>
<td style="text-align: left;">Cat genes (Felis_catus_9.0)</td>
<td style="text-align: left;">Felis_catus_9.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">fheteroclitus_gene_ensembl</td>
<td style="text-align: left;">Mummichog genes (Fundulus_heteroclitus-3.0.2)</td>
<td style="text-align: left;">Fundulus_heteroclitus-3.0.2</td>
</tr>
<tr class="even">
<td style="text-align: left;">gaculeatus_gene_ensembl</td>
<td style="text-align: left;">Stickleback genes (GAculeatus_UGA_version5)</td>
<td style="text-align: left;">GAculeatus_UGA_version5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">gevgoodei_gene_ensembl</td>
<td style="text-align: left;">Goodes thornscrub tortoise genes (rGopEvg1_v1.p)</td>
<td style="text-align: left;">rGopEvg1_v1.p</td>
</tr>
<tr class="even">
<td style="text-align: left;">gfortis_gene_ensembl</td>
<td style="text-align: left;">Medium ground-finch genes (GeoFor_1.0)</td>
<td style="text-align: left;">GeoFor_1.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ggallus_gene_ensembl</td>
<td style="text-align: left;">Chicken genes (bGalGal1.mat.broiler.GRCg7b)</td>
<td style="text-align: left;">bGalGal1.mat.broiler.GRCg7b</td>
</tr>
<tr class="even">
<td style="text-align: left;">ggorilla_gene_ensembl</td>
<td style="text-align: left;">Gorilla genes (gorGor4)</td>
<td style="text-align: left;">gorGor4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">gmorhua_gene_ensembl</td>
<td style="text-align: left;">Atlantic cod genes (gadMor3.0)</td>
<td style="text-align: left;">gadMor3.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">hburtoni_gene_ensembl</td>
<td style="text-align: left;">Burton’s mouthbrooder genes (AstBur1.0)</td>
<td style="text-align: left;">AstBur1.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">hcomes_gene_ensembl</td>
<td style="text-align: left;">Tiger tail seahorse genes (H_comes_QL1_v1)</td>
<td style="text-align: left;">H_comes_QL1_v1</td>
</tr>
<tr class="even">
<td style="text-align: left;">hgfemale_gene_ensembl</td>
<td style="text-align: left;">Naked mole-rat female genes (Naked_mole-rat_maternal)</td>
<td style="text-align: left;">Naked_mole-rat_maternal</td>
</tr>
<tr class="odd">
<td style="text-align: left;">hhucho_gene_ensembl</td>
<td style="text-align: left;">Huchen genes (ASM331708v1)</td>
<td style="text-align: left;">ASM331708v1</td>
</tr>
<tr class="even">
<td style="text-align: left;">hsapiens_gene_ensembl</td>
<td style="text-align: left;">Human genes (GRCh38.p14)</td>
<td style="text-align: left;">GRCh38.p14</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ipunctatus_gene_ensembl</td>
<td style="text-align: left;">Channel catfish genes (ASM400665v3)</td>
<td style="text-align: left;">ASM400665v3</td>
</tr>
<tr class="even">
<td style="text-align: left;">itridecemlineatus_gene_ensembl</td>
<td style="text-align: left;">Squirrel genes (SpeTri2.0)</td>
<td style="text-align: left;">SpeTri2.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">jjaculus_gene_ensembl</td>
<td style="text-align: left;">Lesser Egyptian jerboa genes (JacJac1.0)</td>
<td style="text-align: left;">JacJac1.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">kmarmoratus_gene_ensembl</td>
<td style="text-align: left;">Mangrove rivulus genes (ASM164957v1)</td>
<td style="text-align: left;">ASM164957v1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">lafricana_gene_ensembl</td>
<td style="text-align: left;">Elephant genes (Loxafr3.0)</td>
<td style="text-align: left;">Loxafr3.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">lbergylta_gene_ensembl</td>
<td style="text-align: left;">Ballan wrasse genes (BallGen_V1)</td>
<td style="text-align: left;">BallGen_V1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">lcalcarifer_gene_ensembl</td>
<td style="text-align: left;">Barramundi perch genes (ASB_HGAPassembly_v1)</td>
<td style="text-align: left;">ASB_HGAPassembly_v1</td>
</tr>
<tr class="even">
<td style="text-align: left;">lchalumnae_gene_ensembl</td>
<td style="text-align: left;">Coelacanth genes (LatCha1)</td>
<td style="text-align: left;">LatCha1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">lcrocea_gene_ensembl</td>
<td style="text-align: left;">Large yellow croaker genes (L_crocea_2.0)</td>
<td style="text-align: left;">L_crocea_2.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">llaticaudata_gene_ensembl</td>
<td style="text-align: left;">Blue-ringed sea krait genes (latLat_1.0)</td>
<td style="text-align: left;">latLat_1.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">lleishanense_gene_ensembl</td>
<td style="text-align: left;">Leishan spiny toad genes (ASM966780v1)</td>
<td style="text-align: left;">ASM966780v1</td>
</tr>
<tr class="even">
<td style="text-align: left;">loculatus_gene_ensembl</td>
<td style="text-align: left;">Spotted gar genes (LepOcu1)</td>
<td style="text-align: left;">LepOcu1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">marmatus_gene_ensembl</td>
<td style="text-align: left;">Zig-zag eel genes (fMasArm1.2)</td>
<td style="text-align: left;">fMasArm1.2</td>
</tr>
<tr class="even">
<td style="text-align: left;">mauratus_gene_ensembl</td>
<td style="text-align: left;">Golden Hamster genes (MesAur1.0)</td>
<td style="text-align: left;">MesAur1.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mcaroli_gene_ensembl</td>
<td style="text-align: left;">Ryukyu mouse genes (CAROLI_EIJ_v1.1)</td>
<td style="text-align: left;">CAROLI_EIJ_v1.1</td>
</tr>
<tr class="even">
<td style="text-align: left;">mdomestica_gene_ensembl</td>
<td style="text-align: left;">Opossum genes (ASM229v1)</td>
<td style="text-align: left;">ASM229v1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mfascicularis_gene_ensembl</td>
<td style="text-align: left;">Crab-eating macaque genes (Macaca_fascicularis_6.0)</td>
<td style="text-align: left;">Macaca_fascicularis_6.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">mgallopavo_gene_ensembl</td>
<td style="text-align: left;">Turkey genes (Turkey_5.1)</td>
<td style="text-align: left;">Turkey_5.1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mleucophaeus_gene_ensembl</td>
<td style="text-align: left;">Drill genes (Mleu.le_1.0)</td>
<td style="text-align: left;">Mleu.le_1.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">mlucifugus_gene_ensembl</td>
<td style="text-align: left;">Microbat genes (Myoluc2.0)</td>
<td style="text-align: left;">Myoluc2.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mmmarmota_gene_ensembl</td>
<td style="text-align: left;">Alpine marmot genes (marMar2.1)</td>
<td style="text-align: left;">marMar2.1</td>
</tr>
<tr class="even">
<td style="text-align: left;">mmonoceros_gene_ensembl</td>
<td style="text-align: left;">Narwhal genes (NGI_Narwhal_1)</td>
<td style="text-align: left;">NGI_Narwhal_1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mmoschiferus_gene_ensembl</td>
<td style="text-align: left;">Siberian musk deer genes (MosMos_v2_BIUU_UCD)</td>
<td style="text-align: left;">MosMos_v2_BIUU_UCD</td>
</tr>
<tr class="even">
<td style="text-align: left;">mmulatta_gene_ensembl</td>
<td style="text-align: left;">Macaque genes (Mmul_10)</td>
<td style="text-align: left;">Mmul_10</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mmurdjan_gene_ensembl</td>
<td style="text-align: left;">Pinecone soldierfish genes (fMyrMur1.1)</td>
<td style="text-align: left;">fMyrMur1.1</td>
</tr>
<tr class="even">
<td style="text-align: left;">mmurinus_gene_ensembl</td>
<td style="text-align: left;">Mouse Lemur genes (Mmur_3.0)</td>
<td style="text-align: left;">Mmur_3.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mmusculus_gene_ensembl</td>
<td style="text-align: left;">Mouse genes (GRCm39)</td>
<td style="text-align: left;">GRCm39</td>
</tr>
<tr class="even">
<td style="text-align: left;">mnemestrina_gene_ensembl</td>
<td style="text-align: left;">Pig-tailed macaque genes (Mnem_1.0)</td>
<td style="text-align: left;">Mnem_1.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mochrogaster_gene_ensembl</td>
<td style="text-align: left;">Prairie vole genes (MicOch1.0)</td>
<td style="text-align: left;">MicOch1.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">mpahari_gene_ensembl</td>
<td style="text-align: left;">Shrew mouse genes (PAHARI_EIJ_v1.1)</td>
<td style="text-align: left;">PAHARI_EIJ_v1.1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mpfuro_gene_ensembl</td>
<td style="text-align: left;">Ferret genes (MusPutFur1.0)</td>
<td style="text-align: left;">MusPutFur1.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">mspicilegus_gene_ensembl</td>
<td style="text-align: left;">Steppe mouse genes (MUSP714)</td>
<td style="text-align: left;">MUSP714</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mspretus_gene_ensembl</td>
<td style="text-align: left;">Algerian mouse genes (SPRET_EiJ_v1)</td>
<td style="text-align: left;">SPRET_EiJ_v1</td>
</tr>
<tr class="even">
<td style="text-align: left;">mzebra_gene_ensembl</td>
<td style="text-align: left;">Zebra mbuna genes (M_zebra_UMD2a)</td>
<td style="text-align: left;">M_zebra_UMD2a</td>
</tr>
<tr class="odd">
<td style="text-align: left;">nbrichardi_gene_ensembl</td>
<td style="text-align: left;">Lyretail cichlid genes (NeoBri1.0)</td>
<td style="text-align: left;">NeoBri1.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">neugenii_gene_ensembl</td>
<td style="text-align: left;">Wallaby genes (Meug_1.0)</td>
<td style="text-align: left;">Meug_1.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">nfurzeri_gene_ensembl</td>
<td style="text-align: left;">Turquoise killifish genes (Nfu_20140520)</td>
<td style="text-align: left;">Nfu_20140520</td>
</tr>
<tr class="even">
<td style="text-align: left;">ngalili_gene_ensembl</td>
<td style="text-align: left;">Upper Galilee mountains blind mole rat genes (S.galili_v1.0)</td>
<td style="text-align: left;">S.galili_v1.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">nleucogenys_gene_ensembl</td>
<td style="text-align: left;">Gibbon genes (Nleu_3.0)</td>
<td style="text-align: left;">Nleu_3.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">nnaja_gene_ensembl</td>
<td style="text-align: left;">Indian cobra genes (Nana_v5)</td>
<td style="text-align: left;">Nana_v5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">nscutatus_gene_ensembl</td>
<td style="text-align: left;">Mainland tiger snake genes (TS10Xv2-PRI)</td>
<td style="text-align: left;">TS10Xv2-PRI</td>
</tr>
<tr class="even">
<td style="text-align: left;">nvison_gene_ensembl</td>
<td style="text-align: left;">American mink genes (NNQGG.v01)</td>
<td style="text-align: left;">NNQGG.v01</td>
</tr>
<tr class="odd">
<td style="text-align: left;">odegus_gene_ensembl</td>
<td style="text-align: left;">Degu genes (OctDeg1.0)</td>
<td style="text-align: left;">OctDeg1.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">okisutch_gene_ensembl</td>
<td style="text-align: left;">Coho salmon genes (Okis_V2)</td>
<td style="text-align: left;">Okis_V2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">oprinceps_gene_ensembl</td>
<td style="text-align: left;">Pika genes (OchPri2.0-Ens)</td>
<td style="text-align: left;">OchPri2.0-Ens</td>
</tr>
</tbody>
</table>
</div>
</div>
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span></span>
<span><span class="co"># 查找智人</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listDatasets.html">searchDatasets</a></span><span class="op">(</span>mart <span class="op">=</span> <span class="va">ensembl</span>, pattern <span class="op">=</span> <span class="st">"hsapiens"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="kable-table">
<table class="caption-top table table-sm table-striped small">
<thead><tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">dataset</th>
<th style="text-align: left;">description</th>
<th style="text-align: left;">version</th>
</tr></thead>
<tbody><tr class="odd">
<td style="text-align: left;">80</td>
<td style="text-align: left;">hsapiens_gene_ensembl</td>
<td style="text-align: left;">Human genes (GRCh38.p14)</td>
<td style="text-align: left;">GRCh38.p14</td>
</tr></tbody>
</table>
</div>
</div>
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span></span>
<span><span class="co"># 选择数据集</span></span>
<span><span class="va">dataset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useDataset.html">useDataset</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="st">"hsapiens_gene_ensembl"</span>, mart <span class="op">=</span> <span class="va">ensembl</span><span class="op">)</span></span>
<span><span class="va">dataset</span></span>
<span><span class="co">#&gt; Object of class 'Mart':</span></span>
<span><span class="co">#&gt;   Using the ENSEMBL_MART_ENSEMBL BioMart database</span></span>
<span><span class="co">#&gt;   Using the hsapiens_gene_ensembl dataset</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><a href="https://asia.ensembl.org/index.html">GRCh38（Genome Research Consortium human genome build 38）</a></p>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># 人类基因BioMart</span></span>
<span><span class="va">ensembl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useEnsembl.html">useEnsembl</a></span><span class="op">(</span>biomart <span class="op">=</span> <span class="st">"ensembl"</span>, </span>
<span>                   dataset <span class="op">=</span> <span class="st">"hsapiens_gene_ensembl"</span>, </span>
<span>                   mirror <span class="op">=</span> <span class="st">"asia"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="映射" class="level3" data-number="3.1.1"><h3 data-number="3.1.1" class="anchored" data-anchor-id="映射">
<span class="header-section-number">3.1.1</span> 映射</h3>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listMarts.html">listMarts</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="kable-table">
<table class="caption-top table table-sm table-striped small">
<thead><tr class="header">
<th style="text-align: left;">biomart</th>
<th style="text-align: left;">version</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ENSEMBL_MART_ENSEMBL</td>
<td style="text-align: left;">Ensembl Genes 113</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSEMBL_MART_MOUSE</td>
<td style="text-align: left;">Mouse strains 113</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSEMBL_MART_SNP</td>
<td style="text-align: left;">Ensembl Variation 113</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSEMBL_MART_FUNCGEN</td>
<td style="text-align: left;">Ensembl Regulation 113</td>
</tr>
</tbody>
</table>
</div>
</div>
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># 连接到Ensembl BioMart</span></span>
<span><span class="va">mart</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html">useMart</a></span><span class="op">(</span><span class="st">"ENSEMBL_MART_ENSEMBL"</span>, dataset <span class="op">=</span> <span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listAttributes.html">listAttributes</a></span><span class="op">(</span><span class="va">mart</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="datatables html-widget html-fill-item" id="htmlwidget-bec484340289eba7bef6" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-bec484340289eba7bef6">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417","418","419","420","421","422","423","424","425","426","427","428","429","430","431","432","433","434","435","436","437","438","439","440","441","442","443","444","445","446","447","448","449","450","451","452","453","454","455","456","457","458","459","460","461","462","463","464","465","466","467","468","469","470","471","472","473","474","475","476","477","478","479","480","481","482","483","484","485","486","487","488","489","490","491","492","493","494","495","496","497","498","499","500","501","502","503","504","505","506","507","508","509","510","511","512","513","514","515","516","517","518","519","520","521","522","523","524","525","526","527","528","529","530","531","532","533","534","535","536","537","538","539","540","541","542","543","544","545","546","547","548","549","550","551","552","553","554","555","556","557","558","559","560","561","562","563","564","565","566","567","568","569","570","571","572","573","574","575","576","577","578","579","580","581","582","583","584","585","586","587","588","589","590","591","592","593","594","595","596","597","598","599","600","601","602","603","604","605","606","607","608","609","610","611","612","613","614","615","616","617","618","619","620","621","622","623","624","625","626","627","628","629","630","631","632","633","634","635","636","637","638","639","640","641","642","643","644","645","646","647","648","649","650","651","652","653","654","655","656","657","658","659","660","661","662","663","664","665","666","667","668","669","670","671","672","673","674","675","676","677","678","679","680","681","682","683","684","685","686","687","688","689","690","691","692","693","694","695","696","697","698","699","700","701","702","703","704","705","706","707","708","709","710","711","712","713","714","715","716","717","718","719","720","721","722","723","724","725","726","727","728","729","730","731","732","733","734","735","736","737","738","739","740","741","742","743","744","745","746","747","748","749","750","751","752","753","754","755","756","757","758","759","760","761","762","763","764","765","766","767","768","769","770","771","772","773","774","775","776","777","778","779","780","781","782","783","784","785","786","787","788","789","790","791","792","793","794","795","796","797","798","799","800","801","802","803","804","805","806","807","808","809","810","811","812","813","814","815","816","817","818","819","820","821","822","823","824","825","826","827","828","829","830","831","832","833","834","835","836","837","838","839","840","841","842","843","844","845","846","847","848","849","850","851","852","853","854","855","856","857","858","859","860","861","862","863","864","865","866","867","868","869","870","871","872","873","874","875","876","877","878","879","880","881","882","883","884","885","886","887","888","889","890","891","892","893","894","895","896","897","898","899","900","901","902","903","904","905","906","907","908","909","910","911","912","913","914","915","916","917","918","919","920","921","922","923","924","925","926","927","928","929","930","931","932","933","934","935","936","937","938","939","940","941","942","943","944","945","946","947","948","949","950","951","952","953","954","955","956","957","958","959","960","961","962","963","964","965","966","967","968","969","970","971","972","973","974","975","976","977","978","979","980","981","982","983","984","985","986","987","988","989","990","991","992","993","994","995","996","997","998","999","1000","1001","1002","1003","1004","1005","1006","1007","1008","1009","1010","1011","1012","1013","1014","1015","1016","1017","1018","1019","1020","1021","1022","1023","1024","1025","1026","1027","1028","1029","1030","1031","1032","1033","1034","1035","1036","1037","1038","1039","1040","1041","1042","1043","1044","1045","1046","1047","1048","1049","1050","1051","1052","1053","1054","1055","1056","1057","1058","1059","1060","1061","1062","1063","1064","1065","1066","1067","1068","1069","1070","1071","1072","1073","1074","1075","1076","1077","1078","1079","1080","1081","1082","1083","1084","1085","1086","1087","1088","1089","1090","1091","1092","1093","1094","1095","1096","1097","1098","1099","1100","1101","1102","1103","1104","1105","1106","1107","1108","1109","1110","1111","1112","1113","1114","1115","1116","1117","1118","1119","1120","1121","1122","1123","1124","1125","1126","1127","1128","1129","1130","1131","1132","1133","1134","1135","1136","1137","1138","1139","1140","1141","1142","1143","1144","1145","1146","1147","1148","1149","1150","1151","1152","1153","1154","1155","1156","1157","1158","1159","1160","1161","1162","1163","1164","1165","1166","1167","1168","1169","1170","1171","1172","1173","1174","1175","1176","1177","1178","1179","1180","1181","1182","1183","1184","1185","1186","1187","1188","1189","1190","1191","1192","1193","1194","1195","1196","1197","1198","1199","1200","1201","1202","1203","1204","1205","1206","1207","1208","1209","1210","1211","1212","1213","1214","1215","1216","1217","1218","1219","1220","1221","1222","1223","1224","1225","1226","1227","1228","1229","1230","1231","1232","1233","1234","1235","1236","1237","1238","1239","1240","1241","1242","1243","1244","1245","1246","1247","1248","1249","1250","1251","1252","1253","1254","1255","1256","1257","1258","1259","1260","1261","1262","1263","1264","1265","1266","1267","1268","1269","1270","1271","1272","1273","1274","1275","1276","1277","1278","1279","1280","1281","1282","1283","1284","1285","1286","1287","1288","1289","1290","1291","1292","1293","1294","1295","1296","1297","1298","1299","1300","1301","1302","1303","1304","1305","1306","1307","1308","1309","1310","1311","1312","1313","1314","1315","1316","1317","1318","1319","1320","1321","1322","1323","1324","1325","1326","1327","1328","1329","1330","1331","1332","1333","1334","1335","1336","1337","1338","1339","1340","1341","1342","1343","1344","1345","1346","1347","1348","1349","1350","1351","1352","1353","1354","1355","1356","1357","1358","1359","1360","1361","1362","1363","1364","1365","1366","1367","1368","1369","1370","1371","1372","1373","1374","1375","1376","1377","1378","1379","1380","1381","1382","1383","1384","1385","1386","1387","1388","1389","1390","1391","1392","1393","1394","1395","1396","1397","1398","1399","1400","1401","1402","1403","1404","1405","1406","1407","1408","1409","1410","1411","1412","1413","1414","1415","1416","1417","1418","1419","1420","1421","1422","1423","1424","1425","1426","1427","1428","1429","1430","1431","1432","1433","1434","1435","1436","1437","1438","1439","1440","1441","1442","1443","1444","1445","1446","1447","1448","1449","1450","1451","1452","1453","1454","1455","1456","1457","1458","1459","1460","1461","1462","1463","1464","1465","1466","1467","1468","1469","1470","1471","1472","1473","1474","1475","1476","1477","1478","1479","1480","1481","1482","1483","1484","1485","1486","1487","1488","1489","1490","1491","1492","1493","1494","1495","1496","1497","1498","1499","1500","1501","1502","1503","1504","1505","1506","1507","1508","1509","1510","1511","1512","1513","1514","1515","1516","1517","1518","1519","1520","1521","1522","1523","1524","1525","1526","1527","1528","1529","1530","1531","1532","1533","1534","1535","1536","1537","1538","1539","1540","1541","1542","1543","1544","1545","1546","1547","1548","1549","1550","1551","1552","1553","1554","1555","1556","1557","1558","1559","1560","1561","1562","1563","1564","1565","1566","1567","1568","1569","1570","1571","1572","1573","1574","1575","1576","1577","1578","1579","1580","1581","1582","1583","1584","1585","1586","1587","1588","1589","1590","1591","1592","1593","1594","1595","1596","1597","1598","1599","1600","1601","1602","1603","1604","1605","1606","1607","1608","1609","1610","1611","1612","1613","1614","1615","1616","1617","1618","1619","1620","1621","1622","1623","1624","1625","1626","1627","1628","1629","1630","1631","1632","1633","1634","1635","1636","1637","1638","1639","1640","1641","1642","1643","1644","1645","1646","1647","1648","1649","1650","1651","1652","1653","1654","1655","1656","1657","1658","1659","1660","1661","1662","1663","1664","1665","1666","1667","1668","1669","1670","1671","1672","1673","1674","1675","1676","1677","1678","1679","1680","1681","1682","1683","1684","1685","1686","1687","1688","1689","1690","1691","1692","1693","1694","1695","1696","1697","1698","1699","1700","1701","1702","1703","1704","1705","1706","1707","1708","1709","1710","1711","1712","1713","1714","1715","1716","1717","1718","1719","1720","1721","1722","1723","1724","1725","1726","1727","1728","1729","1730","1731","1732","1733","1734","1735","1736","1737","1738","1739","1740","1741","1742","1743","1744","1745","1746","1747","1748","1749","1750","1751","1752","1753","1754","1755","1756","1757","1758","1759","1760","1761","1762","1763","1764","1765","1766","1767","1768","1769","1770","1771","1772","1773","1774","1775","1776","1777","1778","1779","1780","1781","1782","1783","1784","1785","1786","1787","1788","1789","1790","1791","1792","1793","1794","1795","1796","1797","1798","1799","1800","1801","1802","1803","1804","1805","1806","1807","1808","1809","1810","1811","1812","1813","1814","1815","1816","1817","1818","1819","1820","1821","1822","1823","1824","1825","1826","1827","1828","1829","1830","1831","1832","1833","1834","1835","1836","1837","1838","1839","1840","1841","1842","1843","1844","1845","1846","1847","1848","1849","1850","1851","1852","1853","1854","1855","1856","1857","1858","1859","1860","1861","1862","1863","1864","1865","1866","1867","1868","1869","1870","1871","1872","1873","1874","1875","1876","1877","1878","1879","1880","1881","1882","1883","1884","1885","1886","1887","1888","1889","1890","1891","1892","1893","1894","1895","1896","1897","1898","1899","1900","1901","1902","1903","1904","1905","1906","1907","1908","1909","1910","1911","1912","1913","1914","1915","1916","1917","1918","1919","1920","1921","1922","1923","1924","1925","1926","1927","1928","1929","1930","1931","1932","1933","1934","1935","1936","1937","1938","1939","1940","1941","1942","1943","1944","1945","1946","1947","1948","1949","1950","1951","1952","1953","1954","1955","1956","1957","1958","1959","1960","1961","1962","1963","1964","1965","1966","1967","1968","1969","1970","1971","1972","1973","1974","1975","1976","1977","1978","1979","1980","1981","1982","1983","1984","1985","1986","1987","1988","1989","1990","1991","1992","1993","1994","1995","1996","1997","1998","1999","2000","2001","2002","2003","2004","2005","2006","2007","2008","2009","2010","2011","2012","2013","2014","2015","2016","2017","2018","2019","2020","2021","2022","2023","2024","2025","2026","2027","2028","2029","2030","2031","2032","2033","2034","2035","2036","2037","2038","2039","2040","2041","2042","2043","2044","2045","2046","2047","2048","2049","2050","2051","2052","2053","2054","2055","2056","2057","2058","2059","2060","2061","2062","2063","2064","2065","2066","2067","2068","2069","2070","2071","2072","2073","2074","2075","2076","2077","2078","2079","2080","2081","2082","2083","2084","2085","2086","2087","2088","2089","2090","2091","2092","2093","2094","2095","2096","2097","2098","2099","2100","2101","2102","2103","2104","2105","2106","2107","2108","2109"],["ensembl_gene_id","ensembl_gene_id_version","ensembl_transcript_id","ensembl_transcript_id_version","ensembl_peptide_id","ensembl_peptide_id_version","ensembl_exon_id","description","chromosome_name","start_position","end_position","strand","band","transcript_start","transcript_end","transcription_start_site","transcript_length","transcript_tsl","transcript_gencode_basic","transcript_primary_basic","transcript_appris","transcript_is_canonical","transcript_mane_select","transcript_mane_plus_clinical","external_gene_name","external_gene_source","external_transcript_name","external_transcript_source_name","transcript_count","percentage_gene_gc_content","gene_biotype","transcript_biotype","source","transcript_source","version","transcript_version","peptide_version","external_synonym","phenotype_description","source_name","study_external_id","strain_name","strain_gender","p_value","go_id","name_1006","definition_1006","go_linkage_type","namespace_1003","goslim_goa_accession","goslim_goa_description","biogrid","ccds","chembl","dbass3_id","dbass3_name","dbass5_id","dbass5_name","entrezgene_trans_name","embl","arrayexpress","genecards","hgnc_symbol","hgnc_id","hpa_id","hpa_accession","protein_id","ens_lrg_gene","ens_lrg_transcript","merops","mim_gene_description","mim_gene_accession","mim_morbid_description","mim_morbid_accession","mirbase_id","mirbase_accession","mirbase_trans_name","entrezgene_accession","entrezgene_description","entrezgene_id","pdb","reactome","reactome_gene","reactome_transcript","refseq_mrna","refseq_mrna_predicted","refseq_ncrna","refseq_ncrna_predicted","refseq_peptide","refseq_peptide_predicted","rfam","rfam_trans_name","rnacentral","hgnc_trans_name","ucsc","uniparc","uniprot_gn_id","uniprot_gn_symbol","uniprot_isoform","uniprotswissprot","uniprotsptrembl","wikigene_name","wikigene_id","wikigene_description","affy_hc_g110","affy_hg_focus","affy_hg_u133a_2","affy_hg_u133b","affy_hg_u133_plus_2","affy_hg_u95a","affy_hg_u95av2","affy_hg_u95b","affy_hg_u95c","affy_hg_u95d","affy_hg_u95e","affy_hta_2_0","affy_ht_hg_u133_plus_pm","affy_huex_1_0_st_v2","affy_hugenefl","affy_hugene_1_0_st_v1","affy_hugene_2_0_st_v1","affy_hugene_2_1_st_v1","affy_primeview","affy_u133_x3p","agilent_cgh_44b","agilent_gpl19072","agilent_gpl26966","agilent_gpl6848","agilent_sureprint_g3_ge_8x60k","agilent_sureprint_g3_ge_8x60k_v2","agilent_wholegenome","agilent_wholegenome_4x44k_v1","agilent_wholegenome_4x44k_v2","codelink_codelink","illumina_humanref_8_v3","illumina_humanwg_6_v3","phalanx_onearray","cdd","cdd_start","cdd_end","gene3d","gene3d_start","gene3d_end","hamap","hamap_start","hamap_end","ncbifam","ncbifam_start","ncbifam_end","hmmpanther","hmmpanther_start","hmmpanther_end","pfam","pfam_start","pfam_end","pirsf","pirsf_start","pirsf_end","prints","prints_start","prints_end","scanprosite","scanprosite_start","scanprosite_end","pfscan","pfscan_start","pfscan_end","sfld","sfld_start","sfld_end","smart","smart_start","smart_end","superfamily","superfamily_start","superfamily_end","tigrfam","tigrfam_start","tigrfam_end","interpro","interpro_short_description","interpro_description","interpro_start","interpro_end","alphafold","alphafold_start","alphafold_end","mobidblite","mobidblite_start","mobidblite_end","ncoils","ncoils_start","ncoils_end","seg","seg_start","seg_end","sifts_import","sifts_import_start","sifts_import_end","signalp","signalp_start","signalp_end","tmhmm","tmhmm_start","tmhmm_end","ensembl_gene_id","ensembl_gene_id_version","version","ensembl_transcript_id","ensembl_transcript_id_version","transcript_version","ensembl_peptide_id","ensembl_peptide_id_version","peptide_version","chromosome_name","start_position","end_position","transcript_start","transcript_end","transcription_start_site","transcript_length","strand","external_gene_name","external_gene_source","5_utr_start","5_utr_end","3_utr_start","3_utr_end","cds_length","transcript_count","description","gene_biotype","exon_chrom_start","exon_chrom_end","is_constitutive","rank","phase","end_phase","cdna_coding_start","cdna_coding_end","genomic_coding_start","genomic_coding_end","ensembl_exon_id","cds_start","cds_end","ensembl_gene_id","ensembl_gene_id_version","version","ensembl_transcript_id","ensembl_transcript_id_version","transcript_version","ensembl_peptide_id","ensembl_peptide_id_version","peptide_version","chromosome_name","start_position","end_position","strand","band","external_gene_name","external_gene_source","transcript_count","percentage_gene_gc_content","description","cabingdonii_homolog_ensembl_gene","cabingdonii_homolog_associated_gene_name","cabingdonii_homolog_ensembl_peptide","cabingdonii_homolog_chromosome","cabingdonii_homolog_chrom_start","cabingdonii_homolog_chrom_end","cabingdonii_homolog_canonical_transcript_protein","cabingdonii_homolog_subtype","cabingdonii_homolog_orthology_type","cabingdonii_homolog_perc_id","cabingdonii_homolog_perc_id_r1","cabingdonii_homolog_goc_score","cabingdonii_homolog_wga_coverage","cabingdonii_homolog_orthology_confidence","mspretus_homolog_ensembl_gene","mspretus_homolog_associated_gene_name","mspretus_homolog_ensembl_peptide","mspretus_homolog_chromosome","mspretus_homolog_chrom_start","mspretus_homolog_chrom_end","mspretus_homolog_canonical_transcript_protein","mspretus_homolog_subtype","mspretus_homolog_orthology_type","mspretus_homolog_perc_id","mspretus_homolog_perc_id_r1","mspretus_homolog_goc_score","mspretus_homolog_wga_coverage","mspretus_homolog_orthology_confidence","mmmarmota_homolog_ensembl_gene","mmmarmota_homolog_associated_gene_name","mmmarmota_homolog_ensembl_peptide","mmmarmota_homolog_chromosome","mmmarmota_homolog_chrom_start","mmmarmota_homolog_chrom_end","mmmarmota_homolog_canonical_transcript_protein","mmmarmota_homolog_subtype","mmmarmota_homolog_orthology_type","mmmarmota_homolog_perc_id","mmmarmota_homolog_perc_id_r1","mmmarmota_homolog_goc_score","mmmarmota_homolog_wga_coverage","mmmarmota_homolog_orthology_confidence","bbbison_homolog_ensembl_gene","bbbison_homolog_associated_gene_name","bbbison_homolog_ensembl_peptide","bbbison_homolog_chromosome","bbbison_homolog_chrom_start","bbbison_homolog_chrom_end","bbbison_homolog_canonical_transcript_protein","bbbison_homolog_subtype","bbbison_homolog_orthology_type","bbbison_homolog_perc_id","bbbison_homolog_perc_id_r1","bbbison_homolog_goc_score","bbbison_homolog_wga_coverage","bbbison_homolog_orthology_confidence","nvison_homolog_ensembl_gene","nvison_homolog_associated_gene_name","nvison_homolog_ensembl_peptide","nvison_homolog_chromosome","nvison_homolog_chrom_start","nvison_homolog_chrom_end","nvison_homolog_canonical_transcript_protein","nvison_homolog_subtype","nvison_homolog_orthology_type","nvison_homolog_perc_id","nvison_homolog_perc_id_r1","nvison_homolog_goc_score","nvison_homolog_wga_coverage","nvison_homolog_orthology_confidence","cdromedarius_homolog_ensembl_gene","cdromedarius_homolog_associated_gene_name","cdromedarius_homolog_ensembl_peptide","cdromedarius_homolog_chromosome","cdromedarius_homolog_chrom_start","cdromedarius_homolog_chrom_end","cdromedarius_homolog_canonical_transcript_protein","cdromedarius_homolog_subtype","cdromedarius_homolog_orthology_type","cdromedarius_homolog_perc_id","cdromedarius_homolog_perc_id_r1","cdromedarius_homolog_goc_score","cdromedarius_homolog_wga_coverage","cdromedarius_homolog_orthology_confidence","dnovemcinctus_homolog_ensembl_gene","dnovemcinctus_homolog_associated_gene_name","dnovemcinctus_homolog_ensembl_peptide","dnovemcinctus_homolog_chromosome","dnovemcinctus_homolog_chrom_start","dnovemcinctus_homolog_chrom_end","dnovemcinctus_homolog_canonical_transcript_protein","dnovemcinctus_homolog_subtype","dnovemcinctus_homolog_orthology_type","dnovemcinctus_homolog_perc_id","dnovemcinctus_homolog_perc_id_r1","dnovemcinctus_homolog_goc_score","dnovemcinctus_homolog_wga_coverage","dnovemcinctus_homolog_orthology_confidence","gmorhua_homolog_ensembl_gene","gmorhua_homolog_associated_gene_name","gmorhua_homolog_ensembl_peptide","gmorhua_homolog_chromosome","gmorhua_homolog_chrom_start","gmorhua_homolog_chrom_end","gmorhua_homolog_canonical_transcript_protein","gmorhua_homolog_subtype","gmorhua_homolog_orthology_type","gmorhua_homolog_perc_id","gmorhua_homolog_perc_id_r1","gmorhua_homolog_goc_score","gmorhua_homolog_orthology_confidence","charengus_homolog_ensembl_gene","charengus_homolog_associated_gene_name","charengus_homolog_ensembl_peptide","charengus_homolog_chromosome","charengus_homolog_chrom_start","charengus_homolog_chrom_end","charengus_homolog_canonical_transcript_protein","charengus_homolog_subtype","charengus_homolog_orthology_type","charengus_homolog_perc_id","charengus_homolog_perc_id_r1","charengus_homolog_goc_score","charengus_homolog_orthology_confidence","cporosus_homolog_ensembl_gene","cporosus_homolog_associated_gene_name","cporosus_homolog_ensembl_peptide","cporosus_homolog_chromosome","cporosus_homolog_chrom_start","cporosus_homolog_chrom_end","cporosus_homolog_canonical_transcript_protein","cporosus_homolog_subtype","cporosus_homolog_orthology_type","cporosus_homolog_perc_id","cporosus_homolog_perc_id_r1","cporosus_homolog_goc_score","cporosus_homolog_wga_coverage","cporosus_homolog_orthology_confidence","lbergylta_homolog_ensembl_gene","lbergylta_homolog_associated_gene_name","lbergylta_homolog_ensembl_peptide","lbergylta_homolog_chromosome","lbergylta_homolog_chrom_start","lbergylta_homolog_chrom_end","lbergylta_homolog_canonical_transcript_protein","lbergylta_homolog_subtype","lbergylta_homolog_orthology_type","lbergylta_homolog_perc_id","lbergylta_homolog_perc_id_r1","lbergylta_homolog_goc_score","lbergylta_homolog_wga_coverage","lbergylta_homolog_orthology_confidence","lcalcarifer_homolog_ensembl_gene","lcalcarifer_homolog_associated_gene_name","lcalcarifer_homolog_ensembl_peptide","lcalcarifer_homolog_chromosome","lcalcarifer_homolog_chrom_start","lcalcarifer_homolog_chrom_end","lcalcarifer_homolog_canonical_transcript_protein","lcalcarifer_homolog_subtype","lcalcarifer_homolog_orthology_type","lcalcarifer_homolog_perc_id","lcalcarifer_homolog_perc_id_r1","lcalcarifer_homolog_goc_score","lcalcarifer_homolog_wga_coverage","lcalcarifer_homolog_orthology_confidence","dleucas_homolog_ensembl_gene","dleucas_homolog_associated_gene_name","dleucas_homolog_ensembl_peptide","dleucas_homolog_chromosome","dleucas_homolog_chrom_start","dleucas_homolog_chrom_end","dleucas_homolog_canonical_transcript_protein","dleucas_homolog_subtype","dleucas_homolog_orthology_type","dleucas_homolog_perc_id","dleucas_homolog_perc_id_r1","dleucas_homolog_goc_score","dleucas_homolog_wga_coverage","dleucas_homolog_orthology_confidence","bmusculus_homolog_ensembl_gene","bmusculus_homolog_associated_gene_name","bmusculus_homolog_ensembl_peptide","bmusculus_homolog_chromosome","bmusculus_homolog_chrom_start","bmusculus_homolog_chrom_end","bmusculus_homolog_canonical_transcript_protein","bmusculus_homolog_subtype","bmusculus_homolog_orthology_type","bmusculus_homolog_perc_id","bmusculus_homolog_perc_id_r1","bmusculus_homolog_goc_score","bmusculus_homolog_wga_coverage","bmusculus_homolog_orthology_confidence","llaticaudata_homolog_ensembl_gene","llaticaudata_homolog_associated_gene_name","llaticaudata_homolog_ensembl_peptide","llaticaudata_homolog_chromosome","llaticaudata_homolog_chrom_start","llaticaudata_homolog_chrom_end","llaticaudata_homolog_canonical_transcript_protein","llaticaudata_homolog_subtype","llaticaudata_homolog_orthology_type","llaticaudata_homolog_perc_id","llaticaudata_homolog_perc_id_r1","llaticaudata_homolog_goc_score","llaticaudata_homolog_wga_coverage","llaticaudata_homolog_orthology_confidence","hburtoni_homolog_ensembl_gene","hburtoni_homolog_associated_gene_name","hburtoni_homolog_ensembl_peptide","hburtoni_homolog_chromosome","hburtoni_homolog_chrom_start","hburtoni_homolog_chrom_end","hburtoni_homolog_canonical_transcript_protein","hburtoni_homolog_subtype","hburtoni_homolog_orthology_type","hburtoni_homolog_perc_id","hburtoni_homolog_perc_id_r1","hburtoni_homolog_goc_score","hburtoni_homolog_wga_coverage","hburtoni_homolog_orthology_confidence","cintestinalis_homolog_ensembl_gene","cintestinalis_homolog_associated_gene_name","cintestinalis_homolog_ensembl_peptide","cintestinalis_homolog_chromosome","cintestinalis_homolog_chrom_start","cintestinalis_homolog_chrom_end","cintestinalis_homolog_canonical_transcript_protein","cintestinalis_homolog_subtype","cintestinalis_homolog_orthology_type","cintestinalis_homolog_perc_id","cintestinalis_homolog_perc_id_r1","cintestinalis_homolog_wga_coverage","cintestinalis_homolog_orthology_confidence","csavignyi_homolog_ensembl_gene","csavignyi_homolog_associated_gene_name","csavignyi_homolog_ensembl_peptide","csavignyi_homolog_chromosome","csavignyi_homolog_chrom_start","csavignyi_homolog_chrom_end","csavignyi_homolog_canonical_transcript_protein","csavignyi_homolog_subtype","csavignyi_homolog_orthology_type","csavignyi_homolog_perc_id","csavignyi_homolog_perc_id_r1","csavignyi_homolog_wga_coverage","csavignyi_homolog_orthology_confidence","celegans_homolog_ensembl_gene","celegans_homolog_associated_gene_name","celegans_homolog_ensembl_peptide","celegans_homolog_chromosome","celegans_homolog_chrom_start","celegans_homolog_chrom_end","celegans_homolog_canonical_transcript_protein","celegans_homolog_subtype","celegans_homolog_orthology_type","celegans_homolog_perc_id","celegans_homolog_perc_id_r1","celegans_homolog_orthology_confidence","fcatus_homolog_ensembl_gene","fcatus_homolog_associated_gene_name","fcatus_homolog_ensembl_peptide","fcatus_homolog_chromosome","fcatus_homolog_chrom_start","fcatus_homolog_chrom_end","fcatus_homolog_canonical_transcript_protein","fcatus_homolog_subtype","fcatus_homolog_orthology_type","fcatus_homolog_perc_id","fcatus_homolog_perc_id_r1","fcatus_homolog_goc_score","fcatus_homolog_wga_coverage","fcatus_homolog_orthology_confidence","cwagneri_homolog_ensembl_gene","cwagneri_homolog_associated_gene_name","cwagneri_homolog_ensembl_peptide","cwagneri_homolog_chromosome","cwagneri_homolog_chrom_start","cwagneri_homolog_chrom_end","cwagneri_homolog_canonical_transcript_protein","cwagneri_homolog_subtype","cwagneri_homolog_orthology_type","cwagneri_homolog_perc_id","cwagneri_homolog_perc_id_r1","cwagneri_homolog_goc_score","cwagneri_homolog_wga_coverage","cwagneri_homolog_orthology_confidence","cgobio_homolog_ensembl_gene","cgobio_homolog_associated_gene_name","cgobio_homolog_ensembl_peptide","cgobio_homolog_chromosome","cgobio_homolog_chrom_start","cgobio_homolog_chrom_end","cgobio_homolog_canonical_transcript_protein","cgobio_homolog_subtype","cgobio_homolog_orthology_type","cgobio_homolog_perc_id","cgobio_homolog_perc_id_r1","cgobio_homolog_goc_score","cgobio_homolog_wga_coverage","cgobio_homolog_orthology_confidence","ipunctatus_homolog_ensembl_gene","ipunctatus_homolog_associated_gene_name","ipunctatus_homolog_ensembl_peptide","ipunctatus_homolog_chromosome","ipunctatus_homolog_chrom_start","ipunctatus_homolog_chrom_end","ipunctatus_homolog_canonical_transcript_protein","ipunctatus_homolog_subtype","ipunctatus_homolog_orthology_type","ipunctatus_homolog_perc_id","ipunctatus_homolog_perc_id_r1","ipunctatus_homolog_goc_score","ipunctatus_homolog_orthology_confidence","ggallus_homolog_ensembl_gene","ggallus_homolog_associated_gene_name","ggallus_homolog_ensembl_peptide","ggallus_homolog_chromosome","ggallus_homolog_chrom_start","ggallus_homolog_chrom_end","ggallus_homolog_canonical_transcript_protein","ggallus_homolog_subtype","ggallus_homolog_orthology_type","ggallus_homolog_perc_id","ggallus_homolog_perc_id_r1","ggallus_homolog_goc_score","ggallus_homolog_wga_coverage","ggallus_homolog_orthology_confidence","cgchok1gshd_homolog_ensembl_gene","cgchok1gshd_homolog_associated_gene_name","cgchok1gshd_homolog_ensembl_peptide","cgchok1gshd_homolog_chromosome","cgchok1gshd_homolog_chrom_start","cgchok1gshd_homolog_chrom_end","cgchok1gshd_homolog_canonical_transcript_protein","cgchok1gshd_homolog_subtype","cgchok1gshd_homolog_orthology_type","cgchok1gshd_homolog_perc_id","cgchok1gshd_homolog_perc_id_r1","cgchok1gshd_homolog_goc_score","cgchok1gshd_homolog_wga_coverage","cgchok1gshd_homolog_orthology_confidence","atestudineus_homolog_ensembl_gene","atestudineus_homolog_associated_gene_name","atestudineus_homolog_ensembl_peptide","atestudineus_homolog_chromosome","atestudineus_homolog_chrom_start","atestudineus_homolog_chrom_end","atestudineus_homolog_canonical_transcript_protein","atestudineus_homolog_subtype","atestudineus_homolog_orthology_type","atestudineus_homolog_perc_id","atestudineus_homolog_perc_id_r1","atestudineus_homolog_goc_score","atestudineus_homolog_orthology_confidence","aocellaris_homolog_ensembl_gene","aocellaris_homolog_associated_gene_name","aocellaris_homolog_ensembl_peptide","aocellaris_homolog_chromosome","aocellaris_homolog_chrom_start","aocellaris_homolog_chrom_end","aocellaris_homolog_canonical_transcript_protein","aocellaris_homolog_subtype","aocellaris_homolog_orthology_type","aocellaris_homolog_perc_id","aocellaris_homolog_perc_id_r1","aocellaris_homolog_goc_score","aocellaris_homolog_orthology_confidence","lchalumnae_homolog_ensembl_gene","lchalumnae_homolog_associated_gene_name","lchalumnae_homolog_ensembl_peptide","lchalumnae_homolog_chromosome","lchalumnae_homolog_chrom_start","lchalumnae_homolog_chrom_end","lchalumnae_homolog_canonical_transcript_protein","lchalumnae_homolog_subtype","lchalumnae_homolog_orthology_type","lchalumnae_homolog_perc_id","lchalumnae_homolog_perc_id_r1","lchalumnae_homolog_goc_score","lchalumnae_homolog_wga_coverage","lchalumnae_homolog_orthology_confidence","okisutch_homolog_ensembl_gene","okisutch_homolog_associated_gene_name","okisutch_homolog_ensembl_peptide","okisutch_homolog_chromosome","okisutch_homolog_chrom_start","okisutch_homolog_chrom_end","okisutch_homolog_canonical_transcript_protein","okisutch_homolog_subtype","okisutch_homolog_orthology_type","okisutch_homolog_perc_id","okisutch_homolog_perc_id_r1","okisutch_homolog_goc_score","okisutch_homolog_wga_coverage","okisutch_homolog_orthology_confidence","falbicollis_homolog_ensembl_gene","falbicollis_homolog_associated_gene_name","falbicollis_homolog_ensembl_peptide","falbicollis_homolog_chromosome","falbicollis_homolog_chrom_start","falbicollis_homolog_chrom_end","falbicollis_homolog_canonical_transcript_protein","falbicollis_homolog_subtype","falbicollis_homolog_orthology_type","falbicollis_homolog_perc_id","falbicollis_homolog_perc_id_r1","falbicollis_homolog_goc_score","falbicollis_homolog_orthology_confidence","cccarpio_homolog_ensembl_gene","cccarpio_homolog_associated_gene_name","cccarpio_homolog_ensembl_peptide","cccarpio_homolog_chromosome","cccarpio_homolog_chrom_start","cccarpio_homolog_chrom_end","cccarpio_homolog_canonical_transcript_protein","cccarpio_homolog_subtype","cccarpio_homolog_orthology_type","cccarpio_homolog_perc_id","cccarpio_homolog_perc_id_r1","cccarpio_homolog_goc_score","cccarpio_homolog_orthology_confidence","btaurus_homolog_ensembl_gene","btaurus_homolog_associated_gene_name","btaurus_homolog_ensembl_peptide","btaurus_homolog_chromosome","btaurus_homolog_chrom_start","btaurus_homolog_chrom_end","btaurus_homolog_canonical_transcript_protein","btaurus_homolog_subtype","btaurus_homolog_orthology_type","btaurus_homolog_perc_id","btaurus_homolog_perc_id_r1","btaurus_homolog_goc_score","btaurus_homolog_wga_coverage","btaurus_homolog_orthology_confidence","mfascicularis_homolog_ensembl_gene","mfascicularis_homolog_associated_gene_name","mfascicularis_homolog_ensembl_peptide","mfascicularis_homolog_chromosome","mfascicularis_homolog_chrom_start","mfascicularis_homolog_chrom_end","mfascicularis_homolog_canonical_transcript_protein","mfascicularis_homolog_subtype","mfascicularis_homolog_orthology_type","mfascicularis_homolog_perc_id","mfascicularis_homolog_perc_id_r1","mfascicularis_homolog_goc_score","mfascicularis_homolog_wga_coverage","mfascicularis_homolog_orthology_confidence","odegus_homolog_ensembl_gene","odegus_homolog_associated_gene_name","odegus_homolog_ensembl_peptide","odegus_homolog_chromosome","odegus_homolog_chrom_start","odegus_homolog_chrom_end","odegus_homolog_canonical_transcript_protein","odegus_homolog_subtype","odegus_homolog_orthology_type","odegus_homolog_perc_id","odegus_homolog_perc_id_r1","odegus_homolog_goc_score","odegus_homolog_wga_coverage","odegus_homolog_orthology_confidence","dclupeoides_homolog_ensembl_gene","dclupeoides_homolog_associated_gene_name","dclupeoides_homolog_ensembl_peptide","dclupeoides_homolog_chromosome","dclupeoides_homolog_chrom_start","dclupeoides_homolog_chrom_end","dclupeoides_homolog_canonical_transcript_protein","dclupeoides_homolog_subtype","dclupeoides_homolog_orthology_type","dclupeoides_homolog_perc_id","dclupeoides_homolog_perc_id_r1","dclupeoides_homolog_goc_score","dclupeoides_homolog_orthology_confidence","cldingo_homolog_ensembl_gene","cldingo_homolog_associated_gene_name","cldingo_homolog_ensembl_peptide","cldingo_homolog_chromosome","cldingo_homolog_chrom_start","cldingo_homolog_chrom_end","cldingo_homolog_canonical_transcript_protein","cldingo_homolog_subtype","cldingo_homolog_orthology_type","cldingo_homolog_perc_id","cldingo_homolog_perc_id_r1","cldingo_homolog_goc_score","cldingo_homolog_wga_coverage","cldingo_homolog_orthology_confidence","clfamiliaris_homolog_ensembl_gene","clfamiliaris_homolog_associated_gene_name","clfamiliaris_homolog_ensembl_peptide","clfamiliaris_homolog_chromosome","clfamiliaris_homolog_chrom_start","clfamiliaris_homolog_chrom_end","clfamiliaris_homolog_canonical_transcript_protein","clfamiliaris_homolog_subtype","clfamiliaris_homolog_orthology_type","clfamiliaris_homolog_perc_id","clfamiliaris_homolog_perc_id_r1","clfamiliaris_homolog_goc_score","clfamiliaris_homolog_wga_coverage","clfamiliaris_homolog_orthology_confidence","bgrunniens_homolog_ensembl_gene","bgrunniens_homolog_associated_gene_name","bgrunniens_homolog_ensembl_peptide","bgrunniens_homolog_chromosome","bgrunniens_homolog_chrom_start","bgrunniens_homolog_chrom_end","bgrunniens_homolog_canonical_transcript_protein","bgrunniens_homolog_subtype","bgrunniens_homolog_orthology_type","bgrunniens_homolog_perc_id","bgrunniens_homolog_perc_id_r1","bgrunniens_homolog_goc_score","bgrunniens_homolog_wga_coverage","bgrunniens_homolog_orthology_confidence","easinus_homolog_ensembl_gene","easinus_homolog_associated_gene_name","easinus_homolog_ensembl_peptide","easinus_homolog_chromosome","easinus_homolog_chrom_start","easinus_homolog_chrom_end","easinus_homolog_canonical_transcript_protein","easinus_homolog_subtype","easinus_homolog_orthology_type","easinus_homolog_perc_id","easinus_homolog_perc_id_r1","easinus_homolog_goc_score","easinus_homolog_orthology_confidence","mleucophaeus_homolog_ensembl_gene","mleucophaeus_homolog_associated_gene_name","mleucophaeus_homolog_ensembl_peptide","mleucophaeus_homolog_chromosome","mleucophaeus_homolog_chrom_start","mleucophaeus_homolog_chrom_end","mleucophaeus_homolog_canonical_transcript_protein","mleucophaeus_homolog_subtype","mleucophaeus_homolog_orthology_type","mleucophaeus_homolog_perc_id","mleucophaeus_homolog_perc_id_r1","mleucophaeus_homolog_goc_score","mleucophaeus_homolog_wga_coverage","mleucophaeus_homolog_orthology_confidence","dmelanogaster_homolog_ensembl_gene","dmelanogaster_homolog_associated_gene_name","dmelanogaster_homolog_ensembl_peptide","dmelanogaster_homolog_chromosome","dmelanogaster_homolog_chrom_start","dmelanogaster_homolog_chrom_end","dmelanogaster_homolog_canonical_transcript_protein","dmelanogaster_homolog_subtype","dmelanogaster_homolog_orthology_type","dmelanogaster_homolog_perc_id","dmelanogaster_homolog_perc_id_r1","dmelanogaster_homolog_orthology_confidence","applatyrhynchos_homolog_ensembl_gene","applatyrhynchos_homolog_associated_gene_name","applatyrhynchos_homolog_ensembl_peptide","applatyrhynchos_homolog_chromosome","applatyrhynchos_homolog_chrom_start","applatyrhynchos_homolog_chrom_end","applatyrhynchos_homolog_canonical_transcript_protein","applatyrhynchos_homolog_subtype","applatyrhynchos_homolog_orthology_type","applatyrhynchos_homolog_perc_id","applatyrhynchos_homolog_perc_id_r1","applatyrhynchos_homolog_goc_score","applatyrhynchos_homolog_wga_coverage","applatyrhynchos_homolog_orthology_confidence","acalliptera_homolog_ensembl_gene","acalliptera_homolog_associated_gene_name","acalliptera_homolog_ensembl_peptide","acalliptera_homolog_chromosome","acalliptera_homolog_chrom_start","acalliptera_homolog_chrom_end","acalliptera_homolog_canonical_transcript_protein","acalliptera_homolog_subtype","acalliptera_homolog_orthology_type","acalliptera_homolog_perc_id","acalliptera_homolog_perc_id_r1","acalliptera_homolog_goc_score","acalliptera_homolog_orthology_confidence","eelectricus_homolog_ensembl_gene","eelectricus_homolog_associated_gene_name","eelectricus_homolog_ensembl_peptide","eelectricus_homolog_chromosome","eelectricus_homolog_chrom_start","eelectricus_homolog_chrom_end","eelectricus_homolog_canonical_transcript_protein","eelectricus_homolog_subtype","eelectricus_homolog_orthology_type","eelectricus_homolog_perc_id","eelectricus_homolog_perc_id_r1","eelectricus_homolog_goc_score","eelectricus_homolog_orthology_confidence","lafricana_homolog_ensembl_gene","lafricana_homolog_associated_gene_name","lafricana_homolog_ensembl_peptide","lafricana_homolog_chromosome","lafricana_homolog_chrom_start","lafricana_homolog_chrom_end","lafricana_homolog_canonical_transcript_protein","lafricana_homolog_subtype","lafricana_homolog_orthology_type","lafricana_homolog_perc_id","lafricana_homolog_perc_id_r1","lafricana_homolog_goc_score","lafricana_homolog_wga_coverage","lafricana_homolog_orthology_confidence","cmilii_homolog_ensembl_gene","cmilii_homolog_associated_gene_name","cmilii_homolog_ensembl_peptide","cmilii_homolog_chromosome","cmilii_homolog_chrom_start","cmilii_homolog_chrom_end","cmilii_homolog_canonical_transcript_protein","cmilii_homolog_subtype","cmilii_homolog_orthology_type","cmilii_homolog_perc_id","cmilii_homolog_perc_id_r1","cmilii_homolog_wga_coverage","cmilii_homolog_orthology_confidence","dlabrax_homolog_ensembl_gene","dlabrax_homolog_associated_gene_name","dlabrax_homolog_ensembl_peptide","dlabrax_homolog_chromosome","dlabrax_homolog_chrom_start","dlabrax_homolog_chrom_end","dlabrax_homolog_canonical_transcript_protein","dlabrax_homolog_subtype","dlabrax_homolog_orthology_type","dlabrax_homolog_perc_id","dlabrax_homolog_perc_id_r1","dlabrax_homolog_goc_score","dlabrax_homolog_orthology_confidence","mpfuro_homolog_ensembl_gene","mpfuro_homolog_associated_gene_name","mpfuro_homolog_ensembl_peptide","mpfuro_homolog_chromosome","mpfuro_homolog_chrom_start","mpfuro_homolog_chrom_end","mpfuro_homolog_canonical_transcript_protein","mpfuro_homolog_subtype","mpfuro_homolog_orthology_type","mpfuro_homolog_perc_id","mpfuro_homolog_perc_id_r1","mpfuro_homolog_goc_score","mpfuro_homolog_wga_coverage","mpfuro_homolog_orthology_confidence","amelanoleuca_homolog_ensembl_gene","amelanoleuca_homolog_associated_gene_name","amelanoleuca_homolog_ensembl_peptide","amelanoleuca_homolog_chromosome","amelanoleuca_homolog_chrom_start","amelanoleuca_homolog_chrom_end","amelanoleuca_homolog_canonical_transcript_protein","amelanoleuca_homolog_subtype","amelanoleuca_homolog_orthology_type","amelanoleuca_homolog_perc_id","amelanoleuca_homolog_perc_id_r1","amelanoleuca_homolog_goc_score","amelanoleuca_homolog_orthology_confidence","nleucogenys_homolog_ensembl_gene","nleucogenys_homolog_associated_gene_name","nleucogenys_homolog_ensembl_peptide","nleucogenys_homolog_chromosome","nleucogenys_homolog_chrom_start","nleucogenys_homolog_chrom_end","nleucogenys_homolog_canonical_transcript_protein","nleucogenys_homolog_subtype","nleucogenys_homolog_orthology_type","nleucogenys_homolog_perc_id","nleucogenys_homolog_perc_id_r1","nleucogenys_homolog_goc_score","nleucogenys_homolog_wga_coverage","nleucogenys_homolog_orthology_confidence","chircus_homolog_ensembl_gene","chircus_homolog_associated_gene_name","chircus_homolog_ensembl_peptide","chircus_homolog_chromosome","chircus_homolog_chrom_start","chircus_homolog_chrom_end","chircus_homolog_canonical_transcript_protein","chircus_homolog_subtype","chircus_homolog_orthology_type","chircus_homolog_perc_id","chircus_homolog_perc_id_r1","chircus_homolog_goc_score","chircus_homolog_wga_coverage","chircus_homolog_orthology_confidence","mauratus_homolog_ensembl_gene","mauratus_homolog_associated_gene_name","mauratus_homolog_ensembl_peptide","mauratus_homolog_chromosome","mauratus_homolog_chrom_start","mauratus_homolog_chrom_end","mauratus_homolog_canonical_transcript_protein","mauratus_homolog_subtype","mauratus_homolog_orthology_type","mauratus_homolog_perc_id","mauratus_homolog_perc_id_r1","mauratus_homolog_goc_score","mauratus_homolog_wga_coverage","mauratus_homolog_orthology_confidence","acchrysaetos_homolog_ensembl_gene","acchrysaetos_homolog_associated_gene_name","acchrysaetos_homolog_ensembl_peptide","acchrysaetos_homolog_chromosome","acchrysaetos_homolog_chrom_start","acchrysaetos_homolog_chrom_end","acchrysaetos_homolog_canonical_transcript_protein","acchrysaetos_homolog_subtype","acchrysaetos_homolog_orthology_type","acchrysaetos_homolog_perc_id","acchrysaetos_homolog_perc_id_r1","acchrysaetos_homolog_goc_score","acchrysaetos_homolog_wga_coverage","acchrysaetos_homolog_orthology_confidence","cauratus_homolog_ensembl_gene","cauratus_homolog_associated_gene_name","cauratus_homolog_ensembl_peptide","cauratus_homolog_chromosome","cauratus_homolog_chrom_start","cauratus_homolog_chrom_end","cauratus_homolog_canonical_transcript_protein","cauratus_homolog_subtype","cauratus_homolog_orthology_type","cauratus_homolog_perc_id","cauratus_homolog_perc_id_r1","cauratus_homolog_goc_score","cauratus_homolog_wga_coverage","cauratus_homolog_orthology_confidence","gevgoodei_homolog_ensembl_gene","gevgoodei_homolog_associated_gene_name","gevgoodei_homolog_ensembl_peptide","gevgoodei_homolog_chromosome","gevgoodei_homolog_chrom_start","gevgoodei_homolog_chrom_end","gevgoodei_homolog_canonical_transcript_protein","gevgoodei_homolog_subtype","gevgoodei_homolog_orthology_type","gevgoodei_homolog_perc_id","gevgoodei_homolog_perc_id_r1","gevgoodei_homolog_goc_score","gevgoodei_homolog_wga_coverage","gevgoodei_homolog_orthology_confidence","ggorilla_homolog_ensembl_gene","ggorilla_homolog_associated_gene_name","ggorilla_homolog_ensembl_peptide","ggorilla_homolog_chromosome","ggorilla_homolog_chrom_start","ggorilla_homolog_chrom_end","ggorilla_homolog_canonical_transcript_protein","ggorilla_homolog_subtype","ggorilla_homolog_orthology_type","ggorilla_homolog_perc_id","ggorilla_homolog_perc_id_r1","ggorilla_homolog_goc_score","ggorilla_homolog_wga_coverage","ggorilla_homolog_orthology_confidence","acarolinensis_homolog_ensembl_gene","acarolinensis_homolog_associated_gene_name","acarolinensis_homolog_ensembl_peptide","acarolinensis_homolog_chromosome","acarolinensis_homolog_chrom_start","acarolinensis_homolog_chrom_end","acarolinensis_homolog_canonical_transcript_protein","acarolinensis_homolog_subtype","acarolinensis_homolog_orthology_type","acarolinensis_homolog_perc_id","acarolinensis_homolog_perc_id_r1","acarolinensis_homolog_goc_score","acarolinensis_homolog_orthology_confidence","cporcellus_homolog_ensembl_gene","cporcellus_homolog_associated_gene_name","cporcellus_homolog_ensembl_peptide","cporcellus_homolog_chromosome","cporcellus_homolog_chrom_start","cporcellus_homolog_chrom_end","cporcellus_homolog_canonical_transcript_protein","cporcellus_homolog_subtype","cporcellus_homolog_orthology_type","cporcellus_homolog_perc_id","cporcellus_homolog_perc_id_r1","cporcellus_homolog_goc_score","cporcellus_homolog_wga_coverage","cporcellus_homolog_orthology_confidence","eburgeri_homolog_ensembl_gene","eburgeri_homolog_associated_gene_name","eburgeri_homolog_ensembl_peptide","eburgeri_homolog_chromosome","eburgeri_homolog_chrom_start","eburgeri_homolog_chrom_end","eburgeri_homolog_canonical_transcript_protein","eburgeri_homolog_subtype","eburgeri_homolog_orthology_type","eburgeri_homolog_perc_id","eburgeri_homolog_perc_id_r1","eburgeri_homolog_wga_coverage","eburgeri_homolog_orthology_confidence","eeuropaeus_homolog_ensembl_gene","eeuropaeus_homolog_associated_gene_name","eeuropaeus_homolog_ensembl_peptide","eeuropaeus_homolog_chromosome","eeuropaeus_homolog_chrom_start","eeuropaeus_homolog_chrom_end","eeuropaeus_homolog_canonical_transcript_protein","eeuropaeus_homolog_subtype","eeuropaeus_homolog_orthology_type","eeuropaeus_homolog_perc_id","eeuropaeus_homolog_perc_id_r1","eeuropaeus_homolog_goc_score","eeuropaeus_homolog_wga_coverage","eeuropaeus_homolog_orthology_confidence","ecaballus_homolog_ensembl_gene","ecaballus_homolog_associated_gene_name","ecaballus_homolog_ensembl_peptide","ecaballus_homolog_chromosome","ecaballus_homolog_chrom_start","ecaballus_homolog_chrom_end","ecaballus_homolog_canonical_transcript_protein","ecaballus_homolog_subtype","ecaballus_homolog_orthology_type","ecaballus_homolog_perc_id","ecaballus_homolog_perc_id_r1","ecaballus_homolog_goc_score","ecaballus_homolog_wga_coverage","ecaballus_homolog_orthology_confidence","hhucho_homolog_ensembl_gene","hhucho_homolog_associated_gene_name","hhucho_homolog_ensembl_peptide","hhucho_homolog_chromosome","hhucho_homolog_chrom_start","hhucho_homolog_chrom_end","hhucho_homolog_canonical_transcript_protein","hhucho_homolog_subtype","hhucho_homolog_orthology_type","hhucho_homolog_perc_id","hhucho_homolog_perc_id_r1","hhucho_homolog_goc_score","hhucho_homolog_wga_coverage","hhucho_homolog_orthology_confidence","bihybrid_homolog_ensembl_gene","bihybrid_homolog_associated_gene_name","bihybrid_homolog_ensembl_peptide","bihybrid_homolog_chromosome","bihybrid_homolog_chrom_start","bihybrid_homolog_chrom_end","bihybrid_homolog_canonical_transcript_protein","bihybrid_homolog_subtype","bihybrid_homolog_orthology_type","bihybrid_homolog_perc_id","bihybrid_homolog_perc_id_r1","bihybrid_homolog_goc_score","bihybrid_homolog_wga_coverage","bihybrid_homolog_orthology_confidence","nnaja_homolog_ensembl_gene","nnaja_homolog_associated_gene_name","nnaja_homolog_ensembl_peptide","nnaja_homolog_chromosome","nnaja_homolog_chrom_start","nnaja_homolog_chrom_end","nnaja_homolog_canonical_transcript_protein","nnaja_homolog_subtype","nnaja_homolog_orthology_type","nnaja_homolog_perc_id","nnaja_homolog_perc_id_r1","nnaja_homolog_goc_score","nnaja_homolog_wga_coverage","nnaja_homolog_orthology_confidence","cjaponica_homolog_ensembl_gene","cjaponica_homolog_associated_gene_name","cjaponica_homolog_ensembl_peptide","cjaponica_homolog_chromosome","cjaponica_homolog_chrom_start","cjaponica_homolog_chrom_end","cjaponica_homolog_canonical_transcript_protein","cjaponica_homolog_subtype","cjaponica_homolog_orthology_type","cjaponica_homolog_perc_id","cjaponica_homolog_perc_id_r1","cjaponica_homolog_goc_score","cjaponica_homolog_wga_coverage","cjaponica_homolog_orthology_confidence","dordii_homolog_ensembl_gene","dordii_homolog_associated_gene_name","dordii_homolog_ensembl_peptide","dordii_homolog_chromosome","dordii_homolog_chrom_start","dordii_homolog_chrom_end","dordii_homolog_canonical_transcript_protein","dordii_homolog_subtype","dordii_homolog_orthology_type","dordii_homolog_perc_id","dordii_homolog_perc_id_r1","dordii_homolog_goc_score","dordii_homolog_wga_coverage","dordii_homolog_orthology_confidence","lcrocea_homolog_ensembl_gene","lcrocea_homolog_associated_gene_name","lcrocea_homolog_ensembl_peptide","lcrocea_homolog_chromosome","lcrocea_homolog_chrom_start","lcrocea_homolog_chrom_end","lcrocea_homolog_canonical_transcript_protein","lcrocea_homolog_subtype","lcrocea_homolog_orthology_type","lcrocea_homolog_perc_id","lcrocea_homolog_perc_id_r1","lcrocea_homolog_goc_score","lcrocea_homolog_wga_coverage","lcrocea_homolog_orthology_confidence","lleishanense_homolog_ensembl_gene","lleishanense_homolog_associated_gene_name","lleishanense_homolog_ensembl_peptide","lleishanense_homolog_chromosome","lleishanense_homolog_chrom_start","lleishanense_homolog_chrom_end","lleishanense_homolog_canonical_transcript_protein","lleishanense_homolog_subtype","lleishanense_homolog_orthology_type","lleishanense_homolog_perc_id","lleishanense_homolog_perc_id_r1","lleishanense_homolog_goc_score","lleishanense_homolog_wga_coverage","lleishanense_homolog_orthology_confidence","jjaculus_homolog_ensembl_gene","jjaculus_homolog_associated_gene_name","jjaculus_homolog_ensembl_peptide","jjaculus_homolog_chromosome","jjaculus_homolog_chrom_start","jjaculus_homolog_chrom_end","jjaculus_homolog_canonical_transcript_protein","jjaculus_homolog_subtype","jjaculus_homolog_orthology_type","jjaculus_homolog_perc_id","jjaculus_homolog_perc_id_r1","jjaculus_homolog_goc_score","jjaculus_homolog_wga_coverage","jjaculus_homolog_orthology_confidence","etelfairi_homolog_ensembl_gene","etelfairi_homolog_associated_gene_name","etelfairi_homolog_ensembl_peptide","etelfairi_homolog_chromosome","etelfairi_homolog_chrom_start","etelfairi_homolog_chrom_end","etelfairi_homolog_canonical_transcript_protein","etelfairi_homolog_subtype","etelfairi_homolog_orthology_type","etelfairi_homolog_perc_id","etelfairi_homolog_perc_id_r1","etelfairi_homolog_goc_score","etelfairi_homolog_wga_coverage","etelfairi_homolog_orthology_confidence","clanigera_homolog_ensembl_gene","clanigera_homolog_associated_gene_name","clanigera_homolog_ensembl_peptide","clanigera_homolog_chromosome","clanigera_homolog_chrom_start","clanigera_homolog_chrom_end","clanigera_homolog_canonical_transcript_protein","clanigera_homolog_subtype","clanigera_homolog_orthology_type","clanigera_homolog_perc_id","clanigera_homolog_perc_id_r1","clanigera_homolog_goc_score","clanigera_homolog_wga_coverage","clanigera_homolog_orthology_confidence","clumpus_homolog_ensembl_gene","clumpus_homolog_associated_gene_name","clumpus_homolog_ensembl_peptide","clumpus_homolog_chromosome","clumpus_homolog_chrom_start","clumpus_homolog_chrom_end","clumpus_homolog_canonical_transcript_protein","clumpus_homolog_subtype","clumpus_homolog_orthology_type","clumpus_homolog_perc_id","clumpus_homolog_perc_id_r1","clumpus_homolog_goc_score","clumpus_homolog_wga_coverage","clumpus_homolog_orthology_confidence","nbrichardi_homolog_ensembl_gene","nbrichardi_homolog_associated_gene_name","nbrichardi_homolog_ensembl_peptide","nbrichardi_homolog_chromosome","nbrichardi_homolog_chrom_start","nbrichardi_homolog_chrom_end","nbrichardi_homolog_canonical_transcript_protein","nbrichardi_homolog_subtype","nbrichardi_homolog_orthology_type","nbrichardi_homolog_perc_id","nbrichardi_homolog_perc_id_r1","nbrichardi_homolog_goc_score","nbrichardi_homolog_wga_coverage","nbrichardi_homolog_orthology_confidence","anancymaae_homolog_ensembl_gene","anancymaae_homolog_associated_gene_name","anancymaae_homolog_ensembl_peptide","anancymaae_homolog_chromosome","anancymaae_homolog_chrom_start","anancymaae_homolog_chrom_end","anancymaae_homolog_canonical_transcript_protein","anancymaae_homolog_subtype","anancymaae_homolog_orthology_type","anancymaae_homolog_perc_id","anancymaae_homolog_perc_id_r1","anancymaae_homolog_goc_score","anancymaae_homolog_wga_coverage","anancymaae_homolog_orthology_confidence","mmulatta_homolog_ensembl_gene","mmulatta_homolog_associated_gene_name","mmulatta_homolog_ensembl_peptide","mmulatta_homolog_chromosome","mmulatta_homolog_chrom_start","mmulatta_homolog_chrom_end","mmulatta_homolog_canonical_transcript_protein","mmulatta_homolog_subtype","mmulatta_homolog_orthology_type","mmulatta_homolog_perc_id","mmulatta_homolog_perc_id_r1","mmulatta_homolog_goc_score","mmulatta_homolog_wga_coverage","mmulatta_homolog_orthology_confidence","nscutatus_homolog_ensembl_gene","nscutatus_homolog_associated_gene_name","nscutatus_homolog_ensembl_peptide","nscutatus_homolog_chromosome","nscutatus_homolog_chrom_start","nscutatus_homolog_chrom_end","nscutatus_homolog_canonical_transcript_protein","nscutatus_homolog_subtype","nscutatus_homolog_orthology_type","nscutatus_homolog_perc_id","nscutatus_homolog_perc_id_r1","nscutatus_homolog_goc_score","nscutatus_homolog_wga_coverage","nscutatus_homolog_orthology_confidence","kmarmoratus_homolog_ensembl_gene","kmarmoratus_homolog_associated_gene_name","kmarmoratus_homolog_ensembl_peptide","kmarmoratus_homolog_chromosome","kmarmoratus_homolog_chrom_start","kmarmoratus_homolog_chrom_end","kmarmoratus_homolog_canonical_transcript_protein","kmarmoratus_homolog_subtype","kmarmoratus_homolog_orthology_type","kmarmoratus_homolog_perc_id","kmarmoratus_homolog_perc_id_r1","kmarmoratus_homolog_goc_score","kmarmoratus_homolog_wga_coverage","kmarmoratus_homolog_orthology_confidence","gfortis_homolog_ensembl_gene","gfortis_homolog_associated_gene_name","gfortis_homolog_ensembl_peptide","gfortis_homolog_chromosome","gfortis_homolog_chrom_start","gfortis_homolog_chrom_end","gfortis_homolog_canonical_transcript_protein","gfortis_homolog_subtype","gfortis_homolog_orthology_type","gfortis_homolog_perc_id","gfortis_homolog_perc_id_r1","gfortis_homolog_goc_score","gfortis_homolog_wga_coverage","gfortis_homolog_orthology_confidence","amexicanus_homolog_ensembl_gene","amexicanus_homolog_associated_gene_name","amexicanus_homolog_ensembl_peptide","amexicanus_homolog_chromosome","amexicanus_homolog_chrom_start","amexicanus_homolog_chrom_end","amexicanus_homolog_canonical_transcript_protein","amexicanus_homolog_subtype","amexicanus_homolog_orthology_type","amexicanus_homolog_perc_id","amexicanus_homolog_perc_id_r1","amexicanus_homolog_goc_score","amexicanus_homolog_wga_coverage","amexicanus_homolog_orthology_confidence","mlucifugus_homolog_ensembl_gene","mlucifugus_homolog_associated_gene_name","mlucifugus_homolog_ensembl_peptide","mlucifugus_homolog_chromosome","mlucifugus_homolog_chrom_start","mlucifugus_homolog_chrom_end","mlucifugus_homolog_canonical_transcript_protein","mlucifugus_homolog_subtype","mlucifugus_homolog_orthology_type","mlucifugus_homolog_perc_id","mlucifugus_homolog_perc_id_r1","mlucifugus_homolog_goc_score","mlucifugus_homolog_wga_coverage","mlucifugus_homolog_orthology_confidence","acitrinellus_homolog_ensembl_gene","acitrinellus_homolog_associated_gene_name","acitrinellus_homolog_ensembl_peptide","acitrinellus_homolog_chromosome","acitrinellus_homolog_chrom_start","acitrinellus_homolog_chrom_end","acitrinellus_homolog_canonical_transcript_protein","acitrinellus_homolog_subtype","acitrinellus_homolog_orthology_type","acitrinellus_homolog_perc_id","acitrinellus_homolog_perc_id_r1","acitrinellus_homolog_goc_score","acitrinellus_homolog_wga_coverage","acitrinellus_homolog_orthology_confidence","mmusculus_homolog_ensembl_gene","mmusculus_homolog_associated_gene_name","mmusculus_homolog_ensembl_peptide","mmusculus_homolog_chromosome","mmusculus_homolog_chrom_start","mmusculus_homolog_chrom_end","mmusculus_homolog_canonical_transcript_protein","mmusculus_homolog_subtype","mmusculus_homolog_orthology_type","mmusculus_homolog_perc_id","mmusculus_homolog_perc_id_r1","mmusculus_homolog_goc_score","mmusculus_homolog_wga_coverage","mmusculus_homolog_orthology_confidence","mmurinus_homolog_ensembl_gene","mmurinus_homolog_associated_gene_name","mmurinus_homolog_ensembl_peptide","mmurinus_homolog_chromosome","mmurinus_homolog_chrom_start","mmurinus_homolog_chrom_end","mmurinus_homolog_canonical_transcript_protein","mmurinus_homolog_subtype","mmurinus_homolog_orthology_type","mmurinus_homolog_perc_id","mmurinus_homolog_perc_id_r1","mmurinus_homolog_goc_score","mmurinus_homolog_wga_coverage","mmurinus_homolog_orthology_confidence","fheteroclitus_homolog_ensembl_gene","fheteroclitus_homolog_associated_gene_name","fheteroclitus_homolog_ensembl_peptide","fheteroclitus_homolog_chromosome","fheteroclitus_homolog_chrom_start","fheteroclitus_homolog_chrom_end","fheteroclitus_homolog_canonical_transcript_protein","fheteroclitus_homolog_subtype","fheteroclitus_homolog_orthology_type","fheteroclitus_homolog_perc_id","fheteroclitus_homolog_perc_id_r1","fheteroclitus_homolog_goc_score","fheteroclitus_homolog_wga_coverage","fheteroclitus_homolog_orthology_confidence","hgfemale_homolog_ensembl_gene","hgfemale_homolog_associated_gene_name","hgfemale_homolog_ensembl_peptide","hgfemale_homolog_chromosome","hgfemale_homolog_chrom_start","hgfemale_homolog_chrom_end","hgfemale_homolog_canonical_transcript_protein","hgfemale_homolog_subtype","hgfemale_homolog_orthology_type","hgfemale_homolog_perc_id","hgfemale_homolog_perc_id_r1","hgfemale_homolog_goc_score","hgfemale_homolog_orthology_confidence","mmonoceros_homolog_ensembl_gene","mmonoceros_homolog_associated_gene_name","mmonoceros_homolog_ensembl_peptide","mmonoceros_homolog_chromosome","mmonoceros_homolog_chrom_start","mmonoceros_homolog_chrom_end","mmonoceros_homolog_canonical_transcript_protein","mmonoceros_homolog_subtype","mmonoceros_homolog_orthology_type","mmonoceros_homolog_perc_id","mmonoceros_homolog_perc_id_r1","mmonoceros_homolog_goc_score","mmonoceros_homolog_wga_coverage","mmonoceros_homolog_orthology_confidence","elucius_homolog_ensembl_gene","elucius_homolog_associated_gene_name","elucius_homolog_ensembl_peptide","elucius_homolog_chromosome","elucius_homolog_chrom_start","elucius_homolog_chrom_end","elucius_homolog_canonical_transcript_protein","elucius_homolog_subtype","elucius_homolog_orthology_type","elucius_homolog_perc_id","elucius_homolog_perc_id_r1","elucius_homolog_goc_score","elucius_homolog_orthology_confidence","mdomestica_homolog_ensembl_gene","mdomestica_homolog_associated_gene_name","mdomestica_homolog_ensembl_peptide","mdomestica_homolog_chromosome","mdomestica_homolog_chrom_start","mdomestica_homolog_chrom_end","mdomestica_homolog_canonical_transcript_protein","mdomestica_homolog_subtype","mdomestica_homolog_orthology_type","mdomestica_homolog_perc_id","mdomestica_homolog_perc_id_r1","mdomestica_homolog_goc_score","mdomestica_homolog_wga_coverage","mdomestica_homolog_orthology_confidence","apercula_homolog_ensembl_gene","apercula_homolog_associated_gene_name","apercula_homolog_ensembl_peptide","apercula_homolog_chromosome","apercula_homolog_chrom_start","apercula_homolog_chrom_end","apercula_homolog_canonical_transcript_protein","apercula_homolog_subtype","apercula_homolog_orthology_type","apercula_homolog_perc_id","apercula_homolog_perc_id_r1","apercula_homolog_goc_score","apercula_homolog_wga_coverage","apercula_homolog_orthology_confidence","cpbellii_homolog_ensembl_gene","cpbellii_homolog_associated_gene_name","cpbellii_homolog_ensembl_peptide","cpbellii_homolog_chromosome","cpbellii_homolog_chrom_start","cpbellii_homolog_chrom_end","cpbellii_homolog_canonical_transcript_protein","cpbellii_homolog_subtype","cpbellii_homolog_orthology_type","cpbellii_homolog_perc_id","cpbellii_homolog_perc_id_r1","cpbellii_homolog_goc_score","cpbellii_homolog_wga_coverage","cpbellii_homolog_orthology_confidence","cimitator_homolog_ensembl_gene","cimitator_homolog_associated_gene_name","cimitator_homolog_ensembl_peptide","cimitator_homolog_chromosome","cimitator_homolog_chrom_start","cimitator_homolog_chrom_end","cimitator_homolog_canonical_transcript_protein","cimitator_homolog_subtype","cimitator_homolog_orthology_type","cimitator_homolog_perc_id","cimitator_homolog_perc_id_r1","cimitator_homolog_goc_score","cimitator_homolog_wga_coverage","cimitator_homolog_orthology_confidence","mnemestrina_homolog_ensembl_gene","mnemestrina_homolog_associated_gene_name","mnemestrina_homolog_ensembl_peptide","mnemestrina_homolog_chromosome","mnemestrina_homolog_chrom_start","mnemestrina_homolog_chrom_end","mnemestrina_homolog_canonical_transcript_protein","mnemestrina_homolog_subtype","mnemestrina_homolog_orthology_type","mnemestrina_homolog_perc_id","mnemestrina_homolog_perc_id_r1","mnemestrina_homolog_goc_score","mnemestrina_homolog_wga_coverage","mnemestrina_homolog_orthology_confidence","oprinceps_homolog_ensembl_gene","oprinceps_homolog_associated_gene_name","oprinceps_homolog_ensembl_peptide","oprinceps_homolog_chromosome","oprinceps_homolog_chrom_start","oprinceps_homolog_chrom_end","oprinceps_homolog_canonical_transcript_protein","oprinceps_homolog_subtype","oprinceps_homolog_orthology_type","oprinceps_homolog_perc_id","oprinceps_homolog_perc_id_r1","oprinceps_homolog_goc_score","oprinceps_homolog_wga_coverage","oprinceps_homolog_orthology_confidence","mmurdjan_homolog_ensembl_gene","mmurdjan_homolog_associated_gene_name","mmurdjan_homolog_ensembl_peptide","mmurdjan_homolog_chromosome","mmurdjan_homolog_chrom_start","mmurdjan_homolog_chrom_end","mmurdjan_homolog_canonical_transcript_protein","mmurdjan_homolog_subtype","mmurdjan_homolog_orthology_type","mmurdjan_homolog_perc_id","mmurdjan_homolog_perc_id_r1","mmurdjan_homolog_goc_score","mmurdjan_homolog_wga_coverage","mmurdjan_homolog_orthology_confidence","abrachyrhynchus_homolog_ensembl_gene","abrachyrhynchus_homolog_associated_gene_name","abrachyrhynchus_homolog_ensembl_peptide","abrachyrhynchus_homolog_chromosome","abrachyrhynchus_homolog_chrom_start","abrachyrhynchus_homolog_chrom_end","abrachyrhynchus_homolog_canonical_transcript_protein","abrachyrhynchus_homolog_subtype","abrachyrhynchus_homolog_orthology_type","abrachyrhynchus_homolog_perc_id","abrachyrhynchus_homolog_perc_id_r1","abrachyrhynchus_homolog_goc_score","abrachyrhynchus_homolog_wga_coverage","abrachyrhynchus_homolog_orthology_confidence","mochrogaster_homolog_ensembl_gene","mochrogaster_homolog_associated_gene_name","mochrogaster_homolog_ensembl_peptide","mochrogaster_homolog_chromosome","mochrogaster_homolog_chrom_start","mochrogaster_homolog_chrom_end","mochrogaster_homolog_canonical_transcript_protein","mochrogaster_homolog_subtype","mochrogaster_homolog_orthology_type","mochrogaster_homolog_perc_id","mochrogaster_homolog_perc_id_r1","mochrogaster_homolog_goc_score","mochrogaster_homolog_wga_coverage","mochrogaster_homolog_orthology_confidence","ecalabaricus_homolog_ensembl_gene","ecalabaricus_homolog_associated_gene_name","ecalabaricus_homolog_ensembl_peptide","ecalabaricus_homolog_chromosome","ecalabaricus_homolog_chrom_start","ecalabaricus_homolog_chrom_end","ecalabaricus_homolog_canonical_transcript_protein","ecalabaricus_homolog_subtype","ecalabaricus_homolog_orthology_type","ecalabaricus_homolog_perc_id","ecalabaricus_homolog_perc_id_r1","ecalabaricus_homolog_goc_score","ecalabaricus_homolog_wga_coverage","ecalabaricus_homolog_orthology_confidence","mcaroli_homolog_ensembl_gene","mcaroli_homolog_associated_gene_name","mcaroli_homolog_ensembl_peptide","mcaroli_homolog_chromosome","mcaroli_homolog_chrom_start","mcaroli_homolog_chrom_end","mcaroli_homolog_canonical_transcript_protein","mcaroli_homolog_subtype","mcaroli_homolog_orthology_type","mcaroli_homolog_perc_id","mcaroli_homolog_perc_id_r1","mcaroli_homolog_goc_score","mcaroli_homolog_wga_coverage","mcaroli_homolog_orthology_confidence","cvariegatus_homolog_ensembl_gene","cvariegatus_homolog_associated_gene_name","cvariegatus_homolog_ensembl_peptide","cvariegatus_homolog_chromosome","cvariegatus_homolog_chrom_start","cvariegatus_homolog_chrom_end","cvariegatus_homolog_canonical_transcript_protein","cvariegatus_homolog_subtype","cvariegatus_homolog_orthology_type","cvariegatus_homolog_perc_id","cvariegatus_homolog_perc_id_r1","cvariegatus_homolog_goc_score","cvariegatus_homolog_wga_coverage","cvariegatus_homolog_orthology_confidence","mpahari_homolog_ensembl_gene","mpahari_homolog_associated_gene_name","mpahari_homolog_ensembl_peptide","mpahari_homolog_chromosome","mpahari_homolog_chrom_start","mpahari_homolog_chrom_end","mpahari_homolog_canonical_transcript_protein","mpahari_homolog_subtype","mpahari_homolog_orthology_type","mpahari_homolog_perc_id","mpahari_homolog_perc_id_r1","mpahari_homolog_goc_score","mpahari_homolog_wga_coverage","mpahari_homolog_orthology_confidence","bsplendens_homolog_ensembl_gene","bsplendens_homolog_associated_gene_name","bsplendens_homolog_ensembl_peptide","bsplendens_homolog_chromosome","bsplendens_homolog_chrom_start","bsplendens_homolog_chrom_end","bsplendens_homolog_canonical_transcript_protein","bsplendens_homolog_subtype","bsplendens_homolog_orthology_type","bsplendens_homolog_perc_id","bsplendens_homolog_perc_id_r1","bsplendens_homolog_goc_score","bsplendens_homolog_wga_coverage","bsplendens_homolog_orthology_confidence","mmoschiferus_homolog_ensembl_gene","mmoschiferus_homolog_associated_gene_name","mmoschiferus_homolog_ensembl_peptide","mmoschiferus_homolog_chromosome","mmoschiferus_homolog_chrom_start","mmoschiferus_homolog_chrom_end","mmoschiferus_homolog_canonical_transcript_protein","mmoschiferus_homolog_subtype","mmoschiferus_homolog_orthology_type","mmoschiferus_homolog_perc_id","mmoschiferus_homolog_perc_id_r1","mmoschiferus_homolog_goc_score","mmoschiferus_homolog_wga_coverage","mmoschiferus_homolog_orthology_confidence","choffmanni_homolog_ensembl_gene","choffmanni_homolog_associated_gene_name","choffmanni_homolog_ensembl_peptide","choffmanni_homolog_chromosome","choffmanni_homolog_chrom_start","choffmanni_homolog_chrom_end","choffmanni_homolog_canonical_transcript_protein","choffmanni_homolog_subtype","choffmanni_homolog_orthology_type","choffmanni_homolog_perc_id","choffmanni_homolog_perc_id_r1","choffmanni_homolog_goc_score","choffmanni_homolog_wga_coverage","choffmanni_homolog_orthology_confidence","catys_homolog_ensembl_gene","catys_homolog_associated_gene_name","catys_homolog_ensembl_peptide","catys_homolog_chromosome","catys_homolog_chrom_start","catys_homolog_chrom_end","catys_homolog_canonical_transcript_protein","catys_homolog_subtype","catys_homolog_orthology_type","catys_homolog_perc_id","catys_homolog_perc_id_r1","catys_homolog_goc_score","catys_homolog_wga_coverage","catys_homolog_orthology_confidence","apolyacanthus_homolog_ensembl_gene","apolyacanthus_homolog_associated_gene_name","apolyacanthus_homolog_ensembl_peptide","apolyacanthus_homolog_chromosome","apolyacanthus_homolog_chrom_start","apolyacanthus_homolog_chrom_end","apolyacanthus_homolog_canonical_transcript_protein","apolyacanthus_homolog_subtype","apolyacanthus_homolog_orthology_type","apolyacanthus_homolog_perc_id","apolyacanthus_homolog_perc_id_r1","apolyacanthus_homolog_goc_score","apolyacanthus_homolog_wga_coverage","apolyacanthus_homolog_orthology_confidence","loculatus_homolog_ensembl_gene","loculatus_homolog_associated_gene_name","loculatus_homolog_ensembl_peptide","loculatus_homolog_chromosome","loculatus_homolog_chrom_start","loculatus_homolog_chrom_end","loculatus_homolog_canonical_transcript_protein","loculatus_homolog_subtype","loculatus_homolog_orthology_type","loculatus_homolog_perc_id","loculatus_homolog_perc_id_r1","loculatus_homolog_goc_score","loculatus_homolog_wga_coverage","loculatus_homolog_orthology_confidence","itridecemlineatus_homolog_ensembl_gene","itridecemlineatus_homolog_associated_gene_name","itridecemlineatus_homolog_ensembl_peptide","itridecemlineatus_homolog_chromosome","itridecemlineatus_homolog_chrom_start","itridecemlineatus_homolog_chrom_end","itridecemlineatus_homolog_canonical_transcript_protein","itridecemlineatus_homolog_subtype","itridecemlineatus_homolog_orthology_type","itridecemlineatus_homolog_perc_id","itridecemlineatus_homolog_perc_id_r1","itridecemlineatus_homolog_goc_score","itridecemlineatus_homolog_wga_coverage","itridecemlineatus_homolog_orthology_confidence","mspicilegus_homolog_ensembl_gene","mspicilegus_homolog_associated_gene_name","mspicilegus_homolog_ensembl_peptide","mspicilegus_homolog_chromosome","mspicilegus_homolog_chrom_start","mspicilegus_homolog_chrom_end","mspicilegus_homolog_canonical_transcript_protein","mspicilegus_homolog_subtype","mspicilegus_homolog_orthology_type","mspicilegus_homolog_perc_id","mspicilegus_homolog_perc_id_r1","mspicilegus_homolog_goc_score","mspicilegus_homolog_wga_coverage","mspicilegus_homolog_orthology_confidence","gaculeatus_homolog_ensembl_gene","gaculeatus_homolog_associated_gene_name","gaculeatus_homolog_ensembl_peptide","gaculeatus_homolog_chromosome","gaculeatus_homolog_chrom_start","gaculeatus_homolog_chrom_end","gaculeatus_homolog_canonical_transcript_protein","gaculeatus_homolog_subtype","gaculeatus_homolog_orthology_type","gaculeatus_homolog_perc_id","gaculeatus_homolog_perc_id_r1","gaculeatus_homolog_goc_score","gaculeatus_homolog_orthology_confidence","csyrichta_homolog_ensembl_gene","csyrichta_homolog_associated_gene_name","csyrichta_homolog_ensembl_peptide","csyrichta_homolog_chromosome","csyrichta_homolog_chrom_start","csyrichta_homolog_chrom_end","csyrichta_homolog_canonical_transcript_protein","csyrichta_homolog_subtype","csyrichta_homolog_orthology_type","csyrichta_homolog_perc_id","csyrichta_homolog_perc_id_r1","csyrichta_homolog_goc_score","csyrichta_homolog_wga_coverage","csyrichta_homolog_orthology_confidence","hcomes_homolog_ensembl_gene","hcomes_homolog_associated_gene_name","hcomes_homolog_ensembl_peptide","hcomes_homolog_chromosome","hcomes_homolog_chrom_start","hcomes_homolog_chrom_end","hcomes_homolog_canonical_transcript_protein","hcomes_homolog_subtype","hcomes_homolog_orthology_type","hcomes_homolog_perc_id","hcomes_homolog_perc_id_r1","hcomes_homolog_goc_score","hcomes_homolog_wga_coverage","hcomes_homolog_orthology_confidence","csemilaevis_homolog_ensembl_gene","csemilaevis_homolog_associated_gene_name","csemilaevis_homolog_ensembl_peptide","csemilaevis_homolog_chromosome","csemilaevis_homolog_chrom_start","csemilaevis_homolog_chrom_end","csemilaevis_homolog_canonical_transcript_protein","csemilaevis_homolog_subtype","csemilaevis_homolog_orthology_type","csemilaevis_homolog_perc_id","csemilaevis_homolog_perc_id_r1","csemilaevis_homolog_goc_score","csemilaevis_homolog_wga_coverage","csemilaevis_homolog_orthology_confidence","mgallopavo_homolog_ensembl_gene","mgallopavo_homolog_associated_gene_name","mgallopavo_homolog_ensembl_peptide","mgallopavo_homolog_chromosome","mgallopavo_homolog_chrom_start","mgallopavo_homolog_chrom_end","mgallopavo_homolog_canonical_transcript_protein","mgallopavo_homolog_subtype","mgallopavo_homolog_orthology_type","mgallopavo_homolog_perc_id","mgallopavo_homolog_perc_id_r1","mgallopavo_homolog_goc_score","mgallopavo_homolog_orthology_confidence","nfurzeri_homolog_ensembl_gene","nfurzeri_homolog_associated_gene_name","nfurzeri_homolog_ensembl_peptide","nfurzeri_homolog_chromosome","nfurzeri_homolog_chrom_start","nfurzeri_homolog_chrom_end","nfurzeri_homolog_canonical_transcript_protein","nfurzeri_homolog_subtype","nfurzeri_homolog_orthology_type","nfurzeri_homolog_perc_id","nfurzeri_homolog_perc_id_r1","nfurzeri_homolog_goc_score","nfurzeri_homolog_wga_coverage","nfurzeri_homolog_orthology_confidence","ngalili_homolog_ensembl_gene","ngalili_homolog_associated_gene_name","ngalili_homolog_ensembl_peptide","ngalili_homolog_chromosome","ngalili_homolog_chrom_start","ngalili_homolog_chrom_end","ngalili_homolog_canonical_transcript_protein","ngalili_homolog_subtype","ngalili_homolog_orthology_type","ngalili_homolog_perc_id","ngalili_homolog_perc_id_r1","ngalili_homolog_goc_score","ngalili_homolog_wga_coverage","ngalili_homolog_orthology_confidence","csabaeus_homolog_ensembl_gene","csabaeus_homolog_associated_gene_name","csabaeus_homolog_ensembl_peptide","csabaeus_homolog_chromosome","csabaeus_homolog_chrom_start","csabaeus_homolog_chrom_end","csabaeus_homolog_canonical_transcript_protein","csabaeus_homolog_subtype","csabaeus_homolog_orthology_type","csabaeus_homolog_perc_id","csabaeus_homolog_perc_id_r1","csabaeus_homolog_goc_score","csabaeus_homolog_wga_coverage","csabaeus_homolog_orthology_confidence","neugenii_homolog_ensembl_gene","neugenii_homolog_associated_gene_name","neugenii_homolog_ensembl_peptide","neugenii_homolog_chromosome","neugenii_homolog_chrom_start","neugenii_homolog_chrom_end","neugenii_homolog_canonical_transcript_protein","neugenii_homolog_subtype","neugenii_homolog_orthology_type","neugenii_homolog_perc_id","neugenii_homolog_perc_id_r1","neugenii_homolog_goc_score","neugenii_homolog_wga_coverage","neugenii_homolog_orthology_confidence","cjacchus_homolog_ensembl_gene","cjacchus_homolog_associated_gene_name","cjacchus_homolog_ensembl_peptide","cjacchus_homolog_chromosome","cjacchus_homolog_chrom_start","cjacchus_homolog_chrom_end","cjacchus_homolog_canonical_transcript_protein","cjacchus_homolog_subtype","cjacchus_homolog_orthology_type","cjacchus_homolog_perc_id","cjacchus_homolog_perc_id_r1","cjacchus_homolog_goc_score","cjacchus_homolog_wga_coverage","cjacchus_homolog_orthology_confidence","bmutus_homolog_ensembl_gene","bmutus_homolog_associated_gene_name","bmutus_homolog_ensembl_peptide","bmutus_homolog_chromosome","bmutus_homolog_chrom_start","bmutus_homolog_chrom_end","bmutus_homolog_canonical_transcript_protein","bmutus_homolog_subtype","bmutus_homolog_orthology_type","bmutus_homolog_perc_id","bmutus_homolog_perc_id_r1","bmutus_homolog_goc_score","bmutus_homolog_wga_coverage","bmutus_homolog_orthology_confidence","chyarkandensis_homolog_ensembl_gene","chyarkandensis_homolog_associated_gene_name","chyarkandensis_homolog_ensembl_peptide","chyarkandensis_homolog_chromosome","chyarkandensis_homolog_chrom_start","chyarkandensis_homolog_chrom_end","chyarkandensis_homolog_canonical_transcript_protein","chyarkandensis_homolog_subtype","chyarkandensis_homolog_orthology_type","chyarkandensis_homolog_perc_id","chyarkandensis_homolog_perc_id_r1","chyarkandensis_homolog_goc_score","chyarkandensis_homolog_wga_coverage","chyarkandensis_homolog_orthology_confidence","mzebra_homolog_ensembl_gene","mzebra_homolog_associated_gene_name","mzebra_homolog_ensembl_peptide","mzebra_homolog_chromosome","mzebra_homolog_chrom_start","mzebra_homolog_chrom_end","mzebra_homolog_canonical_transcript_protein","mzebra_homolog_subtype","mzebra_homolog_orthology_type","mzebra_homolog_perc_id","mzebra_homolog_perc_id_r1","mzebra_homolog_goc_score","mzebra_homolog_wga_coverage","mzebra_homolog_orthology_confidence","drerio_homolog_ensembl_gene","drerio_homolog_associated_gene_name","drerio_homolog_ensembl_peptide","drerio_homolog_chromosome","drerio_homolog_chrom_start","drerio_homolog_chrom_end","drerio_homolog_canonical_transcript_protein","drerio_homolog_subtype","drerio_homolog_orthology_type","drerio_homolog_perc_id","drerio_homolog_perc_id_r1","drerio_homolog_goc_score","drerio_homolog_wga_coverage","drerio_homolog_orthology_confidence","marmatus_homolog_ensembl_gene","marmatus_homolog_associated_gene_name","marmatus_homolog_ensembl_peptide","marmatus_homolog_chromosome","marmatus_homolog_chrom_start","marmatus_homolog_chrom_end","marmatus_homolog_canonical_transcript_protein","marmatus_homolog_subtype","marmatus_homolog_orthology_type","marmatus_homolog_perc_id","marmatus_homolog_perc_id_r1","marmatus_homolog_goc_score","marmatus_homolog_orthology_confidence","hsapiens_paralog_ensembl_gene","hsapiens_paralog_associated_gene_name","hsapiens_paralog_ensembl_peptide","hsapiens_paralog_chromosome","hsapiens_paralog_chrom_start","hsapiens_paralog_chrom_end","hsapiens_paralog_canonical_transcript_protein","hsapiens_paralog_subtype","hsapiens_paralog_orthology_type","hsapiens_paralog_perc_id","hsapiens_paralog_perc_id_r1","ensembl_gene_id","ensembl_gene_id_version","version","ensembl_transcript_id","ensembl_transcript_id_version","transcript_version","ensembl_peptide_id","ensembl_peptide_id_version","peptide_version","chromosome_name","start_position","end_position","strand","band","external_gene_name","external_gene_source","transcript_count","percentage_gene_gc_content","description","variation_name","germ_line_variation_source","source_description","allele","validated","mapweight","minor_allele","minor_allele_freq","minor_allele_count","clinical_significance","transcript_location","snp_chromosome_strand","peptide_location","chromosome_start","chromosome_end","polyphen_prediction_2076","polyphen_score_2076","sift_prediction_2076","sift_score_2076","distance_to_transcript_2076","cds_start_2076","cds_end_2076","peptide_shift","synonymous_status","allele_string_2076","ensembl_gene_id","ensembl_gene_id_version","version","ensembl_transcript_id","ensembl_transcript_id_version","transcript_version","ensembl_peptide_id","ensembl_peptide_id_version","peptide_version","chromosome_name","start_position","end_position","strand","band","external_gene_name","external_gene_source","transcript_count","percentage_gene_gc_content","description","somatic_variation_name","somatic_source_name","somatic_source_description","somatic_allele","somatic_validated","somatic_mapweight","somatic_transcript_location","somatic_snp_chromosome_strand","somatic_peptide_location","somatic_chromosome_start","somatic_chromosome_end","mart_transcript_variation_som__dm_distance_to_transcript_2076","somatic_cds_start_2076","somatic_cds_end_2076","somatic_synonymous_status","mart_transcript_variation_som__dm_allele_string_2076","transcript_exon_intron","gene_exon_intron","transcript_flank","gene_flank","coding_transcript_flank","coding_gene_flank","5utr","3utr","gene_exon","cdna","coding","peptide","upstream_flank","downstream_flank","ensembl_gene_id","ensembl_gene_id_version","description","external_gene_name","external_gene_source","chromosome_name","start_position","end_position","gene_biotype","version","uniparc","uniprotswissprot","uniprotsptrembl","cdna_coding_start","cdna_coding_end","5_utr_start","5_utr_end","3_utr_start","3_utr_end","ensembl_transcript_id","ensembl_transcript_id_version","ensembl_peptide_id","ensembl_peptide_id_version","transcript_biotype","transcript_version","peptide_version","strand","transcript_start","transcript_end","transcription_start_site","transcript_length","cds_length","cds_start","cds_end","ensembl_exon_id","exon_chrom_start","exon_chrom_end","strand","rank","phase","end_phase","cdna_coding_start","cdna_coding_end","genomic_coding_start","genomic_coding_end","is_constitutive"],["Gene stable ID","Gene stable ID version","Transcript stable ID","Transcript stable ID version","Protein stable ID","Protein stable ID version","Exon stable ID","Gene description","Chromosome/scaffold name","Gene start (bp)","Gene end (bp)","Strand","Karyotype band","Transcript start (bp)","Transcript end (bp)","Transcription start site (TSS)","Transcript length (including UTRs and CDS)","Transcript support level (TSL)","GENCODE basic annotation","GENCODE primary annotation","APPRIS annotation","Ensembl Canonical","RefSeq match transcript (MANE Select)","RefSeq match transcript (MANE Plus Clinical)","Gene name","Source of gene name","Transcript name","Source of transcript name","Transcript count","Gene % GC content","Gene type","Transcript type","Source (gene)","Source (transcript)","Version (gene)","Version (transcript)","Version (protein)","Gene Synonym","Phenotype description","Source name","Study external reference","Strain name","Strain gender","P value","GO term accession","GO term name","GO term definition","GO term evidence code","GO domain","GOSlim GOA Accession(s)","GOSlim GOA Description","BioGRID Interaction data, The General Repository for Interaction Datasets ID","CCDS ID","ChEMBL ID","DataBase of Aberrant 3' Splice Sites ID","DataBase of Aberrant 3' Splice Sites name","DataBase of Aberrant 5' Splice Sites ID","DataBase of Aberrant 5' Splice Sites name","EntrezGene transcript name ID","European Nucleotide Archive ID","Expression Atlas ID","GeneCards ID","HGNC symbol","HGNC ID","Human Protein Atlas ID","Human Protein Atlas accession","INSDC protein ID","LRG display in Ensembl gene ID","LRG display in Ensembl transcript ID","MEROPS - the Peptidase Database ID","MIM gene description","MIM gene accession","MIM morbid description","MIM morbid accession","miRBase ID","miRBase accession","miRBase transcript name ID","NCBI gene (formerly Entrezgene) accession","NCBI gene (formerly Entrezgene) description","NCBI gene (formerly Entrezgene) ID","PDB ID","Reactome ID","Reactome gene ID","Reactome transcript ID","RefSeq mRNA ID","RefSeq mRNA predicted ID","RefSeq ncRNA ID","RefSeq ncRNA predicted ID","RefSeq peptide ID","RefSeq peptide predicted ID","RFAM ID","RFAM transcript name ID","RNAcentral ID","Transcript name ID","UCSC Stable ID","UniParc ID","UniProtKB Gene Name ID","UniProtKB Gene Name symbol","UniProtKB isoform ID","UniProtKB/Swiss-Prot ID","UniProtKB/TrEMBL ID","WikiGene name","WikiGene ID","WikiGene description","AFFY HC G110 probe","AFFY HG Focus probe","AFFY HG U133A 2 probe","AFFY HG U133B probe","AFFY HG U133 Plus 2 probe","AFFY HG U95A probe","AFFY HG U95Av2 probe","AFFY HG U95B probe","AFFY HG U95C probe","AFFY HG U95D probe","AFFY HG U95E probe","AFFY HTA 2 0 probe","AFFY HT HG U133 Plus PM probe","AFFY HuEx 1 0 st v2 probe","AFFY HuGeneFL probe","AFFY HuGene 1 0 st v1 probe","AFFY HuGene 2 0 st v1 probe","AFFY HuGene 2 1 st v1 probe","AFFY PrimeView probe","AFFY U133 X3P probe","AGILENT CGH 44b probe","AGILENT GPL19072 probe","AGILENT GPL26966 probe","AGILENT GPL6848 probe","AGILENT SurePrint G3 GE 8x60k probe","AGILENT SurePrint G3 GE 8x60k v2 probe","AGILENT WholeGenome probe","AGILENT WholeGenome 4x44k v1 probe","AGILENT WholeGenome 4x44k v2 probe","CODELINK CODELINK probe","ILLUMINA HumanRef 8 V3 probe","ILLUMINA HumanWG 6 V3 probe","PHALANX OneArray probe","CDD ID","CDD start","CDD end","Gene3D ID","Gene3D start","Gene3D end","HAMAP ID","HAMAP start","HAMAP end","NCBIFAM ID","NCBIFAM start","NCBIFAM end","PANTHER ID","PANTHER start","PANTHER end","Pfam ID","Pfam start","Pfam end","PIRSF ID","PIRSF start","PIRSF end","Prints ID","Prints start","Prints end","PROSITE patterns ID","PROSITE patterns start","PROSITE patterns end","PROSITE profiles ID","PROSITE profiles start","PROSITE profiles end","SFLD ID","SFLD start","SFLD end","SMART ID","SMART start","SMART end","Superfamily ID","Superfamily start","Superfamily end","TIGRFAM ID","TIGRFAM start","TIGRFAM end","Interpro ID","Interpro Short Description","Interpro Description","Interpro start","Interpro end","AFDB-ENSP mappings","AFDB-ENSP mappings start","AFDB-ENSP mappings end","MobiDB lite","MobiDB lite start","MobiDB lite end","Coiled-coils (Ncoils)","Coiled-coils (Ncoils) start","Coiled-coils (Ncoils) end","Low complexity (Seg)","Low complexity (Seg) start","Low complexity (Seg) end","PDB-ENSP mappings","PDB-ENSP mappings start","PDB-ENSP mappings end","Cleavage site (Signalp)","Cleavage site (Signalp) start","Cleavage site (Signalp) end","Transmembrane helices","Transmembrane helices start","Transmembrane helices end","Gene stable ID","Gene stable ID version","Version (gene)","Transcript stable ID","Transcript stable ID version","Version (transcript)","Protein stable ID","Protein stable ID version","Version (protein)","Chromosome/scaffold name","Gene start (bp)","Gene end (bp)","Transcript start (bp)","Transcript end (bp)","Transcription start site (TSS)","Transcript length (including UTRs and CDS)","Strand","Gene name","Source of gene name","5' UTR start","5' UTR end","3' UTR start","3' UTR end","CDS Length","Transcript count","Gene description","Gene type","Exon region start (bp)","Exon region end (bp)","Constitutive exon","Exon rank in transcript","Start phase","End phase","cDNA coding start","cDNA coding end","Genomic coding start","Genomic coding end","Exon stable ID","CDS start","CDS end","Gene stable ID","Gene stable ID version","Version (gene)","Transcript stable ID","Transcript stable ID version","Version (transcript)","Protein stable ID","Protein stable ID version","Version (protein)","Chromosome/scaffold name","Gene start (bp)","Gene end (bp)","Strand","Karyotype band","Gene name","Source of gene name","Transcript count","Gene % GC content","Gene description","Abingdon island giant tortoise gene stable ID","Abingdon island giant tortoise gene name","Abingdon island giant tortoise protein or transcript stable ID","Abingdon island giant tortoise chromosome/scaffold name","Abingdon island giant tortoise chromosome/scaffold start (bp)","Abingdon island giant tortoise chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Abingdon island giant tortoise","Abingdon island giant tortoise homology type","%id. target Abingdon island giant tortoise gene identical to query gene","%id. query gene identical to target Abingdon island giant tortoise gene","Abingdon island giant tortoise Gene-order conservation score","Abingdon island giant tortoise Whole-genome alignment coverage","Abingdon island giant tortoise orthology confidence [0 low, 1 high]","Algerian mouse gene stable ID","Algerian mouse gene name","Algerian mouse protein or transcript stable ID","Algerian mouse chromosome/scaffold name","Algerian mouse chromosome/scaffold start (bp)","Algerian mouse chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Algerian mouse","Algerian mouse homology type","%id. target Algerian mouse gene identical to query gene","%id. query gene identical to target Algerian mouse gene","Algerian mouse Gene-order conservation score","Algerian mouse Whole-genome alignment coverage","Algerian mouse orthology confidence [0 low, 1 high]","Alpine marmot gene stable ID","Alpine marmot gene name","Alpine marmot protein or transcript stable ID","Alpine marmot chromosome/scaffold name","Alpine marmot chromosome/scaffold start (bp)","Alpine marmot chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Alpine marmot","Alpine marmot homology type","%id. target Alpine marmot gene identical to query gene","%id. query gene identical to target Alpine marmot gene","Alpine marmot Gene-order conservation score","Alpine marmot Whole-genome alignment coverage","Alpine marmot orthology confidence [0 low, 1 high]","American bison gene stable ID","American bison gene name","American bison protein or transcript stable ID","American bison chromosome/scaffold name","American bison chromosome/scaffold start (bp)","American bison chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with American bison","American bison homology type","%id. target American bison gene identical to query gene","%id. query gene identical to target American bison gene","American bison Gene-order conservation score","American bison Whole-genome alignment coverage","American bison orthology confidence [0 low, 1 high]","American mink gene stable ID","American mink gene name","American mink protein or transcript stable ID","American mink chromosome/scaffold name","American mink chromosome/scaffold start (bp)","American mink chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with American mink","American mink homology type","%id. target American mink gene identical to query gene","%id. query gene identical to target American mink gene","American mink Gene-order conservation score","American mink Whole-genome alignment coverage","American mink orthology confidence [0 low, 1 high]","Arabian camel gene stable ID","Arabian camel gene name","Arabian camel protein or transcript stable ID","Arabian camel chromosome/scaffold name","Arabian camel chromosome/scaffold start (bp)","Arabian camel chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Arabian camel","Arabian camel homology type","%id. target Arabian camel gene identical to query gene","%id. query gene identical to target Arabian camel gene","Arabian camel Gene-order conservation score","Arabian camel Whole-genome alignment coverage","Arabian camel orthology confidence [0 low, 1 high]","Armadillo gene stable ID","Armadillo gene name","Armadillo protein or transcript stable ID","Armadillo chromosome/scaffold name","Armadillo chromosome/scaffold start (bp)","Armadillo chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Armadillo","Armadillo homology type","%id. target Armadillo gene identical to query gene","%id. query gene identical to target Armadillo gene","Armadillo Gene-order conservation score","Armadillo Whole-genome alignment coverage","Armadillo orthology confidence [0 low, 1 high]","Atlantic cod gene stable ID","Atlantic cod gene name","Atlantic cod protein or transcript stable ID","Atlantic cod chromosome/scaffold name","Atlantic cod chromosome/scaffold start (bp)","Atlantic cod chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Atlantic cod","Atlantic cod homology type","%id. target Atlantic cod gene identical to query gene","%id. query gene identical to target Atlantic cod gene","Atlantic cod Gene-order conservation score","Atlantic cod orthology confidence [0 low, 1 high]","Atlantic herring  gene stable ID","Atlantic herring  gene name","Atlantic herring  protein or transcript stable ID","Atlantic herring  chromosome/scaffold name","Atlantic herring  chromosome/scaffold start (bp)","Atlantic herring  chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Atlantic herring ","Atlantic herring  homology type","%id. target Atlantic herring  gene identical to query gene","%id. query gene identical to target Atlantic herring  gene","Atlantic herring  Gene-order conservation score","Atlantic herring  orthology confidence [0 low, 1 high]","Australian saltwater crocodile gene stable ID","Australian saltwater crocodile gene name","Australian saltwater crocodile protein or transcript stable ID","Australian saltwater crocodile chromosome/scaffold name","Australian saltwater crocodile chromosome/scaffold start (bp)","Australian saltwater crocodile chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Australian saltwater crocodile","Australian saltwater crocodile homology type","%id. target Australian saltwater crocodile gene identical to query gene","%id. query gene identical to target Australian saltwater crocodile gene","Australian saltwater crocodile Gene-order conservation score","Australian saltwater crocodile Whole-genome alignment coverage","Australian saltwater crocodile orthology confidence [0 low, 1 high]","Ballan wrasse gene stable ID","Ballan wrasse gene name","Ballan wrasse protein or transcript stable ID","Ballan wrasse chromosome/scaffold name","Ballan wrasse chromosome/scaffold start (bp)","Ballan wrasse chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Ballan wrasse","Ballan wrasse homology type","%id. target Ballan wrasse gene identical to query gene","%id. query gene identical to target Ballan wrasse gene","Ballan wrasse Gene-order conservation score","Ballan wrasse Whole-genome alignment coverage","Ballan wrasse orthology confidence [0 low, 1 high]","Barramundi perch gene stable ID","Barramundi perch gene name","Barramundi perch protein or transcript stable ID","Barramundi perch chromosome/scaffold name","Barramundi perch chromosome/scaffold start (bp)","Barramundi perch chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Barramundi perch","Barramundi perch homology type","%id. target Barramundi perch gene identical to query gene","%id. query gene identical to target Barramundi perch gene","Barramundi perch Gene-order conservation score","Barramundi perch Whole-genome alignment coverage","Barramundi perch orthology confidence [0 low, 1 high]","Beluga whale gene stable ID","Beluga whale gene name","Beluga whale protein or transcript stable ID","Beluga whale chromosome/scaffold name","Beluga whale chromosome/scaffold start (bp)","Beluga whale chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Beluga whale","Beluga whale homology type","%id. target Beluga whale gene identical to query gene","%id. query gene identical to target Beluga whale gene","Beluga whale Gene-order conservation score","Beluga whale Whole-genome alignment coverage","Beluga whale orthology confidence [0 low, 1 high]","Blue whale gene stable ID","Blue whale gene name","Blue whale protein or transcript stable ID","Blue whale chromosome/scaffold name","Blue whale chromosome/scaffold start (bp)","Blue whale chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Blue whale","Blue whale homology type","%id. target Blue whale gene identical to query gene","%id. query gene identical to target Blue whale gene","Blue whale Gene-order conservation score","Blue whale Whole-genome alignment coverage","Blue whale orthology confidence [0 low, 1 high]","Blue-ringed sea krait gene stable ID","Blue-ringed sea krait gene name","Blue-ringed sea krait protein or transcript stable ID","Blue-ringed sea krait chromosome/scaffold name","Blue-ringed sea krait chromosome/scaffold start (bp)","Blue-ringed sea krait chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Blue-ringed sea krait","Blue-ringed sea krait homology type","%id. target Blue-ringed sea krait gene identical to query gene","%id. query gene identical to target Blue-ringed sea krait gene","Blue-ringed sea krait Gene-order conservation score","Blue-ringed sea krait Whole-genome alignment coverage","Blue-ringed sea krait orthology confidence [0 low, 1 high]","Burton's mouthbrooder gene stable ID","Burton's mouthbrooder gene name","Burton's mouthbrooder protein or transcript stable ID","Burton's mouthbrooder chromosome/scaffold name","Burton's mouthbrooder chromosome/scaffold start (bp)","Burton's mouthbrooder chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Burton's mouthbrooder","Burton's mouthbrooder homology type","%id. target Burton's mouthbrooder gene identical to query gene","%id. query gene identical to target Burton's mouthbrooder gene","Burton's mouthbrooder Gene-order conservation score","Burton's mouthbrooder Whole-genome alignment coverage","Burton's mouthbrooder orthology confidence [0 low, 1 high]","C.intestinalis gene stable ID","C.intestinalis gene name","C.intestinalis protein or transcript stable ID","C.intestinalis chromosome/scaffold name","C.intestinalis chromosome/scaffold start (bp)","C.intestinalis chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with C.intestinalis","C.intestinalis homology type","%id. target C.intestinalis gene identical to query gene","%id. query gene identical to target C.intestinalis gene","C.intestinalis Whole-genome alignment coverage","C.intestinalis orthology confidence [0 low, 1 high]","C.savignyi gene stable ID","C.savignyi gene name","C.savignyi protein or transcript stable ID","C.savignyi chromosome/scaffold name","C.savignyi chromosome/scaffold start (bp)","C.savignyi chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with C.savignyi","C.savignyi homology type","%id. target C.savignyi gene identical to query gene","%id. query gene identical to target C.savignyi gene","C.savignyi Whole-genome alignment coverage","C.savignyi orthology confidence [0 low, 1 high]","Caenorhabditis elegans (Nematode, N2) gene stable ID","Caenorhabditis elegans (Nematode, N2) gene name","Caenorhabditis elegans (Nematode, N2) protein or transcript stable ID","Caenorhabditis elegans (Nematode, N2) chromosome/scaffold name","Caenorhabditis elegans (Nematode, N2) chromosome/scaffold start (bp)","Caenorhabditis elegans (Nematode, N2) chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Caenorhabditis elegans (Nematode, N2)","Caenorhabditis elegans (Nematode, N2) homology type","%id. target Caenorhabditis elegans (Nematode, N2) gene identical to query gene","%id. query gene identical to target Caenorhabditis elegans (Nematode, N2) gene","Caenorhabditis elegans (Nematode, N2) orthology confidence [0 low, 1 high]","Cat gene stable ID","Cat gene name","Cat protein or transcript stable ID","Cat chromosome/scaffold name","Cat chromosome/scaffold start (bp)","Cat chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Cat","Cat homology type","%id. target Cat gene identical to query gene","%id. query gene identical to target Cat gene","Cat Gene-order conservation score","Cat Whole-genome alignment coverage","Cat orthology confidence [0 low, 1 high]","Chacoan peccary gene stable ID","Chacoan peccary gene name","Chacoan peccary protein or transcript stable ID","Chacoan peccary chromosome/scaffold name","Chacoan peccary chromosome/scaffold start (bp)","Chacoan peccary chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Chacoan peccary","Chacoan peccary homology type","%id. target Chacoan peccary gene identical to query gene","%id. query gene identical to target Chacoan peccary gene","Chacoan peccary Gene-order conservation score","Chacoan peccary Whole-genome alignment coverage","Chacoan peccary orthology confidence [0 low, 1 high]","Channel bull blenny gene stable ID","Channel bull blenny gene name","Channel bull blenny protein or transcript stable ID","Channel bull blenny chromosome/scaffold name","Channel bull blenny chromosome/scaffold start (bp)","Channel bull blenny chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Channel bull blenny","Channel bull blenny homology type","%id. target Channel bull blenny gene identical to query gene","%id. query gene identical to target Channel bull blenny gene","Channel bull blenny Gene-order conservation score","Channel bull blenny Whole-genome alignment coverage","Channel bull blenny orthology confidence [0 low, 1 high]","Channel catfish gene stable ID","Channel catfish gene name","Channel catfish protein or transcript stable ID","Channel catfish chromosome/scaffold name","Channel catfish chromosome/scaffold start (bp)","Channel catfish chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Channel catfish","Channel catfish homology type","%id. target Channel catfish gene identical to query gene","%id. query gene identical to target Channel catfish gene","Channel catfish Gene-order conservation score","Channel catfish orthology confidence [0 low, 1 high]","Chicken gene stable ID","Chicken gene name","Chicken protein or transcript stable ID","Chicken chromosome/scaffold name","Chicken chromosome/scaffold start (bp)","Chicken chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Chicken","Chicken homology type","%id. target Chicken gene identical to query gene","%id. query gene identical to target Chicken gene","Chicken Gene-order conservation score","Chicken Whole-genome alignment coverage","Chicken orthology confidence [0 low, 1 high]","Chinese hamster CHOK1GS gene stable ID","Chinese hamster CHOK1GS gene name","Chinese hamster CHOK1GS protein or transcript stable ID","Chinese hamster CHOK1GS chromosome/scaffold name","Chinese hamster CHOK1GS chromosome/scaffold start (bp)","Chinese hamster CHOK1GS chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Chinese hamster CHOK1GS","Chinese hamster CHOK1GS homology type","%id. target Chinese hamster CHOK1GS gene identical to query gene","%id. query gene identical to target Chinese hamster CHOK1GS gene","Chinese hamster CHOK1GS Gene-order conservation score","Chinese hamster CHOK1GS Whole-genome alignment coverage","Chinese hamster CHOK1GS orthology confidence [0 low, 1 high]","Climbing perch gene stable ID","Climbing perch gene name","Climbing perch protein or transcript stable ID","Climbing perch chromosome/scaffold name","Climbing perch chromosome/scaffold start (bp)","Climbing perch chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Climbing perch","Climbing perch homology type","%id. target Climbing perch gene identical to query gene","%id. query gene identical to target Climbing perch gene","Climbing perch Gene-order conservation score","Climbing perch orthology confidence [0 low, 1 high]","Clown anemonefish gene stable ID","Clown anemonefish gene name","Clown anemonefish protein or transcript stable ID","Clown anemonefish chromosome/scaffold name","Clown anemonefish chromosome/scaffold start (bp)","Clown anemonefish chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Clown anemonefish","Clown anemonefish homology type","%id. target Clown anemonefish gene identical to query gene","%id. query gene identical to target Clown anemonefish gene","Clown anemonefish Gene-order conservation score","Clown anemonefish orthology confidence [0 low, 1 high]","Coelacanth gene stable ID","Coelacanth gene name","Coelacanth protein or transcript stable ID","Coelacanth chromosome/scaffold name","Coelacanth chromosome/scaffold start (bp)","Coelacanth chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Coelacanth","Coelacanth homology type","%id. target Coelacanth gene identical to query gene","%id. query gene identical to target Coelacanth gene","Coelacanth Gene-order conservation score","Coelacanth Whole-genome alignment coverage","Coelacanth orthology confidence [0 low, 1 high]","Coho salmon gene stable ID","Coho salmon gene name","Coho salmon protein or transcript stable ID","Coho salmon chromosome/scaffold name","Coho salmon chromosome/scaffold start (bp)","Coho salmon chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Coho salmon","Coho salmon homology type","%id. target Coho salmon gene identical to query gene","%id. query gene identical to target Coho salmon gene","Coho salmon Gene-order conservation score","Coho salmon Whole-genome alignment coverage","Coho salmon orthology confidence [0 low, 1 high]","Collared flycatcher gene stable ID","Collared flycatcher gene name","Collared flycatcher protein or transcript stable ID","Collared flycatcher chromosome/scaffold name","Collared flycatcher chromosome/scaffold start (bp)","Collared flycatcher chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Collared flycatcher","Collared flycatcher homology type","%id. target Collared flycatcher gene identical to query gene","%id. query gene identical to target Collared flycatcher gene","Collared flycatcher Gene-order conservation score","Collared flycatcher orthology confidence [0 low, 1 high]","Common carp gene stable ID","Common carp gene name","Common carp protein or transcript stable ID","Common carp chromosome/scaffold name","Common carp chromosome/scaffold start (bp)","Common carp chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Common carp","Common carp homology type","%id. target Common carp gene identical to query gene","%id. query gene identical to target Common carp gene","Common carp Gene-order conservation score","Common carp orthology confidence [0 low, 1 high]","Cow gene stable ID","Cow gene name","Cow protein or transcript stable ID","Cow chromosome/scaffold name","Cow chromosome/scaffold start (bp)","Cow chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Cow","Cow homology type","%id. target Cow gene identical to query gene","%id. query gene identical to target Cow gene","Cow Gene-order conservation score","Cow Whole-genome alignment coverage","Cow orthology confidence [0 low, 1 high]","Crab-eating macaque gene stable ID","Crab-eating macaque gene name","Crab-eating macaque protein or transcript stable ID","Crab-eating macaque chromosome/scaffold name","Crab-eating macaque chromosome/scaffold start (bp)","Crab-eating macaque chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Crab-eating macaque","Crab-eating macaque homology type","%id. target Crab-eating macaque gene identical to query gene","%id. query gene identical to target Crab-eating macaque gene","Crab-eating macaque Gene-order conservation score","Crab-eating macaque Whole-genome alignment coverage","Crab-eating macaque orthology confidence [0 low, 1 high]","Degu gene stable ID","Degu gene name","Degu protein or transcript stable ID","Degu chromosome/scaffold name","Degu chromosome/scaffold start (bp)","Degu chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Degu","Degu homology type","%id. target Degu gene identical to query gene","%id. query gene identical to target Degu gene","Degu Gene-order conservation score","Degu Whole-genome alignment coverage","Degu orthology confidence [0 low, 1 high]","Denticle herring gene stable ID","Denticle herring gene name","Denticle herring protein or transcript stable ID","Denticle herring chromosome/scaffold name","Denticle herring chromosome/scaffold start (bp)","Denticle herring chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Denticle herring","Denticle herring homology type","%id. target Denticle herring gene identical to query gene","%id. query gene identical to target Denticle herring gene","Denticle herring Gene-order conservation score","Denticle herring orthology confidence [0 low, 1 high]","Dingo gene stable ID","Dingo gene name","Dingo protein or transcript stable ID","Dingo chromosome/scaffold name","Dingo chromosome/scaffold start (bp)","Dingo chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Dingo","Dingo homology type","%id. target Dingo gene identical to query gene","%id. query gene identical to target Dingo gene","Dingo Gene-order conservation score","Dingo Whole-genome alignment coverage","Dingo orthology confidence [0 low, 1 high]","Dog gene stable ID","Dog gene name","Dog protein or transcript stable ID","Dog chromosome/scaffold name","Dog chromosome/scaffold start (bp)","Dog chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Dog","Dog homology type","%id. target Dog gene identical to query gene","%id. query gene identical to target Dog gene","Dog Gene-order conservation score","Dog Whole-genome alignment coverage","Dog orthology confidence [0 low, 1 high]","Domestic yak gene stable ID","Domestic yak gene name","Domestic yak protein or transcript stable ID","Domestic yak chromosome/scaffold name","Domestic yak chromosome/scaffold start (bp)","Domestic yak chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Domestic yak","Domestic yak homology type","%id. target Domestic yak gene identical to query gene","%id. query gene identical to target Domestic yak gene","Domestic yak Gene-order conservation score","Domestic yak Whole-genome alignment coverage","Domestic yak orthology confidence [0 low, 1 high]","Donkey gene stable ID","Donkey gene name","Donkey protein or transcript stable ID","Donkey chromosome/scaffold name","Donkey chromosome/scaffold start (bp)","Donkey chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Donkey","Donkey homology type","%id. target Donkey gene identical to query gene","%id. query gene identical to target Donkey gene","Donkey Gene-order conservation score","Donkey orthology confidence [0 low, 1 high]","Drill gene stable ID","Drill gene name","Drill protein or transcript stable ID","Drill chromosome/scaffold name","Drill chromosome/scaffold start (bp)","Drill chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Drill","Drill homology type","%id. target Drill gene identical to query gene","%id. query gene identical to target Drill gene","Drill Gene-order conservation score","Drill Whole-genome alignment coverage","Drill orthology confidence [0 low, 1 high]","Drosophila melanogaster (Fruit fly) gene stable ID","Drosophila melanogaster (Fruit fly) gene name","Drosophila melanogaster (Fruit fly) protein or transcript stable ID","Drosophila melanogaster (Fruit fly) chromosome/scaffold name","Drosophila melanogaster (Fruit fly) chromosome/scaffold start (bp)","Drosophila melanogaster (Fruit fly) chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Drosophila melanogaster (Fruit fly)","Drosophila melanogaster (Fruit fly) homology type","%id. target Drosophila melanogaster (Fruit fly) gene identical to query gene","%id. query gene identical to target Drosophila melanogaster (Fruit fly) gene","Drosophila melanogaster (Fruit fly) orthology confidence [0 low, 1 high]","Duck gene stable ID","Duck gene name","Duck protein or transcript stable ID","Duck chromosome/scaffold name","Duck chromosome/scaffold start (bp)","Duck chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Duck","Duck homology type","%id. target Duck gene identical to query gene","%id. query gene identical to target Duck gene","Duck Gene-order conservation score","Duck Whole-genome alignment coverage","Duck orthology confidence [0 low, 1 high]","Eastern happy gene stable ID","Eastern happy gene name","Eastern happy protein or transcript stable ID","Eastern happy chromosome/scaffold name","Eastern happy chromosome/scaffold start (bp)","Eastern happy chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Eastern happy","Eastern happy homology type","%id. target Eastern happy gene identical to query gene","%id. query gene identical to target Eastern happy gene","Eastern happy Gene-order conservation score","Eastern happy orthology confidence [0 low, 1 high]","Electric eel gene stable ID","Electric eel gene name","Electric eel protein or transcript stable ID","Electric eel chromosome/scaffold name","Electric eel chromosome/scaffold start (bp)","Electric eel chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Electric eel","Electric eel homology type","%id. target Electric eel gene identical to query gene","%id. query gene identical to target Electric eel gene","Electric eel Gene-order conservation score","Electric eel orthology confidence [0 low, 1 high]","Elephant gene stable ID","Elephant gene name","Elephant protein or transcript stable ID","Elephant chromosome/scaffold name","Elephant chromosome/scaffold start (bp)","Elephant chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Elephant","Elephant homology type","%id. target Elephant gene identical to query gene","%id. query gene identical to target Elephant gene","Elephant Gene-order conservation score","Elephant Whole-genome alignment coverage","Elephant orthology confidence [0 low, 1 high]","Elephant shark gene stable ID","Elephant shark gene name","Elephant shark protein or transcript stable ID","Elephant shark chromosome/scaffold name","Elephant shark chromosome/scaffold start (bp)","Elephant shark chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Elephant shark","Elephant shark homology type","%id. target Elephant shark gene identical to query gene","%id. query gene identical to target Elephant shark gene","Elephant shark Whole-genome alignment coverage","Elephant shark orthology confidence [0 low, 1 high]","European seabass gene stable ID","European seabass gene name","European seabass protein or transcript stable ID","European seabass chromosome/scaffold name","European seabass chromosome/scaffold start (bp)","European seabass chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with European seabass","European seabass homology type","%id. target European seabass gene identical to query gene","%id. query gene identical to target European seabass gene","European seabass Gene-order conservation score","European seabass orthology confidence [0 low, 1 high]","Ferret gene stable ID","Ferret gene name","Ferret protein or transcript stable ID","Ferret chromosome/scaffold name","Ferret chromosome/scaffold start (bp)","Ferret chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Ferret","Ferret homology type","%id. target Ferret gene identical to query gene","%id. query gene identical to target Ferret gene","Ferret Gene-order conservation score","Ferret Whole-genome alignment coverage","Ferret orthology confidence [0 low, 1 high]","Giant panda gene stable ID","Giant panda gene name","Giant panda protein or transcript stable ID","Giant panda chromosome/scaffold name","Giant panda chromosome/scaffold start (bp)","Giant panda chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Giant panda","Giant panda homology type","%id. target Giant panda gene identical to query gene","%id. query gene identical to target Giant panda gene","Giant panda Gene-order conservation score","Giant panda orthology confidence [0 low, 1 high]","Gibbon gene stable ID","Gibbon gene name","Gibbon protein or transcript stable ID","Gibbon chromosome/scaffold name","Gibbon chromosome/scaffold start (bp)","Gibbon chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Gibbon","Gibbon homology type","%id. target Gibbon gene identical to query gene","%id. query gene identical to target Gibbon gene","Gibbon Gene-order conservation score","Gibbon Whole-genome alignment coverage","Gibbon orthology confidence [0 low, 1 high]","Goat gene stable ID","Goat gene name","Goat protein or transcript stable ID","Goat chromosome/scaffold name","Goat chromosome/scaffold start (bp)","Goat chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Goat","Goat homology type","%id. target Goat gene identical to query gene","%id. query gene identical to target Goat gene","Goat Gene-order conservation score","Goat Whole-genome alignment coverage","Goat orthology confidence [0 low, 1 high]","Golden Hamster gene stable ID","Golden Hamster gene name","Golden Hamster protein or transcript stable ID","Golden Hamster chromosome/scaffold name","Golden Hamster chromosome/scaffold start (bp)","Golden Hamster chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Golden Hamster","Golden Hamster homology type","%id. target Golden Hamster gene identical to query gene","%id. query gene identical to target Golden Hamster gene","Golden Hamster Gene-order conservation score","Golden Hamster Whole-genome alignment coverage","Golden Hamster orthology confidence [0 low, 1 high]","Golden eagle gene stable ID","Golden eagle gene name","Golden eagle protein or transcript stable ID","Golden eagle chromosome/scaffold name","Golden eagle chromosome/scaffold start (bp)","Golden eagle chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Golden eagle","Golden eagle homology type","%id. target Golden eagle gene identical to query gene","%id. query gene identical to target Golden eagle gene","Golden eagle Gene-order conservation score","Golden eagle Whole-genome alignment coverage","Golden eagle orthology confidence [0 low, 1 high]","Goldfish gene stable ID","Goldfish gene name","Goldfish protein or transcript stable ID","Goldfish chromosome/scaffold name","Goldfish chromosome/scaffold start (bp)","Goldfish chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Goldfish","Goldfish homology type","%id. target Goldfish gene identical to query gene","%id. query gene identical to target Goldfish gene","Goldfish Gene-order conservation score","Goldfish Whole-genome alignment coverage","Goldfish orthology confidence [0 low, 1 high]","Goodes thornscrub tortoise gene stable ID","Goodes thornscrub tortoise gene name","Goodes thornscrub tortoise protein or transcript stable ID","Goodes thornscrub tortoise chromosome/scaffold name","Goodes thornscrub tortoise chromosome/scaffold start (bp)","Goodes thornscrub tortoise chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Goodes thornscrub tortoise","Goodes thornscrub tortoise homology type","%id. target Goodes thornscrub tortoise gene identical to query gene","%id. query gene identical to target Goodes thornscrub tortoise gene","Goodes thornscrub tortoise Gene-order conservation score","Goodes thornscrub tortoise Whole-genome alignment coverage","Goodes thornscrub tortoise orthology confidence [0 low, 1 high]","Gorilla gene stable ID","Gorilla gene name","Gorilla protein or transcript stable ID","Gorilla chromosome/scaffold name","Gorilla chromosome/scaffold start (bp)","Gorilla chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Gorilla","Gorilla homology type","%id. target Gorilla gene identical to query gene","%id. query gene identical to target Gorilla gene","Gorilla Gene-order conservation score","Gorilla Whole-genome alignment coverage","Gorilla orthology confidence [0 low, 1 high]","Green anole gene stable ID","Green anole gene name","Green anole protein or transcript stable ID","Green anole chromosome/scaffold name","Green anole chromosome/scaffold start (bp)","Green anole chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Green anole","Green anole homology type","%id. target Green anole gene identical to query gene","%id. query gene identical to target Green anole gene","Green anole Gene-order conservation score","Green anole orthology confidence [0 low, 1 high]","Guinea Pig gene stable ID","Guinea Pig gene name","Guinea Pig protein or transcript stable ID","Guinea Pig chromosome/scaffold name","Guinea Pig chromosome/scaffold start (bp)","Guinea Pig chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Guinea Pig","Guinea Pig homology type","%id. target Guinea Pig gene identical to query gene","%id. query gene identical to target Guinea Pig gene","Guinea Pig Gene-order conservation score","Guinea Pig Whole-genome alignment coverage","Guinea Pig orthology confidence [0 low, 1 high]","Hagfish gene stable ID","Hagfish gene name","Hagfish protein or transcript stable ID","Hagfish chromosome/scaffold name","Hagfish chromosome/scaffold start (bp)","Hagfish chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Hagfish","Hagfish homology type","%id. target Hagfish gene identical to query gene","%id. query gene identical to target Hagfish gene","Hagfish Whole-genome alignment coverage","Hagfish orthology confidence [0 low, 1 high]","Hedgehog gene stable ID","Hedgehog gene name","Hedgehog protein or transcript stable ID","Hedgehog chromosome/scaffold name","Hedgehog chromosome/scaffold start (bp)","Hedgehog chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Hedgehog","Hedgehog homology type","%id. target Hedgehog gene identical to query gene","%id. query gene identical to target Hedgehog gene","Hedgehog Gene-order conservation score","Hedgehog Whole-genome alignment coverage","Hedgehog orthology confidence [0 low, 1 high]","Horse gene stable ID","Horse gene name","Horse protein or transcript stable ID","Horse chromosome/scaffold name","Horse chromosome/scaffold start (bp)","Horse chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Horse","Horse homology type","%id. target Horse gene identical to query gene","%id. query gene identical to target Horse gene","Horse Gene-order conservation score","Horse Whole-genome alignment coverage","Horse orthology confidence [0 low, 1 high]","Huchen gene stable ID","Huchen gene name","Huchen protein or transcript stable ID","Huchen chromosome/scaffold name","Huchen chromosome/scaffold start (bp)","Huchen chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Huchen","Huchen homology type","%id. target Huchen gene identical to query gene","%id. query gene identical to target Huchen gene","Huchen Gene-order conservation score","Huchen Whole-genome alignment coverage","Huchen orthology confidence [0 low, 1 high]","Hybrid - Bos Indicus gene stable ID","Hybrid - Bos Indicus gene name","Hybrid - Bos Indicus protein or transcript stable ID","Hybrid - Bos Indicus chromosome/scaffold name","Hybrid - Bos Indicus chromosome/scaffold start (bp)","Hybrid - Bos Indicus chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Hybrid - Bos Indicus","Hybrid - Bos Indicus homology type","%id. target Hybrid - Bos Indicus gene identical to query gene","%id. query gene identical to target Hybrid - Bos Indicus gene","Hybrid - Bos Indicus Gene-order conservation score","Hybrid - Bos Indicus Whole-genome alignment coverage","Hybrid - Bos Indicus orthology confidence [0 low, 1 high]","Indian cobra gene stable ID","Indian cobra gene name","Indian cobra protein or transcript stable ID","Indian cobra chromosome/scaffold name","Indian cobra chromosome/scaffold start (bp)","Indian cobra chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Indian cobra","Indian cobra homology type","%id. target Indian cobra gene identical to query gene","%id. query gene identical to target Indian cobra gene","Indian cobra Gene-order conservation score","Indian cobra Whole-genome alignment coverage","Indian cobra orthology confidence [0 low, 1 high]","Japanese quail gene stable ID","Japanese quail gene name","Japanese quail protein or transcript stable ID","Japanese quail chromosome/scaffold name","Japanese quail chromosome/scaffold start (bp)","Japanese quail chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Japanese quail","Japanese quail homology type","%id. target Japanese quail gene identical to query gene","%id. query gene identical to target Japanese quail gene","Japanese quail Gene-order conservation score","Japanese quail Whole-genome alignment coverage","Japanese quail orthology confidence [0 low, 1 high]","Kangaroo rat gene stable ID","Kangaroo rat gene name","Kangaroo rat protein or transcript stable ID","Kangaroo rat chromosome/scaffold name","Kangaroo rat chromosome/scaffold start (bp)","Kangaroo rat chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Kangaroo rat","Kangaroo rat homology type","%id. target Kangaroo rat gene identical to query gene","%id. query gene identical to target Kangaroo rat gene","Kangaroo rat Gene-order conservation score","Kangaroo rat Whole-genome alignment coverage","Kangaroo rat orthology confidence [0 low, 1 high]","Large yellow croaker gene stable ID","Large yellow croaker gene name","Large yellow croaker protein or transcript stable ID","Large yellow croaker chromosome/scaffold name","Large yellow croaker chromosome/scaffold start (bp)","Large yellow croaker chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Large yellow croaker","Large yellow croaker homology type","%id. target Large yellow croaker gene identical to query gene","%id. query gene identical to target Large yellow croaker gene","Large yellow croaker Gene-order conservation score","Large yellow croaker Whole-genome alignment coverage","Large yellow croaker orthology confidence [0 low, 1 high]","Leishan spiny toad gene stable ID","Leishan spiny toad gene name","Leishan spiny toad protein or transcript stable ID","Leishan spiny toad chromosome/scaffold name","Leishan spiny toad chromosome/scaffold start (bp)","Leishan spiny toad chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Leishan spiny toad","Leishan spiny toad homology type","%id. target Leishan spiny toad gene identical to query gene","%id. query gene identical to target Leishan spiny toad gene","Leishan spiny toad Gene-order conservation score","Leishan spiny toad Whole-genome alignment coverage","Leishan spiny toad orthology confidence [0 low, 1 high]","Lesser Egyptian jerboa gene stable ID","Lesser Egyptian jerboa gene name","Lesser Egyptian jerboa protein or transcript stable ID","Lesser Egyptian jerboa chromosome/scaffold name","Lesser Egyptian jerboa chromosome/scaffold start (bp)","Lesser Egyptian jerboa chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Lesser Egyptian jerboa","Lesser Egyptian jerboa homology type","%id. target Lesser Egyptian jerboa gene identical to query gene","%id. query gene identical to target Lesser Egyptian jerboa gene","Lesser Egyptian jerboa Gene-order conservation score","Lesser Egyptian jerboa Whole-genome alignment coverage","Lesser Egyptian jerboa orthology confidence [0 low, 1 high]","Lesser hedgehog tenrec gene stable ID","Lesser hedgehog tenrec gene name","Lesser hedgehog tenrec protein or transcript stable ID","Lesser hedgehog tenrec chromosome/scaffold name","Lesser hedgehog tenrec chromosome/scaffold start (bp)","Lesser hedgehog tenrec chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Lesser hedgehog tenrec","Lesser hedgehog tenrec homology type","%id. target Lesser hedgehog tenrec gene identical to query gene","%id. query gene identical to target Lesser hedgehog tenrec gene","Lesser hedgehog tenrec Gene-order conservation score","Lesser hedgehog tenrec Whole-genome alignment coverage","Lesser hedgehog tenrec orthology confidence [0 low, 1 high]","Long-tailed chinchilla gene stable ID","Long-tailed chinchilla gene name","Long-tailed chinchilla protein or transcript stable ID","Long-tailed chinchilla chromosome/scaffold name","Long-tailed chinchilla chromosome/scaffold start (bp)","Long-tailed chinchilla chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Long-tailed chinchilla","Long-tailed chinchilla homology type","%id. target Long-tailed chinchilla gene identical to query gene","%id. query gene identical to target Long-tailed chinchilla gene","Long-tailed chinchilla Gene-order conservation score","Long-tailed chinchilla Whole-genome alignment coverage","Long-tailed chinchilla orthology confidence [0 low, 1 high]","Lumpfish gene stable ID","Lumpfish gene name","Lumpfish protein or transcript stable ID","Lumpfish chromosome/scaffold name","Lumpfish chromosome/scaffold start (bp)","Lumpfish chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Lumpfish","Lumpfish homology type","%id. target Lumpfish gene identical to query gene","%id. query gene identical to target Lumpfish gene","Lumpfish Gene-order conservation score","Lumpfish Whole-genome alignment coverage","Lumpfish orthology confidence [0 low, 1 high]","Lyretail cichlid gene stable ID","Lyretail cichlid gene name","Lyretail cichlid protein or transcript stable ID","Lyretail cichlid chromosome/scaffold name","Lyretail cichlid chromosome/scaffold start (bp)","Lyretail cichlid chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Lyretail cichlid","Lyretail cichlid homology type","%id. target Lyretail cichlid gene identical to query gene","%id. query gene identical to target Lyretail cichlid gene","Lyretail cichlid Gene-order conservation score","Lyretail cichlid Whole-genome alignment coverage","Lyretail cichlid orthology confidence [0 low, 1 high]","Ma's night monkey gene stable ID","Ma's night monkey gene name","Ma's night monkey protein or transcript stable ID","Ma's night monkey chromosome/scaffold name","Ma's night monkey chromosome/scaffold start (bp)","Ma's night monkey chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Ma's night monkey","Ma's night monkey homology type","%id. target Ma's night monkey gene identical to query gene","%id. query gene identical to target Ma's night monkey gene","Ma's night monkey Gene-order conservation score","Ma's night monkey Whole-genome alignment coverage","Ma's night monkey orthology confidence [0 low, 1 high]","Macaque gene stable ID","Macaque gene name","Macaque protein or transcript stable ID","Macaque chromosome/scaffold name","Macaque chromosome/scaffold start (bp)","Macaque chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Macaque","Macaque homology type","%id. target Macaque gene identical to query gene","%id. query gene identical to target Macaque gene","Macaque Gene-order conservation score","Macaque Whole-genome alignment coverage","Macaque orthology confidence [0 low, 1 high]","Mainland tiger snake gene stable ID","Mainland tiger snake gene name","Mainland tiger snake protein or transcript stable ID","Mainland tiger snake chromosome/scaffold name","Mainland tiger snake chromosome/scaffold start (bp)","Mainland tiger snake chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Mainland tiger snake","Mainland tiger snake homology type","%id. target Mainland tiger snake gene identical to query gene","%id. query gene identical to target Mainland tiger snake gene","Mainland tiger snake Gene-order conservation score","Mainland tiger snake Whole-genome alignment coverage","Mainland tiger snake orthology confidence [0 low, 1 high]","Mangrove rivulus gene stable ID","Mangrove rivulus gene name","Mangrove rivulus protein or transcript stable ID","Mangrove rivulus chromosome/scaffold name","Mangrove rivulus chromosome/scaffold start (bp)","Mangrove rivulus chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Mangrove rivulus","Mangrove rivulus homology type","%id. target Mangrove rivulus gene identical to query gene","%id. query gene identical to target Mangrove rivulus gene","Mangrove rivulus Gene-order conservation score","Mangrove rivulus Whole-genome alignment coverage","Mangrove rivulus orthology confidence [0 low, 1 high]","Medium ground-finch gene stable ID","Medium ground-finch gene name","Medium ground-finch protein or transcript stable ID","Medium ground-finch chromosome/scaffold name","Medium ground-finch chromosome/scaffold start (bp)","Medium ground-finch chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Medium ground-finch","Medium ground-finch homology type","%id. target Medium ground-finch gene identical to query gene","%id. query gene identical to target Medium ground-finch gene","Medium ground-finch Gene-order conservation score","Medium ground-finch Whole-genome alignment coverage","Medium ground-finch orthology confidence [0 low, 1 high]","Mexican tetra gene stable ID","Mexican tetra gene name","Mexican tetra protein or transcript stable ID","Mexican tetra chromosome/scaffold name","Mexican tetra chromosome/scaffold start (bp)","Mexican tetra chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Mexican tetra","Mexican tetra homology type","%id. target Mexican tetra gene identical to query gene","%id. query gene identical to target Mexican tetra gene","Mexican tetra Gene-order conservation score","Mexican tetra Whole-genome alignment coverage","Mexican tetra orthology confidence [0 low, 1 high]","Microbat gene stable ID","Microbat gene name","Microbat protein or transcript stable ID","Microbat chromosome/scaffold name","Microbat chromosome/scaffold start (bp)","Microbat chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Microbat","Microbat homology type","%id. target Microbat gene identical to query gene","%id. query gene identical to target Microbat gene","Microbat Gene-order conservation score","Microbat Whole-genome alignment coverage","Microbat orthology confidence [0 low, 1 high]","Midas cichlid gene stable ID","Midas cichlid gene name","Midas cichlid protein or transcript stable ID","Midas cichlid chromosome/scaffold name","Midas cichlid chromosome/scaffold start (bp)","Midas cichlid chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Midas cichlid","Midas cichlid homology type","%id. target Midas cichlid gene identical to query gene","%id. query gene identical to target Midas cichlid gene","Midas cichlid Gene-order conservation score","Midas cichlid Whole-genome alignment coverage","Midas cichlid orthology confidence [0 low, 1 high]","Mouse gene stable ID","Mouse gene name","Mouse protein or transcript stable ID","Mouse chromosome/scaffold name","Mouse chromosome/scaffold start (bp)","Mouse chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Mouse","Mouse homology type","%id. target Mouse gene identical to query gene","%id. query gene identical to target Mouse gene","Mouse Gene-order conservation score","Mouse Whole-genome alignment coverage","Mouse orthology confidence [0 low, 1 high]","Mouse Lemur gene stable ID","Mouse Lemur gene name","Mouse Lemur protein or transcript stable ID","Mouse Lemur chromosome/scaffold name","Mouse Lemur chromosome/scaffold start (bp)","Mouse Lemur chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Mouse Lemur","Mouse Lemur homology type","%id. target Mouse Lemur gene identical to query gene","%id. query gene identical to target Mouse Lemur gene","Mouse Lemur Gene-order conservation score","Mouse Lemur Whole-genome alignment coverage","Mouse Lemur orthology confidence [0 low, 1 high]","Mummichog gene stable ID","Mummichog gene name","Mummichog protein or transcript stable ID","Mummichog chromosome/scaffold name","Mummichog chromosome/scaffold start (bp)","Mummichog chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Mummichog","Mummichog homology type","%id. target Mummichog gene identical to query gene","%id. query gene identical to target Mummichog gene","Mummichog Gene-order conservation score","Mummichog Whole-genome alignment coverage","Mummichog orthology confidence [0 low, 1 high]","Naked mole-rat female gene stable ID","Naked mole-rat female gene name","Naked mole-rat female protein or transcript stable ID","Naked mole-rat female chromosome/scaffold name","Naked mole-rat female chromosome/scaffold start (bp)","Naked mole-rat female chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Naked mole-rat female","Naked mole-rat female homology type","%id. target Naked mole-rat female gene identical to query gene","%id. query gene identical to target Naked mole-rat female gene","Naked mole-rat female Gene-order conservation score","Naked mole-rat female orthology confidence [0 low, 1 high]","Narwhal gene stable ID","Narwhal gene name","Narwhal protein or transcript stable ID","Narwhal chromosome/scaffold name","Narwhal chromosome/scaffold start (bp)","Narwhal chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Narwhal","Narwhal homology type","%id. target Narwhal gene identical to query gene","%id. query gene identical to target Narwhal gene","Narwhal Gene-order conservation score","Narwhal Whole-genome alignment coverage","Narwhal orthology confidence [0 low, 1 high]","Northern pike gene stable ID","Northern pike gene name","Northern pike protein or transcript stable ID","Northern pike chromosome/scaffold name","Northern pike chromosome/scaffold start (bp)","Northern pike chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Northern pike","Northern pike homology type","%id. target Northern pike gene identical to query gene","%id. query gene identical to target Northern pike gene","Northern pike Gene-order conservation score","Northern pike orthology confidence [0 low, 1 high]","Opossum gene stable ID","Opossum gene name","Opossum protein or transcript stable ID","Opossum chromosome/scaffold name","Opossum chromosome/scaffold start (bp)","Opossum chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Opossum","Opossum homology type","%id. target Opossum gene identical to query gene","%id. query gene identical to target Opossum gene","Opossum Gene-order conservation score","Opossum Whole-genome alignment coverage","Opossum orthology confidence [0 low, 1 high]","Orange clownfish gene stable ID","Orange clownfish gene name","Orange clownfish protein or transcript stable ID","Orange clownfish chromosome/scaffold name","Orange clownfish chromosome/scaffold start (bp)","Orange clownfish chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Orange clownfish","Orange clownfish homology type","%id. target Orange clownfish gene identical to query gene","%id. query gene identical to target Orange clownfish gene","Orange clownfish Gene-order conservation score","Orange clownfish Whole-genome alignment coverage","Orange clownfish orthology confidence [0 low, 1 high]","Painted turtle gene stable ID","Painted turtle gene name","Painted turtle protein or transcript stable ID","Painted turtle chromosome/scaffold name","Painted turtle chromosome/scaffold start (bp)","Painted turtle chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Painted turtle","Painted turtle homology type","%id. target Painted turtle gene identical to query gene","%id. query gene identical to target Painted turtle gene","Painted turtle Gene-order conservation score","Painted turtle Whole-genome alignment coverage","Painted turtle orthology confidence [0 low, 1 high]","Panamanian white-faced capuchin gene stable ID","Panamanian white-faced capuchin gene name","Panamanian white-faced capuchin protein or transcript stable ID","Panamanian white-faced capuchin chromosome/scaffold name","Panamanian white-faced capuchin chromosome/scaffold start (bp)","Panamanian white-faced capuchin chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Panamanian white-faced capuchin","Panamanian white-faced capuchin homology type","%id. target Panamanian white-faced capuchin gene identical to query gene","%id. query gene identical to target Panamanian white-faced capuchin gene","Panamanian white-faced capuchin Gene-order conservation score","Panamanian white-faced capuchin Whole-genome alignment coverage","Panamanian white-faced capuchin orthology confidence [0 low, 1 high]","Pig-tailed macaque gene stable ID","Pig-tailed macaque gene name","Pig-tailed macaque protein or transcript stable ID","Pig-tailed macaque chromosome/scaffold name","Pig-tailed macaque chromosome/scaffold start (bp)","Pig-tailed macaque chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Pig-tailed macaque","Pig-tailed macaque homology type","%id. target Pig-tailed macaque gene identical to query gene","%id. query gene identical to target Pig-tailed macaque gene","Pig-tailed macaque Gene-order conservation score","Pig-tailed macaque Whole-genome alignment coverage","Pig-tailed macaque orthology confidence [0 low, 1 high]","Pika gene stable ID","Pika gene name","Pika protein or transcript stable ID","Pika chromosome/scaffold name","Pika chromosome/scaffold start (bp)","Pika chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Pika","Pika homology type","%id. target Pika gene identical to query gene","%id. query gene identical to target Pika gene","Pika Gene-order conservation score","Pika Whole-genome alignment coverage","Pika orthology confidence [0 low, 1 high]","Pinecone soldierfish gene stable ID","Pinecone soldierfish gene name","Pinecone soldierfish protein or transcript stable ID","Pinecone soldierfish chromosome/scaffold name","Pinecone soldierfish chromosome/scaffold start (bp)","Pinecone soldierfish chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Pinecone soldierfish","Pinecone soldierfish homology type","%id. target Pinecone soldierfish gene identical to query gene","%id. query gene identical to target Pinecone soldierfish gene","Pinecone soldierfish Gene-order conservation score","Pinecone soldierfish Whole-genome alignment coverage","Pinecone soldierfish orthology confidence [0 low, 1 high]","Pink-footed goose gene stable ID","Pink-footed goose gene name","Pink-footed goose protein or transcript stable ID","Pink-footed goose chromosome/scaffold name","Pink-footed goose chromosome/scaffold start (bp)","Pink-footed goose chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Pink-footed goose","Pink-footed goose homology type","%id. target Pink-footed goose gene identical to query gene","%id. query gene identical to target Pink-footed goose gene","Pink-footed goose Gene-order conservation score","Pink-footed goose Whole-genome alignment coverage","Pink-footed goose orthology confidence [0 low, 1 high]","Prairie vole gene stable ID","Prairie vole gene name","Prairie vole protein or transcript stable ID","Prairie vole chromosome/scaffold name","Prairie vole chromosome/scaffold start (bp)","Prairie vole chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Prairie vole","Prairie vole homology type","%id. target Prairie vole gene identical to query gene","%id. query gene identical to target Prairie vole gene","Prairie vole Gene-order conservation score","Prairie vole Whole-genome alignment coverage","Prairie vole orthology confidence [0 low, 1 high]","Reedfish gene stable ID","Reedfish gene name","Reedfish protein or transcript stable ID","Reedfish chromosome/scaffold name","Reedfish chromosome/scaffold start (bp)","Reedfish chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Reedfish","Reedfish homology type","%id. target Reedfish gene identical to query gene","%id. query gene identical to target Reedfish gene","Reedfish Gene-order conservation score","Reedfish Whole-genome alignment coverage","Reedfish orthology confidence [0 low, 1 high]","Ryukyu mouse gene stable ID","Ryukyu mouse gene name","Ryukyu mouse protein or transcript stable ID","Ryukyu mouse chromosome/scaffold name","Ryukyu mouse chromosome/scaffold start (bp)","Ryukyu mouse chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Ryukyu mouse","Ryukyu mouse homology type","%id. target Ryukyu mouse gene identical to query gene","%id. query gene identical to target Ryukyu mouse gene","Ryukyu mouse Gene-order conservation score","Ryukyu mouse Whole-genome alignment coverage","Ryukyu mouse orthology confidence [0 low, 1 high]","Sheepshead minnow gene stable ID","Sheepshead minnow gene name","Sheepshead minnow protein or transcript stable ID","Sheepshead minnow chromosome/scaffold name","Sheepshead minnow chromosome/scaffold start (bp)","Sheepshead minnow chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Sheepshead minnow","Sheepshead minnow homology type","%id. target Sheepshead minnow gene identical to query gene","%id. query gene identical to target Sheepshead minnow gene","Sheepshead minnow Gene-order conservation score","Sheepshead minnow Whole-genome alignment coverage","Sheepshead minnow orthology confidence [0 low, 1 high]","Shrew mouse gene stable ID","Shrew mouse gene name","Shrew mouse protein or transcript stable ID","Shrew mouse chromosome/scaffold name","Shrew mouse chromosome/scaffold start (bp)","Shrew mouse chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Shrew mouse","Shrew mouse homology type","%id. target Shrew mouse gene identical to query gene","%id. query gene identical to target Shrew mouse gene","Shrew mouse Gene-order conservation score","Shrew mouse Whole-genome alignment coverage","Shrew mouse orthology confidence [0 low, 1 high]","Siamese fighting fish gene stable ID","Siamese fighting fish gene name","Siamese fighting fish protein or transcript stable ID","Siamese fighting fish chromosome/scaffold name","Siamese fighting fish chromosome/scaffold start (bp)","Siamese fighting fish chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Siamese fighting fish","Siamese fighting fish homology type","%id. target Siamese fighting fish gene identical to query gene","%id. query gene identical to target Siamese fighting fish gene","Siamese fighting fish Gene-order conservation score","Siamese fighting fish Whole-genome alignment coverage","Siamese fighting fish orthology confidence [0 low, 1 high]","Siberian musk deer gene stable ID","Siberian musk deer gene name","Siberian musk deer protein or transcript stable ID","Siberian musk deer chromosome/scaffold name","Siberian musk deer chromosome/scaffold start (bp)","Siberian musk deer chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Siberian musk deer","Siberian musk deer homology type","%id. target Siberian musk deer gene identical to query gene","%id. query gene identical to target Siberian musk deer gene","Siberian musk deer Gene-order conservation score","Siberian musk deer Whole-genome alignment coverage","Siberian musk deer orthology confidence [0 low, 1 high]","Sloth gene stable ID","Sloth gene name","Sloth protein or transcript stable ID","Sloth chromosome/scaffold name","Sloth chromosome/scaffold start (bp)","Sloth chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Sloth","Sloth homology type","%id. target Sloth gene identical to query gene","%id. query gene identical to target Sloth gene","Sloth Gene-order conservation score","Sloth Whole-genome alignment coverage","Sloth orthology confidence [0 low, 1 high]","Sooty mangabey gene stable ID","Sooty mangabey gene name","Sooty mangabey protein or transcript stable ID","Sooty mangabey chromosome/scaffold name","Sooty mangabey chromosome/scaffold start (bp)","Sooty mangabey chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Sooty mangabey","Sooty mangabey homology type","%id. target Sooty mangabey gene identical to query gene","%id. query gene identical to target Sooty mangabey gene","Sooty mangabey Gene-order conservation score","Sooty mangabey Whole-genome alignment coverage","Sooty mangabey orthology confidence [0 low, 1 high]","Spiny chromis gene stable ID","Spiny chromis gene name","Spiny chromis protein or transcript stable ID","Spiny chromis chromosome/scaffold name","Spiny chromis chromosome/scaffold start (bp)","Spiny chromis chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Spiny chromis","Spiny chromis homology type","%id. target Spiny chromis gene identical to query gene","%id. query gene identical to target Spiny chromis gene","Spiny chromis Gene-order conservation score","Spiny chromis Whole-genome alignment coverage","Spiny chromis orthology confidence [0 low, 1 high]","Spotted gar gene stable ID","Spotted gar gene name","Spotted gar protein or transcript stable ID","Spotted gar chromosome/scaffold name","Spotted gar chromosome/scaffold start (bp)","Spotted gar chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Spotted gar","Spotted gar homology type","%id. target Spotted gar gene identical to query gene","%id. query gene identical to target Spotted gar gene","Spotted gar Gene-order conservation score","Spotted gar Whole-genome alignment coverage","Spotted gar orthology confidence [0 low, 1 high]","Squirrel gene stable ID","Squirrel gene name","Squirrel protein or transcript stable ID","Squirrel chromosome/scaffold name","Squirrel chromosome/scaffold start (bp)","Squirrel chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Squirrel","Squirrel homology type","%id. target Squirrel gene identical to query gene","%id. query gene identical to target Squirrel gene","Squirrel Gene-order conservation score","Squirrel Whole-genome alignment coverage","Squirrel orthology confidence [0 low, 1 high]","Steppe mouse gene stable ID","Steppe mouse gene name","Steppe mouse protein or transcript stable ID","Steppe mouse chromosome/scaffold name","Steppe mouse chromosome/scaffold start (bp)","Steppe mouse chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Steppe mouse","Steppe mouse homology type","%id. target Steppe mouse gene identical to query gene","%id. query gene identical to target Steppe mouse gene","Steppe mouse Gene-order conservation score","Steppe mouse Whole-genome alignment coverage","Steppe mouse orthology confidence [0 low, 1 high]","Stickleback gene stable ID","Stickleback gene name","Stickleback protein or transcript stable ID","Stickleback chromosome/scaffold name","Stickleback chromosome/scaffold start (bp)","Stickleback chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Stickleback","Stickleback homology type","%id. target Stickleback gene identical to query gene","%id. query gene identical to target Stickleback gene","Stickleback Gene-order conservation score","Stickleback orthology confidence [0 low, 1 high]","Tarsier gene stable ID","Tarsier gene name","Tarsier protein or transcript stable ID","Tarsier chromosome/scaffold name","Tarsier chromosome/scaffold start (bp)","Tarsier chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Tarsier","Tarsier homology type","%id. target Tarsier gene identical to query gene","%id. query gene identical to target Tarsier gene","Tarsier Gene-order conservation score","Tarsier Whole-genome alignment coverage","Tarsier orthology confidence [0 low, 1 high]","Tiger tail seahorse gene stable ID","Tiger tail seahorse gene name","Tiger tail seahorse protein or transcript stable ID","Tiger tail seahorse chromosome/scaffold name","Tiger tail seahorse chromosome/scaffold start (bp)","Tiger tail seahorse chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Tiger tail seahorse","Tiger tail seahorse homology type","%id. target Tiger tail seahorse gene identical to query gene","%id. query gene identical to target Tiger tail seahorse gene","Tiger tail seahorse Gene-order conservation score","Tiger tail seahorse Whole-genome alignment coverage","Tiger tail seahorse orthology confidence [0 low, 1 high]","Tongue sole gene stable ID","Tongue sole gene name","Tongue sole protein or transcript stable ID","Tongue sole chromosome/scaffold name","Tongue sole chromosome/scaffold start (bp)","Tongue sole chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Tongue sole","Tongue sole homology type","%id. target Tongue sole gene identical to query gene","%id. query gene identical to target Tongue sole gene","Tongue sole Gene-order conservation score","Tongue sole Whole-genome alignment coverage","Tongue sole orthology confidence [0 low, 1 high]","Turkey gene stable ID","Turkey gene name","Turkey protein or transcript stable ID","Turkey chromosome/scaffold name","Turkey chromosome/scaffold start (bp)","Turkey chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Turkey","Turkey homology type","%id. target Turkey gene identical to query gene","%id. query gene identical to target Turkey gene","Turkey Gene-order conservation score","Turkey orthology confidence [0 low, 1 high]","Turquoise killifish gene stable ID","Turquoise killifish gene name","Turquoise killifish protein or transcript stable ID","Turquoise killifish chromosome/scaffold name","Turquoise killifish chromosome/scaffold start (bp)","Turquoise killifish chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Turquoise killifish","Turquoise killifish homology type","%id. target Turquoise killifish gene identical to query gene","%id. query gene identical to target Turquoise killifish gene","Turquoise killifish Gene-order conservation score","Turquoise killifish Whole-genome alignment coverage","Turquoise killifish orthology confidence [0 low, 1 high]","Upper Galilee mountains blind mole rat gene stable ID","Upper Galilee mountains blind mole rat gene name","Upper Galilee mountains blind mole rat protein or transcript stable ID","Upper Galilee mountains blind mole rat chromosome/scaffold name","Upper Galilee mountains blind mole rat chromosome/scaffold start (bp)","Upper Galilee mountains blind mole rat chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Upper Galilee mountains blind mole rat","Upper Galilee mountains blind mole rat homology type","%id. target Upper Galilee mountains blind mole rat gene identical to query gene","%id. query gene identical to target Upper Galilee mountains blind mole rat gene","Upper Galilee mountains blind mole rat Gene-order conservation score","Upper Galilee mountains blind mole rat Whole-genome alignment coverage","Upper Galilee mountains blind mole rat orthology confidence [0 low, 1 high]","Vervet-AGM gene stable ID","Vervet-AGM gene name","Vervet-AGM protein or transcript stable ID","Vervet-AGM chromosome/scaffold name","Vervet-AGM chromosome/scaffold start (bp)","Vervet-AGM chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Vervet-AGM","Vervet-AGM homology type","%id. target Vervet-AGM gene identical to query gene","%id. query gene identical to target Vervet-AGM gene","Vervet-AGM Gene-order conservation score","Vervet-AGM Whole-genome alignment coverage","Vervet-AGM orthology confidence [0 low, 1 high]","Wallaby gene stable ID","Wallaby gene name","Wallaby protein or transcript stable ID","Wallaby chromosome/scaffold name","Wallaby chromosome/scaffold start (bp)","Wallaby chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Wallaby","Wallaby homology type","%id. target Wallaby gene identical to query gene","%id. query gene identical to target Wallaby gene","Wallaby Gene-order conservation score","Wallaby Whole-genome alignment coverage","Wallaby orthology confidence [0 low, 1 high]","White-tufted-ear marmoset gene stable ID","White-tufted-ear marmoset gene name","White-tufted-ear marmoset protein or transcript stable ID","White-tufted-ear marmoset chromosome/scaffold name","White-tufted-ear marmoset chromosome/scaffold start (bp)","White-tufted-ear marmoset chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with White-tufted-ear marmoset","White-tufted-ear marmoset homology type","%id. target White-tufted-ear marmoset gene identical to query gene","%id. query gene identical to target White-tufted-ear marmoset gene","White-tufted-ear marmoset Gene-order conservation score","White-tufted-ear marmoset Whole-genome alignment coverage","White-tufted-ear marmoset orthology confidence [0 low, 1 high]","Wild yak gene stable ID","Wild yak gene name","Wild yak protein or transcript stable ID","Wild yak chromosome/scaffold name","Wild yak chromosome/scaffold start (bp)","Wild yak chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Wild yak","Wild yak homology type","%id. target Wild yak gene identical to query gene","%id. query gene identical to target Wild yak gene","Wild yak Gene-order conservation score","Wild yak Whole-genome alignment coverage","Wild yak orthology confidence [0 low, 1 high]","Yarkand deer gene stable ID","Yarkand deer gene name","Yarkand deer protein or transcript stable ID","Yarkand deer chromosome/scaffold name","Yarkand deer chromosome/scaffold start (bp)","Yarkand deer chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Yarkand deer","Yarkand deer homology type","%id. target Yarkand deer gene identical to query gene","%id. query gene identical to target Yarkand deer gene","Yarkand deer Gene-order conservation score","Yarkand deer Whole-genome alignment coverage","Yarkand deer orthology confidence [0 low, 1 high]","Zebra mbuna gene stable ID","Zebra mbuna gene name","Zebra mbuna protein or transcript stable ID","Zebra mbuna chromosome/scaffold name","Zebra mbuna chromosome/scaffold start (bp)","Zebra mbuna chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Zebra mbuna","Zebra mbuna homology type","%id. target Zebra mbuna gene identical to query gene","%id. query gene identical to target Zebra mbuna gene","Zebra mbuna Gene-order conservation score","Zebra mbuna Whole-genome alignment coverage","Zebra mbuna orthology confidence [0 low, 1 high]","Zebrafish gene stable ID","Zebrafish gene name","Zebrafish protein or transcript stable ID","Zebrafish chromosome/scaffold name","Zebrafish chromosome/scaffold start (bp)","Zebrafish chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Zebrafish","Zebrafish homology type","%id. target Zebrafish gene identical to query gene","%id. query gene identical to target Zebrafish gene","Zebrafish Gene-order conservation score","Zebrafish Whole-genome alignment coverage","Zebrafish orthology confidence [0 low, 1 high]","Zig-zag eel gene stable ID","Zig-zag eel gene name","Zig-zag eel protein or transcript stable ID","Zig-zag eel chromosome/scaffold name","Zig-zag eel chromosome/scaffold start (bp)","Zig-zag eel chromosome/scaffold end (bp)","Query protein or transcript ID","Last common ancestor with Zig-zag eel","Zig-zag eel homology type","%id. target Zig-zag eel gene identical to query gene","%id. query gene identical to target Zig-zag eel gene","Zig-zag eel Gene-order conservation score","Zig-zag eel orthology confidence [0 low, 1 high]","Human paralogue gene stable ID","Human paralogue associated gene name","Human paralogue protein or transcript ID","Human paralogue chromosome/scaffold name","Human paralogue chromosome/scaffold start (bp)","Human paralogue chromosome/scaffold end (bp)","Paralogue query protein or transcript ID","Paralogue last common ancestor with Human","Human paralogue homology type","Paralogue %id. target Human gene identical to query gene","Paralogue %id. query gene identical to target Human gene","Gene stable ID","Gene stable ID version","Version (gene)","Transcript stable ID","Transcript stable ID version","Version (transcript)","Protein stable ID","Protein stable ID version","Version (protein)","Chromosome/scaffold name","Gene start (bp)","Gene end (bp)","Strand","Karyotype band","Gene name","Source of gene name","Transcript count","Gene % GC content","Gene description","Variant name","Variant source","Variant source description","Variant alleles","Variant supporting evidence","Mapweight","Minor allele","Minor allele frequency","Minor allele count","Clinical significance","Transcript location (bp)","Variant chromosome Strand","Protein location (aa)","chromosome/scaffold position start (bp)","Chromosome/scaffold position end (bp)","PolyPhen prediction","PolyPhen score","SIFT prediction","SIFT score","Distance to transcript","CDS start","CDS end","Protein allele","Variant consequence","Consequence specific allele","Gene stable ID","Gene stable ID version","Version (gene)","Transcript stable ID","Transcript stable ID version","Version (transcript)","Protein stable ID","Protein stable ID version","Version (protein)","Chromosome/scaffold name","Gene start (bp)","Gene end (bp)","Strand","Karyotype band","Gene name","Source of gene name","Transcript count","Gene % GC content","Gene description","Variant name","Variant source","Variant source description","Variant alleles","Variant supporting evidence","Mapweight","Transcript location (bp)","Variant chromosome/scaffold strand","Protein location (aa)","Chromosome/scaffold position start (bp)","Chromosome/scaffold position end (bp)","Distance to transcript","CDS start","CDS end","Variant consequence","Consequence specific allele","Unspliced (Transcript)","Unspliced (Gene)","Flank (Transcript)","Flank (Gene)","Flank-coding region (Transcript)","Flank-coding region (Gene)","5' UTR","3' UTR","Exon sequences","cDNA sequences","Coding sequence","Peptide","upstream_flank","downstream_flank","Gene stable ID","Gene stable ID version","Gene description","Gene name","Source of gene name","Chromosome/scaffold name","Gene start (bp)","Gene end (bp)","Gene type","Version (gene)","UniParc ID","UniProtKB/Swiss-Prot ID","UniProtKB/TrEMBL ID","CDS start (within cDNA)","CDS end (within cDNA)","5' UTR start","5' UTR end","3' UTR start","3' UTR end","Transcript stable ID","Transcript stable ID version","Protein stable ID","Protein stable ID version","Transcript type","Version (transcript)","Version (protein)","Strand","Transcript start (bp)","Transcript end (bp)","Transcription start site (TSS)","Transcript length (including UTRs and CDS)","CDS Length","CDS start","CDS end","Exon stable ID","Exon region start (bp)","Exon region end (bp)","Strand","Exon rank in transcript","Start phase","End phase","cDNA coding start","cDNA coding end","Genomic coding start","Genomic coding end","Constitutive exon"],["feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","feature_page","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","structure","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","homologs","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","snp_somatic","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences","sequences"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>description<\/th>\n      <th>page<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"name","targets":1},{"name":"description","targets":2},{"name":"page","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span></span>
<span></span>
<span></span>
<span></span>
<span><span class="va">gene_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"1017"</span>, <span class="st">"1018"</span>, <span class="st">"1019"</span><span class="op">)</span>  </span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/getBM.html">getBM</a></span><span class="op">(</span>attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"entrezgene_id"</span>, <span class="st">"ensembl_gene_id"</span>, <span class="st">"external_gene_name"</span><span class="op">)</span>,</span>
<span>                  filters <span class="op">=</span> <span class="st">"entrezgene_id"</span>,</span>
<span>                  values <span class="op">=</span> <span class="va">gene_ids</span>,</span>
<span>                  mart <span class="op">=</span> <span class="va">mart</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="kable-table">
<table class="caption-top table table-sm table-striped small">
<thead><tr class="header">
<th style="text-align: right;">entrezgene_id</th>
<th style="text-align: left;">ensembl_gene_id</th>
<th style="text-align: left;">external_gene_name</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1017</td>
<td style="text-align: left;">ENSG00000123374</td>
<td style="text-align: left;">CDK2</td>
</tr>
<tr class="even">
<td style="text-align: right;">1018</td>
<td style="text-align: left;">ENSG00000250506</td>
<td style="text-align: left;">CDK3</td>
</tr>
<tr class="odd">
<td style="text-align: right;">1019</td>
<td style="text-align: left;">ENSG00000135446</td>
<td style="text-align: left;">CDK4</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section><section id="使用-ensembl-的存档版本" class="level3" data-number="3.1.2"><h3 data-number="3.1.2" class="anchored" data-anchor-id="使用-ensembl-的存档版本">
<span class="header-section-number">3.1.2</span> <strong>使用 Ensembl 的存档版本</strong>
</h3>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listEnsemblArchives.html">listEnsemblArchives</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listEnsembl.html">listEnsembl</a></span><span class="op">(</span>version <span class="op">=</span> <span class="fl">112</span><span class="op">)</span></span>
<span><span class="va">ensembl_112</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useEnsembl.html">useEnsembl</a></span><span class="op">(</span>biomart <span class="op">=</span> <span class="st">'genes'</span>, </span>
<span>                       dataset <span class="op">=</span> <span class="st">'hsapiens_gene_ensembl'</span>,</span>
<span>                       mirror <span class="op">=</span> <span class="st">"asia"</span>,</span>
<span>                       version <span class="op">=</span> <span class="fl">112</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="使用-ensembl-基因组" class="level3" data-number="3.1.3"><h3 data-number="3.1.3" class="anchored" data-anchor-id="使用-ensembl-基因组">
<span class="header-section-number">3.1.3</span> <strong>使用 Ensembl 基因组</strong>
</h3>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listEnsembl.html">listEnsemblGenomes</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="kable-table">
<table class="caption-top table table-sm table-striped small">
<thead><tr class="header">
<th style="text-align: left;">biomart</th>
<th style="text-align: left;">version</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">protists_mart</td>
<td style="text-align: left;">Ensembl Protists Genes 60</td>
</tr>
<tr class="even">
<td style="text-align: left;">protists_variations</td>
<td style="text-align: left;">Ensembl Protists Variations 60</td>
</tr>
<tr class="odd">
<td style="text-align: left;">fungi_mart</td>
<td style="text-align: left;">Ensembl Fungi Genes 60</td>
</tr>
<tr class="even">
<td style="text-align: left;">fungi_variations</td>
<td style="text-align: left;">Ensembl Fungi Variations 60</td>
</tr>
<tr class="odd">
<td style="text-align: left;">metazoa_mart</td>
<td style="text-align: left;">Ensembl Metazoa Genes 60</td>
</tr>
<tr class="even">
<td style="text-align: left;">metazoa_variations</td>
<td style="text-align: left;">Ensembl Metazoa Variations 60</td>
</tr>
<tr class="odd">
<td style="text-align: left;">plants_mart</td>
<td style="text-align: left;">Ensembl Plants Genes 60</td>
</tr>
<tr class="even">
<td style="text-align: left;">plants_variations</td>
<td style="text-align: left;">Ensembl Plants Variations 60</td>
</tr>
</tbody>
</table>
</div>
</div>
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">ensembl_plants</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useEnsembl.html">useEnsemblGenomes</a></span><span class="op">(</span>biomart <span class="op">=</span> <span class="st">"plants_mart"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listDatasets.html">searchDatasets</a></span><span class="op">(</span><span class="va">ensembl_plants</span>, pattern <span class="op">=</span> <span class="st">"Arabidopsis"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="kable-table">
<table class="caption-top table table-sm table-striped small">
<thead><tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">dataset</th>
<th style="text-align: left;">description</th>
<th style="text-align: left;">version</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">4</td>
<td style="text-align: left;">ahalleri_eg_gene</td>
<td style="text-align: left;">Arabidopsis halleri genes (Ahal2.2)</td>
<td style="text-align: left;">Ahal2.2</td>
</tr>
<tr class="even">
<td style="text-align: left;">6</td>
<td style="text-align: left;">alyrata_eg_gene</td>
<td style="text-align: left;">Arabidopsis lyrata genes (v.1.0)</td>
<td style="text-align: left;">v.1.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">11</td>
<td style="text-align: left;">athaliana_eg_gene</td>
<td style="text-align: left;">Arabidopsis thaliana genes (TAIR10)</td>
<td style="text-align: left;">TAIR10</td>
</tr>
</tbody>
</table>
</div>
</div>
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">ensembl_arabidopsis</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useEnsembl.html">useEnsemblGenomes</a></span><span class="op">(</span>biomart <span class="op">=</span> <span class="st">"plants_mart"</span>, </span>
<span>                                         dataset <span class="op">=</span> <span class="st">"athaliana_eg_gene"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section></section><section id="annotationhub" class="level2" data-number="3.2"><h2 data-number="3.2" class="anchored" data-anchor-id="annotationhub">
<span class="header-section-number">3.2</span> <a href="https://bioconductor.org/packages/release/bioc/html/AnnotationHub.html">AnnotationHub</a>
</h2>
<p><a href="https://www.jingege.wang/2023/12/12/%E5%9F%BA%E4%BA%8Eannotationhub%E6%9E%84%E5%BB%BA%E6%89%80%E9%9C%80orgdb%EF%BC%8C%E7%94%A8%E4%BA%8Eclusterprofiler%E5%AF%8C%E9%9B%86%E5%88%86%E6%9E%90/">基于annotationhub构建所需orgdb，用于clusterprofiler富集分析</a></p>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">conflicts_prefer</span><span class="op">(</span><span class="fu">BiocGenerics</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html">setdiff</a></span><span class="op">)</span></span>
<span><span class="co"># proxy &lt;- httr::use_proxy(Sys.getenv('http_proxy'))</span></span>
<span><span class="co"># httr::set_config(proxy)</span></span>
<span><span class="co"># AnnotationHub::setAnnotationHubOption("PROXY", proxy)</span></span>
<span></span>
<span><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">AnnotationHub</a></span><span class="op">(</span>cache <span class="op">=</span> <span class="st">"D:/AnnotationHub"</span><span class="op">)</span></span>
<span><span class="va">ah</span></span>
<span><span class="co">#&gt; AnnotationHub with 72098 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2024-10-28</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl, BroadInstitute, UCSC, ftp://ftp.ncbi.nlm.nih.gov/g...</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens, Mus musculus, Drosophila melanogaster, Rattus norv...</span></span>
<span><span class="co">#&gt; # $rdataclass: GRanges, TwoBitFile, BigWigFile, EnsDb, Rle, OrgDb, SQLiteFil...</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH5012"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                                             </span></span>
<span><span class="co">#&gt;   AH5012   | Chromosome Band                                   </span></span>
<span><span class="co">#&gt;   AH5013   | STS Markers                                       </span></span>
<span><span class="co">#&gt;   AH5014   | FISH Clones                                       </span></span>
<span><span class="co">#&gt;   AH5015   | Recomb Rate                                       </span></span>
<span><span class="co">#&gt;   AH5016   | ENCODE Pilot                                      </span></span>
<span><span class="co">#&gt;   ...        ...                                               </span></span>
<span><span class="co">#&gt;   AH119504 | Ensembl 113 EnsDb for Xiphophorus maculatus       </span></span>
<span><span class="co">#&gt;   AH119505 | Ensembl 113 EnsDb for Xenopus tropicalis          </span></span>
<span><span class="co">#&gt;   AH119506 | Ensembl 113 EnsDb for Zonotrichia albicollis      </span></span>
<span><span class="co">#&gt;   AH119507 | Ensembl 113 EnsDb for Zalophus californianus      </span></span>
<span><span class="co">#&gt;   AH119508 | Ensembl 113 EnsDb for Zosterops lateralis melanops</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># 当前可用数据的更新时间</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">snapshotDate</a></span><span class="op">(</span><span class="va">ah</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "2024-10-28"</span></span>
<span></span>
<span><span class="co"># 缓存路径</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">hubCache</a></span><span class="op">(</span><span class="va">ah</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "D:/AnnotationHub"</span></span>
<span></span>
<span><span class="co"># https://annotationhub.bioconductor.org</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">hubUrl</a></span><span class="op">(</span><span class="va">ah</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "https://annotationhub.bioconductor.org"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">isLocalHub</a></span><span class="op">(</span><span class="va">ah</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dbconn.html">dbconn</a></span><span class="op">(</span><span class="va">ah</span><span class="op">)</span></span>
<span><span class="co">#&gt; &lt;SQLiteConnection&gt;</span></span>
<span><span class="co">#&gt;   Path: D:\AnnotationHub\39283aec6c3b_annotationhub.sqlite3</span></span>
<span><span class="co">#&gt;   Extensions: TRUE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">ah</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 72098</span></span>
<span><span class="va">ah</span><span class="op">$</span><span class="va">dataprovider</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "UCSC"                                                                                                      </span></span>
<span><span class="co">#&gt;  [2] "Ensembl"                                                                                                   </span></span>
<span><span class="co">#&gt;  [3] "RefNet"                                                                                                    </span></span>
<span><span class="co">#&gt;  [4] "Inparanoid8"                                                                                               </span></span>
<span><span class="co">#&gt;  [5] "NHLBI"                                                                                                     </span></span>
<span><span class="co">#&gt;  [6] "ChEA"                                                                                                      </span></span>
<span><span class="co">#&gt;  [7] "Pazar"                                                                                                     </span></span>
<span><span class="co">#&gt;  [8] "NIH Pathway Interaction Database"                                                                          </span></span>
<span><span class="co">#&gt;  [9] "Haemcode"                                                                                                  </span></span>
<span><span class="co">#&gt; [10] "BroadInstitute"                                                                                            </span></span>
<span><span class="co">#&gt; [11] "PRIDE"                                                                                                     </span></span>
<span><span class="co">#&gt; [12] "Gencode"                                                                                                   </span></span>
<span><span class="co">#&gt; [13] "CRIBI"                                                                                                     </span></span>
<span><span class="co">#&gt; [14] "Genoscope"                                                                                                 </span></span>
<span><span class="co">#&gt; [15] "MISO, VAST-TOOLS, UCSC"                                                                                    </span></span>
<span><span class="co">#&gt; [16] "Stanford"                                                                                                  </span></span>
<span><span class="co">#&gt; [17] "dbSNP"                                                                                                     </span></span>
<span><span class="co">#&gt; [18] "BioMart"                                                                                                   </span></span>
<span><span class="co">#&gt; [19] "GeneOntology"                                                                                              </span></span>
<span><span class="co">#&gt; [20] "KEGG"                                                                                                      </span></span>
<span><span class="co">#&gt; [21] "URGI"                                                                                                      </span></span>
<span><span class="co">#&gt; [22] "EMBL-EBI"                                                                                                  </span></span>
<span><span class="co">#&gt; [23] "MicrosporidiaDB"                                                                                           </span></span>
<span><span class="co">#&gt; [24] "FungiDB"                                                                                                   </span></span>
<span><span class="co">#&gt; [25] "TriTrypDB"                                                                                                 </span></span>
<span><span class="co">#&gt; [26] "ToxoDB"                                                                                                    </span></span>
<span><span class="co">#&gt; [27] "AmoebaDB"                                                                                                  </span></span>
<span><span class="co">#&gt; [28] "PlasmoDB"                                                                                                  </span></span>
<span><span class="co">#&gt; [29] "PiroplasmaDB"                                                                                              </span></span>
<span><span class="co">#&gt; [30] "CryptoDB"                                                                                                  </span></span>
<span><span class="co">#&gt; [31] "TrichDB"                                                                                                   </span></span>
<span><span class="co">#&gt; [32] "GiardiaDB"                                                                                                 </span></span>
<span><span class="co">#&gt; [33] "The Gene Ontology Consortium"                                                                              </span></span>
<span><span class="co">#&gt; [34] "ENCODE Project"                                                                                            </span></span>
<span><span class="co">#&gt; [35] "SchistoDB"                                                                                                 </span></span>
<span><span class="co">#&gt; [36] "NCBI/UniProt"                                                                                              </span></span>
<span><span class="co">#&gt; [37] "GENCODE"                                                                                                   </span></span>
<span><span class="co">#&gt; [38] "http://www.pantherdb.org"                                                                                  </span></span>
<span><span class="co">#&gt; [39] "RMBase v2.0"                                                                                               </span></span>
<span><span class="co">#&gt; [40] "snoRNAdb"                                                                                                  </span></span>
<span><span class="co">#&gt; [41] "tRNAdb"                                                                                                    </span></span>
<span><span class="co">#&gt; [42] "NCBI"                                                                                                      </span></span>
<span><span class="co">#&gt; [43] "DrugAge, DrugBank, Broad Institute"                                                                        </span></span>
<span><span class="co">#&gt; [44] "DrugAge"                                                                                                   </span></span>
<span><span class="co">#&gt; [45] "DrugBank"                                                                                                  </span></span>
<span><span class="co">#&gt; [46] "Broad Institute"                                                                                           </span></span>
<span><span class="co">#&gt; [47] "HMDB, EMBL-EBI, EPA"                                                                                       </span></span>
<span><span class="co">#&gt; [48] "STRING"                                                                                                    </span></span>
<span><span class="co">#&gt; [49] "OMA"                                                                                                       </span></span>
<span><span class="co">#&gt; [50] "OrthoDB"                                                                                                   </span></span>
<span><span class="co">#&gt; [51] "PathBank"                                                                                                  </span></span>
<span><span class="co">#&gt; [52] "EBI/EMBL"                                                                                                  </span></span>
<span><span class="co">#&gt; [53] "NCBI,DBCLS"                                                                                                </span></span>
<span><span class="co">#&gt; [54] "FANTOM5,DLRP,IUPHAR,HPRD,STRING,SWISSPROT,TREMBL,ENSEMBL,CELLPHONEDB,BADERLAB,SINGLECELLSIGNALR,HOMOLOGENE"</span></span>
<span><span class="co">#&gt; [55] "WikiPathways"                                                                                              </span></span>
<span><span class="co">#&gt; [56] "VAST-TOOLS"                                                                                                </span></span>
<span><span class="co">#&gt; [57] "pyGenomeTracks "                                                                                           </span></span>
<span><span class="co">#&gt; [58] "NA"                                                                                                        </span></span>
<span><span class="co">#&gt; [59] "UoE"                                                                                                       </span></span>
<span><span class="co">#&gt; [60] "TargetScan,miRTarBase,USCS,ENSEMBL"                                                                        </span></span>
<span><span class="co">#&gt; [61] "TargetScan"                                                                                                </span></span>
<span><span class="co">#&gt; [62] "QuickGO"                                                                                                   </span></span>
<span><span class="co">#&gt; [63] "CIS-BP"                                                                                                    </span></span>
<span><span class="co">#&gt; [64] "CTCFBSDB 2.0"                                                                                              </span></span>
<span><span class="co">#&gt; [65] "HOCOMOCO v11"                                                                                              </span></span>
<span><span class="co">#&gt; [66] "JASPAR 2022"                                                                                               </span></span>
<span><span class="co">#&gt; [67] "Jolma 2013"                                                                                                </span></span>
<span><span class="co">#&gt; [68] "SwissRegulon"                                                                                              </span></span>
<span><span class="co">#&gt; [69] "ENCODE SCREEN v3"                                                                                          </span></span>
<span><span class="co">#&gt; [70] "MassBank"                                                                                                  </span></span>
<span><span class="co">#&gt; [71] "excluderanges"                                                                                             </span></span>
<span><span class="co">#&gt; [72] "ENCODE"                                                                                                    </span></span>
<span><span class="co">#&gt; [73] "GitHub"                                                                                                    </span></span>
<span><span class="co">#&gt; [74] "Stanford.edu"                                                                                              </span></span>
<span><span class="co">#&gt; [75] "Publication"                                                                                               </span></span>
<span><span class="co">#&gt; [76] "CHM13"                                                                                                     </span></span>
<span><span class="co">#&gt; [77] "UCSChub"                                                                                                   </span></span>
<span><span class="co">#&gt; [78] "Google DeepMind"                                                                                           </span></span>
<span><span class="co">#&gt; [79] "UWashington"                                                                                               </span></span>
<span><span class="co">#&gt; [80] "Bioconductor"                                                                                              </span></span>
<span><span class="co">#&gt; [81] "ENCODE cCREs"                                                                                              </span></span>
<span><span class="co">#&gt; [82] "The Human Phenotype Ontology"                                                                              </span></span>
<span><span class="co">#&gt; [83] "MGI"                                                                                                       </span></span>
<span><span class="co">#&gt; [84] "ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html">unique</a></span><span class="op">(</span><span class="va">ah</span><span class="op">$</span><span class="va">species</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="datatables html-widget html-fill-item" id="htmlwidget-f324e0346f41b95c9e32" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f324e0346f41b95c9e32">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417","418","419","420","421","422","423","424","425","426","427","428","429","430","431","432","433","434","435","436","437","438","439","440","441","442","443","444","445","446","447","448","449","450","451","452","453","454","455","456","457","458","459","460","461","462","463","464","465","466","467","468","469","470","471","472","473","474","475","476","477","478","479","480","481","482","483","484","485","486","487","488","489","490","491","492","493","494","495","496","497","498","499","500","501","502","503","504","505","506","507","508","509","510","511","512","513","514","515","516","517","518","519","520","521","522","523","524","525","526","527","528","529","530","531","532","533","534","535","536","537","538","539","540","541","542","543","544","545","546","547","548","549","550","551","552","553","554","555","556","557","558","559","560","561","562","563","564","565","566","567","568","569","570","571","572","573","574","575","576","577","578","579","580","581","582","583","584","585","586","587","588","589","590","591","592","593","594","595","596","597","598","599","600","601","602","603","604","605","606","607","608","609","610","611","612","613","614","615","616","617","618","619","620","621","622","623","624","625","626","627","628","629","630","631","632","633","634","635","636","637","638","639","640","641","642","643","644","645","646","647","648","649","650","651","652","653","654","655","656","657","658","659","660","661","662","663","664","665","666","667","668","669","670","671","672","673","674","675","676","677","678","679","680","681","682","683","684","685","686","687","688","689","690","691","692","693","694","695","696","697","698","699","700","701","702","703","704","705","706","707","708","709","710","711","712","713","714","715","716","717","718","719","720","721","722","723","724","725","726","727","728","729","730","731","732","733","734","735","736","737","738","739","740","741","742","743","744","745","746","747","748","749","750","751","752","753","754","755","756","757","758","759","760","761","762","763","764","765","766","767","768","769","770","771","772","773","774","775","776","777","778","779","780","781","782","783","784","785","786","787","788","789","790","791","792","793","794","795","796","797","798","799","800","801","802","803","804","805","806","807","808","809","810","811","812","813","814","815","816","817","818","819","820","821","822","823","824","825","826","827","828","829","830","831","832","833","834","835","836","837","838","839","840","841","842","843","844","845","846","847","848","849","850","851","852","853","854","855","856","857","858","859","860","861","862","863","864","865","866","867","868","869","870","871","872","873","874","875","876","877","878","879","880","881","882","883","884","885","886","887","888","889","890","891","892","893","894","895","896","897","898","899","900","901","902","903","904","905","906","907","908","909","910","911","912","913","914","915","916","917","918","919","920","921","922","923","924","925","926","927","928","929","930","931","932","933","934","935","936","937","938","939","940","941","942","943","944","945","946","947","948","949","950","951","952","953","954","955","956","957","958","959","960","961","962","963","964","965","966","967","968","969","970","971","972","973","974","975","976","977","978","979","980","981","982","983","984","985","986","987","988","989","990","991","992","993","994","995","996","997","998","999","1000","1001","1002","1003","1004","1005","1006","1007","1008","1009","1010","1011","1012","1013","1014","1015","1016","1017","1018","1019","1020","1021","1022","1023","1024","1025","1026","1027","1028","1029","1030","1031","1032","1033","1034","1035","1036","1037","1038","1039","1040","1041","1042","1043","1044","1045","1046","1047","1048","1049","1050","1051","1052","1053","1054","1055","1056","1057","1058","1059","1060","1061","1062","1063","1064","1065","1066","1067","1068","1069","1070","1071","1072","1073","1074","1075","1076","1077","1078","1079","1080","1081","1082","1083","1084","1085","1086","1087","1088","1089","1090","1091","1092","1093","1094","1095","1096","1097","1098","1099","1100","1101","1102","1103","1104","1105","1106","1107","1108","1109","1110","1111","1112","1113","1114","1115","1116","1117","1118","1119","1120","1121","1122","1123","1124","1125","1126","1127","1128","1129","1130","1131","1132","1133","1134","1135","1136","1137","1138","1139","1140","1141","1142","1143","1144","1145","1146","1147","1148","1149","1150","1151","1152","1153","1154","1155","1156","1157","1158","1159","1160","1161","1162","1163","1164","1165","1166","1167","1168","1169","1170","1171","1172","1173","1174","1175","1176","1177","1178","1179","1180","1181","1182","1183","1184","1185","1186","1187","1188","1189","1190","1191","1192","1193","1194","1195","1196","1197","1198","1199","1200","1201","1202","1203","1204","1205","1206","1207","1208","1209","1210","1211","1212","1213","1214","1215","1216","1217","1218","1219","1220","1221","1222","1223","1224","1225","1226","1227","1228","1229","1230","1231","1232","1233","1234","1235","1236","1237","1238","1239","1240","1241","1242","1243","1244","1245","1246","1247","1248","1249","1250","1251","1252","1253","1254","1255","1256","1257","1258","1259","1260","1261","1262","1263","1264","1265","1266","1267","1268","1269","1270","1271","1272","1273","1274","1275","1276","1277","1278","1279","1280","1281","1282","1283","1284","1285","1286","1287","1288","1289","1290","1291","1292","1293","1294","1295","1296","1297","1298","1299","1300","1301","1302","1303","1304","1305","1306","1307","1308","1309","1310","1311","1312","1313","1314","1315","1316","1317","1318","1319","1320","1321","1322","1323","1324","1325","1326","1327","1328","1329","1330","1331","1332","1333","1334","1335","1336","1337","1338","1339","1340","1341","1342","1343","1344","1345","1346","1347","1348","1349","1350","1351","1352","1353","1354","1355","1356","1357","1358","1359","1360","1361","1362","1363","1364","1365","1366","1367","1368","1369","1370","1371","1372","1373","1374","1375","1376","1377","1378","1379","1380","1381","1382","1383","1384","1385","1386","1387","1388","1389","1390","1391","1392","1393","1394","1395","1396","1397","1398","1399","1400","1401","1402","1403","1404","1405","1406","1407","1408","1409","1410","1411","1412","1413","1414","1415","1416","1417","1418","1419","1420","1421","1422","1423","1424","1425","1426","1427","1428","1429","1430","1431","1432","1433","1434","1435","1436","1437","1438","1439","1440","1441","1442","1443","1444","1445","1446","1447","1448","1449","1450","1451","1452","1453","1454","1455","1456","1457","1458","1459","1460","1461","1462","1463","1464","1465","1466","1467","1468","1469","1470","1471","1472","1473","1474","1475","1476","1477","1478","1479","1480","1481","1482","1483","1484","1485","1486","1487","1488","1489","1490","1491","1492","1493","1494","1495","1496","1497","1498","1499","1500","1501","1502","1503","1504","1505","1506","1507","1508","1509","1510","1511","1512","1513","1514","1515","1516","1517","1518","1519","1520","1521","1522","1523","1524","1525","1526","1527","1528","1529","1530","1531","1532","1533","1534","1535","1536","1537","1538","1539","1540","1541","1542","1543","1544","1545","1546","1547","1548","1549","1550","1551","1552","1553","1554","1555","1556","1557","1558","1559","1560","1561","1562","1563","1564","1565","1566","1567","1568","1569","1570","1571","1572","1573","1574","1575","1576","1577","1578","1579","1580","1581","1582","1583","1584","1585","1586","1587","1588","1589","1590","1591","1592","1593","1594","1595","1596","1597","1598","1599","1600","1601","1602","1603","1604","1605","1606","1607","1608","1609","1610","1611","1612","1613","1614","1615","1616","1617","1618","1619","1620","1621","1622","1623","1624","1625","1626","1627","1628","1629","1630","1631","1632","1633","1634","1635","1636","1637","1638","1639","1640","1641","1642","1643","1644","1645","1646","1647","1648","1649","1650","1651","1652","1653","1654","1655","1656","1657","1658","1659","1660","1661","1662","1663","1664","1665","1666","1667","1668","1669","1670","1671","1672","1673","1674","1675","1676","1677","1678","1679","1680","1681","1682","1683","1684","1685","1686","1687","1688","1689","1690","1691","1692","1693","1694","1695","1696","1697","1698","1699","1700","1701","1702","1703","1704","1705","1706","1707","1708","1709","1710","1711","1712","1713","1714","1715","1716","1717","1718","1719","1720","1721","1722","1723","1724","1725","1726","1727","1728","1729","1730","1731","1732","1733","1734","1735","1736","1737","1738","1739","1740","1741","1742","1743","1744","1745","1746","1747","1748","1749","1750","1751","1752","1753","1754","1755","1756","1757","1758","1759","1760","1761","1762","1763","1764","1765","1766","1767","1768","1769","1770","1771","1772","1773","1774","1775","1776","1777","1778","1779","1780","1781","1782","1783","1784","1785","1786","1787","1788","1789","1790","1791","1792","1793","1794","1795","1796","1797","1798","1799","1800","1801","1802","1803","1804","1805","1806","1807","1808","1809","1810","1811","1812","1813","1814","1815","1816","1817","1818","1819","1820","1821","1822","1823","1824","1825","1826","1827","1828","1829","1830","1831","1832","1833","1834","1835","1836","1837","1838","1839","1840","1841","1842","1843","1844","1845","1846","1847","1848","1849","1850","1851","1852","1853","1854","1855","1856","1857","1858","1859","1860","1861","1862","1863","1864","1865","1866","1867","1868","1869","1870","1871","1872","1873","1874","1875","1876","1877","1878","1879","1880","1881","1882","1883","1884","1885","1886","1887","1888","1889","1890","1891","1892","1893","1894","1895","1896","1897","1898","1899","1900","1901","1902","1903","1904","1905","1906","1907","1908","1909","1910","1911","1912","1913","1914","1915","1916","1917","1918","1919","1920","1921","1922","1923","1924","1925","1926","1927","1928","1929","1930","1931","1932","1933","1934","1935","1936","1937","1938","1939","1940","1941","1942","1943","1944","1945","1946","1947","1948","1949","1950","1951","1952","1953","1954","1955","1956","1957","1958","1959","1960","1961","1962","1963","1964","1965","1966","1967","1968","1969","1970","1971","1972","1973","1974","1975","1976","1977","1978","1979","1980","1981","1982","1983","1984","1985","1986","1987","1988","1989","1990","1991","1992","1993","1994","1995","1996","1997","1998","1999","2000","2001","2002","2003","2004","2005","2006","2007","2008","2009","2010","2011","2012","2013","2014","2015","2016","2017","2018","2019","2020","2021","2022","2023","2024","2025","2026","2027","2028","2029","2030","2031","2032","2033","2034","2035","2036","2037","2038","2039","2040","2041","2042","2043","2044","2045","2046","2047","2048","2049","2050","2051","2052","2053","2054","2055","2056","2057","2058","2059","2060","2061","2062","2063","2064","2065","2066","2067","2068","2069","2070","2071","2072","2073","2074","2075","2076","2077","2078","2079","2080","2081","2082","2083","2084","2085","2086","2087","2088","2089","2090","2091","2092","2093","2094","2095","2096","2097","2098","2099","2100","2101","2102","2103","2104","2105","2106","2107","2108","2109","2110","2111","2112","2113","2114","2115","2116","2117","2118","2119","2120","2121","2122","2123","2124","2125","2126","2127","2128","2129","2130","2131","2132","2133","2134","2135","2136","2137","2138","2139","2140","2141","2142","2143","2144","2145","2146","2147","2148","2149","2150","2151","2152","2153","2154","2155","2156","2157","2158","2159","2160","2161","2162","2163","2164","2165","2166","2167","2168","2169","2170","2171","2172","2173","2174","2175","2176","2177","2178","2179","2180","2181","2182","2183","2184","2185","2186","2187","2188","2189","2190","2191","2192","2193","2194","2195","2196","2197","2198","2199","2200","2201","2202","2203","2204","2205","2206","2207","2208","2209","2210","2211","2212","2213","2214","2215","2216","2217","2218","2219","2220","2221","2222","2223","2224","2225","2226","2227","2228","2229","2230","2231","2232","2233","2234","2235","2236","2237","2238","2239","2240","2241","2242","2243","2244","2245","2246","2247","2248","2249","2250","2251","2252","2253","2254","2255","2256","2257","2258","2259","2260","2261","2262","2263","2264","2265","2266","2267","2268","2269","2270","2271","2272","2273","2274","2275","2276","2277","2278","2279","2280","2281","2282","2283","2284","2285","2286","2287","2288","2289","2290","2291","2292","2293","2294","2295","2296","2297","2298","2299","2300","2301","2302","2303","2304","2305","2306","2307","2308","2309","2310","2311","2312","2313","2314","2315","2316","2317","2318","2319","2320","2321","2322","2323","2324","2325","2326","2327","2328","2329","2330","2331","2332","2333","2334","2335","2336","2337","2338","2339","2340","2341","2342","2343","2344","2345","2346","2347","2348","2349","2350","2351","2352","2353","2354","2355","2356","2357","2358","2359","2360","2361","2362","2363","2364","2365","2366","2367","2368","2369","2370","2371","2372","2373","2374","2375","2376","2377","2378","2379","2380","2381","2382","2383","2384","2385","2386","2387","2388","2389","2390","2391","2392","2393","2394","2395","2396","2397","2398","2399","2400","2401","2402","2403","2404","2405","2406","2407","2408","2409","2410","2411","2412","2413","2414","2415","2416","2417","2418","2419","2420","2421","2422","2423","2424","2425","2426","2427","2428","2429","2430","2431","2432","2433","2434","2435","2436","2437","2438","2439","2440","2441","2442","2443","2444","2445","2446","2447","2448","2449","2450","2451","2452","2453","2454","2455","2456","2457","2458","2459","2460","2461","2462","2463","2464","2465","2466","2467","2468","2469","2470","2471","2472","2473","2474","2475","2476","2477","2478","2479","2480","2481","2482","2483","2484","2485","2486","2487","2488","2489","2490","2491","2492","2493","2494","2495","2496","2497","2498","2499","2500","2501","2502","2503","2504","2505","2506","2507","2508","2509","2510","2511","2512","2513","2514","2515","2516","2517","2518","2519","2520","2521","2522","2523","2524","2525","2526","2527","2528","2529","2530","2531","2532","2533","2534","2535","2536","2537","2538","2539","2540","2541","2542","2543","2544","2545","2546","2547","2548","2549","2550","2551","2552","2553","2554","2555","2556","2557","2558","2559","2560","2561","2562","2563","2564","2565","2566","2567","2568","2569","2570","2571","2572","2573","2574","2575","2576","2577","2578","2579","2580","2581","2582","2583","2584","2585","2586","2587","2588","2589","2590","2591","2592","2593","2594","2595","2596","2597","2598","2599","2600","2601","2602","2603","2604","2605","2606","2607","2608","2609","2610","2611","2612","2613","2614","2615","2616","2617","2618","2619","2620","2621","2622","2623","2624","2625","2626","2627","2628","2629","2630","2631","2632","2633","2634","2635","2636","2637","2638","2639","2640","2641","2642","2643","2644","2645","2646","2647","2648","2649","2650","2651","2652","2653","2654","2655","2656","2657","2658","2659","2660","2661","2662","2663","2664","2665","2666","2667","2668","2669","2670","2671","2672","2673","2674","2675","2676","2677","2678","2679","2680","2681","2682","2683","2684","2685","2686","2687","2688","2689","2690","2691","2692","2693","2694","2695","2696","2697","2698","2699","2700","2701","2702","2703","2704","2705","2706","2707","2708","2709","2710","2711","2712","2713","2714","2715","2716","2717","2718","2719","2720","2721","2722","2723","2724","2725","2726","2727","2728","2729","2730","2731","2732","2733","2734","2735","2736","2737","2738","2739","2740","2741","2742","2743","2744","2745","2746","2747","2748","2749","2750","2751","2752","2753","2754","2755","2756","2757","2758","2759","2760","2761","2762","2763","2764","2765","2766","2767","2768","2769","2770","2771","2772","2773","2774","2775","2776","2777","2778","2779","2780","2781","2782","2783","2784","2785","2786","2787","2788","2789","2790","2791","2792","2793","2794","2795","2796","2797","2798","2799","2800","2801","2802","2803","2804","2805","2806","2807","2808","2809","2810","2811","2812","2813","2814","2815","2816","2817","2818","2819","2820","2821","2822","2823","2824","2825","2826","2827","2828","2829","2830","2831","2832","2833","2834","2835","2836","2837","2838","2839","2840","2841","2842","2843","2844","2845","2846","2847","2848","2849","2850","2851","2852","2853","2854","2855","2856","2857","2858","2859","2860","2861","2862","2863","2864","2865","2866","2867","2868","2869","2870","2871","2872","2873","2874","2875","2876","2877","2878","2879","2880","2881","2882","2883","2884","2885","2886","2887","2888","2889","2890","2891","2892","2893","2894","2895","2896","2897","2898","2899","2900","2901","2902","2903","2904","2905","2906","2907","2908","2909","2910","2911","2912","2913","2914","2915","2916","2917","2918","2919","2920","2921","2922","2923","2924","2925","2926","2927","2928","2929","2930","2931","2932","2933","2934","2935","2936","2937","2938","2939","2940","2941","2942","2943","2944","2945","2946","2947","2948","2949","2950","2951","2952","2953","2954","2955","2956","2957","2958","2959","2960","2961","2962","2963","2964","2965","2966","2967","2968","2969","2970","2971","2972","2973","2974","2975","2976","2977","2978","2979","2980","2981","2982","2983","2984","2985","2986","2987","2988","2989","2990","2991","2992","2993","2994"],["Homo sapiens","Vicugna pacos","Dasypus novemcinctus","Otolemur garnettii","Papio hamadryas","Papio anubis","Felis catus","Pan troglodytes","Bos taurus","Canis familiaris","Tursiops truncatus","Loxodonta africana","Nomascus leucogenys","Gorilla gorilla","Cavia porcellus","Equus caballus","Dipodomys ordii","Trichechus manatus","Callithrix jacchus","Pteropus vampyrus","Myotis lucifugus","Mus musculus","Microcebus murinus","Heterocephalus glaber","Monodelphis domestica","Pongo abelii","Ailuropoda melanoleuca","Sus scrofa","Ochotona princeps","Ornithorhynchus anatinus","Oryctolagus cuniculus","Rattus norvegicus","Macaca mulatta","Procavia capensis","Ovis aries","Sorex araneus","Petromyzon marinus","Anolis carolinensis","Oryzias latipes","Geospiza fortis","Oreochromis niloticus","Chrysemys picta","Gasterosteus aculeatus","Tetraodon nigroviridis","Meleagris gallopavo","Xenopus tropicalis","Taeniopygia guttata","Danio rerio","Ciona intestinalis","Branchiostoma floridae","Strongylocentrotus purpuratus","Apis mellifera","Anopheles gambiae","Drosophila ananassae","Drosophila erecta","Drosophila grimshawi","Drosophila melanogaster","Drosophila mojavensis","Drosophila persimilis","Drosophila pseudoobscura","Drosophila sechellia","Drosophila simulans","Drosophila virilis","Drosophila yakuba","Caenorhabditis brenneri","Caenorhabditis briggsae","Caenorhabditis elegans","Caenorhabditis japonica","Caenorhabditis remanei","Pristionchus pacificus","Aplysia californica","Saccharomyces cerevisiae","Spermophilus tridecemlineatus","Saimiri sciureus","Sarcophilus harrisii","Echinops telfairi","Tupaia belangeri","Macropus eugenii","Ceratotherium simum","Melopsittacus undulatus","Gallus gallus","Latimeria chalumnae","Fugu rubripes","Takifugu rubripes","Tarsius syrichta","Xiphophorus maculatus","Choloepus hoffmanni","Ciona savignyi","Erinaceus europaeus","Gadus morhua","Ictidomys tridecemlineatus","Mustela putorius furo","Pelodiscus sinensis","Anas platyrhynchos","Ficedula albicollis","Astyanax mexicanus","Lepisosteus oculatus","Aedes aegypti","Aquifex aeolicus VF5","Aureococcus anophagefferens","Agaricus bisporus var. burnettii JB137-S8","Ajellomyces capsulatus G186AR","Atta cephalotes","Anopheles darlingi","Auricularia delicata","Acromyrmex echinatior","Ashbya gossypii ATCC 10895","Arthroderma gypseum","Aspergillus kawachii IFO 4308","Brugia malayi","Bombyx mori","Brassica rapa","Bacillus subtilis","Candida albicans","Coprinopsis cinerea okayama7#130","Camponotus floridanus","Crassostrea gigas","[Candida] glabrata CBS 138","Chaetomium globosum CBS 148.51","Cricetulus griseus","Cryptosporidium hominis","Coccidioides immitis","Debaryomyces hansenii","Danaus plexippus","Daphnia pulex","Dictyostelium purpureum","Laccaria bicolor S238N-H82","Leishmania infantum","Leptospira interrogans","Loa loa","Leptosphaeria maculans","Leishmania major","Lachancea thermotolerans","Methanosarcina acetivorans","Metarhizium acridum","Monosiga brevicollis","Marssonina brunnea","Puccinia graminis","Pediculus humanus","Piriformospora indica","Plasmodium knowlesi","Perkinsus marinus","Parastagonospora nodorum SN15","Polysphondylium pallidum","Physcomitrella patens","Postia placenta","Phytophthora ramorum","Phytophthora sojae","Pichia sorbitophila","Paramecium tetraurelia","Populus trichocarpa","Toxoplasma gondii","Thermotoga maritima","Tuber melanosporum","Theileria parva","Thalassiosira pseudonana","Trichophyton rubrum","Trichinella spiralis","Talaromyces stipitatus","Tetrahymena thermophila","Trichomonas vaginalis","Thermodesulfovibrio yellowstonii","Ustilago maydis","Uncinocarpus reesii","Volvox carteri","Arthrobotrys oligospora","Acyrthosiphon pisum","Amphimedon queenslandica","Arabidopsis thaliana","Beauveria bassiana","Babesia bovis","Batrachochytrium dendrobatidis","Brachypodium distachyon","Botryotinia fuckeliana","Blastocystis hominis","Bradyrhizobium japonicum","Bacteroides thetaiotaomicron","Chloroflexus aurantiacus","Colletotrichum gloeosporioides","Clavispora lusitaniae","Cordyceps militaris","Cryptococcus neoformans","Capsaspora owczarzaki","Cryptosporidium parvum","Culex quinquefasciatus","Chlamydomonas reinhardtii","Clonorchis sinensis","Chlamydia trachomatis","Chlorella variabilis","Dictyostelium discoideum","Deinococcus radiodurans","Dictyoglomus turgidum","Drosophila willistoni","Edhazardia aedis","Enterocytozoon bieneusi","Escherichia coli","Encephalitozoon cuniculi","Eremothecium cymbalariae","Exophiala dermatitidis","Entamoeba histolytica","Emericella nidulans FGSC A4","Ectocarpus siliculosus","Fusobacterium nucleatum","Fusarium pseudograminearum","Fibroporia radiculosa","Grosmannia clavigera","Pseudogymnoascus destructans","Gaeumannomyces graminis","Giardia intestinalis","Glarea lozoyensis","Glycine max","Geobacter sulfurreducens","Gloeobacter violaceus","Gibberella zeae","Hyaloperonospora arabidopsidis","Halobacterium salinarum","Harpegnathos saltator","Hypocrea virens","Hordeum vulgare","Ichthyophthirius multifiliis","Ixodes scapularis","Kazachstania africana","Korarchaeum cryptofilum","Kluyveromyces lactis","Pichia pastoris","Leishmania braziliensis","Lodderomyces elongisporus","Malassezia globosa","Mycosphaerella graminicola IPO323","Meyerozyma guilliermondii","Methanocaldococcus jannaschii","Magnaporthe oryzae","Mixia osmundae","Moniliophthora perniciosa","Macrophomina phaseolina","Mycobacterium tuberculosis","Micromonas sp.","Neospora caninum","Naumovozyma castellii","Nosema ceranae","Neurospora crassa","Aspergillus fumigatus Af293","Naegleria gruberi","[Nectria] haematococca mpVI 77-13-4","Nematocida parisii","Nematostella vectensis","Nasonia vitripennis","Oikopleura dioica","Oryza sativa","Ostreococcus tauri","Pyrobaculum aerophilum","Pseudomonas aeruginosa","Pteropus alecto","Plasmodium berghei","Paracoccidioides brasiliensis","Phanerochaete carnosa","Plasmodium chabaudi","Penicillium digitatum","Plasmodium falciparum","Phytophthora infestans","Pneumocystis jirovecii","Pyrococcus kodakaraensis","Phaeodactylum tricornutum","Pythium ultimum","Plasmodium vivax","Rhodopirellula baltica","Ricinus communis","Rhizopus delemar","Rhodotorula glutinis","Sorghum bicolor","Streptomyces coelicolor","Schizophyllum commune","Solenopsis invicta","Setaria italica","Serpula lacrymans","Solanum lycopersicum","Sordaria macrospora","Schistosoma mansoni","Selaginella moellendorffii","Spathaspora passalidarum","Schizosaccharomyces pombe","Sporisorium reilianum","Sclerotinia sclerotiorum","Sulfolobus solfataricus","Scheffersomyces stipitis","Solanum tuberosum","Salpingoeca sp.","Synechocystis sp.","Trichoplax adhaerens","Theileria annulata","Trichosporon asahii","Tetrapisispora blattae","Tribolium castaneum","Tupaia chinensis","Trypanosoma cruzi","Torulaspora delbrueckii","Myceliophthora thermophila","Trachipleistophora hominis","Vittaforma corneae","Vavraia culicis","Verticillium dahliae","Vanderwaltozyma polyspora","Vitis vinifera","Wuchereria bancrofti","Wickerhamomyces ciferrii","Wallemia sebi","Yarrowia lipolytica","Drosophila pseudoobscura pseudoobscura","Gorilla gorilla gorilla","Canis lupus familiaris","Trichechus manatus latirostris","Saimiri boliviensis","Alligator mississippiensis","Chrysemys picta bellii",null,"Chlorocebus sabaeus","Poecilia formosa","Erwinia carotovora","mus musculus 129s1svimj","mus musculus aj","mus musculus akrj","mus musculus balbcj","mus musculus c3hhej","mus musculus c57bl6nj","mus musculus casteij","mus musculus cbaj","mus musculus dba2j","mus musculus fvbnj","mus musculus lpj","mus musculus nodshiltj","mus musculus nzohlltj","mus musculus pwkphj","mus musculus wsbeij","mus spretus spreteij","mustela putorius furo","Carlito syrichta","Notamacropus eugenii","Cavia aperea","Chinchilla lanigera","cricetulus griseus chok1gshd","cricetulus griseus crigri","Fukomys damarensis","heterocephalus glaber female","heterocephalus glaber male","Jaculus jaculus","Mesocricetus auratus","Microtus ochrogaster","Mus caroli","Mus pahari","Nannospalax galili","Octodon degus","peromyscus maniculatus bairdii","Aotus nancymaae","Cebus capucinus","Cercocebus atys","colobus angolensis palliatus","Macaca fascicularis","Macaca nemestrina","Mandrillus leucophaeus","Pan paniscus","Propithecus coquereli","Rhinopithecus bieti","Rhinopithecus roxellana","saimiri boliviensis boliviensis","Capra hircus","Mus spretus","Eptatretus burgeri","Panthera pardus","panthera tigris altaica","Acanthochromis polyacanthus","Amphilophus citrinellus","Amphiprion ocellaris","Amphiprion percula","Anabas testudineus","Astatotilapia calliptera","Cynoglossus semilaevis","Cyprinodon variegatus","Esox lucius","Fundulus heteroclitus","Gambusia affinis","Haplochromis burtoni","Hippocampus comes","Ictalurus punctatus","Kryptolebias marmoratus","Labrus bergylta","Mastacembelus armatus","Maylandia zebra","Mola mola","Monopterus albus","Neolamprologus brichardi","oryzias latipes hni","oryzias latipes hsok","Oryzias melastigma","Paramormyrops kingsleyae","Periophthalmus magnuspinnatus","Poecilia latipinna","Poecilia mexicana","Poecilia reticulata","Pundamilia nyererei","Pygocentrus nattereri","Scleropages formosus","Scophthalmus maximus","Seriola dumerili","seriola lalandi dorsalis","Stegastes partitus","Xiphophorus couchianus","Cebus capucinus imitator","Aspergillus versicolor","Aspergillus sydowii","Phytophthora cinnamomi","Trypanosoma vivax","Trypanosoma congolense","Aspergillus wentii","Aspergillus zonatus","Entamoeba nuttalli","Fusarium oxysporum","Plasmodium cynomolgi","Babesia microti","Aspergillus luchuensis","Saprolegnia diclina","Aspergillus glaucus","Chromera velia","Vitrella brassicaformis","Encephalitozoon romaleae","Leishmania aethiopica","Leishmania tropica","Pythium aphanidermatum","Pythium arrhenomanes","Pythium irregulare","Pythium iwayamai","Pythium vexans","Plasmodium inui","Anncaliia algerae","Acanthamoeba castellanii","Fusarium fujikuroi","Leishmania panamensis","Cryptococcus gattii","Leishmania sp.","Ordospora colligata","Spraguea lophii","Aspergillus campestris","Aspergillus steynii","Aspergillus novofumigatus","Aspergillus ochraceoroseus","Sporothrix schenckii","Theileria equi","Trypanosoma brucei","Coccidioides posadasii","Aspergillus nidulans","Fusarium graminearum","Phanerochaete chrysosporium","Cryptococcus deuterogattii","Aspergillus fumigatus","Aspergillus fischeri","Botrytis cinerea","Aspergillus flavus","Fusarium verticillioides","Zymoseptoria tritici","Histoplasma capsulatum","Aspergillus terreus","Aspergillus clavatus","Plasmodium yoelii","Entamoeba dispar","Entamoeba invadens","Aspergillus niger","Schizosaccharomyces japonicus","Eimeria tenella","Trypanosoma rangeli","Trichoderma reesei","Cryptosporidium muris","Talaromyces marneffei","Schizosaccharomyces octosporus","Penicillium rubens","Paracoccidioides lutzii","Aspergillus oryzae","Neurospora tetrasperma","Neurospora discreta","Tremella mesenterica","Nosema bombycis","Allomyces macrogynus","Aspergillus carbonarius","Spizellomyces punctatus","Aspergillus aculeatus","Saprolegnia parasitica","Melampsora larici-populina","Mucor circinelloides","Phytophthora parasitica","Phycomyces blakesleeanus","Phytophthora capsici","Aspergillus brasiliensis","Aspergillus tubingensis","Theileria orientalis","Encephalitozoon intestinalis","Albugo laibachii","Encephalitozoon hellem","Albugo candida","Leishmania mexicana","Leishmania donovani","Aphanomyces astaci","Aphanomyces invadans","Babesia bigemina","Babesia ovata","Cryptosporidium andersoni","Cyclospora cayetanensis","Crithidia fasciculata","Cytauxzoon felis","Cryptosporidium meleagridis","Cystoisospora suis","Cryptosporidium tyzzeri","Cryptosporidium ubiquitum","Eimeria acervulina","Eimeria brunetti","Enterospora canceri","Eimeria falciformis","Enterocytozoon hepatopenaei","Eimeria maxima","Eimeria mitis","Endotrypanum monterogeii","Entamoeba moshkovskii","Eimeria necatrix","Eimeria praecox","Gregarina niphandrodes","Hepatospora eriocheir","Hammondia hammondi","Leishmania arabica","Leishmania enriettii","Leishmania gerbilli","Leptomonas pyrrhocoris","Leptomonas seymouri","Leishmania tarentolae","Leishmania turanica","Mitosporidium daphniae","Nematocida ausubeli","Nematocida displodere","Naegleria fowleri","Plasmodium billcollinsi","Plasmodium coatneyi","Plasmodium fragile","Plasmodium gaboni","Plasmodium gallinaceum","Plasmodium malariae","Pseudoloma neurophilia","Plasmodium ovale","Plasmodium reichenowi","Plasmodium relictum","Plasmodium vinckei","Sarcocystis neurona","Spironucleus salmonicida","Trypanosoma evansi","Trypanosoma grayi","Gopherus agassizii","Phascolarctos cinereus","Sphenodon punctatus","Ursus americanus","Ursus maritimus","Vulpes vulpes","canis lupus dingo","chrysemys picta bellii","equus asinus asinus","Anser brachyrhynchus","Apteryx haastii","Apteryx owenii","Apteryx rowi","Bos mutus","Chelonoidis abingdonii","Cyanistes caeruleus","Castor canadensis","Coturnix japonica","Crocodylus porosus","Calidris pugnax","Calidris pygmaea","Dromaius novaehollandiae","Junco hyemalis","Lepidothrix coronata","Mus spicilegus","Meriones unguiculatus","Manacus vitellinus","Numida meleagris","Nothoprocta perdicaria","Notechis scutatus","Neovison vison","Parus major","Prolemur simus","Piliocolobus tephrosceles","Pogona vitticeps","Serinus canaria","Spermophilus dauricus","Salvator merianae","Theropithecus gelada","Urocitellus parryii","Zonotrichia albicollis","anas platyrhynchos platyrhynchos","bison bison bison","cricetulus griseus picr","lonchura striata domestica","marmota marmota marmota","Callorhinchus milii","Electrophorus electricus","Hucho hucho","Lates calcarifer","Vombatus ursinus","Acanthamoeba castellanii str. Neff","Entamoeba dispar SAW760","Entamoeba histolytica HM-1:IMSS","Entamoeba histolytica HM-1:IMSS-A","Entamoeba histolytica HM-1:IMSS-B","Entamoeba histolytica HM-3:IMSS","Entamoeba histolytica KU27","Entamoeba invadens IP1","Entamoeba nuttalli P19","Chromera velia CCMP2878","Cryptosporidium hominis TU502","Cryptosporidium muris RN66","Cryptosporidium parvum Iowa II","Vitrella brassicaformis CCMP3155","Albugo candida 2VRR","Albugo laibachii Nc14","Allomyces macrogynus ATCC 38327","Aspergillus aculeatus ATCC 16872","Aspergillus brasiliensis CBS 101740","Aspergillus campestris IBT 28561","Aspergillus carbonarius ITEM 5010","Aspergillus clavatus NRRL 1","Aspergillus fischeri NRRL 181","Aspergillus flavus NRRL3357","Aspergillus fumigatus A1163","Aspergillus glaucus CBS 516.65","Aspergillus luchuensis CBS 106.47","Aspergillus nidulans FGSC A4","Aspergillus niger ATCC 1015","Aspergillus niger CBS 513.88","Aspergillus novofumigatus IBT 16806","Aspergillus ochraceoroseus IBT 24754","Aspergillus oryzae RIB40","Aspergillus steynii IBT 23096","Aspergillus sydowii CBS 593.65","Aspergillus terreus NIH2624","Aspergillus tubingensis CBS 134.48","Aspergillus versicolor CBS 583.65","Aspergillus wentii DTO 134E9","Batrachochytrium dendrobatidis JEL423","Botrytis cinerea B05.10","Candida albicans SC5314","Candida albicans WO-1","Candida glabrata CBS 138","Candida parapsilosis CDC317","Clavispora lusitaniae ATCC 42720","Coccidioides immitis H538.4","Coccidioides immitis RS","Coccidioides posadasii C735 delta SOWgp","Coccidioides posadasii RMSCC 3488","Coccidioides posadasii str. Silveira","Cryptococcus gattii CA1873","Cryptococcus gattii EJB2","Cryptococcus gattii NT-10","Cryptococcus gattii VGIV IND107","Cryptococcus gattii WM276","Cryptococcus neoformans var. grubii H99","Cryptococcus neoformans var. neoformans B-3501A","Cryptococcus neoformans var. neoformans JEC21","Fusarium fujikuroi IMI 58289","Fusarium graminearum PH-1","Fusarium oxysporum Fo47","Fusarium oxysporum f. sp. cubense race 1","Fusarium oxysporum f. sp. cubense race 4","Fusarium oxysporum f. sp. cubense tropical race 4 54006","Fusarium oxysporum f. sp. lycopersici 4287","Fusarium oxysporum f. sp. melonis 26406","Fusarium proliferatum ET1","Fusarium verticillioides 7600","Histoplasma capsulatum G186AR","Histoplasma capsulatum G217B","Histoplasma capsulatum H143","Histoplasma capsulatum H88","Histoplasma capsulatum NAm1","Hyaloperonospora arabidopsidis Emoy2","Kwoniella bestiolae CBS 10118","Kwoniella dejecticola CBS 10117","Malassezia globosa CBS 7966","Melampsora larici-populina 98AG31","Mucor circinelloides f. lusitanicus CBS 277.49","Neurospora crassa OR74A","Neurospora discreta FGSC 8579","Neurospora tetrasperma FGSC 2508","Paracoccidioides brasiliensis Pb03","Paracoccidioides brasiliensis Pb18","Paracoccidioides lutzii Pb01","Penicillium rubens Wisconsin 54-1255","Phanerochaete chrysosporium RP-78","Phycomyces blakesleeanus NRRL 1555(-)","Phytophthora capsici LT1534","Phytophthora cinnamomi var. cinnamomi CBS 144.22","Phytophthora infestans T30-4","Phytophthora parasitica INRA-310","Pneumocystis jirovecii SE8","Puccinia graminis f. sp. tritici CRL 75-36-700-3","Pythium aphanidermatum DAOM BR444","Pythium arrhenomanes ATCC 12531","Pythium irregulare DAOM BR486","Pythium iwayamai DAOM BR242034","Rhizopus delemar RA 99-880","Saprolegnia diclina VS20","Saprolegnia parasitica CBS 223.65","Schizosaccharomyces japonicus yFS275","Schizosaccharomyces octosporus yFS286","Schizosaccharomyces pombe 972h-","Sclerotinia sclerotiorum 1980 UF-70","Sordaria macrospora k-hell","Spizellomyces punctatus DAOM BR117","Sporisorium reilianum SRZ2","Sporothrix brasiliensis 5110","Sporothrix schenckii 1099-18","Talaromyces marneffei ATCC 18224","Talaromyces stipitatus ATCC 10500","Tremella mesenterica DSM 1558","Trichoderma reesei QM6a","Uncinocarpus reesii 1704","Ustilago maydis 521","Yarrowia lipolytica CLIB122","Zymoseptoria tritici IPO323","Penicilliopsis zonata CBS 506.65","Cryptococcus gattii VGII R265","Fusarium sp. FOSC 3-a","Fusarium proliferatum","Lomentospora prolificans","Pyricularia oryzae 70-15","Globisporangium ultimum DAOM BR144","Globisporangium ultimum var. sporangiiferum BR650","Phytopythium vexans DAOM BR484","Saccharomyces cerevisiae S288C","Scedosporium apiospermum","Giardia intestinalis ATCC 50803","Giardia intestinalis ATCC 50581","Giardia lamblia P15","Amphiamblys sp. WSBS2006","Anncaliia algerae PRA109","Anncaliia algerae PRA339","Edhazardia aedis USNM 41457","Encephalitozoon cuniculi EC1","Encephalitozoon cuniculi EC2","Encephalitozoon cuniculi EC3","Encephalitozoon cuniculi EcunIII-L","Encephalitozoon cuniculi GB-M1","Encephalitozoon hellem ATCC 50504","Encephalitozoon intestinalis ATCC 50506","Encephalitozoon romaleae SJ-2008","Enterocytozoon bieneusi H348","Nematocida parisii ERTm1","Nematocida parisii ERTm3","Nosema bombycis CQ1","Nosema ceranae BRL01","Ordospora colligata OC4","Spraguea lophii 42_110","Vavraia culicis subsp. floridensis","Vittaforma corneae ATCC 50505","Babesia bovis T2Bo","Babesia microti strain RI","Theileria annulata strain Ankara","Theileria equi strain WA","Theileria orientalis strain Shintoku","Theileria parva strain Muguga","Babesia divergens","Plasmodium berghei ANKA","Plasmodium chabaudi chabaudi","Plasmodium cynomolgi strain B","Plasmodium cynomolgi strain M","Plasmodium falciparum 3D7","Plasmodium falciparum 7G8","Plasmodium falciparum Dd2","Plasmodium falciparum HB3","Plasmodium inui San Antonio 1","Plasmodium knowlesi strain H","Plasmodium vivax Sal-1","Plasmodium yoelii yoelii 17XNL","Schistosoma haematobium","Eimeria tenella strain Houghton","Neospora caninum Liverpool","Toxoplasma gondii ARI","Toxoplasma gondii FOU","Toxoplasma gondii GAB2-2007-GAL-DOM2","Toxoplasma gondii GT1","Toxoplasma gondii MAS","Toxoplasma gondii ME49","Toxoplasma gondii RH","Toxoplasma gondii RUB","Toxoplasma gondii TgCatPRC2","Toxoplasma gondii VAND","Toxoplasma gondii VEG","Toxoplasma gondii p89","Trichomonas vaginalis G3","Leishmania aethiopica L147","Leishmania braziliensis MHOM/BR/75/M2903","Leishmania braziliensis MHOM/BR/75/M2904","Leishmania donovani BPK282A1","Leishmania infantum JPCM5","Leishmania major strain Friedlin","Leishmania major strain LV39c5","Leishmania major strain SD 75.1","Leishmania mexicana MHOM/GT/2001/U1103","Leishmania panamensis MHOM/COL/81/L13","Leishmania sp. MAR LEM2494","Leishmania tropica L590","Trypanosoma brucei brucei TREU927","Trypanosoma brucei gambiense DAL972","Trypanosoma congolense IL3000","Trypanosoma cruzi strain CL Brener","Trypanosoma rangeli SC58","Trypanosoma vivax Y486","Blechomonas ayalai","Bodo saltans","Leishmania amazonensis","Paratrypanosoma confusum","Trypanosoma cruzi Dm28c","Trypanosoma theileri","Betta splendens","Cottoperca gobio","Clupea harengus","Denticeps clupeoides","Erpetoichthys calabaricus","Gouania willdenowi","Larimichthys crocea","Parambassis ranga","Myxococcus xanthus DK 1622","astyanax mexicanus pachon","bos indicus hybrid","bos taurus hybrid","sus scrofa bamei","sus scrofa berkshire","sus scrofa hampshire","sus scrofa jinhua","sus scrofa landrace","sus scrofa largewhite","sus scrofa meishan","sus scrofa pietrain","sus scrofa rongchang","sus scrofa tibetan","sus scrofa usmarc","sus scrofa wuzhishan","Amazona collaria","Athene cunicularia","Anser cygnoides","Accipiter nisus","Bos grunniens","Camelus dromedarius","Chrysolophus pictus","Catagonus wagneri","Erythrura gouldiae","Echeneis naucrates","Lynx canadensis","Moschus moschiferus","Myripristis murdjan","Neogobius melanostomus","Oreochromis aureus","Physeter catodon","Phasianus colchicus","Pavo cristatus","Podarcis muralis","Pseudonaja textilis","Rhinolophus ferrumequinum","Sinocyclocheilus anshuiensis","Sparus aurata","Salarias fasciatus","Sinocyclocheilus grahami","Strigops habroptila","Sphaeramia orbicularis","Sinocyclocheilus rhinocerous","Salmo salar","Suricata suricatta","Salmo trutta","Varanus komodoensis","aquila chrysaetos chrysaetos","canis lupus familiarisbasenji","canis lupus familiarisgreatdane","cyprinus carpio germanmirror","cyprinus carpio hebaored","cyprinus carpio huanghe","struthio camelus australis","terrapene carolina triunguis","Carassius auratus","Cyprinus carpio","Camarhynchus parvulus","Cyanoderma ruficeps","Chelydra serpentina","Dicentrarchus labrax","Delphinapterus leucas","Gopherus evgoodei","Laticauda laticaudata","Oncorhynchus mykiss","Oncorhynchus tshawytscha","Pelusios castaneus","Panthera leo","canis lupus familiaris","Oryzias javanicus","Oryzias sinensis","Stachyris ruficeps","ursus thibetanus thibetanus","zosterops lateralis melanops","Anas zonorhyncha","Bubo bubo","Buteo japonicus","Balaenoptera musculus","Cyclopterus lumpus","Corvus moneduloides","Catharus ustulatus","Falco tinnunculus","Leptobrachium leishanense","Monodon monoceros","Nothobranchius furzeri","Naja naja","Oncorhynchus kisutch","Otus sunia","Phocoena sinus","Sander lucioperca","Sciurus vulgaris","Zalophus californianus","acanthochromis polyacanthus","accipiter nisus","ailuropoda melanoleuca","amazona collaria","amphilophus citrinellus","amphiprion ocellaris","amphiprion percula","anabas testudineus","anas platyrhynchos","anas platyrhynchos_platyrhynchos","anas zonorhyncha","anolis carolinensis","anser brachyrhynchus","anser cygnoides","aotus nancymaae","apteryx haastii","apteryx owenii","apteryx rowi","aquila chrysaetos_chrysaetos","astatotilapia calliptera","astyanax mexicanus","astyanax mexicanus_pachon","athene cunicularia","balaenoptera musculus","betta splendens","bison bison_bison","bos grunniens","bos indicus_hybrid","bos mutus","bos taurus","bos taurus_hybrid","bubo bubo","buteo japonicus","caenorhabditis elegans","cairina moschata_domestica","calidris pugnax","calidris pygmaea","callithrix jacchus","callorhinchus milii","camarhynchus parvulus","camelus dromedarius","canis lupus_dingo","canis lupus_familiaris","canis lupus_familiarisbasenji","canis lupus_familiarisgreatdane","capra hircus","capra hircus_blackbengal","carassius auratus","carlito syrichta","castor canadensis","catagonus wagneri","catharus ustulatus","cavia aperea","cavia porcellus","cebus capucinus","cercocebus atys","cervus hanglu_yarkandensis","chelonoidis abingdonii","chelydra serpentina","chinchilla lanigera","chlorocebus sabaeus","choloepus hoffmanni","chrysemys picta_bellii","chrysolophus pictus","ciona intestinalis","ciona savignyi","clupea harengus","colobus angolensis_palliatus","corvus moneduloides","cottoperca gobio","coturnix japonica","cricetulus griseus_chok1gshd","cricetulus griseus_crigri","cricetulus griseus_picr","crocodylus porosus","cyanistes caeruleus","cyclopterus lumpus","cynoglossus semilaevis","cyprinodon variegatus","cyprinus carpio","cyprinus carpio_germanmirror","cyprinus carpio_hebaored","cyprinus carpio_huanghe","danio rerio","dasypus novemcinctus","delphinapterus leucas","denticeps clupeoides","dicentrarchus labrax","dipodomys ordii","dromaius novaehollandiae","drosophila melanogaster","echeneis naucrates","echinops telfairi","electrophorus electricus","eptatretus burgeri","equus asinus_asinus","equus caballus","erinaceus europaeus","erpetoichthys calabaricus","erythrura gouldiae","esox lucius","falco tinnunculus","felis catus","ficedula albicollis","fukomys damarensis","fundulus heteroclitus","gadus morhua","gallus gallus","gambusia affinis","gasterosteus aculeatus","geospiza fortis","gopherus agassizii","gopherus evgoodei","gorilla gorilla","gouania willdenowi","haplochromis burtoni","heterocephalus glaber_female","heterocephalus glaber_male","hippocampus comes","homo sapiens","hucho hucho","ictalurus punctatus","ictidomys tridecemlineatus","jaculus jaculus","junco hyemalis","kryptolebias marmoratus","labrus bergylta","larimichthys crocea","lates calcarifer","laticauda laticaudata","latimeria chalumnae","lepidothrix coronata","lepisosteus oculatus","leptobrachium leishanense","lonchura striata_domestica","loxodonta africana","lynx canadensis","macaca fascicularis","macaca mulatta","macaca nemestrina","malurus cyaneus_samueli","manacus vitellinus","mandrillus leucophaeus","marmota marmota_marmota","mastacembelus armatus","maylandia zebra","meleagris gallopavo","melopsittacus undulatus","meriones unguiculatus","mesocricetus auratus","microcebus murinus","microtus ochrogaster","mola mola","monodelphis domestica","monodon monoceros","monopterus albus","moschus moschiferus","mus caroli","mus musculus","mus musculus_129s1svimj","mus musculus_aj","mus musculus_akrj","mus musculus_balbcj","mus musculus_c3hhej","mus musculus_c57bl6nj","mus musculus_casteij","mus musculus_cbaj","mus musculus_dba2j","mus musculus_fvbnj","mus musculus_lpj","mus musculus_nodshiltj","mus musculus_nzohlltj","mus musculus_pwkphj","mus musculus_wsbeij","mus pahari","mus spicilegus","mus spretus","mustela putorius_furo","myotis lucifugus","myripristis murdjan","naja naja","nannospalax galili","neogobius melanostomus","neolamprologus brichardi","neovison vison","nomascus leucogenys","notamacropus eugenii","notechis scutatus","nothobranchius furzeri","nothoprocta perdicaria","numida meleagris","ochotona princeps","octodon degus","oncorhynchus kisutch","oncorhynchus mykiss","oncorhynchus tshawytscha","oreochromis aureus","oreochromis niloticus","ornithorhynchus anatinus","oryctolagus cuniculus","oryzias javanicus","oryzias latipes","oryzias latipes_hni","oryzias latipes_hsok","oryzias melastigma","oryzias sinensis","otolemur garnettii","otus sunia","ovis aries","ovis aries_rambouillet","pan paniscus","pan troglodytes","panthera leo","panthera pardus","panthera tigris_altaica","papio anubis","parambassis ranga","paramormyrops kingsleyae","parus major","pavo cristatus","pelodiscus sinensis","pelusios castaneus","periophthalmus magnuspinnatus","peromyscus maniculatus_bairdii","petromyzon marinus","phascolarctos cinereus","phasianus colchicus","phocoena sinus","physeter catodon","piliocolobus tephrosceles","podarcis muralis","poecilia formosa","poecilia latipinna","poecilia mexicana","poecilia reticulata","pogona vitticeps","pongo abelii","procavia capensis","prolemur simus","propithecus coquereli","pseudonaja textilis","pteropus vampyrus","pundamilia nyererei","pygocentrus nattereri","rattus norvegicus","rhinolophus ferrumequinum","rhinopithecus bieti","rhinopithecus roxellana","saccharomyces cerevisiae","saimiri boliviensis_boliviensis","salarias fasciatus","salmo salar","salmo trutta","salvator merianae","sander lucioperca","sarcophilus harrisii","sciurus vulgaris","scleropages formosus","scophthalmus maximus","serinus canaria","seriola dumerili","seriola lalandi_dorsalis","sinocyclocheilus anshuiensis","sinocyclocheilus grahami","sinocyclocheilus rhinocerous","sorex araneus","sparus aurata","spermophilus dauricus","sphaeramia orbicularis","sphenodon punctatus","stachyris ruficeps","stegastes partitus","strigops habroptila","strix occidentalis_caurina","struthio camelus_australis","suricata suricatta","sus scrofa","sus scrofa_bamei","sus scrofa_berkshire","sus scrofa_hampshire","sus scrofa_jinhua","sus scrofa_landrace","sus scrofa_largewhite","sus scrofa_meishan","sus scrofa_pietrain","sus scrofa_rongchang","sus scrofa_tibetan","sus scrofa_usmarc","sus scrofa_wuzhishan","taeniopygia guttata","takifugu rubripes","terrapene carolina_triunguis","tetraodon nigroviridis","theropithecus gelada","tupaia belangeri","tursiops truncatus","urocitellus parryii","ursus americanus","ursus maritimus","ursus thibetanus_thibetanus","varanus komodoensis","vicugna pacos","vombatus ursinus","vulpes vulpes","xenopus tropicalis","xiphophorus couchianus","xiphophorus maculatus","zalophus californianus","zonotrichia albicollis","zosterops lateralis_melanops","Anopheles gambiae str. PEST","Bacillus subtilis subsp. subtilis str. 168","Cyanobacteria","Dickeya dadantii 3937","Dictyostelium discoideum AX4","Escherichia coli str. K-12 substr. MG1655","Escherichia coli O157:H7 str. Sakai","Mesorhizobium japonicum MAFF 303099","Lotus japonicus","Medicago truncatula","Nostoc sp. PCC 7120 = FACHB-418","Oryza sativa Japonica Group","Pseudomonas aeruginosa PAO1","Staphylococcus argenteus","Streptomyces coelicolor A3(2)","Xenopus laevis","Zea mays","Aquila chrysaetos chrysaetos","Bison bison bison","Bos indicus x Bos taurus","Canis lupus dingo","Marmota marmota marmota","Peromyscus maniculatus bairdii","Panthera tigris altaica","Seriola lalandi dorsalis","Branchiostoma lanceolatum","Strigamia maritima","Schmidtea mediterranea","canis lupus_familiarisboxer","Apteryx australis","Aquila chrysaetos canadensis","Balaenoptera acutorostrata scammoni","Galeopterus variegatus","Manis pentadactyla","Nanorana parkeri","Nasalis larvatus","Neomonachus schauinslandi","Pongo pygmaeus abelii","Thamnophis sirtalis","cyprinus carpio_carpio","canis lupus_familiarisgsd","gallus gallus_gca000002315v5","gallus gallus_gca016700215v2","Cebus imitator","Anas platyrhynchos platyrhynchos","Colobus angolensis palliatus","Cyprinus carpio carpio","Cervus hanglu yarkandensis","Cairina moschata domestica","Equus asinus asinus","Equus asinus","Lonchura striata domestica","Malurus cyaneus samueli","Mus musculus castaneus","Mus musculus musculus","Mus musculus domesticus","Saimiri boliviensis boliviensis","Struthio camelus australis","Strix occidentalis caurina","Terrapene carolina triunguis","Ursus thibetanus thibetanus","Zosterops lateralis melanops","cebus imitator","equus asinus","Coffea arabica","Gasterosteus aculeatus aculeatus","Triticum aestivum","Triticum aestivum_subsp._aestivum","Triticum vulgare","Brassica napus","Arachis hypogaea","Hibiscus syriacus","Acridium cancellatum","Schistocerca cancellata","Triticum dicoccoides","Triticum turgidum_subsp._dicoccoides","Triticum turgidum_var._dicoccoides","Dendrohyas sarda","Hyla arborea_sarda","Hyla sarda","Locusta gregaria","Schistocerca gregaria","Gossypium hirsutum","Gossypium hirsutum_subsp._mexicanum","Gossypium lanceolatum","Gossypium purpurascens","Camelina sativa","Myagrum sativum","Carassius auratus_gibelio","Carassius gibelio_gibelio","Carassius gibelio","Carassius gibelio_subsp._gibelio","Cyprinus gibelio","Schistocerca piceifrons","Papaver somniferum","Zingiber officinale","Trichomonas vaginalis_G3","Trichomonas vaginalis_strain_G3","Helianthus annuus","Schistocerca americana","Acipenser ruthenus","Schistocerca serialis_cubense","Panicum virgatum","Nicotiana tabacum","Oncorhynchus nerka_mykiss","Parasalmo mykiss","Salmo mykiss","Schistocerca nitens","Schistocerca vaga","Salvia splendens","Carassius carassius","Cyprinus carassius","Vicia villosa","Camellia sinensis","Thea sinensis","Oncorhynchus keta","Salmo keta","Pisum sativum","Raphanus sativus","Oncorhyncus kisutch","Salmo kisatch","Lolium rigidum","Aegilops squarrosa_subsp._squarrosa","Aegilops squarrosa","Aegilops tauschii","Patropyrum tauschii_subsp._tauschii","Patropyrum tauschii","Triticum aegilops","Triticum tauschii","Cryptomeria japonica","Cupressus japonica","Coregonus clupeaformis","Salmo clupeaformis","Oncorhynchus gorbuscha","Salmo gorbuscha","Glycine max_subsp._soja","Glycine soja","Salmo fontinalis","Salvelinus fontinalis","Phaseolus max","Chenopodium quinoa","Hordeum sativum","Hordeum vulgare_subsp._vulgare","Hordeum vulgare_var._nudum","Hordeum vulgare_var._vulgare","Festuca perennis_(L.)_Columbus_&amp;_J.P.Sm.,_2010","Festuca perennis","Lolium perenne","Lolium vulgare","Barbus grahami","Gossypium arboreum","Brassica oleracea","Malus sylvestris","Pyrus malus_var._sylvestris","Tetragonopterus mexicanus","Arachis stenosperma","Prosopis alba","Lactuca sativa","Dreissena polymorpha","Mytilus polymorphus","Hydractinia symbiolongicarpus","Triticum urartu","Hevea brasiliensis","Siphonia brasiliensis","Oncorhynchus tschawytscha","Salmo tshawytscha","Arachis ipaensis","Zea mays_var._japonica","Salmo namaycush","Salvelinus namaycush","Capsicum annuum","Brienomyrus brachyistius","Marcusenius brachyistius","Convolvulus nil","Ipomoea nil","Pharbitis nil","Olea europaea_subsp._europaea_var._sylvestris","Olea europaea_var._oleaster","Olea europaea_var._sylvestris","Olea europea_subsp._sylvestris","Alosa sapidissima","Clupea sapidissima","Carpiodes asiaticus","Myxocyprinus asiaticus","Actinidia eriantha","Gossypium raimondii","Salmo alpinus","Salvelinus alpinus","Catostomus texanus","Xyrauchen texanus","Doryrhamphus excisus","Quercus lobata","Malus communis","Malus domestica","Malus pumila_auct.","Malus pumila_var._domestica","Malus sylvestris_var._domestica","Malus x_domestica","Pyrus malus","Pyrus malus_var._domestica","Quercus suber","Oncorhynchus nerka","Salmo nerka","Nicotiana tomentosiformis","Carya illinoensis","Carya illinoinensis","Mercenaria mercenaria","Venus mercenaria","Quercus robur","Durio zibethinus","Pongo pygmaeus_abelii","Pongo pygmaeus_abeli","Mya arenaria","Arachis duranensis","Arachis spegazzinii","Pyrus x_bretschneideri","Trifolium pratense","Gorilla gorilla_gorilla","Cobitis anguillicaudata","Misgurnus anguillicaudatus","Scaphiopus bombifrons","Spea bombifrons","Haliotis rufenscens","Haliotis rufescens","Oreochromis nilotica","Perca nilotica","Tilapia nilotica","Acropora convexa","Acropora millepora","Acropora singularis","Cebus apella","Sapajus apella","Simia apella","Eucalyptus grandis","Dasypus fenestratus","Callithrix jacchus_jacchus","Simia jacchus","Pistacia vera","greater Indian_fruit_bat","Pteropus giganteus","Pteropus medius","Salvia miltiorhiza","Salvia miltiorrhiza","Daphnia pulicaria","Magnolia sinica","Manglietia sinica","Manglietiastrum sinicum","Pachylarnax sinica_(Y.W.Law)_N.H.Xia_&amp;_C.Y.Wu","Rosa chinensis","Rosa indica_auct.,_non_L.","Mytilus californianus","Vespertilio vampyrus","Chinemys reevesii","Chinemys reevesi","Emys reevesii","Geoclemys reevesii","Geoclemys reevessi","Mauremys reevesii","Mauremys reevesi","Choloepus brasiliensis_Fitzinger_1871","Choloepus brasiliensis","Choloepus didactylus","Simia nemestrina","Lotus corniculatus_var._japonicus","Nicotiana sylvestris","Tupaia belangeri_chinensis","Clarias gariepinus","Clarias lazera","Silurus gariepinus","Barbus tetrazona","Capoeta tetrazona","Puntigrus tetrazona","Puntius tetrazona","Systomus tetrazona","Lycium ferocissimum","Nicotiana attenuata","Haliotis rubra","Aedes albopictus","Stegomyia albopicta","Spinacia oleracea","Paramecium aurelia_syngen_4","Salvia hispanica","Crassostrea virginica","Ostrea virginica","Felis domesticus","Felis silvestris_catus","Anubis baboon","Papio cynocephalus_anubis","Papio doguera","Papio hamadryas_anubis","Papio hamadryas_doguera","Simia anubis","Pongo pygmaeus","Simia pygmaeus","Sorex etruscus","Suncus etruscus","Mimosa cineraria","Prosopis cineraria","Nycticebus coucang","Tardigradus coucang","Rhododendron vialii","Ixodes dammini","Lupinus angustifolius","Ipomoea triloba","Emiliania huxleyi_CCMP1516","Emiliania huxleyi_CCMP2090","Mangifera indica","Rana temporaria","Magallana gigas","Ostrea gigas","Etheostoma spectabile","Poecilichthys spectabilis","Macadamia integrifolia","Megalobrama amblycephala","Halichoerus grypus","Phoca grypus","Juglans regia","Selaginella moellendorfii","Pleuronectes platessa","Presbytis francoisi","Trachypithecus francoisi","Tripterygium wilfordii","Argiope bruennichi","Lepus cuniculus","Huro salmoides","Labrus salmoides","Labrus salmonides","Micropterus nigricans","Micropterus salmoides","Solanum stenotomum","Neosciurus carolinensis","Sciurus carolinensis","Cervus elaphus","Polyodon spathula","Squalus spathula","Gadus chalcogrammus","Theragra chalcogramma","Bos bubalis","Bubalus arnee_bubalis","Bubalus bubalis","Pleuronectes solea","Solea solea","Solea vulgaris","Mastomys coucha","Praomys coucha","Impatiens glandulifera","Dermacentor andersoni","Felis nebulosa","Neofelis nebulosa","Pteropus egyptiacus","Rousettus aegyptiacus","Rousettus aegypticus","Rousettus egyptiacus","Phoenix dactylifera","Pimephales promelas","Ostrea edulis","Peromyscus maniculatus_bairdii","Gasterosteus pungitius","Pungitius pungitius","Populus alba","Cricetus auratus","Golden hamsters","Syrian hamsters","Chromis aureus","Oreochromis aurea","Daucus carota_subsp._sativus","Daucus carota_var._sativus","Dermacentor silvarum","Hylobates syndactylus","Simia syndactyla","Symphalangus syndactylus","Saccharolobus solfataricus","Felis geoffroyi","Leopardus geoffroyi","Oncifelis geoffroyi","Felis yagouaroundi","Herpailurus yagouaroundi","Herpailurus yaguarondi","Puma yagouaroundii","Puma yagouaroundi","Cervus canadensis","Populus diversifolia","Populus euphratica","Cucurbita pepo_subsp._pepo","Cucurbita pepo_var._medullosa","Cucurbita pepo_var._pepo","Macaca cynomolgus","Macaca irus","Simia fascicularis","Emys muticus","Geoclemmys mutica","Mauremys mutica","Viverra suricatta","Hylobates moloch","Simia moloch","Solanum dulcamara","Cucurbita moschata","Coffea eugeniodes","Coffea eugenioides","Cucurbita maxima","Centropristis striata","Labrus striatus","Oryza sativa_(japonica_cultivar-group)","Oryza sativa_Japonica_Group","Oryza sativa_subsp._japonica","Mus jaculus","Dioscorea cayenensis_subsp._rotundata_(Poir.)_J.Miege,_1968","Dioscorea cayenensis_subsp._rotundata","Dioscorea rotundata","Cercopithecus aethiops_sabaeus","Cercopithecus sabaeus","Cercopithecus sabeus","Chlorocebus aethiops_sabaeus","Chlorocebus aethiops_sabeus","Chlorocebus sabeus","Simia sabaea","Marmota monax","Mus monax","Pygathrix roxellana","Semnopithecus roxellana","Callorhinus ursinus","Callorhynus ursius","Phoca ursina","Cricetulus barabensis_griseus","Elephantulus edwardii","Macroscelides edwardii","Cobitis heteroclita","Neothunnus macropterus","Scomber albacares","Thunnus albacares","Telopea speciosissima","Danio aesculapii","Danio sp._'snakeskin'","Danio sp._snakeskin","Apodemus sylvaticus","Mus sylvaticus","Sylvaemus sylvaticus","Populus balsamifera_subsp._trichocarpa","Mercurialis ambigua","Mercurialis annua","Eugenia oleosa","Syzygium oleosum","Citellus tridecemlineatus","Ovis ammon_aries","Ovis orientalis_aries","Ovis ovis","Solanum verrucosum","Leo pardus","Microtus oregoni","Arabidopsis lyrata_subsp._lyrata","Arabis lyrata_subsp._lyrata","Arabis lyrata","Cardaminopsis lyrata","Manihot esculenta","Manihot utilissima","Mustela erminea","Dolichos unguiculatus","Phaseolus unguiculatus","Vigna unguiculata","Lycopersicon pennellii_(Correll)_D'Arcy,_1982","Solanum pennellii_Correll,_1958","Solanum pennellii","Panicum viride","Setaria viridis","Musa AA_Group","Musa acuminata_AA_Group","Musa acuminata","Musa nana","Gymnostomus macrolepis","Onychostoma macrolepis","Scaphesthes macrolepis","Varicorhinus macrolepis","Varicorhinus (Scaphesthes)_macrolepis","Oryza glaberrima","Pelteobagrus fulvidraco","Pimelodus fulvidraco","Pseudobagrus fulvidraco","Tachysurus fulvidraco","Hylobates concolor_leucogenys","Hylobates concolor_leucogyneus","Hylobates leucogenys_leucogenys","Hylobates leucogenys","Nomascus leucogenys_leucogenys","Nomascus leukogenys","Nannospalax ehrenbergi_galili","Spalax galili","Equus przewalskii_f._caballus","Equus przewalskii_forma_caballus","Thunnus maccoyii","Thynnus maccoyii","Chromis diagramma","Simochromis diagramma","Diplophysa dalaica","Triplophysa dalaica","Felis tigris","Panthera tigris","Echinus purpuratus","Lucioperca lucioperca","Perca lucioperca","Stizostedion lucioperca","Dipodomys spectabilis","Acinonyx jubatus","Felis jubata","Conyza canadensis","Erigeron canadensis","Mustela lutreola","Camelus bactrianus_ferus","Camelus ferus","Cajanus cajan","Didelphys domestica","Pygathrix bieti","Hesperomys eremicus","Peromyscus eremicus","Arabidopsis salsuginea","Eutrema salsugineum","Hesperis salsuginea","Sisymbrium salsugineum","Stenophragma salsugineum","Thellungiella salsuginea","Thelypodium salsugineum","Coetomys damarensis","Cryptomys damarensis","Leptonychotes weddellii","Leptonychotes weddelli","Otaria weddellii","Grammomys dolichurus_surdaster","Grammomys surdaster","Thamnomys surdaster","Solanum aracc-papa","Andropogon sorghum","Sorghum bicolor_subsp._bicolor","Sorghum nervosum","Sorghum saccharatum","Sorghum vulgare","Holocentrus calcarifer","Hippopotamus amphibius_kiboko","Ixodes sanguineus","Rhipicephalus sanguineus","Clupea harengus_harengus","Bos indicus_x_Bos_taurus","Bos primigenius_indicus_x_Bos_primigenius_taurus","Bos taurus_indicus_x_Bos_taurus_taurus","Bos taurus_x_Bos_indicus","Chrysochloris asiatica","Talpa asiatica","Bufo bufo","Rana bufo","Metriaclima zebra","Pseudotropheus sp._'Pseudotropheus_zebra_complex'","Pseudotropheus zebra","Silurus punctatus","Oryx dammah","Asparagus litoralis","Asparagus officinalis","Amaranthus gangeticus","Amaranthus mangostanus","Amaranthus tricolor","Pneumatophorus japonicus","Scomber japonicus","Lutra lutra","Peromyscus leucopus","Perca fluviatilis","Pagothenia bernacchii","Pseudotrematomus bernacchii","Trematomus bernacchii","Trematomus bernacchi","Lipurus cinereus","Mustela furo","Mustela putorius_furo","Chaetochloa italica","Panicum italicum","Pennisetum macrochaetum","Setaria viridis_subsp._italica","Elaeis guineensis","Mus rattus","Rattus rattoides","Rattus rattus","Rattus wroughtoni","Acropora digitifera","Madrepora digitifera","Echinops telfairii","Madrepora verrucosa","Pocillopora danae","Pocillopora verrucosa","Myotis daubentonii","Myotis daubentoni","Vespertilio daubentonii","Limia formosa","Mollienesia formosa","Phyllostomus discolor","Peromyscus californicus_insignis","Peromyscus californicus_subsp._insignis","Galago garnettii","Galago garnetti","Arvicanthis niloticus","Mus niloticus","Didelphis ursina","Phaseolus angularis","Vigna angularis","Haitia acuta","Physa acuta","Physa heterostropha","Physa integra","Physella acuta","Physella heterostropha","Physella integra","Ctenopharyngodon idella","Ctenopharyngodon idellus","Leuciscus idella","Thalassophryne amazonica","Cyprinus rohita","Labeo rohita","Talpa occidentalis","Bombina bombina","Rana bombina","Cavia aperea_porcellus","Cavia cobaya","Mus porcellus","Odocoileus virginianus","Amphibalanus amphitrite","Balanus amphitrite","Panicum hallii","Angill angill","Anguilla anguilla_anguilla","Anguilla anguilla","Muraena anguilla","Delphinus orca","Orcinus orca","Cannabis sativa","Penaeus bubulus","Penaeus carinatus","Penaeus durbani","Penaeus monodon","Penaeus (Penaeus)_monodon","Didelphis vulpecula","Trichosurus vulpecula","Vespertilio lucifugus","Bromus distachyos","Aotus nancymai","Rhamnus zizyphus","Ziziphus jujuba","Micropterus dolomieu","Micropterus velox","Lycopersicon esculentum","Lycopersicon esculentum_var._esculentum","Solanum esculentum","Solanum lycopersicum_var._humboldtii","Meles meles","Ursus meles","Platypus anatinus","Felis uncia","Panthera uncia","Uncia uncia","Crocodilus mississipiensis","Myrmecophaga aculeata","Tachyglossus aculeatus","Colossoma macropomum","Myletes macropomus","Cordylus capensis","Cordylus (Hemicordylus)_capensis","Hemicordylus capensis","Pseudocordylus capensis","Zonurus capensis","Eptesicus fuscus","Vespertilio fuscus","Dromiciops australis","Dromiciops gliroides","Camelus pacos","Lama guanicoe_pacos","Lama pacos","Mollienesia latipinna","Elephas maximus_indicus","Corylus avellana","Ostrea maxima","Pecten maximus","Felis viverrina","Prionailurus viverrinus","Gymnodraco acuticeps","Thalarctos maritimus","Lemur catta","Myotis myotis","Vespertilio myotis","Lytechinus pictus","Psammechinus pictus","Litopenaeus vannamei","Penaeus (Litopenaeus)_vannamei","Penaeus vannamei","Ursus arctos","Vitis riparia","Felis bengalensis","Prionailurus bengalensis","Clethrionomys glareolus","Mus glareolus","Myodes glareolus","Mustela nigripes","Putorius nigripes","Alopex lagopus","Canis lagopus","Vulpes lagopus","Cercocebus torquatus_atys","Simia atys","Lepidosiren annectens","Protopterus annectens","Rhinocryptis annectens","Cerasus avium","Prunus avium","Prunus cerasus_var._avium","Procambarus clarkii","Sorex fumeus","Macrorhinus angustirostris","Mirounga angustirostris","Beta vulgaris_subsp._vulgaris","Beta vulgaris_subsp._vulgaris_var._altissima","Beta vulgaris_Sugar_Beet_Group","Beta vulgaris_var._altissima","Eumetopias jubatus","Phoca jubata","Centruroides sculpturatus","Diceros bicornis_minor","Cicer arietinum","Cleome hassleriana_Chodat,_1898","Tarenaya hassleriana","Sebastes umbrosus","Sebastichthys umbrosus","Eriocheir chinensis","Eriocheir japonica_sinensis","Eriocheir sinensis","Labrax labrax","Morone labrax","Perca labrax","Roccus labrax","Sciaena labrax","Acanthopagrus latus","Sparus latus","Xiphophorus hellerii","Xiphophorus helleri","Acanthochromis polyacathus","Dascyllus polyacanthus","Mustela vison","Neogale vison","Lingula anatina","Lingula lingua","Lingula nipponica","Lingula unguis","Madrepora faveolata","Montastraea faveolata","Montastrea faveolata","Orbicella faveolata","Chinchilla velligera","Chinchilla villidera","Mirounga leonina","Phoca leonina","Perognathus longimembris_pacificus","Cynocephalus variegatus","Galeopithecus variegatus","Vigna radiata","Vitis vinifera_subsp._vinifera","Characodon multiradiatus","Girardinichthys multiradiatus","Marmota flaviventris","Phaseolus calcaratus","Phaseolus chrysanthos","Phaseolus chrysanthus","Vigna calcarata","Vigna umbellata","Balaenoptera acutorostrata","Canis procyonoides","Nyctereutes procyonoides","Amphioxus floridae","Moschus berezovskii","Erythranthe guttata","Mimulus guttatus_subsp._guttatus","Mimulus guttatus","Camelus bactrianus","Desmodus rotundus","Phyllostoma rotundum","Octopus sinensis","Physeter macrocephalus","Alexandromys fortis","Microtus fortis","Dendronephthya gigantea","Canis hyaena","Hyaena hyaena","Helicophagus hypophthalmus","Pangasianodon hypophthalmus","Pangasius hypophthalmus","Pangasius sutchi","Pseudochaenichthys georgianus","Capsella rubella","Perkinsus marinus_ATCC_50983","Holocentrus leopardus","Plectropomus leopardus","Hippocampus zosterae","Artibeus jamaicensis","Citrus sinensis","Citrus x_sinensis","Punica granatum","Abrus cyaneus","Abrus precatorius","Polypterus senegalus","Acomys russatus","Mus russatus","Hemibagrus wyckioides","Macrones wyckioides","Mystus wyckioides","Melanotaenia boesemani","Sturnira hondurensis","Amphilophus centrarchus","Archocentrus centrarchus","Cichlasoma centrarchus","Heros centrarchus","Delphinus melas","Globicephala melaena","Globicephala melas","Manis javanica","Phyllostomus hastatus","Vespertilio hastatus","Scyliorhinus canicula","Squalus canicula","Silurana tropicalis","Xenopus laevis_tropicalis","Xenopus (Silurana)_tropicalis","Pipistrellus kuhlii","Pipistrellus kuhli","Vespertilio kuhlii","Solea senegalensis","Mugil cephalotus","Mugil cephalus","Mugil galapagensis","Mugil japonicus","Siphostoma scovelli","Syngnathus scovelli","Capra aegagrus_hircus","Poeciliopsis prolifica","Gopherus flavomarginatus","Lontra canadensis","Lutra canadensis","Hesperomys torridus","Onychomys torridus","Elephas africanus","Loxodonta africana_africana","Boophilus microplus","Rhipicephalus (Boophilus)_microplus","Rhipicephalus microplus","Molossus molossus","Vespertilio molossus","Lagenorhynchus obliquidens","Delphinus cymodoce","Delphinus truncatus","Tursiops cymodoce","Morone flavescens","Perca flavescens","Euarctos americanus","Arvicola nivalis","Chionomys nivalis","Microtus nivalis","Felis rufus","Lynx rufus","Myotis brandtii","Vespertilio brandtii","Astatotilapia burtoni","Chromis burtoni","Silurus meridionalis","Silurus soldatovi_meridionalis","Cucumis melo","Hydra attenuata","Hydra carnea","Hydra littoralis","Hydra magnipapillata","Hydra vulgaris","Anoplopoma fimbria","Gadus fimbria","Alosa alosa","Clupea alosa","Chelonia mydas","Testudo mydas","Ctenocephalides felis","Stylophora pistillata","Cyrtodiopsis dalmanii","Diopsis dalmanni","Teleopsis dalmanni","Rhagoletis zephyria","Rhodamnia argentea","Labrus celidotus","Notolabrus celidotus","Budorcas taxicolor","Nelumbo nucifera","Arvicola amphibius","Arvicola terrestris_(Linnaeus,_1758)","Mus amphibius","Daphnia magna","Phaseolus vulgaris","Psammomys obesus","Simia syrichta","Cyprinodon tularosa","Arvicola princeps","Equus caballus_przewalskii","Equus ferus_przewalskii","Equus przewalskii","Phoca vitulina","Coecilia bivitatum","Rhinatrema bivitattum","Rhinatrema bivittatum","Lagomys curzoniae","Ochotona curzonae","Ochotona curzoniae","Kogia breviceps","Physeter breviceps","Clupea cyprinoides","Megalops cyprinoides","Diospyros lotus","Hippoglossus stenolepis","Phacochoerus africanus","Corythoichthys intestinalis","Syngnatus intestinalis","Papio leucophaeus","Simia leucophaea","Epinephelus fuscoguttatus","Perca summana_fuscoguttata","Asterias miniata","Asterina miniata","Patiria miniata","Rhinolophus rouxii_sinicus","Rhinolophus sinicus","Lampris incognitus","Monachus schauinslandi","Hippoglossus hippoglossus","Pleuronectes hippoglossus","Andrographis paniculata","Etheostoma cragini","Perca chuatsi","Siniperca chuatsi","Colobus angolensis_palliatus","Notothenia coriiceps","Hypomesus transpacificus","Dermochelys coriacea","Testudo coriacea","Bufo bufo_gargarizans","Bufo gargarizans","Bufo japonicus_gargarizans","Delphinus leucas","Fugu flavidus","Takifugu flavidus","Pteronotus mesoamericanus","Pteronotus parnellii_mesoamericanus","Citrus clementina","Citrus deliciosa_x_Citrus_sinensis","Citrus x_clementina","Sphaeroides rubripes","Tetraodon rubripes","Homarus americanus","Osteoglossum formosum","Pseudosciaena amblyceps","Pseudosciaena crocea","Sciaena crocea","Fragaria vesca","Folsomia candida","Limulus polyphemus","Monoculus polyphemus","Doryrhamphus dactyliophorus","Dunckerocampus dactyliophorus","Syngnathus dactyliophorus","Epinephelus lanceolatus","Holocentrus lanceolatus","Promicrops lanceolatus","Mizuhopecten yessoensis","Patinopecten yessoensis","Patiopecten yessoensis","Pecten yessoensis","Platypoecilus maculatus","Triplophysa rosa","Antechinus flavipes","Phascogale flavipes","Balaena musculus","Vespertilio ferrumequinum","Oryza brachyantha","Testudo picta","Trachemys picta","Tetrahymena thermophila_SB210","Amygdalus communis","Amygdalus dulcis","Prunus amygdalus","Prunus communis","Prunus dulcis","Prunus dulcis_var._sativa","Poecilia latipes","Sarcophilus laniarius_(Owen,_1838)","Sarcophilus laniarius","Ursinus harrisii","Ictalurus furcatus","Pimelodus furcatus","Amphioxus belcheri","Branchiostoma belcheri","Gigantopelta aegis","Echinus variegatus","Lytechinus variegatus","Diaphorina citri","Epinephelus moara","Serranus moara","Stegodyphus dumicola","Boleophthalmus pectinirostris","Gobius pectinirostris","Austrofundulus limnaeus","Columba livia_domestica","Columba livia","Pleuronectes maximus","Psetta maxima","Rhombus maximus","Sesamum indicum","Sesamum orientale","Armeniaca mume","Prunus mume","Myotis davidii","Vespertilio Davidii","Didelphys agilis","Gracilinanus agilis","Acanthophacelus reticulata","Poecilia (Acanthophacelus)_reticulata","Poecilia latipinna_reticulata","Australorbis glabratus","Biomphalaria glabrata","Planorbis glabratus","Hypudaeus ochrogaster","Amygdalus persica","Persica vulgaris","Prunus persica","Prunus persica_var._densa","Chiloscyllium plagiosum","Scyllium plagiosum","Cheilinus undulatus","Phodopus roborovskii","Caenorhabditis vulgaris","Lamprologus brichardi","Gymnopis unicolor","Microcaecilia unicolor","Rhinatrema unicolor","Sciaena jaculatrix","Toxotes jaculatrix","Bos indicus","Bos primigenius_indicus","Bos taurus_indicus","Lacerta sicula_raffonei","Podarcis raffoneae","Podarcis raffonei","Podarcis wagleriana_raffonei","Benincasa cerifera","Benincasa hispida","Benincasa pruriens","Cucurbita hispida","Lagenaria siceraria_var._hispida","Dendrobium catenatum","Marsupenaeus japonicus","Penaeus japonicus","Penaeus (Marsupenaeus)_japonicus","Penaeus (Melicertus)_japonicus","Chaetodon argus","Scatophagus argus","Chanos chanos","Mugil chanos","Bison bison_bison","Bos bison_bison","Amblyraja radiata","Raja radiata","Hipposideros armiger","Rhinolophus armiger","Cynoglossus (Arelia)_semilaevis","Alecto japonica","Anneissia japonica","Oxycomanthus japonicus","Ananas comosus","Ananas comosus_var._comosus","Ananas lucidus","Bromelia comosa","Callionymus splendidus","Pterosynchiropus splendidus","Synchiropus splendidus","Neophocaena asiaeorientalis_asiaeorientalis","Coluber guttatus","Elaphe guttata","Pantherophis guttatus","Pollicipes cornucopia","Pollicipes pollicipes","Pseudoliparis swirei","Chelonoidis abingdoni","Chelonoidis nigra_abingdonii","Geochelone nigra_abigdonii","Geochelone nigra_abingdoni","Geochelone nigra_ephippium","Testudo abingdonii","Rhincodon typus","Ricinus sanguineus","Malania oleifera","Ceratotherium simum_simum","Rivulus marmoratus","Patella vulgata","Rhagoletis pomonella","Squalus fasciatus","Squalus tigrinus","Stegostoma fasciatum","Stegostoma tigrinum","Cistudo triunguis","Terrapene mexicana_triunguis","Terrapene triunguis","Odobenus rosmarus_divergens","Manatus latirostris","Trichechus manatus_latirostris","Carcharodon carcharias","Squalus carcharias","Macrognathus armatus","Anas boschas","Anas domesticus","Anas platyrhynchos_f._domestica","Theobroma cacao","Diabrotica virgifera_virgifera","Actinia diaphana","Aiptasia pallida","Aiptasia pulchella","Dysactis pallida","Exaiptasia diaphana","Exaiptasia pallida","Syngnathus acus_rubescens","Syngnathus acus","Syngnathus rubescens","Caretta caretta","Testudo caretta","Guillardia theta_CCMP2712","Anarrhichthys ocellatus","Trionyx sinensis","Hippoglossus olivaceus","Paralichthys olivaceus","Xiphias gladius","Bos grunniens_mutus","Poephagus mutus","Alligator sinensis","Morus notabilis","Nymphaea colorata","Photinus pyralis","Pomacea canaliculata","Haplochromis nyererei","Caranx dumerili","Macrosteles (Macrosteles)_quadrilineatus","Macrosteles quadrilineatus","Enhydra lutris_kenyoni","Fluta alba","Muraena alba","Caecilia seraphini","Caecilia Seraphini","Geotrypetes seraphini","Hypogeophis seraphini","Chaetodon rostratus","Chelmon rostratus","Cucumis sativus","Cyrtodactylus macularius","Eublepharis macularius","Felis concolor","Panthera concolor","Puma concolor","Cancer chinensis","Fenneropenaeus chinensis","Penaeus chinensis","Pomacentrus partitus","Phascum patens","Physcomitrella patens_subsp._patens","Physcomitrium patens","Anas jamaicensis","Oxyura jamaicensis","Drosophila miranda","Lottia gigantea","Eurytemora affinis","Temora affinis","Crotalus tigris","Argentina anserina","Potentilla anserina","Achaearanea tepidariorum","Parasteatoda tepidariorum","Theridion tepidariorum","Uranotaenia lowii","Cynolebias whitei","Nematolebias whitei","Simpsonichthys whitei","Sceloporus undulatus","Stellio undulatus","Helobdella robusta","Styela clava","Manis afer_afer","Orycteropus afer_afer","Leucoraja erinacea","Raja erinacea","Raja erinaceus","Raja erinacia","Phytophthora nicotianae_INRA-310","Anas olor","Cygnus olor","Lacerta agilis","Millepora damicornis","Pocillopora caespitosa_laysanensis","Pocillopora damicornis_laysanensis","Pocillopora damicornis","Morone saxatilis","Perca saxatilis","Miniopterus natalensis","Miniopterus schreibersii_natalensis","Vespertilio natalensis","Anas cygnoid","Actinia tenebrosa","Neptunus trituberculatus","Portunus (Portunus)_trituberculatus","Portunus trituberculatus","Lacerta vivipara","Zootoca vivipara","Propithecus verreauxi_coquereli","Jatropha curcas","Rhabditis briggsae","Cherax quadricarinatus","Homalodisca coagulata","Homalodisca vitripennis","Tettigonia coagulata","Tettigonia vitripennis","Python bivittatus","Python molurus_bivittatus","Chrysemys scripta_elegans","Emys elegans","Pseudemys scripta_elegans","Trachemys scripta_elegans","Protobothrops mucrosquamatus","Trigonocephalus mucrosquamatus","Trimeresurus mucrosquamatus","Paramacrobiotus metropolitanus","Lipotes vexillifer","Poephila guttata","Taenopygia guttata","Phalaenopsis equestris","Stauroglottis equestris","Balanoglossus kowalevskii","Saccoglossus kowalevskii","Saccoglossus kowalevskyi","Phasianus meleagris","Momordica charantia","Sphaerodactylus townsendi","Eutainia elegans","Thamnophis elegans","Corvus hawaiiensis","Manacus candei","Pipra candei","Euleptes europaea","Euleptes europea","Phyllodactylus europaea","Phyllodactylus europaeus","Ptyodactylus caudivolvolus","Altirana parkeri","Ahaetulla prasina","Dryophis prasinus","Fusarium oxysporum_f._sp._lycopersici_4287","Heteropelma chrysocephalum","Neopelma chrysocephalum","Musca domestica","Pristis pectinata","Ischnura elegans","Fringilla chalybeata","Vidua chalybeata","Coturnix coturnix_japanica","Coturnix coturnix_japonica","Coturnix coturnix_Japonicus","Coturnix japonica_japonica","Gekko japonicus","Platydactylus japonicus","Nilaparvata lugens","Ardea americana","Grus americana","Grus americanus","Harpia harpyja","Vultur harpyja","Pipra filicauda","Herrania umbratica","Ilyonectria robusta","Cupidonia cupido_pallidicincta","Tympanuchus pallidicinctus","Topomyia yanbarensis","Parus atricapillus","Poecile atricapilla","Poecile atricapillus","Corapipo altera","Acyrthosiphum pisum","Saprolegnia parasitica_CBS_223.65","Fringilla macroura","Vidua macroura","Carica papaya","Chiroxiphia lanceolata","Pipra lanceolata","Octopus bimaculoides","Lagopus muta","Tetrao mutus","Bradysia coprophila","Sciara coprophila","Coluber sirtalis","Falco peregrinus","Falco cherrug","Asteracanthion distichum","Asterias attenuata","Asterias clathrata","Asterias disticha","Asterias gigantea","Asterias pallida","Asterias rubens","Asterias stimpsoni","Asterias vulgaris","Manduca sexta","Sphinx sexta","Condylura cristata","Sorex cristatus","Cuculus canorus","Pezoporus wallicus","Aedes (Stegomyia)_aegypti","Culex aegypti","Stegomyia aegypti","Falco naumanni","Corvus kubaryi","Necator americanus","Larus tridactylus","Rissa tridactyla","Culex (Culex)_pipiens_pallens","Culex pipiens_pallens","Turdus ustulatus","Accipiter gentilis","Accipiter gentillis","Falco gentilis","Amborella trichopoda","Falco biarmicus","Lagopus leucura","Lagopus leucurus","Tetrao leucurus","Falco rusticolus","Corvus brachyrhynchos","Uloborus diversus","Phytophthora infestans_strain_T30-4","Phytophthora infestans_T30-4","Empidonax traillii","Muscicapa traillii","Strix alba","Tyto alba","Lepeophtheirus salmonis","Gavialis gangeticus","Lacerta gangetica","Daphnia carinata","Aphis gossypii","Ampithoe aztecus","Hyalella azteca","Hyalella knickerbockeri","Colletotrichum lupini","Gloeosporium lupini","Sphaeroforma arctica_JP610","Suillus fuscotomentosus","Mollisia scopiformis","Phialocephala scopiformis","Muscicapa cayanensis","Myiozetetes cayanensis","Hyaloscypha bicolor_E","Psittacus undulatus","Fringilla montana","Passer montanus","Coccinella axyridis","Harmonia axyridis","Aimophila crissalis","Kieneria crissalis","Kieneria crissalis_(Vigors,_1839)","Melozone crissalis","Pipilo crissalis","Pipilo fuscus_crissalis","Stomoxis calcitrans","Stomoxys calcitrans","Anas atrata","Cygnus atratus","Culex fatigans","Culex pipiens_fatigans","Culex pipiens_quinquefasciatus","Hirundo rustica","Acanthaster planci","Asterias planci","Molothrus ater","Oriolus ater","Laccaria bicolor_S238N-H82","Anastrepha obliqua","Tephritis obliqua","Grapholitha glycinivorella","Leguminivora glycinivorella","Ammodromus caudacutus_nelsoni","Ammospiza nelsoni","Nylanderia fulva","Paratrechina fulva","Agelaius phoeniceus","Agelaius phoniceus","Oriolus phoeniceus","Colletotrichum fructicola","Colletotrichum ignotum","Euthrips occidentalis","Frankliniella brunnescens","Frankliniella californica","Frankliniella occidentalis_brunnescens","Frankliniella occidentalis","Motacilla alba_alba","Fusarium solani","Fusisporium solani","Neocosmospora solani","Sitophilus oryzae","Corvus cornix_cornix","Fringilla canaria_Linnaeus,_1758","Serinus canarius","Drosophila subpulchrella","Chlamydomonas smithii","Puccinia striiformis_f._sp._tritici","Bactrocera cucurbitae","Bactrocera (Zeugodacus)_cucurbitae","Zeugodacus cucurbitae","Zeugodacus (Zeugodacus)_cucurbitae","Antrodia serialis","Fomitopsis serialis","Neoantrodia serialis","Polyporus serialis","Drosophila suzukii","Leucophenga suzukii","Aedes smithii","Wyeomyia smithii","Montifringilla ruficollis","Pyrgilauda ruficollis","Gymnogyps californianus","Vultur californianus","Bactrocera (Bactrocera)_dorsalis","Bactrocera (Bactrocera)_invadens","Bactrocera dorsalis","Bactrocera invadens","Bactrocera papayae","Bactrocera philippinensis","Dacus dorsalis","Trichoplusia ni","Leptothorax curvispinosus","Temnothorax curvispinosus","Saprolegnia declina_VS20","Saprolegnia diclina_VS20","Fringilla albicollis","Bactrocera neohumeralis","Dacus tryoni_var._neohumeralis","Sphaeria pertusa","Trematosphaeria pertusa","Fusarium oxysporum_var._redolens","Fusarium redolens","Anastrepha ludens","Trypeta ludens","Cantharellus anzutake","Malurus melanocephalus","Muscicapa melanocephala","Melitaea cinxia","Papilio cinxia","Maniola jurtina","Papilio jurtina","Anas fuligula","Aythya fuligula","Phalaena mori","Botys furnacalis","Ostrinia furnacalis","Priapula caudata","Priapulus caudatus","Apanteles glomeratus","Cotesia glomerata","Ichneumon glomeratus","Centrocercus urophasianus","Centrocerus urophasianus","Tetrao urophasianus","Montifringilla taczanowskii_(Przewalski,_1876)","Onychostruthus taczanowskii","Monomorium pharaonis","Daktulosphaira vitifoliae","Pemphigus vitifoliae","Viteus vitifoliae","Helicoverpa armigera","Heliothis armigera","Heliothis (Helicoverpa)_armigera","Noctua armigera","Drosophila biarmipes","Myzus (Nectarosiphon)_persicae","Myzus persicae","Lucilia sericata","Phaenicia sericata","Tinamus guttatus","Fringilla georgiana","Melospiza georgiana","Helicoverpa zea","Heliothis zea","Phalaena zea","Drosophila annanassae","Fusarium odoratissimum_NRRL_54006","Coccinella 7-punctata","Coccinella septempunctata","Spodoptera frugiperda","Tigriopus californicus","Tisbe californica","Muscicapa albicollis","Mytilidion resinicola","Mytilinidion resinicola","Halyomorpha halys","Phycomyces blakesleeanus_NRRL_1555(-)","Monoraphidium neglectum","Sturnus vulgaris","Bactrocera tryoni","Tephritis tryoni","Apus apus","Hirundo apus","Suillus paluster","Suillus discolor","Suillus tomentosus_var._discolor","Trichina spiralis","Onthophagus taurus","Epistrophe balteatus","Episyrphus balteatus","Episyrphus (Episyrphus)_balteatus","Musca balteata","Leptinotarsa decemlineata","Leptinotarsa decimlineata","Stilodes decemlineata","Boletus plorans","Suillus plorans","Dryobates pubescens","Picoides pubescens_(Linnaeus,_1766)","Picoides pubescens","Fusarium proliferatum_ET1","Fusarium oxysporum_Fo47","Schizophyllum commune_H4-8","Depressaria gossypiella","Pectinophora gossypiella","Parus humilis","Podoces humilis","Pseudopoces humilis","Pseudopodoces humilis","Ascidia intestinalis","Distomum viverrini","Opisthorchis viverrini","Puccinia graminis_f._sp._tritici_CRL_75-36-700-3","Plutella xylostella","Melampsora larici-populina_98AG31","Drosophila obscura","Fusarium verticillioides_7600","Anoplophora glabripennis","Anoplophora nobilis","Cerosterna glabripennis","Melanauster nobilis","Calypte anna","Ornismya anna","Microdochium trichocladiopsis","Anopheles merus","Bactrocera (Daculus)_oleae","Bactrocera (Dacus)_oleae","Bactrocera oleae","Dacus oleae","Musca oleae","Fusarium mangiferae","Contarinia nasturtii","Parastagonospora nodorum_SN15","Zasmidium cellare_ATCC_36951","Drosophila mauritiana","Eupeodes corollae","Eupeodes (Eupeodes)_corollae","Metasyrphus corollae","Spodoptera litura","Sitodiplosis mosellana","Microgaster mediator","Microplitis medianus","Microplitis mediator","Drosophila kikkawai","Diaporthe citri","Phomopsis citri","Mesites unicolor","Mesitornis unicolor","Suillus subaureus","Colletotrichum capsici","Colletotrichum dematium_f._truncatum_(Schwein.)_Arx,_1957","Colletotrichum truncatum","Glomerella glycines","Vermicularia capsici","Vermicularia truncata","Anisochrysa carnea","Chrysopa carnea","Chrysoperla carnea","Drosophila takahashii","Lucilia cuprina","Falco albicilla","Haliaeetus albicilla","Antrostomus carolinensis","Caprimulgus carolinensis","Colias crocea","Colias croceus","Papilio croceus","Leptidea sinapis","Papilio sinapis","Anopheles arabiensis","Drosophila ficusphila","Vollenhovia emeryi","Hermetia illucens","Fusarium vanettenii_77-13-4","Nectria haematococca_mpVI_77-13-4","Thrips palmi","Falco leucocephalus","Haliaeetus leucocephalus","Malaya genurostris","Colletotrichum gloeosporioides_(Penz.)_Penz._&amp;_Sacc.,_1884","Glomerella cingulata","Glomerella rufomaculans-vaccinii","Gnomoniopsis cingulata","Vermicularia gloeosporioides","Acanthamoeba castellanii_Neff_strain","Acanthamoeba castellanii_strain_Neff","Acanthamoeba castellanii_str._Neff","Drosophila albomicans","Drosophila nasuta_albomicans","Spinulophila albomicans","Diaporthe amygdali","Fusicoccum amygdali","Phomopsis amygdali","Pelecanus crispus","Pelecanus philippensis_crispus","Drosophila rhopaloa","Aphantopus hyperantus","Maniola hyperantus","Papilio hyperantus","Drosophila serrata","Leptopilina heterotoma","Peronospora halstedii","Plasmopara halstedii","Cuculus discolor","Leptosomus discolor","Drosophila santomea","Sipha flava","Drosophila teissieri","Aptenodytes forsteri","Phaethon lepturus","Drosophila bipectinata","Fulmaris glacialis","Fulmarus glacialis","Procellaria glacialis","Ardea garzetta","Egretta garzetta","Anopheles mysorensis","Anopheles stephensi_mysorensis","Anopheles stephensi","Anopheles stephensi_var._mysorensis","Neocellia intermedia_Rothwell,_1907","Neocellia intermedia","Cryptotermes secundus","Pestalotiopsis fici_W106-1","Aricia agestis","Papilio agestis","Polyommatus agestis","Artogeia napi","Papilio napi","Pieris napi","Drosophila eugracilis","Wasmannia auropunctata","Oppia nitens","Adelges cooleyi","Chermes cooleyi","Gilletteella cooleyi","Acanthisitta chloris","Sitta chloris","Agrilus feretrius","Agrilus marcopoli","Agrilus planipennis","Drosophila elegans","Hyposmocoma kahamanoa","Cariama cristata","Palamedea cristata","Aleurodes tabaci","Aleyrodes tabaci","Bemisia tabaci","Ibis nippon","Nipponia nippon","Balearica gibbericeps","Balearica pavonina_gibbericeps","Balearica regulorum_gibbericepse","Balearica regulorum_gibbericeps","Bombus affinis","Varroa jacobsoni","Drosophila gunungcola","Colius striatus","Tauraco erythrolophus","Colletotrichum aenigma","Colletotrichum communis","Colletotrichum dianesei","Colletotrichum endomangiferae","Colletotrichum hymenocallidis","Colletotrichum jasmini-sambac","Colletotrichum melanocaulon","Colletotrichum siamense","Pterocles gutturalis","Aethina tumida","Galleria mellonella","Phalaena mellonella","Bicyclus anynana","Mycalesis anynana","Leptopilina boulardi","Zootermopsis nevadensis","Achroia grisella","Tinea grisella","Acremonium falciforme","Cephalosporium falciforme","Fusarium falciforme","Neocosmospora falciformis","Drosophila mohavensis","Drosophila innubila","Bombus huntii","Cuculus indicator","Indicator indicator","Dendroctonus ponderosae","Ardea helias","Eurypyga helias","Dracunculus loa","Cadophora gregata","Cephalosporium gregatum","Phialophora gregata","Nestor notabilis_notabilis","Nestor notabilis","Colymbus stellatus","Gavia stellata","Cynthia cardui","Papilio cardui","Vanessa cardui","Cladosporium fulvum","Fulvia fulva","Mycovellosiella fulva","Passalora fulva","Plodia interpunctella","Tinea interpunctella","Stilbospora angustata","Truncatella angustata","Truncatella truncata","Chlamydotis macqueenii_macqueenii","Chlamydotis macqueenii_macqueeni","Chlamydotis macqueenii","Chlamydotis undulata_macqueenii","Otis macqueenii","Anopheles funestus","Fusarium fujikuroi_IMI_58289","Cephalosporium keratoplasticum_(nom._inval.)","Fusarium keratoplasticum","Neocosmospora keratoplastica","Drosophila lebanonensis","Scaptodrosophila lebanonensis","Merops nubicus","Coniothyrium fuckelii_var._sporulosum","Coniothyrium sporulosum","Paraconiothyrium sporulosum","Paraconyotrichium sporulosum","Paraphaeosphaeria sporulosa","Fusarium venenatum","Fusarium venetum","Amyelois transitella","Nephopteryx transitella","Pelecanus carbo","Phalacrocorax carbo","Naegleria lovaniensis","Papilio machaon","Gaeumannomyces tritici_R3-111a-1","Papilio aegeria","Pararge aegeria","Lophyrus lecontei","Neodiprion lecontei","Sclerotinia sclerotiorum_1980_UF-70","Aegialitis vocifera","Charadrius vociferous","Charadrius vociferus","Oxyechus vociferus","Sus scrofa domesticus"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>.<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":".","targets":1}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html">unique</a></span><span class="op">(</span><span class="va">ah</span><span class="op">$</span><span class="va">rdataclass</span><span class="op">)</span> </span>
<span><span class="co">#&gt;  [1] "GRanges"                           "data.frame"                       </span></span>
<span><span class="co">#&gt;  [3] "Inparanoid8Db"                     "TwoBitFile"                       </span></span>
<span><span class="co">#&gt;  [5] "ChainFile"                         "SQLiteConnection"                 </span></span>
<span><span class="co">#&gt;  [7] "biopax"                            "BigWigFile"                       </span></span>
<span><span class="co">#&gt;  [9] "AAStringSet"                       "MSnSet"                           </span></span>
<span><span class="co">#&gt; [11] "mzRident"                          "list"                             </span></span>
<span><span class="co">#&gt; [13] "TxDb"                              "Rle"                              </span></span>
<span><span class="co">#&gt; [15] "EnsDb"                             "VcfFile"                          </span></span>
<span><span class="co">#&gt; [17] "igraph"                            "data.frame, DNAStringSet, GRanges"</span></span>
<span><span class="co">#&gt; [19] "sqlite"                            "data.table"                       </span></span>
<span><span class="co">#&gt; [21] "character"                         "SQLite"                           </span></span>
<span><span class="co">#&gt; [23] "SQLiteFile"                        "Tibble"                           </span></span>
<span><span class="co">#&gt; [25] "Rda"                               "FaFile"                           </span></span>
<span><span class="co">#&gt; [27] "String"                            "CompDb"                           </span></span>
<span><span class="co">#&gt; [29] "OrgDb"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="子集" class="level3" data-number="3.2.1"><h3 data-number="3.2.1" class="anchored" data-anchor-id="子集">
<span class="header-section-number">3.2.1</span> 子集</h3>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">ah</span><span class="op">[</span><span class="va">ah</span><span class="op">$</span><span class="va">dataprovider</span><span class="op">==</span><span class="st">"UCSC"</span><span class="op">]</span></span>
<span><span class="co">#&gt; AnnotationHub with 11204 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2024-10-28</span></span>
<span><span class="co">#&gt; # $dataprovider: UCSC</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens, Mus musculus, Drosophila melanogaster, Bos taurus,...</span></span>
<span><span class="co">#&gt; # $rdataclass: GRanges, Rle, ChainFile, TxDb, TwoBitFile</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH5012"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                                    </span></span>
<span><span class="co">#&gt;   AH5012   | Chromosome Band                          </span></span>
<span><span class="co">#&gt;   AH5013   | STS Markers                              </span></span>
<span><span class="co">#&gt;   AH5014   | FISH Clones                              </span></span>
<span><span class="co">#&gt;   AH5015   | Recomb Rate                              </span></span>
<span><span class="co">#&gt;   AH5016   | ENCODE Pilot                             </span></span>
<span><span class="co">#&gt;   ...        ...                                      </span></span>
<span><span class="co">#&gt;   AH114096 | TxDb.Mmusculus.UCSC.mm39.refGene.sqlite  </span></span>
<span><span class="co">#&gt;   AH116719 | TxDb.Hsapiens.UCSC.hg38.refGene.sqlite   </span></span>
<span><span class="co">#&gt;   AH116720 | TxDb.Mmusculus.UCSC.mm39.refGene.sqlite  </span></span>
<span><span class="co">#&gt;   AH117076 | TxDb.Hsapiens.UCSC.hg38.knownGene.sqlite </span></span>
<span><span class="co">#&gt;   AH117077 | TxDb.Mmusculus.UCSC.mm39.knownGene.sqlite</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html">subset</a></span><span class="op">(</span><span class="va">ah</span>, <span class="va">species</span> <span class="op">==</span> <span class="st">"homo sapiens"</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 71 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2024-10-28</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl</span></span>
<span><span class="co">#&gt; # $species: homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: GRanges, TwoBitFile</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH83649"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                                           </span></span>
<span><span class="co">#&gt;   AH83649  | Homo_sapiens.GRCh38.101.abinitio.gtf            </span></span>
<span><span class="co">#&gt;   AH83650  | Homo_sapiens.GRCh38.101.chr.gtf                 </span></span>
<span><span class="co">#&gt;   AH83651  | Homo_sapiens.GRCh38.101.chr_patch_hapl_scaff.gtf</span></span>
<span><span class="co">#&gt;   AH83652  | Homo_sapiens.GRCh38.101.gtf                     </span></span>
<span><span class="co">#&gt;   AH84627  | Homo_sapiens.GRCh38.cdna.all.2bit               </span></span>
<span><span class="co">#&gt;   ...        ...                                             </span></span>
<span><span class="co">#&gt;   AH110100 | Homo_sapiens.GRCh38.108.gtf                     </span></span>
<span><span class="co">#&gt;   AH110866 | Homo_sapiens.GRCh38.109.abinitio.gtf            </span></span>
<span><span class="co">#&gt;   AH110867 | Homo_sapiens.GRCh38.109.chr.gtf                 </span></span>
<span><span class="co">#&gt;   AH110868 | Homo_sapiens.GRCh38.109.chr_patch_hapl_scaff.gtf</span></span>
<span><span class="co">#&gt;   AH110869 | Homo_sapiens.GRCh38.109.gtf</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="查询" class="level3" data-number="3.2.2"><h3 data-number="3.2.2" class="anchored" data-anchor-id="查询">
<span class="header-section-number">3.2.2</span> 查询</h3>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Ensembl"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 37283 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2024-10-28</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl, FANTOM5,DLRP,IUPHAR,HPRD,STRING,SWISSPROT,TREMBL,E...</span></span>
<span><span class="co">#&gt; # $species: Mus musculus, Sus scrofa, Homo sapiens, Rattus norvegicus, Danio...</span></span>
<span><span class="co">#&gt; # $rdataclass: TwoBitFile, GRanges, EnsDb, SQLiteFile, data.frame, OrgDb, li...</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH5046"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                                             </span></span>
<span><span class="co">#&gt;   AH5046   | Ensembl Genes                                     </span></span>
<span><span class="co">#&gt;   AH5160   | Ensembl Genes                                     </span></span>
<span><span class="co">#&gt;   AH5311   | Ensembl Genes                                     </span></span>
<span><span class="co">#&gt;   AH5434   | Ensembl Genes                                     </span></span>
<span><span class="co">#&gt;   AH5435   | Ensembl EST Genes                                 </span></span>
<span><span class="co">#&gt;   ...        ...                                               </span></span>
<span><span class="co">#&gt;   AH119504 | Ensembl 113 EnsDb for Xiphophorus maculatus       </span></span>
<span><span class="co">#&gt;   AH119505 | Ensembl 113 EnsDb for Xenopus tropicalis          </span></span>
<span><span class="co">#&gt;   AH119506 | Ensembl 113 EnsDb for Zonotrichia albicollis      </span></span>
<span><span class="co">#&gt;   AH119507 | Ensembl 113 EnsDb for Zalophus californianus      </span></span>
<span><span class="co">#&gt;   AH119508 | Ensembl 113 EnsDb for Zosterops lateralis melanops</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Gencode"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 5426 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2024-10-28</span></span>
<span><span class="co">#&gt; # $dataprovider: UCSC, GENCODE, Gencode, ENCODE, BroadInstitute</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens, Mus musculus, Rattus norvegicus, Pan troglodytes, ...</span></span>
<span><span class="co">#&gt; # $rdataclass: GRanges, list, TxDb, SQLiteConnection</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH5073"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                                                </span></span>
<span><span class="co">#&gt;   AH5073   | Affy RNA Loc                                         </span></span>
<span><span class="co">#&gt;   AH5079   | CSHL Small RNA-seq                                   </span></span>
<span><span class="co">#&gt;   AH5080   | GIS RNA PET                                          </span></span>
<span><span class="co">#&gt;   AH5084   | RIKEN CAGE Loc                                       </span></span>
<span><span class="co">#&gt;   AH5088   | SUNY SwitchGear                                      </span></span>
<span><span class="co">#&gt;   ...        ...                                                  </span></span>
<span><span class="co">#&gt;   AH107317 | hg19.Crawford.wgEncodeDukeMapabilityRegionsExcludable</span></span>
<span><span class="co">#&gt;   AH116727 | ENCODE_dELS_regions                                  </span></span>
<span><span class="co">#&gt;   AH116728 | ENCODE_pELS_regions                                  </span></span>
<span><span class="co">#&gt;   AH116729 | ENCODE_PLS_regions                                   </span></span>
<span><span class="co">#&gt;   AH116730 | GENCODE basic gene annotation v40</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Web界面</span></span>
<span><span class="fu">BiocHubsShiny</span><span class="fu">::</span><span class="fu">BiocHubsShiny</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="images/BiocHubsShiny.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:80.0%"></p>
</figure>
</div>
</section></section><section id="bsgenome" class="level2" data-number="3.3"><h2 data-number="3.3" class="anchored" data-anchor-id="bsgenome">
<span class="header-section-number">3.3</span> BSgenome</h2>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/BSgenome">BSgenome</a></span><span class="op">)</span></span>
<span><span class="fu">conflicts_prefer</span><span class="op">(</span><span class="fu">Biostrings</span><span class="fu">::</span><span class="va">setdiff</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BSgenome/man/available.genomes.html">available.genomes</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;   [1] "BSgenome.Alyrata.JGI.v1"                           </span></span>
<span><span class="co">#&gt;   [2] "BSgenome.Amellifera.BeeBase.assembly4"             </span></span>
<span><span class="co">#&gt;   [3] "BSgenome.Amellifera.NCBI.AmelHAv3.1"               </span></span>
<span><span class="co">#&gt;   [4] "BSgenome.Amellifera.UCSC.apiMel2"                  </span></span>
<span><span class="co">#&gt;   [5] "BSgenome.Amellifera.UCSC.apiMel2.masked"           </span></span>
<span><span class="co">#&gt;   [6] "BSgenome.Aofficinalis.NCBI.V1"                     </span></span>
<span><span class="co">#&gt;   [7] "BSgenome.Athaliana.TAIR.04232008"                  </span></span>
<span><span class="co">#&gt;   [8] "BSgenome.Athaliana.TAIR.TAIR9"                     </span></span>
<span><span class="co">#&gt;   [9] "BSgenome.Btaurus.UCSC.bosTau3"                     </span></span>
<span><span class="co">#&gt;  [10] "BSgenome.Btaurus.UCSC.bosTau3.masked"              </span></span>
<span><span class="co">#&gt;  [11] "BSgenome.Btaurus.UCSC.bosTau4"                     </span></span>
<span><span class="co">#&gt;  [12] "BSgenome.Btaurus.UCSC.bosTau4.masked"              </span></span>
<span><span class="co">#&gt;  [13] "BSgenome.Btaurus.UCSC.bosTau6"                     </span></span>
<span><span class="co">#&gt;  [14] "BSgenome.Btaurus.UCSC.bosTau6.masked"              </span></span>
<span><span class="co">#&gt;  [15] "BSgenome.Btaurus.UCSC.bosTau8"                     </span></span>
<span><span class="co">#&gt;  [16] "BSgenome.Btaurus.UCSC.bosTau9"                     </span></span>
<span><span class="co">#&gt;  [17] "BSgenome.Btaurus.UCSC.bosTau9.masked"              </span></span>
<span><span class="co">#&gt;  [18] "BSgenome.Carietinum.NCBI.v1"                       </span></span>
<span><span class="co">#&gt;  [19] "BSgenome.Celegans.UCSC.ce10"                       </span></span>
<span><span class="co">#&gt;  [20] "BSgenome.Celegans.UCSC.ce11"                       </span></span>
<span><span class="co">#&gt;  [21] "BSgenome.Celegans.UCSC.ce2"                        </span></span>
<span><span class="co">#&gt;  [22] "BSgenome.Celegans.UCSC.ce6"                        </span></span>
<span><span class="co">#&gt;  [23] "BSgenome.Cfamiliaris.UCSC.canFam2"                 </span></span>
<span><span class="co">#&gt;  [24] "BSgenome.Cfamiliaris.UCSC.canFam2.masked"          </span></span>
<span><span class="co">#&gt;  [25] "BSgenome.Cfamiliaris.UCSC.canFam3"                 </span></span>
<span><span class="co">#&gt;  [26] "BSgenome.Cfamiliaris.UCSC.canFam3.masked"          </span></span>
<span><span class="co">#&gt;  [27] "BSgenome.Cjacchus.UCSC.calJac3"                    </span></span>
<span><span class="co">#&gt;  [28] "BSgenome.Cjacchus.UCSC.calJac4"                    </span></span>
<span><span class="co">#&gt;  [29] "BSgenome.CneoformansVarGrubiiKN99.NCBI.ASM221672v1"</span></span>
<span><span class="co">#&gt;  [30] "BSgenome.Creinhardtii.JGI.v5.6"                    </span></span>
<span><span class="co">#&gt;  [31] "BSgenome.Dmelanogaster.UCSC.dm2"                   </span></span>
<span><span class="co">#&gt;  [32] "BSgenome.Dmelanogaster.UCSC.dm2.masked"            </span></span>
<span><span class="co">#&gt;  [33] "BSgenome.Dmelanogaster.UCSC.dm3"                   </span></span>
<span><span class="co">#&gt;  [34] "BSgenome.Dmelanogaster.UCSC.dm3.masked"            </span></span>
<span><span class="co">#&gt;  [35] "BSgenome.Dmelanogaster.UCSC.dm6"                   </span></span>
<span><span class="co">#&gt;  [36] "BSgenome.Drerio.UCSC.danRer10"                     </span></span>
<span><span class="co">#&gt;  [37] "BSgenome.Drerio.UCSC.danRer11"                     </span></span>
<span><span class="co">#&gt;  [38] "BSgenome.Drerio.UCSC.danRer5"                      </span></span>
<span><span class="co">#&gt;  [39] "BSgenome.Drerio.UCSC.danRer5.masked"               </span></span>
<span><span class="co">#&gt;  [40] "BSgenome.Drerio.UCSC.danRer6"                      </span></span>
<span><span class="co">#&gt;  [41] "BSgenome.Drerio.UCSC.danRer6.masked"               </span></span>
<span><span class="co">#&gt;  [42] "BSgenome.Drerio.UCSC.danRer7"                      </span></span>
<span><span class="co">#&gt;  [43] "BSgenome.Drerio.UCSC.danRer7.masked"               </span></span>
<span><span class="co">#&gt;  [44] "BSgenome.Dvirilis.Ensembl.dvircaf1"                </span></span>
<span><span class="co">#&gt;  [45] "BSgenome.Ecoli.NCBI.20080805"                      </span></span>
<span><span class="co">#&gt;  [46] "BSgenome.Gaculeatus.UCSC.gasAcu1"                  </span></span>
<span><span class="co">#&gt;  [47] "BSgenome.Gaculeatus.UCSC.gasAcu1.masked"           </span></span>
<span><span class="co">#&gt;  [48] "BSgenome.Ggallus.UCSC.galGal3"                     </span></span>
<span><span class="co">#&gt;  [49] "BSgenome.Ggallus.UCSC.galGal3.masked"              </span></span>
<span><span class="co">#&gt;  [50] "BSgenome.Ggallus.UCSC.galGal4"                     </span></span>
<span><span class="co">#&gt;  [51] "BSgenome.Ggallus.UCSC.galGal4.masked"              </span></span>
<span><span class="co">#&gt;  [52] "BSgenome.Ggallus.UCSC.galGal5"                     </span></span>
<span><span class="co">#&gt;  [53] "BSgenome.Ggallus.UCSC.galGal6"                     </span></span>
<span><span class="co">#&gt;  [54] "BSgenome.Gmax.NCBI.Gmv40"                          </span></span>
<span><span class="co">#&gt;  [55] "BSgenome.Hsapiens.1000genomes.hs37d5"              </span></span>
<span><span class="co">#&gt;  [56] "BSgenome.Hsapiens.NCBI.GRCh38"                     </span></span>
<span><span class="co">#&gt;  [57] "BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0"              </span></span>
<span><span class="co">#&gt;  [58] "BSgenome.Hsapiens.UCSC.hg17"                       </span></span>
<span><span class="co">#&gt;  [59] "BSgenome.Hsapiens.UCSC.hg17.masked"                </span></span>
<span><span class="co">#&gt;  [60] "BSgenome.Hsapiens.UCSC.hg18"                       </span></span>
<span><span class="co">#&gt;  [61] "BSgenome.Hsapiens.UCSC.hg18.masked"                </span></span>
<span><span class="co">#&gt;  [62] "BSgenome.Hsapiens.UCSC.hg19"                       </span></span>
<span><span class="co">#&gt;  [63] "BSgenome.Hsapiens.UCSC.hg19.masked"                </span></span>
<span><span class="co">#&gt;  [64] "BSgenome.Hsapiens.UCSC.hg38"                       </span></span>
<span><span class="co">#&gt;  [65] "BSgenome.Hsapiens.UCSC.hg38.dbSNP151.major"        </span></span>
<span><span class="co">#&gt;  [66] "BSgenome.Hsapiens.UCSC.hg38.dbSNP151.minor"        </span></span>
<span><span class="co">#&gt;  [67] "BSgenome.Hsapiens.UCSC.hg38.masked"                </span></span>
<span><span class="co">#&gt;  [68] "BSgenome.Hsapiens.UCSC.hs1"                        </span></span>
<span><span class="co">#&gt;  [69] "BSgenome.Mdomestica.UCSC.monDom5"                  </span></span>
<span><span class="co">#&gt;  [70] "BSgenome.Mfascicularis.NCBI.5.0"                   </span></span>
<span><span class="co">#&gt;  [71] "BSgenome.Mfascicularis.NCBI.6.0"                   </span></span>
<span><span class="co">#&gt;  [72] "BSgenome.Mfuro.UCSC.musFur1"                       </span></span>
<span><span class="co">#&gt;  [73] "BSgenome.Mmulatta.UCSC.rheMac10"                   </span></span>
<span><span class="co">#&gt;  [74] "BSgenome.Mmulatta.UCSC.rheMac2"                    </span></span>
<span><span class="co">#&gt;  [75] "BSgenome.Mmulatta.UCSC.rheMac2.masked"             </span></span>
<span><span class="co">#&gt;  [76] "BSgenome.Mmulatta.UCSC.rheMac3"                    </span></span>
<span><span class="co">#&gt;  [77] "BSgenome.Mmulatta.UCSC.rheMac3.masked"             </span></span>
<span><span class="co">#&gt;  [78] "BSgenome.Mmulatta.UCSC.rheMac8"                    </span></span>
<span><span class="co">#&gt;  [79] "BSgenome.Mmusculus.UCSC.mm10"                      </span></span>
<span><span class="co">#&gt;  [80] "BSgenome.Mmusculus.UCSC.mm10.masked"               </span></span>
<span><span class="co">#&gt;  [81] "BSgenome.Mmusculus.UCSC.mm39"                      </span></span>
<span><span class="co">#&gt;  [82] "BSgenome.Mmusculus.UCSC.mm8"                       </span></span>
<span><span class="co">#&gt;  [83] "BSgenome.Mmusculus.UCSC.mm8.masked"                </span></span>
<span><span class="co">#&gt;  [84] "BSgenome.Mmusculus.UCSC.mm9"                       </span></span>
<span><span class="co">#&gt;  [85] "BSgenome.Mmusculus.UCSC.mm9.masked"                </span></span>
<span><span class="co">#&gt;  [86] "BSgenome.Osativa.MSU.MSU7"                         </span></span>
<span><span class="co">#&gt;  [87] "BSgenome.Ppaniscus.UCSC.panPan1"                   </span></span>
<span><span class="co">#&gt;  [88] "BSgenome.Ppaniscus.UCSC.panPan2"                   </span></span>
<span><span class="co">#&gt;  [89] "BSgenome.Ptroglodytes.UCSC.panTro2"                </span></span>
<span><span class="co">#&gt;  [90] "BSgenome.Ptroglodytes.UCSC.panTro2.masked"         </span></span>
<span><span class="co">#&gt;  [91] "BSgenome.Ptroglodytes.UCSC.panTro3"                </span></span>
<span><span class="co">#&gt;  [92] "BSgenome.Ptroglodytes.UCSC.panTro3.masked"         </span></span>
<span><span class="co">#&gt;  [93] "BSgenome.Ptroglodytes.UCSC.panTro5"                </span></span>
<span><span class="co">#&gt;  [94] "BSgenome.Ptroglodytes.UCSC.panTro6"                </span></span>
<span><span class="co">#&gt;  [95] "BSgenome.Rnorvegicus.UCSC.rn4"                     </span></span>
<span><span class="co">#&gt;  [96] "BSgenome.Rnorvegicus.UCSC.rn4.masked"              </span></span>
<span><span class="co">#&gt;  [97] "BSgenome.Rnorvegicus.UCSC.rn5"                     </span></span>
<span><span class="co">#&gt;  [98] "BSgenome.Rnorvegicus.UCSC.rn5.masked"              </span></span>
<span><span class="co">#&gt;  [99] "BSgenome.Rnorvegicus.UCSC.rn6"                     </span></span>
<span><span class="co">#&gt; [100] "BSgenome.Rnorvegicus.UCSC.rn7"                     </span></span>
<span><span class="co">#&gt; [101] "BSgenome.Scerevisiae.UCSC.sacCer1"                 </span></span>
<span><span class="co">#&gt; [102] "BSgenome.Scerevisiae.UCSC.sacCer2"                 </span></span>
<span><span class="co">#&gt; [103] "BSgenome.Scerevisiae.UCSC.sacCer3"                 </span></span>
<span><span class="co">#&gt; [104] "BSgenome.Sscrofa.UCSC.susScr11"                    </span></span>
<span><span class="co">#&gt; [105] "BSgenome.Sscrofa.UCSC.susScr3"                     </span></span>
<span><span class="co">#&gt; [106] "BSgenome.Sscrofa.UCSC.susScr3.masked"              </span></span>
<span><span class="co">#&gt; [107] "BSgenome.Tgondii.ToxoDB.7.0"                       </span></span>
<span><span class="co">#&gt; [108] "BSgenome.Tguttata.UCSC.taeGut1"                    </span></span>
<span><span class="co">#&gt; [109] "BSgenome.Tguttata.UCSC.taeGut1.masked"             </span></span>
<span><span class="co">#&gt; [110] "BSgenome.Tguttata.UCSC.taeGut2"                    </span></span>
<span><span class="co">#&gt; [111] "BSgenome.Vvinifera.URGI.IGGP12Xv0"                 </span></span>
<span><span class="co">#&gt; [112] "BSgenome.Vvinifera.URGI.IGGP12Xv2"                 </span></span>
<span><span class="co">#&gt; [113] "BSgenome.Vvinifera.URGI.IGGP8X"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>从UCSC hg38中加载智人的全基因组序列，计算14号染色体GC含量</p>
<p><a href="https://bioconductor.org/packages/release/data/annotation/html/BSgenome.Hsapiens.UCSC.hg38.html">BSgenome.Hsapiens.UCSC.hg38</a></p>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg38</span><span class="op">)</span></span>
<span><span class="va">chr14_range</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="st">"chr14"</span>,</span>
<span>                      ranges <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                       end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html">seqlengths</a></span><span class="op">(</span><span class="va">Hsapiens</span><span class="op">)</span><span class="op">[</span><span class="st">"chr14"</span><span class="op">]</span><span class="op">)</span></span>
<span>                      <span class="op">)</span></span>
<span><span class="va">chr14_dna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html">getSeq</a></span><span class="op">(</span><span class="va">Hsapiens</span>, <span class="va">chr14_range</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/letterFrequency.html">letterFrequency</a></span><span class="op">(</span><span class="va">chr14_dna</span>, letters <span class="op">=</span> <span class="st">"GC"</span>, as.prob<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt;            G|C</span></span>
<span><span class="co">#&gt; [1,] 0.3454924</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="annotationdbi" class="level2" data-number="3.4"><h2 data-number="3.4" class="anchored" data-anchor-id="annotationdbi">
<span class="header-section-number">3.4</span> AnnotationDbi</h2>
<p><a href="https://bioconductor.org/packages/release/bioc/html/AnnotationDbi.html" class="uri">https://bioconductor.org/packages/release/bioc/html/AnnotationDbi.html</a></p>
<table class="caption-top table">
<thead><tr class="header">
<th style="text-align: left;">Package Name</th>
<th style="text-align: left;">Contents</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><a href="http://bioconductor.org/packages/org.Hs.eg.db/">org.Hs.eg.db</a></td>
<td style="text-align: left;">Genome wide annotation for Human；useful for mapping between gene IDs, Names, Symbols, GO and KEGG identifiers, etc.</td>
</tr>
<tr class="even">
<td style="text-align: left;">org.Mm.eg.db</td>
<td style="text-align: left;">Genome wide annotation for Mouse</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO.db</td>
<td style="text-align: left;">A set of annotation maps describing the entire Gene Ontology</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="https://bioconductor.org/packages/release/data/annotation/html/HPO.db.html">HPO.db</a></td>
<td style="text-align: left;">A set of annotation maps describing the entire Human Phenotype Ontology</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="https://bioconductor.org/packages/release/data/annotation/html/MPO.db.html">MPO.db</a></td>
<td style="text-align: left;">A set of annotation maps describing the Mouse Phenotype Ontology</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<section id="orgdb" class="level3" data-number="3.4.1"><h3 data-number="3.4.1" class="anchored" data-anchor-id="orgdb">
<span class="header-section-number">3.4.1</span> OrgDb</h3>
<p><a href="https://bioconductor.org/packages/release/data/annotation/html/org.Hs.eg.db.html">org.Hs.eg.db</a></p>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/AnnotationDbi">AnnotationDbi</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">keytypes</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span> </span>
<span><span class="co">#&gt;  [1] "ACCNUM"       "ALIAS"        "ENSEMBL"      "ENSEMBLPROT"  "ENSEMBLTRANS"</span></span>
<span><span class="co">#&gt;  [6] "ENTREZID"     "ENZYME"       "EVIDENCE"     "EVIDENCEALL"  "GENENAME"    </span></span>
<span><span class="co">#&gt; [11] "GENETYPE"     "GO"           "GOALL"        "IPI"          "MAP"         </span></span>
<span><span class="co">#&gt; [16] "OMIM"         "ONTOLOGY"     "ONTOLOGYALL"  "PATH"         "PFAM"        </span></span>
<span><span class="co">#&gt; [21] "PMID"         "PROSITE"      "REFSEQ"       "SYMBOL"       "UCSCKG"      </span></span>
<span><span class="co">#&gt; [26] "UNIPROT"</span></span>
<span><span class="co"># 等价 AnnotationDbi::columns(org.Hs.eg.db)</span></span>
<span></span>
<span><span class="co"># 示例 GeneId</span></span>
<span><span class="va">gene_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"1017"</span>, <span class="st">"1018"</span>, <span class="st">"1019"</span><span class="op">)</span>  <span class="co"># 替换为你自己的 GeneId</span></span>
<span></span>
<span><span class="co"># 进行映射</span></span>
<span><span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, </span>
<span>                       keys <span class="op">=</span> <span class="va">gene_ids</span>, </span>
<span>                       columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GENENAME"</span>,<span class="st">"SYMBOL"</span>,<span class="st">"GO"</span>,<span class="st">"ONTOLOGY"</span><span class="op">)</span>, </span>
<span>                       keytype <span class="op">=</span> <span class="st">"ENTREZID"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="kable-table">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 12%">
<col style="width: 36%">
<col style="width: 9%">
<col style="width: 15%">
<col style="width: 12%">
<col style="width: 12%">
</colgroup>
<thead><tr class="header">
<th style="text-align: left;">ENTREZID</th>
<th style="text-align: left;">GENENAME</th>
<th style="text-align: left;">SYMBOL</th>
<th style="text-align: left;">GO</th>
<th style="text-align: left;">EVIDENCE</th>
<th style="text-align: left;">ONTOLOGY</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1017</td>
<td style="text-align: left;">cyclin dependent kinase 2</td>
<td style="text-align: left;">CDK2</td>
<td style="text-align: left;">GO:0000082</td>
<td style="text-align: left;">IBA</td>
<td style="text-align: left;">BP</td>
</tr>
<tr class="even">
<td style="text-align: left;">1017</td>
<td style="text-align: left;">cyclin dependent kinase 2</td>
<td style="text-align: left;">CDK2</td>
<td style="text-align: left;">GO:0000082</td>
<td style="text-align: left;">NAS</td>
<td style="text-align: left;">BP</td>
</tr>
<tr class="odd">
<td style="text-align: left;">1017</td>
<td style="text-align: left;">cyclin dependent kinase 2</td>
<td style="text-align: left;">CDK2</td>
<td style="text-align: left;">GO:0000086</td>
<td style="text-align: left;">NAS</td>
<td style="text-align: left;">BP</td>
</tr>
<tr class="even">
<td style="text-align: left;">1017</td>
<td style="text-align: left;">cyclin dependent kinase 2</td>
<td style="text-align: left;">CDK2</td>
<td style="text-align: left;">GO:0000122</td>
<td style="text-align: left;">IEA</td>
<td style="text-align: left;">BP</td>
</tr>
<tr class="odd">
<td style="text-align: left;">1017</td>
<td style="text-align: left;">cyclin dependent kinase 2</td>
<td style="text-align: left;">CDK2</td>
<td style="text-align: left;">GO:0000287</td>
<td style="text-align: left;">IEA</td>
<td style="text-align: left;">MF</td>
</tr>
<tr class="even">
<td style="text-align: left;">1017</td>
<td style="text-align: left;">cyclin dependent kinase 2</td>
<td style="text-align: left;">CDK2</td>
<td style="text-align: left;">GO:0000307</td>
<td style="text-align: left;">IBA</td>
<td style="text-align: left;">CC</td>
</tr>
</tbody>
</table>
</div>
</div>
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span></span>
<span></span>
<span><span class="co"># mapIds can only use one column.</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">mapIds</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, </span>
<span>     keys <span class="op">=</span> <span class="va">gene_ids</span>, </span>
<span>     column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GENENAME"</span><span class="op">)</span>, </span>
<span>     keytype <span class="op">=</span> <span class="st">"ENTREZID"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="kable-table">
<table class="caption-top table table-sm table-striped small">
<thead><tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">.</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1017</td>
<td style="text-align: left;">cyclin dependent kinase 2</td>
</tr>
<tr class="even">
<td style="text-align: left;">1018</td>
<td style="text-align: left;">cyclin dependent kinase 3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">1019</td>
<td style="text-align: left;">cyclin dependent kinase 4</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section></section><section id="experimenthub" class="level2" data-number="3.5"><h2 data-number="3.5" class="anchored" data-anchor-id="experimenthub">
<span class="header-section-number">3.5</span> ExperimentHub</h2>


<!-- -->

</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/wanganlin00\.github\.io\/Bioinformatics\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./biodata_structure.html" class="pagination-link" aria-label="生物数据结构">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">生物数据结构</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./ExpressionSet.html" class="pagination-link" aria-label="`ExpressionSet`">
        <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title"><code>ExpressionSet</code></span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb25" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span><span class="co"> </span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="co">  cache: true</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="fu"># 注释数据库</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://bioconductor.org/packages/release/workflows/html/annotation.html&gt;</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="fu">## [BioMart](https://bioconductor.org/packages/release/bioc/html/biomaRt.html)</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>即Ensembl ID</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>查找智人（homo sapien）基因组</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://bioconductor.org/packages/release/bioc/vignettes/biomaRt/inst/doc/accessing_ensembl.html#introduction&gt;</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("biomaRt")</span></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a><span class="fu">listEnsembl</span>()</span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a><span class="co"># 连接到Ensembl BioMart</span></span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> <span class="fu">useEnsembl</span>(<span class="at">biomart =</span> <span class="st">"genes"</span>)</span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>ensembl</span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a><span class="co"># 列出数据集</span></span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a><span class="fu">listDatasets</span>(ensembl) <span class="sc">|&gt;</span> <span class="fu">as_tibble</span>()</span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a><span class="co"># 查找智人</span></span>
<span id="cb25-34"><a href="#cb25-34" aria-hidden="true" tabindex="-1"></a><span class="fu">searchDatasets</span>(<span class="at">mart =</span> ensembl, <span class="at">pattern =</span> <span class="st">"hsapiens"</span>)</span>
<span id="cb25-35"><a href="#cb25-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-36"><a href="#cb25-36" aria-hidden="true" tabindex="-1"></a><span class="co"># 选择数据集</span></span>
<span id="cb25-37"><a href="#cb25-37" aria-hidden="true" tabindex="-1"></a>dataset <span class="ot">&lt;-</span> <span class="fu">useDataset</span>(<span class="at">dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>, <span class="at">mart =</span> ensembl)</span>
<span id="cb25-38"><a href="#cb25-38" aria-hidden="true" tabindex="-1"></a>dataset</span>
<span id="cb25-39"><a href="#cb25-39" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-40"><a href="#cb25-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-41"><a href="#cb25-41" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">GRCh38（Genome Research Consortium human genome build 38）</span><span class="co">](https://asia.ensembl.org/index.html)</span></span>
<span id="cb25-42"><a href="#cb25-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-45"><a href="#cb25-45" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb25-46"><a href="#cb25-46" aria-hidden="true" tabindex="-1"></a><span class="co"># 人类基因BioMart</span></span>
<span id="cb25-47"><a href="#cb25-47" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> <span class="fu">useEnsembl</span>(<span class="at">biomart =</span> <span class="st">"ensembl"</span>, </span>
<span id="cb25-48"><a href="#cb25-48" aria-hidden="true" tabindex="-1"></a>                   <span class="at">dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>, </span>
<span id="cb25-49"><a href="#cb25-49" aria-hidden="true" tabindex="-1"></a>                   <span class="at">mirror =</span> <span class="st">"asia"</span>)</span>
<span id="cb25-50"><a href="#cb25-50" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-51"><a href="#cb25-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-52"><a href="#cb25-52" aria-hidden="true" tabindex="-1"></a><span class="fu">### 映射</span></span>
<span id="cb25-53"><a href="#cb25-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-56"><a href="#cb25-56" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb25-57"><a href="#cb25-57" aria-hidden="true" tabindex="-1"></a><span class="fu">listMarts</span>()</span>
<span id="cb25-58"><a href="#cb25-58" aria-hidden="true" tabindex="-1"></a><span class="co"># 连接到Ensembl BioMart</span></span>
<span id="cb25-59"><a href="#cb25-59" aria-hidden="true" tabindex="-1"></a>mart <span class="ot">&lt;-</span> <span class="fu">useMart</span>(<span class="st">"ENSEMBL_MART_ENSEMBL"</span>, <span class="at">dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>)</span>
<span id="cb25-60"><a href="#cb25-60" aria-hidden="true" tabindex="-1"></a><span class="fu">listAttributes</span>(mart) <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">datatable</span>()</span>
<span id="cb25-61"><a href="#cb25-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-62"><a href="#cb25-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-63"><a href="#cb25-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-64"><a href="#cb25-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-65"><a href="#cb25-65" aria-hidden="true" tabindex="-1"></a>gene_ids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"1017"</span>, <span class="st">"1018"</span>, <span class="st">"1019"</span>)  </span>
<span id="cb25-66"><a href="#cb25-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-67"><a href="#cb25-67" aria-hidden="true" tabindex="-1"></a><span class="fu">getBM</span>(<span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">"entrezgene_id"</span>, <span class="st">"ensembl_gene_id"</span>, <span class="st">"external_gene_name"</span>),</span>
<span id="cb25-68"><a href="#cb25-68" aria-hidden="true" tabindex="-1"></a>                  <span class="at">filters =</span> <span class="st">"entrezgene_id"</span>,</span>
<span id="cb25-69"><a href="#cb25-69" aria-hidden="true" tabindex="-1"></a>                  <span class="at">values =</span> gene_ids,</span>
<span id="cb25-70"><a href="#cb25-70" aria-hidden="true" tabindex="-1"></a>                  <span class="at">mart =</span> mart)</span>
<span id="cb25-71"><a href="#cb25-71" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb25-72"><a href="#cb25-72" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-73"><a href="#cb25-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-74"><a href="#cb25-74" aria-hidden="true" tabindex="-1"></a><span class="fu">### **使用 Ensembl 的存档版本**</span></span>
<span id="cb25-75"><a href="#cb25-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-76"><a href="#cb25-76" aria-hidden="true" tabindex="-1"></a><span class="in">```{r eval=FALSE}</span></span>
<span id="cb25-77"><a href="#cb25-77" aria-hidden="true" tabindex="-1"></a><span class="in">listEnsemblArchives()</span></span>
<span id="cb25-78"><a href="#cb25-78" aria-hidden="true" tabindex="-1"></a><span class="in">listEnsembl(version = 112)</span></span>
<span id="cb25-79"><a href="#cb25-79" aria-hidden="true" tabindex="-1"></a><span class="in">ensembl_112 &lt;- useEnsembl(biomart = 'genes', </span></span>
<span id="cb25-80"><a href="#cb25-80" aria-hidden="true" tabindex="-1"></a><span class="in">                       dataset = 'hsapiens_gene_ensembl',</span></span>
<span id="cb25-81"><a href="#cb25-81" aria-hidden="true" tabindex="-1"></a><span class="in">                       mirror = "asia",</span></span>
<span id="cb25-82"><a href="#cb25-82" aria-hidden="true" tabindex="-1"></a><span class="in">                       version = 112)</span></span>
<span id="cb25-83"><a href="#cb25-83" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-84"><a href="#cb25-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-85"><a href="#cb25-85" aria-hidden="true" tabindex="-1"></a><span class="fu">### **使用 Ensembl 基因组**</span></span>
<span id="cb25-86"><a href="#cb25-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-89"><a href="#cb25-89" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb25-90"><a href="#cb25-90" aria-hidden="true" tabindex="-1"></a><span class="fu">listEnsemblGenomes</span>()</span>
<span id="cb25-91"><a href="#cb25-91" aria-hidden="true" tabindex="-1"></a>ensembl_plants <span class="ot">&lt;-</span> <span class="fu">useEnsemblGenomes</span>(<span class="at">biomart =</span> <span class="st">"plants_mart"</span>)</span>
<span id="cb25-92"><a href="#cb25-92" aria-hidden="true" tabindex="-1"></a><span class="fu">searchDatasets</span>(ensembl_plants, <span class="at">pattern =</span> <span class="st">"Arabidopsis"</span>)</span>
<span id="cb25-93"><a href="#cb25-93" aria-hidden="true" tabindex="-1"></a>ensembl_arabidopsis <span class="ot">&lt;-</span> <span class="fu">useEnsemblGenomes</span>(<span class="at">biomart =</span> <span class="st">"plants_mart"</span>, </span>
<span id="cb25-94"><a href="#cb25-94" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">dataset =</span> <span class="st">"athaliana_eg_gene"</span>)</span>
<span id="cb25-95"><a href="#cb25-95" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-96"><a href="#cb25-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-97"><a href="#cb25-97" aria-hidden="true" tabindex="-1"></a><span class="fu">## [AnnotationHub](https://bioconductor.org/packages/release/bioc/html/AnnotationHub.html)</span></span>
<span id="cb25-98"><a href="#cb25-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-99"><a href="#cb25-99" aria-hidden="true" tabindex="-1"></a><span class="in">```{r eval=FALSE, include=FALSE}</span></span>
<span id="cb25-100"><a href="#cb25-100" aria-hidden="true" tabindex="-1"></a><span class="in">if (!require("AnnotationHub", quietly = TRUE))</span></span>
<span id="cb25-101"><a href="#cb25-101" aria-hidden="true" tabindex="-1"></a><span class="in">  BiocManager::install("AnnotationHub")</span></span>
<span id="cb25-102"><a href="#cb25-102" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-103"><a href="#cb25-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-104"><a href="#cb25-104" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">基于annotationhub构建所需orgdb，用于clusterprofiler富集分析</span><span class="co">](https://www.jingege.wang/2023/12/12/%E5%9F%BA%E4%BA%8Eannotationhub%E6%9E%84%E5%BB%BA%E6%89%80%E9%9C%80orgdb%EF%BC%8C%E7%94%A8%E4%BA%8Eclusterprofiler%E5%AF%8C%E9%9B%86%E5%88%86%E6%9E%90/)</span></span>
<span id="cb25-105"><a href="#cb25-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-108"><a href="#cb25-108" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb25-109"><a href="#cb25-109" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(AnnotationHub))</span>
<span id="cb25-110"><a href="#cb25-110" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(BiocGenerics<span class="sc">::</span>setdiff)</span>
<span id="cb25-111"><a href="#cb25-111" aria-hidden="true" tabindex="-1"></a><span class="co"># proxy &lt;- httr::use_proxy(Sys.getenv('http_proxy'))</span></span>
<span id="cb25-112"><a href="#cb25-112" aria-hidden="true" tabindex="-1"></a><span class="co"># httr::set_config(proxy)</span></span>
<span id="cb25-113"><a href="#cb25-113" aria-hidden="true" tabindex="-1"></a><span class="co"># AnnotationHub::setAnnotationHubOption("PROXY", proxy)</span></span>
<span id="cb25-114"><a href="#cb25-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-115"><a href="#cb25-115" aria-hidden="true" tabindex="-1"></a>ah <span class="ot">&lt;-</span> <span class="fu">AnnotationHub</span>(<span class="at">cache =</span> <span class="st">"D:/AnnotationHub"</span>)</span>
<span id="cb25-116"><a href="#cb25-116" aria-hidden="true" tabindex="-1"></a>ah</span>
<span id="cb25-117"><a href="#cb25-117" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-118"><a href="#cb25-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-121"><a href="#cb25-121" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb25-122"><a href="#cb25-122" aria-hidden="true" tabindex="-1"></a><span class="co"># 当前可用数据的更新时间</span></span>
<span id="cb25-123"><a href="#cb25-123" aria-hidden="true" tabindex="-1"></a><span class="fu">snapshotDate</span>(ah)</span>
<span id="cb25-124"><a href="#cb25-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-125"><a href="#cb25-125" aria-hidden="true" tabindex="-1"></a><span class="co"># 缓存路径</span></span>
<span id="cb25-126"><a href="#cb25-126" aria-hidden="true" tabindex="-1"></a><span class="fu">hubCache</span>(ah)</span>
<span id="cb25-127"><a href="#cb25-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-128"><a href="#cb25-128" aria-hidden="true" tabindex="-1"></a><span class="co"># https://annotationhub.bioconductor.org</span></span>
<span id="cb25-129"><a href="#cb25-129" aria-hidden="true" tabindex="-1"></a><span class="fu">hubUrl</span>(ah)</span>
<span id="cb25-130"><a href="#cb25-130" aria-hidden="true" tabindex="-1"></a><span class="fu">isLocalHub</span>(ah)</span>
<span id="cb25-131"><a href="#cb25-131" aria-hidden="true" tabindex="-1"></a><span class="fu">dbconn</span>(ah)</span>
<span id="cb25-132"><a href="#cb25-132" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-133"><a href="#cb25-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-136"><a href="#cb25-136" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb25-137"><a href="#cb25-137" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(ah)</span>
<span id="cb25-138"><a href="#cb25-138" aria-hidden="true" tabindex="-1"></a>ah<span class="sc">$</span>dataprovider <span class="sc">|&gt;</span> <span class="fu">unique</span>()</span>
<span id="cb25-139"><a href="#cb25-139" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-140"><a href="#cb25-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-143"><a href="#cb25-143" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb25-144"><a href="#cb25-144" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(ah<span class="sc">$</span>species) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">datatable</span>()</span>
<span id="cb25-145"><a href="#cb25-145" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(ah<span class="sc">$</span>rdataclass) </span>
<span id="cb25-146"><a href="#cb25-146" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-147"><a href="#cb25-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-148"><a href="#cb25-148" aria-hidden="true" tabindex="-1"></a><span class="fu">### 子集</span></span>
<span id="cb25-149"><a href="#cb25-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-152"><a href="#cb25-152" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb25-153"><a href="#cb25-153" aria-hidden="true" tabindex="-1"></a>ah[ah<span class="sc">$</span>dataprovider<span class="sc">==</span><span class="st">"UCSC"</span>]</span>
<span id="cb25-154"><a href="#cb25-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-155"><a href="#cb25-155" aria-hidden="true" tabindex="-1"></a><span class="fu">subset</span>(ah, species <span class="sc">==</span> <span class="st">"homo sapiens"</span>)</span>
<span id="cb25-156"><a href="#cb25-156" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-157"><a href="#cb25-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-158"><a href="#cb25-158" aria-hidden="true" tabindex="-1"></a><span class="fu">### 查询</span></span>
<span id="cb25-159"><a href="#cb25-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-162"><a href="#cb25-162" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb25-163"><a href="#cb25-163" aria-hidden="true" tabindex="-1"></a><span class="fu">query</span>(ah, <span class="fu">c</span>(<span class="st">"Ensembl"</span>))</span>
<span id="cb25-164"><a href="#cb25-164" aria-hidden="true" tabindex="-1"></a><span class="fu">query</span>(ah, <span class="fu">c</span>(<span class="st">"Gencode"</span>))</span>
<span id="cb25-165"><a href="#cb25-165" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-166"><a href="#cb25-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-167"><a href="#cb25-167" aria-hidden="true" tabindex="-1"></a><span class="in">```{r eval=FALSE}</span></span>
<span id="cb25-168"><a href="#cb25-168" aria-hidden="true" tabindex="-1"></a><span class="in"># Web界面</span></span>
<span id="cb25-169"><a href="#cb25-169" aria-hidden="true" tabindex="-1"></a><span class="in">BiocHubsShiny::BiocHubsShiny()</span></span>
<span id="cb25-170"><a href="#cb25-170" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-171"><a href="#cb25-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-172"><a href="#cb25-172" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/BiocHubsShiny.png)</span>{fig-align="center" width="80%"}</span>
<span id="cb25-173"><a href="#cb25-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-174"><a href="#cb25-174" aria-hidden="true" tabindex="-1"></a><span class="fu">## BSgenome</span></span>
<span id="cb25-175"><a href="#cb25-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-178"><a href="#cb25-178" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb25-179"><a href="#cb25-179" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BSgenome)</span>
<span id="cb25-180"><a href="#cb25-180" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(Biostrings<span class="sc">::</span>setdiff)</span>
<span id="cb25-181"><a href="#cb25-181" aria-hidden="true" tabindex="-1"></a><span class="fu">available.genomes</span>()</span>
<span id="cb25-182"><a href="#cb25-182" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-183"><a href="#cb25-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-184"><a href="#cb25-184" aria-hidden="true" tabindex="-1"></a>从UCSC hg38中加载智人的全基因组序列，计算14号染色体GC含量</span>
<span id="cb25-185"><a href="#cb25-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-186"><a href="#cb25-186" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">BSgenome.Hsapiens.UCSC.hg38</span><span class="co">](https://bioconductor.org/packages/release/data/annotation/html/BSgenome.Hsapiens.UCSC.hg38.html)</span></span>
<span id="cb25-187"><a href="#cb25-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-188"><a href="#cb25-188" aria-hidden="true" tabindex="-1"></a><span class="in">```{r eval=FALSE, include=FALSE}</span></span>
<span id="cb25-189"><a href="#cb25-189" aria-hidden="true" tabindex="-1"></a><span class="in">install.packages("D:/.GitHub/_packages/BSgenome.Hsapiens.UCSC.hg38_1.4.5.tar.gz", repos = NULL, type = "source")</span></span>
<span id="cb25-190"><a href="#cb25-190" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-191"><a href="#cb25-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-194"><a href="#cb25-194" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb25-195"><a href="#cb25-195" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BSgenome.Hsapiens.UCSC.hg38)</span>
<span id="cb25-196"><a href="#cb25-196" aria-hidden="true" tabindex="-1"></a>chr14_range <span class="ot">=</span> <span class="fu">GRanges</span>(<span class="at">seqnames =</span> <span class="st">"chr14"</span>,</span>
<span id="cb25-197"><a href="#cb25-197" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ranges =</span>  <span class="fu">IRanges</span>(<span class="at">start =</span> <span class="dv">1</span>, </span>
<span id="cb25-198"><a href="#cb25-198" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">end =</span> <span class="fu">seqlengths</span>(Hsapiens)[<span class="st">"chr14"</span>])</span>
<span id="cb25-199"><a href="#cb25-199" aria-hidden="true" tabindex="-1"></a>                      )</span>
<span id="cb25-200"><a href="#cb25-200" aria-hidden="true" tabindex="-1"></a>chr14_dna <span class="ot">&lt;-</span> <span class="fu">getSeq</span>(Hsapiens, chr14_range)</span>
<span id="cb25-201"><a href="#cb25-201" aria-hidden="true" tabindex="-1"></a><span class="fu">letterFrequency</span>(chr14_dna, <span class="at">letters =</span> <span class="st">"GC"</span>, <span class="at">as.prob=</span><span class="cn">TRUE</span>)</span>
<span id="cb25-202"><a href="#cb25-202" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-203"><a href="#cb25-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-204"><a href="#cb25-204" aria-hidden="true" tabindex="-1"></a><span class="fu">## AnnotationDbi</span></span>
<span id="cb25-205"><a href="#cb25-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-206"><a href="#cb25-206" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://bioconductor.org/packages/release/bioc/html/AnnotationDbi.html&gt;</span></span>
<span id="cb25-207"><a href="#cb25-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-208"><a href="#cb25-208" aria-hidden="true" tabindex="-1"></a>| Package Name | Contents |</span>
<span id="cb25-209"><a href="#cb25-209" aria-hidden="true" tabindex="-1"></a>|:------------------------------|:----------------------------------------|</span>
<span id="cb25-210"><a href="#cb25-210" aria-hidden="true" tabindex="-1"></a>| <span class="co">[</span><span class="ot">org.Hs.eg.db</span><span class="co">](http://bioconductor.org/packages/org.Hs.eg.db/)</span> | Genome wide annotation for Human；useful for mapping between gene IDs, Names, Symbols, GO and KEGG identifiers, etc. |</span>
<span id="cb25-211"><a href="#cb25-211" aria-hidden="true" tabindex="-1"></a>| org.Mm.eg.db | Genome wide annotation for Mouse |</span>
<span id="cb25-212"><a href="#cb25-212" aria-hidden="true" tabindex="-1"></a>| GO.db | A set of annotation maps describing the entire Gene Ontology |</span>
<span id="cb25-213"><a href="#cb25-213" aria-hidden="true" tabindex="-1"></a>| <span class="co">[</span><span class="ot">HPO.db</span><span class="co">](https://bioconductor.org/packages/release/data/annotation/html/HPO.db.html)</span> | A set of annotation maps describing the entire Human Phenotype Ontology |</span>
<span id="cb25-214"><a href="#cb25-214" aria-hidden="true" tabindex="-1"></a>| <span class="co">[</span><span class="ot">MPO.db</span><span class="co">](https://bioconductor.org/packages/release/data/annotation/html/MPO.db.html)</span> | A set of annotation maps describing the Mouse Phenotype Ontology |</span>
<span id="cb25-215"><a href="#cb25-215" aria-hidden="true" tabindex="-1"></a>|  |  |</span>
<span id="cb25-216"><a href="#cb25-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-217"><a href="#cb25-217" aria-hidden="true" tabindex="-1"></a><span class="fu">### OrgDb</span></span>
<span id="cb25-218"><a href="#cb25-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-219"><a href="#cb25-219" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">org.Hs.eg.db</span><span class="co">](https://bioconductor.org/packages/release/data/annotation/html/org.Hs.eg.db.html)</span></span>
<span id="cb25-220"><a href="#cb25-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-221"><a href="#cb25-221" aria-hidden="true" tabindex="-1"></a><span class="in">```{r eval=FALSE, include=FALSE}</span></span>
<span id="cb25-222"><a href="#cb25-222" aria-hidden="true" tabindex="-1"></a><span class="in">install.packages("D:/.GitHub/_packages/org.Hs.eg.db_3.20.0.tar.gz", </span></span>
<span id="cb25-223"><a href="#cb25-223" aria-hidden="true" tabindex="-1"></a><span class="in">                 repos = NULL, type = "source")</span></span>
<span id="cb25-224"><a href="#cb25-224" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-225"><a href="#cb25-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-228"><a href="#cb25-228" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb25-229"><a href="#cb25-229" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi)</span>
<span id="cb25-230"><a href="#cb25-230" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb25-231"><a href="#cb25-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-232"><a href="#cb25-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-233"><a href="#cb25-233" aria-hidden="true" tabindex="-1"></a><span class="fu">keytypes</span>(org.Hs.eg.db) </span>
<span id="cb25-234"><a href="#cb25-234" aria-hidden="true" tabindex="-1"></a><span class="co"># 等价 AnnotationDbi::columns(org.Hs.eg.db)</span></span>
<span id="cb25-235"><a href="#cb25-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-236"><a href="#cb25-236" aria-hidden="true" tabindex="-1"></a><span class="co"># 示例 GeneId</span></span>
<span id="cb25-237"><a href="#cb25-237" aria-hidden="true" tabindex="-1"></a>gene_ids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"1017"</span>, <span class="st">"1018"</span>, <span class="st">"1019"</span>)  <span class="co"># 替换为你自己的 GeneId</span></span>
<span id="cb25-238"><a href="#cb25-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-239"><a href="#cb25-239" aria-hidden="true" tabindex="-1"></a><span class="co"># 进行映射</span></span>
<span id="cb25-240"><a href="#cb25-240" aria-hidden="true" tabindex="-1"></a>AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(org.Hs.eg.db, </span>
<span id="cb25-241"><a href="#cb25-241" aria-hidden="true" tabindex="-1"></a>                       <span class="at">keys =</span> gene_ids, </span>
<span id="cb25-242"><a href="#cb25-242" aria-hidden="true" tabindex="-1"></a>                       <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"GENENAME"</span>,<span class="st">"SYMBOL"</span>,<span class="st">"GO"</span>,<span class="st">"ONTOLOGY"</span>), </span>
<span id="cb25-243"><a href="#cb25-243" aria-hidden="true" tabindex="-1"></a>                       <span class="at">keytype =</span> <span class="st">"ENTREZID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-244"><a href="#cb25-244" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>()</span>
<span id="cb25-245"><a href="#cb25-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-246"><a href="#cb25-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-247"><a href="#cb25-247" aria-hidden="true" tabindex="-1"></a><span class="co"># mapIds can only use one column.</span></span>
<span id="cb25-248"><a href="#cb25-248" aria-hidden="true" tabindex="-1"></a><span class="fu">mapIds</span>(org.Hs.eg.db, </span>
<span id="cb25-249"><a href="#cb25-249" aria-hidden="true" tabindex="-1"></a>     <span class="at">keys =</span> gene_ids, </span>
<span id="cb25-250"><a href="#cb25-250" aria-hidden="true" tabindex="-1"></a>     <span class="at">column =</span> <span class="fu">c</span>(<span class="st">"GENENAME"</span>), </span>
<span id="cb25-251"><a href="#cb25-251" aria-hidden="true" tabindex="-1"></a>     <span class="at">keytype =</span> <span class="st">"ENTREZID"</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb25-252"><a href="#cb25-252" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb25-253"><a href="#cb25-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-254"><a href="#cb25-254" aria-hidden="true" tabindex="-1"></a><span class="fu">## ExperimentHub</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/Wanganlin00/Bioinformatics/edit/main/annotation_database.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/Wanganlin00/Bioinformatics/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div></div></footer></body></html>