<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Bioinformatics - 4&nbsp; 分析概述</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./basics.html" rel="next">
<link href="./SCE.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
</head>
<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./index.html">Single-cell Analysis</a></li><li class="breadcrumb-item"><a href="./analysis.html"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">分析概述</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Bioinformatics</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/Wanganlin00/Bioinformatics" rel="" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./genomics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title"><code>SummarizeExperiment</code>结构</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Single-cell Analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">前言</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./import.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">获取scRNA-seq数据集</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./SCE.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title"><code>SingleCellExperiment</code>结构</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./analysis.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">分析概述</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./basics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Basics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./QC.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">质量控制</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./normalization.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Normalization</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./feature_selection.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">特征选择</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Seurat</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Seurat.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Seurat</span></span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li><a href="#outline" id="toc-outline" class="nav-link active" data-scroll-target="#outline"><span class="header-section-number">4.1</span> outline</a></li>
  <li><a href="#%E5%BF%AB%E9%80%9F%E5%85%A5%E9%97%A8%E7%AE%80%E5%8D%95" id="toc-快速入门简单" class="nav-link" data-scroll-target="#%E5%BF%AB%E9%80%9F%E5%85%A5%E9%97%A8%E7%AE%80%E5%8D%95"><span class="header-section-number">4.2</span> 快速入门（简单）</a></li>
  <li><a href="#%E5%BF%AB%E9%80%9F%E5%BC%80%E5%A7%8B%E5%A4%9A%E6%89%B9%E6%AC%A1" id="toc-快速开始多批次" class="nav-link" data-scroll-target="#%E5%BF%AB%E9%80%9F%E5%BC%80%E5%A7%8B%E5%A4%9A%E6%89%B9%E6%AC%A1"><span class="header-section-number">4.3</span> 快速开始（多批次）</a></li>
  </ul><div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/Wanganlin00/Bioinformatics/edit/main/analysis.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/Wanganlin00/Bioinformatics/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">
<span class="chapter-number">4</span>&nbsp; <span class="chapter-title">分析概述</span>
</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header><section id="outline" class="level2" data-number="4.1"><h2 data-number="4.1" class="anchored" data-anchor-id="outline">
<span class="header-section-number">4.1</span> outline</h2>
<p>本章概述了典型scRNA-seq分析工作流程的框架（<a href="#fig-workflow">图&nbsp;<span class="quarto-unresolved-ref">fig-workflow</span></a>）</p>
<div id="fig-workflow" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="images/workflow.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">图&nbsp;4.1: 典型 scRNA-seq 分析工作流程示意图。每个阶段（用虚线分隔）由许多特定步骤组成，其中许多步骤对实例进行操作和修改</figcaption></figure>
</div>
</section><section id="快速入门简单" class="level2" data-number="4.2"><h2 data-number="4.2" class="anchored" data-anchor-id="快速入门简单">
<span class="header-section-number">4.2</span> 快速入门（简单）</h2>
<p>原书使用来自Macosko等人（2015）的基于液滴的视网膜数据集，该数据集在scRNAseq包中提供。</p>
<p>从计数矩阵开始，以聚类结束( <a href="#fig-workflow">图&nbsp;<span class="quarto-unresolved-ref">fig-workflow</span></a> )。</p>
<div class="cell" data-hash="analysis_cache/html/unnamed-chunk-2_30103117c51a1521914de034e2c9afab">
<details><summary>数据集信息</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">scRNAseq</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scRNAseq/man/MacoskoRetinaData.html">MacoskoRetinaData</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">sce</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 24658 49300 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(24658): KITL TMTC3 ... 1110059M19RIK GM20861</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(49300): r1_GGCCGCAGTCCG r1_CTTGTGCGGGAA ... p1_TAACGCGCTCCT</span></span>
<span><span class="co">#&gt;   p1_ATTCTTGTTCTT</span></span>
<span><span class="co">#&gt; colData names(2): cell.id cluster</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">assay</span><span class="op">(</span><span class="va">sce</span>, <span class="st">"counts"</span><span class="op">)</span></span>
<span><span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 49300 rows and 2 columns</span></span>
<span><span class="co">#&gt;                         cell.id   cluster</span></span>
<span><span class="co">#&gt;                     &lt;character&gt; &lt;integer&gt;</span></span>
<span><span class="co">#&gt; r1_GGCCGCAGTCCG r1_GGCCGCAGTCCG         2</span></span>
<span><span class="co">#&gt; r1_CTTGTGCGGGAA r1_CTTGTGCGGGAA         2</span></span>
<span><span class="co">#&gt; r1_GCGCAACTGCTC r1_GCGCAACTGCTC         2</span></span>
<span><span class="co">#&gt; r1_GATTGGGAGGCA r1_GATTGGGAGGCA         2</span></span>
<span><span class="co">#&gt; r1_CCTCCTAGTTGG r1_CCTCCTAGTTGG        NA</span></span>
<span><span class="co">#&gt; ...                         ...       ...</span></span>
<span><span class="co">#&gt; p1_TCAAAAGCCGGG p1_TCAAAAGCCGGG        24</span></span>
<span><span class="co">#&gt; p1_ATTAAGTTCCAA p1_ATTAAGTTCCAA        34</span></span>
<span><span class="co">#&gt; p1_CTGTCTGAGACC p1_CTGTCTGAGACC         2</span></span>
<span><span class="co">#&gt; p1_TAACGCGCTCCT p1_TAACGCGCTCCT        24</span></span>
<span><span class="co">#&gt; p1_ATTCTTGTTCTT p1_ATTCTTGTTCTT        24</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="analysis_cache/html/unnamed-chunk-3_81adb4d8832fdc8d5732a800ee3d258f">
<details><summary>质量控制</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Quality control (using mitochondrial genes)线粒体基因</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/">scater</a></span><span class="op">)</span></span>
<span><span class="va">is.mito</span> <span class="op">&lt;-</span> <span class="fu">str_detect</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>,<span class="st">"^MT-"</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">is.mito</span><span class="op">)</span> <span class="co"># 31个</span></span>
<span><span class="co">#&gt; [1] 31</span></span>
<span></span>
<span><span class="co">#?perCellQCMetrics</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/perCellQCMetrics.html">perCellQCMetrics</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 49300 rows and 3 columns</span></span>
<span><span class="co">#&gt;                       sum  detected     total</span></span>
<span><span class="co">#&gt;                 &lt;numeric&gt; &lt;integer&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; r1_GGCCGCAGTCCG     37487      7243     37487</span></span>
<span><span class="co">#&gt; r1_CTTGTGCGGGAA     32047      6933     32047</span></span>
<span><span class="co">#&gt; r1_GCGCAACTGCTC     28148      6397     28148</span></span>
<span><span class="co">#&gt; r1_GATTGGGAGGCA     20367      5740     20367</span></span>
<span><span class="co">#&gt; r1_CCTCCTAGTTGG     19560      5779     19560</span></span>
<span><span class="co">#&gt; ...                   ...       ...       ...</span></span>
<span><span class="co">#&gt; p1_TCAAAAGCCGGG       817       537       817</span></span>
<span><span class="co">#&gt; p1_ATTAAGTTCCAA       817       574       817</span></span>
<span><span class="co">#&gt; p1_CTGTCTGAGACC       817       637       817</span></span>
<span><span class="co">#&gt; p1_TAACGCGCTCCT       816       488       816</span></span>
<span><span class="co">#&gt; p1_ATTCTTGTTCTT       816       484       816</span></span>
<span><span class="va">qcstats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/perCellQCMetrics.html">perCellQCMetrics</a></span><span class="op">(</span><span class="va">sce</span>, subsets<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>Mito<span class="op">=</span><span class="va">is.mito</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">qcstats</span></span>
<span><span class="co">#&gt; DataFrame with 49300 rows and 6 columns</span></span>
<span><span class="co">#&gt;                       sum  detected subsets_Mito_sum subsets_Mito_detected</span></span>
<span><span class="co">#&gt;                 &lt;numeric&gt; &lt;integer&gt;        &lt;numeric&gt;             &lt;integer&gt;</span></span>
<span><span class="co">#&gt; r1_GGCCGCAGTCCG     37487      7243              427                    14</span></span>
<span><span class="co">#&gt; r1_CTTGTGCGGGAA     32047      6933              503                    15</span></span>
<span><span class="co">#&gt; r1_GCGCAACTGCTC     28148      6397              460                    13</span></span>
<span><span class="co">#&gt; r1_GATTGGGAGGCA     20367      5740              326                    11</span></span>
<span><span class="co">#&gt; r1_CCTCCTAGTTGG     19560      5779              264                     9</span></span>
<span><span class="co">#&gt; ...                   ...       ...              ...                   ...</span></span>
<span><span class="co">#&gt; p1_TCAAAAGCCGGG       817       537               13                     4</span></span>
<span><span class="co">#&gt; p1_ATTAAGTTCCAA       817       574               10                     5</span></span>
<span><span class="co">#&gt; p1_CTGTCTGAGACC       817       637               24                     7</span></span>
<span><span class="co">#&gt; p1_TAACGCGCTCCT       816       488               27                     5</span></span>
<span><span class="co">#&gt; p1_ATTCTTGTTCTT       816       484               16                     4</span></span>
<span><span class="co">#&gt;                 subsets_Mito_percent     total</span></span>
<span><span class="co">#&gt;                            &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; r1_GGCCGCAGTCCG              1.13906     37487</span></span>
<span><span class="co">#&gt; r1_CTTGTGCGGGAA              1.56957     32047</span></span>
<span><span class="co">#&gt; r1_GCGCAACTGCTC              1.63422     28148</span></span>
<span><span class="co">#&gt; r1_GATTGGGAGGCA              1.60063     20367</span></span>
<span><span class="co">#&gt; r1_CCTCCTAGTTGG              1.34969     19560</span></span>
<span><span class="co">#&gt; ...                              ...       ...</span></span>
<span><span class="co">#&gt; p1_TCAAAAGCCGGG              1.59119       817</span></span>
<span><span class="co">#&gt; p1_ATTAAGTTCCAA              1.22399       817</span></span>
<span><span class="co">#&gt; p1_CTGTCTGAGACC              2.93758       817</span></span>
<span><span class="co">#&gt; p1_TAACGCGCTCCT              3.30882       816</span></span>
<span><span class="co">#&gt; p1_ATTCTTGTTCTT              1.96078       816</span></span>
<span></span>
<span><span class="co">#?quickPerCellQC</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/quickPerCellQC.html">quickPerCellQC</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">colData</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 49287 rows and 8 columns</span></span>
<span><span class="co">#&gt;                         cell.id   cluster       sum  detected     total</span></span>
<span><span class="co">#&gt;                     &lt;character&gt; &lt;integer&gt; &lt;numeric&gt; &lt;integer&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; r1_GGCCGCAGTCCG r1_GGCCGCAGTCCG         2     37487      7243     37487</span></span>
<span><span class="co">#&gt; r1_CTTGTGCGGGAA r1_CTTGTGCGGGAA         2     32047      6933     32047</span></span>
<span><span class="co">#&gt; r1_GCGCAACTGCTC r1_GCGCAACTGCTC         2     28148      6397     28148</span></span>
<span><span class="co">#&gt; r1_GATTGGGAGGCA r1_GATTGGGAGGCA         2     20367      5740     20367</span></span>
<span><span class="co">#&gt; r1_CCTCCTAGTTGG r1_CCTCCTAGTTGG        NA     19560      5779     19560</span></span>
<span><span class="co">#&gt; ...                         ...       ...       ...       ...       ...</span></span>
<span><span class="co">#&gt; p1_TCAAAAGCCGGG p1_TCAAAAGCCGGG        24       817       537       817</span></span>
<span><span class="co">#&gt; p1_ATTAAGTTCCAA p1_ATTAAGTTCCAA        34       817       574       817</span></span>
<span><span class="co">#&gt; p1_CTGTCTGAGACC p1_CTGTCTGAGACC         2       817       637       817</span></span>
<span><span class="co">#&gt; p1_TAACGCGCTCCT p1_TAACGCGCTCCT        24       816       488       816</span></span>
<span><span class="co">#&gt; p1_ATTCTTGTTCTT p1_ATTCTTGTTCTT        24       816       484       816</span></span>
<span><span class="co">#&gt;                     low_lib_size   low_n_features   discard</span></span>
<span><span class="co">#&gt;                 &lt;outlier.filter&gt; &lt;outlier.filter&gt; &lt;logical&gt;</span></span>
<span><span class="co">#&gt; r1_GGCCGCAGTCCG            FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; r1_CTTGTGCGGGAA            FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; r1_GCGCAACTGCTC            FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; r1_GATTGGGAGGCA            FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; r1_CCTCCTAGTTGG            FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; ...                          ...              ...       ...</span></span>
<span><span class="co">#&gt; p1_TCAAAAGCCGGG            FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; p1_ATTAAGTTCCAA            FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; p1_CTGTCTGAGACC            FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; p1_TAACGCGCTCCT            FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; p1_ATTCTTGTTCTT            FALSE            FALSE     FALSE</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/quickPerCellQC.html">quickPerCellQC</a></span><span class="op">(</span><span class="va">qcstats</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 49300 rows and 3 columns</span></span>
<span><span class="co">#&gt;           low_lib_size   low_n_features   discard</span></span>
<span><span class="co">#&gt;       &lt;outlier.filter&gt; &lt;outlier.filter&gt; &lt;logical&gt;</span></span>
<span><span class="co">#&gt; 1                FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; 2                FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; 3                FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; 4                FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; 5                FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; ...                ...              ...       ...</span></span>
<span><span class="co">#&gt; 49296            FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; 49297            FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; 49298            FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; 49299            FALSE            FALSE     FALSE</span></span>
<span><span class="co">#&gt; 49300            FALSE            FALSE     FALSE</span></span>
<span><span class="va">filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/quickPerCellQC.html">quickPerCellQC</a></span><span class="op">(</span><span class="va">qcstats</span>, percent_subsets<span class="op">=</span><span class="st">"subsets_Mito_percent"</span><span class="op">)</span></span>
<span><span class="va">filtered</span></span>
<span><span class="co">#&gt; DataFrame with 49300 rows and 4 columns</span></span>
<span><span class="co">#&gt;           low_lib_size   low_n_features high_subsets_Mito_percent   discard</span></span>
<span><span class="co">#&gt;       &lt;outlier.filter&gt; &lt;outlier.filter&gt;          &lt;outlier.filter&gt; &lt;logical&gt;</span></span>
<span><span class="co">#&gt; 1                FALSE            FALSE                     FALSE     FALSE</span></span>
<span><span class="co">#&gt; 2                FALSE            FALSE                     FALSE     FALSE</span></span>
<span><span class="co">#&gt; 3                FALSE            FALSE                     FALSE     FALSE</span></span>
<span><span class="co">#&gt; 4                FALSE            FALSE                     FALSE     FALSE</span></span>
<span><span class="co">#&gt; 5                FALSE            FALSE                     FALSE     FALSE</span></span>
<span><span class="co">#&gt; ...                ...              ...                       ...       ...</span></span>
<span><span class="co">#&gt; 49296            FALSE            FALSE                     FALSE     FALSE</span></span>
<span><span class="co">#&gt; 49297            FALSE            FALSE                     FALSE     FALSE</span></span>
<span><span class="co">#&gt; 49298            FALSE            FALSE                     FALSE     FALSE</span></span>
<span><span class="co">#&gt; 49299            FALSE            FALSE                     FALSE     FALSE</span></span>
<span><span class="co">#&gt; 49300            FALSE            FALSE                     FALSE     FALSE</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span>, <span class="op">!</span><span class="va">filtered</span><span class="op">$</span><span class="va">discard</span><span class="op">]</span> <span class="co"># discard 丢弃</span></span>
<span><span class="va">sce</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 24658 45877 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(24658): KITL TMTC3 ... 1110059M19RIK GM20861</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(45877): r1_GGCCGCAGTCCG r1_CTTGTGCGGGAA ... p1_TAACGCGCTCCT</span></span>
<span><span class="co">#&gt;   p1_ATTCTTGTTCTT</span></span>
<span><span class="co">#&gt; colData names(2): cell.id cluster</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="analysis_cache/html/unnamed-chunk-4_236a7d94d5e82fe6ac51d8015bedcb3b">
<details><summary>Normalization</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">scuttle</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html">logNormCounts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="fu">assays</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of length 2</span></span>
<span><span class="co">#&gt; names(2): counts logcounts</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="analysis_cache/html/unnamed-chunk-5_e16b7564f3dd5bdc667062ccd9ee76c0">
<details><summary>Feature selection</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span></span>
<span><span class="va">dec</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVar.html">modelGeneVar</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>  <span class="co"># ?modelGeneVar  对每个基因的对数表达谱的方差进行建模，根据拟合的均值-方差趋势将其分解为技术和生物学成分。</span></span>
<span><span class="va">dec</span></span>
<span><span class="co">#&gt; DataFrame with 24658 rows and 6 columns</span></span>
<span><span class="co">#&gt;                      mean       total        tech          bio   p.value</span></span>
<span><span class="co">#&gt;                 &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;    &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; KITL           0.00884872 0.011881960 0.011546899  3.35060e-04 0.4118020</span></span>
<span><span class="co">#&gt; TMTC3          0.05143737 0.075214907 0.066708042  8.50687e-03 0.1636316</span></span>
<span><span class="co">#&gt; CEP290         0.57977970 0.841422026 0.689765311  1.51657e-01 0.0456074</span></span>
<span><span class="co">#&gt; 4930430F08RIK  0.05284121 0.069214001 0.068514734  6.99267e-04 0.4687536</span></span>
<span><span class="co">#&gt; 1700017N19RIK  0.00047060 0.000537273 0.000614846 -7.75727e-05 0.8337798</span></span>
<span><span class="co">#&gt; ...                   ...         ...         ...          ...       ...</span></span>
<span><span class="co">#&gt; VSIG1         7.47516e-06 2.56352e-06 9.76708e-06 -7.20357e-06         1</span></span>
<span><span class="co">#&gt; GM16390       0.00000e+00 0.00000e+00 0.00000e+00  0.00000e+00       NaN</span></span>
<span><span class="co">#&gt; GM25207       3.60098e-06 5.94891e-07 4.70506e-06 -4.11017e-06         1</span></span>
<span><span class="co">#&gt; 1110059M19RIK 0.00000e+00 0.00000e+00 0.00000e+00  0.00000e+00       NaN</span></span>
<span><span class="co">#&gt; GM20861       0.00000e+00 0.00000e+00 0.00000e+00  0.00000e+00       NaN</span></span>
<span><span class="co">#&gt;                     FDR</span></span>
<span><span class="co">#&gt;               &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; KITL           1.000000</span></span>
<span><span class="co">#&gt; TMTC3          1.000000</span></span>
<span><span class="co">#&gt; CEP290         0.791672</span></span>
<span><span class="co">#&gt; 4930430F08RIK  1.000000</span></span>
<span><span class="co">#&gt; 1700017N19RIK  1.000000</span></span>
<span><span class="co">#&gt; ...                 ...</span></span>
<span><span class="co">#&gt; VSIG1                 1</span></span>
<span><span class="co">#&gt; GM16390             NaN</span></span>
<span><span class="co">#&gt; GM25207               1</span></span>
<span><span class="co">#&gt; 1110059M19RIK       NaN</span></span>
<span><span class="co">#&gt; GM20861             NaN</span></span>
<span><span class="va">hvg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/getTopHVGs.html">getTopHVGs</a></span><span class="op">(</span><span class="va">dec</span>, prop<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span> <span class="co"># ?getTopHVGs 根据 modelGeneVar() 或相关函数的方差建模统计数据，定义一组高度可变的基因。</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">hvg</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 914</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="analysis_cache/html/unnamed-chunk-6_99301aa81ca6a5e7f7d4860254dbe1f8">
<details><summary>数据降维</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">### PCA</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/">scater</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html">runPCA</a></span><span class="op">(</span><span class="va">sce</span>, ncomponents<span class="op">=</span><span class="fl">25</span>, subset_row<span class="op">=</span><span class="va">hvg</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html">reducedDim</a></span><span class="op">(</span><span class="va">sce</span>,<span class="st">"PCA"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 45877    25</span></span>
<span></span>
<span><span class="co">### Clustering</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">bluster</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/colLabels.html">colLabels</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce</span>, use.dimred<span class="op">=</span><span class="st">'PCA'</span>,</span>
<span>                               BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/NNGraphParam-class.html">NNGraphParam</a></span><span class="op">(</span>cluster.fun<span class="op">=</span><span class="st">"louvain"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/colLabels.html">colLabels</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    1    2    3    4    5    6    7    8    9   10   11   12   13 </span></span>
<span><span class="co">#&gt; 3515 2360  182 4519 1889 2599  193 6501  496 2735 6154 5210 9524</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="analysis_cache/html/fig-umap1_d315b3a1ad644314f965f29b22ebacce">
<details><summary>可视化</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">sce</span>, dimred <span class="op">=</span> <span class="st">'PCA'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plot_reddim.html">plotUMAP</a></span><span class="op">(</span><span class="va">sce</span>, colour_by<span class="op">=</span><span class="st">"label"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="fig-umap1" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="analysis_files/figure-html/fig-umap1-1.png" class="img-fluid figure-img" width="576"></p>
<figcaption class="figure-caption">图&nbsp;4.2: 视网膜数据集的UMAP 图，其中每个点都是一个细胞，并由分配的聚类标识着色。</figcaption></figure>
</div>
</div>
</div>
<div class="cell" data-hash="analysis_cache/html/unnamed-chunk-8_b6ec35df1411acc65416682cdfbffe25">
<details><summary>Marker detection</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/findMarkers.html">findMarkers</a></span><span class="op">(</span><span class="va">sce</span>, test.type<span class="op">=</span><span class="st">"wilcox"</span>, direction<span class="op">=</span><span class="st">"up"</span>, lfc<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="co"># ?findMarkers 通过检测细胞组配对之间的差异表达，找到细胞组（如聚类）的候选标记基因。</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="快速开始多批次" class="level2 page-columns page-full" data-number="4.3"><h2 data-number="4.3" class="anchored" data-anchor-id="快速开始多批次">
<span class="header-section-number">4.3</span> 快速开始（多批次）</h2>
<p><a href="http://bioconductor.org/books/3.18/OSCA.workflows/segerstolpe-human-pancreas-smart-seq2.html#segerstolpe-human-pancreas-smart-seq2">工作流第 8 章</a>中提供了对同一数据集的更详细分析以及每个步骤的理由。</p>
<div class="cell" data-hash="analysis_cache/html/unnamed-chunk-9_c2514d1fa64635fe6330ad402a02dbd5">
<details><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span></span>
<span><span class="va">sce1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scRNAseq/man/SegerstolpePancreasData.html">SegerstolpePancreasData</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Quality control (using ERCCs).</span></span>
<span><span class="va">qcstats1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/perCellQCMetrics.html">perCellQCMetrics</a></span><span class="op">(</span><span class="va">sce1</span><span class="op">)</span></span>
<span><span class="va">filtered1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/quickPerCellQC.html">quickPerCellQC</a></span><span class="op">(</span><span class="va">qcstats1</span>, percent_subsets<span class="op">=</span><span class="st">"altexps_ERCC_percent"</span><span class="op">)</span></span>
<span><span class="va">sce1</span> <span class="op">&lt;-</span> <span class="va">sce1</span><span class="op">[</span>, <span class="op">!</span><span class="va">filtered1</span><span class="op">$</span><span class="va">discard</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Normalization.</span></span>
<span><span class="va">sce1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html">logNormCounts</a></span><span class="op">(</span><span class="va">sce1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Feature selection, blocking on the individual of origin.</span></span>
<span><span class="va">dec1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVar.html">modelGeneVar</a></span><span class="op">(</span><span class="va">sce1</span>, block<span class="op">=</span><span class="va">sce1</span><span class="op">$</span><span class="va">individual</span><span class="op">)</span></span>
<span><span class="va">hvg1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/getTopHVGs.html">getTopHVGs</a></span><span class="op">(</span><span class="va">dec1</span>, prop<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Batch correction.</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">batchelor</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span></span>
<span><span class="va">sce1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/batchelor/man/correctExperiments.html">correctExperiments</a></span><span class="op">(</span><span class="va">sce1</span>, batch<span class="op">=</span><span class="va">sce1</span><span class="op">$</span><span class="va">individual</span>, </span>
<span>    subset.row<span class="op">=</span><span class="va">hvg1</span>, correct.all<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Clustering.</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/colLabels.html">colLabels</a></span><span class="op">(</span><span class="va">sce1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce1</span>, use.dimred<span class="op">=</span><span class="st">'corrected'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/colLabels.html">colLabels</a></span><span class="op">(</span><span class="va">sce1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10  11  12 </span></span>
<span><span class="co">#&gt; 462  61 138 104 283  70 968 216  36 116  10  22</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="page-columns page-full">
<details><summary>可视化</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">sce1</span>, dimred <span class="op">=</span> <span class="st">'corrected'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plot_reddim.html">plotUMAP</a></span><span class="op">(</span><span class="va">sce1</span>, colour_by<span class="op">=</span><span class="st">"label"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plot_reddim.html">plotUMAP</a></span><span class="op">(</span><span class="va">sce1</span>, colour_by<span class="op">=</span><span class="st">"individual"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div id="fig-umap2" class="cell column-page-right quarto-layout-panel" data-hash="analysis_cache/html/fig-umap2_f533dc8c2248b8048efac1c2f50c08f6">
<figure class="figure"><div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-umap2-1" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="analysis_files/figure-html/fig-umap2-1.png" class="img-fluid figure-img" data-ref-parent="fig-umap2" width="576"></p>
<figcaption class="figure-caption">(a) 根据聚类标记</figcaption></figure>
</div>
</div>
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-umap2-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="analysis_files/figure-html/fig-umap2-2.png" class="img-fluid figure-img" data-ref-parent="fig-umap2" width="576"></p>
<figcaption class="figure-caption">(b) 根据起源个体</figcaption></figure>
</div>
</div>
</div>
<p></p><figcaption class="figure-caption">图&nbsp;4.3: 胰腺数据集的 UMAP 图，其中每个点都是一个细胞，并由分配的聚类标识（左）或起源个体（右）着色。</figcaption><p></p>
</figure>
</div>
</div>
<div class="cell" data-hash="analysis_cache/html/unnamed-chunk-11_0c039687afdee5906935b0c366103ee6">
<details><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Marker detection, blocking on the individual of origin.</span></span>
<span><span class="va">markers1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/findMarkers.html">findMarkers</a></span><span class="op">(</span><span class="va">sce1</span>, test.type<span class="op">=</span><span class="st">"wilcox"</span>, direction<span class="op">=</span><span class="st">"up"</span>, lfc<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->

</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./SCE.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title"><code>SingleCellExperiment</code>结构</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./basics.html" class="pagination-link">
        <span class="nav-page-text">Basics</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb11" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># 分析概述</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">## outline</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>本章概述了典型scRNA-seq分析工作流程的框架（@fig-workflow）</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="al">![典型 scRNA-seq 分析工作流程示意图。每个阶段（用虚线分隔）由许多特定步骤组成，其中许多步骤对实例进行操作和修改](images/workflow.png)</span>{#fig-workflow fig-align="center"}</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="fu">## 快速入门（简单）</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>原书使用来自Macosko等人（2015）的基于液滴的视网膜数据集，该数据集在scRNAseq包中提供。</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(GenomicRanges<span class="sc">::</span>setdiff)</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleCellExperiment)</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>从计数矩阵开始，以聚类结束( @fig-workflow )。</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache: true</span></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "数据集信息"</span></span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scRNAseq)</span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">MacoskoRetinaData</span>()</span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>sce</span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">assay</span>(sce, <span class="st">"counts"</span>)</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(sce)</span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache: true</span></span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "质量控制"</span></span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Quality control (using mitochondrial genes)线粒体基因</span></span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a>is.mito <span class="ot">&lt;-</span> <span class="fu">str_detect</span>(<span class="fu">rownames</span>(sce),<span class="st">"^MT-"</span>) </span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(is.mito) <span class="co"># 31个</span></span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a><span class="co">#?perCellQCMetrics</span></span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a><span class="fu">perCellQCMetrics</span>(sce)</span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a>qcstats <span class="ot">&lt;-</span> <span class="fu">perCellQCMetrics</span>(sce, <span class="at">subsets=</span><span class="fu">list</span>(<span class="at">Mito=</span>is.mito))</span>
<span id="cb11-50"><a href="#cb11-50" aria-hidden="true" tabindex="-1"></a>qcstats</span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-52"><a href="#cb11-52" aria-hidden="true" tabindex="-1"></a><span class="co">#?quickPerCellQC</span></span>
<span id="cb11-53"><a href="#cb11-53" aria-hidden="true" tabindex="-1"></a><span class="fu">quickPerCellQC</span>(sce) <span class="sc">|&gt;</span> <span class="fu">colData</span>()</span>
<span id="cb11-54"><a href="#cb11-54" aria-hidden="true" tabindex="-1"></a><span class="fu">quickPerCellQC</span>(qcstats)</span>
<span id="cb11-55"><a href="#cb11-55" aria-hidden="true" tabindex="-1"></a>filtered <span class="ot">&lt;-</span> <span class="fu">quickPerCellQC</span>(qcstats, <span class="at">percent_subsets=</span><span class="st">"subsets_Mito_percent"</span>)</span>
<span id="cb11-56"><a href="#cb11-56" aria-hidden="true" tabindex="-1"></a>filtered</span>
<span id="cb11-57"><a href="#cb11-57" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> sce[, <span class="sc">!</span>filtered<span class="sc">$</span>discard] <span class="co"># discard 丢弃</span></span>
<span id="cb11-58"><a href="#cb11-58" aria-hidden="true" tabindex="-1"></a>sce</span>
<span id="cb11-59"><a href="#cb11-59" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-60"><a href="#cb11-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-63"><a href="#cb11-63" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-64"><a href="#cb11-64" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "Normalization"</span></span>
<span id="cb11-65"><a href="#cb11-65" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scuttle)</span>
<span id="cb11-66"><a href="#cb11-66" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">logNormCounts</span>(sce)</span>
<span id="cb11-67"><a href="#cb11-67" aria-hidden="true" tabindex="-1"></a><span class="fu">assays</span>(sce)</span>
<span id="cb11-68"><a href="#cb11-68" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-69"><a href="#cb11-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-72"><a href="#cb11-72" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-73"><a href="#cb11-73" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "Feature selection"</span></span>
<span id="cb11-74"><a href="#cb11-74" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scran)</span>
<span id="cb11-75"><a href="#cb11-75" aria-hidden="true" tabindex="-1"></a>dec <span class="ot">&lt;-</span> <span class="fu">modelGeneVar</span>(sce)  <span class="co"># ?modelGeneVar  对每个基因的对数表达谱的方差进行建模，根据拟合的均值-方差趋势将其分解为技术和生物学成分。</span></span>
<span id="cb11-76"><a href="#cb11-76" aria-hidden="true" tabindex="-1"></a>dec</span>
<span id="cb11-77"><a href="#cb11-77" aria-hidden="true" tabindex="-1"></a>hvg <span class="ot">&lt;-</span> <span class="fu">getTopHVGs</span>(dec, <span class="at">prop=</span><span class="fl">0.1</span>) <span class="co"># ?getTopHVGs 根据 modelGeneVar() 或相关函数的方差建模统计数据，定义一组高度可变的基因。</span></span>
<span id="cb11-78"><a href="#cb11-78" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(hvg)</span>
<span id="cb11-79"><a href="#cb11-79" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-80"><a href="#cb11-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-83"><a href="#cb11-83" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-84"><a href="#cb11-84" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "数据降维"</span></span>
<span id="cb11-85"><a href="#cb11-85" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache: true</span></span>
<span id="cb11-86"><a href="#cb11-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-87"><a href="#cb11-87" aria-hidden="true" tabindex="-1"></a><span class="do">### PCA</span></span>
<span id="cb11-88"><a href="#cb11-88" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb11-89"><a href="#cb11-89" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb11-90"><a href="#cb11-90" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(sce, <span class="at">ncomponents=</span><span class="dv">25</span>, <span class="at">subset_row=</span>hvg)</span>
<span id="cb11-91"><a href="#cb11-91" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">reducedDim</span>(sce,<span class="st">"PCA"</span>))</span>
<span id="cb11-92"><a href="#cb11-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-93"><a href="#cb11-93" aria-hidden="true" tabindex="-1"></a><span class="do">### Clustering</span></span>
<span id="cb11-94"><a href="#cb11-94" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(bluster)</span>
<span id="cb11-95"><a href="#cb11-95" aria-hidden="true" tabindex="-1"></a><span class="fu">colLabels</span>(sce) <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce, <span class="at">use.dimred=</span><span class="st">'PCA'</span>,</span>
<span id="cb11-96"><a href="#cb11-96" aria-hidden="true" tabindex="-1"></a>                               <span class="at">BLUSPARAM=</span><span class="fu">NNGraphParam</span>(<span class="at">cluster.fun=</span><span class="st">"louvain"</span>))</span>
<span id="cb11-97"><a href="#cb11-97" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">colLabels</span>(sce))</span>
<span id="cb11-98"><a href="#cb11-98" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-99"><a href="#cb11-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-102"><a href="#cb11-102" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-103"><a href="#cb11-103" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "可视化"</span></span>
<span id="cb11-104"><a href="#cb11-104" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache: true</span></span>
<span id="cb11-105"><a href="#cb11-105" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-umap1</span></span>
<span id="cb11-106"><a href="#cb11-106" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "视网膜数据集的UMAP 图，其中每个点都是一个细胞，并由分配的聚类标识着色。"</span></span>
<span id="cb11-107"><a href="#cb11-107" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(sce, <span class="at">dimred =</span> <span class="st">'PCA'</span>)</span>
<span id="cb11-108"><a href="#cb11-108" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUMAP</span>(sce, <span class="at">colour_by=</span><span class="st">"label"</span>)</span>
<span id="cb11-109"><a href="#cb11-109" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-110"><a href="#cb11-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-113"><a href="#cb11-113" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-114"><a href="#cb11-114" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "Marker detection"</span></span>
<span id="cb11-115"><a href="#cb11-115" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache: true</span></span>
<span id="cb11-116"><a href="#cb11-116" aria-hidden="true" tabindex="-1"></a>markers <span class="ot">&lt;-</span> <span class="fu">findMarkers</span>(sce, <span class="at">test.type=</span><span class="st">"wilcox"</span>, <span class="at">direction=</span><span class="st">"up"</span>, <span class="at">lfc=</span><span class="dv">1</span>)<span class="co"># ?findMarkers 通过检测细胞组配对之间的差异表达，找到细胞组（如聚类）的候选标记基因。</span></span>
<span id="cb11-117"><a href="#cb11-117" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-118"><a href="#cb11-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-119"><a href="#cb11-119" aria-hidden="true" tabindex="-1"></a><span class="fu">## 快速开始（多批次）</span></span>
<span id="cb11-120"><a href="#cb11-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-121"><a href="#cb11-121" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">工作流第 8 章</span><span class="co">](http://bioconductor.org/books/3.18/OSCA.workflows/segerstolpe-human-pancreas-smart-seq2.html#segerstolpe-human-pancreas-smart-seq2)</span>中提供了对同一数据集的更详细分析以及每个步骤的理由。</span>
<span id="cb11-122"><a href="#cb11-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-125"><a href="#cb11-125" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-126"><a href="#cb11-126" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache: true</span></span>
<span id="cb11-127"><a href="#cb11-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-128"><a href="#cb11-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-129"><a href="#cb11-129" aria-hidden="true" tabindex="-1"></a>sce1 <span class="ot">&lt;-</span> <span class="fu">SegerstolpePancreasData</span>()</span>
<span id="cb11-130"><a href="#cb11-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-131"><a href="#cb11-131" aria-hidden="true" tabindex="-1"></a><span class="co"># Quality control (using ERCCs).</span></span>
<span id="cb11-132"><a href="#cb11-132" aria-hidden="true" tabindex="-1"></a>qcstats1 <span class="ot">&lt;-</span> <span class="fu">perCellQCMetrics</span>(sce1)</span>
<span id="cb11-133"><a href="#cb11-133" aria-hidden="true" tabindex="-1"></a>filtered1 <span class="ot">&lt;-</span> <span class="fu">quickPerCellQC</span>(qcstats1, <span class="at">percent_subsets=</span><span class="st">"altexps_ERCC_percent"</span>)</span>
<span id="cb11-134"><a href="#cb11-134" aria-hidden="true" tabindex="-1"></a>sce1 <span class="ot">&lt;-</span> sce1[, <span class="sc">!</span>filtered1<span class="sc">$</span>discard]</span>
<span id="cb11-135"><a href="#cb11-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-136"><a href="#cb11-136" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalization.</span></span>
<span id="cb11-137"><a href="#cb11-137" aria-hidden="true" tabindex="-1"></a>sce1 <span class="ot">&lt;-</span> <span class="fu">logNormCounts</span>(sce1)</span>
<span id="cb11-138"><a href="#cb11-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-139"><a href="#cb11-139" aria-hidden="true" tabindex="-1"></a><span class="co"># Feature selection, blocking on the individual of origin.</span></span>
<span id="cb11-140"><a href="#cb11-140" aria-hidden="true" tabindex="-1"></a>dec1 <span class="ot">&lt;-</span> <span class="fu">modelGeneVar</span>(sce1, <span class="at">block=</span>sce1<span class="sc">$</span>individual)</span>
<span id="cb11-141"><a href="#cb11-141" aria-hidden="true" tabindex="-1"></a>hvg1 <span class="ot">&lt;-</span> <span class="fu">getTopHVGs</span>(dec1, <span class="at">prop=</span><span class="fl">0.1</span>)</span>
<span id="cb11-142"><a href="#cb11-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-143"><a href="#cb11-143" aria-hidden="true" tabindex="-1"></a><span class="co"># Batch correction.</span></span>
<span id="cb11-144"><a href="#cb11-144" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(batchelor)</span>
<span id="cb11-145"><a href="#cb11-145" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb11-146"><a href="#cb11-146" aria-hidden="true" tabindex="-1"></a>sce1 <span class="ot">&lt;-</span> <span class="fu">correctExperiments</span>(sce1, <span class="at">batch=</span>sce1<span class="sc">$</span>individual, </span>
<span id="cb11-147"><a href="#cb11-147" aria-hidden="true" tabindex="-1"></a>    <span class="at">subset.row=</span>hvg1, <span class="at">correct.all=</span><span class="cn">TRUE</span>)</span>
<span id="cb11-148"><a href="#cb11-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-149"><a href="#cb11-149" aria-hidden="true" tabindex="-1"></a><span class="co"># Clustering.</span></span>
<span id="cb11-150"><a href="#cb11-150" aria-hidden="true" tabindex="-1"></a><span class="fu">colLabels</span>(sce1) <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce1, <span class="at">use.dimred=</span><span class="st">'corrected'</span>)</span>
<span id="cb11-151"><a href="#cb11-151" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">colLabels</span>(sce1))</span>
<span id="cb11-152"><a href="#cb11-152" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-153"><a href="#cb11-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-156"><a href="#cb11-156" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-157"><a href="#cb11-157" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-umap2</span></span>
<span id="cb11-158"><a href="#cb11-158" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "可视化"</span></span>
<span id="cb11-159"><a href="#cb11-159" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache: true</span></span>
<span id="cb11-160"><a href="#cb11-160" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "胰腺数据集的 UMAP 图，其中每个点都是一个细胞，并由分配的聚类标识（左）或起源个体（右）着色。"</span></span>
<span id="cb11-161"><a href="#cb11-161" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-subcap:</span></span>
<span id="cb11-162"><a href="#cb11-162" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "根据聚类标记"</span></span>
<span id="cb11-163"><a href="#cb11-163" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "根据起源个体"</span></span>
<span id="cb11-164"><a href="#cb11-164" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 2</span></span>
<span id="cb11-165"><a href="#cb11-165" aria-hidden="true" tabindex="-1"></a><span class="co">#| column: page-right</span></span>
<span id="cb11-166"><a href="#cb11-166" aria-hidden="true" tabindex="-1"></a>sce1 <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(sce1, <span class="at">dimred =</span> <span class="st">'corrected'</span>)</span>
<span id="cb11-167"><a href="#cb11-167" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUMAP</span>(sce1, <span class="at">colour_by=</span><span class="st">"label"</span>)</span>
<span id="cb11-168"><a href="#cb11-168" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUMAP</span>(sce1, <span class="at">colour_by=</span><span class="st">"individual"</span>)</span>
<span id="cb11-169"><a href="#cb11-169" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-170"><a href="#cb11-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-173"><a href="#cb11-173" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-174"><a href="#cb11-174" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker detection, blocking on the individual of origin.</span></span>
<span id="cb11-175"><a href="#cb11-175" aria-hidden="true" tabindex="-1"></a>markers1 <span class="ot">&lt;-</span> <span class="fu">findMarkers</span>(sce1, <span class="at">test.type=</span><span class="st">"wilcox"</span>, <span class="at">direction=</span><span class="st">"up"</span>, <span class="at">lfc=</span><span class="dv">1</span>)</span>
<span id="cb11-176"><a href="#cb11-176" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>