<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Bioinformatics - 10&nbsp; 聚类</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./MarkerGeneDetection.html" rel="next">
<link href="./DimensionalityReduction.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
</head>
<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./prerequisites.html">单细胞分析</a></li><li class="breadcrumb-item"><a href="./Clustering.html"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">聚类</span></a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Bioinformatics</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/Wanganlin00/Bioinformatics" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">参考资料</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">上游分析</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Seq_to_Matrix.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Sequences Data to Expression Matrix</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">单细胞分析</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./prerequisites.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">写在前面</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./import.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">获取scRNA-seq表达值矩阵</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./SCE.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title"><code>SingleCellExperiment</code></span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">工作流</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./QC.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">质量控制</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./normalization.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">标准化</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./feature_selection.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">特征选择</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./DimensionalityReduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">降维</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Clustering.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">聚类</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./MarkerGeneDetection.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">conflicts_prefer(dplyr::filter)</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./CellTypeAnnotation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">细胞类型注释</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Seurat</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Seurat.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Seurat</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">下游分析</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./DifferentExpressionGeneAnalysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">差异表达基因分析</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./EnrichmentAnalysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">富集分析</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PseudotimeAnalysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">伪时序分析</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true">
 <span class="menu-text">基因组学</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Biostrings.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title"><code>Biostrings</code>包</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./eQTL.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">eQTL</span></span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li>
<a href="#graph-based-clustering" id="toc-graph-based-clustering" class="nav-link active" data-scroll-target="#graph-based-clustering"><span class="header-section-number">10.1</span> Graph-based clustering</a>
  <ul>
<li>
<a href="#%E5%8F%82%E6%95%B0" id="toc-参数" class="nav-link" data-scroll-target="#%E5%8F%82%E6%95%B0"><span class="header-section-number">10.1.1</span> 参数</a>
  <ul class="collapse">
<li><a href="#k-%E6%9C%80%E8%BF%91%E9%82%BB%E6%95%B0%E7%9B%AE" id="toc-k-最近邻数目" class="nav-link" data-scroll-target="#k-%E6%9C%80%E8%BF%91%E9%82%BB%E6%95%B0%E7%9B%AE"><span class="header-section-number">10.1.1.1</span> <code>k</code> 最近邻数目</a></li>
  <li><a href="#type-%E8%BE%B9%E7%BC%98%E5%8A%A0%E6%9D%83%E6%96%B9%E6%B3%95" id="toc-type-边缘加权方法" class="nav-link" data-scroll-target="#type-%E8%BE%B9%E7%BC%98%E5%8A%A0%E6%9D%83%E6%96%B9%E6%B3%95"><span class="header-section-number">10.1.1.2</span> <code>type</code> 边缘加权方法</a></li>
  <li><a href="#cluster.fun-%E7%B0%87%E8%AF%86%E5%88%AB%E6%96%B9%E6%B3%95" id="toc-cluster.fun-簇识别方法" class="nav-link" data-scroll-target="#cluster.fun-%E7%B0%87%E8%AF%86%E5%88%AB%E6%96%B9%E6%B3%95"><span class="header-section-number">10.1.1.3</span> <code>cluster.fun</code> 簇识别方法</a></li>
  </ul>
</li>
  </ul>
</li>
  <li>
<a href="#k-means-%E8%81%9A%E7%B1%BB" id="toc-k-means-聚类" class="nav-link" data-scroll-target="#k-means-%E8%81%9A%E7%B1%BB"><span class="header-section-number">10.2</span> K-means 聚类</a>
  <ul>
<li><a href="#mini-batch-k-means" id="toc-mini-batch-k-means" class="nav-link" data-scroll-target="#mini-batch-k-means"><span class="header-section-number">10.2.1</span> mini-batch k-means</a></li>
  <li><a href="#two-step-modekmeans-nngraph" id="toc-two-step-modekmeans-nngraph" class="nav-link" data-scroll-target="#two-step-modekmeans-nngraph"><span class="header-section-number">10.2.2</span> “two-step” mode：Kmeans-nnGraph</a></li>
  </ul>
</li>
  <li>
<a href="#hierarchical-clustering" id="toc-hierarchical-clustering" class="nav-link" data-scroll-target="#hierarchical-clustering"><span class="header-section-number">10.3</span> Hierarchical clustering</a>
  <ul>
<li><a href="#two-step-mode-kmeans-hierarchical" id="toc-two-step-mode-kmeans-hierarchical" class="nav-link" data-scroll-target="#two-step-mode-kmeans-hierarchical"><span class="header-section-number">10.3.1</span> “two-step” mode ：Kmeans-Hierarchical</a></li>
  <li><a href="#two-step-mode-kmeans-affinity" id="toc-two-step-mode-kmeans-affinity" class="nav-link" data-scroll-target="#two-step-mode-kmeans-affinity"><span class="header-section-number">10.3.2</span> “two-step” mode ：Kmeans-Affinity</a></li>
  </ul>
</li>
  <li><a href="#subclustering" id="toc-subclustering" class="nav-link" data-scroll-target="#subclustering"><span class="header-section-number">10.4</span> Subclustering</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/Wanganlin00/Bioinformatics/edit/main/Clustering.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/Wanganlin00/Bioinformatics/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./prerequisites.html">单细胞分析</a></li><li class="breadcrumb-item"><a href="./Clustering.html"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">聚类</span></a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">
<span class="chapter-number">10</span>&nbsp; <span class="chapter-title">聚类</span>
</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><p>聚类是一种无监督学习过程，旨在通过计算基因之间的欧几里得距离来识别具有相似转录组学谱的细胞，使用易于理解的离散标签来描述群体异质性，而不是试图理解细胞真正所在的高维流形。</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span></span>
<span><span class="co">#--- loading ---#</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">DropletUtils</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="fu">DropletUtils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DropletUtils/man/read10xCounts.html">read10xCounts</a></span><span class="op">(</span><span class="st">"data/OSCA/raw_gene_bc_matrices/GRCh38"</span>,</span>
<span>                                        col.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#--- gene-annotation ---#</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/">scater</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/uniquifyFeatureNames.html">uniquifyFeatureNames</a></span><span class="op">(</span></span>
<span>    <span class="fu">rowData</span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span><span class="op">$</span><span class="va">ID</span>, <span class="fu">rowData</span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span><span class="op">$</span><span class="va">Symbol</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v86</span><span class="op">)</span></span>
<span><span class="va">location</span> <span class="op">&lt;-</span> <span class="fu">mapIds</span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v86</span>, keys<span class="op">=</span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span><span class="op">$</span><span class="va">ID</span>, </span>
<span>    column<span class="op">=</span><span class="st">"SEQNAME"</span>, keytype<span class="op">=</span><span class="st">"GENEID"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#--- cell-detection ---#</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">e.out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DropletUtils/man/emptyDrops.html">emptyDrops</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html">counts</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="va">sce.pbmc</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">e.out</span><span class="op">$</span><span class="va">FDR</span> <span class="op">&lt;=</span> <span class="fl">0.001</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co">#--- quality-control ---#</span></span>
<span><span class="va">stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/perCellQCMetrics.html">perCellQCMetrics</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, subsets<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>Mito<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">location</span><span class="op">==</span><span class="st">"MT"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">high.mito</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/isOutlier.html">isOutlier</a></span><span class="op">(</span><span class="va">stats</span><span class="op">$</span><span class="va">subsets_Mito_percent</span>, type<span class="op">=</span><span class="st">"higher"</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="va">sce.pbmc</span><span class="op">[</span>,<span class="op">!</span><span class="va">high.mito</span><span class="op">]</span></span>
<span></span>
<span><span class="co">#--- normalization ---#</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1000</span><span class="op">)</span></span>
<span><span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/quickCluster.html">quickCluster</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/computeSumFactors.html">computeSumFactors</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, cluster<span class="op">=</span><span class="va">clusters</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html">logNormCounts</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#--- variance-modelling ---#</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1001</span><span class="op">)</span></span>
<span><span class="va">dec.pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVarByPoisson.html">modelGeneVarByPoisson</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span></span>
<span><span class="va">top.pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/getTopHVGs.html">getTopHVGs</a></span><span class="op">(</span><span class="va">dec.pbmc</span>, prop<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#--- dimensionality-reduction ---#</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">10000</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/denoisePCA.html">denoisePCA</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, subset.row<span class="op">=</span><span class="va">top.pbmc</span>, technical<span class="op">=</span><span class="va">dec.pbmc</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">100000</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runTSNE.html">runTSNE</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, dimred<span class="op">=</span><span class="st">"PCA"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1000000</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, dimred<span class="op">=</span><span class="st">"PCA"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce.pbmc</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 33694 3985 </span></span>
<span><span class="co">#&gt; metadata(1): Samples</span></span>
<span><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span><span class="co">#&gt; rownames(33694): RP11-34P13.3 FAM138A ... AC213203.1 FAM231B</span></span>
<span><span class="co">#&gt; rowData names(2): ID Symbol</span></span>
<span><span class="co">#&gt; colnames(3985): AAACCTGAGAAGGCCT-1 AAACCTGAGACAGACC-1 ...</span></span>
<span><span class="co">#&gt;   TTTGTCAGTTAAGACA-1 TTTGTCATCCCAAGAT-1</span></span>
<span><span class="co">#&gt; colData names(3): Sample Barcode sizeFactor</span></span>
<span><span class="co">#&gt; reducedDimNames(3): PCA TSNE UMAP</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="graph-based-clustering" class="level2" data-number="10.1"><h2 data-number="10.1" class="anchored" data-anchor-id="graph-based-clustering">
<span class="header-section-number">10.1</span> Graph-based clustering</h2>
<p>基于图的聚类因其在 Seurat 中的使用而广为普及，是一种灵活且可扩展的技术，用于聚类大型 scRNA-seq 数据集。首先构建一个图，其中每个节点都是一个连接到高维空间中 nearest neighbors 的细胞。根据所涉及的细胞之间的相似性对边缘进行加权，对关系更密切的细胞赋予更高的权重。然后应用算法来识别在同一群落中的细胞比在非同一群落的细胞更紧密的“簇”。</p>
<p>基于图的聚类的主要优势在于其可伸缩性。 它只需要一个 k-nearest neighbor 搜索，避免了对簇的形状或每个簇内细胞的分布做出强有力的假设，例如k-means（有利于球面簇）或高斯混合模型（需要正态性）。</p>
<p>主要缺点是在图构造之后，不会保留超出邻近细胞关系的信息。在细胞密度差异的数据产生一些实际后果。</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span></span>
<span><span class="va">nn.clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">nn.clusters</span><span class="op">)</span></span>
<span><span class="co">#&gt; nn.clusters</span></span>
<span><span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16 </span></span>
<span><span class="co">#&gt; 205 508 541  56 374 125  46 432 302 867  47 155 166  61  84  16</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/">scater</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/colLabels.html">colLabels</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">nn.clusters</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html">plotReducedDim</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, <span class="st">"TSNE"</span>, colour_by<span class="op">=</span><span class="st">"label"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="Clustering_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>默认情况下，使用每个细胞的 10 个最近邻来构造共享的最近邻图。如果两个细胞的任何最近邻共享，则通过一条边连接， 边权重由共享最近邻的最高平均秩定义。<code>walktrap</code></p>
<p>显式指定参数,改变参数切换不同的聚类算法</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">bluster</span><span class="op">)</span></span>
<span><span class="va">nn.clusters2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, </span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/NNGraphParam-class.html">SNNGraphParam</a></span><span class="op">(</span>k<span class="op">=</span><span class="fl">10</span>, type<span class="op">=</span><span class="st">"rank"</span>, cluster.fun<span class="op">=</span><span class="st">"walktrap"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">nn.clusters2</span><span class="op">)</span></span>
<span><span class="co">#&gt; nn.clusters2</span></span>
<span><span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16 </span></span>
<span><span class="co">#&gt; 205 508 541  56 374 125  46 432 302 867  47 155 166  61  84  16</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>通过在<code><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells()</a></code>中指定<code>full=TRUE</code>来获取图本身，这样做将返回聚类分析期间使用的所有中间结构，包括 <a href="https://cran.r-project.org/web/packages/igraph/index.html">igraph</a> 包中的图形对象。 该图可以使用力导向布局(force-directed layout)进行可视化</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">nn.clust.info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, full<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">nn.clust.info</span><span class="op">$</span><span class="va">objects</span><span class="op">$</span><span class="va">graph</span></span>
<span><span class="co">#&gt; IGRAPH d6212fa U-W- 3985 133130 -- </span></span>
<span><span class="co">#&gt; + attr: weight (e/n)</span></span>
<span><span class="co">#&gt; + edges from d6212fa:</span></span>
<span><span class="co">#&gt;  [1]  4--12  8--13 11--15 12--17  4--17 12--18 17--18  8--20  1--21  8--23</span></span>
<span><span class="co">#&gt; [11]  2--25 17--27 18--27 11--30 15--30 10--31 25--33  2--33 19--37  9--38</span></span>
<span><span class="co">#&gt; [21] 23--41  4--43 12--43 17--43 21--45  2--47 33--47 22--48 18--49 22--51</span></span>
<span><span class="co">#&gt; [31] 48--51 16--52 11--54 15--54 30--54 26--57 21--58 47--58 52--59 13--59</span></span>
<span><span class="co">#&gt; [41] 23--60  9--60 41--60 52--60 22--61 51--61 48--61 55--63 22--64 61--64</span></span>
<span><span class="co">#&gt; [51] 48--64 51--64  1--65 42--67 16--68 55--69 50--70 59--70 40--71 55--72</span></span>
<span><span class="co">#&gt; [61] 63--73 39--75 69--75 30--76 11--76 15--76  8--78 41--78 20--79  8--79</span></span>
<span><span class="co">#&gt; [71]  6--82 76--82 15--82 11--85 30--85 76--85 15--85 54--85 14--87 66--87</span></span>
<span><span class="co">#&gt; + ... omitted several edges</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">11000</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html">reducedDim</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, <span class="st">"force"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/layout_with_fr.html">layout_with_fr</a></span><span class="op">(</span><span class="va">nn.clust.info</span><span class="op">$</span><span class="va">objects</span><span class="op">$</span><span class="va">graph</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html">plotReducedDim</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, colour_by<span class="op">=</span><span class="st">"label"</span>, dimred<span class="op">=</span><span class="st">"force"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="fig-Force-directed-layout--shared-nearest-neighbor-graph" class="quarto-figure quarto-figure-center quarto-float anchored" width="672">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-Force-directed-layout--shared-nearest-neighbor-graph-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Clustering_files/figure-html/fig-Force-directed-layout--shared-nearest-neighbor-graph-1.png" id="fig-Force-directed-layout--shared-nearest-neighbor-graph" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-Force-directed-layout--shared-nearest-neighbor-graph-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10.1
</figcaption></figure>
</div>
</div>
</div>
<p>此外，该图还可用于生成有关基于图的聚类行为的详细诊断。</p>
<section id="参数" class="level3" data-number="10.1.1"><h3 data-number="10.1.1" class="anchored" data-anchor-id="参数">
<span class="header-section-number">10.1.1</span> 参数</h3>
<section id="k-最近邻数目" class="level4" data-number="10.1.1.1"><h4 data-number="10.1.1.1" class="anchored" data-anchor-id="k-最近邻数目">
<span class="header-section-number">10.1.1.1</span> <code>k</code> 最近邻数目</h4>
<p>用于构造图的最近邻的数量，这控制了聚类的分辨率，其中更高的分辨率（k越小）是更互联的图和更多的聚类。 最近邻越多，聚类越少。</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># More resolved.</span></span>
<span><span class="va">clust.5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/NNGraphParam-class.html">NNGraphParam</a></span><span class="op">(</span>k<span class="op">=</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">clust.5</span><span class="op">)</span></span>
<span><span class="co">#&gt; clust.5</span></span>
<span><span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20 </span></span>
<span><span class="co">#&gt; 523 302 125  45 172 573 249 439 293  95 772 142  38  18  62  38  30  16  15   9 </span></span>
<span><span class="co">#&gt;  21  22 </span></span>
<span><span class="co">#&gt;  16  13</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Less resolved.</span></span>
<span><span class="va">clust.50</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/NNGraphParam-class.html">NNGraphParam</a></span><span class="op">(</span>k<span class="op">=</span><span class="fl">50</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">clust.50</span><span class="op">)</span></span>
<span><span class="co">#&gt; clust.50</span></span>
<span><span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10 </span></span>
<span><span class="co">#&gt; 869 514 194 478 539 944 138 175  89  45</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="type-边缘加权方法" class="level4" data-number="10.1.1.2"><h4 data-number="10.1.1.2" class="anchored" data-anchor-id="type-边缘加权方法">
<span class="header-section-number">10.1.1.2</span> <code>type</code> 边缘加权方法</h4>
<p><code>type="number"</code>将根据两个细胞之间共享的最近邻的数量对边进行加权。</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">clust.num</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, </span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/NNGraphParam-class.html">NNGraphParam</a></span><span class="op">(</span>type<span class="op">=</span><span class="st">"number"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">clust.num</span><span class="op">)</span></span>
<span><span class="co">#&gt; clust.num</span></span>
<span><span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19 </span></span>
<span><span class="co">#&gt; 199 541 354 294 458 123  47  45 170 397 838 150  40  80 136  50  31  16  16</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><code>type="jaccard"</code>将根据两组最近邻的 Jaccard 指数对边缘进行加权。</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">clust.jaccard</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, </span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/NNGraphParam-class.html">NNGraphParam</a></span><span class="op">(</span>type<span class="op">=</span><span class="st">"jaccard"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">clust.jaccard</span><span class="op">)</span></span>
<span><span class="co">#&gt; clust.jaccard</span></span>
<span><span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17 </span></span>
<span><span class="co">#&gt; 201 541 740 233 352  47 124 841  45 361  40 154  80  61 133  16  16</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>不对边缘加权</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">clust.none</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, </span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/NNGraphParam-class.html">KNNGraphParam</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">clust.none</span><span class="op">)</span></span>
<span><span class="co">#&gt; clust.none</span></span>
<span><span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16 </span></span>
<span><span class="co">#&gt; 541 446  54 170 699  45 126 172 128 907 286  47 158 137  54  15</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="cluster.fun-簇识别方法" class="level4" data-number="10.1.1.3"><h4 data-number="10.1.1.3" class="anchored" data-anchor-id="cluster.fun-簇识别方法">
<span class="header-section-number">10.1.1.3</span> <code>cluster.fun</code> 簇识别方法</h4>
<p><a href="https://cran.r-project.org/web/packages/igraph/index.html">igraph</a></p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/">igraph</a></span><span class="op">)</span></span>
<span><span class="va">clust.walktrap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, </span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/NNGraphParam-class.html">NNGraphParam</a></span><span class="op">(</span>cluster.fun<span class="op">=</span><span class="st">"walktrap"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">clust.louvain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, </span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/NNGraphParam-class.html">NNGraphParam</a></span><span class="op">(</span>cluster.fun<span class="op">=</span><span class="st">"louvain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">clust.infomap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, </span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/NNGraphParam-class.html">NNGraphParam</a></span><span class="op">(</span>cluster.fun<span class="op">=</span><span class="st">"infomap"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">clust.fast</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, </span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/NNGraphParam-class.html">NNGraphParam</a></span><span class="op">(</span>cluster.fun<span class="op">=</span><span class="st">"fast_greedy"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">clust.labprop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, </span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/NNGraphParam-class.html">NNGraphParam</a></span><span class="op">(</span>cluster.fun<span class="op">=</span><span class="st">"label_prop"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">clust.eigen</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, </span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/NNGraphParam-class.html">NNGraphParam</a></span><span class="op">(</span>cluster.fun<span class="op">=</span><span class="st">"leading_eigen"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span>
<span><span class="co"># Using a large pseudo-count for a smoother color transition</span></span>
<span><span class="co"># between 0 and 1 cell in each 'tab'.</span></span>
<span><span class="va">tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Infomap"</span>, <span class="va">clust.infomap</span><span class="op">)</span>, </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Walktrap"</span>, <span class="va">clust.walktrap</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ivw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">tab</span><span class="op">+</span><span class="fl">10</span><span class="op">)</span>, main<span class="op">=</span><span class="st">"Infomap vs Walktrap"</span>,</span>
<span>    color<span class="op">=</span><span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html">viridis</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span>, silent<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">gridExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span><span class="va">ivw</span><span class="op">[[</span><span class="fl">4</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="fig-Infomap_vs_Walktrap" class="quarto-figure quarto-figure-center quarto-float anchored" width="672">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-Infomap_vs_Walktrap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Clustering_files/figure-html/fig-Infomap_vs_Walktrap-1.png" id="fig-Infomap_vs_Walktrap" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-Infomap_vs_Walktrap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10.2
</figcaption></figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Fast"</span>, <span class="va">clust.fast</span><span class="op">)</span>, </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Walktrap"</span>, <span class="va">clust.walktrap</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fvw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">tab</span><span class="op">+</span><span class="fl">10</span><span class="op">)</span>, main<span class="op">=</span><span class="st">"Fast-greedy vs Walktrap"</span>,</span>
<span>    color<span class="op">=</span><span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html">viridis</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span>, silent<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">gridExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span><span class="va">fvw</span><span class="op">[[</span><span class="fl">4</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="fig-Fast-greedy_vs_Walktrap" class="quarto-figure quarto-figure-center quarto-float anchored" width="672">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-Fast-greedy_vs_Walktrap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Clustering_files/figure-html/fig-Fast-greedy_vs_Walktrap-1.png" id="fig-Fast-greedy_vs_Walktrap" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-Fast-greedy_vs_Walktrap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10.3
</figcaption></figure>
</div>
</div>
</div>
<p>涉及<a href="https://bioconductor.org/packages/3.18/bioc/html/scran.html">scran</a>的管道默认为基于秩（rank）的权重，后跟 Walktrap 聚类。 相比之下，Seurat 使用基于 Jaccard 的权重，然后是 Louvain 聚类。</p>
</section></section></section><section id="k-means-聚类" class="level2" data-number="10.2"><h2 data-number="10.2" class="anchored" data-anchor-id="k-means-聚类">
<span class="header-section-number">10.2</span> K-means 聚类</h2>
<p>k-means 聚类是一种经典的向量量化技术，可将细胞分成 k 个簇。通过随机初始配置 k 个质心，最小化簇内的平方和以实现每个细胞都被分配给最接近质心的簇， 。 我们通常设置 k以较大的值（例如细胞数的平方根）来获得细粒度（ fine-grained）的簇。</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">clust.kmeans</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, </span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/KmeansParam-class.html">KmeansParam</a></span><span class="op">(</span>centers<span class="op">=</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">clust.kmeans</span><span class="op">)</span></span>
<span><span class="co">#&gt; clust.kmeans</span></span>
<span><span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10 </span></span>
<span><span class="co">#&gt; 548  46 408 270 539 199 148 783 163 881</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/colLabels.html">colLabels</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">clust.kmeans</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html">plotReducedDim</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, <span class="st">"TSNE"</span>, colour_by<span class="op">=</span><span class="st">"label"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="Clustering_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">clust.kmeans2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, </span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/KmeansParam-class.html">KmeansParam</a></span><span class="op">(</span>centers<span class="op">=</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">clust.kmeans2</span><span class="op">)</span></span>
<span><span class="co">#&gt; clust.kmeans2</span></span>
<span><span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20 </span></span>
<span><span class="co">#&gt; 243  28 202 361 282 166 388 150 114 537 170  96  46 131 162 118 201 257 288  45</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/colLabels.html">colLabels</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">clust.kmeans2</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plot_reddim.html">plotTSNE</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, colour_by<span class="op">=</span><span class="st">"label"</span>, text_by<span class="op">=</span><span class="st">"label"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="Clustering_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<section id="mini-batch-k-means" class="level3" data-number="10.2.1"><h3 data-number="10.2.1" class="anchored" data-anchor-id="mini-batch-k-means">
<span class="header-section-number">10.2.1</span> mini-batch k-means</h3>
<p><a href="https://bioconductor.org/packages/3.18/bioc/html/mbkmeans.html">mbkmeans</a></p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">clust.mbkmeans</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>,</span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/MbkmeansParam-class.html">MbkmeansParam</a></span><span class="op">(</span>centers<span class="op">=</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">clust.mbkmeans</span><span class="op">)</span></span>
<span><span class="co">#&gt; clust.mbkmeans</span></span>
<span><span class="co">#&gt;    1    2    3    4    5    6    7    8    9   10 </span></span>
<span><span class="co">#&gt;  304  339  490   46  304  560   16  176  727 1023</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>就其本身而言， k-means 存在几个缺点：</p>
<ol type="1">
<li>它隐式偏向于半径相等的球形簇，这可能会导致在包含大小和形状不规则的分组的数据集上出现不直观的分区。</li>
<li>簇个数（质心数，centers） k 必须事先指定，并表示聚类分辨率的上限。k小于细胞类型数量时将始终导致两 种细胞类型的共聚类。相比之下，其他方法（如基于图的聚类）将遵循强分离，即使相关分辨率参数设置为低值。</li>
<li>它取决于随机选择的初始质心。 这需要多次运行来验证簇是否稳定。</li>
</ol></section><section id="two-step-modekmeans-nngraph" class="level3" data-number="10.2.2"><h3 data-number="10.2.2" class="anchored" data-anchor-id="two-step-modekmeans-nngraph">
<span class="header-section-number">10.2.2</span> “two-step” mode：Kmeans-nnGraph</h3>
<p>k-means 首先用于获得经过基于图的聚类的代表性质心。 然后，将每个细胞放置在相同的基于图的簇中，这也是k-means聚类分配的簇。</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Setting the seed due to the randomness of k-means.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">0101010</span><span class="op">)</span></span>
<span><span class="va">kgraph.clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>,</span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/TwoStepParam-class.html">TwoStepParam</a></span><span class="op">(</span></span>
<span>        first<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/KmeansParam-class.html">KmeansParam</a></span><span class="op">(</span>centers<span class="op">=</span><span class="fl">1000</span><span class="op">)</span>,</span>
<span>        second<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/NNGraphParam-class.html">NNGraphParam</a></span><span class="op">(</span>k<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">kgraph.clusters</span><span class="op">)</span></span>
<span><span class="co">#&gt; kgraph.clusters</span></span>
<span><span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10  11  12 </span></span>
<span><span class="co">#&gt; 191 854 506 541 541 892  46 120  29 132  47  86</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plot_reddim.html">plotTSNE</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, colour_by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="va">kgraph.clusters</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="fig-K-Graph_cluster" class="quarto-figure quarto-figure-center quarto-float anchored" width="672">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-K-Graph_cluster-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Clustering_files/figure-html/fig-K-Graph_cluster-1.png" id="fig-K-Graph_cluster" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-K-Graph_cluster-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10.4
</figcaption></figure>
</div>
</div>
</div>
</section></section><section id="hierarchical-clustering" class="level2" data-number="10.3"><h2 data-number="10.3" class="anchored" data-anchor-id="hierarchical-clustering">
<span class="header-section-number">10.3</span> Hierarchical clustering</h2>
<p>分层聚类是一种古老的技术，它根据样本彼此之间的相对相似性将样本排列到层次结构中。 大多数实现通过将最相似的示例加入到新集群中，然后将相似的集群加入到更大的集群中，依此类推，直到所有样本都属于单个集群。 此过程将获得一个树状图，该树状图定义了粒度逐渐增加的聚类。 分层聚类方法的变体主要区别在于它们选择如何执行集聚。 例如，完全连接旨在合并具有最小最大距离的簇，而 Ward 方法旨在最小化簇内方差的增加。</p>
<p>分层聚类的主要优势在于树状图的产生，可以定量捕获不同分辨率下的亚群之间的关系 <a href="#fig-hclust" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-hclust</span></a> 。 以高分辨率切割树状图也保证产生嵌套在以低分辨率切割获得的聚类中的聚类，这可能有助于解释 <a href="#fig-hclust-dynamicTreeCut" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-hclust-dynamicTreeCut</span></a> 。 树状图也是细胞从相对较新的共同祖先传代而来的数据的自然表示。</p>
<p>缺点是计算成本高，速度太慢。</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#--- loading ---#</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">scRNAseq</span><span class="op">)</span></span>
<span><span class="va">sce.416b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scRNAseq/man/LunSpikeInData.html">LunSpikeInData</a></span><span class="op">(</span>which<span class="op">=</span><span class="st">"416b"</span><span class="op">)</span> </span>
<span><span class="va">sce.416b</span><span class="op">$</span><span class="va">block</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">sce.416b</span><span class="op">$</span><span class="va">block</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#--- gene-annotation ---#</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span></span>
<span><span class="va">ens.mm.v97</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span><span class="op">[[</span><span class="st">"AH73905"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span><span class="op">$</span><span class="va">ENSEMBL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span></span>
<span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span> <span class="op">&lt;-</span> <span class="fu">mapIds</span><span class="op">(</span><span class="va">ens.mm.v97</span>, keys<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span>,</span>
<span>    keytype<span class="op">=</span><span class="st">"GENEID"</span>, column<span class="op">=</span><span class="st">"SYMBOL"</span><span class="op">)</span></span>
<span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span><span class="op">$</span><span class="va">SEQNAME</span> <span class="op">&lt;-</span> <span class="fu">mapIds</span><span class="op">(</span><span class="va">ens.mm.v97</span>, keys<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span>,</span>
<span>    keytype<span class="op">=</span><span class="st">"GENEID"</span>, column<span class="op">=</span><span class="st">"SEQNAME"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/">scater</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/uniquifyFeatureNames.html">uniquifyFeatureNames</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span><span class="op">$</span><span class="va">ENSEMBL</span>, </span>
<span>    <span class="fu">rowData</span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#--- quality-control ---#</span></span>
<span><span class="va">mito</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html">which</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span><span class="op">$</span><span class="va">SEQNAME</span><span class="op">==</span><span class="st">"MT"</span><span class="op">)</span></span>
<span><span class="va">stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/perCellQCMetrics.html">perCellQCMetrics</a></span><span class="op">(</span><span class="va">sce.416b</span>, subsets<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>Mt<span class="op">=</span><span class="va">mito</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">qc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/quickPerCellQC.html">quickPerCellQC</a></span><span class="op">(</span><span class="va">stats</span>, percent_subsets<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"subsets_Mt_percent"</span>,</span>
<span>    <span class="st">"altexps_ERCC_percent"</span><span class="op">)</span>, batch<span class="op">=</span><span class="va">sce.416b</span><span class="op">$</span><span class="va">block</span><span class="op">)</span></span>
<span><span class="va">sce.416b</span> <span class="op">&lt;-</span> <span class="va">sce.416b</span><span class="op">[</span>,<span class="op">!</span><span class="va">qc</span><span class="op">$</span><span class="va">discard</span><span class="op">]</span></span>
<span></span>
<span><span class="co">#--- normalization ---#</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span></span>
<span><span class="va">sce.416b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/computeSumFactors.html">computeSumFactors</a></span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span></span>
<span><span class="va">sce.416b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html">logNormCounts</a></span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#--- variance-modelling ---#</span></span>
<span><span class="va">dec.416b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVarWithSpikes.html">modelGeneVarWithSpikes</a></span><span class="op">(</span><span class="va">sce.416b</span>, <span class="st">"ERCC"</span>, block<span class="op">=</span><span class="va">sce.416b</span><span class="op">$</span><span class="va">block</span><span class="op">)</span></span>
<span><span class="va">chosen.hvgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/getTopHVGs.html">getTopHVGs</a></span><span class="op">(</span><span class="va">dec.416b</span>, prop<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#--- batch-correction ---#</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/limma/">limma</a></span><span class="op">)</span></span>
<span><span class="fu">assay</span><span class="op">(</span><span class="va">sce.416b</span>, <span class="st">"corrected"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/removeBatchEffect.html">removeBatchEffect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">logcounts</a></span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span>, </span>
<span>    design<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span><span class="va">sce.416b</span><span class="op">$</span><span class="va">phenotype</span><span class="op">)</span>, batch<span class="op">=</span><span class="va">sce.416b</span><span class="op">$</span><span class="va">block</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#--- dimensionality-reduction ---#</span></span>
<span><span class="va">sce.416b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html">runPCA</a></span><span class="op">(</span><span class="va">sce.416b</span>, ncomponents<span class="op">=</span><span class="fl">10</span>, subset_row<span class="op">=</span><span class="va">chosen.hvgs</span>,</span>
<span>    exprs_values<span class="op">=</span><span class="st">"corrected"</span>, BSPARAM<span class="op">=</span><span class="fu">BiocSingular</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/BiocSingularParam.html">ExactParam</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1010</span><span class="op">)</span></span>
<span><span class="va">sce.416b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runTSNE.html">runTSNE</a></span><span class="op">(</span><span class="va">sce.416b</span>, dimred<span class="op">=</span><span class="st">"PCA"</span>, perplexity<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce.416b</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 46604 185 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(3): counts logcounts corrected</span></span>
<span><span class="co">#&gt; rownames(46604): 4933401J01Rik Gm26206 ... CAAA01147332.1</span></span>
<span><span class="co">#&gt;   CBFB-MYH11-mcherry</span></span>
<span><span class="co">#&gt; rowData names(4): Length ENSEMBL SYMBOL SEQNAME</span></span>
<span><span class="co">#&gt; colnames(185): SLX-9555.N701_S502.C89V9ANXX.s_1.r_1</span></span>
<span><span class="co">#&gt;   SLX-9555.N701_S503.C89V9ANXX.s_1.r_1 ...</span></span>
<span><span class="co">#&gt;   SLX-11312.N712_S507.H5H5YBBXX.s_8.r_1</span></span>
<span><span class="co">#&gt;   SLX-11312.N712_S517.H5H5YBBXX.s_8.r_1</span></span>
<span><span class="co">#&gt; colData names(10): Source Name cell line ... block sizeFactor</span></span>
<span><span class="co">#&gt; reducedDimNames(2): PCA TSNE</span></span>
<span><span class="co">#&gt; mainExpName: endogenous</span></span>
<span><span class="co">#&gt; altExpNames(2): ERCC SIRV</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">hclust.416b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.416b</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>,</span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/HclustParam-class.html">HclustParam</a></span><span class="op">(</span>method<span class="op">=</span><span class="st">"ward.D2"</span><span class="op">)</span>, full<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">tree.416b</span> <span class="op">&lt;-</span> <span class="va">hclust.416b</span><span class="op">$</span><span class="va">objects</span><span class="op">$</span><span class="va">hclust</span></span>
<span></span>
<span><span class="co"># Making a prettier dendrogram.</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://talgalili.github.io/dendextend/">dendextend</a></span><span class="op">)</span></span>
<span><span class="va">tree.416b</span><span class="op">$</span><span class="va">labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_along</a></span><span class="op">(</span><span class="va">tree.416b</span><span class="op">$</span><span class="va">labels</span><span class="op">)</span></span>
<span><span class="va">dend</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html">as.dendrogram</a></span><span class="op">(</span><span class="va">tree.416b</span>, hang<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">combined.fac</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">sce.416b</span><span class="op">$</span><span class="va">block</span>, <span class="st">"."</span>, </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">" .*"</span>, <span class="st">""</span>, <span class="va">sce.416b</span><span class="op">$</span><span class="va">phenotype</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://talgalili.github.io/dendextend/reference/labels_colors.html">labels_colors</a></span><span class="op">(</span><span class="va">dend</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"20160113.wild"</span><span class="op">=</span><span class="st">"blue"</span>,</span>
<span>    <span class="st">"20160113.induced"</span><span class="op">=</span><span class="st">"red"</span>,</span>
<span>    <span class="st">"20160325.wild"</span><span class="op">=</span><span class="st">"dodgerblue"</span>,</span>
<span>    <span class="st">"20160325.induced"</span><span class="op">=</span><span class="st">"salmon"</span></span>
<span><span class="op">)</span><span class="op">[</span><span class="va">combined.fac</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/stats/order.dendrogram.html">order.dendrogram</a></span><span class="op">(</span><span class="va">dend</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">dend</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="fig-hclust" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-hclust-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Clustering_files/figure-html/fig-hclust-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-hclust-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10.5: 根据致癌基因诱导状态（红色为诱导，蓝色为对照）和起始板（浅色或深色）进行着色
</figcaption></figure>
</div>
</div>
</div>
<p>为了获得显式聚类 <a href="#fig-hclust-dynamicTreeCut" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-hclust-dynamicTreeCut</span></a> ，我们通过删除内部分支来“切割”树，使每个子树代表一个不同的聚类。</p>
<p><a href="https://cran.r-project.org/web/packages/dynamicTreeCut/index.html">dynamicTreeCut</a></p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">hclust.dyn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.416b</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>,</span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/HclustParam-class.html">HclustParam</a></span><span class="op">(</span>method<span class="op">=</span><span class="st">"ward.D2"</span>, cut.dynamic<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>        cut.params<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>minClusterSize<span class="op">=</span><span class="fl">10</span>, deepSplit<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="va">hclust.dyn</span><span class="op">)</span></span>
<span><span class="co">#&gt; hclust.dyn</span></span>
<span><span class="co">#&gt;  1  2  3  4 </span></span>
<span><span class="co">#&gt; 78 69 24 14</span></span>
<span></span>
<span><span class="fu"><a href="http://talgalili.github.io/dendextend/reference/labels_colors.html">labels_colors</a></span><span class="op">(</span><span class="va">dend</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">hclust.dyn</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/stats/order.dendrogram.html">order.dendrogram</a></span><span class="op">(</span><span class="va">dend</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">dend</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="fig-hclust-dynamicTreeCut" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-hclust-dynamicTreeCut-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Clustering_files/figure-html/fig-hclust-dynamicTreeCut-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-hclust-dynamicTreeCut-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10.6: 根据dynamic Cut中分配的聚类标识进行着色
</figcaption></figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/colLabels.html">colLabels</a></span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">hclust.dyn</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html">plotReducedDim</a></span><span class="op">(</span><span class="va">sce.416b</span>, <span class="st">"TSNE"</span>, colour_by<span class="op">=</span><span class="st">"label"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="fig-tsne-hclust-dynamicTreeCut" class="quarto-figure quarto-figure-center quarto-float anchored" width="672">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-tsne-hclust-dynamicTreeCut-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Clustering_files/figure-html/fig-tsne-hclust-dynamicTreeCut-1.png" id="fig-tsne-hclust-dynamicTreeCut" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-tsne-hclust-dynamicTreeCut-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10.7
</figcaption></figure>
</div>
</div>
</div>
<section id="two-step-mode-kmeans-hierarchical" class="level3" data-number="10.3.1"><h3 data-number="10.3.1" class="anchored" data-anchor-id="two-step-mode-kmeans-hierarchical">
<span class="header-section-number">10.3.1</span> “two-step” mode ：Kmeans-Hierarchical</h3>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Setting the seed due to the randomness of k-means.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1111</span><span class="op">)</span></span>
<span><span class="va">khclust.info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>,</span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/TwoStepParam-class.html">TwoStepParam</a></span><span class="op">(</span></span>
<span>        first<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/KmeansParam-class.html">KmeansParam</a></span><span class="op">(</span>centers<span class="op">=</span><span class="fl">1000</span><span class="op">)</span>,</span>
<span>        second<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/HclustParam-class.html">HclustParam</a></span><span class="op">(</span>method<span class="op">=</span><span class="st">"ward.D2"</span>, cut.dynamic<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>            cut.param<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>deepSplit<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="co"># for higher resolution.</span></span>
<span>    <span class="op">)</span>,</span>
<span>    full<span class="op">=</span><span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="va">khclust.info</span><span class="op">$</span><span class="va">clusters</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18 </span></span>
<span><span class="co">#&gt; 649 374 347 328 281 243 218 223 161 157 172 133 135 167  93 117  72 115</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plot_reddim.html">plotTSNE</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, colour_by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="va">khclust.info</span><span class="op">$</span><span class="va">clusters</span><span class="op">)</span>, </span>
<span>    text_by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="va">khclust.info</span><span class="op">$</span><span class="va">clusters</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="fig-K-H_cluster" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-K-H_cluster-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Clustering_files/figure-html/fig-K-H_cluster-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-K-H_cluster-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10.8: 根据k-means/hierarchical聚类组合中分配的集群的身份进行着色
</figcaption></figure>
</div>
</div>
</div>
<p>还可以检查在质心上构建的树状图 <a href="#fig-dendrogram-centroids" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-dendrogram-centroids</span></a> , 这提供了不同亚群之间相对相似性的更定量的可视化。</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">k.stats</span> <span class="op">&lt;-</span> <span class="va">khclust.info</span><span class="op">$</span><span class="va">objects</span><span class="op">$</span><span class="va">first</span></span>
<span><span class="va">tree.pbmc</span> <span class="op">&lt;-</span> <span class="va">khclust.info</span><span class="op">$</span><span class="va">objects</span><span class="op">$</span><span class="va">second</span><span class="op">$</span><span class="va">hclust</span></span>
<span></span>
<span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">tree.pbmc</span><span class="op">$</span><span class="va">labels</span><span class="op">)</span>, <span class="va">k.stats</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span></span>
<span><span class="va">final.clusters</span> <span class="op">&lt;-</span> <span class="va">khclust.info</span><span class="op">$</span><span class="va">clusters</span><span class="op">[</span><span class="va">m</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># TODO: expose scater color palette for easier re-use,</span></span>
<span><span class="co"># given that the default colors start getting recycled.</span></span>
<span><span class="va">dend</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html">as.dendrogram</a></span><span class="op">(</span><span class="va">tree.pbmc</span>, hang<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://talgalili.github.io/dendextend/reference/labels_colors.html">labels_colors</a></span><span class="op">(</span><span class="va">dend</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">final.clusters</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/stats/order.dendrogram.html">order.dendrogram</a></span><span class="op">(</span><span class="va">dend</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">dend</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="fig-dendrogram-centroids" class="quarto-figure quarto-figure-center quarto-float anchored" width="672">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-dendrogram-centroids-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Clustering_files/figure-html/fig-dendrogram-centroids-1.png" id="fig-dendrogram-centroids" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-dendrogram-centroids-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10.9
</figcaption></figure>
</div>
</div>
</div>
</section><section id="two-step-mode-kmeans-affinity" class="level3" data-number="10.3.2"><h3 data-number="10.3.2" class="anchored" data-anchor-id="two-step-mode-kmeans-affinity">
<span class="header-section-number">10.3.2</span> “two-step” mode ：Kmeans-Affinity</h3>
<p>相同的方法可用于加速任何基于距离矩阵的聚类方法。 例如， 使k-means质心通过亲和性传播（affinity propagation）执行聚类。在这个过程中，每个样本（如质心）选择自己或另一个样本作为其“示例exemplar”， 选择的适用性取决于样本之间的距离、每个样本的其他潜在样本以及具有相同所选样本的其他样本。 这些选择的迭代更新会产生一组聚类，其中每个聚类都是根据分配给同一示例的样本定义的。 与分层聚类不同，它不提供树状图dendrogram，但它也避免了切割树的额外复杂性—— 分辨率主要通过参数<code>q=</code>控制，该参数定义了样本将自己视为示例并因此形成自己的簇的强度。</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Setting the seed due to the randomness of k-means.</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/UBod/apcluster">apcluster</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1111</span><span class="op">)</span></span>
<span><span class="va">kaclust.info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>,</span>
<span>    BLUSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/TwoStepParam-class.html">TwoStepParam</a></span><span class="op">(</span></span>
<span>        first<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/KmeansParam-class.html">KmeansParam</a></span><span class="op">(</span>centers<span class="op">=</span><span class="fl">1000</span><span class="op">)</span>,</span>
<span>        second<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/bluster/man/AffinityParam-class.html">AffinityParam</a></span><span class="op">(</span>q<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span> <span class="co"># larger q =&gt; more clusters</span></span>
<span>    <span class="op">)</span>,</span>
<span>    full<span class="op">=</span><span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="va">kaclust.info</span><span class="op">$</span><span class="va">clusters</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18 </span></span>
<span><span class="co">#&gt; 847 144 392  93 405 264 170 209 379 187  24  20  45 121 536  27  64  58</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plot_reddim.html">plotTSNE</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, colour_by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="va">kaclust.info</span><span class="op">$</span><span class="va">clusters</span><span class="op">)</span>, </span>
<span>    text_by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="va">kaclust.info</span><span class="op">$</span><span class="va">clusters</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="fig-Kmeans-Affinity" class="quarto-figure quarto-figure-center quarto-float anchored" width="672">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-Kmeans-Affinity-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Clustering_files/figure-html/fig-Kmeans-Affinity-1.png" id="fig-Kmeans-Affinity" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-Kmeans-Affinity-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10.10
</figcaption></figure>
</div>
</div>
</div>
</section></section><section id="subclustering" class="level2" data-number="10.4"><h2 data-number="10.4" class="anchored" data-anchor-id="subclustering">
<span class="header-section-number">10.4</span> Subclustering</h2>
<p>提高分辨率的另一种简单方法是在单个聚类中重复特征选择和聚类，这旨在选择与内部结构更相关的 HVGs 和 PCs，通过避免不必要的功能产生的噪音来提高分辨率。 Subclustering 在缺乏不同亚群的情况下根据更适度的异质性分离细胞。</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">clust.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/clusterCells.html">clusterCells</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotExpression.html">plotExpression</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, features<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD3E"</span>, <span class="st">"CCR7"</span>, <span class="st">"CD69"</span>, <span class="st">"CD44"</span><span class="op">)</span>,</span>
<span>    x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="va">clust.full</span><span class="op">)</span>, colour_by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="va">clust.full</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="Clustering_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Repeating modelling and PCA on the subset.</span></span>
<span><span class="va">memory</span> <span class="op">&lt;-</span> <span class="fl">10L</span></span>
<span><span class="va">sce.memory</span> <span class="op">&lt;-</span> <span class="va">sce.pbmc</span><span class="op">[</span>,<span class="va">clust.full</span><span class="op">==</span><span class="va">memory</span><span class="op">]</span></span>
<span><span class="va">dec.memory</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVar.html">modelGeneVar</a></span><span class="op">(</span><span class="va">sce.memory</span><span class="op">)</span></span>
<span><span class="va">sce.memory</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/denoisePCA.html">denoisePCA</a></span><span class="op">(</span><span class="va">sce.memory</span>, technical<span class="op">=</span><span class="va">dec.memory</span>,</span>
<span>    subset.row<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/scran/man/getTopHVGs.html">getTopHVGs</a></span><span class="op">(</span><span class="va">dec.memory</span>, n<span class="op">=</span><span class="fl">5000</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g.memory</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/buildSNNGraph.html">buildSNNGraph</a></span><span class="op">(</span><span class="va">sce.memory</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span><span class="op">)</span></span>
<span><span class="va">clust.memory</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/cluster_walktrap.html">cluster_walktrap</a></span><span class="op">(</span><span class="va">g.memory</span><span class="op">)</span><span class="op">$</span><span class="va">membership</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotExpression.html">plotExpression</a></span><span class="op">(</span><span class="va">sce.memory</span>, features<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD8A"</span>, <span class="st">"CD4"</span><span class="op">)</span>,</span>
<span>    x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">clust.memory</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="Clustering_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>对于子聚类分析，定义一个自定义函数<code><a href="https://rdrr.io/pkg/scran/man/quickSubCluster.html">quickSubCluster()</a></code></p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1000010</span><span class="op">)</span></span>
<span><span class="va">subcluster.out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/quickSubCluster.html">quickSubCluster</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, groups<span class="op">=</span><span class="va">clust.full</span>,</span>
<span>    prepFUN<span class="op">=</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span> <span class="co"># Preparing the subsetted SCE for clustering.</span></span>
<span>        <span class="va">dec</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVar.html">modelGeneVar</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>        <span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/denoisePCA.html">denoisePCA</a></span><span class="op">(</span><span class="va">x</span>, technical<span class="op">=</span><span class="va">dec</span>,</span>
<span>            subset.row<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/scran/man/getTopHVGs.html">getTopHVGs</a></span><span class="op">(</span><span class="va">dec</span>, prop<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span>,</span>
<span>            BSPARAM<span class="op">=</span><span class="fu">BiocSingular</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/BiocSingularParam.html">IrlbaParam</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">}</span>,</span>
<span>    clusterFUN<span class="op">=</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span> <span class="co"># Performing the subclustering in the subset.</span></span>
<span>        <span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/buildSNNGraph.html">buildSNNGraph</a></span><span class="op">(</span><span class="va">x</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span>, k<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span>        <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/cluster_walktrap.html">cluster_walktrap</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">membership</span></span>
<span>    <span class="op">}</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># One SingleCellExperiment object per parent cluster:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">subcluster.out</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "1"  "2"  "3"  "4"  "5"  "6"  "7"  "8"  "9"  "10" "11" "12" "13" "14" "15"</span></span>
<span><span class="co">#&gt; [16] "16"</span></span>
<span><span class="co"># Looking at the subclustering for one example:</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="va">subcluster.out</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">subcluster</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 1.1 1.2 1.3 1.4 1.5 1.6 </span></span>
<span><span class="co">#&gt;  28  22  34  62  11  48</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->

</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/wanganlin00\.github\.io\/Bioinformatics\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./DimensionalityReduction.html" class="pagination-link" aria-label="降维">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">降维</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./MarkerGeneDetection.html" class="pagination-link" aria-label="conflicts_prefer(dplyr::filter)">
        <span class="nav-page-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">conflicts_prefer(dplyr::filter)</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb33" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="in">#| include: false</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="in">conflicts_prefer(GenomicRanges::setdiff)</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="in">#conflicts_prefer(dplyr::filter)</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a><span class="fu"># 聚类</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>聚类是一种无监督学习过程，旨在通过计算基因之间的欧几里得距离来识别具有相似转录组学谱的细胞，使用易于理解的离散标签来描述群体异质性，而不是试图理解细胞真正所在的高维流形。</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: loading-sce.pbmc_to_降维</span></span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a><span class="co">#--- loading ---#</span></span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DropletUtils)</span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a>sce.pbmc <span class="ot">&lt;-</span> DropletUtils<span class="sc">::</span><span class="fu">read10xCounts</span>(<span class="st">"data/OSCA/raw_gene_bc_matrices/GRCh38"</span>,</span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">col.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a><span class="co">#--- gene-annotation ---#</span></span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sce.pbmc) <span class="ot">&lt;-</span> <span class="fu">uniquifyFeatureNames</span>(</span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rowData</span>(sce.pbmc)<span class="sc">$</span>ID, <span class="fu">rowData</span>(sce.pbmc)<span class="sc">$</span>Symbol)</span>
<span id="cb33-27"><a href="#cb33-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-28"><a href="#cb33-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EnsDb.Hsapiens.v86)</span>
<span id="cb33-29"><a href="#cb33-29" aria-hidden="true" tabindex="-1"></a>location <span class="ot">&lt;-</span> <span class="fu">mapIds</span>(EnsDb.Hsapiens.v86, <span class="at">keys=</span><span class="fu">rowData</span>(sce.pbmc)<span class="sc">$</span>ID, </span>
<span id="cb33-30"><a href="#cb33-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">column=</span><span class="st">"SEQNAME"</span>, <span class="at">keytype=</span><span class="st">"GENEID"</span>)</span>
<span id="cb33-31"><a href="#cb33-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-32"><a href="#cb33-32" aria-hidden="true" tabindex="-1"></a><span class="co">#--- cell-detection ---#</span></span>
<span id="cb33-33"><a href="#cb33-33" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">100</span>)</span>
<span id="cb33-34"><a href="#cb33-34" aria-hidden="true" tabindex="-1"></a>e.out <span class="ot">&lt;-</span> <span class="fu">emptyDrops</span>(<span class="fu">counts</span>(sce.pbmc))</span>
<span id="cb33-35"><a href="#cb33-35" aria-hidden="true" tabindex="-1"></a>sce.pbmc <span class="ot">&lt;-</span> sce.pbmc[,<span class="fu">which</span>(e.out<span class="sc">$</span>FDR <span class="sc">&lt;=</span> <span class="fl">0.001</span>)]</span>
<span id="cb33-36"><a href="#cb33-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-37"><a href="#cb33-37" aria-hidden="true" tabindex="-1"></a><span class="co">#--- quality-control ---#</span></span>
<span id="cb33-38"><a href="#cb33-38" aria-hidden="true" tabindex="-1"></a>stats <span class="ot">&lt;-</span> <span class="fu">perCellQCMetrics</span>(sce.pbmc, <span class="at">subsets=</span><span class="fu">list</span>(<span class="at">Mito=</span><span class="fu">which</span>(location<span class="sc">==</span><span class="st">"MT"</span>)))</span>
<span id="cb33-39"><a href="#cb33-39" aria-hidden="true" tabindex="-1"></a>high.mito <span class="ot">&lt;-</span> <span class="fu">isOutlier</span>(stats<span class="sc">$</span>subsets_Mito_percent, <span class="at">type=</span><span class="st">"higher"</span>)</span>
<span id="cb33-40"><a href="#cb33-40" aria-hidden="true" tabindex="-1"></a>sce.pbmc <span class="ot">&lt;-</span> sce.pbmc[,<span class="sc">!</span>high.mito]</span>
<span id="cb33-41"><a href="#cb33-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-42"><a href="#cb33-42" aria-hidden="true" tabindex="-1"></a><span class="co">#--- normalization ---#</span></span>
<span id="cb33-43"><a href="#cb33-43" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scran)</span>
<span id="cb33-44"><a href="#cb33-44" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1000</span>)</span>
<span id="cb33-45"><a href="#cb33-45" aria-hidden="true" tabindex="-1"></a>clusters <span class="ot">&lt;-</span> <span class="fu">quickCluster</span>(sce.pbmc)</span>
<span id="cb33-46"><a href="#cb33-46" aria-hidden="true" tabindex="-1"></a>sce.pbmc <span class="ot">&lt;-</span> <span class="fu">computeSumFactors</span>(sce.pbmc, <span class="at">cluster=</span>clusters)</span>
<span id="cb33-47"><a href="#cb33-47" aria-hidden="true" tabindex="-1"></a>sce.pbmc <span class="ot">&lt;-</span> <span class="fu">logNormCounts</span>(sce.pbmc)</span>
<span id="cb33-48"><a href="#cb33-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-49"><a href="#cb33-49" aria-hidden="true" tabindex="-1"></a><span class="co">#--- variance-modelling ---#</span></span>
<span id="cb33-50"><a href="#cb33-50" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1001</span>)</span>
<span id="cb33-51"><a href="#cb33-51" aria-hidden="true" tabindex="-1"></a>dec.pbmc <span class="ot">&lt;-</span> <span class="fu">modelGeneVarByPoisson</span>(sce.pbmc)</span>
<span id="cb33-52"><a href="#cb33-52" aria-hidden="true" tabindex="-1"></a>top.pbmc <span class="ot">&lt;-</span> <span class="fu">getTopHVGs</span>(dec.pbmc, <span class="at">prop=</span><span class="fl">0.1</span>)</span>
<span id="cb33-53"><a href="#cb33-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-54"><a href="#cb33-54" aria-hidden="true" tabindex="-1"></a><span class="co">#--- dimensionality-reduction ---#</span></span>
<span id="cb33-55"><a href="#cb33-55" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">10000</span>)</span>
<span id="cb33-56"><a href="#cb33-56" aria-hidden="true" tabindex="-1"></a>sce.pbmc <span class="ot">&lt;-</span> <span class="fu">denoisePCA</span>(sce.pbmc, <span class="at">subset.row=</span>top.pbmc, <span class="at">technical=</span>dec.pbmc)</span>
<span id="cb33-57"><a href="#cb33-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-58"><a href="#cb33-58" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">100000</span>)</span>
<span id="cb33-59"><a href="#cb33-59" aria-hidden="true" tabindex="-1"></a>sce.pbmc <span class="ot">&lt;-</span> <span class="fu">runTSNE</span>(sce.pbmc, <span class="at">dimred=</span><span class="st">"PCA"</span>)</span>
<span id="cb33-60"><a href="#cb33-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-61"><a href="#cb33-61" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1000000</span>)</span>
<span id="cb33-62"><a href="#cb33-62" aria-hidden="true" tabindex="-1"></a>sce.pbmc <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(sce.pbmc, <span class="at">dimred=</span><span class="st">"PCA"</span>)</span>
<span id="cb33-63"><a href="#cb33-63" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-64"><a href="#cb33-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-67"><a href="#cb33-67" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-68"><a href="#cb33-68" aria-hidden="true" tabindex="-1"></a>sce.pbmc</span>
<span id="cb33-69"><a href="#cb33-69" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-70"><a href="#cb33-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-71"><a href="#cb33-71" aria-hidden="true" tabindex="-1"></a><span class="fu">## Graph-based clustering</span></span>
<span id="cb33-72"><a href="#cb33-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-73"><a href="#cb33-73" aria-hidden="true" tabindex="-1"></a>基于图的聚类因其在 Seurat 中的使用而广为普及，是一种灵活且可扩展的技术，用于聚类大型 scRNA-seq 数据集。首先构建一个图，其中每个节点都是一个连接到高维空间中 nearest neighbors 的细胞。根据所涉及的细胞之间的相似性对边缘进行加权，对关系更密切的细胞赋予更高的权重。然后应用算法来识别在同一群落中的细胞比在非同一群落的细胞更紧密的“簇”。</span>
<span id="cb33-74"><a href="#cb33-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-75"><a href="#cb33-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-76"><a href="#cb33-76" aria-hidden="true" tabindex="-1"></a>基于图的聚类的主要优势在于其可伸缩性。 它只需要一个 k-nearest neighbor 搜索，避免了对簇的形状或每个簇内细胞的分布做出强有力的假设，例如k-means（有利于球面簇）或高斯混合模型（需要正态性）。</span>
<span id="cb33-77"><a href="#cb33-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-78"><a href="#cb33-78" aria-hidden="true" tabindex="-1"></a>主要缺点是在图构造之后，不会保留超出邻近细胞关系的信息。在细胞密度差异的数据产生一些实际后果。</span>
<span id="cb33-79"><a href="#cb33-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-82"><a href="#cb33-82" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-83"><a href="#cb33-83" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scran)</span>
<span id="cb33-84"><a href="#cb33-84" aria-hidden="true" tabindex="-1"></a>nn.clusters <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>)</span>
<span id="cb33-85"><a href="#cb33-85" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(nn.clusters)</span>
<span id="cb33-86"><a href="#cb33-86" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-87"><a href="#cb33-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-90"><a href="#cb33-90" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-91"><a href="#cb33-91" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb33-92"><a href="#cb33-92" aria-hidden="true" tabindex="-1"></a><span class="fu">colLabels</span>(sce.pbmc) <span class="ot">&lt;-</span> nn.clusters</span>
<span id="cb33-93"><a href="#cb33-93" aria-hidden="true" tabindex="-1"></a><span class="fu">plotReducedDim</span>(sce.pbmc, <span class="st">"TSNE"</span>, <span class="at">colour_by=</span><span class="st">"label"</span>)</span>
<span id="cb33-94"><a href="#cb33-94" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-95"><a href="#cb33-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-96"><a href="#cb33-96" aria-hidden="true" tabindex="-1"></a>默认情况下，使用每个细胞的 10 个最近邻来构造共享的最近邻图。如果两个细胞的任何最近邻共享，则通过一条边连接， 边权重由共享最近邻的最高平均秩定义。<span class="in">` walktrap`</span></span>
<span id="cb33-97"><a href="#cb33-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-98"><a href="#cb33-98" aria-hidden="true" tabindex="-1"></a>显式指定参数,改变参数切换不同的聚类算法</span>
<span id="cb33-99"><a href="#cb33-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-102"><a href="#cb33-102" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-103"><a href="#cb33-103" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(bluster)</span>
<span id="cb33-104"><a href="#cb33-104" aria-hidden="true" tabindex="-1"></a>nn.clusters2 <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, </span>
<span id="cb33-105"><a href="#cb33-105" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">SNNGraphParam</span>(<span class="at">k=</span><span class="dv">10</span>, <span class="at">type=</span><span class="st">"rank"</span>, <span class="at">cluster.fun=</span><span class="st">"walktrap"</span>))</span>
<span id="cb33-106"><a href="#cb33-106" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(nn.clusters2)</span>
<span id="cb33-107"><a href="#cb33-107" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-108"><a href="#cb33-108" aria-hidden="true" tabindex="-1"></a>通过在<span class="in">`clusterCells()`</span>中指定<span class="in">`full=TRUE`</span>来获取图本身，这样做将返回聚类分析期间使用的所有中间结构，包括 <span class="co">[</span><span class="ot">igraph</span><span class="co">](https://cran.r-project.org/web/packages/igraph/index.html)</span> 包中的图形对象。 该图可以使用力导向布局(force-directed layout)进行可视化</span>
<span id="cb33-109"><a href="#cb33-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-112"><a href="#cb33-112" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-113"><a href="#cb33-113" aria-hidden="true" tabindex="-1"></a>nn.clust.info <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, <span class="at">full=</span><span class="cn">TRUE</span>)</span>
<span id="cb33-114"><a href="#cb33-114" aria-hidden="true" tabindex="-1"></a>nn.clust.info<span class="sc">$</span>objects<span class="sc">$</span>graph</span>
<span id="cb33-115"><a href="#cb33-115" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-118"><a href="#cb33-118" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-119"><a href="#cb33-119" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-Force-directed-layout--shared-nearest-neighbor-graph</span></span>
<span id="cb33-120"><a href="#cb33-120" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">11000</span>)</span>
<span id="cb33-121"><a href="#cb33-121" aria-hidden="true" tabindex="-1"></a><span class="fu">reducedDim</span>(sce.pbmc, <span class="st">"force"</span>) <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">layout_with_fr</span>(nn.clust.info<span class="sc">$</span>objects<span class="sc">$</span>graph)</span>
<span id="cb33-122"><a href="#cb33-122" aria-hidden="true" tabindex="-1"></a><span class="fu">plotReducedDim</span>(sce.pbmc, <span class="at">colour_by=</span><span class="st">"label"</span>, <span class="at">dimred=</span><span class="st">"force"</span>)</span>
<span id="cb33-123"><a href="#cb33-123" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-124"><a href="#cb33-124" aria-hidden="true" tabindex="-1"></a>此外，该图还可用于生成有关基于图的聚类行为的详细诊断。</span>
<span id="cb33-125"><a href="#cb33-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-126"><a href="#cb33-126" aria-hidden="true" tabindex="-1"></a><span class="fu">### 参数</span></span>
<span id="cb33-127"><a href="#cb33-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-128"><a href="#cb33-128" aria-hidden="true" tabindex="-1"></a><span class="fu">#### ` k ` 最近邻数目</span></span>
<span id="cb33-129"><a href="#cb33-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-130"><a href="#cb33-130" aria-hidden="true" tabindex="-1"></a>用于构造图的最近邻的数量，这控制了聚类的分辨率，其中更高的分辨率（k越小）是更互联的图和更多的聚类。</span>
<span id="cb33-131"><a href="#cb33-131" aria-hidden="true" tabindex="-1"></a>最近邻越多，聚类越少。</span>
<span id="cb33-132"><a href="#cb33-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-135"><a href="#cb33-135" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-136"><a href="#cb33-136" aria-hidden="true" tabindex="-1"></a><span class="co"># More resolved.</span></span>
<span id="cb33-137"><a href="#cb33-137" aria-hidden="true" tabindex="-1"></a>clust<span class="fl">.5</span> <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, <span class="at">BLUSPARAM=</span><span class="fu">NNGraphParam</span>(<span class="at">k=</span><span class="dv">5</span>))</span>
<span id="cb33-138"><a href="#cb33-138" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(clust<span class="fl">.5</span>)</span>
<span id="cb33-139"><a href="#cb33-139" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-140"><a href="#cb33-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-143"><a href="#cb33-143" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-144"><a href="#cb33-144" aria-hidden="true" tabindex="-1"></a><span class="co"># Less resolved.</span></span>
<span id="cb33-145"><a href="#cb33-145" aria-hidden="true" tabindex="-1"></a>clust<span class="fl">.50</span> <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, <span class="at">BLUSPARAM=</span><span class="fu">NNGraphParam</span>(<span class="at">k=</span><span class="dv">50</span>))</span>
<span id="cb33-146"><a href="#cb33-146" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(clust<span class="fl">.50</span>)</span>
<span id="cb33-147"><a href="#cb33-147" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-148"><a href="#cb33-148" aria-hidden="true" tabindex="-1"></a><span class="fu">#### ` type ` 边缘加权方法</span></span>
<span id="cb33-149"><a href="#cb33-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-150"><a href="#cb33-150" aria-hidden="true" tabindex="-1"></a><span class="in">`type="number" `</span>将根据两个细胞之间共享的最近邻的数量对边进行加权。</span>
<span id="cb33-151"><a href="#cb33-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-154"><a href="#cb33-154" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-155"><a href="#cb33-155" aria-hidden="true" tabindex="-1"></a>clust.num <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, </span>
<span id="cb33-156"><a href="#cb33-156" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">NNGraphParam</span>(<span class="at">type=</span><span class="st">"number"</span>))</span>
<span id="cb33-157"><a href="#cb33-157" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(clust.num)</span>
<span id="cb33-158"><a href="#cb33-158" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-159"><a href="#cb33-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-160"><a href="#cb33-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-161"><a href="#cb33-161" aria-hidden="true" tabindex="-1"></a><span class="in">`type="jaccard" `</span>将根据两组最近邻的 Jaccard 指数对边缘进行加权。</span>
<span id="cb33-162"><a href="#cb33-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-165"><a href="#cb33-165" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-166"><a href="#cb33-166" aria-hidden="true" tabindex="-1"></a>clust.jaccard <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, </span>
<span id="cb33-167"><a href="#cb33-167" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">NNGraphParam</span>(<span class="at">type=</span><span class="st">"jaccard"</span>))</span>
<span id="cb33-168"><a href="#cb33-168" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(clust.jaccard)</span>
<span id="cb33-169"><a href="#cb33-169" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-170"><a href="#cb33-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-171"><a href="#cb33-171" aria-hidden="true" tabindex="-1"></a>不对边缘加权</span>
<span id="cb33-172"><a href="#cb33-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-175"><a href="#cb33-175" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-176"><a href="#cb33-176" aria-hidden="true" tabindex="-1"></a>clust.none <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, </span>
<span id="cb33-177"><a href="#cb33-177" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">KNNGraphParam</span>())</span>
<span id="cb33-178"><a href="#cb33-178" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(clust.none)</span>
<span id="cb33-179"><a href="#cb33-179" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-180"><a href="#cb33-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-181"><a href="#cb33-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-182"><a href="#cb33-182" aria-hidden="true" tabindex="-1"></a><span class="fu">#### ` cluster.fun ` 簇识别方法</span></span>
<span id="cb33-183"><a href="#cb33-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-184"><a href="#cb33-184" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">igraph</span><span class="co">](https://cran.r-project.org/web/packages/igraph/index.html)</span> </span>
<span id="cb33-185"><a href="#cb33-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-188"><a href="#cb33-188" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-189"><a href="#cb33-189" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb33-190"><a href="#cb33-190" aria-hidden="true" tabindex="-1"></a>clust.walktrap <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, </span>
<span id="cb33-191"><a href="#cb33-191" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">NNGraphParam</span>(<span class="at">cluster.fun=</span><span class="st">"walktrap"</span>))</span>
<span id="cb33-192"><a href="#cb33-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-193"><a href="#cb33-193" aria-hidden="true" tabindex="-1"></a>clust.louvain <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, </span>
<span id="cb33-194"><a href="#cb33-194" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">NNGraphParam</span>(<span class="at">cluster.fun=</span><span class="st">"louvain"</span>))</span>
<span id="cb33-195"><a href="#cb33-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-196"><a href="#cb33-196" aria-hidden="true" tabindex="-1"></a>clust.infomap <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, </span>
<span id="cb33-197"><a href="#cb33-197" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">NNGraphParam</span>(<span class="at">cluster.fun=</span><span class="st">"infomap"</span>))</span>
<span id="cb33-198"><a href="#cb33-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-199"><a href="#cb33-199" aria-hidden="true" tabindex="-1"></a>clust.fast <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, </span>
<span id="cb33-200"><a href="#cb33-200" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">NNGraphParam</span>(<span class="at">cluster.fun=</span><span class="st">"fast_greedy"</span>))</span>
<span id="cb33-201"><a href="#cb33-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-202"><a href="#cb33-202" aria-hidden="true" tabindex="-1"></a>clust.labprop <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, </span>
<span id="cb33-203"><a href="#cb33-203" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">NNGraphParam</span>(<span class="at">cluster.fun=</span><span class="st">"label_prop"</span>))</span>
<span id="cb33-204"><a href="#cb33-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-205"><a href="#cb33-205" aria-hidden="true" tabindex="-1"></a>clust.eigen <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, </span>
<span id="cb33-206"><a href="#cb33-206" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">NNGraphParam</span>(<span class="at">cluster.fun=</span><span class="st">"leading_eigen"</span>))</span>
<span id="cb33-207"><a href="#cb33-207" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-208"><a href="#cb33-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-209"><a href="#cb33-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-212"><a href="#cb33-212" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-213"><a href="#cb33-213" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-Infomap_vs_Walktrap</span></span>
<span id="cb33-214"><a href="#cb33-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-215"><a href="#cb33-215" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb33-216"><a href="#cb33-216" aria-hidden="true" tabindex="-1"></a><span class="co"># Using a large pseudo-count for a smoother color transition</span></span>
<span id="cb33-217"><a href="#cb33-217" aria-hidden="true" tabindex="-1"></a><span class="co"># between 0 and 1 cell in each 'tab'.</span></span>
<span id="cb33-218"><a href="#cb33-218" aria-hidden="true" tabindex="-1"></a>tab <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="fu">paste</span>(<span class="st">"Infomap"</span>, clust.infomap), </span>
<span id="cb33-219"><a href="#cb33-219" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste</span>(<span class="st">"Walktrap"</span>, clust.walktrap))</span>
<span id="cb33-220"><a href="#cb33-220" aria-hidden="true" tabindex="-1"></a>ivw <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(<span class="fu">log10</span>(tab<span class="sc">+</span><span class="dv">10</span>), <span class="at">main=</span><span class="st">"Infomap vs Walktrap"</span>,</span>
<span id="cb33-221"><a href="#cb33-221" aria-hidden="true" tabindex="-1"></a>    <span class="at">color=</span>viridis<span class="sc">::</span><span class="fu">viridis</span>(<span class="dv">100</span>), <span class="at">silent=</span><span class="cn">TRUE</span>)</span>
<span id="cb33-222"><a href="#cb33-222" aria-hidden="true" tabindex="-1"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(ivw[[<span class="dv">4</span>]])</span>
<span id="cb33-223"><a href="#cb33-223" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-224"><a href="#cb33-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-227"><a href="#cb33-227" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-228"><a href="#cb33-228" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-Fast-greedy_vs_Walktrap</span></span>
<span id="cb33-229"><a href="#cb33-229" aria-hidden="true" tabindex="-1"></a>tab <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="fu">paste</span>(<span class="st">"Fast"</span>, clust.fast), </span>
<span id="cb33-230"><a href="#cb33-230" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste</span>(<span class="st">"Walktrap"</span>, clust.walktrap))</span>
<span id="cb33-231"><a href="#cb33-231" aria-hidden="true" tabindex="-1"></a>fvw <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(<span class="fu">log10</span>(tab<span class="sc">+</span><span class="dv">10</span>), <span class="at">main=</span><span class="st">"Fast-greedy vs Walktrap"</span>,</span>
<span id="cb33-232"><a href="#cb33-232" aria-hidden="true" tabindex="-1"></a>    <span class="at">color=</span>viridis<span class="sc">::</span><span class="fu">viridis</span>(<span class="dv">100</span>), <span class="at">silent=</span><span class="cn">TRUE</span>)</span>
<span id="cb33-233"><a href="#cb33-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-234"><a href="#cb33-234" aria-hidden="true" tabindex="-1"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(fvw[[<span class="dv">4</span>]])</span>
<span id="cb33-235"><a href="#cb33-235" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-236"><a href="#cb33-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-237"><a href="#cb33-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-238"><a href="#cb33-238" aria-hidden="true" tabindex="-1"></a>涉及<span class="co">[</span><span class="ot"> scran </span><span class="co">](https://bioconductor.org/packages/3.18/bioc/html/scran.html)</span>的管道默认为基于秩（rank）的权重，后跟 Walktrap 聚类。 相比之下，Seurat 使用基于 Jaccard 的权重，然后是 Louvain 聚类。 </span>
<span id="cb33-239"><a href="#cb33-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-240"><a href="#cb33-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-241"><a href="#cb33-241" aria-hidden="true" tabindex="-1"></a><span class="fu">## K-means 聚类</span></span>
<span id="cb33-242"><a href="#cb33-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-243"><a href="#cb33-243" aria-hidden="true" tabindex="-1"></a>k-means 聚类是一种经典的向量量化技术，可将细胞分成 k 个簇。通过随机初始配置 k 个质心，最小化簇内的平方和以实现每个细胞都被分配给最接近质心的簇，</span>
<span id="cb33-244"><a href="#cb33-244" aria-hidden="true" tabindex="-1"></a>。 我们通常设置 k以较大的值（例如细胞数的平方根）来获得细粒度（ fine-grained）的簇。 </span>
<span id="cb33-245"><a href="#cb33-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-248"><a href="#cb33-248" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-249"><a href="#cb33-249" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">100</span>)</span>
<span id="cb33-250"><a href="#cb33-250" aria-hidden="true" tabindex="-1"></a>clust.kmeans <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, </span>
<span id="cb33-251"><a href="#cb33-251" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">KmeansParam</span>(<span class="at">centers=</span><span class="dv">10</span>))</span>
<span id="cb33-252"><a href="#cb33-252" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(clust.kmeans)</span>
<span id="cb33-253"><a href="#cb33-253" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-254"><a href="#cb33-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-257"><a href="#cb33-257" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-258"><a href="#cb33-258" aria-hidden="true" tabindex="-1"></a><span class="fu">colLabels</span>(sce.pbmc) <span class="ot">&lt;-</span> clust.kmeans</span>
<span id="cb33-259"><a href="#cb33-259" aria-hidden="true" tabindex="-1"></a><span class="fu">plotReducedDim</span>(sce.pbmc, <span class="st">"TSNE"</span>, <span class="at">colour_by=</span><span class="st">"label"</span>)</span>
<span id="cb33-260"><a href="#cb33-260" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-261"><a href="#cb33-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-264"><a href="#cb33-264" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-265"><a href="#cb33-265" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">100</span>)</span>
<span id="cb33-266"><a href="#cb33-266" aria-hidden="true" tabindex="-1"></a>clust.kmeans2 <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, </span>
<span id="cb33-267"><a href="#cb33-267" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">KmeansParam</span>(<span class="at">centers=</span><span class="dv">20</span>))</span>
<span id="cb33-268"><a href="#cb33-268" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(clust.kmeans2)</span>
<span id="cb33-269"><a href="#cb33-269" aria-hidden="true" tabindex="-1"></a><span class="fu">colLabels</span>(sce.pbmc) <span class="ot">&lt;-</span> clust.kmeans2</span>
<span id="cb33-270"><a href="#cb33-270" aria-hidden="true" tabindex="-1"></a><span class="fu">plotTSNE</span>(sce.pbmc, <span class="at">colour_by=</span><span class="st">"label"</span>, <span class="at">text_by=</span><span class="st">"label"</span>)</span>
<span id="cb33-271"><a href="#cb33-271" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-272"><a href="#cb33-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-273"><a href="#cb33-273" aria-hidden="true" tabindex="-1"></a><span class="fu">### mini-batch k-means </span></span>
<span id="cb33-274"><a href="#cb33-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-275"><a href="#cb33-275" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">mbkmeans</span><span class="co">](https://bioconductor.org/packages/3.18/bioc/html/mbkmeans.html)</span></span>
<span id="cb33-276"><a href="#cb33-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-279"><a href="#cb33-279" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-280"><a href="#cb33-280" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">100</span>)</span>
<span id="cb33-281"><a href="#cb33-281" aria-hidden="true" tabindex="-1"></a>clust.mbkmeans <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>,</span>
<span id="cb33-282"><a href="#cb33-282" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">MbkmeansParam</span>(<span class="at">centers=</span><span class="dv">10</span>))</span>
<span id="cb33-283"><a href="#cb33-283" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(clust.mbkmeans)</span>
<span id="cb33-284"><a href="#cb33-284" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-285"><a href="#cb33-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-286"><a href="#cb33-286" aria-hidden="true" tabindex="-1"></a>就其本身而言， k-means 存在几个缺点：</span>
<span id="cb33-287"><a href="#cb33-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-288"><a href="#cb33-288" aria-hidden="true" tabindex="-1"></a><span class="ss">1.  </span>它隐式偏向于半径相等的球形簇，这可能会导致在包含大小和形状不规则的分组的数据集上出现不直观的分区。</span>
<span id="cb33-289"><a href="#cb33-289" aria-hidden="true" tabindex="-1"></a><span class="ss">2.  </span>簇个数（质心数，centers） k 必须事先指定，并表示聚类分辨率的上限。k小于细胞类型数量时将始终导致两</span>
<span id="cb33-290"><a href="#cb33-290" aria-hidden="true" tabindex="-1"></a>种细胞类型的共聚类。相比之下，其他方法（如基于图的聚类）将遵循强分离，即使相关分辨率参数设置为低值。</span>
<span id="cb33-291"><a href="#cb33-291" aria-hidden="true" tabindex="-1"></a><span class="ss">3.  </span>它取决于随机选择的初始质心。 这需要多次运行来验证簇是否稳定。</span>
<span id="cb33-292"><a href="#cb33-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-293"><a href="#cb33-293" aria-hidden="true" tabindex="-1"></a><span class="fu">### "two-step" mode：Kmeans-nnGraph</span></span>
<span id="cb33-294"><a href="#cb33-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-295"><a href="#cb33-295" aria-hidden="true" tabindex="-1"></a>k-means 首先用于获得经过基于图的聚类的代表性质心。</span>
<span id="cb33-296"><a href="#cb33-296" aria-hidden="true" tabindex="-1"></a>然后，将每个细胞放置在相同的基于图的簇中，这也是k-means聚类分配的簇。</span>
<span id="cb33-297"><a href="#cb33-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-300"><a href="#cb33-300" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-301"><a href="#cb33-301" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-K-Graph_cluster</span></span>
<span id="cb33-302"><a href="#cb33-302" aria-hidden="true" tabindex="-1"></a><span class="co"># Setting the seed due to the randomness of k-means.</span></span>
<span id="cb33-303"><a href="#cb33-303" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">0101010</span>)</span>
<span id="cb33-304"><a href="#cb33-304" aria-hidden="true" tabindex="-1"></a>kgraph.clusters <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>,</span>
<span id="cb33-305"><a href="#cb33-305" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">TwoStepParam</span>(</span>
<span id="cb33-306"><a href="#cb33-306" aria-hidden="true" tabindex="-1"></a>        <span class="at">first=</span><span class="fu">KmeansParam</span>(<span class="at">centers=</span><span class="dv">1000</span>),</span>
<span id="cb33-307"><a href="#cb33-307" aria-hidden="true" tabindex="-1"></a>        <span class="at">second=</span><span class="fu">NNGraphParam</span>(<span class="at">k=</span><span class="dv">5</span>)</span>
<span id="cb33-308"><a href="#cb33-308" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb33-309"><a href="#cb33-309" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb33-310"><a href="#cb33-310" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(kgraph.clusters)</span>
<span id="cb33-311"><a href="#cb33-311" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-312"><a href="#cb33-312" aria-hidden="true" tabindex="-1"></a><span class="fu">plotTSNE</span>(sce.pbmc, <span class="at">colour_by=</span><span class="fu">I</span>(kgraph.clusters))</span>
<span id="cb33-313"><a href="#cb33-313" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-314"><a href="#cb33-314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-315"><a href="#cb33-315" aria-hidden="true" tabindex="-1"></a><span class="fu">## Hierarchical clustering</span></span>
<span id="cb33-316"><a href="#cb33-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-317"><a href="#cb33-317" aria-hidden="true" tabindex="-1"></a>分层聚类是一种古老的技术，它根据样本彼此之间的相对相似性将样本排列到层次结构中。 大多数实现通过将最相似的示例加入到新集群中，然后将相似的集群加入到更大的集群中，依此类推，直到所有样本都属于单个集群。 此过程将获得一个树状图，该树状图定义了粒度逐渐增加的聚类。 分层聚类方法的变体主要区别在于它们选择如何执行集聚。 例如，完全连接旨在合并具有最小最大距离的簇，而 Ward 方法旨在最小化簇内方差的增加。</span>
<span id="cb33-318"><a href="#cb33-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-319"><a href="#cb33-319" aria-hidden="true" tabindex="-1"></a>分层聚类的主要优势在于树状图的产生，可以定量捕获不同分辨率下的亚群之间的关系 @fig-hclust 。 以高分辨率切割树状图也保证产生嵌套在以低分辨率切割获得的聚类中的聚类，这可能有助于解释 @fig-hclust-dynamicTreeCut 。</span>
<span id="cb33-320"><a href="#cb33-320" aria-hidden="true" tabindex="-1"></a>树状图也是细胞从相对较新的共同祖先传代而来的数据的自然表示。</span>
<span id="cb33-321"><a href="#cb33-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-322"><a href="#cb33-322" aria-hidden="true" tabindex="-1"></a>缺点是计算成本高，速度太慢。</span>
<span id="cb33-323"><a href="#cb33-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-326"><a href="#cb33-326" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-327"><a href="#cb33-327" aria-hidden="true" tabindex="-1"></a><span class="co">#--- loading ---#</span></span>
<span id="cb33-328"><a href="#cb33-328" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scRNAseq)</span>
<span id="cb33-329"><a href="#cb33-329" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b <span class="ot">&lt;-</span> <span class="fu">LunSpikeInData</span>(<span class="at">which=</span><span class="st">"416b"</span>) </span>
<span id="cb33-330"><a href="#cb33-330" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b<span class="sc">$</span>block <span class="ot">&lt;-</span> <span class="fu">factor</span>(sce<span class="fl">.416</span>b<span class="sc">$</span>block)</span>
<span id="cb33-331"><a href="#cb33-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-332"><a href="#cb33-332" aria-hidden="true" tabindex="-1"></a><span class="co">#--- gene-annotation ---#</span></span>
<span id="cb33-333"><a href="#cb33-333" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationHub)</span>
<span id="cb33-334"><a href="#cb33-334" aria-hidden="true" tabindex="-1"></a>ens.mm.v97 <span class="ot">&lt;-</span> <span class="fu">AnnotationHub</span>()[[<span class="st">"AH73905"</span>]]</span>
<span id="cb33-335"><a href="#cb33-335" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(sce<span class="fl">.416</span>b)<span class="sc">$</span>ENSEMBL <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sce<span class="fl">.416</span>b)</span>
<span id="cb33-336"><a href="#cb33-336" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(sce<span class="fl">.416</span>b)<span class="sc">$</span>SYMBOL <span class="ot">&lt;-</span> <span class="fu">mapIds</span>(ens.mm.v97, <span class="at">keys=</span><span class="fu">rownames</span>(sce<span class="fl">.416</span>b),</span>
<span id="cb33-337"><a href="#cb33-337" aria-hidden="true" tabindex="-1"></a>    <span class="at">keytype=</span><span class="st">"GENEID"</span>, <span class="at">column=</span><span class="st">"SYMBOL"</span>)</span>
<span id="cb33-338"><a href="#cb33-338" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(sce<span class="fl">.416</span>b)<span class="sc">$</span>SEQNAME <span class="ot">&lt;-</span> <span class="fu">mapIds</span>(ens.mm.v97, <span class="at">keys=</span><span class="fu">rownames</span>(sce<span class="fl">.416</span>b),</span>
<span id="cb33-339"><a href="#cb33-339" aria-hidden="true" tabindex="-1"></a>    <span class="at">keytype=</span><span class="st">"GENEID"</span>, <span class="at">column=</span><span class="st">"SEQNAME"</span>)</span>
<span id="cb33-340"><a href="#cb33-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-341"><a href="#cb33-341" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb33-342"><a href="#cb33-342" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sce<span class="fl">.416</span>b) <span class="ot">&lt;-</span> <span class="fu">uniquifyFeatureNames</span>(<span class="fu">rowData</span>(sce<span class="fl">.416</span>b)<span class="sc">$</span>ENSEMBL, </span>
<span id="cb33-343"><a href="#cb33-343" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rowData</span>(sce<span class="fl">.416</span>b)<span class="sc">$</span>SYMBOL)</span>
<span id="cb33-344"><a href="#cb33-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-345"><a href="#cb33-345" aria-hidden="true" tabindex="-1"></a><span class="co">#--- quality-control ---#</span></span>
<span id="cb33-346"><a href="#cb33-346" aria-hidden="true" tabindex="-1"></a>mito <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">rowData</span>(sce<span class="fl">.416</span>b)<span class="sc">$</span>SEQNAME<span class="sc">==</span><span class="st">"MT"</span>)</span>
<span id="cb33-347"><a href="#cb33-347" aria-hidden="true" tabindex="-1"></a>stats <span class="ot">&lt;-</span> <span class="fu">perCellQCMetrics</span>(sce<span class="fl">.416</span>b, <span class="at">subsets=</span><span class="fu">list</span>(<span class="at">Mt=</span>mito))</span>
<span id="cb33-348"><a href="#cb33-348" aria-hidden="true" tabindex="-1"></a>qc <span class="ot">&lt;-</span> <span class="fu">quickPerCellQC</span>(stats, <span class="at">percent_subsets=</span><span class="fu">c</span>(<span class="st">"subsets_Mt_percent"</span>,</span>
<span id="cb33-349"><a href="#cb33-349" aria-hidden="true" tabindex="-1"></a>    <span class="st">"altexps_ERCC_percent"</span>), <span class="at">batch=</span>sce<span class="fl">.416</span>b<span class="sc">$</span>block)</span>
<span id="cb33-350"><a href="#cb33-350" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b <span class="ot">&lt;-</span> sce<span class="fl">.416</span>b[,<span class="sc">!</span>qc<span class="sc">$</span>discard]</span>
<span id="cb33-351"><a href="#cb33-351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-352"><a href="#cb33-352" aria-hidden="true" tabindex="-1"></a><span class="co">#--- normalization ---#</span></span>
<span id="cb33-353"><a href="#cb33-353" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scran)</span>
<span id="cb33-354"><a href="#cb33-354" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b <span class="ot">&lt;-</span> <span class="fu">computeSumFactors</span>(sce<span class="fl">.416</span>b)</span>
<span id="cb33-355"><a href="#cb33-355" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b <span class="ot">&lt;-</span> <span class="fu">logNormCounts</span>(sce<span class="fl">.416</span>b)</span>
<span id="cb33-356"><a href="#cb33-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-357"><a href="#cb33-357" aria-hidden="true" tabindex="-1"></a><span class="co">#--- variance-modelling ---#</span></span>
<span id="cb33-358"><a href="#cb33-358" aria-hidden="true" tabindex="-1"></a>dec<span class="fl">.416</span>b <span class="ot">&lt;-</span> <span class="fu">modelGeneVarWithSpikes</span>(sce<span class="fl">.416</span>b, <span class="st">"ERCC"</span>, <span class="at">block=</span>sce<span class="fl">.416</span>b<span class="sc">$</span>block)</span>
<span id="cb33-359"><a href="#cb33-359" aria-hidden="true" tabindex="-1"></a>chosen.hvgs <span class="ot">&lt;-</span> <span class="fu">getTopHVGs</span>(dec<span class="fl">.416</span>b, <span class="at">prop=</span><span class="fl">0.1</span>)</span>
<span id="cb33-360"><a href="#cb33-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-361"><a href="#cb33-361" aria-hidden="true" tabindex="-1"></a><span class="co">#--- batch-correction ---#</span></span>
<span id="cb33-362"><a href="#cb33-362" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb33-363"><a href="#cb33-363" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(sce<span class="fl">.416</span>b, <span class="st">"corrected"</span>) <span class="ot">&lt;-</span> <span class="fu">removeBatchEffect</span>(<span class="fu">logcounts</span>(sce<span class="fl">.416</span>b), </span>
<span id="cb33-364"><a href="#cb33-364" aria-hidden="true" tabindex="-1"></a>    <span class="at">design=</span><span class="fu">model.matrix</span>(<span class="sc">~</span>sce<span class="fl">.416</span>b<span class="sc">$</span>phenotype), <span class="at">batch=</span>sce<span class="fl">.416</span>b<span class="sc">$</span>block)</span>
<span id="cb33-365"><a href="#cb33-365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-366"><a href="#cb33-366" aria-hidden="true" tabindex="-1"></a><span class="co">#--- dimensionality-reduction ---#</span></span>
<span id="cb33-367"><a href="#cb33-367" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(sce<span class="fl">.416</span>b, <span class="at">ncomponents=</span><span class="dv">10</span>, <span class="at">subset_row=</span>chosen.hvgs,</span>
<span id="cb33-368"><a href="#cb33-368" aria-hidden="true" tabindex="-1"></a>    <span class="at">exprs_values=</span><span class="st">"corrected"</span>, <span class="at">BSPARAM=</span>BiocSingular<span class="sc">::</span><span class="fu">ExactParam</span>())</span>
<span id="cb33-369"><a href="#cb33-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-370"><a href="#cb33-370" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1010</span>)</span>
<span id="cb33-371"><a href="#cb33-371" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b <span class="ot">&lt;-</span> <span class="fu">runTSNE</span>(sce<span class="fl">.416</span>b, <span class="at">dimred=</span><span class="st">"PCA"</span>, <span class="at">perplexity=</span><span class="dv">10</span>)</span>
<span id="cb33-372"><a href="#cb33-372" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-373"><a href="#cb33-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-376"><a href="#cb33-376" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-377"><a href="#cb33-377" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b</span>
<span id="cb33-378"><a href="#cb33-378" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-379"><a href="#cb33-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-382"><a href="#cb33-382" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-383"><a href="#cb33-383" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-hclust</span></span>
<span id="cb33-384"><a href="#cb33-384" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "根据致癌基因诱导状态（红色为诱导，蓝色为对照）和起始板（浅色或深色）进行着色"</span></span>
<span id="cb33-385"><a href="#cb33-385" aria-hidden="true" tabindex="-1"></a>hclust<span class="fl">.416</span>b <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce<span class="fl">.416</span>b, <span class="at">use.dimred=</span><span class="st">"PCA"</span>,</span>
<span id="cb33-386"><a href="#cb33-386" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">HclustParam</span>(<span class="at">method=</span><span class="st">"ward.D2"</span>), <span class="at">full=</span><span class="cn">TRUE</span>)</span>
<span id="cb33-387"><a href="#cb33-387" aria-hidden="true" tabindex="-1"></a>tree<span class="fl">.416</span>b <span class="ot">&lt;-</span> hclust<span class="fl">.416</span>b<span class="sc">$</span>objects<span class="sc">$</span>hclust</span>
<span id="cb33-388"><a href="#cb33-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-389"><a href="#cb33-389" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a prettier dendrogram.</span></span>
<span id="cb33-390"><a href="#cb33-390" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dendextend)</span>
<span id="cb33-391"><a href="#cb33-391" aria-hidden="true" tabindex="-1"></a>tree<span class="fl">.416</span>b<span class="sc">$</span>labels <span class="ot">&lt;-</span> <span class="fu">seq_along</span>(tree<span class="fl">.416</span>b<span class="sc">$</span>labels)</span>
<span id="cb33-392"><a href="#cb33-392" aria-hidden="true" tabindex="-1"></a>dend <span class="ot">&lt;-</span> <span class="fu">as.dendrogram</span>(tree<span class="fl">.416</span>b, <span class="at">hang=</span><span class="fl">0.1</span>)</span>
<span id="cb33-393"><a href="#cb33-393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-394"><a href="#cb33-394" aria-hidden="true" tabindex="-1"></a>combined.fac <span class="ot">&lt;-</span> <span class="fu">paste0</span>(sce<span class="fl">.416</span>b<span class="sc">$</span>block, <span class="st">"."</span>, </span>
<span id="cb33-395"><a href="#cb33-395" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, sce<span class="fl">.416</span>b<span class="sc">$</span>phenotype))</span>
<span id="cb33-396"><a href="#cb33-396" aria-hidden="true" tabindex="-1"></a><span class="fu">labels_colors</span>(dend) <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb33-397"><a href="#cb33-397" aria-hidden="true" tabindex="-1"></a>    <span class="st">"20160113.wild"</span><span class="ot">=</span><span class="st">"blue"</span>,</span>
<span id="cb33-398"><a href="#cb33-398" aria-hidden="true" tabindex="-1"></a>    <span class="st">"20160113.induced"</span><span class="ot">=</span><span class="st">"red"</span>,</span>
<span id="cb33-399"><a href="#cb33-399" aria-hidden="true" tabindex="-1"></a>    <span class="st">"20160325.wild"</span><span class="ot">=</span><span class="st">"dodgerblue"</span>,</span>
<span id="cb33-400"><a href="#cb33-400" aria-hidden="true" tabindex="-1"></a>    <span class="st">"20160325.induced"</span><span class="ot">=</span><span class="st">"salmon"</span></span>
<span id="cb33-401"><a href="#cb33-401" aria-hidden="true" tabindex="-1"></a>)[combined.fac][<span class="fu">order.dendrogram</span>(dend)]</span>
<span id="cb33-402"><a href="#cb33-402" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-403"><a href="#cb33-403" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(dend)</span>
<span id="cb33-404"><a href="#cb33-404" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-405"><a href="#cb33-405" aria-hidden="true" tabindex="-1"></a>为了获得显式聚类 @fig-hclust-dynamicTreeCut ，我们通过删除内部分支来“切割”树，使每个子树代表一个不同的聚类。</span>
<span id="cb33-406"><a href="#cb33-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-407"><a href="#cb33-407" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">dynamicTreeCut</span><span class="co">](https://cran.r-project.org/web/packages/dynamicTreeCut/index.html)</span></span>
<span id="cb33-408"><a href="#cb33-408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-411"><a href="#cb33-411" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-412"><a href="#cb33-412" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-hclust-dynamicTreeCut</span></span>
<span id="cb33-413"><a href="#cb33-413" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "根据dynamic Cut中分配的聚类标识进行着色"</span></span>
<span id="cb33-414"><a href="#cb33-414" aria-hidden="true" tabindex="-1"></a>hclust.dyn <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce<span class="fl">.416</span>b, <span class="at">use.dimred=</span><span class="st">"PCA"</span>,</span>
<span id="cb33-415"><a href="#cb33-415" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">HclustParam</span>(<span class="at">method=</span><span class="st">"ward.D2"</span>, <span class="at">cut.dynamic=</span><span class="cn">TRUE</span>,</span>
<span id="cb33-416"><a href="#cb33-416" aria-hidden="true" tabindex="-1"></a>        <span class="at">cut.params=</span><span class="fu">list</span>(<span class="at">minClusterSize=</span><span class="dv">10</span>, <span class="at">deepSplit=</span><span class="dv">1</span>)))</span>
<span id="cb33-417"><a href="#cb33-417" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(hclust.dyn)</span>
<span id="cb33-418"><a href="#cb33-418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-419"><a href="#cb33-419" aria-hidden="true" tabindex="-1"></a><span class="fu">labels_colors</span>(dend) <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(hclust.dyn)[<span class="fu">order.dendrogram</span>(dend)]</span>
<span id="cb33-420"><a href="#cb33-420" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(dend)</span>
<span id="cb33-421"><a href="#cb33-421" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-424"><a href="#cb33-424" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-425"><a href="#cb33-425" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-tsne-hclust-dynamicTreeCut</span></span>
<span id="cb33-426"><a href="#cb33-426" aria-hidden="true" tabindex="-1"></a><span class="fu">colLabels</span>(sce<span class="fl">.416</span>b) <span class="ot">&lt;-</span> <span class="fu">factor</span>(hclust.dyn)</span>
<span id="cb33-427"><a href="#cb33-427" aria-hidden="true" tabindex="-1"></a><span class="fu">plotReducedDim</span>(sce<span class="fl">.416</span>b, <span class="st">"TSNE"</span>, <span class="at">colour_by=</span><span class="st">"label"</span>)</span>
<span id="cb33-428"><a href="#cb33-428" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-429"><a href="#cb33-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-430"><a href="#cb33-430" aria-hidden="true" tabindex="-1"></a><span class="fu">### "two-step" mode ：Kmeans-Hierarchical</span></span>
<span id="cb33-431"><a href="#cb33-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-434"><a href="#cb33-434" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-435"><a href="#cb33-435" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-K-H_cluster</span></span>
<span id="cb33-436"><a href="#cb33-436" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "根据k-means/hierarchical聚类组合中分配的集群的身份进行着色 "</span></span>
<span id="cb33-437"><a href="#cb33-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-438"><a href="#cb33-438" aria-hidden="true" tabindex="-1"></a><span class="co"># Setting the seed due to the randomness of k-means.</span></span>
<span id="cb33-439"><a href="#cb33-439" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1111</span>)</span>
<span id="cb33-440"><a href="#cb33-440" aria-hidden="true" tabindex="-1"></a>khclust.info <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>,</span>
<span id="cb33-441"><a href="#cb33-441" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">TwoStepParam</span>(</span>
<span id="cb33-442"><a href="#cb33-442" aria-hidden="true" tabindex="-1"></a>        <span class="at">first=</span><span class="fu">KmeansParam</span>(<span class="at">centers=</span><span class="dv">1000</span>),</span>
<span id="cb33-443"><a href="#cb33-443" aria-hidden="true" tabindex="-1"></a>        <span class="at">second=</span><span class="fu">HclustParam</span>(<span class="at">method=</span><span class="st">"ward.D2"</span>, <span class="at">cut.dynamic=</span><span class="cn">TRUE</span>,</span>
<span id="cb33-444"><a href="#cb33-444" aria-hidden="true" tabindex="-1"></a>            <span class="at">cut.param=</span><span class="fu">list</span>(<span class="at">deepSplit=</span><span class="dv">3</span>)) <span class="co"># for higher resolution.</span></span>
<span id="cb33-445"><a href="#cb33-445" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb33-446"><a href="#cb33-446" aria-hidden="true" tabindex="-1"></a>    <span class="at">full=</span><span class="cn">TRUE</span></span>
<span id="cb33-447"><a href="#cb33-447" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb33-448"><a href="#cb33-448" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(khclust.info<span class="sc">$</span>clusters)</span>
<span id="cb33-449"><a href="#cb33-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-450"><a href="#cb33-450" aria-hidden="true" tabindex="-1"></a><span class="fu">plotTSNE</span>(sce.pbmc, <span class="at">colour_by=</span><span class="fu">I</span>(khclust.info<span class="sc">$</span>clusters), </span>
<span id="cb33-451"><a href="#cb33-451" aria-hidden="true" tabindex="-1"></a>    <span class="at">text_by=</span><span class="fu">I</span>(khclust.info<span class="sc">$</span>clusters))</span>
<span id="cb33-452"><a href="#cb33-452" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-453"><a href="#cb33-453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-454"><a href="#cb33-454" aria-hidden="true" tabindex="-1"></a>还可以检查在质心上构建的树状图 @fig-dendrogram-centroids , 这提供了不同亚群之间相对相似性的更定量的可视化。</span>
<span id="cb33-455"><a href="#cb33-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-458"><a href="#cb33-458" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-459"><a href="#cb33-459" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-dendrogram-centroids</span></span>
<span id="cb33-460"><a href="#cb33-460" aria-hidden="true" tabindex="-1"></a>k.stats <span class="ot">&lt;-</span> khclust.info<span class="sc">$</span>objects<span class="sc">$</span>first</span>
<span id="cb33-461"><a href="#cb33-461" aria-hidden="true" tabindex="-1"></a>tree.pbmc <span class="ot">&lt;-</span> khclust.info<span class="sc">$</span>objects<span class="sc">$</span>second<span class="sc">$</span>hclust</span>
<span id="cb33-462"><a href="#cb33-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-463"><a href="#cb33-463" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">match</span>(<span class="fu">as.integer</span>(tree.pbmc<span class="sc">$</span>labels), k.stats<span class="sc">$</span>cluster)</span>
<span id="cb33-464"><a href="#cb33-464" aria-hidden="true" tabindex="-1"></a>final.clusters <span class="ot">&lt;-</span> khclust.info<span class="sc">$</span>clusters[m]</span>
<span id="cb33-465"><a href="#cb33-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-466"><a href="#cb33-466" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: expose scater color palette for easier re-use,</span></span>
<span id="cb33-467"><a href="#cb33-467" aria-hidden="true" tabindex="-1"></a><span class="co"># given that the default colors start getting recycled.</span></span>
<span id="cb33-468"><a href="#cb33-468" aria-hidden="true" tabindex="-1"></a>dend <span class="ot">&lt;-</span> <span class="fu">as.dendrogram</span>(tree.pbmc, <span class="at">hang=</span><span class="fl">0.1</span>)</span>
<span id="cb33-469"><a href="#cb33-469" aria-hidden="true" tabindex="-1"></a><span class="fu">labels_colors</span>(dend) <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(final.clusters)[<span class="fu">order.dendrogram</span>(dend)]</span>
<span id="cb33-470"><a href="#cb33-470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-471"><a href="#cb33-471" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(dend)</span>
<span id="cb33-472"><a href="#cb33-472" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-473"><a href="#cb33-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-474"><a href="#cb33-474" aria-hidden="true" tabindex="-1"></a><span class="fu">### "two-step" mode ：Kmeans-Affinity</span></span>
<span id="cb33-475"><a href="#cb33-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-476"><a href="#cb33-476" aria-hidden="true" tabindex="-1"></a>相同的方法可用于加速任何基于距离矩阵的聚类方法。 例如， 使k-means质心通过亲和性传播（affinity propagation）执行聚类。在这个过程中，每个样本（如质心）选择自己或另一个样本作为其“示例exemplar”， 选择的适用性取决于样本之间的距离、每个样本的其他潜在样本以及具有相同所选样本的其他样本。 这些选择的迭代更新会产生一组聚类，其中每个聚类都是根据分配给同一示例的样本定义的。</span>
<span id="cb33-477"><a href="#cb33-477" aria-hidden="true" tabindex="-1"></a>与分层聚类不同，它不提供树状图dendrogram，但它也避免了切割树的额外复杂性—— 分辨率主要通过参数<span class="in">`q=`</span>控制，该参数定义了样本将自己视为示例并因此形成自己的簇的强度。</span>
<span id="cb33-478"><a href="#cb33-478" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-481"><a href="#cb33-481" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-482"><a href="#cb33-482" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-Kmeans-Affinity</span></span>
<span id="cb33-483"><a href="#cb33-483" aria-hidden="true" tabindex="-1"></a><span class="co"># Setting the seed due to the randomness of k-means.</span></span>
<span id="cb33-484"><a href="#cb33-484" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(apcluster)</span>
<span id="cb33-485"><a href="#cb33-485" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1111</span>)</span>
<span id="cb33-486"><a href="#cb33-486" aria-hidden="true" tabindex="-1"></a>kaclust.info <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>,</span>
<span id="cb33-487"><a href="#cb33-487" aria-hidden="true" tabindex="-1"></a>    <span class="at">BLUSPARAM=</span><span class="fu">TwoStepParam</span>(</span>
<span id="cb33-488"><a href="#cb33-488" aria-hidden="true" tabindex="-1"></a>        <span class="at">first=</span><span class="fu">KmeansParam</span>(<span class="at">centers=</span><span class="dv">1000</span>),</span>
<span id="cb33-489"><a href="#cb33-489" aria-hidden="true" tabindex="-1"></a>        <span class="at">second=</span><span class="fu">AffinityParam</span>(<span class="at">q=</span><span class="fl">0.1</span>) <span class="co"># larger q =&gt; more clusters</span></span>
<span id="cb33-490"><a href="#cb33-490" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb33-491"><a href="#cb33-491" aria-hidden="true" tabindex="-1"></a>    <span class="at">full=</span><span class="cn">TRUE</span></span>
<span id="cb33-492"><a href="#cb33-492" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb33-493"><a href="#cb33-493" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(kaclust.info<span class="sc">$</span>clusters)</span>
<span id="cb33-494"><a href="#cb33-494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-495"><a href="#cb33-495" aria-hidden="true" tabindex="-1"></a><span class="fu">plotTSNE</span>(sce.pbmc, <span class="at">colour_by=</span><span class="fu">I</span>(kaclust.info<span class="sc">$</span>clusters), </span>
<span id="cb33-496"><a href="#cb33-496" aria-hidden="true" tabindex="-1"></a>    <span class="at">text_by=</span><span class="fu">I</span>(kaclust.info<span class="sc">$</span>clusters))</span>
<span id="cb33-497"><a href="#cb33-497" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-498"><a href="#cb33-498" aria-hidden="true" tabindex="-1"></a><span class="fu">## Subclustering</span></span>
<span id="cb33-499"><a href="#cb33-499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-500"><a href="#cb33-500" aria-hidden="true" tabindex="-1"></a>提高分辨率的另一种简单方法是在单个聚类中重复特征选择和聚类，这旨在选择与内部结构更相关的 HVGs 和 PCs，通过避免不必要的功能产生的噪音来提高分辨率。 Subclustering 在缺乏不同亚群的情况下根据更适度的异质性分离细胞。</span>
<span id="cb33-501"><a href="#cb33-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-504"><a href="#cb33-504" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-505"><a href="#cb33-505" aria-hidden="true" tabindex="-1"></a>clust.full <span class="ot">&lt;-</span> <span class="fu">clusterCells</span>(sce.pbmc, <span class="at">use.dimred=</span><span class="st">"PCA"</span>)</span>
<span id="cb33-506"><a href="#cb33-506" aria-hidden="true" tabindex="-1"></a><span class="fu">plotExpression</span>(sce.pbmc, <span class="at">features=</span><span class="fu">c</span>(<span class="st">"CD3E"</span>, <span class="st">"CCR7"</span>, <span class="st">"CD69"</span>, <span class="st">"CD44"</span>),</span>
<span id="cb33-507"><a href="#cb33-507" aria-hidden="true" tabindex="-1"></a>    <span class="at">x=</span><span class="fu">I</span>(clust.full), <span class="at">colour_by=</span><span class="fu">I</span>(clust.full))</span>
<span id="cb33-508"><a href="#cb33-508" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-511"><a href="#cb33-511" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-512"><a href="#cb33-512" aria-hidden="true" tabindex="-1"></a><span class="co"># Repeating modelling and PCA on the subset.</span></span>
<span id="cb33-513"><a href="#cb33-513" aria-hidden="true" tabindex="-1"></a>memory <span class="ot">&lt;-</span> <span class="dv">10</span>L</span>
<span id="cb33-514"><a href="#cb33-514" aria-hidden="true" tabindex="-1"></a>sce.memory <span class="ot">&lt;-</span> sce.pbmc[,clust.full<span class="sc">==</span>memory]</span>
<span id="cb33-515"><a href="#cb33-515" aria-hidden="true" tabindex="-1"></a>dec.memory <span class="ot">&lt;-</span> <span class="fu">modelGeneVar</span>(sce.memory)</span>
<span id="cb33-516"><a href="#cb33-516" aria-hidden="true" tabindex="-1"></a>sce.memory <span class="ot">&lt;-</span> <span class="fu">denoisePCA</span>(sce.memory, <span class="at">technical=</span>dec.memory,</span>
<span id="cb33-517"><a href="#cb33-517" aria-hidden="true" tabindex="-1"></a>    <span class="at">subset.row=</span><span class="fu">getTopHVGs</span>(dec.memory, <span class="at">n=</span><span class="dv">5000</span>))</span>
<span id="cb33-518"><a href="#cb33-518" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-519"><a href="#cb33-519" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-522"><a href="#cb33-522" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-523"><a href="#cb33-523" aria-hidden="true" tabindex="-1"></a>g.memory <span class="ot">&lt;-</span> <span class="fu">buildSNNGraph</span>(sce.memory, <span class="at">use.dimred=</span><span class="st">"PCA"</span>)</span>
<span id="cb33-524"><a href="#cb33-524" aria-hidden="true" tabindex="-1"></a>clust.memory <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">cluster_walktrap</span>(g.memory)<span class="sc">$</span>membership</span>
<span id="cb33-525"><a href="#cb33-525" aria-hidden="true" tabindex="-1"></a><span class="fu">plotExpression</span>(sce.memory, <span class="at">features=</span><span class="fu">c</span>(<span class="st">"CD8A"</span>, <span class="st">"CD4"</span>),</span>
<span id="cb33-526"><a href="#cb33-526" aria-hidden="true" tabindex="-1"></a>    <span class="at">x=</span><span class="fu">I</span>(<span class="fu">factor</span>(clust.memory)))</span>
<span id="cb33-527"><a href="#cb33-527" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-528"><a href="#cb33-528" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-529"><a href="#cb33-529" aria-hidden="true" tabindex="-1"></a>对于子聚类分析，定义一个自定义函数<span class="in">`quickSubCluster()`</span></span>
<span id="cb33-530"><a href="#cb33-530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-533"><a href="#cb33-533" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb33-534"><a href="#cb33-534" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1000010</span>)</span>
<span id="cb33-535"><a href="#cb33-535" aria-hidden="true" tabindex="-1"></a>subcluster.out <span class="ot">&lt;-</span> <span class="fu">quickSubCluster</span>(sce.pbmc, <span class="at">groups=</span>clust.full,</span>
<span id="cb33-536"><a href="#cb33-536" aria-hidden="true" tabindex="-1"></a>    <span class="at">prepFUN=</span><span class="cf">function</span>(x) { <span class="co"># Preparing the subsetted SCE for clustering.</span></span>
<span id="cb33-537"><a href="#cb33-537" aria-hidden="true" tabindex="-1"></a>        dec <span class="ot">&lt;-</span> <span class="fu">modelGeneVar</span>(x)</span>
<span id="cb33-538"><a href="#cb33-538" aria-hidden="true" tabindex="-1"></a>        input <span class="ot">&lt;-</span> <span class="fu">denoisePCA</span>(x, <span class="at">technical=</span>dec,</span>
<span id="cb33-539"><a href="#cb33-539" aria-hidden="true" tabindex="-1"></a>            <span class="at">subset.row=</span><span class="fu">getTopHVGs</span>(dec, <span class="at">prop=</span><span class="fl">0.1</span>),</span>
<span id="cb33-540"><a href="#cb33-540" aria-hidden="true" tabindex="-1"></a>            <span class="at">BSPARAM=</span>BiocSingular<span class="sc">::</span><span class="fu">IrlbaParam</span>())</span>
<span id="cb33-541"><a href="#cb33-541" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb33-542"><a href="#cb33-542" aria-hidden="true" tabindex="-1"></a>    <span class="at">clusterFUN=</span><span class="cf">function</span>(x) { <span class="co"># Performing the subclustering in the subset.</span></span>
<span id="cb33-543"><a href="#cb33-543" aria-hidden="true" tabindex="-1"></a>        g <span class="ot">&lt;-</span> <span class="fu">buildSNNGraph</span>(x, <span class="at">use.dimred=</span><span class="st">"PCA"</span>, <span class="at">k=</span><span class="dv">20</span>)</span>
<span id="cb33-544"><a href="#cb33-544" aria-hidden="true" tabindex="-1"></a>        igraph<span class="sc">::</span><span class="fu">cluster_walktrap</span>(g)<span class="sc">$</span>membership</span>
<span id="cb33-545"><a href="#cb33-545" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb33-546"><a href="#cb33-546" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb33-547"><a href="#cb33-547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-548"><a href="#cb33-548" aria-hidden="true" tabindex="-1"></a><span class="co"># One SingleCellExperiment object per parent cluster:</span></span>
<span id="cb33-549"><a href="#cb33-549" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(subcluster.out)</span>
<span id="cb33-550"><a href="#cb33-550" aria-hidden="true" tabindex="-1"></a><span class="co"># Looking at the subclustering for one example:</span></span>
<span id="cb33-551"><a href="#cb33-551" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(subcluster.out[[<span class="dv">1</span>]]<span class="sc">$</span>subcluster)</span>
<span id="cb33-552"><a href="#cb33-552" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/Wanganlin00/Bioinformatics/edit/main/Clustering.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/Wanganlin00/Bioinformatics/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div></div></footer></body></html>