<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>转录组下游分析</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="TranscriptomicsDownstreamAnalysis_files/libs/clipboard/clipboard.min.js"></script>
<script src="TranscriptomicsDownstreamAnalysis_files/libs/quarto-html/quarto.js"></script>
<script src="TranscriptomicsDownstreamAnalysis_files/libs/quarto-html/popper.min.js"></script>
<script src="TranscriptomicsDownstreamAnalysis_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="TranscriptomicsDownstreamAnalysis_files/libs/quarto-html/anchor.min.js"></script>
<link href="TranscriptomicsDownstreamAnalysis_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="TranscriptomicsDownstreamAnalysis_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="TranscriptomicsDownstreamAnalysis_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="TranscriptomicsDownstreamAnalysis_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="TranscriptomicsDownstreamAnalysis_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#assays" id="toc-assays" class="nav-link active" data-scroll-target="#assays"><span class="header-section-number">0.1</span> Assays</a>
  <ul class="collapse">
  <li><a href="#logcount-mrna-count" id="toc-logcount-mrna-count" class="nav-link" data-scroll-target="#logcount-mrna-count"><span class="header-section-number">0.1.1</span> logCount→ mRNA Count</a></li>
  </ul></li>
  <li><a href="#sample-metadata" id="toc-sample-metadata" class="nav-link" data-scroll-target="#sample-metadata"><span class="header-section-number">0.2</span> Sample metadata</a></li>
  <li><a href="#建立deg对象" id="toc-建立deg对象" class="nav-link" data-scroll-target="#建立deg对象"><span class="header-section-number">0.3</span> 建立DEG对象</a>
  <ul class="collapse">
  <li><a href="#预过滤" id="toc-预过滤" class="nav-link" data-scroll-target="#预过滤"><span class="header-section-number">0.3.1</span> 预过滤</a></li>
  </ul></li>
  <li><a href="#差异表达基因分析" id="toc-差异表达基因分析" class="nav-link" data-scroll-target="#差异表达基因分析"><span class="header-section-number">1</span> 差异表达基因分析</a>
  <ul class="collapse">
  <li><a href="#结果分析" id="toc-结果分析" class="nav-link" data-scroll-target="#结果分析"><span class="header-section-number">1.0.1</span> 结果分析</a></li>
  <li><a href="#热图" id="toc-热图" class="nav-link" data-scroll-target="#热图"><span class="header-section-number">1.0.2</span> 热图</a></li>
  <li><a href="#挑选指定第一个基因看它在同一个病人的配对表达情况" id="toc-挑选指定第一个基因看它在同一个病人的配对表达情况" class="nav-link" data-scroll-target="#挑选指定第一个基因看它在同一个病人的配对表达情况"><span class="header-section-number">1.0.3</span> 挑选指定第一个基因看它在同一个病人的配对表达情况</a></li>
  <li><a href="#火山图" id="toc-火山图" class="nav-link" data-scroll-target="#火山图"><span class="header-section-number">1.0.4</span> 火山图</a></li>
  <li><a href="#lfc收缩和pagj" id="toc-lfc收缩和pagj" class="nav-link" data-scroll-target="#lfc收缩和pagj"><span class="header-section-number">1.0.5</span> LFC收缩和pagj</a></li>
  <li><a href="#ma图-mean-absdeviation-plot" id="toc-ma图-mean-absdeviation-plot" class="nav-link" data-scroll-target="#ma图-mean-absdeviation-plot"><span class="header-section-number">1.0.6</span> MA图 Mean-AbsDeviation plot</a></li>
  <li><a href="#readcounts" id="toc-readcounts" class="nav-link" data-scroll-target="#readcounts"><span class="header-section-number">1.0.7</span> readCounts</a></li>
  <li><a href="#异常观测" id="toc-异常观测" class="nav-link" data-scroll-target="#异常观测"><span class="header-section-number">1.0.8</span> 异常观测</a></li>
  <li><a href="#多因素设计" id="toc-多因素设计" class="nav-link" data-scroll-target="#多因素设计"><span class="header-section-number">1.0.9</span> 多因素设计</a></li>
  <li><a href="#导出到-csv-文件" id="toc-导出到-csv-文件" class="nav-link" data-scroll-target="#导出到-csv-文件"><span class="header-section-number">1.0.10</span> 导出到 CSV 文件</a></li>
  </ul></li>
  <li><a href="#富集分析" id="toc-富集分析" class="nav-link" data-scroll-target="#富集分析"><span class="header-section-number">2</span> 富集分析</a>
  <ul class="collapse">
  <li><a href="#数据整理" id="toc-数据整理" class="nav-link" data-scroll-target="#数据整理"><span class="header-section-number">2.0.1</span> 数据整理</a></li>
  <li><a href="#kegg-通路富集" id="toc-kegg-通路富集" class="nav-link" data-scroll-target="#kegg-通路富集"><span class="header-section-number">2.1</span> KEGG 通路富集</a>
  <ul class="collapse">
  <li><a href="#水平条图" id="toc-水平条图" class="nav-link" data-scroll-target="#水平条图"><span class="header-section-number">2.1.1</span> 水平条图</a></li>
  <li><a href="#点图" id="toc-点图" class="nav-link" data-scroll-target="#点图"><span class="header-section-number">2.1.2</span> 点图</a></li>
  </ul></li>
  <li><a href="#go富集分析" id="toc-go富集分析" class="nav-link" data-scroll-target="#go富集分析"><span class="header-section-number">2.2</span> GO富集分析</a>
  <ul class="collapse">
  <li><a href="#可视化" id="toc-可视化" class="nav-link" data-scroll-target="#可视化"><span class="header-section-number">2.2.1</span> 可视化</a></li>
  </ul></li>
  <li><a href="#gsea" id="toc-gsea" class="nav-link" data-scroll-target="#gsea"><span class="header-section-number">2.3</span> GSEA</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">转录组下游分析</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="assays" class="level2" data-number="0.1">
<h2 data-number="0.1" class="anchored" data-anchor-id="assays"><span class="header-section-number">0.1</span> Assays</h2>
<section id="logcount-mrna-count" class="level3" data-number="0.1.1">
<h3 data-number="0.1.1" class="anchored" data-anchor-id="logcount-mrna-count"><span class="header-section-number">0.1.1</span> logCount→ mRNA Count</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-1_673a624f1f571be8fa977c7c2b724c5e">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 读取计数矩阵 log2(count+1)</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>raw_logCount <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/TCGA-CHOL.htseq_counts.tsv"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Rows: 60488 Columns: 46</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Delimiter: "\t"</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; chr  (1): Ensembl_ID</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; dbl (45): TCGA-ZD-A8I3-01A, TCGA-W5-AA2U-11A, TCGA-W5-AA30-01A, TCGA-W5-AA38...</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co"># 60488 个gene ×  45 个sample</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-2_756114177ba794b33d1fe46765a22008">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 基因编码  BiocManager::install("rtracklayer")</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>gtf_v22 <span class="ot">&lt;-</span> rtracklayer<span class="sc">::</span><span class="fu">import</span>(<span class="st">'data/gencode.v22.annotation.gtf.gz'</span>) <span class="sc">|&gt;</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a> <span class="fu">as_tibble</span>()</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(gtf_v22,<span class="at">file =</span> <span class="st">"data/gtf_v22.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-3_a8995a199efad7ca20cdf0c08022edba">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 提取mRNA相关基因</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>gtf_mrna_v22 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gtf_v22,</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">c</span>(<span class="st">"gene_id"</span>,<span class="st">"gene_type"</span>, <span class="st">"gene_name"</span>))<span class="sc">|&gt;</span> </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(gene_type <span class="sc">==</span> <span class="st">"protein_coding"</span>) <span class="sc">|&gt;</span>  </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>()</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>gtf_mrna_v22 </span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 19,814 × 3</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    gene_id            gene_type      gene_name </span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;              &lt;chr&gt;          &lt;chr&gt;     </span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 ENSG00000186092.4  protein_coding OR4F5     </span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 ENSG00000279928.1  protein_coding FO538757.3</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 ENSG00000279457.2  protein_coding FO538757.2</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 ENSG00000278566.1  protein_coding OR4F29    </span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 ENSG00000273547.1  protein_coding OR4F16    </span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 ENSG00000187634.9  protein_coding SAMD11    </span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 ENSG00000188976.9  protein_coding NOC2L     </span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 ENSG00000187961.12 protein_coding KLHL17    </span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 ENSG00000187583.9  protein_coding PLEKHN1   </span></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 ENSG00000187642.8  protein_coding PERM1     </span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ 19,804 more rows</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(gtf_mrna_v22,<span class="at">file =</span> <span class="st">"data/gtf_mrna_v22.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-4_de7d6334e49fa258aee314d591fb9c38">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 根据注释文件把Ensembl ID转换为gene ID</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>gene_logCount <span class="ot">&lt;-</span> raw_logCount <span class="sc">|&gt;</span> </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(gtf_mrna_v22, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Ensembl_ID"</span> <span class="ot">=</span> <span class="st">"gene_id"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name, <span class="fu">starts_with</span>(<span class="st">"TCGA"</span>) )</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>gene_logCount</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 19,814 × 46</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    gene_name `TCGA-ZD-A8I3-01A` `TCGA-W5-AA2U-11A` `TCGA-W5-AA30-01A`</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                  &lt;dbl&gt;              &lt;dbl&gt;              &lt;dbl&gt;</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 TSPAN6                 12.4               11.3               12.3 </span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 TNMD                    1                  1                  0   </span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 DPM1                   10.2                9.36              10.7 </span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 SCYL3                   9.56               8.44              11.4 </span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 C1orf112                8.54               5.13              10.0 </span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 FGR                     7.17               8.29               8.57</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 CFH                    14.1               15.1               11.7 </span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 FUCA2                  11.7               10.6               11.6 </span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 GCLC                   11.1               12.2               11.7 </span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 NFYA                   10.5                9.04              10.9 </span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ 19,804 more rows</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ 42 more variables: `TCGA-W5-AA38-01A` &lt;dbl&gt;, `TCGA-W5-AA30-11A` &lt;dbl&gt;,</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   `TCGA-YR-A95A-01A` &lt;dbl&gt;, `TCGA-W5-AA36-01A` &lt;dbl&gt;,</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   `TCGA-W5-AA2U-01A` &lt;dbl&gt;, `TCGA-ZH-A8Y2-01A` &lt;dbl&gt;,</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   `TCGA-3X-AAVB-01A` &lt;dbl&gt;, `TCGA-W5-AA2Z-01A` &lt;dbl&gt;,</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   `TCGA-W5-AA2X-01A` &lt;dbl&gt;, `TCGA-W5-AA34-11A` &lt;dbl&gt;,</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   `TCGA-ZH-A8Y6-01A` &lt;dbl&gt;, `TCGA-ZH-A8Y1-01A` &lt;dbl&gt;, …</span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(gene_logCount,<span class="st">"data/gene_logCount.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>基因名是有重复的，其中有100个出现了2次，有1个出现了3次</p>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-5_7f0ebc8fe48ceb7cad0c4715ed75b6a1">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>gene_logCount <span class="sc">|&gt;</span> <span class="fu">summarise</span>(<span class="at">n=</span><span class="fu">n</span>(),<span class="at">.by =</span> gene_name) <span class="sc">|&gt;</span> </span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(n<span class="sc">==</span><span class="dv">1</span>) </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 19,611 × 2</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    gene_name     n</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;     &lt;int&gt;</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 TSPAN6        1</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 TNMD          1</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 DPM1          1</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 SCYL3         1</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 C1orf112      1</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 FGR           1</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 CFH           1</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 FUCA2         1</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 GCLC          1</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 NFYA          1</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ 19,601 more rows</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>gene_logCount <span class="sc">|&gt;</span> <span class="fu">summarise</span>(<span class="at">n=</span><span class="fu">n</span>(),<span class="at">.by =</span> gene_name) <span class="sc">|&gt;</span> </span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(n<span class="sc">==</span><span class="dv">2</span>) </span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 100 × 2</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    gene_name     n</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;     &lt;int&gt;</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 CD99          2</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 IDS           2</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 MATR3         2</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 PI4K2B        2</span></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 CFHR2         2</span></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 IL12RB2       2</span></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 TAF9          2</span></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 KIAA0391      2</span></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 SIGLEC5       2</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 PIK3R2        2</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ 90 more rows</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>gene_logCount <span class="sc">|&gt;</span> <span class="fu">summarise</span>(<span class="at">n=</span><span class="fu">n</span>(),<span class="at">.by =</span> gene_name) <span class="sc">|&gt;</span> </span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(n<span class="sc">==</span><span class="dv">3</span>) </span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 1 × 2</span></span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   gene_name     n</span></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;     &lt;int&gt;</span></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 C12orf79      3</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>先去对数（ log2(Count+1））</p>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-6_3c54ce328b74ab66edada1385864a16f">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>gene_Count <span class="ot">&lt;-</span> gene_logCount <span class="sc">|&gt;</span> </span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a> <span class="fu">mutate_if</span>(is.numeric,<span class="sc">~</span><span class="dv">2</span><span class="sc">^</span>(.)<span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>gene_Count</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 19,814 × 46</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    gene_name `TCGA-ZD-A8I3-01A` `TCGA-W5-AA2U-11A` `TCGA-W5-AA30-01A`</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                  &lt;dbl&gt;              &lt;dbl&gt;              &lt;dbl&gt;</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 TSPAN6                 5254.               2476              5132 </span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 TNMD                      1                   1                 0 </span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 DPM1                   1212.                655              1644.</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 SCYL3                   753                 346              2652 </span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 C1orf112                370                  34              1044.</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 FGR                     143                 311               378 </span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 CFH                   17588.              35982              3352 </span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 FUCA2                  3235.               1589              2998 </span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 GCLC                   2180                4711              3248.</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 NFYA                   1411                 525              1916 </span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ 19,804 more rows</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ 42 more variables: `TCGA-W5-AA38-01A` &lt;dbl&gt;, `TCGA-W5-AA30-11A` &lt;dbl&gt;,</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   `TCGA-YR-A95A-01A` &lt;dbl&gt;, `TCGA-W5-AA36-01A` &lt;dbl&gt;,</span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   `TCGA-W5-AA2U-01A` &lt;dbl&gt;, `TCGA-ZH-A8Y2-01A` &lt;dbl&gt;,</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   `TCGA-3X-AAVB-01A` &lt;dbl&gt;, `TCGA-W5-AA2Z-01A` &lt;dbl&gt;,</span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   `TCGA-W5-AA2X-01A` &lt;dbl&gt;, `TCGA-W5-AA34-11A` &lt;dbl&gt;,</span></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   `TCGA-ZH-A8Y6-01A` &lt;dbl&gt;, `TCGA-ZH-A8Y1-01A` &lt;dbl&gt;, …</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>根据基因id分组，重复，则取平均值去重，去掉重复的102行</p>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-7_6d699d48c1c9248cf435b28c8951e1c5">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>Counts <span class="ot">&lt;-</span> gene_Count <span class="sc">|&gt;</span> </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a> <span class="fu">group_by</span>(gene_name) <span class="sc">|&gt;</span> </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a> <span class="fu">summarise_all</span>(mean,<span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>Counts <span class="sc">|&gt;</span> <span class="fu">summarise</span>(<span class="at">n=</span><span class="fu">n</span>(),<span class="at">.by =</span> gene_name) <span class="sc">|&gt;</span> </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(n<span class="sc">==</span><span class="dv">1</span>) </span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 19,712 × 2</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    gene_name     n</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;     &lt;int&gt;</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 A1BG          1</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 A1CF          1</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 A2M           1</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 A2ML1         1</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 A3GALT2       1</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 A4GALT        1</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 A4GNT         1</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 AAAS          1</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 AACS          1</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 AADAC         1</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ 19,702 more rows</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>提取lncRNA、miRNA等也是类似的操作</p>
<div id="summarise_all" class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-8_4693683f242609e22b8deebedabdb772">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tibble</span>(</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">x=</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>),</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">y=</span><span class="fu">c</span>(<span class="dv">3</span><span class="sc">:</span><span class="dv">7</span>),</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">g=</span><span class="fu">factor</span>(<span class="fu">c</span>(<span class="st">'A'</span>,<span class="st">'B'</span>,<span class="st">'A'</span>,<span class="st">'C'</span>,<span class="st">'C'</span>))</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>)<span class="sc">|&gt;</span> </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(g) <span class="sc">|&gt;</span> <span class="fu">summarise_all</span>(<span class="fu">list</span>(<span class="st">"均数"</span><span class="ot">=</span>mean, <span class="st">"中位数"</span><span class="ot">=</span>median))</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 3 × 5</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   g     x_均数 y_均数 x_中位数 y_中位数</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;fct&gt;  &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 A        2      4        2        4  </span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 B        2      4        2        4  </span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 C        4.5    6.5      4.5      6.5</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
</div>
</section>
</section>
<section id="sample-metadata" class="level2" data-number="0.2">
<h2 data-number="0.2" class="anchored" data-anchor-id="sample-metadata"><span class="header-section-number">0.2</span> Sample metadata</h2>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-9_7b82268ad7a12c01854b25aee22c8697">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co">#首先载入下载好胆管癌的样本信息文件</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>phenotype_file <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">'data/TCGA-CHOL.GDC_phenotype.tsv'</span>)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Rows: 71 Columns: 122</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Delimiter: "\t"</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; chr (69): submitter_id.samples, batch_number, bcr, bcr_followup_barcode, bcr...</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; dbl (47): age_at_initial_pathologic_diagnosis, albumin_result_lower_limit, a...</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; lgl  (6): withdrawn, releasable.project, days_to_sample_procurement.samples,...</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ Use `spec()` to retrieve the full column specification for this data.</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>phenotype_file[<span class="dv">1</span>,]</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 1 × 122</span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   submitter_id.samples age_at_initial_pathologic_diagno…¹ albumin_result_lower…²</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                                             &lt;dbl&gt;                  &lt;dbl&gt;</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 TCGA-ZH-A8Y2-01A                                     59                     NA</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ abbreviated names: ¹​age_at_initial_pathologic_diagnosis,</span></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   ²​albumin_result_lower_limit</span></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ 119 more variables: albumin_result_specified_value &lt;dbl&gt;,</span></span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   albumin_result_upper_limit &lt;dbl&gt;, batch_number &lt;chr&gt;, bcr &lt;chr&gt;,</span></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   bcr_followup_barcode &lt;chr&gt;, bcr_followup_uuid &lt;chr&gt;, submitter_id &lt;chr&gt;,</span></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   bilirubin_lower_limit &lt;dbl&gt;, bilirubin_upper_limit &lt;dbl&gt;,</span></span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   ca_19_9_level &lt;dbl&gt;, ca_19_9_level_lower &lt;dbl&gt;, …</span></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>phenotypeID <span class="ot">&lt;-</span> phenotype_file<span class="sc">$</span>submitter_id.samples</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a><span class="co"># 71 个sample × 122 个样本变量</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-10_98965880f0887a4cfedacdbb382039b8">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 选择样本</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(phenotype_file<span class="sc">$</span>neoplasm_histologic_grade)<span class="co"># 组织学分级</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; G1 G2 G3 G4 </span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 31 33  3</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(phenotype_file<span class="sc">$</span>tumor_stage.diagnoses) <span class="co">#分期</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   stage i  stage ii stage iib stage iii  stage iv stage iva stage ivb </span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        30        16         6         5         2         3         6</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-11_ca954966bd14cf90dea79cc7594e9f00">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="do">## 找到所需列变量的索引，最好用正则表达式</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>col_index <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"(neoplasm_histologic_grade)|(tumor_stage.diagnoses)"</span>,<span class="fu">colnames</span>(phenotype_file))</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>col_index</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 39 98</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 或者</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>col_index_logical <span class="ot">&lt;-</span> <span class="fu">str_detect</span>(<span class="fu">colnames</span>(phenotype_file),<span class="st">"(neoplasm_histologic_grade)|(tumor_stage.diagnoses)"</span>)</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>col_index_logical</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [13] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [25] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [37] FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [49] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [61] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [73] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [85] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [97] FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [109] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [121] FALSE FALSE</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>col_index <span class="ot">&lt;-</span> <span class="fu">which</span>(col_index_logical<span class="sc">==</span><span class="cn">TRUE</span>)</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>col_index</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 39 98</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-12_5e82e5234ba00b734a90fd4c24b75783">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 取出需要的列的样本信息</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>hs <span class="ot">&lt;-</span> phenotype_file[,col_index[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]]</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>hs <span class="ot">&lt;-</span> <span class="fu">cbind</span>(phenotypeID,hs)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hs)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        phenotypeID neoplasm_histologic_grade tumor_stage.diagnoses</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 TCGA-ZH-A8Y2-01A                        G2               stage i</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 TCGA-ZH-A8Y7-01A                        G3             stage ivb</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 TCGA-W7-A93O-01A                        G3             stage iib</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 TCGA-W7-A93O-11A                        G3             stage iib</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 TCGA-W5-AA30-01A                        G3               stage i</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 TCGA-W5-AA30-11A                        G3               stage i</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-13_648c570151a452c73d5848d976d53f31">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 挑选组织学为G2或G3和分期为i或ii期的样本，数量多</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>sample <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(hs,neoplasm_histologic_grade<span class="sc">==</span><span class="st">"G2"</span><span class="sc">|</span>neoplasm_histologic_grade<span class="sc">==</span><span class="st">"G3"</span> ) <span class="sc">|&gt;</span> </span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a> dplyr<span class="sc">::</span><span class="fu">filter</span>(tumor_stage.diagnoses<span class="sc">==</span><span class="st">"stage i"</span><span class="sc">|</span>tumor_stage.diagnoses<span class="sc">==</span><span class="st">"stage ii"</span>)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sample)</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        phenotypeID neoplasm_histologic_grade tumor_stage.diagnoses</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 TCGA-ZH-A8Y2-01A                        G2               stage i</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 TCGA-W5-AA30-01A                        G3               stage i</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 TCGA-W5-AA30-11A                        G3               stage i</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 TCGA-W5-AA2K-01A                        G3               stage i</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 TCGA-W5-AA2R-01A                        G3               stage i</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 TCGA-W5-AA2R-11A                        G3               stage i</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>到这里，就从71个样本中挑出了45个样本，但是，我们只要这45个样本中成对样本的表达数据，即同一个病人既有癌旁<del>正常</del>细胞的表达蛋白，又有肿瘤细胞的异常表达蛋白。</p>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-14_8715907310db664cc74300e2d705f76b">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 前缀1~12字符</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>sample <span class="ot">&lt;-</span> sample <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">patient_id =</span> <span class="fu">str_sub</span>(sample[,<span class="dv">1</span>],<span class="dv">1</span>,<span class="dv">12</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a> <span class="fu">group_by</span>(patient_id) <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">|&gt;</span> <span class="fu">ungroup</span>()</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>sample <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(sample,n<span class="sc">==</span><span class="dv">2</span>) <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>n)</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>sample <span class="co"># 筛选出 15对 成对数据</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 30 × 4</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    phenotypeID      neoplasm_histologic_grade tumor_stage.diagnoses patient_id  </span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;            &lt;chr&gt;                     &lt;chr&gt;                 &lt;chr&gt;       </span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 TCGA-W5-AA30-01A G3                        stage i               TCGA-W5-AA30</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 TCGA-W5-AA30-11A G3                        stage i               TCGA-W5-AA30</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 TCGA-W5-AA2R-01A G3                        stage i               TCGA-W5-AA2R</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 TCGA-W5-AA2R-11A G3                        stage i               TCGA-W5-AA2R</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 TCGA-4G-AAZR-01A G2                        stage ii              TCGA-4G-AAZR</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 TCGA-4G-AAZR-11A G2                        stage ii              TCGA-4G-AAZR</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 TCGA-W5-AA2Z-01A G3                        stage ii              TCGA-W5-AA2Z</span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 TCGA-W5-AA2Z-11A G3                        stage ii              TCGA-W5-AA2Z</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 TCGA-W5-AA31-01A G3                        stage i               TCGA-W5-AA31</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 TCGA-W5-AA31-11A G3                        stage i               TCGA-W5-AA31</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # ℹ 20 more rows</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>并且要匹配到<code>Counts</code> 中出现的45个样本中的成对样本</p>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-15_d2fc2b461a9d998f349edc155fddb370">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 表达矩阵  </span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>patient_of_counts <span class="ot">&lt;-</span> <span class="fu">str_sub</span>(<span class="fu">colnames</span>(Counts),<span class="dv">1</span>,<span class="dv">12</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(patient_of_counts)<span class="sc">==</span><span class="dv">2</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; patient_of_counts</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    gene_name TCGA-3X-AAV9 TCGA-3X-AAVA TCGA-3X-AAVB TCGA-3X-AAVC TCGA-3X-AAVE </span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        FALSE        FALSE        FALSE        FALSE        FALSE        FALSE </span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-4G-AAZO TCGA-4G-AAZT TCGA-W5-AA2G TCGA-W5-AA2H TCGA-W5-AA2I TCGA-W5-AA2O </span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        FALSE        FALSE        FALSE        FALSE         TRUE        FALSE </span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2Q TCGA-W5-AA2R TCGA-W5-AA2T TCGA-W5-AA2U TCGA-W5-AA2W TCGA-W5-AA2X </span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         TRUE         TRUE        FALSE         TRUE        FALSE         TRUE </span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2Z TCGA-W5-AA30 TCGA-W5-AA31 TCGA-W5-AA33 TCGA-W5-AA34 TCGA-W5-AA36 </span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        FALSE         TRUE         TRUE        FALSE         TRUE        FALSE </span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA38 TCGA-W5-AA39 TCGA-W6-AA0S TCGA-WD-A7RX TCGA-YR-A95A TCGA-ZD-A8I3 </span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        FALSE        FALSE        FALSE        FALSE        FALSE        FALSE </span></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-ZH-A8Y1 TCGA-ZH-A8Y2 TCGA-ZH-A8Y4 TCGA-ZH-A8Y5 TCGA-ZH-A8Y6 TCGA-ZH-A8Y8 </span></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        FALSE        FALSE        FALSE        FALSE        FALSE        FALSE </span></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-ZU-A8S4 </span></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         TRUE</span></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>paired_patient <span class="ot">&lt;-</span> <span class="fu">table</span>(patient_of_counts)[<span class="fu">table</span>(patient_of_counts)<span class="sc">==</span><span class="dv">2</span>]<span class="sc">|&gt;</span> </span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a> <span class="fu">names</span>()  <span class="co"># 筛选出9对成对数据</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>取<code>sample</code>和<code>paired_patient</code>的交集，8对成对数据</p>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-16_46a589c30dcb3aef3c2b9ebed0e2de78">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 取交集</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>final_patient <span class="ot">&lt;-</span> lubridate<span class="sc">::</span><span class="fu">intersect</span>(sample<span class="sc">$</span>patient_id,paired_patient)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>final_patient</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] "TCGA-W5-AA30" "TCGA-W5-AA2R" "TCGA-W5-AA31" "TCGA-W5-AA2U" "TCGA-W5-AA2Q"</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [6] "TCGA-W5-AA2I" "TCGA-W5-AA34" "TCGA-ZU-A8S4"</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>sample <span class="ot">&lt;-</span> sample <span class="sc">|&gt;</span> </span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample<span class="sc">$</span>patient_id <span class="sc">%in%</span> final_patient) <span class="sc">|&gt;</span> </span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>patient_id)<span class="sc">|&gt;</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(phenotypeID)<span class="sc">|&gt;</span> </span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"grade"</span><span class="ot">=</span><span class="st">"neoplasm_histologic_grade"</span>,</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"stage"</span><span class="ot">=</span><span class="st">"tumor_stage.diagnoses"</span>)</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>sample</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 16 × 3</span></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    phenotypeID      grade stage   </span></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;            &lt;chr&gt; &lt;chr&gt;   </span></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 TCGA-W5-AA2I-01A G2    stage i </span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 TCGA-W5-AA2I-11A G2    stage i </span></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 TCGA-W5-AA2Q-01A G2    stage ii</span></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 TCGA-W5-AA2Q-11A G2    stage ii</span></span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 TCGA-W5-AA2R-01A G3    stage i </span></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 TCGA-W5-AA2R-11A G3    stage i </span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 TCGA-W5-AA2U-01A G3    stage i </span></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 TCGA-W5-AA2U-11A G3    stage i </span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 TCGA-W5-AA30-01A G3    stage i </span></span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 TCGA-W5-AA30-11A G3    stage i </span></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11 TCGA-W5-AA31-01A G3    stage i </span></span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12 TCGA-W5-AA31-11A G3    stage i </span></span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13 TCGA-W5-AA34-01A G2    stage i </span></span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14 TCGA-W5-AA34-11A G2    stage i </span></span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15 TCGA-ZU-A8S4-01A G3    stage i </span></span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 16 TCGA-ZU-A8S4-11A G3    stage i</span></span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>exprCounts <span class="ot">&lt;-</span><span class="fu">cbind</span>(Counts[,<span class="dv">1</span>], Counts[,patient_of_counts <span class="sc">%in%</span> final_patient])</span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>exprCounts <span class="ot">&lt;-</span>exprCounts[,<span class="fu">order</span>(<span class="fu">colnames</span>(exprCounts))]</span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(exprCounts)</span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   gene_name TCGA-W5-AA2I-01A TCGA-W5-AA2I-11A TCGA-W5-AA2Q-01A TCGA-W5-AA2Q-11A</span></span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1      A1BG         6.022368        12.344851         2.584963        12.300353</span></span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2      A1CF        11.853310        13.403412        10.766529        12.528698</span></span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3       A2M        12.134426        16.354680        13.839696        15.652733</span></span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4     A2ML1         1.000000         1.000000         3.000000         1.000000</span></span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5   A3GALT2         1.584963         1.584963         0.000000         0.000000</span></span>
<span id="cb16-44"><a href="#cb16-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6    A4GALT         6.303781         4.954196         8.754888         7.011227</span></span>
<span id="cb16-45"><a href="#cb16-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   TCGA-W5-AA2R-01A TCGA-W5-AA2R-11A TCGA-W5-AA2U-01A TCGA-W5-AA2U-11A</span></span>
<span id="cb16-46"><a href="#cb16-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1         4.643856        13.207167         4.321928        12.512987</span></span>
<span id="cb16-47"><a href="#cb16-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2        13.305919        13.642954        12.375311        13.517054</span></span>
<span id="cb16-48"><a href="#cb16-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3        14.002639        16.724647        14.376329        17.008801</span></span>
<span id="cb16-49"><a href="#cb16-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4         1.584963         1.584963         1.584963         1.000000</span></span>
<span id="cb16-50"><a href="#cb16-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5         1.584963         1.000000         0.000000         0.000000</span></span>
<span id="cb16-51"><a href="#cb16-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6         9.554589         6.507795        10.457381         6.149747</span></span>
<span id="cb16-52"><a href="#cb16-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   TCGA-W5-AA30-01A TCGA-W5-AA30-11A TCGA-W5-AA31-01A TCGA-W5-AA31-11A</span></span>
<span id="cb16-53"><a href="#cb16-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1         3.584963        12.910643        10.490851        12.795431</span></span>
<span id="cb16-54"><a href="#cb16-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2        11.173052        13.079318        12.532599        14.060189</span></span>
<span id="cb16-55"><a href="#cb16-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3        14.753739        17.780239        16.037032        17.310249</span></span>
<span id="cb16-56"><a href="#cb16-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4         3.584963         2.000000         3.000000         0.000000</span></span>
<span id="cb16-57"><a href="#cb16-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5         0.000000         0.000000         1.584963         0.000000</span></span>
<span id="cb16-58"><a href="#cb16-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6         8.422065         6.930737        10.700440         6.285402</span></span>
<span id="cb16-59"><a href="#cb16-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   TCGA-W5-AA34-01A TCGA-W5-AA34-11A TCGA-ZU-A8S4-01A TCGA-ZU-A8S4-11A</span></span>
<span id="cb16-60"><a href="#cb16-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1         3.700440        12.760928         4.754888        12.859535</span></span>
<span id="cb16-61"><a href="#cb16-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2        10.075479        13.854381         9.527477        13.330077</span></span>
<span id="cb16-62"><a href="#cb16-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3        12.342630        17.759088        14.328394        16.730218</span></span>
<span id="cb16-63"><a href="#cb16-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4         2.584963         2.584963         1.584963         1.000000</span></span>
<span id="cb16-64"><a href="#cb16-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5         1.000000         0.000000         1.000000         1.000000</span></span>
<span id="cb16-65"><a href="#cb16-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6         9.876517         5.247928        10.406205         5.392317</span></span>
<span id="cb16-66"><a href="#cb16-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-67"><a href="#cb16-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-68"><a href="#cb16-68" aria-hidden="true" tabindex="-1"></a><span class="co"># 去除所有样本都不表达的基因</span></span>
<span id="cb16-69"><a href="#cb16-69" aria-hidden="true" tabindex="-1"></a>exprCounts <span class="ot">&lt;-</span> exprCounts <span class="sc">%&gt;%</span></span>
<span id="cb16-70"><a href="#cb16-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">rowsum =</span> <span class="fu">apply</span>(exprCounts[,<span class="sc">-</span><span class="dv">1</span>],<span class="dv">1</span>,sum)) <span class="sc">%&gt;%</span></span>
<span id="cb16-71"><a href="#cb16-71" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(rowsum<span class="sc">!=</span><span class="dv">0</span>)</span>
<span id="cb16-72"><a href="#cb16-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-73"><a href="#cb16-73" aria-hidden="true" tabindex="-1"></a>exprCounts <span class="ot">&lt;-</span> exprCounts <span class="sc">|&gt;</span> </span>
<span id="cb16-74"><a href="#cb16-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"gene_name"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb16-75"><a href="#cb16-75" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>rowsum)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-17_b889962581228ec633a436c6cf6e074c">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 检查计数矩阵和列数据，看看它们在样本顺序方面是否一致。</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(exprCounts)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] "TCGA-W5-AA2I-01A" "TCGA-W5-AA2I-11A" "TCGA-W5-AA2Q-01A" "TCGA-W5-AA2Q-11A"</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [5] "TCGA-W5-AA2R-01A" "TCGA-W5-AA2R-11A" "TCGA-W5-AA2U-01A" "TCGA-W5-AA2U-11A"</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] "TCGA-W5-AA30-01A" "TCGA-W5-AA30-11A" "TCGA-W5-AA31-01A" "TCGA-W5-AA31-11A"</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] "TCGA-W5-AA34-01A" "TCGA-W5-AA34-11A" "TCGA-ZU-A8S4-01A" "TCGA-ZU-A8S4-11A"</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>sample<span class="sc">$</span>phenotypeID</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] "TCGA-W5-AA2I-01A" "TCGA-W5-AA2I-11A" "TCGA-W5-AA2Q-01A" "TCGA-W5-AA2Q-11A"</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [5] "TCGA-W5-AA2R-01A" "TCGA-W5-AA2R-11A" "TCGA-W5-AA2U-01A" "TCGA-W5-AA2U-11A"</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] "TCGA-W5-AA30-01A" "TCGA-W5-AA30-11A" "TCGA-W5-AA31-01A" "TCGA-W5-AA31-11A"</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] "TCGA-W5-AA34-01A" "TCGA-W5-AA34-11A" "TCGA-ZU-A8S4-01A" "TCGA-ZU-A8S4-11A"</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(sample<span class="sc">$</span>phenotypeID <span class="sc">%in%</span> <span class="fu">colnames</span>(exprCounts))</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(sample<span class="sc">$</span>phenotypeID <span class="sc">==</span> <span class="fu">colnames</span>(exprCounts))</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="建立deg对象" class="level2" data-number="0.3">
<h2 data-number="0.3" class="anchored" data-anchor-id="建立deg对象"><span class="header-section-number">0.3</span> 建立DEG对象</h2>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-18_251fd1b2ce2e41aa26ddf993cdf58774">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(exprCounts)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 18756    16</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(sample)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 16  3</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>exprCounts<span class="ot">=</span><span class="fu">ceiling</span>(exprCounts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>根据TCGA样本的命名可以区分正常组织和肿瘤样本的测序结果 其中编号01-09表示肿瘤，10-19表示正常</p>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-19_4ed13f36073d6578fcd3851fd18682d2">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 字符14~15，</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>sample<span class="sc">$</span>condition <span class="ot">&lt;-</span> <span class="fu">factor</span>(</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a> <span class="fu">ifelse</span>(<span class="fu">as.numeric</span>(<span class="fu">str_sub</span>(sample<span class="sc">$</span>phenotypeID,<span class="dv">14</span>,<span class="dv">15</span>)) <span class="sc">&lt;</span> <span class="dv">10</span>,<span class="st">'tumor'</span>,<span class="st">'normal'</span>),</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a> <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"normal"</span>,<span class="st">"tumor"</span>)</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a> ) </span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>sample <span class="ot">&lt;-</span> <span class="fu">column_to_rownames</span>(sample,<span class="at">var =</span><span class="st">"phenotypeID"</span> )</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>sample</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                  grade    stage condition</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2I-01A    G2  stage i     tumor</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2I-11A    G2  stage i    normal</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2Q-01A    G2 stage ii     tumor</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2Q-11A    G2 stage ii    normal</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2R-01A    G3  stage i     tumor</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2R-11A    G3  stage i    normal</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2U-01A    G3  stage i     tumor</span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2U-11A    G3  stage i    normal</span></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA30-01A    G3  stage i     tumor</span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA30-11A    G3  stage i    normal</span></span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA31-01A    G3  stage i     tumor</span></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA31-11A    G3  stage i    normal</span></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA34-01A    G2  stage i     tumor</span></span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA34-11A    G2  stage i    normal</span></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-ZU-A8S4-01A    G3  stage i     tumor</span></span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-ZU-A8S4-11A    G3  stage i    normal</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-20_ca3ded1ac3224852c4b92175a673f9b7">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(GenomicRanges<span class="sc">::</span>setdiff)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [conflicted] Will prefer GenomicRanges::setdiff over any other package.</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> exprCounts,</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> sample,</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> condition)</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>dds</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; class: DESeqDataSet </span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; dim: 18756 16 </span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metadata(1): version</span></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; assays(1): counts</span></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; rownames(18756): A1BG A1CF ... ZZZ3 pk</span></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; rowData names(0):</span></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; colnames(16): TCGA-W5-AA2I-01A TCGA-W5-AA2I-11A ... TCGA-ZU-A8S4-01A</span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   TCGA-ZU-A8S4-11A</span></span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; colData names(3): grade stage condition</span></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a><span class="fu">counts</span>(dds)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,]</span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      TCGA-W5-AA2I-01A TCGA-W5-AA2I-11A TCGA-W5-AA2Q-01A TCGA-W5-AA2Q-11A</span></span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A1BG                7               13                3               13</span></span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A1CF               12               14               11               13</span></span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A2M                13               17               14               16</span></span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      TCGA-W5-AA2R-01A TCGA-W5-AA2R-11A TCGA-W5-AA2U-01A TCGA-W5-AA2U-11A</span></span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A1BG                5               14                5               13</span></span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A1CF               14               14               13               14</span></span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A2M                15               17               15               18</span></span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      TCGA-W5-AA30-01A TCGA-W5-AA30-11A TCGA-W5-AA31-01A TCGA-W5-AA31-11A</span></span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A1BG                4               13               11               13</span></span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A1CF               12               14               13               15</span></span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A2M                15               18               17               18</span></span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      TCGA-W5-AA34-01A TCGA-W5-AA34-11A TCGA-ZU-A8S4-01A TCGA-ZU-A8S4-11A</span></span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A1BG                4               13                5               13</span></span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A1CF               11               14               10               14</span></span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A2M                13               18               15               17</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="预过滤" class="level3" data-number="0.3.1">
<h3 data-number="0.3.1" class="anchored" data-anchor-id="预过滤"><span class="header-section-number">0.3.1</span> 预过滤</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-21_f4237b2d3a3a6629f0b46de7ed74778f">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>smallestGroupSize <span class="ot">&lt;-</span> <span class="dv">8</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> smallestGroupSize</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[keep,]</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>dds</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; class: DESeqDataSet </span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; dim: 8637 16 </span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metadata(1): version</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; assays(1): counts</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; rownames(8637): A1BG A1CF ... ZZZ3 pk</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; rowData names(0):</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; colnames(16): TCGA-W5-AA2I-01A TCGA-W5-AA2I-11A ... TCGA-ZU-A8S4-01A</span></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   TCGA-ZU-A8S4-11A</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; colData names(3): grade stage condition</span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">$</span>condition</span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] tumor  normal tumor  normal tumor  normal tumor  normal tumor  normal</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] tumor  normal tumor  normal tumor  normal</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Levels: normal tumor</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">@</span>colData</span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; DataFrame with 16 rows and 3 columns</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                        grade       stage condition</span></span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                  &lt;character&gt; &lt;character&gt;  &lt;factor&gt;</span></span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2I-01A          G2     stage i    tumor </span></span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2I-11A          G2     stage i    normal</span></span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2Q-01A          G2    stage ii    tumor </span></span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2Q-11A          G2    stage ii    normal</span></span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2R-01A          G3     stage i    tumor </span></span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ...                      ...         ...       ...</span></span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA31-11A          G3     stage i    normal</span></span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA34-01A          G2     stage i    tumor </span></span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA34-11A          G2     stage i    normal</span></span>
<span id="cb21-33"><a href="#cb21-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-ZU-A8S4-01A          G3     stage i    tumor </span></span>
<span id="cb21-34"><a href="#cb21-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-ZU-A8S4-11A          G3     stage i    normal</span></span>
<span id="cb21-35"><a href="#cb21-35" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">@</span>assays</span>
<span id="cb21-36"><a href="#cb21-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; An object of class "SimpleAssays"</span></span>
<span id="cb21-37"><a href="#cb21-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot "data":</span></span>
<span id="cb21-38"><a href="#cb21-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; List of length 1</span></span>
<span id="cb21-39"><a href="#cb21-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; names(1): counts</span></span>
<span id="cb21-40"><a href="#cb21-40" aria-hidden="true" tabindex="-1"></a><span class="fu">counts</span>(dds)[<span class="dv">1</span>,]</span>
<span id="cb21-41"><a href="#cb21-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2I-01A TCGA-W5-AA2I-11A TCGA-W5-AA2Q-01A TCGA-W5-AA2Q-11A </span></span>
<span id="cb21-42"><a href="#cb21-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                7               13                3               13 </span></span>
<span id="cb21-43"><a href="#cb21-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2R-01A TCGA-W5-AA2R-11A TCGA-W5-AA2U-01A TCGA-W5-AA2U-11A </span></span>
<span id="cb21-44"><a href="#cb21-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                5               14                5               13 </span></span>
<span id="cb21-45"><a href="#cb21-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA30-01A TCGA-W5-AA30-11A TCGA-W5-AA31-01A TCGA-W5-AA31-11A </span></span>
<span id="cb21-46"><a href="#cb21-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                4               13               11               13 </span></span>
<span id="cb21-47"><a href="#cb21-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA34-01A TCGA-W5-AA34-11A TCGA-ZU-A8S4-01A TCGA-ZU-A8S4-11A </span></span>
<span id="cb21-48"><a href="#cb21-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                4               13                5               13</span></span>
<span id="cb21-49"><a href="#cb21-49" aria-hidden="true" tabindex="-1"></a>pre_filtered_exprCounts <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds)  <span class="co"># 14815×16</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="差异表达基因分析" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> 差异表达基因分析</h1>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-22_ddbcc1cd7bec2367e3b2c0087a279dfc">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; estimating size factors</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; estimating dispersions</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; mean-dispersion relationship</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -- note: fitType='parametric', but the dispersion trend was not well captured by the</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    function: y = a/x + b, and a local regression fit was automatically substituted.</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    specify fitType='local' or 'mean' to avoid this message next time.</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; final dispersion estimates</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; fitting model and testing</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(<span class="at">object =</span> dds,</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"condition"</span>,<span class="st">"tumor"</span>,<span class="st">"normal"</span>),</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>               <span class="at">filter =</span> <span class="fu">rowMeans</span>(<span class="fu">counts</span>(dds, <span class="at">normalized=</span><span class="cn">TRUE</span>)),</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="co">#theta = 0.99,</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a><span class="fu">metadata</span>(res)[<span class="st">"alpha"</span>]</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $alpha</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 0.1</span></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>res</span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; log2 fold change (MLE): condition tumor vs normal </span></span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Wald test p-value: condition tumor vs normal </span></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; DataFrame with 8637 rows and 6 columns</span></span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         baseMean log2FoldChange     lfcSE       stat      pvalue       padj</span></span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;   &lt;numeric&gt;  &lt;numeric&gt;</span></span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A1BG     9.52464     -1.4212155  0.329197  -4.317225 1.58003e-05 0.00174958</span></span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A1CF    13.05085     -0.3877281  0.239729  -1.617362 1.05800e-01 0.99971561</span></span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A2M     16.06687     -0.4143537  0.276715  -1.497401 1.34289e-01 0.99971561</span></span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; AAAS    11.01328      0.0472444  0.240213   0.196677 8.44080e-01 0.99971561</span></span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; AACS     9.72926      0.2609899  0.281901   0.925820 3.54539e-01 0.99971561</span></span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ...          ...            ...       ...        ...         ...        ...</span></span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ZYG11B   11.3580    -0.13352238  0.237200 -0.5629100    0.573496   0.999716</span></span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ZYX      12.5775     0.00636048  0.237028  0.0268343    0.978592   0.999716</span></span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ZZEF1    10.9441     0.06469033  0.241048  0.2683707    0.788414   0.999716</span></span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ZZZ3     10.8275     0.03405370  0.242467  0.1404468    0.888307   0.999716</span></span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; pk       11.9390     0.07584185  0.236057  0.3212861    0.747994   0.999716</span></span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a><span class="co"># 根据padj 升序</span></span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a>resOrdered <span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj),]</span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resOrdered)</span>
<span id="cb22-42"><a href="#cb22-42" aria-hidden="true" tabindex="-1"></a><span class="co"># 导出到csv</span></span>
<span id="cb22-43"><a href="#cb22-43" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(res, <span class="at">file=</span><span class="st">"data/resOrdered.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="结果分析" class="level3" data-number="1.0.1">
<h3 data-number="1.0.1" class="anchored" data-anchor-id="结果分析"><span class="header-section-number">1.0.1</span> 结果分析</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-23_dc083de3da194eade004c45fadbbfcc0">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mcols</span>(resOrdered)<span class="sc">$</span>description</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] "mean of normalized counts for all samples"        </span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [2] "log2 fold change (MLE): condition tumor vs normal"</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [3] "standard error: condition tumor vs normal"        </span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [4] "Wald statistic: condition tumor vs normal"        </span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [5] "Wald test p-value: condition tumor vs normal"     </span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [6] "BH adjusted p-values"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>pvalue，是统计学检验变量，代表差异显著性，一般认为P &lt; 0.05 为显著， P &lt;0.01 为非常显著。其含义为：由抽样误差导致样本间差异的概率小于0.05 或0.01。</p>
<p>padj，是对pvalue进行多重假设检验校正后的结果。转录组测序的差异表达分析是对大量的基因表达值进行的独立统计假设检验，存在假阳性（Family-Wise Error Rate，FWER），Bonferroni矫正</p>
<p>qvalue，False Discovery Rate，FDR，所有假阳性结果占所有拒绝零假设结果的比例，假阳性结果的比例不超过预设的FDR水平（0.05/0.1）,基于pvalue的分布。多重假设检验校正方法：Benjamini-Hochberg，</p>
<p>log2FoldChange：对数倍数变化，1.5倍差异即0.585，2倍差异即1。</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="data/FoldChange.jpg" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-24_624d83438275b967b2ed76dfa4ab4d42">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>DEG <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data/resOrdered.csv"</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; New names:</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Rows: 8637 Columns: 7</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ── Column specification</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ──────────────────────────────────────────────────────── Delimiter: "," chr</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; (1): ...1 dbl (6): baseMean, log2FoldChange, lfcSE, stat, pvalue, padj</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ Use `spec()` to retrieve the full column specification for this data. ℹ</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; • `` -&gt; `...1`</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>DESeq2_DEG <span class="ot">=</span> <span class="fu">na.omit</span>(DEG) </span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>nrDEG<span class="ot">=</span>DESeq2_DEG</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="热图" class="level3" data-number="1.0.2">
<h3 data-number="1.0.2" class="anchored" data-anchor-id="热图"><span class="header-section-number">1.0.2</span> 热图</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-25_3017f71f2b7bf34cedac40ec8a5b1c42">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>pre_filtered_exprCounts[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      TCGA-W5-AA2I-01A TCGA-W5-AA2I-11A TCGA-W5-AA2Q-01A TCGA-W5-AA2Q-11A</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A1BG                7               13                3               13</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A1CF               12               14               11               13</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A2M                13               17               14               16</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; AAAS               12               10               12               10</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>log2exprCounts <span class="ot">&lt;-</span> <span class="fu">log2</span>(pre_filtered_exprCounts<span class="sc">+</span><span class="dv">1</span>)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>DESeq2_DEG <span class="ot">&lt;-</span> DESeq2_DEG <span class="sc">|&gt;</span> <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"...1"</span>)</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(log2exprCounts[<span class="fu">rownames</span>(<span class="fu">head</span>(DESeq2_DEG,<span class="at">n=</span><span class="dv">20</span>)) , ])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TranscriptomicsDownstreamAnalysis_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="挑选指定第一个基因看它在同一个病人的配对表达情况" class="level3" data-number="1.0.3">
<h3 data-number="1.0.3" class="anchored" data-anchor-id="挑选指定第一个基因看它在同一个病人的配对表达情况"><span class="header-section-number">1.0.3</span> 挑选指定第一个基因看它在同一个病人的配对表达情况</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-26_967791d69791fc13a14e75f9e6d9585f">
<pre><code>#&gt; [conflicted] Removing existing preference.
#&gt; [conflicted] Will prefer GenomicRanges::setdiff over any other package.</code></pre>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-27_fe8596d5b6be7146fb91849bac41cc49">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">g=</span>sample<span class="sc">$</span>condition,</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">p=</span><span class="fu">colnames</span>(exprCounts),</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">v=</span><span class="fu">as.numeric</span>(exprCounts[<span class="fu">rownames</span>(DESeq2_DEG[<span class="dv">1</span>,]),])) </span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggpaired</span>(df, <span class="at">x =</span> <span class="st">"g"</span>, <span class="at">y =</span> <span class="st">"v"</span>,<span class="at">color =</span> <span class="st">"g"</span>, </span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">line.color =</span> <span class="st">"gray"</span>, <span class="at">line.size =</span> <span class="fl">0.4</span>,<span class="at">palette =</span> <span class="st">"npg"</span>)<span class="sc">+</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_compare_means</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TranscriptomicsDownstreamAnalysis_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="火山图" class="level3" data-number="1.0.4">
<h3 data-number="1.0.4" class="anchored" data-anchor-id="火山图"><span class="header-section-number">1.0.4</span> 火山图</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-28_279ce03f01c582a54b5a1d391a17cf4d">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>nrDEG<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">if_else</span>(nrDEG<span class="sc">$</span>padj<span class="sc">&lt;</span><span class="fl">0.2</span><span class="sc">&amp;</span> <span class="fu">abs</span>(nrDEG<span class="sc">$</span>log2FoldChange)<span class="sc">&gt;=</span><span class="fl">0.585</span>,</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>            <span class="fu">if_else</span>(nrDEG<span class="sc">$</span>log2FoldChange<span class="sc">&gt;=</span><span class="fl">0.585</span>,<span class="st">"up"</span>,<span class="st">"down"</span>),</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>            <span class="st">"normal"</span>,),</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">levels =</span><span class="fu">c</span>(<span class="st">"up"</span>,<span class="st">"down"</span>,<span class="st">"normal"</span>) )</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(nrDEG<span class="sc">$</span>group)</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     up   down normal </span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     10    243   8384</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>gene_labels <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(nrDEG,nrDEG<span class="sc">$</span>padj<span class="sc">&lt;</span><span class="fl">0.2</span> <span class="sc">&amp;</span> </span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">abs</span>(nrDEG<span class="sc">$</span>log2FoldChange)<span class="sc">&gt;=</span><span class="fl">0.585</span>)<span class="sc">|&gt;</span> </span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n=</span><span class="dv">10</span>) <span class="sc">|&gt;</span> <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"...1"</span>)</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(nrDEG,<span class="fu">aes</span>( <span class="at">x =</span> log2FoldChange,<span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj),<span class="at">color=</span>group,<span class="at">shape=</span>group))<span class="sc">+</span></span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>,<span class="at">size=</span><span class="dv">2</span>)<span class="sc">+</span></span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"red"</span>,<span class="st">"green"</span>,<span class="st">"gray"</span>))<span class="sc">+</span></span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>    <span class="co">#scale_shape_manual(values = c(2,25,4))+</span></span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.585</span>,<span class="dv">0</span>,<span class="fl">0.585</span>),<span class="at">lty=</span><span class="dv">3</span>,<span class="at">color=</span><span class="st">"grey25"</span>,<span class="at">lwd=</span><span class="fl">0.8</span>)<span class="sc">+</span></span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.2</span>),<span class="at">lty=</span><span class="dv">3</span>,<span class="at">color=</span><span class="st">"grey25"</span>,<span class="at">lwd=</span><span class="fl">0.8</span>)<span class="sc">+</span></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>    ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(</span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">data =</span> gene_labels,</span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">label=</span><span class="fu">row.names</span>(gene_labels)),</span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">color=</span><span class="st">"black"</span>,</span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">size=</span><span class="dv">2</span>)<span class="sc">+</span></span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TranscriptomicsDownstreamAnalysis_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="lfc收缩和pagj" class="level3" data-number="1.0.5">
<h3 data-number="1.0.5" class="anchored" data-anchor-id="lfc收缩和pagj"><span class="header-section-number">1.0.5</span> LFC收缩和pagj</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-29_e5fe5f05f525ca59b093cad47fce738b">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">resultsNames</span>(dds)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] "Intercept"                 "condition_tumor_vs_normal"</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("apeglm")  自适应 t 先验收缩估计器  默认</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>resLFC <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds, <span class="at">coef=</span><span class="st">"condition_tumor_vs_normal"</span>, <span class="at">type=</span><span class="st">"apeglm"</span>)</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; using 'apeglm' for LFC shrinkage. If used in published research, please cite:</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     sequence count data: removing the noise and preserving large differences.</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a><span class="co"># `normal`是原始的 DESeq2 收缩估计器，是先验的自适应正态分布</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>resNorm <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds, <span class="at">coef=</span><span class="dv">2</span>, <span class="at">type=</span><span class="st">"normal"</span>)</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; using 'normal' for LFC shrinkage, the Normal prior from Love et al (2014).</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Note that type='apeglm' and type='ashr' have shown to have less bias than type='normal'.</span></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; See ?lfcShrink for more details on shrinkage type, and the DESeq2 vignette.</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Reference: https://doi.org/10.1093/bioinformatics/bty895</span></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("ashr")</span></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>resAsh <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds, <span class="at">coef=</span><span class="dv">2</span>, <span class="at">type=</span><span class="st">"ashr"</span>)</span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; using 'ashr' for LFC shrinkage. If used in published research, please cite:</span></span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.</span></span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     https://doi.org/10.1093/biostatistics/kxw041</span></span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res)</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     baseMean     log2FoldChange          lfcSE             stat         </span></span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Min.   : 6.53   Min.   :-2.613850   Min.   :0.2358   Min.   :-6.58439  </span></span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1st Qu.:10.15   1st Qu.:-0.097222   1st Qu.:0.2375   1st Qu.:-0.39214  </span></span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Median :10.92   Median : 0.003097   Median :0.2437   Median : 0.01303  </span></span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Mean   :11.10   Mean   :-0.036818   Mean   :0.2556   Mean   :-0.13382  </span></span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3rd Qu.:11.82   3rd Qu.: 0.091066   3rd Qu.:0.2587   3rd Qu.: 0.35789  </span></span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Max.   :20.23   Max.   : 1.165006   Max.   :1.0519   Max.   : 3.34424  </span></span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      pvalue            padj          </span></span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Min.   :0.0000   Min.   :0.0000004  </span></span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1st Qu.:0.4660   1st Qu.:0.9997156  </span></span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Median :0.7058   Median :0.9997156  </span></span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Mean   :0.6369   Mean   :0.9600961  </span></span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3rd Qu.:0.8680   3rd Qu.:0.9997156  </span></span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Max.   :0.9998   Max.   :0.9998196</span></span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 178</span></span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a>res05 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb29-43"><a href="#cb29-43" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res05)</span>
<span id="cb29-44"><a href="#cb29-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-45"><a href="#cb29-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; out of 8637 with nonzero total read count</span></span>
<span id="cb29-46"><a href="#cb29-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; adjusted p-value &lt; 0.05</span></span>
<span id="cb29-47"><a href="#cb29-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; LFC &gt; 0 (up)       : 1, 0.012%</span></span>
<span id="cb29-48"><a href="#cb29-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; LFC &lt; 0 (down)     : 177, 2%</span></span>
<span id="cb29-49"><a href="#cb29-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; outliers [1]       : 0, 0%</span></span>
<span id="cb29-50"><a href="#cb29-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; low counts [2]     : 0, 0%</span></span>
<span id="cb29-51"><a href="#cb29-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; (mean count &lt; 7)</span></span>
<span id="cb29-52"><a href="#cb29-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] see 'cooksCutoff' argument of ?results</span></span>
<span id="cb29-53"><a href="#cb29-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [2] see 'independentFiltering' argument of ?results</span></span>
<span id="cb29-54"><a href="#cb29-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-55"><a href="#cb29-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-56"><a href="#cb29-56" aria-hidden="true" tabindex="-1"></a><span class="do">## 先验信息</span></span>
<span id="cb29-57"><a href="#cb29-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-58"><a href="#cb29-58" aria-hidden="true" tabindex="-1"></a><span class="fu">priorInfo</span>(resLFC)</span>
<span id="cb29-59"><a href="#cb29-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $type</span></span>
<span id="cb29-60"><a href="#cb29-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] "apeglm"</span></span>
<span id="cb29-61"><a href="#cb29-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-62"><a href="#cb29-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $package</span></span>
<span id="cb29-63"><a href="#cb29-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] "apeglm"</span></span>
<span id="cb29-64"><a href="#cb29-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-65"><a href="#cb29-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $version</span></span>
<span id="cb29-66"><a href="#cb29-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] '1.24.0'</span></span>
<span id="cb29-67"><a href="#cb29-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-68"><a href="#cb29-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $prior.control</span></span>
<span id="cb29-69"><a href="#cb29-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $prior.control$no.shrink</span></span>
<span id="cb29-70"><a href="#cb29-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 1</span></span>
<span id="cb29-71"><a href="#cb29-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-72"><a href="#cb29-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $prior.control$prior.mean</span></span>
<span id="cb29-73"><a href="#cb29-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 0</span></span>
<span id="cb29-74"><a href="#cb29-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-75"><a href="#cb29-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $prior.control$prior.scale</span></span>
<span id="cb29-76"><a href="#cb29-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 0.001</span></span>
<span id="cb29-77"><a href="#cb29-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-78"><a href="#cb29-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $prior.control$prior.df</span></span>
<span id="cb29-79"><a href="#cb29-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 1</span></span>
<span id="cb29-80"><a href="#cb29-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-81"><a href="#cb29-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $prior.control$prior.no.shrink.mean</span></span>
<span id="cb29-82"><a href="#cb29-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 0</span></span>
<span id="cb29-83"><a href="#cb29-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-84"><a href="#cb29-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $prior.control$prior.no.shrink.scale</span></span>
<span id="cb29-85"><a href="#cb29-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 15</span></span>
<span id="cb29-86"><a href="#cb29-86" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-87"><a href="#cb29-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $prior.control$prior.var</span></span>
<span id="cb29-88"><a href="#cb29-88" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 1e-06</span></span>
<span id="cb29-89"><a href="#cb29-89" aria-hidden="true" tabindex="-1"></a><span class="fu">priorInfo</span>(resNorm)</span>
<span id="cb29-90"><a href="#cb29-90" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $type</span></span>
<span id="cb29-91"><a href="#cb29-91" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] "normal"</span></span>
<span id="cb29-92"><a href="#cb29-92" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-93"><a href="#cb29-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $package</span></span>
<span id="cb29-94"><a href="#cb29-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] "DESeq2"</span></span>
<span id="cb29-95"><a href="#cb29-95" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-96"><a href="#cb29-96" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $version</span></span>
<span id="cb29-97"><a href="#cb29-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] '1.42.1'</span></span>
<span id="cb29-98"><a href="#cb29-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-99"><a href="#cb29-99" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $betaPriorVar</span></span>
<span id="cb29-100"><a href="#cb29-100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      Intercept conditiontumor </span></span>
<span id="cb29-101"><a href="#cb29-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   1.000000e+06   6.080778e-02</span></span>
<span id="cb29-102"><a href="#cb29-102" aria-hidden="true" tabindex="-1"></a><span class="fu">priorInfo</span>(resAsh)</span>
<span id="cb29-103"><a href="#cb29-103" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $type</span></span>
<span id="cb29-104"><a href="#cb29-104" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] "ashr"</span></span>
<span id="cb29-105"><a href="#cb29-105" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-106"><a href="#cb29-106" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $package</span></span>
<span id="cb29-107"><a href="#cb29-107" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] "ashr"</span></span>
<span id="cb29-108"><a href="#cb29-108" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-109"><a href="#cb29-109" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $version</span></span>
<span id="cb29-110"><a href="#cb29-110" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] '2.2.63'</span></span>
<span id="cb29-111"><a href="#cb29-111" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-112"><a href="#cb29-112" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $fitted_g</span></span>
<span id="cb29-113"><a href="#cb29-113" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $pi</span></span>
<span id="cb29-114"><a href="#cb29-114" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] 0.964379719 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000</span></span>
<span id="cb29-115"><a href="#cb29-115" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 0.026291632</span></span>
<span id="cb29-116"><a href="#cb29-116" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] 0.009328649 0.000000000 0.000000000 0.000000000 0.000000000</span></span>
<span id="cb29-117"><a href="#cb29-117" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-118"><a href="#cb29-118" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $mean</span></span>
<span id="cb29-119"><a href="#cb29-119" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span id="cb29-120"><a href="#cb29-120" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-121"><a href="#cb29-121" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $sd</span></span>
<span id="cb29-122"><a href="#cb29-122" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] 0.02002812 0.02832403 0.04005623 0.05664807 0.08011246 0.11329613</span></span>
<span id="cb29-123"><a href="#cb29-123" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] 0.16022493 0.22659227 0.32044986 0.45318454 0.64089972 0.90636908</span></span>
<span id="cb29-124"><a href="#cb29-124" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] 1.28179944 1.81273815 2.56359888 3.62547630 5.12719776</span></span>
<span id="cb29-125"><a href="#cb29-125" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb29-126"><a href="#cb29-126" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,"class")</span></span>
<span id="cb29-127"><a href="#cb29-127" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] "normalmix"</span></span>
<span id="cb29-128"><a href="#cb29-128" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,"row.names")</span></span>
<span id="cb29-129"><a href="#cb29-129" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="ma图-mean-absdeviation-plot" class="level3" data-number="1.0.6">
<h3 data-number="1.0.6" class="anchored" data-anchor-id="ma图-mean-absdeviation-plot"><span class="header-section-number">1.0.6</span> MA图 Mean-AbsDeviation plot</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-30_562debb85c4bfdce61c747dc83ff465e">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 1.将DESeq结果转换为dataframe</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DESeq2_DEG)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;          baseMean log2FoldChange     lfcSE      stat       pvalue         padj</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ADH4    11.448069      -1.860929 0.2826274 -6.584390 4.567573e-11 3.945013e-07</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CYP2A6  11.106269      -1.762143 0.2777187 -6.345064 2.223333e-10 9.601462e-07</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CYP3A4  11.171913      -1.729042 0.2760280 -6.264010 3.752012e-10 1.080204e-06</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; SULT2A1 10.111359      -1.914233 0.3130939 -6.113927 9.720853e-10 2.098975e-06</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; BHMT    10.395426      -1.727164 0.2880450 -5.996161 2.020365e-09 3.489979e-06</span></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; F9       9.917009      -1.931105 0.3279613 -5.888209 3.904033e-09 5.619856e-06</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. 对gene添加“是否呈现显著差异表达”的标签</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>DESeq2_DEG<span class="sc">$</span>significant <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">ifelse</span>(DESeq2_DEG<span class="sc">$</span>padj <span class="sc">&lt;</span><span class="fl">0.05</span>, <span class="st">"Significant"</span>, <span class="st">"NS"</span>),</span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a><span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Significant"</span>, <span class="st">"NS"</span>))</span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(DESeq2_DEG<span class="sc">$</span>significant)</span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Significant          NS </span></span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         178        8459</span></span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a><span class="co"># 以baseMean作为x，log2FoldChange作为y   以significant作为分类变量</span></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(DESeq2_DEG, <span class="fu">aes</span>(baseMean, log2FoldChange, <span class="at">colour=</span>significant)) <span class="sc">+</span> </span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a> <span class="fu">scale_y_continuous</span>(<span class="at">limits=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">3</span>, <span class="dv">3</span>)) <span class="sc">+</span> </span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a> <span class="fu">scale_x_log10</span>() <span class="sc">+</span> </span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">colour=</span><span class="st">"black"</span>, <span class="at">linewidth=</span><span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"mean of normalized counts"</span>, <span class="at">y=</span><span class="st">"log2FoldChange"</span>) <span class="sc">+</span> </span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a> <span class="fu">scale_colour_manual</span>(<span class="at">name=</span><span class="st">"padj"</span>, </span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>                     <span class="at">values=</span><span class="fu">c</span>(<span class="st">"Significant"</span><span class="ot">=</span><span class="st">"blue"</span>,<span class="st">"NS"</span><span class="ot">=</span><span class="st">"grey50"</span>)) <span class="sc">+</span></span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a> <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TranscriptomicsDownstreamAnalysis_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-31_c9fff3d3a6b30f5357b73eb73c200658">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>xlim <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">25</span>)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>ylim <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">3</span>,<span class="dv">3</span>)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resOrdered,  <span class="at">xlim=</span>xlim, <span class="at">ylim=</span>ylim,<span class="at">main=</span><span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TranscriptomicsDownstreamAnalysis_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid" width="672"></p>
</div>
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC, <span class="at">xlim=</span>xlim, <span class="at">ylim=</span>ylim, <span class="at">main=</span><span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TranscriptomicsDownstreamAnalysis_files/figure-html/unnamed-chunk-31-2.png" class="img-fluid" width="672"></p>
</div>
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resNorm, <span class="at">xlim=</span>xlim, <span class="at">ylim=</span>ylim, <span class="at">main=</span><span class="st">"normal"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TranscriptomicsDownstreamAnalysis_files/figure-html/unnamed-chunk-31-3.png" class="img-fluid" width="672"></p>
</div>
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resAsh, <span class="at">xlim=</span>xlim, <span class="at">ylim=</span>ylim, <span class="at">main=</span><span class="st">"ashr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TranscriptomicsDownstreamAnalysis_files/figure-html/unnamed-chunk-31-4.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="readcounts" class="level3" data-number="1.0.7">
<h3 data-number="1.0.7" class="anchored" data-anchor-id="readcounts"><span class="header-section-number">1.0.7</span> readCounts</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-32_24fd2b30a6ba199a608267b017b71072">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co">#  the counts of reads for a single gene across the groups</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="fu">which.min</span>(resOrdered<span class="sc">$</span>padj), </span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">intgroup=</span><span class="st">"condition"</span>,<span class="at">returnData=</span><span class="cn">TRUE</span>) <span class="sc">|&gt;</span> </span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>condition, <span class="at">y=</span>count)) <span class="sc">+</span> </span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">position=</span><span class="fu">position_jitter</span>(<span class="at">w=</span><span class="fl">0.1</span>,<span class="at">h=</span><span class="dv">0</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TranscriptomicsDownstreamAnalysis_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="异常观测" class="level3" data-number="1.0.8">
<h3 data-number="1.0.8" class="anchored" data-anchor-id="异常观测"><span class="header-section-number">1.0.8</span> 异常观测</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-33_569d8e352d5c7c16a94ab5e93695393e">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>W <span class="ot">&lt;-</span> resOrdered<span class="sc">$</span>stat</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>maxCooks <span class="ot">&lt;-</span> <span class="fu">apply</span>(<span class="fu">assays</span>(dds)[[<span class="st">"cooks"</span>]],<span class="dv">1</span>,max)</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="sc">!</span><span class="fu">is.na</span>(W)</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">rank</span>(W[idx]), maxCooks[idx], <span class="at">xlab=</span><span class="st">"rank of Wald statistic"</span>, </span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">ylab=</span><span class="st">"maximum Cook's distance per gene"</span>,</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">5</span>), <span class="at">cex=</span>.<span class="dv">4</span>, <span class="at">col=</span><span class="fu">rgb</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,.<span class="dv">3</span>))</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">ncol</span>(dds)</span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="fu">qf</span>(.<span class="dv">99</span>, p, m <span class="sc">-</span> p))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TranscriptomicsDownstreamAnalysis_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="多因素设计" class="level3" data-number="1.0.9">
<h3 data-number="1.0.9" class="anchored" data-anchor-id="多因素设计"><span class="header-section-number">1.0.9</span> 多因素设计</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-34_d7caf4237ab2eb33d699de236d94cdc3">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(dds)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; DataFrame with 16 rows and 5 columns</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                        grade       stage condition sizeFactor replaceable</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                  &lt;character&gt; &lt;character&gt;  &lt;factor&gt;  &lt;numeric&gt;   &lt;logical&gt;</span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2I-01A          G2     stage i    tumor    1.023464        TRUE</span></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2I-11A          G2     stage i    normal   0.941406        TRUE</span></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2Q-01A          G2    stage ii    tumor    1.077360        TRUE</span></span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2Q-11A          G2    stage ii    normal   0.913683        TRUE</span></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA2R-01A          G3     stage i    tumor    1.069652        TRUE</span></span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ...                      ...         ...       ...        ...         ...</span></span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA31-11A          G3     stage i    normal   0.961966        TRUE</span></span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA34-01A          G2     stage i    tumor    1.046461        TRUE</span></span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-W5-AA34-11A          G2     stage i    normal   0.980547        TRUE</span></span>
<span id="cb37-14"><a href="#cb37-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-ZU-A8S4-01A          G3     stage i    tumor    1.047917        TRUE</span></span>
<span id="cb37-15"><a href="#cb37-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; TCGA-ZU-A8S4-11A          G3     stage i    normal   0.937547        TRUE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="导出到-csv-文件" class="level3" data-number="1.0.10">
<h3 data-number="1.0.10" class="anchored" data-anchor-id="导出到-csv-文件"><span class="header-section-number">1.0.10</span> 导出到 CSV 文件</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-35_7d5562d278c0f5da835a5089a85a72b7">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>resSig <span class="ot">&lt;-</span> <span class="fu">subset</span>(resOrdered, padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>resSig</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; log2 fold change (MLE): condition tumor vs normal </span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Wald test p-value: condition tumor vs normal </span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; DataFrame with 178 rows and 6 columns</span></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;            baseMean log2FoldChange     lfcSE      stat      pvalue        padj</span></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;           &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ADH4        11.4481       -1.86093  0.282627  -6.58439 4.56757e-11 3.94501e-07</span></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CYP2A6      11.1063       -1.76214  0.277719  -6.34506 2.22333e-10 9.60146e-07</span></span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; CYP3A4      11.1719       -1.72904  0.276028  -6.26401 3.75201e-10 1.08020e-06</span></span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; SULT2A1     10.1114       -1.91423  0.313094  -6.11393 9.72085e-10 2.09898e-06</span></span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; BHMT        10.3954       -1.72716  0.288045  -5.99616 2.02037e-09 3.48998e-06</span></span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ...             ...            ...       ...       ...         ...         ...</span></span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; OIT3        8.60279      -1.316532  0.397119  -3.31521 0.000915740   0.0454554</span></span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; C6         12.65249      -0.810609  0.245057  -3.30784 0.000940179   0.0464019</span></span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; SERPINA10  11.63048      -0.803304  0.243531  -3.29857 0.000971785   0.0476892</span></span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; F12        12.78240      -0.804257  0.244270  -3.29250 0.000993022   0.0481839</span></span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; KLKB1      11.31962      -0.804306  0.244229  -3.29325 0.000990373   0.0481839</span></span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-21"><a href="#cb38-21" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(<span class="fu">as.data.frame</span>(resSig), </span>
<span id="cb38-22"><a href="#cb38-22" aria-hidden="true" tabindex="-1"></a> <span class="at">file=</span><span class="st">"data/padjust_0.05_resOrdered_tumor_normal.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="富集分析" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> 富集分析</h1>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-36_2c85ce76538d55198fd29b53b8ccd662">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install('clusterProfiler')</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("org.Hs.eg.db")</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>DEG <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/resOrdered.csv"</span>)</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>DESeq2_DEG <span class="ot">=</span> <span class="fu">na.omit</span>(DEG) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>对于有参考基因组物种的分析，可以在相关软件包中直接加载该物种的背景基因集 （1）对于常见的模式物种，例如人类，有些专门的 R 包数据库，例如人类参考基因组 hg19 的 <code>library(org.Hs.eg.db)</code></p>
<section id="数据整理" class="level3" data-number="2.0.1">
<h3 data-number="2.0.1" class="anchored" data-anchor-id="数据整理"><span class="header-section-number">2.0.1</span> 数据整理</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-38_82d464aa0e2785d85f9b80e8972c5137">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>nrDEG <span class="ot">&lt;-</span> DESeq2_DEG</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(nrDEG)</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;         X  baseMean log2FoldChange     lfcSE      stat       pvalue</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1    ADH4 11.448069      -1.860929 0.2826274 -6.584390 4.567573e-11</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2  CYP2A6 11.106269      -1.762143 0.2777187 -6.345064 2.223333e-10</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3  CYP3A4 11.171913      -1.729042 0.2760280 -6.264010 3.752012e-10</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 SULT2A1 10.111359      -1.914233 0.3130939 -6.113927 9.720853e-10</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5    BHMT 10.395426      -1.727164 0.2880450 -5.996161 2.020365e-09</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6      F9  9.917009      -1.931105 0.3279613 -5.888209 3.904033e-09</span></span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;           padj</span></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 3.945013e-07</span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 9.601462e-07</span></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 1.080204e-06</span></span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 2.098975e-06</span></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 3.489979e-06</span></span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 5.619856e-06</span></span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a>nrDEG<span class="sc">$</span>expression <span class="ot">&lt;-</span> <span class="fu">as.factor</span>( </span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>( nrDEG<span class="sc">$</span>padj<span class="sc">&lt;</span> <span class="fl">0.2</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(nrDEG<span class="sc">$</span>log2FoldChange) <span class="sc">&gt;=</span> <span class="dv">1</span>,</span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a>            <span class="fu">ifelse</span>( nrDEG<span class="sc">$</span>log2FoldChange <span class="sc">&gt;=</span> <span class="dv">1</span> , <span class="st">'up'</span>, <span class="st">'down'</span> ),</span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a>            <span class="st">'normal'</span>))</span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(nrDEG<span class="sc">$</span>expression)</span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   down normal     up </span></span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    145   8489      3</span></span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-27"><a href="#cb40-27" aria-hidden="true" tabindex="-1"></a>nrDEG <span class="ot">&lt;-</span> nrDEG <span class="sc">|&gt;</span> </span>
<span id="cb40-28"><a href="#cb40-28" aria-hidden="true" tabindex="-1"></a> <span class="fu">inner_join</span>(gtf_mrna_v22, <span class="at">by =</span> <span class="fu">c</span>( <span class="st">"X"</span><span class="ot">=</span> <span class="st">"gene_name"</span>))</span>
<span id="cb40-29"><a href="#cb40-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-30"><a href="#cb40-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-31"><a href="#cb40-31" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(nrDEG) <span class="ot">&lt;-</span> <span class="fu">str_sub</span>(nrDEG<span class="sc">$</span>gene_id, <span class="at">start =</span> <span class="dv">1</span>, <span class="at">end =</span> <span class="dv">15</span>)</span>
<span id="cb40-32"><a href="#cb40-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-33"><a href="#cb40-33" aria-hidden="true" tabindex="-1"></a>nrDEG<span class="sc">$</span>ENSEMBL <span class="ot">&lt;-</span> <span class="fu">rownames</span>( nrDEG )</span>
<span id="cb40-34"><a href="#cb40-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-35"><a href="#cb40-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-36"><a href="#cb40-36" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb40-37"><a href="#cb40-37" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb40-38"><a href="#cb40-38" aria-hidden="true" tabindex="-1"></a>AnnotationDbi<span class="sc">::</span><span class="fu">keytypes</span>(org.Hs.eg.db)</span>
<span id="cb40-39"><a href="#cb40-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] "ACCNUM"       "ALIAS"        "ENSEMBL"      "ENSEMBLPROT"  "ENSEMBLTRANS"</span></span>
<span id="cb40-40"><a href="#cb40-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [6] "ENTREZID"     "ENZYME"       "EVIDENCE"     "EVIDENCEALL"  "GENENAME"    </span></span>
<span id="cb40-41"><a href="#cb40-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] "GENETYPE"     "GO"           "GOALL"        "IPI"          "MAP"         </span></span>
<span id="cb40-42"><a href="#cb40-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [16] "OMIM"         "ONTOLOGY"     "ONTOLOGYALL"  "PATH"         "PFAM"        </span></span>
<span id="cb40-43"><a href="#cb40-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [21] "PMID"         "PROSITE"      "REFSEQ"       "SYMBOL"       "UCSCKG"      </span></span>
<span id="cb40-44"><a href="#cb40-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [26] "UNIPROT"</span></span>
<span id="cb40-45"><a href="#cb40-45" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">keys</span>(org.Hs.eg.db, <span class="st">"ENSEMBL"</span>))</span>
<span id="cb40-46"><a href="#cb40-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] "ENSG00000121410" "ENSG00000175899" "ENSG00000291190" "ENSG00000171428"</span></span>
<span id="cb40-47"><a href="#cb40-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [5] "ENSG00000156006" "ENSG00000196136"</span></span>
<span id="cb40-48"><a href="#cb40-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-49"><a href="#cb40-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-50"><a href="#cb40-50" aria-hidden="true" tabindex="-1"></a>valid_ensembl <span class="ot">&lt;-</span> <span class="fu">rownames</span>(nrDEG) <span class="sc">%in%</span> <span class="fu">keys</span>(org.Hs.eg.db, <span class="st">"ENSEMBL"</span>)  </span>
<span id="cb40-51"><a href="#cb40-51" aria-hidden="true" tabindex="-1"></a>nrDEG_filtered <span class="ot">&lt;-</span> nrDEG[valid_ensembl, ]  </span>
<span id="cb40-52"><a href="#cb40-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-53"><a href="#cb40-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-54"><a href="#cb40-54" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">bitr</span>(</span>
<span id="cb40-55"><a href="#cb40-55" aria-hidden="true" tabindex="-1"></a> <span class="at">geneID=</span> <span class="fu">rownames</span>(nrDEG_filtered),</span>
<span id="cb40-56"><a href="#cb40-56" aria-hidden="true" tabindex="-1"></a> <span class="at">fromType =</span> <span class="st">"ENSEMBL"</span>, </span>
<span id="cb40-57"><a href="#cb40-57" aria-hidden="true" tabindex="-1"></a> <span class="at">toType =</span> <span class="fu">c</span>( <span class="st">"ENTREZID"</span>,<span class="st">"SYMBOL"</span> ), </span>
<span id="cb40-58"><a href="#cb40-58" aria-hidden="true" tabindex="-1"></a> <span class="at">OrgDb =</span> org.Hs.eg.db )</span>
<span id="cb40-59"><a href="#cb40-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span id="cb40-60"><a href="#cb40-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-61"><a href="#cb40-61" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(df)</span>
<span id="cb40-62"><a href="#cb40-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;           ENSEMBL ENTREZID  SYMBOL</span></span>
<span id="cb40-63"><a href="#cb40-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 ENSG00000198099      127    ADH4</span></span>
<span id="cb40-64"><a href="#cb40-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 ENSG00000255974     1548  CYP2A6</span></span>
<span id="cb40-65"><a href="#cb40-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 ENSG00000160868     1576  CYP3A4</span></span>
<span id="cb40-66"><a href="#cb40-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 ENSG00000105398     6822 SULT2A1</span></span>
<span id="cb40-67"><a href="#cb40-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 ENSG00000145692      635    BHMT</span></span>
<span id="cb40-68"><a href="#cb40-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 ENSG00000101981     2158      F9</span></span>
<span id="cb40-69"><a href="#cb40-69" aria-hidden="true" tabindex="-1"></a>nrDEG_filtered<span class="sc">$</span>SYMBOL <span class="ot">=</span> <span class="fu">rownames</span>(nrDEG_filtered)</span>
<span id="cb40-70"><a href="#cb40-70" aria-hidden="true" tabindex="-1"></a>nrDEG <span class="ot">=</span> <span class="fu">inner_join</span>(nrDEG_filtered, df, <span class="at">by=</span><span class="st">'ENSEMBL'</span> )</span>
<span id="cb40-71"><a href="#cb40-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-72"><a href="#cb40-72" aria-hidden="true" tabindex="-1"></a>gene_up <span class="ot">=</span> nrDEG[ nrDEG<span class="sc">$</span>expression <span class="sc">==</span> <span class="st">'up'</span>, <span class="st">"ENTREZID"</span> ] </span>
<span id="cb40-73"><a href="#cb40-73" aria-hidden="true" tabindex="-1"></a>gene_down <span class="ot">=</span> nrDEG[ nrDEG<span class="sc">$</span>expression <span class="sc">==</span> <span class="st">'down'</span>, <span class="st">"ENTREZID"</span>]</span>
<span id="cb40-74"><a href="#cb40-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-75"><a href="#cb40-75" aria-hidden="true" tabindex="-1"></a>gene_all <span class="ot">=</span> <span class="fu">as.character</span>(nrDEG[ ,<span class="st">"ENTREZID"</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="kegg-通路富集" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="kegg-通路富集"><span class="header-section-number">2.1</span> KEGG 通路富集</h2>
<p>对于各列内容：</p>
<p>ID和Description，富集到的通路和功能描述；</p>
<p>GeneRatio和BgRatio，分别为富集到该通路中的基因数目/给定基因的总数目，以及该通路中背景基因总数目/该物种所有已知的KEGG功能基因数目；</p>
<p>pvalue、p.adjust和qvalue，p值、校正后p值和q值信息；</p>
<p>geneID和Count，富集到该通路中的基因名称和数目。</p>
<p>注：本示例分析中使用的entrze id，故这里也显示的entrze id。如使用其它类型的基因名称，如symbol id等类型，可在结果中对entrze id和symbol id做个转换。</p>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-39_b73546663b708597937f30646ecfdce0">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>kegg_up <span class="ot">&lt;-</span> clusterProfiler<span class="sc">::</span><span class="fu">enrichKEGG</span>(<span class="at">gene =</span> gene_up ,</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">organism=</span>  <span class="st">'hsa'</span>,</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">keyType =</span> <span class="st">"kegg"</span>,</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pvalueCutoff  =</span>  <span class="fl">0.1</span>,</span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>,  <span class="co"># FDR多重假设检验校正 qvalue</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">universe=</span>  gene_all,</span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">minGSSize =</span> <span class="dv">10</span>,</span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">maxGSSize =</span> <span class="dv">500</span>,</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">qvalueCutoff  =</span>  <span class="fl">0.05</span> )</span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Reading KEGG annotation online: "https://rest.kegg.jp/link/hsa/pathway"...</span></span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Reading KEGG annotation online: "https://rest.kegg.jp/list/pathway/hsa"...</span></span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a>kegg_down <span class="ot">&lt;-</span> clusterProfiler<span class="sc">::</span><span class="fu">enrichKEGG</span>( <span class="at">gene =</span>  gene_down ,</span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">organism=</span>  <span class="st">'hsa'</span>,</span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">keyType =</span> <span class="st">"kegg"</span>,</span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pvalueCutoff  =</span>  <span class="fl">0.05</span>,</span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>,</span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">universe=</span>  gene_all,</span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">minGSSize =</span> <span class="dv">10</span>,</span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">maxGSSize =</span> <span class="dv">500</span>,</span>
<span id="cb41-20"><a href="#cb41-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">qvalueCutoff  =</span>  <span class="fl">0.05</span> )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-40_1102ac341cedfce5b93ef2fe7d79d2f4">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>df_up<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(kegg_up)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>df_down<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(kegg_down)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-41_2b6680304d708f8cd6832b30168ea91c">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>df_up <span class="ot">&lt;-</span> df_up[df_up<span class="sc">$</span>pvalue <span class="sc">&lt;</span> <span class="fl">0.05</span>, ]</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>df_down <span class="ot">&lt;-</span> df_down[df_down<span class="sc">$</span>pvalue <span class="sc">&lt;</span> <span class="fl">0.05</span>, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-42_dce49f7c83b0c6b761239aa59440cfce">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>df_up <span class="ot">&lt;-</span> df_up <span class="sc">|&gt;</span> </span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group=</span><span class="dv">1</span>)</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>df_down <span class="ot">&lt;-</span> df_down <span class="sc">|&gt;</span> </span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group=</span><span class="sc">-</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-43_7d9b37382a4701c7576640f314abe6ac">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">rbind</span>(df_up,df_down)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dat)</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                      category</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04960                   Organismal Systems</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04972                   Organismal Systems</span></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04080 Environmental Information Processing</span></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa05150                       Human Diseases</span></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04115                   Cellular Processes</span></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04976                   Organismal Systems</span></span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                  subcategory       ID</span></span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04960                    Excretory system hsa04960</span></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04972                    Digestive system hsa04972</span></span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04080 Signaling molecules and interaction hsa04080</span></span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa05150       Infectious disease: bacterial hsa05150</span></span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04115               Cell growth and death hsa04115</span></span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04976                    Digestive system hsa04976</span></span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                        Description GeneRatio BgRatio     pvalue</span></span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04960 Aldosterone-regulated sodium reabsorption       1/3 19/4319 0.01314256</span></span>
<span id="cb45-19"><a href="#cb45-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04972                      Pancreatic secretion       1/3 35/4319 0.02412024</span></span>
<span id="cb45-20"><a href="#cb45-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04080   Neuroactive ligand-receptor interaction       1/3 36/4319 0.02480363</span></span>
<span id="cb45-21"><a href="#cb45-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa05150           Staphylococcus aureus infection       1/3 37/4319 0.02548670</span></span>
<span id="cb45-22"><a href="#cb45-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04115                     p53 signaling pathway       1/3 50/4319 0.03433761</span></span>
<span id="cb45-23"><a href="#cb45-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04976                            Bile secretion       1/3 51/4319 0.03501622</span></span>
<span id="cb45-24"><a href="#cb45-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;            p.adjust      qvalue geneID Count group</span></span>
<span id="cb45-25"><a href="#cb45-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04960 0.04668829 0.006143196   2810     1     1</span></span>
<span id="cb45-26"><a href="#cb45-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04972 0.04668829 0.006143196   6344     1     1</span></span>
<span id="cb45-27"><a href="#cb45-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04080 0.04668829 0.006143196   6344     1     1</span></span>
<span id="cb45-28"><a href="#cb45-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa05150 0.04668829 0.006143196  25984     1     1</span></span>
<span id="cb45-29"><a href="#cb45-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04115 0.04668829 0.006143196   2810     1     1</span></span>
<span id="cb45-30"><a href="#cb45-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; hsa04976 0.04668829 0.006143196   6344     1     1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="水平条图" class="level3" data-number="2.1.1">
<h3 data-number="2.1.1" class="anchored" data-anchor-id="水平条图"><span class="header-section-number">2.1.1</span> 水平条图</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-44_2df329ad8fac0d444c34fbda38797171">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Negative_log10pvalue <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(dat<span class="sc">$</span>pvalue)</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Group_log10pvalue <span class="ot">=</span> dat<span class="sc">$</span>Negative_log10pvalue <span class="sc">*</span> dat<span class="sc">$</span>group</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">=</span> dat[<span class="fu">order</span>( dat<span class="sc">$</span>Group_log10pvalue, <span class="at">decreasing =</span> F ), ]</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>ggKEGG <span class="ot">&lt;-</span> <span class="fu">ggplot</span>( dat,<span class="fu">aes</span>(<span class="at">x =</span>  <span class="fu">reorder</span>(Description,</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">order</span>(Group_log10pvalue,<span class="at">decreasing=</span>F)),</span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">y =</span> Group_log10pvalue, <span class="at">fill =</span> group)) <span class="sc">+</span> </span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>( <span class="at">stat =</span> <span class="st">"identity"</span> ) <span class="sc">+</span> </span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradient</span>( <span class="at">low =</span> <span class="st">"blue"</span>, <span class="at">high =</span> <span class="st">"red"</span> ) <span class="sc">+</span> </span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>( <span class="at">name =</span> <span class="st">"Pathway names"</span> ) <span class="sc">+</span></span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>( <span class="at">name =</span> <span class="st">"log10 pvalue"</span> ) <span class="sc">+</span></span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>( <span class="at">plot.title =</span> <span class="fu">element_text</span>( <span class="at">hjust =</span> <span class="fl">0.5</span> ) ,</span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a>           <span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>( <span class="st">"KEGG Pathway Enrichment"</span> ) </span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a>ggKEGG</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TranscriptomicsDownstreamAnalysis_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="点图" class="level3" data-number="2.1.2">
<h3 data-number="2.1.2" class="anchored" data-anchor-id="点图"><span class="header-section-number">2.1.2</span> 点图</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-45_a9f52f8e0e127622790d058f0af212f8">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># GeneRatio 转化成数值型</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>GR <span class="ot">&lt;-</span> dat<span class="sc">$</span>GeneRatio <span class="sc">|&gt;</span> </span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_split</span>(<span class="at">pattern =</span> <span class="st">"/"</span>,<span class="at">simplify =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">as_tibble</span>()</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: The `x` argument of `as_tibble.matrix()` must have unique column names if</span></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `.name_repair` is omitted as of tibble 2.0.0.</span></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ Using compatibility `.name_repair`.</span></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>GeneRatio <span class="ot">&lt;-</span> <span class="fu">parse_number</span>(GR<span class="sc">$</span>V1)<span class="sc">/</span><span class="fu">parse_number</span>(GR<span class="sc">$</span>V2)</span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> dat[<span class="fu">order</span>(dat<span class="sc">$</span>GeneRatio,<span class="at">decreasing=</span>F),]</span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> dat,<span class="fu">aes</span>(<span class="at">x=</span>GeneRatio,</span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a>                      <span class="at">y=</span><span class="fu">reorder</span>(Description,<span class="fu">order</span>(GeneRatio,<span class="at">decreasing=</span>F)),</span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a>                      <span class="at">color =</span> Negative_log10pvalue,</span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a>                      <span class="at">size=</span>Count</span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a>                      )</span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a>       )<span class="sc">+</span></span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>( <span class="at">name =</span> <span class="st">"GeneRatio"</span> ,<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="fl">1.0</span>,<span class="fl">0.1</span>))<span class="sc">+</span></span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"blue"</span>, <span class="at">high =</span> <span class="st">"red"</span> )<span class="sc">+</span></span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>( <span class="at">name =</span> <span class="st">"Pathway names"</span> ) <span class="sc">+</span></span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>( <span class="at">plot.title =</span> <span class="fu">element_text</span>( <span class="at">hjust =</span> <span class="fl">0.5</span> )) <span class="sc">+</span></span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>( <span class="st">"KEGG Pathway Enrichment "</span> )<span class="sc">+</span></span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">color=</span><span class="st">"-log10 pvalue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TranscriptomicsDownstreamAnalysis_files/figure-html/unnamed-chunk-45-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="go富集分析" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="go富集分析"><span class="header-section-number">2.2</span> GO富集分析</h2>
<p>对于输出的GO富集结果表格中的各列内容：</p>
<p>ONTOLOGY，GO的BP（生物学过程）、CC（细胞组分）或MF（分子功能）；</p>
<p>ID和Description，富集到的GO term及其描述；</p>
<p>GeneRatio和BgRatio，分别为富集到该GO term中的基因数目/给定基因的总数目，以及该GO term中背景基因总数目/该物种所有已知GO功能基因数目；</p>
<p>pvalue、p.adjust和qvalue，p值、校正后p值和q值；</p>
<p>geneID和Count，富集到该GO term中的基因名称和数目。</p>
<p>注：以上示例分析中使用的entrze id，故输出的基因名称也显示的entrze id。如期望显示其它类型的基因名称，如symbol id等类型，除了更改为使用symbol id的基因名称做分析外，还可以通过基因名称转换的方式对entrze id和symbol id做个转换。</p>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-46_742a9a05ea09f5f54ea6d80a80f6410f">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>GO_EA<span class="ot">&lt;-</span> clusterProfiler<span class="sc">::</span><span class="fu">enrichGO</span>(<span class="at">gene =</span> gene_all ,</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">OrgDb =</span> <span class="st">"org.Hs.eg.db"</span>,</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">keyType =</span> <span class="st">"ENTREZID"</span>,</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">ont =</span> <span class="st">"ALL"</span>,</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">pvalueCutoff  =</span>  <span class="fl">0.05</span> ,</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">qvalueCutoff  =</span>  <span class="fl">0.05</span>,</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">readable =</span> F)</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>GO_EA_tbl <span class="ot">&lt;-</span>  <span class="fu">as_tibble</span>(GO_EA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="可视化" class="level3" data-number="2.2.1">
<h3 data-number="2.2.1" class="anchored" data-anchor-id="可视化"><span class="header-section-number">2.2.1</span> 可视化</h3>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-47_99de7b129b0bf0140196bc6268108ae0">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co">#clusterProfiler 包里的一些默认作图方法，例如</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(GO_EA)  <span class="co">#富集柱形图</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TranscriptomicsDownstreamAnalysis_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid" width="672"></p>
</div>
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(GO_EA)  <span class="co">#富集气泡图</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TranscriptomicsDownstreamAnalysis_files/figure-html/unnamed-chunk-47-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="TranscriptomicsDownstreamAnalysis_cache/html/unnamed-chunk-48_87ea747e88306a3cfa35cb8f2f8f1516">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">&lt;-</span> GO_EA_tbl[GO_EA_tbl<span class="sc">$</span>pvalue<span class="sc">&lt;</span> <span class="fl">0.05</span>, ]</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>dat2<span class="sc">$</span>Negative_log10pvalue <span class="ot">&lt;-</span>  <span class="sc">-</span><span class="fu">log10</span>(dat2<span class="sc">$</span>pvalue)</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">=</span> dat2[<span class="fu">order</span>( dat2<span class="sc">$</span>Negative_log10pvalue, <span class="at">decreasing =</span> F ), ]</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">&lt;-</span> dat2 <span class="sc">|&gt;</span> </span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(ONTOLOGY) <span class="sc">|&gt;</span> </span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_max</span>(<span class="at">order_by =</span> Negative_log10pvalue,<span class="at">n=</span><span class="dv">10</span>) <span class="sc">|&gt;</span> </span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>()</span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(dat2,<span class="fu">aes</span>(<span class="at">x =</span> Negative_log10pvalue, </span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span> <span class="fu">reorder</span>(Description,</span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">order</span>(ONTOLOGY,Negative_log10pvalue,<span class="at">decreasing=</span>F)),</span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">fill =</span> ONTOLOGY ))<span class="sc">+</span> </span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>( <span class="at">stat =</span> <span class="st">"identity"</span> ) <span class="sc">+</span></span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>( <span class="at">name =</span> <span class="st">"-log10 pvalue"</span> ) <span class="sc">+</span></span>
<span id="cb51-17"><a href="#cb51-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>( <span class="at">name =</span> <span class="st">"Pathway names"</span> ) <span class="sc">+</span></span>
<span id="cb51-18"><a href="#cb51-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb51-19"><a href="#cb51-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>( <span class="at">plot.title =</span> <span class="fu">element_text</span>( <span class="at">hjust =</span> <span class="fl">0.5</span> ) ) <span class="sc">+</span></span>
<span id="cb51-20"><a href="#cb51-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>( <span class="st">"GO Pathway Enrichment"</span> ) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TranscriptomicsDownstreamAnalysis_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="gsea" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="gsea"><span class="header-section-number">2.3</span> GSEA</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://img-blog.csdnimg.cn/20181105104509344.gif" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">GSEA</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="data/GSEA.gif" class="img-fluid figure-img"></p>
</figure>
</div>
<!-- -->

</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb52" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "转录组下游分析"</span></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a><span class="an">knitr:</span><span class="co"> </span></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a><span class="co">  opts_chunk: </span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a><span class="co">    comment: "#&gt;"</span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a><span class="co">    collapse: TRUE</span></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span><span class="co"> </span></span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a><span class="co">  message: false</span></span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a><span class="co">  cache: true</span></span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a><span class="co">  </span></span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a><span class="an">toc:</span><span class="co"> true</span></span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a><span class="an">number-sections:</span><span class="co"> true</span></span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a><span class="an">code-fold:</span><span class="co"> show</span></span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a><span class="an">code-link:</span><span class="co"> true</span></span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a><span class="an">code-tools:</span><span class="co"> true</span></span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span><span class="co"> html</span></span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-22"><a href="#cb52-22" aria-hidden="true" tabindex="-1"></a><span class="fu">## Assays</span></span>
<span id="cb52-23"><a href="#cb52-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-24"><a href="#cb52-24" aria-hidden="true" tabindex="-1"></a><span class="fu">### logCount→ mRNA Count</span></span>
<span id="cb52-25"><a href="#cb52-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-28"><a href="#cb52-28" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-29"><a href="#cb52-29" aria-hidden="true" tabindex="-1"></a><span class="co"># 读取计数矩阵 log2(count+1)</span></span>
<span id="cb52-30"><a href="#cb52-30" aria-hidden="true" tabindex="-1"></a>raw_logCount <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/TCGA-CHOL.htseq_counts.tsv"</span>)</span>
<span id="cb52-31"><a href="#cb52-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-32"><a href="#cb52-32" aria-hidden="true" tabindex="-1"></a><span class="co"># 60488 个gene ×  45 个sample</span></span>
<span id="cb52-33"><a href="#cb52-33" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-34"><a href="#cb52-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-37"><a href="#cb52-37" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-38"><a href="#cb52-38" aria-hidden="true" tabindex="-1"></a><span class="co"># 基因编码  BiocManager::install("rtracklayer")</span></span>
<span id="cb52-39"><a href="#cb52-39" aria-hidden="true" tabindex="-1"></a>gtf_v22 <span class="ot">&lt;-</span> rtracklayer<span class="sc">::</span><span class="fu">import</span>(<span class="st">'data/gencode.v22.annotation.gtf.gz'</span>) <span class="sc">|&gt;</span></span>
<span id="cb52-40"><a href="#cb52-40" aria-hidden="true" tabindex="-1"></a> <span class="fu">as_tibble</span>()</span>
<span id="cb52-41"><a href="#cb52-41" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(gtf_v22,<span class="at">file =</span> <span class="st">"data/gtf_v22.csv"</span>)</span>
<span id="cb52-42"><a href="#cb52-42" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-43"><a href="#cb52-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-46"><a href="#cb52-46" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-47"><a href="#cb52-47" aria-hidden="true" tabindex="-1"></a><span class="co"># 提取mRNA相关基因</span></span>
<span id="cb52-48"><a href="#cb52-48" aria-hidden="true" tabindex="-1"></a>gtf_mrna_v22 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gtf_v22,</span>
<span id="cb52-49"><a href="#cb52-49" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">c</span>(<span class="st">"gene_id"</span>,<span class="st">"gene_type"</span>, <span class="st">"gene_name"</span>))<span class="sc">|&gt;</span> </span>
<span id="cb52-50"><a href="#cb52-50" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(gene_type <span class="sc">==</span> <span class="st">"protein_coding"</span>) <span class="sc">|&gt;</span>  </span>
<span id="cb52-51"><a href="#cb52-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>()</span>
<span id="cb52-52"><a href="#cb52-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-53"><a href="#cb52-53" aria-hidden="true" tabindex="-1"></a>gtf_mrna_v22 </span>
<span id="cb52-54"><a href="#cb52-54" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(gtf_mrna_v22,<span class="at">file =</span> <span class="st">"data/gtf_mrna_v22.csv"</span>)</span>
<span id="cb52-55"><a href="#cb52-55" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-56"><a href="#cb52-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-59"><a href="#cb52-59" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-60"><a href="#cb52-60" aria-hidden="true" tabindex="-1"></a><span class="co"># 根据注释文件把Ensembl ID转换为gene ID</span></span>
<span id="cb52-61"><a href="#cb52-61" aria-hidden="true" tabindex="-1"></a>gene_logCount <span class="ot">&lt;-</span> raw_logCount <span class="sc">|&gt;</span> </span>
<span id="cb52-62"><a href="#cb52-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(gtf_mrna_v22, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Ensembl_ID"</span> <span class="ot">=</span> <span class="st">"gene_id"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb52-63"><a href="#cb52-63" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name, <span class="fu">starts_with</span>(<span class="st">"TCGA"</span>) )</span>
<span id="cb52-64"><a href="#cb52-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-65"><a href="#cb52-65" aria-hidden="true" tabindex="-1"></a>gene_logCount</span>
<span id="cb52-66"><a href="#cb52-66" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(gene_logCount,<span class="st">"data/gene_logCount.csv"</span>)</span>
<span id="cb52-67"><a href="#cb52-67" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-68"><a href="#cb52-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-69"><a href="#cb52-69" aria-hidden="true" tabindex="-1"></a>基因名是有重复的，其中有100个出现了2次，有1个出现了3次</span>
<span id="cb52-70"><a href="#cb52-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-73"><a href="#cb52-73" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-74"><a href="#cb52-74" aria-hidden="true" tabindex="-1"></a>gene_logCount <span class="sc">|&gt;</span> <span class="fu">summarise</span>(<span class="at">n=</span><span class="fu">n</span>(),<span class="at">.by =</span> gene_name) <span class="sc">|&gt;</span> </span>
<span id="cb52-75"><a href="#cb52-75" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(n<span class="sc">==</span><span class="dv">1</span>) </span>
<span id="cb52-76"><a href="#cb52-76" aria-hidden="true" tabindex="-1"></a>gene_logCount <span class="sc">|&gt;</span> <span class="fu">summarise</span>(<span class="at">n=</span><span class="fu">n</span>(),<span class="at">.by =</span> gene_name) <span class="sc">|&gt;</span> </span>
<span id="cb52-77"><a href="#cb52-77" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(n<span class="sc">==</span><span class="dv">2</span>) </span>
<span id="cb52-78"><a href="#cb52-78" aria-hidden="true" tabindex="-1"></a>gene_logCount <span class="sc">|&gt;</span> <span class="fu">summarise</span>(<span class="at">n=</span><span class="fu">n</span>(),<span class="at">.by =</span> gene_name) <span class="sc">|&gt;</span> </span>
<span id="cb52-79"><a href="#cb52-79" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(n<span class="sc">==</span><span class="dv">3</span>) </span>
<span id="cb52-80"><a href="#cb52-80" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-81"><a href="#cb52-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-82"><a href="#cb52-82" aria-hidden="true" tabindex="-1"></a>先去对数（ log2(Count+1））</span>
<span id="cb52-83"><a href="#cb52-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-86"><a href="#cb52-86" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-87"><a href="#cb52-87" aria-hidden="true" tabindex="-1"></a>gene_Count <span class="ot">&lt;-</span> gene_logCount <span class="sc">|&gt;</span> </span>
<span id="cb52-88"><a href="#cb52-88" aria-hidden="true" tabindex="-1"></a> <span class="fu">mutate_if</span>(is.numeric,<span class="sc">~</span><span class="dv">2</span><span class="sc">^</span>(.)<span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb52-89"><a href="#cb52-89" aria-hidden="true" tabindex="-1"></a>gene_Count</span>
<span id="cb52-90"><a href="#cb52-90" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-91"><a href="#cb52-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-92"><a href="#cb52-92" aria-hidden="true" tabindex="-1"></a>根据基因id分组，重复，则取平均值去重，去掉重复的102行</span>
<span id="cb52-93"><a href="#cb52-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-96"><a href="#cb52-96" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-97"><a href="#cb52-97" aria-hidden="true" tabindex="-1"></a>Counts <span class="ot">&lt;-</span> gene_Count <span class="sc">|&gt;</span> </span>
<span id="cb52-98"><a href="#cb52-98" aria-hidden="true" tabindex="-1"></a> <span class="fu">group_by</span>(gene_name) <span class="sc">|&gt;</span> </span>
<span id="cb52-99"><a href="#cb52-99" aria-hidden="true" tabindex="-1"></a> <span class="fu">summarise_all</span>(mean,<span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb52-100"><a href="#cb52-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-101"><a href="#cb52-101" aria-hidden="true" tabindex="-1"></a>Counts <span class="sc">|&gt;</span> <span class="fu">summarise</span>(<span class="at">n=</span><span class="fu">n</span>(),<span class="at">.by =</span> gene_name) <span class="sc">|&gt;</span> </span>
<span id="cb52-102"><a href="#cb52-102" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(n<span class="sc">==</span><span class="dv">1</span>) </span>
<span id="cb52-103"><a href="#cb52-103" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-104"><a href="#cb52-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-105"><a href="#cb52-105" aria-hidden="true" tabindex="-1"></a>提取lncRNA、miRNA等也是类似的操作</span>
<span id="cb52-106"><a href="#cb52-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-107"><a href="#cb52-107" aria-hidden="true" tabindex="-1"></a>::: {#summarise_all .callout-note}</span>
<span id="cb52-110"><a href="#cb52-110" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-111"><a href="#cb52-111" aria-hidden="true" tabindex="-1"></a><span class="fu">tibble</span>(</span>
<span id="cb52-112"><a href="#cb52-112" aria-hidden="true" tabindex="-1"></a>    <span class="at">x=</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>),</span>
<span id="cb52-113"><a href="#cb52-113" aria-hidden="true" tabindex="-1"></a>    <span class="at">y=</span><span class="fu">c</span>(<span class="dv">3</span><span class="sc">:</span><span class="dv">7</span>),</span>
<span id="cb52-114"><a href="#cb52-114" aria-hidden="true" tabindex="-1"></a>    <span class="at">g=</span><span class="fu">factor</span>(<span class="fu">c</span>(<span class="st">'A'</span>,<span class="st">'B'</span>,<span class="st">'A'</span>,<span class="st">'C'</span>,<span class="st">'C'</span>))</span>
<span id="cb52-115"><a href="#cb52-115" aria-hidden="true" tabindex="-1"></a>)<span class="sc">|&gt;</span> </span>
<span id="cb52-116"><a href="#cb52-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(g) <span class="sc">|&gt;</span> <span class="fu">summarise_all</span>(<span class="fu">list</span>(<span class="st">"均数"</span><span class="ot">=</span>mean, <span class="st">"中位数"</span><span class="ot">=</span>median))</span>
<span id="cb52-117"><a href="#cb52-117" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-118"><a href="#cb52-118" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb52-119"><a href="#cb52-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-120"><a href="#cb52-120" aria-hidden="true" tabindex="-1"></a><span class="fu">## Sample metadata</span></span>
<span id="cb52-121"><a href="#cb52-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-124"><a href="#cb52-124" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-125"><a href="#cb52-125" aria-hidden="true" tabindex="-1"></a><span class="co">#首先载入下载好胆管癌的样本信息文件</span></span>
<span id="cb52-126"><a href="#cb52-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-127"><a href="#cb52-127" aria-hidden="true" tabindex="-1"></a>phenotype_file <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">'data/TCGA-CHOL.GDC_phenotype.tsv'</span>)</span>
<span id="cb52-128"><a href="#cb52-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-129"><a href="#cb52-129" aria-hidden="true" tabindex="-1"></a>phenotype_file[<span class="dv">1</span>,]</span>
<span id="cb52-130"><a href="#cb52-130" aria-hidden="true" tabindex="-1"></a>phenotypeID <span class="ot">&lt;-</span> phenotype_file<span class="sc">$</span>submitter_id.samples</span>
<span id="cb52-131"><a href="#cb52-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-132"><a href="#cb52-132" aria-hidden="true" tabindex="-1"></a><span class="co"># 71 个sample × 122 个样本变量</span></span>
<span id="cb52-133"><a href="#cb52-133" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-134"><a href="#cb52-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-137"><a href="#cb52-137" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-138"><a href="#cb52-138" aria-hidden="true" tabindex="-1"></a><span class="co"># 选择样本</span></span>
<span id="cb52-139"><a href="#cb52-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-140"><a href="#cb52-140" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(phenotype_file<span class="sc">$</span>neoplasm_histologic_grade)<span class="co"># 组织学分级</span></span>
<span id="cb52-141"><a href="#cb52-141" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(phenotype_file<span class="sc">$</span>tumor_stage.diagnoses) <span class="co">#分期</span></span>
<span id="cb52-142"><a href="#cb52-142" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-143"><a href="#cb52-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-146"><a href="#cb52-146" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-147"><a href="#cb52-147" aria-hidden="true" tabindex="-1"></a><span class="do">## 找到所需列变量的索引，最好用正则表达式</span></span>
<span id="cb52-148"><a href="#cb52-148" aria-hidden="true" tabindex="-1"></a>col_index <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"(neoplasm_histologic_grade)|(tumor_stage.diagnoses)"</span>,<span class="fu">colnames</span>(phenotype_file))</span>
<span id="cb52-149"><a href="#cb52-149" aria-hidden="true" tabindex="-1"></a>col_index</span>
<span id="cb52-150"><a href="#cb52-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-151"><a href="#cb52-151" aria-hidden="true" tabindex="-1"></a><span class="co"># 或者</span></span>
<span id="cb52-152"><a href="#cb52-152" aria-hidden="true" tabindex="-1"></a>col_index_logical <span class="ot">&lt;-</span> <span class="fu">str_detect</span>(<span class="fu">colnames</span>(phenotype_file),<span class="st">"(neoplasm_histologic_grade)|(tumor_stage.diagnoses)"</span>)</span>
<span id="cb52-153"><a href="#cb52-153" aria-hidden="true" tabindex="-1"></a>col_index_logical</span>
<span id="cb52-154"><a href="#cb52-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-155"><a href="#cb52-155" aria-hidden="true" tabindex="-1"></a>col_index <span class="ot">&lt;-</span> <span class="fu">which</span>(col_index_logical<span class="sc">==</span><span class="cn">TRUE</span>)</span>
<span id="cb52-156"><a href="#cb52-156" aria-hidden="true" tabindex="-1"></a>col_index</span>
<span id="cb52-157"><a href="#cb52-157" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-158"><a href="#cb52-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-161"><a href="#cb52-161" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-162"><a href="#cb52-162" aria-hidden="true" tabindex="-1"></a><span class="co"># 取出需要的列的样本信息</span></span>
<span id="cb52-163"><a href="#cb52-163" aria-hidden="true" tabindex="-1"></a>hs <span class="ot">&lt;-</span> phenotype_file[,col_index[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]]</span>
<span id="cb52-164"><a href="#cb52-164" aria-hidden="true" tabindex="-1"></a>hs <span class="ot">&lt;-</span> <span class="fu">cbind</span>(phenotypeID,hs)</span>
<span id="cb52-165"><a href="#cb52-165" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hs)</span>
<span id="cb52-166"><a href="#cb52-166" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-167"><a href="#cb52-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-170"><a href="#cb52-170" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-171"><a href="#cb52-171" aria-hidden="true" tabindex="-1"></a><span class="co"># 挑选组织学为G2或G3和分期为i或ii期的样本，数量多</span></span>
<span id="cb52-172"><a href="#cb52-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-173"><a href="#cb52-173" aria-hidden="true" tabindex="-1"></a>sample <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(hs,neoplasm_histologic_grade<span class="sc">==</span><span class="st">"G2"</span><span class="sc">|</span>neoplasm_histologic_grade<span class="sc">==</span><span class="st">"G3"</span> ) <span class="sc">|&gt;</span> </span>
<span id="cb52-174"><a href="#cb52-174" aria-hidden="true" tabindex="-1"></a> dplyr<span class="sc">::</span><span class="fu">filter</span>(tumor_stage.diagnoses<span class="sc">==</span><span class="st">"stage i"</span><span class="sc">|</span>tumor_stage.diagnoses<span class="sc">==</span><span class="st">"stage ii"</span>)</span>
<span id="cb52-175"><a href="#cb52-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-176"><a href="#cb52-176" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sample)</span>
<span id="cb52-177"><a href="#cb52-177" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-178"><a href="#cb52-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-179"><a href="#cb52-179" aria-hidden="true" tabindex="-1"></a>到这里，就从71个样本中挑出了45个样本，但是，我们只要这45个样本中成对样本的表达数据，即同一个病人既有癌旁~~正常~~细胞的表达蛋白，又有肿瘤细胞的异常表达蛋白。</span>
<span id="cb52-180"><a href="#cb52-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-183"><a href="#cb52-183" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-184"><a href="#cb52-184" aria-hidden="true" tabindex="-1"></a><span class="co"># 前缀1~12字符</span></span>
<span id="cb52-185"><a href="#cb52-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-186"><a href="#cb52-186" aria-hidden="true" tabindex="-1"></a>sample <span class="ot">&lt;-</span> sample <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">patient_id =</span> <span class="fu">str_sub</span>(sample[,<span class="dv">1</span>],<span class="dv">1</span>,<span class="dv">12</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb52-187"><a href="#cb52-187" aria-hidden="true" tabindex="-1"></a> <span class="fu">group_by</span>(patient_id) <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">|&gt;</span> <span class="fu">ungroup</span>()</span>
<span id="cb52-188"><a href="#cb52-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-189"><a href="#cb52-189" aria-hidden="true" tabindex="-1"></a>sample <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(sample,n<span class="sc">==</span><span class="dv">2</span>) <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>n)</span>
<span id="cb52-190"><a href="#cb52-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-191"><a href="#cb52-191" aria-hidden="true" tabindex="-1"></a>sample <span class="co"># 筛选出 15对 成对数据</span></span>
<span id="cb52-192"><a href="#cb52-192" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-193"><a href="#cb52-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-194"><a href="#cb52-194" aria-hidden="true" tabindex="-1"></a>并且要匹配到<span class="in">`Counts`</span> 中出现的45个样本中的成对样本</span>
<span id="cb52-195"><a href="#cb52-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-198"><a href="#cb52-198" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-199"><a href="#cb52-199" aria-hidden="true" tabindex="-1"></a><span class="co"># 表达矩阵  </span></span>
<span id="cb52-200"><a href="#cb52-200" aria-hidden="true" tabindex="-1"></a>patient_of_counts <span class="ot">&lt;-</span> <span class="fu">str_sub</span>(<span class="fu">colnames</span>(Counts),<span class="dv">1</span>,<span class="dv">12</span>)</span>
<span id="cb52-201"><a href="#cb52-201" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(patient_of_counts)<span class="sc">==</span><span class="dv">2</span></span>
<span id="cb52-202"><a href="#cb52-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-203"><a href="#cb52-203" aria-hidden="true" tabindex="-1"></a>paired_patient <span class="ot">&lt;-</span> <span class="fu">table</span>(patient_of_counts)[<span class="fu">table</span>(patient_of_counts)<span class="sc">==</span><span class="dv">2</span>]<span class="sc">|&gt;</span> </span>
<span id="cb52-204"><a href="#cb52-204" aria-hidden="true" tabindex="-1"></a> <span class="fu">names</span>()  <span class="co"># 筛选出9对成对数据</span></span>
<span id="cb52-205"><a href="#cb52-205" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-206"><a href="#cb52-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-207"><a href="#cb52-207" aria-hidden="true" tabindex="-1"></a>取<span class="in">`sample`</span>和<span class="in">`paired_patient`</span>的交集，8对成对数据</span>
<span id="cb52-208"><a href="#cb52-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-211"><a href="#cb52-211" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-212"><a href="#cb52-212" aria-hidden="true" tabindex="-1"></a><span class="co"># 取交集</span></span>
<span id="cb52-213"><a href="#cb52-213" aria-hidden="true" tabindex="-1"></a>final_patient <span class="ot">&lt;-</span> lubridate<span class="sc">::</span><span class="fu">intersect</span>(sample<span class="sc">$</span>patient_id,paired_patient)</span>
<span id="cb52-214"><a href="#cb52-214" aria-hidden="true" tabindex="-1"></a>final_patient</span>
<span id="cb52-215"><a href="#cb52-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-216"><a href="#cb52-216" aria-hidden="true" tabindex="-1"></a>sample <span class="ot">&lt;-</span> sample <span class="sc">|&gt;</span> </span>
<span id="cb52-217"><a href="#cb52-217" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample<span class="sc">$</span>patient_id <span class="sc">%in%</span> final_patient) <span class="sc">|&gt;</span> </span>
<span id="cb52-218"><a href="#cb52-218" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>patient_id)<span class="sc">|&gt;</span></span>
<span id="cb52-219"><a href="#cb52-219" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(phenotypeID)<span class="sc">|&gt;</span> </span>
<span id="cb52-220"><a href="#cb52-220" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"grade"</span><span class="ot">=</span><span class="st">"neoplasm_histologic_grade"</span>,</span>
<span id="cb52-221"><a href="#cb52-221" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"stage"</span><span class="ot">=</span><span class="st">"tumor_stage.diagnoses"</span>)</span>
<span id="cb52-222"><a href="#cb52-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-223"><a href="#cb52-223" aria-hidden="true" tabindex="-1"></a>sample</span>
<span id="cb52-224"><a href="#cb52-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-225"><a href="#cb52-225" aria-hidden="true" tabindex="-1"></a>exprCounts <span class="ot">&lt;-</span><span class="fu">cbind</span>(Counts[,<span class="dv">1</span>], Counts[,patient_of_counts <span class="sc">%in%</span> final_patient])</span>
<span id="cb52-226"><a href="#cb52-226" aria-hidden="true" tabindex="-1"></a>exprCounts <span class="ot">&lt;-</span>exprCounts[,<span class="fu">order</span>(<span class="fu">colnames</span>(exprCounts))]</span>
<span id="cb52-227"><a href="#cb52-227" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(exprCounts)</span>
<span id="cb52-228"><a href="#cb52-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-229"><a href="#cb52-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-230"><a href="#cb52-230" aria-hidden="true" tabindex="-1"></a><span class="co"># 去除所有样本都不表达的基因</span></span>
<span id="cb52-231"><a href="#cb52-231" aria-hidden="true" tabindex="-1"></a>exprCounts <span class="ot">&lt;-</span> exprCounts <span class="sc">%&gt;%</span></span>
<span id="cb52-232"><a href="#cb52-232" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">rowsum =</span> <span class="fu">apply</span>(exprCounts[,<span class="sc">-</span><span class="dv">1</span>],<span class="dv">1</span>,sum)) <span class="sc">%&gt;%</span></span>
<span id="cb52-233"><a href="#cb52-233" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(rowsum<span class="sc">!=</span><span class="dv">0</span>)</span>
<span id="cb52-234"><a href="#cb52-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-235"><a href="#cb52-235" aria-hidden="true" tabindex="-1"></a>exprCounts <span class="ot">&lt;-</span> exprCounts <span class="sc">|&gt;</span> </span>
<span id="cb52-236"><a href="#cb52-236" aria-hidden="true" tabindex="-1"></a>    <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"gene_name"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb52-237"><a href="#cb52-237" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>rowsum)</span>
<span id="cb52-238"><a href="#cb52-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-239"><a href="#cb52-239" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-240"><a href="#cb52-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-243"><a href="#cb52-243" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-244"><a href="#cb52-244" aria-hidden="true" tabindex="-1"></a><span class="co"># 检查计数矩阵和列数据，看看它们在样本顺序方面是否一致。</span></span>
<span id="cb52-245"><a href="#cb52-245" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(exprCounts)</span>
<span id="cb52-246"><a href="#cb52-246" aria-hidden="true" tabindex="-1"></a>sample<span class="sc">$</span>phenotypeID</span>
<span id="cb52-247"><a href="#cb52-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-248"><a href="#cb52-248" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(sample<span class="sc">$</span>phenotypeID <span class="sc">%in%</span> <span class="fu">colnames</span>(exprCounts))</span>
<span id="cb52-249"><a href="#cb52-249" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(sample<span class="sc">$</span>phenotypeID <span class="sc">==</span> <span class="fu">colnames</span>(exprCounts))</span>
<span id="cb52-250"><a href="#cb52-250" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-251"><a href="#cb52-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-252"><a href="#cb52-252" aria-hidden="true" tabindex="-1"></a><span class="fu">## 建立DEG对象</span></span>
<span id="cb52-253"><a href="#cb52-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-256"><a href="#cb52-256" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-257"><a href="#cb52-257" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(exprCounts)</span>
<span id="cb52-258"><a href="#cb52-258" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(sample)</span>
<span id="cb52-259"><a href="#cb52-259" aria-hidden="true" tabindex="-1"></a>exprCounts<span class="ot">=</span><span class="fu">ceiling</span>(exprCounts)</span>
<span id="cb52-260"><a href="#cb52-260" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-261"><a href="#cb52-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-262"><a href="#cb52-262" aria-hidden="true" tabindex="-1"></a>根据TCGA样本的命名可以区分正常组织和肿瘤样本的测序结果 其中编号01-09表示肿瘤，10-19表示正常</span>
<span id="cb52-263"><a href="#cb52-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-266"><a href="#cb52-266" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-267"><a href="#cb52-267" aria-hidden="true" tabindex="-1"></a><span class="co"># 字符14~15，</span></span>
<span id="cb52-268"><a href="#cb52-268" aria-hidden="true" tabindex="-1"></a>sample<span class="sc">$</span>condition <span class="ot">&lt;-</span> <span class="fu">factor</span>(</span>
<span id="cb52-269"><a href="#cb52-269" aria-hidden="true" tabindex="-1"></a> <span class="fu">ifelse</span>(<span class="fu">as.numeric</span>(<span class="fu">str_sub</span>(sample<span class="sc">$</span>phenotypeID,<span class="dv">14</span>,<span class="dv">15</span>)) <span class="sc">&lt;</span> <span class="dv">10</span>,<span class="st">'tumor'</span>,<span class="st">'normal'</span>),</span>
<span id="cb52-270"><a href="#cb52-270" aria-hidden="true" tabindex="-1"></a> <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"normal"</span>,<span class="st">"tumor"</span>)</span>
<span id="cb52-271"><a href="#cb52-271" aria-hidden="true" tabindex="-1"></a> ) </span>
<span id="cb52-272"><a href="#cb52-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-273"><a href="#cb52-273" aria-hidden="true" tabindex="-1"></a>sample <span class="ot">&lt;-</span> <span class="fu">column_to_rownames</span>(sample,<span class="at">var =</span><span class="st">"phenotypeID"</span> )</span>
<span id="cb52-274"><a href="#cb52-274" aria-hidden="true" tabindex="-1"></a>sample</span>
<span id="cb52-275"><a href="#cb52-275" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-276"><a href="#cb52-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-279"><a href="#cb52-279" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-280"><a href="#cb52-280" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb52-281"><a href="#cb52-281" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(GenomicRanges<span class="sc">::</span>setdiff)</span>
<span id="cb52-282"><a href="#cb52-282" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> exprCounts,</span>
<span id="cb52-283"><a href="#cb52-283" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> sample,</span>
<span id="cb52-284"><a href="#cb52-284" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> condition)</span>
<span id="cb52-285"><a href="#cb52-285" aria-hidden="true" tabindex="-1"></a>dds</span>
<span id="cb52-286"><a href="#cb52-286" aria-hidden="true" tabindex="-1"></a><span class="fu">counts</span>(dds)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,]</span>
<span id="cb52-287"><a href="#cb52-287" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-288"><a href="#cb52-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-289"><a href="#cb52-289" aria-hidden="true" tabindex="-1"></a><span class="fu">### 预过滤</span></span>
<span id="cb52-290"><a href="#cb52-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-293"><a href="#cb52-293" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-294"><a href="#cb52-294" aria-hidden="true" tabindex="-1"></a>smallestGroupSize <span class="ot">&lt;-</span> <span class="dv">8</span></span>
<span id="cb52-295"><a href="#cb52-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-296"><a href="#cb52-296" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> smallestGroupSize</span>
<span id="cb52-297"><a href="#cb52-297" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[keep,]</span>
<span id="cb52-298"><a href="#cb52-298" aria-hidden="true" tabindex="-1"></a>dds</span>
<span id="cb52-299"><a href="#cb52-299" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">$</span>condition</span>
<span id="cb52-300"><a href="#cb52-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-301"><a href="#cb52-301" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">@</span>colData</span>
<span id="cb52-302"><a href="#cb52-302" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">@</span>assays</span>
<span id="cb52-303"><a href="#cb52-303" aria-hidden="true" tabindex="-1"></a><span class="fu">counts</span>(dds)[<span class="dv">1</span>,]</span>
<span id="cb52-304"><a href="#cb52-304" aria-hidden="true" tabindex="-1"></a>pre_filtered_exprCounts <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds)  <span class="co"># 14815×16</span></span>
<span id="cb52-305"><a href="#cb52-305" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-306"><a href="#cb52-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-307"><a href="#cb52-307" aria-hidden="true" tabindex="-1"></a><span class="fu"># 差异表达基因分析</span></span>
<span id="cb52-308"><a href="#cb52-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-311"><a href="#cb52-311" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-312"><a href="#cb52-312" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb52-313"><a href="#cb52-313" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(<span class="at">object =</span> dds,</span>
<span id="cb52-314"><a href="#cb52-314" aria-hidden="true" tabindex="-1"></a>               <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"condition"</span>,<span class="st">"tumor"</span>,<span class="st">"normal"</span>),</span>
<span id="cb52-315"><a href="#cb52-315" aria-hidden="true" tabindex="-1"></a>               <span class="at">filter =</span> <span class="fu">rowMeans</span>(<span class="fu">counts</span>(dds, <span class="at">normalized=</span><span class="cn">TRUE</span>)),</span>
<span id="cb52-316"><a href="#cb52-316" aria-hidden="true" tabindex="-1"></a><span class="co">#theta = 0.99,</span></span>
<span id="cb52-317"><a href="#cb52-317" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb52-318"><a href="#cb52-318" aria-hidden="true" tabindex="-1"></a><span class="fu">metadata</span>(res)[<span class="st">"alpha"</span>]</span>
<span id="cb52-319"><a href="#cb52-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-320"><a href="#cb52-320" aria-hidden="true" tabindex="-1"></a>res</span>
<span id="cb52-321"><a href="#cb52-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-322"><a href="#cb52-322" aria-hidden="true" tabindex="-1"></a><span class="co"># 根据padj 升序</span></span>
<span id="cb52-323"><a href="#cb52-323" aria-hidden="true" tabindex="-1"></a>resOrdered <span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj),]</span>
<span id="cb52-324"><a href="#cb52-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-325"><a href="#cb52-325" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resOrdered)</span>
<span id="cb52-326"><a href="#cb52-326" aria-hidden="true" tabindex="-1"></a><span class="co"># 导出到csv</span></span>
<span id="cb52-327"><a href="#cb52-327" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(res, <span class="at">file=</span><span class="st">"data/resOrdered.csv"</span>)</span>
<span id="cb52-328"><a href="#cb52-328" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-329"><a href="#cb52-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-330"><a href="#cb52-330" aria-hidden="true" tabindex="-1"></a><span class="fu">### 结果分析</span></span>
<span id="cb52-331"><a href="#cb52-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-334"><a href="#cb52-334" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-335"><a href="#cb52-335" aria-hidden="true" tabindex="-1"></a><span class="fu">mcols</span>(resOrdered)<span class="sc">$</span>description</span>
<span id="cb52-336"><a href="#cb52-336" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-337"><a href="#cb52-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-338"><a href="#cb52-338" aria-hidden="true" tabindex="-1"></a>pvalue，是统计学检验变量，代表差异显著性，一般认为P <span class="sc">\&lt;</span> 0.05 为显著， P <span class="sc">\&lt;</span>0.01 为非常显著。其含义为：由抽样误差导致样本间差异的概率小于0.05 或0.01。</span>
<span id="cb52-339"><a href="#cb52-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-340"><a href="#cb52-340" aria-hidden="true" tabindex="-1"></a>padj，是对pvalue进行多重假设检验校正后的结果。转录组测序的差异表达分析是对大量的基因表达值进行的独立统计假设检验，存在假阳性（Family-Wise Error Rate，FWER），Bonferroni矫正</span>
<span id="cb52-341"><a href="#cb52-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-342"><a href="#cb52-342" aria-hidden="true" tabindex="-1"></a>qvalue，False Discovery Rate，FDR，所有假阳性结果占所有拒绝零假设结果的比例，假阳性结果的比例不超过预设的FDR水平（0.05/0.1）,基于pvalue的分布。多重假设检验校正方法：Benjamini-Hochberg，</span>
<span id="cb52-343"><a href="#cb52-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-344"><a href="#cb52-344" aria-hidden="true" tabindex="-1"></a>log2FoldChange：对数倍数变化，1.5倍差异即0.585，2倍差异即1。</span>
<span id="cb52-345"><a href="#cb52-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-346"><a href="#cb52-346" aria-hidden="true" tabindex="-1"></a><span class="al">![](data/FoldChange.jpg)</span>{fig-align="center"}</span>
<span id="cb52-347"><a href="#cb52-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-350"><a href="#cb52-350" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-351"><a href="#cb52-351" aria-hidden="true" tabindex="-1"></a>DEG <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data/resOrdered.csv"</span>)</span>
<span id="cb52-352"><a href="#cb52-352" aria-hidden="true" tabindex="-1"></a>DESeq2_DEG <span class="ot">=</span> <span class="fu">na.omit</span>(DEG) </span>
<span id="cb52-353"><a href="#cb52-353" aria-hidden="true" tabindex="-1"></a>nrDEG<span class="ot">=</span>DESeq2_DEG</span>
<span id="cb52-354"><a href="#cb52-354" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-355"><a href="#cb52-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-356"><a href="#cb52-356" aria-hidden="true" tabindex="-1"></a><span class="fu">### 热图</span></span>
<span id="cb52-357"><a href="#cb52-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-360"><a href="#cb52-360" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-361"><a href="#cb52-361" aria-hidden="true" tabindex="-1"></a>pre_filtered_exprCounts[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb52-362"><a href="#cb52-362" aria-hidden="true" tabindex="-1"></a>log2exprCounts <span class="ot">&lt;-</span> <span class="fu">log2</span>(pre_filtered_exprCounts<span class="sc">+</span><span class="dv">1</span>)</span>
<span id="cb52-363"><a href="#cb52-363" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb52-364"><a href="#cb52-364" aria-hidden="true" tabindex="-1"></a>DESeq2_DEG <span class="ot">&lt;-</span> DESeq2_DEG <span class="sc">|&gt;</span> <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"...1"</span>)</span>
<span id="cb52-365"><a href="#cb52-365" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(log2exprCounts[<span class="fu">rownames</span>(<span class="fu">head</span>(DESeq2_DEG,<span class="at">n=</span><span class="dv">20</span>)) , ])</span>
<span id="cb52-366"><a href="#cb52-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-367"><a href="#cb52-367" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-368"><a href="#cb52-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-369"><a href="#cb52-369" aria-hidden="true" tabindex="-1"></a><span class="fu">### 挑选指定第一个基因看它在同一个病人的配对表达情况</span></span>
<span id="cb52-370"><a href="#cb52-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-373"><a href="#cb52-373" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-374"><a href="#cb52-374" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb52-375"><a href="#cb52-375" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(GenomicRanges<span class="sc">::</span>setdiff)</span>
<span id="cb52-376"><a href="#cb52-376" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-377"><a href="#cb52-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-380"><a href="#cb52-380" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-381"><a href="#cb52-381" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb52-382"><a href="#cb52-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-383"><a href="#cb52-383" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">g=</span>sample<span class="sc">$</span>condition,</span>
<span id="cb52-384"><a href="#cb52-384" aria-hidden="true" tabindex="-1"></a>                 <span class="at">p=</span><span class="fu">colnames</span>(exprCounts),</span>
<span id="cb52-385"><a href="#cb52-385" aria-hidden="true" tabindex="-1"></a>                 <span class="at">v=</span><span class="fu">as.numeric</span>(exprCounts[<span class="fu">rownames</span>(DESeq2_DEG[<span class="dv">1</span>,]),])) </span>
<span id="cb52-386"><a href="#cb52-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-387"><a href="#cb52-387" aria-hidden="true" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggpaired</span>(df, <span class="at">x =</span> <span class="st">"g"</span>, <span class="at">y =</span> <span class="st">"v"</span>,<span class="at">color =</span> <span class="st">"g"</span>, </span>
<span id="cb52-388"><a href="#cb52-388" aria-hidden="true" tabindex="-1"></a>                 <span class="at">line.color =</span> <span class="st">"gray"</span>, <span class="at">line.size =</span> <span class="fl">0.4</span>,<span class="at">palette =</span> <span class="st">"npg"</span>)<span class="sc">+</span></span>
<span id="cb52-389"><a href="#cb52-389" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_compare_means</span>()</span>
<span id="cb52-390"><a href="#cb52-390" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-391"><a href="#cb52-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-392"><a href="#cb52-392" aria-hidden="true" tabindex="-1"></a><span class="fu">### 火山图</span></span>
<span id="cb52-393"><a href="#cb52-393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-396"><a href="#cb52-396" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-397"><a href="#cb52-397" aria-hidden="true" tabindex="-1"></a>nrDEG<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(</span>
<span id="cb52-398"><a href="#cb52-398" aria-hidden="true" tabindex="-1"></a>    <span class="fu">if_else</span>(nrDEG<span class="sc">$</span>padj<span class="sc">&lt;</span><span class="fl">0.2</span><span class="sc">&amp;</span> <span class="fu">abs</span>(nrDEG<span class="sc">$</span>log2FoldChange)<span class="sc">&gt;=</span><span class="fl">0.585</span>,</span>
<span id="cb52-399"><a href="#cb52-399" aria-hidden="true" tabindex="-1"></a>            <span class="fu">if_else</span>(nrDEG<span class="sc">$</span>log2FoldChange<span class="sc">&gt;=</span><span class="fl">0.585</span>,<span class="st">"up"</span>,<span class="st">"down"</span>),</span>
<span id="cb52-400"><a href="#cb52-400" aria-hidden="true" tabindex="-1"></a>            <span class="st">"normal"</span>,),</span>
<span id="cb52-401"><a href="#cb52-401" aria-hidden="true" tabindex="-1"></a>    <span class="at">levels =</span><span class="fu">c</span>(<span class="st">"up"</span>,<span class="st">"down"</span>,<span class="st">"normal"</span>) )</span>
<span id="cb52-402"><a href="#cb52-402" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(nrDEG<span class="sc">$</span>group)</span>
<span id="cb52-403"><a href="#cb52-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-404"><a href="#cb52-404" aria-hidden="true" tabindex="-1"></a>gene_labels <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(nrDEG,nrDEG<span class="sc">$</span>padj<span class="sc">&lt;</span><span class="fl">0.2</span> <span class="sc">&amp;</span> </span>
<span id="cb52-405"><a href="#cb52-405" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">abs</span>(nrDEG<span class="sc">$</span>log2FoldChange)<span class="sc">&gt;=</span><span class="fl">0.585</span>)<span class="sc">|&gt;</span> </span>
<span id="cb52-406"><a href="#cb52-406" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n=</span><span class="dv">10</span>) <span class="sc">|&gt;</span> <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"...1"</span>)</span>
<span id="cb52-407"><a href="#cb52-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-408"><a href="#cb52-408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-409"><a href="#cb52-409" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(nrDEG,<span class="fu">aes</span>( <span class="at">x =</span> log2FoldChange,<span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj),<span class="at">color=</span>group,<span class="at">shape=</span>group))<span class="sc">+</span></span>
<span id="cb52-410"><a href="#cb52-410" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>,<span class="at">size=</span><span class="dv">2</span>)<span class="sc">+</span></span>
<span id="cb52-411"><a href="#cb52-411" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"red"</span>,<span class="st">"green"</span>,<span class="st">"gray"</span>))<span class="sc">+</span></span>
<span id="cb52-412"><a href="#cb52-412" aria-hidden="true" tabindex="-1"></a>    <span class="co">#scale_shape_manual(values = c(2,25,4))+</span></span>
<span id="cb52-413"><a href="#cb52-413" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.585</span>,<span class="dv">0</span>,<span class="fl">0.585</span>),<span class="at">lty=</span><span class="dv">3</span>,<span class="at">color=</span><span class="st">"grey25"</span>,<span class="at">lwd=</span><span class="fl">0.8</span>)<span class="sc">+</span></span>
<span id="cb52-414"><a href="#cb52-414" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.2</span>),<span class="at">lty=</span><span class="dv">3</span>,<span class="at">color=</span><span class="st">"grey25"</span>,<span class="at">lwd=</span><span class="fl">0.8</span>)<span class="sc">+</span></span>
<span id="cb52-415"><a href="#cb52-415" aria-hidden="true" tabindex="-1"></a>    ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(</span>
<span id="cb52-416"><a href="#cb52-416" aria-hidden="true" tabindex="-1"></a>        <span class="at">data =</span> gene_labels,</span>
<span id="cb52-417"><a href="#cb52-417" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">label=</span><span class="fu">row.names</span>(gene_labels)),</span>
<span id="cb52-418"><a href="#cb52-418" aria-hidden="true" tabindex="-1"></a>        <span class="at">color=</span><span class="st">"black"</span>,</span>
<span id="cb52-419"><a href="#cb52-419" aria-hidden="true" tabindex="-1"></a>        <span class="at">size=</span><span class="dv">2</span>)<span class="sc">+</span></span>
<span id="cb52-420"><a href="#cb52-420" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb52-421"><a href="#cb52-421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-422"><a href="#cb52-422" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-423"><a href="#cb52-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-424"><a href="#cb52-424" aria-hidden="true" tabindex="-1"></a><span class="fu">### LFC收缩和pagj</span></span>
<span id="cb52-425"><a href="#cb52-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-428"><a href="#cb52-428" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-429"><a href="#cb52-429" aria-hidden="true" tabindex="-1"></a><span class="fu">resultsNames</span>(dds)</span>
<span id="cb52-430"><a href="#cb52-430" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("apeglm")  自适应 t 先验收缩估计器  默认</span></span>
<span id="cb52-431"><a href="#cb52-431" aria-hidden="true" tabindex="-1"></a>resLFC <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds, <span class="at">coef=</span><span class="st">"condition_tumor_vs_normal"</span>, <span class="at">type=</span><span class="st">"apeglm"</span>)</span>
<span id="cb52-432"><a href="#cb52-432" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-433"><a href="#cb52-433" aria-hidden="true" tabindex="-1"></a><span class="co"># `normal`是原始的 DESeq2 收缩估计器，是先验的自适应正态分布</span></span>
<span id="cb52-434"><a href="#cb52-434" aria-hidden="true" tabindex="-1"></a>resNorm <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds, <span class="at">coef=</span><span class="dv">2</span>, <span class="at">type=</span><span class="st">"normal"</span>)</span>
<span id="cb52-435"><a href="#cb52-435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-436"><a href="#cb52-436" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("ashr")</span></span>
<span id="cb52-437"><a href="#cb52-437" aria-hidden="true" tabindex="-1"></a>resAsh <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds, <span class="at">coef=</span><span class="dv">2</span>, <span class="at">type=</span><span class="st">"ashr"</span>)</span>
<span id="cb52-438"><a href="#cb52-438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-439"><a href="#cb52-439" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res)</span>
<span id="cb52-440"><a href="#cb52-440" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb52-441"><a href="#cb52-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-442"><a href="#cb52-442" aria-hidden="true" tabindex="-1"></a>res05 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb52-443"><a href="#cb52-443" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res05)</span>
<span id="cb52-444"><a href="#cb52-444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-445"><a href="#cb52-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-446"><a href="#cb52-446" aria-hidden="true" tabindex="-1"></a><span class="do">## 先验信息</span></span>
<span id="cb52-447"><a href="#cb52-447" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-448"><a href="#cb52-448" aria-hidden="true" tabindex="-1"></a><span class="fu">priorInfo</span>(resLFC)</span>
<span id="cb52-449"><a href="#cb52-449" aria-hidden="true" tabindex="-1"></a><span class="fu">priorInfo</span>(resNorm)</span>
<span id="cb52-450"><a href="#cb52-450" aria-hidden="true" tabindex="-1"></a><span class="fu">priorInfo</span>(resAsh)</span>
<span id="cb52-451"><a href="#cb52-451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-452"><a href="#cb52-452" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-453"><a href="#cb52-453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-454"><a href="#cb52-454" aria-hidden="true" tabindex="-1"></a><span class="fu">### MA图 Mean-AbsDeviation plot</span></span>
<span id="cb52-455"><a href="#cb52-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-458"><a href="#cb52-458" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-459"><a href="#cb52-459" aria-hidden="true" tabindex="-1"></a><span class="co"># 1.将DESeq结果转换为dataframe</span></span>
<span id="cb52-460"><a href="#cb52-460" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DESeq2_DEG)</span>
<span id="cb52-461"><a href="#cb52-461" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. 对gene添加“是否呈现显著差异表达”的标签</span></span>
<span id="cb52-462"><a href="#cb52-462" aria-hidden="true" tabindex="-1"></a>DESeq2_DEG<span class="sc">$</span>significant <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">ifelse</span>(DESeq2_DEG<span class="sc">$</span>padj <span class="sc">&lt;</span><span class="fl">0.05</span>, <span class="st">"Significant"</span>, <span class="st">"NS"</span>),</span>
<span id="cb52-463"><a href="#cb52-463" aria-hidden="true" tabindex="-1"></a><span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Significant"</span>, <span class="st">"NS"</span>))</span>
<span id="cb52-464"><a href="#cb52-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-465"><a href="#cb52-465" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(DESeq2_DEG<span class="sc">$</span>significant)</span>
<span id="cb52-466"><a href="#cb52-466" aria-hidden="true" tabindex="-1"></a><span class="co"># 以baseMean作为x，log2FoldChange作为y   以significant作为分类变量</span></span>
<span id="cb52-467"><a href="#cb52-467" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(DESeq2_DEG, <span class="fu">aes</span>(baseMean, log2FoldChange, <span class="at">colour=</span>significant)) <span class="sc">+</span> </span>
<span id="cb52-468"><a href="#cb52-468" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb52-469"><a href="#cb52-469" aria-hidden="true" tabindex="-1"></a> <span class="fu">scale_y_continuous</span>(<span class="at">limits=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">3</span>, <span class="dv">3</span>)) <span class="sc">+</span> </span>
<span id="cb52-470"><a href="#cb52-470" aria-hidden="true" tabindex="-1"></a> <span class="fu">scale_x_log10</span>() <span class="sc">+</span> </span>
<span id="cb52-471"><a href="#cb52-471" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">colour=</span><span class="st">"black"</span>, <span class="at">linewidth=</span><span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb52-472"><a href="#cb52-472" aria-hidden="true" tabindex="-1"></a> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"mean of normalized counts"</span>, <span class="at">y=</span><span class="st">"log2FoldChange"</span>) <span class="sc">+</span> </span>
<span id="cb52-473"><a href="#cb52-473" aria-hidden="true" tabindex="-1"></a> <span class="fu">scale_colour_manual</span>(<span class="at">name=</span><span class="st">"padj"</span>, </span>
<span id="cb52-474"><a href="#cb52-474" aria-hidden="true" tabindex="-1"></a>                     <span class="at">values=</span><span class="fu">c</span>(<span class="st">"Significant"</span><span class="ot">=</span><span class="st">"blue"</span>,<span class="st">"NS"</span><span class="ot">=</span><span class="st">"grey50"</span>)) <span class="sc">+</span></span>
<span id="cb52-475"><a href="#cb52-475" aria-hidden="true" tabindex="-1"></a> <span class="fu">theme_classic</span>()</span>
<span id="cb52-476"><a href="#cb52-476" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-477"><a href="#cb52-477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-480"><a href="#cb52-480" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-481"><a href="#cb52-481" aria-hidden="true" tabindex="-1"></a>xlim <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">25</span>)</span>
<span id="cb52-482"><a href="#cb52-482" aria-hidden="true" tabindex="-1"></a>ylim <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">3</span>,<span class="dv">3</span>)</span>
<span id="cb52-483"><a href="#cb52-483" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resOrdered,  <span class="at">xlim=</span>xlim, <span class="at">ylim=</span>ylim,<span class="at">main=</span><span class="st">"none"</span>)</span>
<span id="cb52-484"><a href="#cb52-484" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC, <span class="at">xlim=</span>xlim, <span class="at">ylim=</span>ylim, <span class="at">main=</span><span class="st">"apeglm"</span>)</span>
<span id="cb52-485"><a href="#cb52-485" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resNorm, <span class="at">xlim=</span>xlim, <span class="at">ylim=</span>ylim, <span class="at">main=</span><span class="st">"normal"</span>)</span>
<span id="cb52-486"><a href="#cb52-486" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resAsh, <span class="at">xlim=</span>xlim, <span class="at">ylim=</span>ylim, <span class="at">main=</span><span class="st">"ashr"</span>)</span>
<span id="cb52-487"><a href="#cb52-487" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-488"><a href="#cb52-488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-489"><a href="#cb52-489" aria-hidden="true" tabindex="-1"></a><span class="fu">### readCounts</span></span>
<span id="cb52-490"><a href="#cb52-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-493"><a href="#cb52-493" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-494"><a href="#cb52-494" aria-hidden="true" tabindex="-1"></a><span class="co">#  the counts of reads for a single gene across the groups</span></span>
<span id="cb52-495"><a href="#cb52-495" aria-hidden="true" tabindex="-1"></a><span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="fu">which.min</span>(resOrdered<span class="sc">$</span>padj), </span>
<span id="cb52-496"><a href="#cb52-496" aria-hidden="true" tabindex="-1"></a>  <span class="at">intgroup=</span><span class="st">"condition"</span>,<span class="at">returnData=</span><span class="cn">TRUE</span>) <span class="sc">|&gt;</span> </span>
<span id="cb52-497"><a href="#cb52-497" aria-hidden="true" tabindex="-1"></a> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>condition, <span class="at">y=</span>count)) <span class="sc">+</span> </span>
<span id="cb52-498"><a href="#cb52-498" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_point</span>(<span class="at">position=</span><span class="fu">position_jitter</span>(<span class="at">w=</span><span class="fl">0.1</span>,<span class="at">h=</span><span class="dv">0</span>))</span>
<span id="cb52-499"><a href="#cb52-499" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-500"><a href="#cb52-500" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-501"><a href="#cb52-501" aria-hidden="true" tabindex="-1"></a><span class="fu">### 异常观测</span></span>
<span id="cb52-502"><a href="#cb52-502" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-505"><a href="#cb52-505" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-506"><a href="#cb52-506" aria-hidden="true" tabindex="-1"></a>W <span class="ot">&lt;-</span> resOrdered<span class="sc">$</span>stat</span>
<span id="cb52-507"><a href="#cb52-507" aria-hidden="true" tabindex="-1"></a>maxCooks <span class="ot">&lt;-</span> <span class="fu">apply</span>(<span class="fu">assays</span>(dds)[[<span class="st">"cooks"</span>]],<span class="dv">1</span>,max)</span>
<span id="cb52-508"><a href="#cb52-508" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="sc">!</span><span class="fu">is.na</span>(W)</span>
<span id="cb52-509"><a href="#cb52-509" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">rank</span>(W[idx]), maxCooks[idx], <span class="at">xlab=</span><span class="st">"rank of Wald statistic"</span>, </span>
<span id="cb52-510"><a href="#cb52-510" aria-hidden="true" tabindex="-1"></a>  <span class="at">ylab=</span><span class="st">"maximum Cook's distance per gene"</span>,</span>
<span id="cb52-511"><a href="#cb52-511" aria-hidden="true" tabindex="-1"></a>  <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">5</span>), <span class="at">cex=</span>.<span class="dv">4</span>, <span class="at">col=</span><span class="fu">rgb</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,.<span class="dv">3</span>))</span>
<span id="cb52-512"><a href="#cb52-512" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">ncol</span>(dds)</span>
<span id="cb52-513"><a href="#cb52-513" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb52-514"><a href="#cb52-514" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="fu">qf</span>(.<span class="dv">99</span>, p, m <span class="sc">-</span> p))</span>
<span id="cb52-515"><a href="#cb52-515" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-516"><a href="#cb52-516" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-517"><a href="#cb52-517" aria-hidden="true" tabindex="-1"></a><span class="fu">### 多因素设计</span></span>
<span id="cb52-518"><a href="#cb52-518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-521"><a href="#cb52-521" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-522"><a href="#cb52-522" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(dds)</span>
<span id="cb52-523"><a href="#cb52-523" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-524"><a href="#cb52-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-525"><a href="#cb52-525" aria-hidden="true" tabindex="-1"></a><span class="fu">### 导出到 CSV 文件</span></span>
<span id="cb52-526"><a href="#cb52-526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-529"><a href="#cb52-529" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-530"><a href="#cb52-530" aria-hidden="true" tabindex="-1"></a>resSig <span class="ot">&lt;-</span> <span class="fu">subset</span>(resOrdered, padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb52-531"><a href="#cb52-531" aria-hidden="true" tabindex="-1"></a>resSig</span>
<span id="cb52-532"><a href="#cb52-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-533"><a href="#cb52-533" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(<span class="fu">as.data.frame</span>(resSig), </span>
<span id="cb52-534"><a href="#cb52-534" aria-hidden="true" tabindex="-1"></a> <span class="at">file=</span><span class="st">"data/padjust_0.05_resOrdered_tumor_normal.csv"</span>)</span>
<span id="cb52-535"><a href="#cb52-535" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-536"><a href="#cb52-536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-537"><a href="#cb52-537" aria-hidden="true" tabindex="-1"></a><span class="fu"># 富集分析</span></span>
<span id="cb52-538"><a href="#cb52-538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-541"><a href="#cb52-541" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-542"><a href="#cb52-542" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install('clusterProfiler')</span></span>
<span id="cb52-543"><a href="#cb52-543" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("org.Hs.eg.db")</span></span>
<span id="cb52-544"><a href="#cb52-544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-545"><a href="#cb52-545" aria-hidden="true" tabindex="-1"></a>DEG <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/resOrdered.csv"</span>)</span>
<span id="cb52-546"><a href="#cb52-546" aria-hidden="true" tabindex="-1"></a>DESeq2_DEG <span class="ot">=</span> <span class="fu">na.omit</span>(DEG) </span>
<span id="cb52-547"><a href="#cb52-547" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-548"><a href="#cb52-548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-549"><a href="#cb52-549" aria-hidden="true" tabindex="-1"></a>对于有参考基因组物种的分析，可以在相关软件包中直接加载该物种的背景基因集 （1）对于常见的模式物种，例如人类，有些专门的 R 包数据库，例如人类参考基因组 hg19 的 <span class="in">`library(org.Hs.eg.db)`</span></span>
<span id="cb52-550"><a href="#cb52-550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-551"><a href="#cb52-551" aria-hidden="true" tabindex="-1"></a><span class="fu">### 数据整理</span></span>
<span id="cb52-552"><a href="#cb52-552" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-555"><a href="#cb52-555" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-556"><a href="#cb52-556" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb52-557"><a href="#cb52-557" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(GenomicRanges<span class="sc">::</span>setdiff)</span>
<span id="cb52-558"><a href="#cb52-558" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-559"><a href="#cb52-559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-562"><a href="#cb52-562" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-563"><a href="#cb52-563" aria-hidden="true" tabindex="-1"></a>nrDEG <span class="ot">&lt;-</span> DESeq2_DEG</span>
<span id="cb52-564"><a href="#cb52-564" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(nrDEG)</span>
<span id="cb52-565"><a href="#cb52-565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-566"><a href="#cb52-566" aria-hidden="true" tabindex="-1"></a>nrDEG<span class="sc">$</span>expression <span class="ot">&lt;-</span> <span class="fu">as.factor</span>( </span>
<span id="cb52-567"><a href="#cb52-567" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>( nrDEG<span class="sc">$</span>padj<span class="sc">&lt;</span> <span class="fl">0.2</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(nrDEG<span class="sc">$</span>log2FoldChange) <span class="sc">&gt;=</span> <span class="dv">1</span>,</span>
<span id="cb52-568"><a href="#cb52-568" aria-hidden="true" tabindex="-1"></a>            <span class="fu">ifelse</span>( nrDEG<span class="sc">$</span>log2FoldChange <span class="sc">&gt;=</span> <span class="dv">1</span> , <span class="st">'up'</span>, <span class="st">'down'</span> ),</span>
<span id="cb52-569"><a href="#cb52-569" aria-hidden="true" tabindex="-1"></a>            <span class="st">'normal'</span>))</span>
<span id="cb52-570"><a href="#cb52-570" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(nrDEG<span class="sc">$</span>expression)</span>
<span id="cb52-571"><a href="#cb52-571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-572"><a href="#cb52-572" aria-hidden="true" tabindex="-1"></a>nrDEG <span class="ot">&lt;-</span> nrDEG <span class="sc">|&gt;</span> </span>
<span id="cb52-573"><a href="#cb52-573" aria-hidden="true" tabindex="-1"></a> <span class="fu">inner_join</span>(gtf_mrna_v22, <span class="at">by =</span> <span class="fu">c</span>( <span class="st">"X"</span><span class="ot">=</span> <span class="st">"gene_name"</span>))</span>
<span id="cb52-574"><a href="#cb52-574" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-575"><a href="#cb52-575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-576"><a href="#cb52-576" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(nrDEG) <span class="ot">&lt;-</span> <span class="fu">str_sub</span>(nrDEG<span class="sc">$</span>gene_id, <span class="at">start =</span> <span class="dv">1</span>, <span class="at">end =</span> <span class="dv">15</span>)</span>
<span id="cb52-577"><a href="#cb52-577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-578"><a href="#cb52-578" aria-hidden="true" tabindex="-1"></a>nrDEG<span class="sc">$</span>ENSEMBL <span class="ot">&lt;-</span> <span class="fu">rownames</span>( nrDEG )</span>
<span id="cb52-579"><a href="#cb52-579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-580"><a href="#cb52-580" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-581"><a href="#cb52-581" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb52-582"><a href="#cb52-582" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb52-583"><a href="#cb52-583" aria-hidden="true" tabindex="-1"></a>AnnotationDbi<span class="sc">::</span><span class="fu">keytypes</span>(org.Hs.eg.db)</span>
<span id="cb52-584"><a href="#cb52-584" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">keys</span>(org.Hs.eg.db, <span class="st">"ENSEMBL"</span>))</span>
<span id="cb52-585"><a href="#cb52-585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-586"><a href="#cb52-586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-587"><a href="#cb52-587" aria-hidden="true" tabindex="-1"></a>valid_ensembl <span class="ot">&lt;-</span> <span class="fu">rownames</span>(nrDEG) <span class="sc">%in%</span> <span class="fu">keys</span>(org.Hs.eg.db, <span class="st">"ENSEMBL"</span>)  </span>
<span id="cb52-588"><a href="#cb52-588" aria-hidden="true" tabindex="-1"></a>nrDEG_filtered <span class="ot">&lt;-</span> nrDEG[valid_ensembl, ]  </span>
<span id="cb52-589"><a href="#cb52-589" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-590"><a href="#cb52-590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-591"><a href="#cb52-591" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">bitr</span>(</span>
<span id="cb52-592"><a href="#cb52-592" aria-hidden="true" tabindex="-1"></a> <span class="at">geneID=</span> <span class="fu">rownames</span>(nrDEG_filtered),</span>
<span id="cb52-593"><a href="#cb52-593" aria-hidden="true" tabindex="-1"></a> <span class="at">fromType =</span> <span class="st">"ENSEMBL"</span>, </span>
<span id="cb52-594"><a href="#cb52-594" aria-hidden="true" tabindex="-1"></a> <span class="at">toType =</span> <span class="fu">c</span>( <span class="st">"ENTREZID"</span>,<span class="st">"SYMBOL"</span> ), </span>
<span id="cb52-595"><a href="#cb52-595" aria-hidden="true" tabindex="-1"></a> <span class="at">OrgDb =</span> org.Hs.eg.db )</span>
<span id="cb52-596"><a href="#cb52-596" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-597"><a href="#cb52-597" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(df)</span>
<span id="cb52-598"><a href="#cb52-598" aria-hidden="true" tabindex="-1"></a>nrDEG_filtered<span class="sc">$</span>SYMBOL <span class="ot">=</span> <span class="fu">rownames</span>(nrDEG_filtered)</span>
<span id="cb52-599"><a href="#cb52-599" aria-hidden="true" tabindex="-1"></a>nrDEG <span class="ot">=</span> <span class="fu">inner_join</span>(nrDEG_filtered, df, <span class="at">by=</span><span class="st">'ENSEMBL'</span> )</span>
<span id="cb52-600"><a href="#cb52-600" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-601"><a href="#cb52-601" aria-hidden="true" tabindex="-1"></a>gene_up <span class="ot">=</span> nrDEG[ nrDEG<span class="sc">$</span>expression <span class="sc">==</span> <span class="st">'up'</span>, <span class="st">"ENTREZID"</span> ] </span>
<span id="cb52-602"><a href="#cb52-602" aria-hidden="true" tabindex="-1"></a>gene_down <span class="ot">=</span> nrDEG[ nrDEG<span class="sc">$</span>expression <span class="sc">==</span> <span class="st">'down'</span>, <span class="st">"ENTREZID"</span>]</span>
<span id="cb52-603"><a href="#cb52-603" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-604"><a href="#cb52-604" aria-hidden="true" tabindex="-1"></a>gene_all <span class="ot">=</span> <span class="fu">as.character</span>(nrDEG[ ,<span class="st">"ENTREZID"</span>])</span>
<span id="cb52-605"><a href="#cb52-605" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-606"><a href="#cb52-606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-607"><a href="#cb52-607" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-608"><a href="#cb52-608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-609"><a href="#cb52-609" aria-hidden="true" tabindex="-1"></a><span class="fu">## KEGG 通路富集</span></span>
<span id="cb52-610"><a href="#cb52-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-611"><a href="#cb52-611" aria-hidden="true" tabindex="-1"></a>对于各列内容：</span>
<span id="cb52-612"><a href="#cb52-612" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-613"><a href="#cb52-613" aria-hidden="true" tabindex="-1"></a>ID和Description，富集到的通路和功能描述；</span>
<span id="cb52-614"><a href="#cb52-614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-615"><a href="#cb52-615" aria-hidden="true" tabindex="-1"></a>GeneRatio和BgRatio，分别为富集到该通路中的基因数目/给定基因的总数目，以及该通路中背景基因总数目/该物种所有已知的KEGG功能基因数目；</span>
<span id="cb52-616"><a href="#cb52-616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-617"><a href="#cb52-617" aria-hidden="true" tabindex="-1"></a>pvalue、p.adjust和qvalue，p值、校正后p值和q值信息；</span>
<span id="cb52-618"><a href="#cb52-618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-619"><a href="#cb52-619" aria-hidden="true" tabindex="-1"></a>geneID和Count，富集到该通路中的基因名称和数目。</span>
<span id="cb52-620"><a href="#cb52-620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-621"><a href="#cb52-621" aria-hidden="true" tabindex="-1"></a>注：本示例分析中使用的entrze id，故这里也显示的entrze id。如使用其它类型的基因名称，如symbol id等类型，可在结果中对entrze id和symbol id做个转换。</span>
<span id="cb52-622"><a href="#cb52-622" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-625"><a href="#cb52-625" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-626"><a href="#cb52-626" aria-hidden="true" tabindex="-1"></a>kegg_up <span class="ot">&lt;-</span> clusterProfiler<span class="sc">::</span><span class="fu">enrichKEGG</span>(<span class="at">gene =</span> gene_up ,</span>
<span id="cb52-627"><a href="#cb52-627" aria-hidden="true" tabindex="-1"></a>                    <span class="at">organism=</span>  <span class="st">'hsa'</span>,</span>
<span id="cb52-628"><a href="#cb52-628" aria-hidden="true" tabindex="-1"></a>                    <span class="at">keyType =</span> <span class="st">"kegg"</span>,</span>
<span id="cb52-629"><a href="#cb52-629" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pvalueCutoff  =</span>  <span class="fl">0.1</span>,</span>
<span id="cb52-630"><a href="#cb52-630" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>,  <span class="co"># FDR多重假设检验校正 qvalue</span></span>
<span id="cb52-631"><a href="#cb52-631" aria-hidden="true" tabindex="-1"></a>                    <span class="at">universe=</span>  gene_all,</span>
<span id="cb52-632"><a href="#cb52-632" aria-hidden="true" tabindex="-1"></a>                    <span class="at">minGSSize =</span> <span class="dv">10</span>,</span>
<span id="cb52-633"><a href="#cb52-633" aria-hidden="true" tabindex="-1"></a>                    <span class="at">maxGSSize =</span> <span class="dv">500</span>,</span>
<span id="cb52-634"><a href="#cb52-634" aria-hidden="true" tabindex="-1"></a>                    <span class="at">qvalueCutoff  =</span>  <span class="fl">0.05</span> )</span>
<span id="cb52-635"><a href="#cb52-635" aria-hidden="true" tabindex="-1"></a>kegg_down <span class="ot">&lt;-</span> clusterProfiler<span class="sc">::</span><span class="fu">enrichKEGG</span>( <span class="at">gene =</span>  gene_down ,</span>
<span id="cb52-636"><a href="#cb52-636" aria-hidden="true" tabindex="-1"></a>                    <span class="at">organism=</span>  <span class="st">'hsa'</span>,</span>
<span id="cb52-637"><a href="#cb52-637" aria-hidden="true" tabindex="-1"></a>                    <span class="at">keyType =</span> <span class="st">"kegg"</span>,</span>
<span id="cb52-638"><a href="#cb52-638" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pvalueCutoff  =</span>  <span class="fl">0.05</span>,</span>
<span id="cb52-639"><a href="#cb52-639" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>,</span>
<span id="cb52-640"><a href="#cb52-640" aria-hidden="true" tabindex="-1"></a>                    <span class="at">universe=</span>  gene_all,</span>
<span id="cb52-641"><a href="#cb52-641" aria-hidden="true" tabindex="-1"></a>                    <span class="at">minGSSize =</span> <span class="dv">10</span>,</span>
<span id="cb52-642"><a href="#cb52-642" aria-hidden="true" tabindex="-1"></a>                    <span class="at">maxGSSize =</span> <span class="dv">500</span>,</span>
<span id="cb52-643"><a href="#cb52-643" aria-hidden="true" tabindex="-1"></a>                    <span class="at">qvalueCutoff  =</span>  <span class="fl">0.05</span> )</span>
<span id="cb52-644"><a href="#cb52-644" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-645"><a href="#cb52-645" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-648"><a href="#cb52-648" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-649"><a href="#cb52-649" aria-hidden="true" tabindex="-1"></a>df_up<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(kegg_up)</span>
<span id="cb52-650"><a href="#cb52-650" aria-hidden="true" tabindex="-1"></a>df_down<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(kegg_down)</span>
<span id="cb52-651"><a href="#cb52-651" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-652"><a href="#cb52-652" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-655"><a href="#cb52-655" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-656"><a href="#cb52-656" aria-hidden="true" tabindex="-1"></a>df_up <span class="ot">&lt;-</span> df_up[df_up<span class="sc">$</span>pvalue <span class="sc">&lt;</span> <span class="fl">0.05</span>, ]</span>
<span id="cb52-657"><a href="#cb52-657" aria-hidden="true" tabindex="-1"></a>df_down <span class="ot">&lt;-</span> df_down[df_down<span class="sc">$</span>pvalue <span class="sc">&lt;</span> <span class="fl">0.05</span>, ]</span>
<span id="cb52-658"><a href="#cb52-658" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-659"><a href="#cb52-659" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-662"><a href="#cb52-662" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-663"><a href="#cb52-663" aria-hidden="true" tabindex="-1"></a>df_up <span class="ot">&lt;-</span> df_up <span class="sc">|&gt;</span> </span>
<span id="cb52-664"><a href="#cb52-664" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group=</span><span class="dv">1</span>)</span>
<span id="cb52-665"><a href="#cb52-665" aria-hidden="true" tabindex="-1"></a>df_down <span class="ot">&lt;-</span> df_down <span class="sc">|&gt;</span> </span>
<span id="cb52-666"><a href="#cb52-666" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group=</span><span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb52-667"><a href="#cb52-667" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-668"><a href="#cb52-668" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-669"><a href="#cb52-669" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-672"><a href="#cb52-672" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-673"><a href="#cb52-673" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">rbind</span>(df_up,df_down)</span>
<span id="cb52-674"><a href="#cb52-674" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dat)</span>
<span id="cb52-675"><a href="#cb52-675" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-676"><a href="#cb52-676" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-677"><a href="#cb52-677" aria-hidden="true" tabindex="-1"></a><span class="fu">### 水平条图</span></span>
<span id="cb52-678"><a href="#cb52-678" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-681"><a href="#cb52-681" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-682"><a href="#cb52-682" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Negative_log10pvalue <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(dat<span class="sc">$</span>pvalue)</span>
<span id="cb52-683"><a href="#cb52-683" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-684"><a href="#cb52-684" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Group_log10pvalue <span class="ot">=</span> dat<span class="sc">$</span>Negative_log10pvalue <span class="sc">*</span> dat<span class="sc">$</span>group</span>
<span id="cb52-685"><a href="#cb52-685" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-686"><a href="#cb52-686" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">=</span> dat[<span class="fu">order</span>( dat<span class="sc">$</span>Group_log10pvalue, <span class="at">decreasing =</span> F ), ]</span>
<span id="cb52-687"><a href="#cb52-687" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-688"><a href="#cb52-688" aria-hidden="true" tabindex="-1"></a>ggKEGG <span class="ot">&lt;-</span> <span class="fu">ggplot</span>( dat,<span class="fu">aes</span>(<span class="at">x =</span>  <span class="fu">reorder</span>(Description,</span>
<span id="cb52-689"><a href="#cb52-689" aria-hidden="true" tabindex="-1"></a>                                       <span class="fu">order</span>(Group_log10pvalue,<span class="at">decreasing=</span>F)),</span>
<span id="cb52-690"><a href="#cb52-690" aria-hidden="true" tabindex="-1"></a>                          <span class="at">y =</span> Group_log10pvalue, <span class="at">fill =</span> group)) <span class="sc">+</span> </span>
<span id="cb52-691"><a href="#cb52-691" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>( <span class="at">stat =</span> <span class="st">"identity"</span> ) <span class="sc">+</span> </span>
<span id="cb52-692"><a href="#cb52-692" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradient</span>( <span class="at">low =</span> <span class="st">"blue"</span>, <span class="at">high =</span> <span class="st">"red"</span> ) <span class="sc">+</span> </span>
<span id="cb52-693"><a href="#cb52-693" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>( <span class="at">name =</span> <span class="st">"Pathway names"</span> ) <span class="sc">+</span></span>
<span id="cb52-694"><a href="#cb52-694" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>( <span class="at">name =</span> <span class="st">"log10 pvalue"</span> ) <span class="sc">+</span></span>
<span id="cb52-695"><a href="#cb52-695" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb52-696"><a href="#cb52-696" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>( <span class="at">plot.title =</span> <span class="fu">element_text</span>( <span class="at">hjust =</span> <span class="fl">0.5</span> ) ,</span>
<span id="cb52-697"><a href="#cb52-697" aria-hidden="true" tabindex="-1"></a>           <span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb52-698"><a href="#cb52-698" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>( <span class="st">"KEGG Pathway Enrichment"</span> ) </span>
<span id="cb52-699"><a href="#cb52-699" aria-hidden="true" tabindex="-1"></a>ggKEGG</span>
<span id="cb52-700"><a href="#cb52-700" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-701"><a href="#cb52-701" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-702"><a href="#cb52-702" aria-hidden="true" tabindex="-1"></a><span class="fu">### 点图</span></span>
<span id="cb52-703"><a href="#cb52-703" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-706"><a href="#cb52-706" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-707"><a href="#cb52-707" aria-hidden="true" tabindex="-1"></a><span class="co"># GeneRatio 转化成数值型</span></span>
<span id="cb52-708"><a href="#cb52-708" aria-hidden="true" tabindex="-1"></a>GR <span class="ot">&lt;-</span> dat<span class="sc">$</span>GeneRatio <span class="sc">|&gt;</span> </span>
<span id="cb52-709"><a href="#cb52-709" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_split</span>(<span class="at">pattern =</span> <span class="st">"/"</span>,<span class="at">simplify =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">as_tibble</span>()</span>
<span id="cb52-710"><a href="#cb52-710" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-711"><a href="#cb52-711" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>GeneRatio <span class="ot">&lt;-</span> <span class="fu">parse_number</span>(GR<span class="sc">$</span>V1)<span class="sc">/</span><span class="fu">parse_number</span>(GR<span class="sc">$</span>V2)</span>
<span id="cb52-712"><a href="#cb52-712" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-713"><a href="#cb52-713" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> dat[<span class="fu">order</span>(dat<span class="sc">$</span>GeneRatio,<span class="at">decreasing=</span>F),]</span>
<span id="cb52-714"><a href="#cb52-714" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-715"><a href="#cb52-715" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> dat,<span class="fu">aes</span>(<span class="at">x=</span>GeneRatio,</span>
<span id="cb52-716"><a href="#cb52-716" aria-hidden="true" tabindex="-1"></a>                      <span class="at">y=</span><span class="fu">reorder</span>(Description,<span class="fu">order</span>(GeneRatio,<span class="at">decreasing=</span>F)),</span>
<span id="cb52-717"><a href="#cb52-717" aria-hidden="true" tabindex="-1"></a>                      <span class="at">color =</span> Negative_log10pvalue,</span>
<span id="cb52-718"><a href="#cb52-718" aria-hidden="true" tabindex="-1"></a>                      <span class="at">size=</span>Count</span>
<span id="cb52-719"><a href="#cb52-719" aria-hidden="true" tabindex="-1"></a>                      )</span>
<span id="cb52-720"><a href="#cb52-720" aria-hidden="true" tabindex="-1"></a>       )<span class="sc">+</span></span>
<span id="cb52-721"><a href="#cb52-721" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb52-722"><a href="#cb52-722" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>( <span class="at">name =</span> <span class="st">"GeneRatio"</span> ,<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="fl">1.0</span>,<span class="fl">0.1</span>))<span class="sc">+</span></span>
<span id="cb52-723"><a href="#cb52-723" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"blue"</span>, <span class="at">high =</span> <span class="st">"red"</span> )<span class="sc">+</span></span>
<span id="cb52-724"><a href="#cb52-724" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>( <span class="at">name =</span> <span class="st">"Pathway names"</span> ) <span class="sc">+</span></span>
<span id="cb52-725"><a href="#cb52-725" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb52-726"><a href="#cb52-726" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>( <span class="at">plot.title =</span> <span class="fu">element_text</span>( <span class="at">hjust =</span> <span class="fl">0.5</span> )) <span class="sc">+</span></span>
<span id="cb52-727"><a href="#cb52-727" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>( <span class="st">"KEGG Pathway Enrichment "</span> )<span class="sc">+</span></span>
<span id="cb52-728"><a href="#cb52-728" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">color=</span><span class="st">"-log10 pvalue"</span>)</span>
<span id="cb52-729"><a href="#cb52-729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-730"><a href="#cb52-730" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-731"><a href="#cb52-731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-732"><a href="#cb52-732" aria-hidden="true" tabindex="-1"></a><span class="fu">## GO富集分析</span></span>
<span id="cb52-733"><a href="#cb52-733" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-734"><a href="#cb52-734" aria-hidden="true" tabindex="-1"></a>对于输出的GO富集结果表格中的各列内容：</span>
<span id="cb52-735"><a href="#cb52-735" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-736"><a href="#cb52-736" aria-hidden="true" tabindex="-1"></a>ONTOLOGY，GO的BP（生物学过程）、CC（细胞组分）或MF（分子功能）；</span>
<span id="cb52-737"><a href="#cb52-737" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-738"><a href="#cb52-738" aria-hidden="true" tabindex="-1"></a>ID和Description，富集到的GO term及其描述；</span>
<span id="cb52-739"><a href="#cb52-739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-740"><a href="#cb52-740" aria-hidden="true" tabindex="-1"></a>GeneRatio和BgRatio，分别为富集到该GO term中的基因数目/给定基因的总数目，以及该GO term中背景基因总数目/该物种所有已知GO功能基因数目；</span>
<span id="cb52-741"><a href="#cb52-741" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-742"><a href="#cb52-742" aria-hidden="true" tabindex="-1"></a>pvalue、p.adjust和qvalue，p值、校正后p值和q值；</span>
<span id="cb52-743"><a href="#cb52-743" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-744"><a href="#cb52-744" aria-hidden="true" tabindex="-1"></a>geneID和Count，富集到该GO term中的基因名称和数目。</span>
<span id="cb52-745"><a href="#cb52-745" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-746"><a href="#cb52-746" aria-hidden="true" tabindex="-1"></a>注：以上示例分析中使用的entrze id，故输出的基因名称也显示的entrze id。如期望显示其它类型的基因名称，如symbol id等类型，除了更改为使用symbol id的基因名称做分析外，还可以通过基因名称转换的方式对entrze id和symbol id做个转换。</span>
<span id="cb52-747"><a href="#cb52-747" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-750"><a href="#cb52-750" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-751"><a href="#cb52-751" aria-hidden="true" tabindex="-1"></a>GO_EA<span class="ot">&lt;-</span> clusterProfiler<span class="sc">::</span><span class="fu">enrichGO</span>(<span class="at">gene =</span> gene_all ,</span>
<span id="cb52-752"><a href="#cb52-752" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">OrgDb =</span> <span class="st">"org.Hs.eg.db"</span>,</span>
<span id="cb52-753"><a href="#cb52-753" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">keyType =</span> <span class="st">"ENTREZID"</span>,</span>
<span id="cb52-754"><a href="#cb52-754" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">ont =</span> <span class="st">"ALL"</span>,</span>
<span id="cb52-755"><a href="#cb52-755" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">pvalueCutoff  =</span>  <span class="fl">0.05</span> ,</span>
<span id="cb52-756"><a href="#cb52-756" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">qvalueCutoff  =</span>  <span class="fl">0.05</span>,</span>
<span id="cb52-757"><a href="#cb52-757" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">readable =</span> F)</span>
<span id="cb52-758"><a href="#cb52-758" aria-hidden="true" tabindex="-1"></a>GO_EA_tbl <span class="ot">&lt;-</span>  <span class="fu">as_tibble</span>(GO_EA)</span>
<span id="cb52-759"><a href="#cb52-759" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-760"><a href="#cb52-760" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-761"><a href="#cb52-761" aria-hidden="true" tabindex="-1"></a><span class="fu">### 可视化</span></span>
<span id="cb52-762"><a href="#cb52-762" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-765"><a href="#cb52-765" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-766"><a href="#cb52-766" aria-hidden="true" tabindex="-1"></a><span class="co">#clusterProfiler 包里的一些默认作图方法，例如</span></span>
<span id="cb52-767"><a href="#cb52-767" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(GO_EA)  <span class="co">#富集柱形图</span></span>
<span id="cb52-768"><a href="#cb52-768" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(GO_EA)  <span class="co">#富集气泡图</span></span>
<span id="cb52-769"><a href="#cb52-769" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-770"><a href="#cb52-770" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-773"><a href="#cb52-773" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb52-774"><a href="#cb52-774" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">&lt;-</span> GO_EA_tbl[GO_EA_tbl<span class="sc">$</span>pvalue<span class="sc">&lt;</span> <span class="fl">0.05</span>, ]</span>
<span id="cb52-775"><a href="#cb52-775" aria-hidden="true" tabindex="-1"></a>dat2<span class="sc">$</span>Negative_log10pvalue <span class="ot">&lt;-</span>  <span class="sc">-</span><span class="fu">log10</span>(dat2<span class="sc">$</span>pvalue)</span>
<span id="cb52-776"><a href="#cb52-776" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">=</span> dat2[<span class="fu">order</span>( dat2<span class="sc">$</span>Negative_log10pvalue, <span class="at">decreasing =</span> F ), ]</span>
<span id="cb52-777"><a href="#cb52-777" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-778"><a href="#cb52-778" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">&lt;-</span> dat2 <span class="sc">|&gt;</span> </span>
<span id="cb52-779"><a href="#cb52-779" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(ONTOLOGY) <span class="sc">|&gt;</span> </span>
<span id="cb52-780"><a href="#cb52-780" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_max</span>(<span class="at">order_by =</span> Negative_log10pvalue,<span class="at">n=</span><span class="dv">10</span>) <span class="sc">|&gt;</span> </span>
<span id="cb52-781"><a href="#cb52-781" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>()</span>
<span id="cb52-782"><a href="#cb52-782" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-783"><a href="#cb52-783" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-784"><a href="#cb52-784" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(dat2,<span class="fu">aes</span>(<span class="at">x =</span> Negative_log10pvalue, </span>
<span id="cb52-785"><a href="#cb52-785" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span> <span class="fu">reorder</span>(Description,</span>
<span id="cb52-786"><a href="#cb52-786" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">order</span>(ONTOLOGY,Negative_log10pvalue,<span class="at">decreasing=</span>F)),</span>
<span id="cb52-787"><a href="#cb52-787" aria-hidden="true" tabindex="-1"></a>                <span class="at">fill =</span> ONTOLOGY ))<span class="sc">+</span> </span>
<span id="cb52-788"><a href="#cb52-788" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>( <span class="at">stat =</span> <span class="st">"identity"</span> ) <span class="sc">+</span></span>
<span id="cb52-789"><a href="#cb52-789" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>( <span class="at">name =</span> <span class="st">"-log10 pvalue"</span> ) <span class="sc">+</span></span>
<span id="cb52-790"><a href="#cb52-790" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>( <span class="at">name =</span> <span class="st">"Pathway names"</span> ) <span class="sc">+</span></span>
<span id="cb52-791"><a href="#cb52-791" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb52-792"><a href="#cb52-792" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>( <span class="at">plot.title =</span> <span class="fu">element_text</span>( <span class="at">hjust =</span> <span class="fl">0.5</span> ) ) <span class="sc">+</span></span>
<span id="cb52-793"><a href="#cb52-793" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>( <span class="st">"GO Pathway Enrichment"</span> ) </span>
<span id="cb52-794"><a href="#cb52-794" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb52-795"><a href="#cb52-795" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-796"><a href="#cb52-796" aria-hidden="true" tabindex="-1"></a><span class="fu">## GSEA</span></span>
<span id="cb52-797"><a href="#cb52-797" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-798"><a href="#cb52-798" aria-hidden="true" tabindex="-1"></a><span class="al">![GSEA](https://img-blog.csdnimg.cn/20181105104509344.gif)</span>{fig-align="center"}</span>
<span id="cb52-799"><a href="#cb52-799" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-800"><a href="#cb52-800" aria-hidden="true" tabindex="-1"></a><span class="al">![](data/GSEA.gif)</span>{fig-align="center"}</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>