<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.32">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>2&nbsp; 生物数据结构 – Bioinformatics</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./annotation_database.html" rel="next">
<link href="./upstream_analysis.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-2486e1f0a3ee9ee1fc393803a1361cdb.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-f90a1b115807ea56b20e50efb3bb25e2.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
</head>
<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./biodata_structure.html">数据结构与数据库</a></li><li class="breadcrumb-item"><a href="./biodata_structure.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">生物数据结构</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Bioinformatics</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/Wanganlin00/Bioinformatics" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">前言</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./upstream_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">上游分析</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">数据结构与数据库</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./biodata_structure.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">生物数据结构</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./annotation_database.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">注释数据库</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ExpressionSet.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title"><code>ExpressionSet</code></span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./SummarizedExperiment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title"><code>SummarizedExperiment</code></span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./SingleCellExperiment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title"><code>SingleCellExperiment</code></span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">下游分析</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./DEGs_Analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">差异表达基因分析</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./EnrichmentAnalysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">富集分析</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PseudotimeAnalysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">伪时序分析</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">单细胞分析</span></span>
  </li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li>
<a href="#s4-%E7%B1%BB" id="toc-s4-类" class="nav-link active" data-scroll-target="#s4-%E7%B1%BB"><span class="header-section-number">2.1</span> S4 类</a>
  <ul>
<li><a href="#dataframe" id="toc-dataframe" class="nav-link" data-scroll-target="#dataframe"><span class="header-section-number">2.1.1</span> DataFrame</a></li>
  <li><a href="#rle" id="toc-rle" class="nav-link" data-scroll-target="#rle"><span class="header-section-number">2.1.2</span> Rle</a></li>
  </ul>
</li>
  <li>
<a href="#biostrings" id="toc-biostrings" class="nav-link" data-scroll-target="#biostrings"><span class="header-section-number">2.2</span> Biostrings</a>
  <ul>
<li><a href="#dna" id="toc-dna" class="nav-link" data-scroll-target="#dna"><span class="header-section-number">2.2.1</span> DNA</a></li>
  <li><a href="#rna" id="toc-rna" class="nav-link" data-scroll-target="#rna"><span class="header-section-number">2.2.2</span> RNA</a></li>
  <li><a href="#%E8%9B%8B%E7%99%BD%E8%B4%A8" id="toc-蛋白质" class="nav-link" data-scroll-target="#%E8%9B%8B%E7%99%BD%E8%B4%A8"><span class="header-section-number">2.2.3</span> 蛋白质</a></li>
  </ul>
</li>
  <li>
<a href="#ranges" id="toc-ranges" class="nav-link" data-scroll-target="#ranges"><span class="header-section-number">2.3</span> Ranges</a>
  <ul>
<li><a href="#intervalranges" id="toc-intervalranges" class="nav-link" data-scroll-target="#intervalranges"><span class="header-section-number">2.3.1</span> IntervalRanges</a></li>
  <li><a href="#genomicranges" id="toc-genomicranges" class="nav-link" data-scroll-target="#genomicranges"><span class="header-section-number">2.3.2</span> GenomicRanges</a></li>
  </ul>
</li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/Wanganlin00/Bioinformatics/edit/main/biodata_structure.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/Wanganlin00/Bioinformatics/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./biodata_structure.html">数据结构与数据库</a></li><li class="breadcrumb-item"><a href="./biodata_structure.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">生物数据结构</span></a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">
<span class="chapter-number">2</span>&nbsp; <span class="chapter-title">生物数据结构</span>
</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><p><a href="https://bioconductor.org/packages/release/workflows/html/sequencing.html" class="uri">https://bioconductor.org/packages/release/workflows/html/sequencing.html</a></p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="images/sequence.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<section id="s4-类" class="level2" data-number="2.1"><h2 data-number="2.1" class="anchored" data-anchor-id="s4-类">
<span class="header-section-number">2.1</span> S4 类</h2>
<section id="dataframe" class="level3" data-number="2.1.1"><h3 data-number="2.1.1" class="anchored" data-anchor-id="dataframe">
<span class="header-section-number">2.1.1</span> DataFrame</h3>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/S4Vectors">S4Vectors</a></span><span class="op">)</span></span>
<span><span class="fu">conflicts_prefer</span><span class="op">(</span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="va">setdiff</span><span class="op">)</span></span>
<span><span class="co"># 创建一个 DataFrame 对象</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html">DataFrame</a></span><span class="op">(</span>a <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, b <span class="op">=</span> <span class="va">letters</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">df</span></span>
<span><span class="co">#&gt; DataFrame with 3 rows and 2 columns</span></span>
<span><span class="co">#&gt;           a           b</span></span>
<span><span class="co">#&gt;   &lt;integer&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; 1         1           a</span></span>
<span><span class="co">#&gt; 2         2           b</span></span>
<span><span class="co">#&gt; 3         3           c</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="rle" class="level3" data-number="2.1.2"><h3 data-number="2.1.2" class="anchored" data-anchor-id="rle">
<span class="header-section-number">2.1.2</span> Rle</h3>
<p><strong>Run-Length Encoding (RLE)</strong> 是一种简单且常用的数据压缩方法，特别适用于存储和处理重复数据序列。<code><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html">class?Rle</a></code></p>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># 创建一个 Rle 对象</span></span>
<span></span>
<span><span class="va">rle_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html">Rle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="fl">5</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"B"</span>, <span class="fl">3</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"C"</span>, <span class="fl">2</span><span class="op">)</span>, <span class="st">"D"</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">rle_obj</span><span class="op">)</span></span>
<span><span class="co">#&gt; character-Rle of length 13 with 5 runs</span></span>
<span><span class="co">#&gt;   Lengths:   5   3   2   1   2</span></span>
<span><span class="co">#&gt;   Values : "A" "B" "C" "D" "A"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html">runLength</a></span><span class="op">(</span><span class="va">rle_obj</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 5 3 2 1 2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>在 RLE 中，连续的重复元素（称为“运行”）被压缩成一个值和一个计数对。这个值表示重复的元素，计数表示它们的数量。例如，序列 <code>[AAAAABBBCCDAA]</code> 可以被编码为 <code>(A5, B3, C2, D1, A2)</code>。</p>
</section></section><section id="biostrings" class="level2" data-number="2.2"><h2 data-number="2.2" class="anchored" data-anchor-id="biostrings">
<span class="header-section-number">2.2</span> Biostrings</h2>
<p>Biological strings</p>
<p><code>DNAstringSet</code>、<code>RNAStringSet</code>、<code>AAStringSet</code>、<code>BStringSet</code></p>
<section id="dna" class="level3" data-number="2.2.1"><h3 data-number="2.2.1" class="anchored" data-anchor-id="dna">
<span class="header-section-number">2.2.1</span> DNA</h3>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># BiocManager::install("Biostrings")</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings">Biostrings</a></span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">Biostrings</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/Biostrings/man/DNAString-class.html">DNA_ALPHABET</a></span></span>
<span><span class="co">#&gt;  [1] "A" "C" "G" "T" "M" "R" "W" "S" "Y" "K" "V" "H" "D" "B" "N" "-" "+" "."</span></span>
<span><span class="va">IUPAC_CODE_MAP</span></span>
<span><span class="co">#&gt;      A      C      G      T      M      R      W      S      Y      K      V </span></span>
<span><span class="co">#&gt;    "A"    "C"    "G"    "T"   "AC"   "AG"   "AT"   "CG"   "CT"   "GT"  "ACG" </span></span>
<span><span class="co">#&gt;      H      D      B      N </span></span>
<span><span class="co">#&gt;  "ACT"  "AGT"  "CGT" "ACGT"</span></span>
<span></span>
<span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html">DNAStringSet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TGCACGTGCATT"</span>,<span class="st">"ACTGCA"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">d</span></span>
<span><span class="co">#&gt; DNAStringSet object of length 2:</span></span>
<span><span class="co">#&gt;     width seq</span></span>
<span><span class="co">#&gt; [1]    12 TGCACGTGCATT</span></span>
<span><span class="co">#&gt; [2]     6 ACTGCA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 2</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html">width</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 12  6</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">rev</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span>
<span><span class="co">#&gt; DNAStringSet object of length 2:</span></span>
<span><span class="co">#&gt;     width seq</span></span>
<span><span class="co">#&gt; [1]     6 ACTGCA</span></span>
<span><span class="co">#&gt; [2]    12 TGCACGTGCATT</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/reverse-methods.html">reverse</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span>
<span><span class="co">#&gt; DNAStringSet object of length 2:</span></span>
<span><span class="co">#&gt;     width seq</span></span>
<span><span class="co">#&gt; [1]    12 TTACGTGCACGT</span></span>
<span><span class="co">#&gt; [2]     6 ACGTCA</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html">reverseComplement</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span>
<span><span class="co">#&gt; DNAStringSet object of length 2:</span></span>
<span><span class="co">#&gt;     width seq</span></span>
<span><span class="co">#&gt; [1]    12 AATGCACGTGCA</span></span>
<span><span class="co">#&gt; [2]     6 TGCAGT</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/translate.html">translate</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span>
<span><span class="co">#&gt; AAStringSet object of length 2:</span></span>
<span><span class="co">#&gt;     width seq</span></span>
<span><span class="co">#&gt; [1]     4 CTCI</span></span>
<span><span class="co">#&gt; [2]     2 TA</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/letterFrequency.html">alphabetFrequency</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span>
<span><span class="co">#&gt;      A C G T M R W S Y K V H D B N - + .</span></span>
<span><span class="co">#&gt; [1,] 2 3 3 4 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span><span class="co">#&gt; [2,] 2 2 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/letterFrequency.html">letterFrequency</a></span><span class="op">(</span><span class="va">d</span>,letters <span class="op">=</span> <span class="st">"GC"</span><span class="op">)</span></span>
<span><span class="co">#&gt;      G|C</span></span>
<span><span class="co">#&gt; [1,]   6</span></span>
<span><span class="co">#&gt; [2,]   3</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/nucleotideFrequency.html">dinucleotideFrequency</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span>
<span><span class="co">#&gt;      AA AC AG AT CA CC CG CT GA GC GG GT TA TC TG TT</span></span>
<span><span class="co">#&gt; [1,]  0  1  0  1  2  0  1  0  0  2  0  1  0  0  2  1</span></span>
<span><span class="co">#&gt; [2,]  0  1  0  0  1  0  0  1  0  1  0  0  0  0  1  0</span></span>
<span></span>
<span><span class="co"># 共识矩阵</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/letterFrequency.html">consensusMatrix</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span>
<span><span class="co">#&gt;   [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10] [,11] [,12]</span></span>
<span><span class="co">#&gt; A    1    0    0    1    0    1    0    0    0     1     0     0</span></span>
<span><span class="co">#&gt; C    0    1    1    0    2    0    0    0    1     0     0     0</span></span>
<span><span class="co">#&gt; G    0    1    0    1    0    1    0    1    0     0     0     0</span></span>
<span><span class="co">#&gt; T    1    0    1    0    0    0    1    0    0     0     1     1</span></span>
<span><span class="co">#&gt; M    0    0    0    0    0    0    0    0    0     0     0     0</span></span>
<span><span class="co">#&gt; R    0    0    0    0    0    0    0    0    0     0     0     0</span></span>
<span><span class="co">#&gt; W    0    0    0    0    0    0    0    0    0     0     0     0</span></span>
<span><span class="co">#&gt; S    0    0    0    0    0    0    0    0    0     0     0     0</span></span>
<span><span class="co">#&gt; Y    0    0    0    0    0    0    0    0    0     0     0     0</span></span>
<span><span class="co">#&gt; K    0    0    0    0    0    0    0    0    0     0     0     0</span></span>
<span><span class="co">#&gt; V    0    0    0    0    0    0    0    0    0     0     0     0</span></span>
<span><span class="co">#&gt; H    0    0    0    0    0    0    0    0    0     0     0     0</span></span>
<span><span class="co">#&gt; D    0    0    0    0    0    0    0    0    0     0     0     0</span></span>
<span><span class="co">#&gt; B    0    0    0    0    0    0    0    0    0     0     0     0</span></span>
<span><span class="co">#&gt; N    0    0    0    0    0    0    0    0    0     0     0     0</span></span>
<span><span class="co">#&gt; -    0    0    0    0    0    0    0    0    0     0     0     0</span></span>
<span><span class="co">#&gt; +    0    0    0    0    0    0    0    0    0     0     0     0</span></span>
<span><span class="co">#&gt; .    0    0    0    0    0    0    0    0    0     0     0     0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>从 Ensembl 的FASTA文件’Homo_sapiens.GRCh38.cdna.all.fa’中下载所有智人cDNA序列</p>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">AnnotationHub</a></span><span class="op">(</span>cache <span class="op">=</span> <span class="st">"D:/AnnotationHub"</span><span class="op">)</span></span>
<span><span class="va">ah</span></span>
<span><span class="va">ah2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"fasta"</span>, <span class="st">"homo sapiens"</span>, <span class="st">"Ensembl"</span>, <span class="st">"cdna"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ah2</span></span>
<span></span>
<span><span class="co"># 下载为TwoBitFile文件</span></span>
<span><span class="va">dna</span> <span class="op">&lt;-</span> <span class="va">ah2</span><span class="op">[[</span><span class="st">"AH68262"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="va">dna</span><span class="op">@</span><span class="va">resource</span></span>
<span><span class="co">#&gt; [1]"C:\\Users\\WANGAN~1\\AppData\\Local\\Temp\\RtmpeUSnD9/BiocFileCache/40e014f2123b_75008"</span></span>
<span></span>
<span><span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html">getSeq</a></span><span class="op">(</span><span class="va">dna</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span></span>
<span><span class="va">dna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/TwoBitFile-class.html">import.2bit</a></span><span class="op">(</span>con <span class="op">=</span> <span class="st">"C:\\Users\\WANGAN~1\\AppData\\Local\\Temp\\RtmpeUSnD9/BiocFileCache/40e014f2123b_75008"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dna</span></span>
<span><span class="co">#&gt; DNAStringSet object of length 187626:</span></span>
<span><span class="co">#&gt;           width seq                                         names               </span></span>
<span><span class="co">#&gt;      [1]     12 GGGACAGGGGGC                                ENST00000632684.1</span></span>
<span><span class="co">#&gt;      [2]      9 CCTTCCTAC                                   ENST00000434970.2</span></span>
<span><span class="co">#&gt;      [3]     13 ACTGGGGGATACG                               ENST00000448914.1</span></span>
<span><span class="co">#&gt;      [4]      8 GAAATAGT                                    ENST00000415118.1</span></span>
<span><span class="co">#&gt;      [5]     12 GGGACAGGGGGC                                ENST00000631435.1</span></span>
<span><span class="co">#&gt;      ...    ... ...</span></span>
<span><span class="co">#&gt; [187622]   1370 GGCTGAGTCTGGGCCCCAGG...AAGATGCAGCCGGGAGGTGA ENST00000639790.1</span></span>
<span><span class="co">#&gt; [187623]    284 GGCGTCTACAAGAGACCTTC...GCCCTAGATGACCACTGTTA ENST00000639660.1</span></span>
<span><span class="co">#&gt; [187624]    105 TGCATCACTCTGGCATTGAC...GTGGACCCCAGAAAGTTAAT ENST00000643577.1</span></span>
<span><span class="co">#&gt; [187625]    900 ATGGGATGTCACCAATCAAT...GGAAGGAGAGGCTGATGTAA ENST00000646356.1</span></span>
<span><span class="co">#&gt; [187626]    930 ATGGGAGTCAACCAATCATG...TGCAGAGGACGCTGTCTATG ENST00000645792.1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="rna" class="level3" data-number="2.2.2"><h3 data-number="2.2.2" class="anchored" data-anchor-id="rna">
<span class="header-section-number">2.2.2</span> RNA</h3>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">Biostrings</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/Biostrings/man/GENETIC_CODE.html">GENETIC_CODE</a></span></span>
<span><span class="co">#&gt; TTT TTC TTA TTG TCT TCC TCA TCG TAT TAC TAA TAG TGT TGC TGA TGG CTT CTC CTA CTG </span></span>
<span><span class="co">#&gt; "F" "F" "L" "L" "S" "S" "S" "S" "Y" "Y" "*" "*" "C" "C" "*" "W" "L" "L" "L" "L" </span></span>
<span><span class="co">#&gt; CCT CCC CCA CCG CAT CAC CAA CAG CGT CGC CGA CGG ATT ATC ATA ATG ACT ACC ACA ACG </span></span>
<span><span class="co">#&gt; "P" "P" "P" "P" "H" "H" "Q" "Q" "R" "R" "R" "R" "I" "I" "I" "M" "T" "T" "T" "T" </span></span>
<span><span class="co">#&gt; AAT AAC AAA AAG AGT AGC AGA AGG GTT GTC GTA GTG GCT GCC GCA GCG GAT GAC GAA GAG </span></span>
<span><span class="co">#&gt; "N" "N" "K" "K" "S" "S" "R" "R" "V" "V" "V" "V" "A" "A" "A" "A" "D" "D" "E" "E" </span></span>
<span><span class="co">#&gt; GGT GGC GGA GGG </span></span>
<span><span class="co">#&gt; "G" "G" "G" "G" </span></span>
<span><span class="co">#&gt; attr(,"alt_init_codons")</span></span>
<span><span class="co">#&gt; [1] "TTG" "CTG"</span></span>
<span><span class="fu">Biostrings</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/Biostrings/man/GENETIC_CODE.html">RNA_GENETIC_CODE</a></span></span>
<span><span class="co">#&gt; UUU UUC UUA UUG UCU UCC UCA UCG UAU UAC UAA UAG UGU UGC UGA UGG CUU CUC CUA CUG </span></span>
<span><span class="co">#&gt; "F" "F" "L" "L" "S" "S" "S" "S" "Y" "Y" "*" "*" "C" "C" "*" "W" "L" "L" "L" "L" </span></span>
<span><span class="co">#&gt; CCU CCC CCA CCG CAU CAC CAA CAG CGU CGC CGA CGG AUU AUC AUA AUG ACU ACC ACA ACG </span></span>
<span><span class="co">#&gt; "P" "P" "P" "P" "H" "H" "Q" "Q" "R" "R" "R" "R" "I" "I" "I" "M" "T" "T" "T" "T" </span></span>
<span><span class="co">#&gt; AAU AAC AAA AAG AGU AGC AGA AGG GUU GUC GUA GUG GCU GCC GCA GCG GAU GAC GAA GAG </span></span>
<span><span class="co">#&gt; "N" "N" "K" "K" "S" "S" "R" "R" "V" "V" "V" "V" "A" "A" "A" "A" "D" "D" "E" "E" </span></span>
<span><span class="co">#&gt; GGU GGC GGA GGG </span></span>
<span><span class="co">#&gt; "G" "G" "G" "G" </span></span>
<span><span class="co">#&gt; attr(,"alt_init_codons")</span></span>
<span><span class="co">#&gt; [1] "UUG" "CUG"</span></span>
<span><span class="va">RNA_ALPHABET</span></span>
<span><span class="co">#&gt;  [1] "A" "C" "G" "U" "M" "R" "W" "S" "Y" "K" "V" "H" "D" "B" "N" "-" "+" "."</span></span>
<span><span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html">RNAStringSet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"AUCG"</span>, <span class="st">"GCAU"</span>, <span class="st">"AUCGAU"</span>, <span class="st">"GCUA"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">r</span></span>
<span><span class="co">#&gt; RNAStringSet object of length 4:</span></span>
<span><span class="co">#&gt;     width seq</span></span>
<span><span class="co">#&gt; [1]     4 AUCG</span></span>
<span><span class="co">#&gt; [2]     4 GCAU</span></span>
<span><span class="co">#&gt; [3]     6 AUCGAU</span></span>
<span><span class="co">#&gt; [4]     4 GCUA</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="蛋白质" class="level3" data-number="2.2.3"><h3 data-number="2.2.3" class="anchored" data-anchor-id="蛋白质">
<span class="header-section-number">2.2.3</span> 蛋白质</h3>
<p>组成人体蛋白质的21种氨基酸，20 + 硒代半胱氨酸（UGA-终止密码子）</p>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">Biostrings</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/Biostrings/man/AMINO_ACID_CODE.html">AMINO_ACID_CODE</a></span></span>
<span><span class="co">#&gt;     A     R     N     D     C     Q     E     G     H     I     L     K     M </span></span>
<span><span class="co">#&gt; "Ala" "Arg" "Asn" "Asp" "Cys" "Gln" "Glu" "Gly" "His" "Ile" "Leu" "Lys" "Met" </span></span>
<span><span class="co">#&gt;     F     P     S     T     W     Y     V     U     O     B     J     Z     X </span></span>
<span><span class="co">#&gt; "Phe" "Pro" "Ser" "Thr" "Trp" "Tyr" "Val" "Sec" "Pyl" "Asx" "Xle" "Glx" "Xaa"</span></span>
<span><span class="va">AA_ALPHABET</span></span>
<span><span class="co">#&gt;  [1] "A" "R" "N" "D" "C" "Q" "E" "G" "H" "I" "L" "K" "M" "F" "P" "S" "T" "W" "Y"</span></span>
<span><span class="co">#&gt; [20] "V" "U" "O" "B" "J" "Z" "X" "*" "-" "+" "."</span></span>
<span></span>
<span><span class="va">protein_strings</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html">AAStringSet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"MATH"</span>, <span class="st">"GCAU"</span>, <span class="st">"MATHMATH"</span>, <span class="st">"CUMA"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">protein_strings</span></span>
<span><span class="co">#&gt; AAStringSet object of length 4:</span></span>
<span><span class="co">#&gt;     width seq</span></span>
<span><span class="co">#&gt; [1]     4 MATH</span></span>
<span><span class="co">#&gt; [2]     4 GCAU</span></span>
<span><span class="co">#&gt; [3]     8 MATHMATH</span></span>
<span><span class="co">#&gt; [4]     4 CUMA</span></span>
<span></span>
<span><span class="co"># 二进制字符串存储 如seq_id、quality</span></span>
<span><span class="va">binary_strings</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html">BStringSet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ERCC010101"</span>, <span class="st">"CRCC110011"</span>, <span class="st">"101010"</span>, <span class="st">"111000"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">binary_strings</span></span>
<span><span class="co">#&gt; BStringSet object of length 4:</span></span>
<span><span class="co">#&gt;     width seq</span></span>
<span><span class="co">#&gt; [1]    10 ERCC010101</span></span>
<span><span class="co">#&gt; [2]    10 CRCC110011</span></span>
<span><span class="co">#&gt; [3]     6 101010</span></span>
<span><span class="co">#&gt; [4]     6 111000</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section></section><section id="ranges" class="level2" data-number="2.3"><h2 data-number="2.3" class="anchored" data-anchor-id="ranges">
<span class="header-section-number">2.3</span> Ranges</h2>
<section id="intervalranges" class="level3" data-number="2.3.1"><h3 data-number="2.3.1" class="anchored" data-anchor-id="intervalranges">
<span class="header-section-number">2.3.1</span> IntervalRanges</h3>
<p><code>IRanges</code></p>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/IRanges">IRanges</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">ranges</span> <span class="op">&lt;-</span> <span class="fu">IRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span></span>
<span>  start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">1</span>, <span class="fl">100</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  width <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">0</span>, <span class="fl">50</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"exton_"</span>, <span class="va">letters</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">26</span>, <span class="fl">10</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">ranges</span></span>
<span><span class="co">#&gt; IRanges object with 10 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;               start       end     width</span></span>
<span><span class="co">#&gt;           &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span></span>
<span><span class="co">#&gt;   exton_h        51        83        33</span></span>
<span><span class="co">#&gt;   exton_n        31        58        28</span></span>
<span><span class="co">#&gt;   exton_g        43        48         6</span></span>
<span><span class="co">#&gt;   exton_f        70        99        30</span></span>
<span><span class="co">#&gt;   exton_x         9        26        18</span></span>
<span><span class="co">#&gt;   exton_r        23        43        21</span></span>
<span><span class="co">#&gt;   exton_u        28        30         3</span></span>
<span><span class="co">#&gt;   exton_m        28        40        13</span></span>
<span><span class="co">#&gt;   exton_e        62        81        20</span></span>
<span><span class="co">#&gt;   exton_a        44        85        42</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="va">ranges</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] 51 31 43 70  9 23 28 28 62 44</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html">width</a></span><span class="op">(</span><span class="va">ranges</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] 33 28  6 30 18 21  3 13 20 42</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/start.html">end</a></span><span class="op">(</span><span class="va">ranges</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] 83 58 48 99 26 43 30 40 81 85</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>没有维度，但有长度</p>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">ranges</span><span class="op">)</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">ranges</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 10</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>重叠检测: 检查不同区间是否有重叠。</p>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">subject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">15</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">overlaps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html">findOverlaps</a></span><span class="op">(</span><span class="va">query</span>, <span class="va">subject</span><span class="op">)</span></span>
<span><span class="va">overlaps</span></span>
<span><span class="co">#&gt; Hits object with 1 hit and 0 metadata columns:</span></span>
<span><span class="co">#&gt;       queryHits subjectHits</span></span>
<span><span class="co">#&gt;       &lt;integer&gt;   &lt;integer&gt;</span></span>
<span><span class="co">#&gt;   [1]         1           1</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   queryLength: 1 / subjectLength: 2</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html">queryHits</a></span><span class="op">(</span><span class="va">overlaps</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>合并区间: 将重叠或相邻的区间合并为一个更大的区间。</p>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">merged</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html">reduce</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">5</span><span class="op">)</span>, end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">merged</span><span class="op">)</span></span>
<span><span class="co">#&gt; IRanges object with 1 range and 0 metadata columns:</span></span>
<span><span class="co">#&gt;           start       end     width</span></span>
<span><span class="co">#&gt;       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span></span>
<span><span class="co">#&gt;   [1]         1        20        20</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">plotRanges</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">xlim</span> <span class="op">=</span> <span class="va">x</span> , </span>
<span>                       <span class="va">main</span> <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/deparse.html">deparse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/substitute.html">substitute</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       <span class="va">col</span> <span class="op">=</span> <span class="st">"black"</span>, <span class="va">sep</span> <span class="op">=</span> <span class="fl">0.5</span>, <span class="va">...</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">height</span> <span class="op">=</span> <span class="fl">1</span></span>
<span>  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/methods/is.html">is</a></span><span class="op">(</span><span class="va">xlim</span>, class2 <span class="op">=</span> <span class="st">"Ranges"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="va">xlim</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="va">xlim</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html">end</a></span><span class="op">(</span><span class="va">xlim</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">bins</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html">disjointBins</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/stats/start.html">end</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">+</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/frame.html">plot.new</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/plot.window.html">plot.window</a></span><span class="op">(</span><span class="va">xlim</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bins</span><span class="op">)</span><span class="op">*</span><span class="op">(</span><span class="va">height</span> <span class="op">+</span> <span class="va">sep</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">ybottom</span> <span class="op">&lt;-</span> <span class="va">bins</span> <span class="op">*</span> <span class="op">(</span><span class="va">sep</span> <span class="op">+</span> <span class="va">height</span><span class="op">)</span> <span class="op">-</span> <span class="va">height</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/rect.html">rect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">-</span><span class="fl">0.5</span>, <span class="va">ybottom</span>, <span class="fu"><a href="https://rdrr.io/r/stats/start.html">end</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">+</span><span class="fl">0.5</span>, <span class="va">ybottom</span> <span class="op">+</span> <span class="va">height</span>,</span>
<span>       col <span class="op">=</span> <span class="va">col</span>, <span class="va">...</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span><span class="op">(</span><span class="va">main</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span>  </span>
<span><span class="op">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">ir</span> <span class="op">&lt;-</span> <span class="fu">IRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span></span>
<span>  start <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">4</span>, <span class="fl">10</span><span class="op">)</span>,</span>
<span>  end <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">6</span>, <span class="fl">3</span>, <span class="fl">8</span>, <span class="fl">10</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">ir</span></span>
<span><span class="co">#&gt; IRanges object with 4 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;           start       end     width</span></span>
<span><span class="co">#&gt;       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span></span>
<span><span class="co">#&gt;   [1]         1         6         6</span></span>
<span><span class="co">#&gt;   [2]         1         3         3</span></span>
<span><span class="co">#&gt;   [3]         4         8         5</span></span>
<span><span class="co">#&gt;   [4]        10        10         1</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html">disjoin</a></span><span class="op">(</span><span class="va">ir</span><span class="op">)</span></span>
<span><span class="co">#&gt; IRanges object with 4 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;           start       end     width</span></span>
<span><span class="co">#&gt;       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span></span>
<span><span class="co">#&gt;   [1]         1         3         3</span></span>
<span><span class="co">#&gt;   [2]         4         6         3</span></span>
<span><span class="co">#&gt;   [3]         7         8         2</span></span>
<span><span class="co">#&gt;   [4]        10        10         1</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html">disjointBins</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="va">ir</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/stats/start.html">end</a></span><span class="op">(</span><span class="va">ir</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="co"># 1,2,3,1 放于第1，2，3，1层</span></span>
<span><span class="co">#&gt; [1] 1 2 3 1</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html">disjointBins</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html">disjoin</a></span><span class="op">(</span><span class="va">ir</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/stats/start.html">end</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html">disjoin</a></span><span class="op">(</span><span class="va">ir</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="co">#放于第1，2，1，1层</span></span>
<span><span class="co">#&gt; [1] 1 2 1 1</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">plotRanges</span><span class="op">(</span><span class="va">ir</span><span class="op">)</span></span>
<span><span class="fu">plotRanges</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html">disjoin</a></span><span class="op">(</span><span class="va">ir</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="biodata_structure_files/figure-html/unnamed-chunk-13-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section><section id="genomicranges" class="level3" data-number="2.3.2"><h3 data-number="2.3.2" class="anchored" data-anchor-id="genomicranges">
<span class="header-section-number">2.3.2</span> GenomicRanges</h3>
<p><strong><code>GRanges</code></strong>：用于表示基因组（染色体）范围的数据结构，例如Promoters，Genes，SNPs，CpG Islands，……</p>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></span><span class="op">)</span></span>
<span><span class="co"># help("GRanges-class")</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">Granges</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span></span>
<span>  seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html">Rle</a></span><span class="op">(</span> values <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'chr1'</span>, <span class="st">'chr2'</span>, <span class="st">'chr3'</span><span class="op">)</span>, lengths <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">18</span>,<span class="fl">2</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>,</span>
<span>  strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"+"</span>,<span class="st">"-"</span>,<span class="st">"*"</span><span class="op">)</span>,each<span class="op">=</span><span class="fl">3</span><span class="op">)</span>,</span>
<span>  score <span class="op">=</span> <span class="fl">101</span><span class="op">:</span><span class="fl">109</span>,</span>
<span>  GC <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span><span class="op">(</span><span class="fl">9</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">Granges</span></span>
<span><span class="co">#&gt; GRanges object with 9 ranges and 2 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames    ranges strand |     score        GC</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1]     chr1       1-3      + |       101  0.507478</span></span>
<span><span class="co">#&gt;   [2]     chr1       3-5      + |       102  0.306769</span></span>
<span><span class="co">#&gt;   [3]     chr1       5-7      + |       103  0.426908</span></span>
<span><span class="co">#&gt;   [4]     chr2       7-9      - |       104  0.693102</span></span>
<span><span class="co">#&gt;   [5]     chr2      9-11      - |       105  0.085136</span></span>
<span><span class="co">#&gt;   [6]     chr2     11-13      - |       106  0.225437</span></span>
<span><span class="co">#&gt;   [7]     chr3     13-15      * |       107  0.274531</span></span>
<span><span class="co">#&gt;   [8]     chr3     15-17      * |       108  0.272305</span></span>
<span><span class="co">#&gt;   [9]     chr3     17-19      * |       109  0.615829</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html">sort</a></span><span class="op">(</span><span class="va">Granges</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 9 ranges and 2 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames    ranges strand |     score        GC</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1]     chr1       1-3      + |       101  0.507478</span></span>
<span><span class="co">#&gt;   [2]     chr1       3-5      + |       102  0.306769</span></span>
<span><span class="co">#&gt;   [3]     chr1       5-7      + |       103  0.426908</span></span>
<span><span class="co">#&gt;   [4]     chr2       7-9      - |       104  0.693102</span></span>
<span><span class="co">#&gt;   [5]     chr2      9-11      - |       105  0.085136</span></span>
<span><span class="co">#&gt;   [6]     chr2     11-13      - |       106  0.225437</span></span>
<span><span class="co">#&gt;   [7]     chr3     13-15      * |       107  0.274531</span></span>
<span><span class="co">#&gt;   [8]     chr3     15-17      * |       108  0.272305</span></span>
<span><span class="co">#&gt;   [9]     chr3     17-19      * |       109  0.615829</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">values</a></span><span class="op">(</span><span class="va">Granges</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html">DataFrame</a></span><span class="op">(</span>score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">9</span><span class="op">)</span>, GC <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span><span class="op">(</span><span class="fl">9</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">Granges</span></span>
<span><span class="co">#&gt; GRanges object with 9 ranges and 2 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames    ranges strand |     score        GC</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1]     chr1       1-3      + | -0.177211 0.2395891</span></span>
<span><span class="co">#&gt;   [2]     chr1       3-5      + |  0.170618 0.7707715</span></span>
<span><span class="co">#&gt;   [3]     chr1       5-7      + |  0.242814 0.3558977</span></span>
<span><span class="co">#&gt;   [4]     chr2       7-9      - | -0.179406 0.5355970</span></span>
<span><span class="co">#&gt;   [5]     chr2      9-11      - | -0.630519 0.0930881</span></span>
<span><span class="co">#&gt;   [6]     chr2     11-13      - |  0.978693 0.1698030</span></span>
<span><span class="co">#&gt;   [7]     chr3     13-15      * |  0.293297 0.8998325</span></span>
<span><span class="co">#&gt;   [8]     chr3     15-17      * | -0.370329 0.4226376</span></span>
<span><span class="co">#&gt;   [9]     chr3     17-19      * |  0.543615 0.7477465</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><code>flank(x, width, start = TRUE, both = FALSE)</code> 用于在基因组范围对象 (GRanges) 上扩展或调整范围。它会基于输入的基因组范围返回其上游或下游的邻近区域。flanking sequence</p>
<ul>
<li>
<code>x</code>: 一个 GRanges 对象。</li>
<li>
<code>width</code>: 你想要的上下游区域的宽度（如上例中的 5）。</li>
<li>
<code>start</code>: 这个参数决定了是否使用起始端（TRUE）还是结束端（FALSE）来生成上下游区域。如果你的基因组是正链，start = TRUE 表示上游。如果是负链，start = TRUE 则表示下游。默认为 TRUE。</li>
<li>
<code>both</code>: 如果设置为 TRUE，则会在范围的两端生成 width 宽度的区域。</li>
</ul>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html">flank</a></span><span class="op">(</span><span class="va">Granges</span>,<span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 9 ranges and 2 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames    ranges strand |     score        GC</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1]     chr1      -4-0      + | -0.177211 0.2395891</span></span>
<span><span class="co">#&gt;   [2]     chr1      -2-2      + |  0.170618 0.7707715</span></span>
<span><span class="co">#&gt;   [3]     chr1       0-4      + |  0.242814 0.3558977</span></span>
<span><span class="co">#&gt;   [4]     chr2     10-14      - | -0.179406 0.5355970</span></span>
<span><span class="co">#&gt;   [5]     chr2     12-16      - | -0.630519 0.0930881</span></span>
<span><span class="co">#&gt;   [6]     chr2     14-18      - |  0.978693 0.1698030</span></span>
<span><span class="co">#&gt;   [7]     chr3      8-12      * |  0.293297 0.8998325</span></span>
<span><span class="co">#&gt;   [8]     chr3     10-14      * | -0.370329 0.4226376</span></span>
<span><span class="co">#&gt;   [9]     chr3     12-16      * |  0.543615 0.7477465</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><code>promoters(x, upstream = 2000, downstream = 200)</code> 通常用于提取基因的启动子区域（promoter regions）。在基因组分析中，启动子区域是靠近基因起始位点的区域，通常被认为是调控基因表达的关键区域。启动子区域是基于基因组范围的起始位置提取的，上游（upstream）和下游（downstream）的长度可以自定义。</p>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html">promoters</a></span><span class="op">(</span><span class="va">Granges</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 9 ranges and 2 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames    ranges strand |     score        GC</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1]     chr1 -1999-200      + | -0.177211 0.2395891</span></span>
<span><span class="co">#&gt;   [2]     chr1 -1997-202      + |  0.170618 0.7707715</span></span>
<span><span class="co">#&gt;   [3]     chr1 -1995-204      + |  0.242814 0.3558977</span></span>
<span><span class="co">#&gt;   [4]     chr2 -190-2009      - | -0.179406 0.5355970</span></span>
<span><span class="co">#&gt;   [5]     chr2 -188-2011      - | -0.630519 0.0930881</span></span>
<span><span class="co">#&gt;   [6]     chr2 -186-2013      - |  0.978693 0.1698030</span></span>
<span><span class="co">#&gt;   [7]     chr3 -1987-212      * |  0.293297 0.8998325</span></span>
<span><span class="co">#&gt;   [8]     chr3 -1985-214      * | -0.370329 0.4226376</span></span>
<span><span class="co">#&gt;   [9]     chr3 -1983-216      * |  0.543615 0.7477465</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomeInfoDb">GenomeInfoDb</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html">genome</a></span><span class="op">(</span><span class="va">Granges</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"hg19"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html">seqinfo</a></span><span class="op">(</span><span class="va">Granges</span><span class="op">)</span></span>
<span><span class="co">#&gt; Seqinfo object with 3 sequences from hg19 genome; no seqlengths:</span></span>
<span><span class="co">#&gt;   seqnames seqlengths isCircular genome</span></span>
<span><span class="co">#&gt;   chr1             NA         NA   hg19</span></span>
<span><span class="co">#&gt;   chr2             NA         NA   hg19</span></span>
<span><span class="co">#&gt;   chr3             NA         NA   hg19</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html">seqlevels</a></span><span class="op">(</span><span class="va">Granges</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "chr1" "chr2" "chr3"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html">seqlengths</a></span><span class="op">(</span><span class="va">Granges</span><span class="op">)</span></span>
<span><span class="co">#&gt; chr1 chr2 chr3 </span></span>
<span><span class="co">#&gt;   NA   NA   NA</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><code>gaps(x)</code>用于计算给定的 GRanges 对象中缺失（gaps）的基因组区域。它返回的是不被输入的 GRanges 对象覆盖的范围（即空白区域），通常用于确定在某些染色体区域中没有基因或特征的部分。</p>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html">gaps</a></span><span class="op">(</span><span class="va">Granges</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames    ranges strand</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   [1]     chr2       1-6      -</span></span>
<span><span class="co">#&gt;   [2]     chr3      1-12      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 3 sequences from hg19 genome; no seqlengths</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkox.org/gggenes/">gggenes</a></span><span class="op">)</span></span>
<span><span class="va">Granges</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rownames_to_column</span><span class="op">(</span>var <span class="op">=</span> <span class="st">"gene"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax<span class="op">=</span> <span class="va">end</span>,y <span class="op">=</span><span class="va">seqnames</span>,fill <span class="op">=</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://wilkox.org/gggenes/reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span> <span class="va">seqnames</span>, scales <span class="op">=</span> <span class="st">"free"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://wilkox.org/gggenes/reference/theme_genes.html">theme_genes</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="biodata_structure_files/figure-html/unnamed-chunk-20-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span></span>
<span><span class="va">gr1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="st">"chr1"</span>, </span>
<span>               ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">20</span><span class="op">)</span>, end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">30</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">gr2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="st">"chr1"</span>, </span>
<span>               ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">25</span><span class="op">)</span>, end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">15</span>, <span class="fl">35</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">gr1</span></span>
<span><span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames    ranges strand</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   [1]     chr1      1-10      *</span></span>
<span><span class="co">#&gt;   [2]     chr1     20-30      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span>
<span><span class="va">gr2</span></span>
<span><span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames    ranges strand</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   [1]     chr1      5-15      *</span></span>
<span><span class="co">#&gt;   [2]     chr1     25-35      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span>
<span><span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html">setdiff</a></span><span class="op">(</span><span class="va">gr1</span>, <span class="va">gr2</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames    ranges strand</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   [1]     chr1       1-4      *</span></span>
<span><span class="co">#&gt;   [2]     chr1     20-24      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html">findOverlaps</a></span><span class="op">(</span><span class="va">gr1</span>,<span class="va">gr2</span><span class="op">)</span></span>
<span><span class="co">#&gt; Hits object with 2 hits and 0 metadata columns:</span></span>
<span><span class="co">#&gt;       queryHits subjectHits</span></span>
<span><span class="co">#&gt;       &lt;integer&gt;   &lt;integer&gt;</span></span>
<span><span class="co">#&gt;   [1]         1           1</span></span>
<span><span class="co">#&gt;   [2]         2           2</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   queryLength: 2 / subjectLength: 2</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gr1</span>, <span class="va">gr2</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames    ranges strand</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   [1]     chr1      1-10      *</span></span>
<span><span class="co">#&gt;   [2]     chr1     20-30      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->

</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/wanganlin00\.github\.io\/Bioinformatics\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./upstream_analysis.html" class="pagination-link" aria-label="上游分析">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">上游分析</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./annotation_database.html" class="pagination-link" aria-label="注释数据库">
        <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">注释数据库</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb22" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># 生物数据结构</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://bioconductor.org/packages/release/workflows/html/sequencing.html&gt;</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/sequence.png)</span>{fig-align="center"}</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="fu">## S4 类</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a><span class="fu">### DataFrame</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(S4Vectors)</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(S4Vectors<span class="sc">::</span>setdiff)</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a><span class="co"># 创建一个 DataFrame 对象</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(<span class="at">a =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">b =</span> letters[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>])</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>df</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a><span class="fu">### Rle</span></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>**Run-Length Encoding (RLE)** 是一种简单且常用的数据压缩方法，特别适用于存储和处理重复数据序列。<span class="in">`class?Rle`</span></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a><span class="co"># 创建一个 Rle 对象</span></span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a>rle_obj <span class="ot">&lt;-</span> <span class="fu">Rle</span>(<span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"A"</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">"B"</span>, <span class="dv">3</span>), <span class="fu">rep</span>(<span class="st">"C"</span>, <span class="dv">2</span>), <span class="st">"D"</span>, <span class="fu">rep</span>(<span class="st">"A"</span>, <span class="dv">2</span>)))</span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(rle_obj)</span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a><span class="fu">runLength</span>(rle_obj)</span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a>在 RLE 中，连续的重复元素（称为“运行”）被压缩成一个值和一个计数对。这个值表示重复的元素，计数表示它们的数量。例如，序列 <span class="in">`[AAAAABBBCCDAA]`</span> 可以被编码为 <span class="in">`(A5, B3, C2, D1, A2)`</span>。</span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a><span class="fu">## Biostrings</span></span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a>Biological strings</span>
<span id="cb22-42"><a href="#cb22-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-43"><a href="#cb22-43" aria-hidden="true" tabindex="-1"></a><span class="in">`DNAstringSet`</span>、<span class="in">`RNAStringSet`</span>、<span class="in">`AAStringSet`</span>、<span class="in">`BStringSet`</span></span>
<span id="cb22-44"><a href="#cb22-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-45"><a href="#cb22-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-46"><a href="#cb22-46" aria-hidden="true" tabindex="-1"></a><span class="fu">### DNA</span></span>
<span id="cb22-47"><a href="#cb22-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-50"><a href="#cb22-50" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-51"><a href="#cb22-51" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("Biostrings")</span></span>
<span id="cb22-52"><a href="#cb22-52" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings) <span class="sc">%&gt;%</span> <span class="fu">suppressMessages</span>()</span>
<span id="cb22-53"><a href="#cb22-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-54"><a href="#cb22-54" aria-hidden="true" tabindex="-1"></a>Biostrings<span class="sc">::</span>DNA_ALPHABET</span>
<span id="cb22-55"><a href="#cb22-55" aria-hidden="true" tabindex="-1"></a>IUPAC_CODE_MAP</span>
<span id="cb22-56"><a href="#cb22-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-57"><a href="#cb22-57" aria-hidden="true" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fu">DNAStringSet</span>(<span class="fu">c</span>(<span class="st">"TGCACGTGCATT"</span>,<span class="st">"ACTGCA"</span>))</span>
<span id="cb22-58"><a href="#cb22-58" aria-hidden="true" tabindex="-1"></a>d</span>
<span id="cb22-59"><a href="#cb22-59" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(d)</span>
<span id="cb22-60"><a href="#cb22-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-61"><a href="#cb22-61" aria-hidden="true" tabindex="-1"></a><span class="fu">width</span>(d)</span>
<span id="cb22-62"><a href="#cb22-62" aria-hidden="true" tabindex="-1"></a><span class="fu">rev</span>(d)</span>
<span id="cb22-63"><a href="#cb22-63" aria-hidden="true" tabindex="-1"></a><span class="fu">reverse</span>(d)</span>
<span id="cb22-64"><a href="#cb22-64" aria-hidden="true" tabindex="-1"></a><span class="fu">reverseComplement</span>(d)</span>
<span id="cb22-65"><a href="#cb22-65" aria-hidden="true" tabindex="-1"></a><span class="fu">translate</span>(d)</span>
<span id="cb22-66"><a href="#cb22-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-67"><a href="#cb22-67" aria-hidden="true" tabindex="-1"></a><span class="fu">alphabetFrequency</span>(d)</span>
<span id="cb22-68"><a href="#cb22-68" aria-hidden="true" tabindex="-1"></a><span class="fu">letterFrequency</span>(d,<span class="at">letters =</span> <span class="st">"GC"</span>)</span>
<span id="cb22-69"><a href="#cb22-69" aria-hidden="true" tabindex="-1"></a><span class="fu">dinucleotideFrequency</span>(d)</span>
<span id="cb22-70"><a href="#cb22-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-71"><a href="#cb22-71" aria-hidden="true" tabindex="-1"></a><span class="co"># 共识矩阵</span></span>
<span id="cb22-72"><a href="#cb22-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-73"><a href="#cb22-73" aria-hidden="true" tabindex="-1"></a><span class="fu">consensusMatrix</span>(d)</span>
<span id="cb22-74"><a href="#cb22-74" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-75"><a href="#cb22-75" aria-hidden="true" tabindex="-1"></a>从 Ensembl 的FASTA文件'Homo_sapiens.GRCh38.cdna.all.fa'中下载所有智人cDNA序列</span>
<span id="cb22-76"><a href="#cb22-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-77"><a href="#cb22-77" aria-hidden="true" tabindex="-1"></a><span class="in">```{r eval=FALSE}</span></span>
<span id="cb22-78"><a href="#cb22-78" aria-hidden="true" tabindex="-1"></a><span class="in">library(AnnotationHub) %&gt;% suppressMessages()</span></span>
<span id="cb22-79"><a href="#cb22-79" aria-hidden="true" tabindex="-1"></a><span class="in">ah &lt;- AnnotationHub(cache = "D:/AnnotationHub")</span></span>
<span id="cb22-80"><a href="#cb22-80" aria-hidden="true" tabindex="-1"></a><span class="in">ah</span></span>
<span id="cb22-81"><a href="#cb22-81" aria-hidden="true" tabindex="-1"></a><span class="in">ah2 &lt;- query(ah, c("fasta", "homo sapiens", "Ensembl", "cdna"))</span></span>
<span id="cb22-82"><a href="#cb22-82" aria-hidden="true" tabindex="-1"></a><span class="in">ah2</span></span>
<span id="cb22-83"><a href="#cb22-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-84"><a href="#cb22-84" aria-hidden="true" tabindex="-1"></a><span class="in"># 下载为TwoBitFile文件</span></span>
<span id="cb22-85"><a href="#cb22-85" aria-hidden="true" tabindex="-1"></a><span class="in">dna &lt;- ah2[["AH68262"]]</span></span>
<span id="cb22-86"><a href="#cb22-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-87"><a href="#cb22-87" aria-hidden="true" tabindex="-1"></a><span class="in">dna@resource</span></span>
<span id="cb22-88"><a href="#cb22-88" aria-hidden="true" tabindex="-1"></a><span class="in">#&gt; [1]"C:\\Users\\WANGAN~1\\AppData\\Local\\Temp\\RtmpeUSnD9/BiocFileCache/40e014f2123b_75008"</span></span>
<span id="cb22-89"><a href="#cb22-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-90"><a href="#cb22-90" aria-hidden="true" tabindex="-1"></a><span class="in">Biostrings::getSeq(dna)</span></span>
<span id="cb22-91"><a href="#cb22-91" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-92"><a href="#cb22-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-93"><a href="#cb22-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-96"><a href="#cb22-96" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-97"><a href="#cb22-97" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rtracklayer)</span>
<span id="cb22-98"><a href="#cb22-98" aria-hidden="true" tabindex="-1"></a>dna <span class="ot">&lt;-</span> <span class="fu">import.2bit</span>(<span class="at">con =</span> <span class="st">"C:</span><span class="sc">\\</span><span class="st">Users</span><span class="sc">\\</span><span class="st">WANGAN~1</span><span class="sc">\\</span><span class="st">AppData</span><span class="sc">\\</span><span class="st">Local</span><span class="sc">\\</span><span class="st">Temp</span><span class="sc">\\</span><span class="st">RtmpeUSnD9/BiocFileCache/40e014f2123b_75008"</span>)</span>
<span id="cb22-99"><a href="#cb22-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-100"><a href="#cb22-100" aria-hidden="true" tabindex="-1"></a>dna</span>
<span id="cb22-101"><a href="#cb22-101" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-102"><a href="#cb22-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-103"><a href="#cb22-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-104"><a href="#cb22-104" aria-hidden="true" tabindex="-1"></a><span class="fu">### RNA</span></span>
<span id="cb22-105"><a href="#cb22-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-108"><a href="#cb22-108" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-109"><a href="#cb22-109" aria-hidden="true" tabindex="-1"></a>Biostrings<span class="sc">::</span>GENETIC_CODE</span>
<span id="cb22-110"><a href="#cb22-110" aria-hidden="true" tabindex="-1"></a>Biostrings<span class="sc">::</span>RNA_GENETIC_CODE</span>
<span id="cb22-111"><a href="#cb22-111" aria-hidden="true" tabindex="-1"></a>RNA_ALPHABET</span>
<span id="cb22-112"><a href="#cb22-112" aria-hidden="true" tabindex="-1"></a>r <span class="ot">&lt;-</span> <span class="fu">RNAStringSet</span>(<span class="fu">c</span>(<span class="st">"AUCG"</span>, <span class="st">"GCAU"</span>, <span class="st">"AUCGAU"</span>, <span class="st">"GCUA"</span>))</span>
<span id="cb22-113"><a href="#cb22-113" aria-hidden="true" tabindex="-1"></a>r</span>
<span id="cb22-114"><a href="#cb22-114" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-115"><a href="#cb22-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-116"><a href="#cb22-116" aria-hidden="true" tabindex="-1"></a><span class="fu">### 蛋白质</span></span>
<span id="cb22-117"><a href="#cb22-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-118"><a href="#cb22-118" aria-hidden="true" tabindex="-1"></a>组成人体蛋白质的21种氨基酸，20 + 硒代半胱氨酸（UGA-终止密码子）</span>
<span id="cb22-119"><a href="#cb22-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-122"><a href="#cb22-122" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-123"><a href="#cb22-123" aria-hidden="true" tabindex="-1"></a>Biostrings<span class="sc">::</span>AMINO_ACID_CODE</span>
<span id="cb22-124"><a href="#cb22-124" aria-hidden="true" tabindex="-1"></a>AA_ALPHABET</span>
<span id="cb22-125"><a href="#cb22-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-126"><a href="#cb22-126" aria-hidden="true" tabindex="-1"></a>protein_strings <span class="ot">&lt;-</span> <span class="fu">AAStringSet</span>(<span class="fu">c</span>(<span class="st">"MATH"</span>, <span class="st">"GCAU"</span>, <span class="st">"MATHMATH"</span>, <span class="st">"CUMA"</span>))</span>
<span id="cb22-127"><a href="#cb22-127" aria-hidden="true" tabindex="-1"></a>protein_strings</span>
<span id="cb22-128"><a href="#cb22-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-129"><a href="#cb22-129" aria-hidden="true" tabindex="-1"></a><span class="co"># 二进制字符串存储 如seq_id、quality</span></span>
<span id="cb22-130"><a href="#cb22-130" aria-hidden="true" tabindex="-1"></a>binary_strings <span class="ot">&lt;-</span> <span class="fu">BStringSet</span>(<span class="fu">c</span>(<span class="st">"ERCC010101"</span>, <span class="st">"CRCC110011"</span>, <span class="st">"101010"</span>, <span class="st">"111000"</span>))</span>
<span id="cb22-131"><a href="#cb22-131" aria-hidden="true" tabindex="-1"></a>binary_strings</span>
<span id="cb22-132"><a href="#cb22-132" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-133"><a href="#cb22-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-134"><a href="#cb22-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-135"><a href="#cb22-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-136"><a href="#cb22-136" aria-hidden="true" tabindex="-1"></a><span class="fu">## Ranges</span></span>
<span id="cb22-137"><a href="#cb22-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-138"><a href="#cb22-138" aria-hidden="true" tabindex="-1"></a><span class="fu">### IntervalRanges</span></span>
<span id="cb22-139"><a href="#cb22-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-140"><a href="#cb22-140" aria-hidden="true" tabindex="-1"></a><span class="in">`IRanges`</span></span>
<span id="cb22-141"><a href="#cb22-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-144"><a href="#cb22-144" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-145"><a href="#cb22-145" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(IRanges)</span>
<span id="cb22-146"><a href="#cb22-146" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">10</span>)</span>
<span id="cb22-147"><a href="#cb22-147" aria-hidden="true" tabindex="-1"></a>ranges <span class="ot">&lt;-</span> IRanges<span class="sc">::</span><span class="fu">IRanges</span>(</span>
<span id="cb22-148"><a href="#cb22-148" aria-hidden="true" tabindex="-1"></a>  <span class="at">start =</span> <span class="fu">round</span>(<span class="fu">runif</span>(<span class="dv">10</span>, <span class="dv">1</span>, <span class="dv">100</span>)),</span>
<span id="cb22-149"><a href="#cb22-149" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="fu">round</span>(<span class="fu">runif</span>(<span class="dv">10</span>, <span class="dv">0</span>, <span class="dv">50</span>)),</span>
<span id="cb22-150"><a href="#cb22-150" aria-hidden="true" tabindex="-1"></a>  <span class="at">names =</span> <span class="fu">paste0</span>(<span class="st">"exton_"</span>, letters[<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">26</span>, <span class="dv">10</span>)])</span>
<span id="cb22-151"><a href="#cb22-151" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb22-152"><a href="#cb22-152" aria-hidden="true" tabindex="-1"></a>ranges</span>
<span id="cb22-153"><a href="#cb22-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-154"><a href="#cb22-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-155"><a href="#cb22-155" aria-hidden="true" tabindex="-1"></a><span class="fu">start</span>(ranges)</span>
<span id="cb22-156"><a href="#cb22-156" aria-hidden="true" tabindex="-1"></a><span class="fu">width</span>(ranges)</span>
<span id="cb22-157"><a href="#cb22-157" aria-hidden="true" tabindex="-1"></a><span class="fu">end</span>(ranges)</span>
<span id="cb22-158"><a href="#cb22-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-159"><a href="#cb22-159" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-160"><a href="#cb22-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-161"><a href="#cb22-161" aria-hidden="true" tabindex="-1"></a>没有维度，但有长度</span>
<span id="cb22-162"><a href="#cb22-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-165"><a href="#cb22-165" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-166"><a href="#cb22-166" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ranges)</span>
<span id="cb22-167"><a href="#cb22-167" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(ranges)</span>
<span id="cb22-168"><a href="#cb22-168" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-169"><a href="#cb22-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-170"><a href="#cb22-170" aria-hidden="true" tabindex="-1"></a>重叠检测: 检查不同区间是否有重叠。</span>
<span id="cb22-171"><a href="#cb22-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-174"><a href="#cb22-174" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-175"><a href="#cb22-175" aria-hidden="true" tabindex="-1"></a>query <span class="ot">&lt;-</span> <span class="fu">IRanges</span>(<span class="dv">1</span>, <span class="dv">10</span>)</span>
<span id="cb22-176"><a href="#cb22-176" aria-hidden="true" tabindex="-1"></a>subject <span class="ot">&lt;-</span> <span class="fu">IRanges</span>(<span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">15</span>), <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">20</span>))</span>
<span id="cb22-177"><a href="#cb22-177" aria-hidden="true" tabindex="-1"></a>overlaps <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(query, subject)</span>
<span id="cb22-178"><a href="#cb22-178" aria-hidden="true" tabindex="-1"></a>overlaps</span>
<span id="cb22-179"><a href="#cb22-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-180"><a href="#cb22-180" aria-hidden="true" tabindex="-1"></a><span class="fu">queryHits</span>(overlaps)</span>
<span id="cb22-181"><a href="#cb22-181" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-182"><a href="#cb22-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-183"><a href="#cb22-183" aria-hidden="true" tabindex="-1"></a>合并区间: 将重叠或相邻的区间合并为一个更大的区间。</span>
<span id="cb22-184"><a href="#cb22-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-187"><a href="#cb22-187" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-188"><a href="#cb22-188" aria-hidden="true" tabindex="-1"></a>merged <span class="ot">&lt;-</span> GenomicRanges<span class="sc">::</span><span class="fu">reduce</span>(<span class="fu">IRanges</span>(<span class="at">start =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">5</span>), <span class="at">end =</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">20</span>)))</span>
<span id="cb22-189"><a href="#cb22-189" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(merged)</span>
<span id="cb22-190"><a href="#cb22-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-191"><a href="#cb22-191" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-192"><a href="#cb22-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-195"><a href="#cb22-195" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-196"><a href="#cb22-196" aria-hidden="true" tabindex="-1"></a>plotRanges <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">xlim =</span> x , </span>
<span id="cb22-197"><a href="#cb22-197" aria-hidden="true" tabindex="-1"></a>                       <span class="at">main =</span><span class="fu">deparse</span>(<span class="fu">substitute</span>(x)),</span>
<span id="cb22-198"><a href="#cb22-198" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">sep =</span> <span class="fl">0.5</span>, ...){</span>
<span id="cb22-199"><a href="#cb22-199" aria-hidden="true" tabindex="-1"></a>  height <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb22-200"><a href="#cb22-200" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is</span>(xlim, <span class="at">class2 =</span> <span class="st">"Ranges"</span>))</span>
<span id="cb22-201"><a href="#cb22-201" aria-hidden="true" tabindex="-1"></a>    xlim <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">min</span>(<span class="fu">start</span>(xlim)), <span class="fu">max</span>(<span class="fu">end</span>(xlim)))</span>
<span id="cb22-202"><a href="#cb22-202" aria-hidden="true" tabindex="-1"></a>  bins <span class="ot">&lt;-</span> <span class="fu">disjointBins</span>(<span class="fu">IRanges</span>(<span class="fu">start</span>(x),<span class="fu">end</span>(x)<span class="sc">+</span><span class="dv">1</span>))</span>
<span id="cb22-203"><a href="#cb22-203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot.new</span>()</span>
<span id="cb22-204"><a href="#cb22-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot.window</span>(xlim, <span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(bins)<span class="sc">*</span>(height <span class="sc">+</span> sep)))</span>
<span id="cb22-205"><a href="#cb22-205" aria-hidden="true" tabindex="-1"></a>  ybottom <span class="ot">&lt;-</span> bins <span class="sc">*</span> (sep <span class="sc">+</span> height) <span class="sc">-</span> height</span>
<span id="cb22-206"><a href="#cb22-206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rect</span>(<span class="fu">start</span>(x)<span class="sc">-</span><span class="fl">0.5</span>, ybottom, <span class="fu">end</span>(x)<span class="sc">+</span><span class="fl">0.5</span>, ybottom <span class="sc">+</span> height,</span>
<span id="cb22-207"><a href="#cb22-207" aria-hidden="true" tabindex="-1"></a>       <span class="at">col =</span> col, ...)</span>
<span id="cb22-208"><a href="#cb22-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">title</span>(main)</span>
<span id="cb22-209"><a href="#cb22-209" aria-hidden="true" tabindex="-1"></a>  <span class="fu">axis</span>(<span class="dv">1</span>)</span>
<span id="cb22-210"><a href="#cb22-210" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-211"><a href="#cb22-211" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb22-212"><a href="#cb22-212" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-213"><a href="#cb22-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-216"><a href="#cb22-216" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-217"><a href="#cb22-217" aria-hidden="true" tabindex="-1"></a>ir <span class="ot">&lt;-</span> IRanges<span class="sc">::</span><span class="fu">IRanges</span>(</span>
<span id="cb22-218"><a href="#cb22-218" aria-hidden="true" tabindex="-1"></a>  <span class="at">start =</span><span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">4</span>, <span class="dv">10</span>),</span>
<span id="cb22-219"><a href="#cb22-219" aria-hidden="true" tabindex="-1"></a>  <span class="at">end =</span>  <span class="fu">c</span>(<span class="dv">6</span>, <span class="dv">3</span>, <span class="dv">8</span>, <span class="dv">10</span>)</span>
<span id="cb22-220"><a href="#cb22-220" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb22-221"><a href="#cb22-221" aria-hidden="true" tabindex="-1"></a>ir</span>
<span id="cb22-222"><a href="#cb22-222" aria-hidden="true" tabindex="-1"></a><span class="fu">disjoin</span>(ir)</span>
<span id="cb22-223"><a href="#cb22-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-224"><a href="#cb22-224" aria-hidden="true" tabindex="-1"></a><span class="fu">disjointBins</span>(<span class="fu">IRanges</span>(<span class="fu">start</span>(ir), <span class="fu">end</span>(ir) <span class="sc">+</span> <span class="dv">1</span>)) <span class="co"># 1,2,3,1 放于第1，2，3，1层</span></span>
<span id="cb22-225"><a href="#cb22-225" aria-hidden="true" tabindex="-1"></a><span class="fu">disjointBins</span>(<span class="fu">IRanges</span>(<span class="fu">start</span>(<span class="fu">disjoin</span>(ir)), <span class="fu">end</span>(<span class="fu">disjoin</span>(ir)) <span class="sc">+</span> <span class="dv">1</span>)) <span class="co">#放于第1，2，1，1层</span></span>
<span id="cb22-226"><a href="#cb22-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-227"><a href="#cb22-227" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">1</span>))</span>
<span id="cb22-228"><a href="#cb22-228" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRanges</span>(ir)</span>
<span id="cb22-229"><a href="#cb22-229" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRanges</span>(<span class="fu">disjoin</span>(ir))</span>
<span id="cb22-230"><a href="#cb22-230" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-231"><a href="#cb22-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-232"><a href="#cb22-232" aria-hidden="true" tabindex="-1"></a><span class="fu">### GenomicRanges</span></span>
<span id="cb22-233"><a href="#cb22-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-234"><a href="#cb22-234" aria-hidden="true" tabindex="-1"></a>**`GRanges`**：用于表示基因组（染色体）范围的数据结构，例如Promoters，Genes，SNPs，CpG Islands，......</span>
<span id="cb22-235"><a href="#cb22-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-238"><a href="#cb22-238" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-239"><a href="#cb22-239" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb22-240"><a href="#cb22-240" aria-hidden="true" tabindex="-1"></a><span class="co"># help("GRanges-class")</span></span>
<span id="cb22-241"><a href="#cb22-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-242"><a href="#cb22-242" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">10</span>)</span>
<span id="cb22-243"><a href="#cb22-243" aria-hidden="true" tabindex="-1"></a>Granges <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(</span>
<span id="cb22-244"><a href="#cb22-244" aria-hidden="true" tabindex="-1"></a>  <span class="at">seqnames =</span> <span class="fu">Rle</span>( <span class="at">values =</span>  <span class="fu">c</span>(<span class="st">'chr1'</span>, <span class="st">'chr2'</span>, <span class="st">'chr3'</span>), <span class="at">lengths =</span>  <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">3</span>)),</span>
<span id="cb22-245"><a href="#cb22-245" aria-hidden="true" tabindex="-1"></a>  <span class="at">ranges =</span> <span class="fu">IRanges</span>(<span class="at">start =</span> <span class="fu">seq</span>(<span class="dv">1</span>,<span class="dv">18</span>,<span class="dv">2</span>), <span class="at">width =</span> <span class="dv">3</span>),</span>
<span id="cb22-246"><a href="#cb22-246" aria-hidden="true" tabindex="-1"></a>  <span class="at">strand =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"+"</span>,<span class="st">"-"</span>,<span class="st">"*"</span>),<span class="at">each=</span><span class="dv">3</span>),</span>
<span id="cb22-247"><a href="#cb22-247" aria-hidden="true" tabindex="-1"></a>  <span class="at">score =</span> <span class="dv">101</span><span class="sc">:</span><span class="dv">109</span>,</span>
<span id="cb22-248"><a href="#cb22-248" aria-hidden="true" tabindex="-1"></a>  <span class="at">GC =</span> <span class="fu">runif</span>(<span class="dv">9</span>)</span>
<span id="cb22-249"><a href="#cb22-249" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb22-250"><a href="#cb22-250" aria-hidden="true" tabindex="-1"></a>Granges</span>
<span id="cb22-251"><a href="#cb22-251" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-252"><a href="#cb22-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-255"><a href="#cb22-255" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-256"><a href="#cb22-256" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(Granges)</span>
<span id="cb22-257"><a href="#cb22-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-258"><a href="#cb22-258" aria-hidden="true" tabindex="-1"></a><span class="fu">values</span>(Granges) <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(<span class="at">score =</span> <span class="fu">rnorm</span>(<span class="dv">9</span>), <span class="at">GC =</span> <span class="fu">runif</span>(<span class="dv">9</span>))</span>
<span id="cb22-259"><a href="#cb22-259" aria-hidden="true" tabindex="-1"></a>Granges</span>
<span id="cb22-260"><a href="#cb22-260" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-261"><a href="#cb22-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-262"><a href="#cb22-262" aria-hidden="true" tabindex="-1"></a><span class="in">`flank(x, width, start = TRUE, both = FALSE)`</span> 用于在基因组范围对象 (GRanges) 上扩展或调整范围。它会基于输入的基因组范围返回其上游或下游的邻近区域。flanking sequence</span>
<span id="cb22-263"><a href="#cb22-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-264"><a href="#cb22-264" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`x`</span>: 一个 GRanges 对象。</span>
<span id="cb22-265"><a href="#cb22-265" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`width`</span>: 你想要的上下游区域的宽度（如上例中的 5）。</span>
<span id="cb22-266"><a href="#cb22-266" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`start`</span>: 这个参数决定了是否使用起始端（TRUE）还是结束端（FALSE）来生成上下游区域。如果你的基因组是正链，start = TRUE 表示上游。如果是负链，start = TRUE 则表示下游。默认为 TRUE。</span>
<span id="cb22-267"><a href="#cb22-267" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`both`</span>: 如果设置为 TRUE，则会在范围的两端生成 width 宽度的区域。</span>
<span id="cb22-268"><a href="#cb22-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-271"><a href="#cb22-271" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-272"><a href="#cb22-272" aria-hidden="true" tabindex="-1"></a><span class="fu">flank</span>(Granges,<span class="dv">5</span>)</span>
<span id="cb22-273"><a href="#cb22-273" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-274"><a href="#cb22-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-275"><a href="#cb22-275" aria-hidden="true" tabindex="-1"></a><span class="in">`promoters(x, upstream = 2000, downstream = 200)`</span> 通常用于提取基因的启动子区域（promoter regions）。在基因组分析中，启动子区域是靠近基因起始位点的区域，通常被认为是调控基因表达的关键区域。启动子区域是基于基因组范围的起始位置提取的，上游（upstream）和下游（downstream）的长度可以自定义。</span>
<span id="cb22-276"><a href="#cb22-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-279"><a href="#cb22-279" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-280"><a href="#cb22-280" aria-hidden="true" tabindex="-1"></a><span class="fu">promoters</span>(Granges)</span>
<span id="cb22-281"><a href="#cb22-281" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-282"><a href="#cb22-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-285"><a href="#cb22-285" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-286"><a href="#cb22-286" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomeInfoDb)</span>
<span id="cb22-287"><a href="#cb22-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-288"><a href="#cb22-288" aria-hidden="true" tabindex="-1"></a><span class="fu">genome</span>(Granges) <span class="ot">&lt;-</span> <span class="st">"hg19"</span></span>
<span id="cb22-289"><a href="#cb22-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-290"><a href="#cb22-290" aria-hidden="true" tabindex="-1"></a><span class="fu">seqinfo</span>(Granges)</span>
<span id="cb22-291"><a href="#cb22-291" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevels</span>(Granges)</span>
<span id="cb22-292"><a href="#cb22-292" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlengths</span>(Granges)</span>
<span id="cb22-293"><a href="#cb22-293" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-294"><a href="#cb22-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-295"><a href="#cb22-295" aria-hidden="true" tabindex="-1"></a><span class="in">`gaps(x)`</span>用于计算给定的 GRanges 对象中缺失（gaps）的基因组区域。它返回的是不被输入的 GRanges 对象覆盖的范围（即空白区域），通常用于确定在某些染色体区域中没有基因或特征的部分。</span>
<span id="cb22-296"><a href="#cb22-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-299"><a href="#cb22-299" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-300"><a href="#cb22-300" aria-hidden="true" tabindex="-1"></a><span class="fu">gaps</span>(Granges)</span>
<span id="cb22-301"><a href="#cb22-301" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-302"><a href="#cb22-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-305"><a href="#cb22-305" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-306"><a href="#cb22-306" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gggenes)</span>
<span id="cb22-307"><a href="#cb22-307" aria-hidden="true" tabindex="-1"></a>Granges <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb22-308"><a href="#cb22-308" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-309"><a href="#cb22-309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">xmin =</span> start, <span class="at">xmax=</span> end,<span class="at">y =</span>seqnames,<span class="at">fill =</span> gene))<span class="sc">+</span></span>
<span id="cb22-310"><a href="#cb22-310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_gene_arrow</span>()<span class="sc">+</span></span>
<span id="cb22-311"><a href="#cb22-311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> seqnames, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb22-312"><a href="#cb22-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set3"</span>)<span class="sc">+</span></span>
<span id="cb22-313"><a href="#cb22-313" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_genes</span>()</span>
<span id="cb22-314"><a href="#cb22-314" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-315"><a href="#cb22-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-318"><a href="#cb22-318" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-319"><a href="#cb22-319" aria-hidden="true" tabindex="-1"></a>gr1 <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames =</span> <span class="st">"chr1"</span>, </span>
<span id="cb22-320"><a href="#cb22-320" aria-hidden="true" tabindex="-1"></a>               <span class="at">ranges =</span> <span class="fu">IRanges</span>(<span class="at">start =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">20</span>), <span class="at">end =</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">30</span>)))</span>
<span id="cb22-321"><a href="#cb22-321" aria-hidden="true" tabindex="-1"></a>gr2 <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames =</span> <span class="st">"chr1"</span>, </span>
<span id="cb22-322"><a href="#cb22-322" aria-hidden="true" tabindex="-1"></a>               <span class="at">ranges =</span> <span class="fu">IRanges</span>(<span class="at">start =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">25</span>), <span class="at">end =</span> <span class="fu">c</span>(<span class="dv">15</span>, <span class="dv">35</span>)))</span>
<span id="cb22-323"><a href="#cb22-323" aria-hidden="true" tabindex="-1"></a>gr1</span>
<span id="cb22-324"><a href="#cb22-324" aria-hidden="true" tabindex="-1"></a>gr2</span>
<span id="cb22-325"><a href="#cb22-325" aria-hidden="true" tabindex="-1"></a>GenomicRanges<span class="sc">::</span><span class="fu">setdiff</span>(gr1, gr2)</span>
<span id="cb22-326"><a href="#cb22-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-327"><a href="#cb22-327" aria-hidden="true" tabindex="-1"></a><span class="fu">findOverlaps</span>(gr1,gr2)</span>
<span id="cb22-328"><a href="#cb22-328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-329"><a href="#cb22-329" aria-hidden="true" tabindex="-1"></a><span class="fu">subsetByOverlaps</span>(gr1, gr2)</span>
<span id="cb22-330"><a href="#cb22-330" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/Wanganlin00/Bioinformatics/edit/main/biodata_structure.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/Wanganlin00/Bioinformatics/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div></div></footer></body></html>