<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Bioinformatics - 5&nbsp; 质量控制</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./normalization.html" rel="next">
<link href="./basics.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
</head>
<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./index.html">Single-cell Analysis</a></li><li class="breadcrumb-item"><a href="./QC.html"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">质量控制</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Bioinformatics</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/Wanganlin00/Bioinformatics" rel="" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./genomics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title"><code>SummarizeExperiment</code>结构</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Single-cell Analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">前言</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./import.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">获取scRNA-seq数据集</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./SCE.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title"><code>SingleCellExperiment</code>结构</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">分析概述</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./basics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Basics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./QC.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">质量控制</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./normalization.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Normalization</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./feature_selection.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">特征选择</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Seurat</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Seurat.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Seurat</span></span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li><a href="#qc-metrics" id="toc-qc-metrics" class="nav-link active" data-scroll-target="#qc-metrics"><span class="header-section-number">5.1</span> QC metrics</a></li>
  <li>
<a href="#%E8%AF%86%E5%88%AB%E4%BD%8E%E8%B4%A8%E9%87%8F%E7%BB%86%E8%83%9E" id="toc-识别低质量细胞" class="nav-link" data-scroll-target="#%E8%AF%86%E5%88%AB%E4%BD%8E%E8%B4%A8%E9%87%8F%E7%BB%86%E8%83%9E"><span class="header-section-number">5.2</span> 识别低质量细胞</a>
  <ul class="collapse">
<li><a href="#%E5%9B%BA%E5%AE%9A%E9%98%88%E5%80%BC" id="toc-固定阈值" class="nav-link" data-scroll-target="#%E5%9B%BA%E5%AE%9A%E9%98%88%E5%80%BC"><span class="header-section-number">5.2.1</span> 固定阈值</a></li>
  <li><a href="#%E8%87%AA%E9%80%82%E5%BA%94%E9%98%88%E5%80%BC" id="toc-自适应阈值" class="nav-link" data-scroll-target="#%E8%87%AA%E9%80%82%E5%BA%94%E9%98%88%E5%80%BC"><span class="header-section-number">5.2.2</span> 自适应阈值</a></li>
  <li><a href="#%E5%85%B6%E4%BB%96%E6%96%B9%E6%B3%95" id="toc-其他方法" class="nav-link" data-scroll-target="#%E5%85%B6%E4%BB%96%E6%96%B9%E6%B3%95"><span class="header-section-number">5.2.3</span> 其他方法</a></li>
  </ul>
</li>
  <li><a href="#%E6%A3%80%E6%9F%A5%E8%AF%8A%E6%96%AD%E5%9B%BE" id="toc-检查诊断图" class="nav-link" data-scroll-target="#%E6%A3%80%E6%9F%A5%E8%AF%8A%E6%96%AD%E5%9B%BE"><span class="header-section-number">5.3</span> 检查诊断图</a></li>
  <li><a href="#%E5%8E%BB%E9%99%A4%E5%8A%A3%E8%B4%A8%E7%BB%86%E8%83%9E" id="toc-去除劣质细胞" class="nav-link" data-scroll-target="#%E5%8E%BB%E9%99%A4%E5%8A%A3%E8%B4%A8%E7%BB%86%E8%83%9E"><span class="header-section-number">5.4</span> 去除劣质细胞</a></li>
  </ul><div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/Wanganlin00/Bioinformatics/edit/main/QC.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/Wanganlin00/Bioinformatics/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">
<span class="chapter-number">5</span>&nbsp; <span class="chapter-title">质量控制</span>
</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header><p>scRNA-seq数据中的低质量文库（library）可能来自多种来源，例如解离过程中的细胞损伤或文库制备失败（如低效的逆转录或PCR扩增），这些通常表现为总计数低、表达基因少、线粒体或尖峰比例高的”细胞”。这些低质量的文库是有问题的，因为它们会导致下游分析中的误导性结果。</p>
<p>为了避免或至少缓解这些问题，我们需要在分析开始时删除有问题的细胞(cell)，此步骤通常称为细胞的质量控制 （QC）。</p>
<div class="cell" data-hash="QC_cache/html/data.416b_fd8f6b128773e70b48e316071af0960c">
<details><summary>加载数据LunSpikeInData:工作流第1章</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">scRNAseq</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jorainer/ensembldb">ensembldb</a></span><span class="op">)</span></span>
<span><span class="va">sce.416b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scRNAseq/man/LunSpikeInData.html">LunSpikeInData</a></span><span class="op">(</span>which<span class="op">=</span><span class="st">"416b"</span><span class="op">)</span> </span>
<span><span class="va">sce.416b</span><span class="op">$</span><span class="va">block</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">sce.416b</span><span class="op">$</span><span class="va">block</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sce.416b</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 46604 192 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(46604): ENSMUSG00000102693 ENSMUSG00000064842 ...</span></span>
<span><span class="co">#&gt;   ENSMUSG00000095742 CBFB-MYH11-mcherry</span></span>
<span><span class="co">#&gt; rowData names(1): Length</span></span>
<span><span class="co">#&gt; colnames(192): SLX-9555.N701_S502.C89V9ANXX.s_1.r_1</span></span>
<span><span class="co">#&gt;   SLX-9555.N701_S503.C89V9ANXX.s_1.r_1 ...</span></span>
<span><span class="co">#&gt;   SLX-11312.N712_S508.H5H5YBBXX.s_8.r_1</span></span>
<span><span class="co">#&gt;   SLX-11312.N712_S517.H5H5YBBXX.s_8.r_1</span></span>
<span><span class="co">#&gt; colData names(9): Source Name cell line ... spike-in addition block</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: endogenous</span></span>
<span><span class="co">#&gt; altExpNames(2): ERCC SIRV</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="qc-metrics" class="level2" data-number="5.1"><h2 data-number="5.1" class="anchored" data-anchor-id="qc-metrics">
<span class="header-section-number">5.1</span> QC metrics</h2>
<ul>
<li><p><strong>The library size</strong> is defined as the total sum of counts across all relevant features for each cell. Cells with small library sizes are of low quality.库大小定义为每个细胞所有基因的计数总和。</p></li>
<li><p><strong>The number of expressed features in each cell</strong> is defined as the number of endogenous genes with non-zero counts for that cell. Any cell with very few expressed genes is likely to be of poor quality. 每个细胞中表达特征的数量被定义为该细胞具有非零计数的内源性基因的数量。</p></li>
<li><p><strong>The proportion of reads mapped to spike-in transcripts</strong> is calculated relative to the total count across all features (including spike-ins) for each cell.High proportions are indicative of poor-quality cells。 相对于每个细胞所有特征（包括加标）的总计数，计算映射到加标转录物中的读数比例。高比例表示细胞质量差</p></li>
<li><p><strong>In the absence of spike-in transcripts, the proportion of reads mapped to genes in the mitochondrial genome</strong> can be used. High proportions are indicative of poor-quality cells。在没有加标转录物的情况下，可以使用映射到线粒体基因组中基因的读数比例。高比例表示细胞质量差。</p></li>
</ul>
<p>对于每个细胞，我们使用&nbsp;<a href="https://bioconductor.org/packages/3.18/scater"><em>scater</em></a>&nbsp;包中的函数<code><a href="https://rdrr.io/pkg/scuttle/man/perCellQCMetrics.html">perCellQCMetrics()</a></code>计算这些 QC 指标,该列<code>sum</code>包含每个细胞文库大小的总计数，该列<code>detected</code>包含检测到的基因数。 该列<code>subsets_Mito_percent</code>包含映射到线粒体转录本的读取百分比，该列<code>altexps_ERCC_percent</code>包含映射到 ERCC 转录本的读取百分比。</p>
<div class="cell" data-hash="QC_cache/html/unnamed-chunk-3_c6e38a7a2b10b4a6bfb49c277bb45ae6">
<details><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce.416b</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 46604 192 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(46604): ENSMUSG00000102693 ENSMUSG00000064842 ...</span></span>
<span><span class="co">#&gt;   ENSMUSG00000095742 CBFB-MYH11-mcherry</span></span>
<span><span class="co">#&gt; rowData names(1): Length</span></span>
<span><span class="co">#&gt; colnames(192): SLX-9555.N701_S502.C89V9ANXX.s_1.r_1</span></span>
<span><span class="co">#&gt;   SLX-9555.N701_S503.C89V9ANXX.s_1.r_1 ...</span></span>
<span><span class="co">#&gt;   SLX-11312.N712_S508.H5H5YBBXX.s_8.r_1</span></span>
<span><span class="co">#&gt;   SLX-11312.N712_S517.H5H5YBBXX.s_8.r_1</span></span>
<span><span class="co">#&gt; colData names(9): Source Name cell line ... spike-in addition block</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: endogenous</span></span>
<span><span class="co">#&gt; altExpNames(2): ERCC SIRV</span></span>
<span><span class="va">location</span> <span class="op">&lt;-</span> <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span></span>
<span><span class="va">location</span></span>
<span><span class="co">#&gt; GRangesList object of length 46604:</span></span>
<span><span class="co">#&gt; $ENSMUSG00000102693</span></span>
<span><span class="co">#&gt; GRanges object with 1 range and 0 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames          ranges strand</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   [1]        1 3073253-3074322      +</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 118 sequences (1 circular) from GRCm38 genome</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $ENSMUSG00000064842</span></span>
<span><span class="co">#&gt; GRanges object with 1 range and 0 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames          ranges strand</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   [1]        1 3102016-3102125      +</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 118 sequences (1 circular) from GRCm38 genome</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $ENSMUSG00000051951</span></span>
<span><span class="co">#&gt; GRanges object with 1 range and 0 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames          ranges strand</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   [1]        1 3205901-3671498      -</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 118 sequences (1 circular) from GRCm38 genome</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ...</span></span>
<span><span class="co">#&gt; &lt;46601 more elements&gt;</span></span>
<span><span class="va">is.mito</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">location</span><span class="op">)</span><span class="op">==</span><span class="st">"MT"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">scuttle</span><span class="op">)</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/perCellQCMetrics.html">perCellQCMetrics</a></span><span class="op">(</span><span class="va">sce.416b</span>, subsets<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>Mito<span class="op">=</span><span class="va">is.mito</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "sum"                   "detected"              "subsets_Mito_sum"     </span></span>
<span><span class="co">#&gt;  [4] "subsets_Mito_detected" "subsets_Mito_percent"  "altexps_ERCC_sum"     </span></span>
<span><span class="co">#&gt;  [7] "altexps_ERCC_detected" "altexps_ERCC_percent"  "altexps_SIRV_sum"     </span></span>
<span><span class="co">#&gt; [10] "altexps_SIRV_detected" "altexps_SIRV_percent"  "total"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">sum</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">#&gt;   27084  856350 1111252 1165865 1328301 4398883</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">altexps_ERCC_percent</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">#&gt;   2.242   4.291   6.031   6.412   8.126  19.429</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>或者 <code><a href="https://rdrr.io/pkg/scuttle/man/addPerCellQCMetrics.html">addPerCellQC()</a></code>计算每个细胞的 QC 统计数据并将其附加到对象的<code>colData</code> 允许我们将所有相关信息保留在单个对象<code>SingleCellExperiment</code>中以供以后操作。</p>
<div class="cell" data-hash="QC_cache/html/unnamed-chunk-4_bfdb4e3da71c470ce2f7abc4a7dc607a">
<details><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce.416b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/addPerCellQCMetrics.html">addPerCellQCMetrics</a></span><span class="op">(</span><span class="va">sce.416b</span>, subsets<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>Mito<span class="op">=</span><span class="va">is.mito</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 192  21</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "Source Name"              "cell line"               </span></span>
<span><span class="co">#&gt;  [3] "cell type"                "single cell well quality"</span></span>
<span><span class="co">#&gt;  [5] "genotype"                 "phenotype"               </span></span>
<span><span class="co">#&gt;  [7] "strain"                   "spike-in addition"       </span></span>
<span><span class="co">#&gt;  [9] "block"                    "sum"                     </span></span>
<span><span class="co">#&gt; [11] "detected"                 "subsets_Mito_sum"        </span></span>
<span><span class="co">#&gt; [13] "subsets_Mito_detected"    "subsets_Mito_percent"    </span></span>
<span><span class="co">#&gt; [15] "altexps_ERCC_sum"         "altexps_ERCC_detected"   </span></span>
<span><span class="co">#&gt; [17] "altexps_ERCC_percent"     "altexps_SIRV_sum"        </span></span>
<span><span class="co">#&gt; [19] "altexps_SIRV_detected"    "altexps_SIRV_percent"    </span></span>
<span><span class="co">#&gt; [21] "total"</span></span>
<span><span class="va">sce.416b</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 46604 192 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(46604): ENSMUSG00000102693 ENSMUSG00000064842 ...</span></span>
<span><span class="co">#&gt;   ENSMUSG00000095742 CBFB-MYH11-mcherry</span></span>
<span><span class="co">#&gt; rowData names(1): Length</span></span>
<span><span class="co">#&gt; colnames(192): SLX-9555.N701_S502.C89V9ANXX.s_1.r_1</span></span>
<span><span class="co">#&gt;   SLX-9555.N701_S503.C89V9ANXX.s_1.r_1 ...</span></span>
<span><span class="co">#&gt;   SLX-11312.N712_S508.H5H5YBBXX.s_8.r_1</span></span>
<span><span class="co">#&gt;   SLX-11312.N712_S517.H5H5YBBXX.s_8.r_1</span></span>
<span><span class="co">#&gt; colData names(21): Source Name cell line ... altexps_SIRV_percent total</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: endogenous</span></span>
<span><span class="co">#&gt; altExpNames(2): ERCC SIRV</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="识别低质量细胞" class="level2" data-number="5.2"><h2 data-number="5.2" class="anchored" data-anchor-id="识别低质量细胞">
<span class="header-section-number">5.2</span> 识别低质量细胞</h2>
<section id="固定阈值" class="level3" data-number="5.2.1"><h3 data-number="5.2.1" class="anchored" data-anchor-id="固定阈值">
<span class="header-section-number">5.2.1</span> 固定阈值</h3>
<p>识别低质量细胞的最简单方法是对 QC 指标应用固定阈值。 例如，如果细胞的文库大小低于 100,000 次读取;表达少于5,000个基因;峰值比例超过10%;或线粒体比例高于10%，我们可能会认为它们的质量较低。</p>
<div class="cell" data-hash="QC_cache/html/unnamed-chunk-5_658a0bf40f8a91d4b991b4a9f552aff0">
<details><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">qc.lib</span> <span class="op">&lt;-</span> <span class="va">df</span><span class="op">$</span><span class="va">sum</span> <span class="op">&lt;</span> <span class="fl">1e5</span></span>
<span><span class="va">qc.nexprs</span> <span class="op">&lt;-</span> <span class="va">df</span><span class="op">$</span><span class="va">detected</span> <span class="op">&lt;</span> <span class="fl">5e3</span></span>
<span><span class="va">qc.spike</span> <span class="op">&lt;-</span> <span class="va">df</span><span class="op">$</span><span class="va">altexps_ERCC_percent</span> <span class="op">&gt;</span> <span class="fl">10</span></span>
<span><span class="va">qc.mito</span> <span class="op">&lt;-</span> <span class="va">df</span><span class="op">$</span><span class="va">subsets_Mito_percent</span> <span class="op">&gt;</span> <span class="fl">10</span></span>
<span><span class="va">discard</span> <span class="op">&lt;-</span> <span class="va">qc.lib</span> <span class="op">|</span> <span class="va">qc.nexprs</span> <span class="op">|</span> <span class="va">qc.spike</span> <span class="op">|</span> <span class="va">qc.mito</span></span>
<span></span>
<span><span class="co"># 汇总</span></span>
<span><span class="fu">tibble</span><span class="op">(</span>LibSize<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">qc.lib</span><span class="op">)</span>, NExprs<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">qc.nexprs</span><span class="op">)</span>,</span>
<span>    SpikeProp<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">qc.spike</span><span class="op">)</span>, MitoProp<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">qc.mito</span><span class="op">)</span>, Total<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">discard</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead><tr class="header">
<th style="text-align: right;">LibSize</th>
<th style="text-align: right;">NExprs</th>
<th style="text-align: right;">SpikeProp</th>
<th style="text-align: right;">MitoProp</th>
<th style="text-align: right;">Total</th>
</tr></thead>
<tbody><tr class="odd">
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">33</td>
</tr></tbody>
</table>
</div>
</div>
</div>
</section><section id="自适应阈值" class="level3" data-number="5.2.2"><h3 data-number="5.2.2" class="anchored" data-anchor-id="自适应阈值">
<span class="header-section-number">5.2.2</span> 自适应阈值</h3>
<p>假设大部分数据集由高质量的细胞组成，然后，我们根据所有细胞中每个指标的中位数绝对偏差（median absolute deviation ，MAD）来识别各种QC指标的异常值细胞。默认情况下，如果一个值在”有问题”的方向上距离中位数超过3个MAD，我们将其视为异常值。即此类过滤器将保留 99% 遵循正态分布的非异常值。</p>
<div class="cell" data-hash="QC_cache/html/unnamed-chunk-6_9819b96fcc77f15918caad850bf16bc9">
<details><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#?perCellQCFilters</span></span>
<span><span class="va">reasons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/perCellQCFilters.html">perCellQCFilters</a></span><span class="op">(</span><span class="va">df</span>, </span>
<span>                            sub.fields<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"subsets_Mito_percent"</span>,</span>
<span>                                         <span class="st">"altexps_ERCC_percent"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">reasons</span><span class="op">)</span><span class="op">)</span> <span class="co"># apply(as.matrix(reasons),2,sum)</span></span>
<span><span class="co">#&gt;              low_lib_size            low_n_features high_subsets_Mito_percent </span></span>
<span><span class="co">#&gt;                         4                         0                         2 </span></span>
<span><span class="co">#&gt; high_altexps_ERCC_percent                   discard </span></span>
<span><span class="co">#&gt;                         1                         6</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">reasons</span><span class="op">$</span><span class="va">discard</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Mode   FALSE    TRUE </span></span>
<span><span class="co">#&gt; logical     186       6</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/attr.html">attr</a></span><span class="op">(</span><span class="va">reasons</span><span class="op">$</span><span class="va">low_lib_size</span>, <span class="st">"thresholds"</span><span class="op">)</span></span>
<span><span class="co">#&gt;    lower   higher </span></span>
<span><span class="co">#&gt; 434082.9      Inf</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/attr.html">attr</a></span><span class="op">(</span><span class="va">reasons</span><span class="op">$</span><span class="va">low_n_features</span>, <span class="st">"thresholds"</span><span class="op">)</span></span>
<span><span class="co">#&gt;    lower   higher </span></span>
<span><span class="co">#&gt; 5231.468      Inf</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="其他方法" class="level3" data-number="5.2.3"><h3 data-number="5.2.3" class="anchored" data-anchor-id="其他方法">
<span class="header-section-number">5.2.3</span> 其他方法</h3>
<p>根据每个细胞的 QC 指标识别高维空间中的异常值</p>
<div class="cell" data-hash="QC_cache/html/unnamed-chunk-7_bf2095170f7bc776276f1afdc8ddd286">
<details><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">sum</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">detected</span><span class="op">)</span>,</span>
<span>    <span class="va">df</span><span class="op">$</span><span class="va">subsets_Mito_percent</span>, <span class="va">df</span><span class="op">$</span><span class="va">altexps_ERCC_percent</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://robustbase.R-forge.R-project.org/">robustbase</a></span><span class="op">)</span></span>
<span><span class="va">outlying</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/robustbase/man/adjOutlyingness.html">adjOutlyingness</a></span><span class="op">(</span><span class="va">stats</span>, only.outlyingness <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">multi.outlier</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/isOutlier.html">isOutlier</a></span><span class="op">(</span><span class="va">outlying</span>, type <span class="op">=</span> <span class="st">"higher"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">multi.outlier</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Mode   FALSE    TRUE </span></span>
<span><span class="co">#&gt; logical     182      10</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section></section><section id="检查诊断图" class="level2" data-number="5.3"><h2 data-number="5.3" class="anchored" data-anchor-id="检查诊断图">
<span class="header-section-number">5.3</span> 检查诊断图</h2>
<div class="cell" data-hash="QC_cache/html/fig-diagnosis-plot_d43529dcf721a5e986db64d916c99afc">
<details><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">colData</span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html">cbind</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">sce.416b</span><span class="op">)</span>, <span class="va">df</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sce.416b</span><span class="op">$</span><span class="va">block</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">sce.416b</span><span class="op">$</span><span class="va">block</span><span class="op">)</span></span>
<span><span class="va">sce.416b</span><span class="op">$</span><span class="va">phenotype</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html">grepl</a></span><span class="op">(</span><span class="st">"induced"</span>, <span class="va">sce.416b</span><span class="op">$</span><span class="va">phenotype</span><span class="op">)</span>,</span>
<span>    <span class="st">"induced"</span>, <span class="st">"wild type"</span><span class="op">)</span></span>
<span><span class="va">sce.416b</span><span class="op">$</span><span class="va">discard</span> <span class="op">&lt;-</span> <span class="va">reasons</span><span class="op">$</span><span class="va">discard</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/">scater</a></span><span class="op">)</span></span>
<span><span class="fu">gridExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html">plotColData</a></span><span class="op">(</span><span class="va">sce.416b</span>, x<span class="op">=</span><span class="st">"block"</span>, y<span class="op">=</span><span class="st">"sum"</span>, colour_by<span class="op">=</span><span class="st">"discard"</span>,</span>
<span>        other_fields<span class="op">=</span><span class="st">"phenotype"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">phenotype</span><span class="op">)</span> <span class="op">+</span> </span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Total count"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html">plotColData</a></span><span class="op">(</span><span class="va">sce.416b</span>, x<span class="op">=</span><span class="st">"block"</span>, y<span class="op">=</span><span class="st">"detected"</span>, colour_by<span class="op">=</span><span class="st">"discard"</span>, </span>
<span>        other_fields<span class="op">=</span><span class="st">"phenotype"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">phenotype</span><span class="op">)</span> <span class="op">+</span> </span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Detected features"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html">plotColData</a></span><span class="op">(</span><span class="va">sce.416b</span>, x<span class="op">=</span><span class="st">"block"</span>, y<span class="op">=</span><span class="st">"subsets_Mito_percent"</span>, </span>
<span>        colour_by<span class="op">=</span><span class="st">"discard"</span>, other_fields<span class="op">=</span><span class="st">"phenotype"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">phenotype</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Mito percent"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html">plotColData</a></span><span class="op">(</span><span class="va">sce.416b</span>, x<span class="op">=</span><span class="st">"block"</span>, y<span class="op">=</span><span class="st">"altexps_ERCC_percent"</span>, </span>
<span>        colour_by<span class="op">=</span><span class="st">"discard"</span>, other_fields<span class="op">=</span><span class="st">"phenotype"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">phenotype</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"ERCC percent"</span><span class="op">)</span>,</span>
<span>    ncol<span class="op">=</span><span class="fl">1</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="fig-diagnosis-plot" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="QC_files/figure-html/fig-diagnosis-plot-1.png" class="img-fluid figure-img" width="576"></p>
<figcaption class="figure-caption">图&nbsp;5.1: 数据集中每个批次和表型的 QC 指标分布。每个点代表一个细胞，并分别根据其是否被丢弃而着色。</figcaption></figure>
</div>
</div>
</div>
<p>另一个有用的诊断涉及绘制线粒体计数与其他一些 QC 指标的比例。 目的是确认没有细胞同时具有大量总计数和大量线粒体计数，以确保我们不会无意中去除恰好具有高度代谢活性的高质量细胞（例如肝细胞）。</p>
<div class="cell" data-hash="QC_cache/html/dataset.zeisel_5eb31ac91146879196c1043b3f82a773">
<details><summary>加载数据ZeiselBrainData：工作流第2章</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">scRNAseq</span><span class="op">)</span></span>
<span><span class="va">sce.zeisel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scRNAseq/man/ZeiselBrainData.html">ZeiselBrainData</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">sce.zeisel</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 20006 3005 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(20006): Tspan12 Tshz1 ... mt-Rnr1 mt-Nd4l</span></span>
<span><span class="co">#&gt; rowData names(1): featureType</span></span>
<span><span class="co">#&gt; colnames(3005): 1772071015_C02 1772071017_G12 ... 1772066098_A12</span></span>
<span><span class="co">#&gt;   1772058148_F03</span></span>
<span><span class="co">#&gt; colData names(10): tissue group # ... level1class level2class</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: endogenous</span></span>
<span><span class="co">#&gt; altExpNames(2): ERCC repeat</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/">scater</a></span><span class="op">)</span></span>
<span><span class="va">sce.zeisel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/aggregateAcrossFeatures.html">aggregateAcrossFeatures</a></span><span class="op">(</span><span class="va">sce.zeisel</span>, </span>
<span>    id<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"_loc[0-9]+$"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce.zeisel</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#--- gene-annotation ---#</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Mm.eg.db</span><span class="op">)</span></span>
<span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce.zeisel</span><span class="op">)</span><span class="op">$</span><span class="va">Ensembl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">mapIds</a></span><span class="op">(</span><span class="va">org.Mm.eg.db</span>, </span>
<span>    keys<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce.zeisel</span><span class="op">)</span>, keytype<span class="op">=</span><span class="st">"SYMBOL"</span>, column<span class="op">=</span><span class="st">"ENSEMBL"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="QC_cache/html/fig-mito-sum_0c2e2b754eb24deefc124f4455c4cc3c">
<details><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce.zeisel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/addPerCellQCMetrics.html">addPerCellQC</a></span><span class="op">(</span></span>
<span>  <span class="va">sce.zeisel</span>,subsets<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>Mt<span class="op">=</span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce.zeisel</span><span class="op">)</span><span class="op">$</span><span class="va">featureType</span><span class="op">==</span><span class="st">"mito"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">qc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/quickPerCellQC.html">quickPerCellQC</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">sce.zeisel</span><span class="op">)</span>, </span>
<span>    sub.fields<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"altexps_ERCC_percent"</span>, <span class="st">"subsets_Mt_percent"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sce.zeisel</span><span class="op">$</span><span class="va">discard</span> <span class="op">&lt;-</span> <span class="va">qc</span><span class="op">$</span><span class="va">discard</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html">plotColData</a></span><span class="op">(</span><span class="va">sce.zeisel</span>, x<span class="op">=</span><span class="st">"sum"</span>, y<span class="op">=</span><span class="st">"subsets_Mt_percent"</span>, colour_by<span class="op">=</span><span class="st">"discard"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="fig-mito-sum" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="QC_files/figure-html/fig-mito-sum-1.png" class="img-fluid figure-img" width="576"></p>
<figcaption class="figure-caption">图&nbsp;5.2: Zeisel 大脑数据集 中分配给线粒体转录本的 UMI 百分比，与 UMI 总数作图。每个点代表一个细胞，并根据它是否被视为低质量并丢弃进行着色。</figcaption></figure>
</div>
</div>
</div>
<div class="cell" data-hash="QC_cache/html/fig-mito-ERCC_4662e87c3fb2b17686e566dfc25475d7">
<details><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html">plotColData</a></span><span class="op">(</span><span class="va">sce.zeisel</span>, x<span class="op">=</span><span class="st">"altexps_ERCC_percent"</span>, y<span class="op">=</span><span class="st">"subsets_Mt_percent"</span>,</span>
<span>    colour_by<span class="op">=</span><span class="st">"discard"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="fig-mito-ERCC" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="QC_files/figure-html/fig-mito-ERCC-1.png" class="img-fluid figure-img" width="576"></p>
<figcaption class="figure-caption">图&nbsp;5.3: Zeisel 大脑数据集中分配给线粒体转录本的 UMI 百分比,与分配给加标转录本的UMI百分比作图。每个点代表一个细胞，并根据它是否被视为低质量并丢弃进行着色。</figcaption></figure>
</div>
</div>
</div>
</section><section id="去除劣质细胞" class="level2" data-number="5.4"><h2 data-number="5.4" class="anchored" data-anchor-id="去除劣质细胞">
<span class="header-section-number">5.4</span> 去除劣质细胞</h2>
<p>对于常规分析，删除</p>
<div class="cell" data-hash="QC_cache/html/unnamed-chunk-12_9008a9ddd0c0e49f90398557094bac94">
<details><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">filtered</span> <span class="op">&lt;-</span> <span class="va">sce.416b</span><span class="op">[</span>,<span class="op">!</span><span class="va">reasons</span><span class="op">$</span><span class="va">discard</span><span class="op">]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>标记</p>
<div class="cell" data-hash="QC_cache/html/unnamed-chunk-13_1b8a6137740447f090ad9a1dddae1a2c">
<details><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">marked</span> <span class="op">&lt;-</span> <span class="va">sce.416b</span></span>
<span><span class="va">marked</span><span class="op">$</span><span class="va">discard</span> <span class="op">&lt;-</span> <span class="va">reasons</span><span class="op">$</span><span class="va">discard</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->

</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./basics.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Basics</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./normalization.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Normalization</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb13" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># 质量控制</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>scRNA-seq数据中的低质量文库（library）可能来自多种来源，例如解离过程中的细胞损伤或文库制备失败（如低效的逆转录或PCR扩增），这些通常表现为总计数低、表达基因少、线粒体或尖峰比例高的"细胞"。这些低质量的文库是有问题的，因为它们会导致下游分析中的误导性结果。</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>为了避免或至少缓解这些问题，我们需要在分析开始时删除有问题的细胞(cell)，此步骤通常称为细胞的质量控制 （QC）。</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(GenomicRanges<span class="sc">::</span>setdiff)</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(dplyr<span class="sc">::</span>filter)</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache: true</span></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: data.416b</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "加载数据LunSpikeInData:工作流第1章" </span></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scRNAseq)</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ensembldb)</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b <span class="ot">&lt;-</span> <span class="fu">LunSpikeInData</span>(<span class="at">which=</span><span class="st">"416b"</span>) </span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b<span class="sc">$</span>block <span class="ot">&lt;-</span> <span class="fu">factor</span>(sce<span class="fl">.416</span>b<span class="sc">$</span>block)</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b</span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a><span class="fu">## QC metrics</span></span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**The library size** is defined as the total sum of counts across all relevant features for each cell. Cells with small library sizes are of low quality.库大小定义为每个细胞所有基因的计数总和。</span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**The number of expressed features in each cell** is defined as the number of endogenous genes with non-zero counts for that cell. Any cell with very few expressed genes is likely to be of poor quality. 每个细胞中表达特征的数量被定义为该细胞具有非零计数的内源性基因的数量。</span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**The proportion of reads mapped to spike-in transcripts** is calculated relative to the total count across all features (including spike-ins) for each cell.High proportions are indicative of poor-quality cells。 相对于每个细胞所有特征（包括加标）的总计数，计算映射到加标转录物中的读数比例。高比例表示细胞质量差</span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**In the absence of spike-in transcripts, the proportion of reads mapped to genes in the mitochondrial genome** can be used. High proportions are indicative of poor-quality cells。在没有加标转录物的情况下，可以使用映射到线粒体基因组中基因的读数比例。高比例表示细胞质量差。</span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>对于每个细胞，我们使用&nbsp;<span class="co">[</span><span class="ot">*scater*</span><span class="co">](https://bioconductor.org/packages/3.18/scater)</span>&nbsp;包中的函数<span class="in">`perCellQCMetrics()`</span>计算这些 QC 指标,该列<span class="in">`sum`</span>包含每个细胞文库大小的总计数，该列<span class="in">`detected`</span>包含检测到的基因数。 该列<span class="in">`subsets_Mito_percent`</span>包含映射到线粒体转录本的读取百分比，该列<span class="in">`altexps_ERCC_percent`</span>包含映射到 ERCC 转录本的读取百分比。</span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b</span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>location <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">rowRanges</span>(sce<span class="fl">.416</span>b)</span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a>location</span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a>is.mito <span class="ot">&lt;-</span> <span class="fu">any</span>(<span class="fu">seqnames</span>(location)<span class="sc">==</span><span class="st">"MT"</span>)</span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scuttle)</span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">perCellQCMetrics</span>(sce<span class="fl">.416</span>b, <span class="at">subsets=</span><span class="fu">list</span>(<span class="at">Mito=</span>is.mito))</span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df)</span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(df<span class="sc">$</span>sum)</span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(df<span class="sc">$</span>altexps_ERCC_percent)</span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-56"><a href="#cb13-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-57"><a href="#cb13-57" aria-hidden="true" tabindex="-1"></a>或者 <span class="in">`addPerCellQC()`</span>计算每个细胞的 QC 统计数据并将其附加到对象的<span class="in">`colData`</span> 允许我们将所有相关信息保留在单个对象<span class="in">`SingleCellExperiment`</span>中以供以后操作。</span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-61"><a href="#cb13-61" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-62"><a href="#cb13-62" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b <span class="ot">&lt;-</span> <span class="fu">addPerCellQCMetrics</span>(sce<span class="fl">.416</span>b, <span class="at">subsets=</span><span class="fu">list</span>(<span class="at">Mito=</span>is.mito))</span>
<span id="cb13-63"><a href="#cb13-63" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">colData</span>(sce<span class="fl">.416</span>b))</span>
<span id="cb13-64"><a href="#cb13-64" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(<span class="fu">colData</span>(sce<span class="fl">.416</span>b))</span>
<span id="cb13-65"><a href="#cb13-65" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b</span>
<span id="cb13-66"><a href="#cb13-66" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-67"><a href="#cb13-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-68"><a href="#cb13-68" aria-hidden="true" tabindex="-1"></a><span class="fu">## 识别低质量细胞</span></span>
<span id="cb13-69"><a href="#cb13-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-70"><a href="#cb13-70" aria-hidden="true" tabindex="-1"></a><span class="fu">### 固定阈值</span></span>
<span id="cb13-71"><a href="#cb13-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-72"><a href="#cb13-72" aria-hidden="true" tabindex="-1"></a>识别低质量细胞的最简单方法是对 QC 指标应用固定阈值。 例如，如果细胞的文库大小低于 100,000 次读取;表达少于5,000个基因;峰值比例超过10%;或线粒体比例高于10%，我们可能会认为它们的质量较低。</span>
<span id="cb13-73"><a href="#cb13-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-76"><a href="#cb13-76" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-77"><a href="#cb13-77" aria-hidden="true" tabindex="-1"></a>qc.lib <span class="ot">&lt;-</span> df<span class="sc">$</span>sum <span class="sc">&lt;</span> <span class="fl">1e5</span></span>
<span id="cb13-78"><a href="#cb13-78" aria-hidden="true" tabindex="-1"></a>qc.nexprs <span class="ot">&lt;-</span> df<span class="sc">$</span>detected <span class="sc">&lt;</span> <span class="fl">5e3</span></span>
<span id="cb13-79"><a href="#cb13-79" aria-hidden="true" tabindex="-1"></a>qc.spike <span class="ot">&lt;-</span> df<span class="sc">$</span>altexps_ERCC_percent <span class="sc">&gt;</span> <span class="dv">10</span></span>
<span id="cb13-80"><a href="#cb13-80" aria-hidden="true" tabindex="-1"></a>qc.mito <span class="ot">&lt;-</span> df<span class="sc">$</span>subsets_Mito_percent <span class="sc">&gt;</span> <span class="dv">10</span></span>
<span id="cb13-81"><a href="#cb13-81" aria-hidden="true" tabindex="-1"></a>discard <span class="ot">&lt;-</span> qc.lib <span class="sc">|</span> qc.nexprs <span class="sc">|</span> qc.spike <span class="sc">|</span> qc.mito</span>
<span id="cb13-82"><a href="#cb13-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-83"><a href="#cb13-83" aria-hidden="true" tabindex="-1"></a><span class="co"># 汇总</span></span>
<span id="cb13-84"><a href="#cb13-84" aria-hidden="true" tabindex="-1"></a><span class="fu">tibble</span>(<span class="at">LibSize=</span><span class="fu">sum</span>(qc.lib), <span class="at">NExprs=</span><span class="fu">sum</span>(qc.nexprs),</span>
<span id="cb13-85"><a href="#cb13-85" aria-hidden="true" tabindex="-1"></a>    <span class="at">SpikeProp=</span><span class="fu">sum</span>(qc.spike), <span class="at">MitoProp=</span><span class="fu">sum</span>(qc.mito), <span class="at">Total=</span><span class="fu">sum</span>(discard))</span>
<span id="cb13-86"><a href="#cb13-86" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-87"><a href="#cb13-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-88"><a href="#cb13-88" aria-hidden="true" tabindex="-1"></a><span class="fu">### 自适应阈值</span></span>
<span id="cb13-89"><a href="#cb13-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-90"><a href="#cb13-90" aria-hidden="true" tabindex="-1"></a>假设大部分数据集由高质量的细胞组成，然后，我们根据所有细胞中每个指标的中位数绝对偏差（median absolute deviation ，MAD）来识别各种QC指标的异常值细胞。默认情况下，如果一个值在"有问题"的方向上距离中位数超过3个MAD，我们将其视为异常值。即此类过滤器将保留 99% 遵循正态分布的非异常值。</span>
<span id="cb13-91"><a href="#cb13-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-94"><a href="#cb13-94" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-95"><a href="#cb13-95" aria-hidden="true" tabindex="-1"></a><span class="co">#?perCellQCFilters</span></span>
<span id="cb13-96"><a href="#cb13-96" aria-hidden="true" tabindex="-1"></a>reasons <span class="ot">&lt;-</span> <span class="fu">perCellQCFilters</span>(df, </span>
<span id="cb13-97"><a href="#cb13-97" aria-hidden="true" tabindex="-1"></a>                            <span class="at">sub.fields=</span><span class="fu">c</span>(<span class="st">"subsets_Mito_percent"</span>,</span>
<span id="cb13-98"><a href="#cb13-98" aria-hidden="true" tabindex="-1"></a>                                         <span class="st">"altexps_ERCC_percent"</span>))</span>
<span id="cb13-99"><a href="#cb13-99" aria-hidden="true" tabindex="-1"></a><span class="fu">colSums</span>(<span class="fu">as.matrix</span>(reasons)) <span class="co"># apply(as.matrix(reasons),2,sum)</span></span>
<span id="cb13-100"><a href="#cb13-100" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(reasons<span class="sc">$</span>discard)</span>
<span id="cb13-101"><a href="#cb13-101" aria-hidden="true" tabindex="-1"></a><span class="fu">attr</span>(reasons<span class="sc">$</span>low_lib_size, <span class="st">"thresholds"</span>)</span>
<span id="cb13-102"><a href="#cb13-102" aria-hidden="true" tabindex="-1"></a><span class="fu">attr</span>(reasons<span class="sc">$</span>low_n_features, <span class="st">"thresholds"</span>)</span>
<span id="cb13-103"><a href="#cb13-103" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-104"><a href="#cb13-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-105"><a href="#cb13-105" aria-hidden="true" tabindex="-1"></a><span class="fu">### 其他方法</span></span>
<span id="cb13-106"><a href="#cb13-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-107"><a href="#cb13-107" aria-hidden="true" tabindex="-1"></a>根据每个细胞的 QC 指标识别高维空间中的异常值</span>
<span id="cb13-108"><a href="#cb13-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-111"><a href="#cb13-111" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-112"><a href="#cb13-112" aria-hidden="true" tabindex="-1"></a>stats <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">log10</span>(df<span class="sc">$</span>sum), <span class="fu">log10</span>(df<span class="sc">$</span>detected),</span>
<span id="cb13-113"><a href="#cb13-113" aria-hidden="true" tabindex="-1"></a>    df<span class="sc">$</span>subsets_Mito_percent, df<span class="sc">$</span>altexps_ERCC_percent)</span>
<span id="cb13-114"><a href="#cb13-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-115"><a href="#cb13-115" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(robustbase)</span>
<span id="cb13-116"><a href="#cb13-116" aria-hidden="true" tabindex="-1"></a>outlying <span class="ot">&lt;-</span> <span class="fu">adjOutlyingness</span>(stats, <span class="at">only.outlyingness =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-117"><a href="#cb13-117" aria-hidden="true" tabindex="-1"></a>multi.outlier <span class="ot">&lt;-</span> <span class="fu">isOutlier</span>(outlying, <span class="at">type =</span> <span class="st">"higher"</span>)</span>
<span id="cb13-118"><a href="#cb13-118" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(multi.outlier)</span>
<span id="cb13-119"><a href="#cb13-119" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-120"><a href="#cb13-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-121"><a href="#cb13-121" aria-hidden="true" tabindex="-1"></a><span class="fu">## 检查诊断图</span></span>
<span id="cb13-122"><a href="#cb13-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-125"><a href="#cb13-125" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-126"><a href="#cb13-126" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache: true</span></span>
<span id="cb13-127"><a href="#cb13-127" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-diagnosis-plot</span></span>
<span id="cb13-128"><a href="#cb13-128" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "数据集中每个批次和表型的 QC 指标分布。每个点代表一个细胞，并分别根据其是否被丢弃而着色。"</span></span>
<span id="cb13-129"><a href="#cb13-129" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(sce<span class="fl">.416</span>b) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">colData</span>(sce<span class="fl">.416</span>b), df)</span>
<span id="cb13-130"><a href="#cb13-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-131"><a href="#cb13-131" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b<span class="sc">$</span>block <span class="ot">&lt;-</span> <span class="fu">factor</span>(sce<span class="fl">.416</span>b<span class="sc">$</span>block)</span>
<span id="cb13-132"><a href="#cb13-132" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b<span class="sc">$</span>phenotype <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"induced"</span>, sce<span class="fl">.416</span>b<span class="sc">$</span>phenotype),</span>
<span id="cb13-133"><a href="#cb13-133" aria-hidden="true" tabindex="-1"></a>    <span class="st">"induced"</span>, <span class="st">"wild type"</span>)</span>
<span id="cb13-134"><a href="#cb13-134" aria-hidden="true" tabindex="-1"></a>sce<span class="fl">.416</span>b<span class="sc">$</span>discard <span class="ot">&lt;-</span> reasons<span class="sc">$</span>discard</span>
<span id="cb13-135"><a href="#cb13-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-136"><a href="#cb13-136" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb13-137"><a href="#cb13-137" aria-hidden="true" tabindex="-1"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(</span>
<span id="cb13-138"><a href="#cb13-138" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plotColData</span>(sce<span class="fl">.416</span>b, <span class="at">x=</span><span class="st">"block"</span>, <span class="at">y=</span><span class="st">"sum"</span>, <span class="at">colour_by=</span><span class="st">"discard"</span>,</span>
<span id="cb13-139"><a href="#cb13-139" aria-hidden="true" tabindex="-1"></a>        <span class="at">other_fields=</span><span class="st">"phenotype"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>phenotype) <span class="sc">+</span> </span>
<span id="cb13-140"><a href="#cb13-140" aria-hidden="true" tabindex="-1"></a>        <span class="fu">scale_y_log10</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Total count"</span>),</span>
<span id="cb13-141"><a href="#cb13-141" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plotColData</span>(sce<span class="fl">.416</span>b, <span class="at">x=</span><span class="st">"block"</span>, <span class="at">y=</span><span class="st">"detected"</span>, <span class="at">colour_by=</span><span class="st">"discard"</span>, </span>
<span id="cb13-142"><a href="#cb13-142" aria-hidden="true" tabindex="-1"></a>        <span class="at">other_fields=</span><span class="st">"phenotype"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>phenotype) <span class="sc">+</span> </span>
<span id="cb13-143"><a href="#cb13-143" aria-hidden="true" tabindex="-1"></a>        <span class="fu">scale_y_log10</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Detected features"</span>),</span>
<span id="cb13-144"><a href="#cb13-144" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plotColData</span>(sce<span class="fl">.416</span>b, <span class="at">x=</span><span class="st">"block"</span>, <span class="at">y=</span><span class="st">"subsets_Mito_percent"</span>, </span>
<span id="cb13-145"><a href="#cb13-145" aria-hidden="true" tabindex="-1"></a>        <span class="at">colour_by=</span><span class="st">"discard"</span>, <span class="at">other_fields=</span><span class="st">"phenotype"</span>) <span class="sc">+</span> </span>
<span id="cb13-146"><a href="#cb13-146" aria-hidden="true" tabindex="-1"></a>        <span class="fu">facet_wrap</span>(<span class="sc">~</span>phenotype) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Mito percent"</span>),</span>
<span id="cb13-147"><a href="#cb13-147" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plotColData</span>(sce<span class="fl">.416</span>b, <span class="at">x=</span><span class="st">"block"</span>, <span class="at">y=</span><span class="st">"altexps_ERCC_percent"</span>, </span>
<span id="cb13-148"><a href="#cb13-148" aria-hidden="true" tabindex="-1"></a>        <span class="at">colour_by=</span><span class="st">"discard"</span>, <span class="at">other_fields=</span><span class="st">"phenotype"</span>) <span class="sc">+</span> </span>
<span id="cb13-149"><a href="#cb13-149" aria-hidden="true" tabindex="-1"></a>        <span class="fu">facet_wrap</span>(<span class="sc">~</span>phenotype) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"ERCC percent"</span>),</span>
<span id="cb13-150"><a href="#cb13-150" aria-hidden="true" tabindex="-1"></a>    <span class="at">ncol=</span><span class="dv">1</span></span>
<span id="cb13-151"><a href="#cb13-151" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb13-152"><a href="#cb13-152" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-153"><a href="#cb13-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-154"><a href="#cb13-154" aria-hidden="true" tabindex="-1"></a>另一个有用的诊断涉及绘制线粒体计数与其他一些 QC 指标的比例。 目的是确认没有细胞同时具有大量总计数和大量线粒体计数，以确保我们不会无意中去除恰好具有高度代谢活性的高质量细胞（例如肝细胞）。</span>
<span id="cb13-155"><a href="#cb13-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-158"><a href="#cb13-158" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-159"><a href="#cb13-159" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: dataset.zeisel </span></span>
<span id="cb13-160"><a href="#cb13-160" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "加载数据ZeiselBrainData：工作流第2章"</span></span>
<span id="cb13-161"><a href="#cb13-161" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb13-162"><a href="#cb13-162" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache: true</span></span>
<span id="cb13-163"><a href="#cb13-163" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scRNAseq)</span>
<span id="cb13-164"><a href="#cb13-164" aria-hidden="true" tabindex="-1"></a>sce.zeisel <span class="ot">&lt;-</span> <span class="fu">ZeiselBrainData</span>()</span>
<span id="cb13-165"><a href="#cb13-165" aria-hidden="true" tabindex="-1"></a>sce.zeisel</span>
<span id="cb13-166"><a href="#cb13-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-167"><a href="#cb13-167" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb13-168"><a href="#cb13-168" aria-hidden="true" tabindex="-1"></a>sce.zeisel <span class="ot">&lt;-</span> <span class="fu">aggregateAcrossFeatures</span>(sce.zeisel, </span>
<span id="cb13-169"><a href="#cb13-169" aria-hidden="true" tabindex="-1"></a>    <span class="at">id=</span><span class="fu">sub</span>(<span class="st">"_loc[0-9]+$"</span>, <span class="st">""</span>, <span class="fu">rownames</span>(sce.zeisel)))</span>
<span id="cb13-170"><a href="#cb13-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-171"><a href="#cb13-171" aria-hidden="true" tabindex="-1"></a><span class="co">#--- gene-annotation ---#</span></span>
<span id="cb13-172"><a href="#cb13-172" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Mm.eg.db)</span>
<span id="cb13-173"><a href="#cb13-173" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(sce.zeisel)<span class="sc">$</span>Ensembl <span class="ot">&lt;-</span> <span class="fu">mapIds</span>(org.Mm.eg.db, </span>
<span id="cb13-174"><a href="#cb13-174" aria-hidden="true" tabindex="-1"></a>    <span class="at">keys=</span><span class="fu">rownames</span>(sce.zeisel), <span class="at">keytype=</span><span class="st">"SYMBOL"</span>, <span class="at">column=</span><span class="st">"ENSEMBL"</span>)</span>
<span id="cb13-175"><a href="#cb13-175" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-176"><a href="#cb13-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-179"><a href="#cb13-179" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-180"><a href="#cb13-180" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-mito-sum</span></span>
<span id="cb13-181"><a href="#cb13-181" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Zeisel 大脑数据集 中分配给线粒体转录本的 UMI 百分比，与 UMI 总数作图。每个点代表一个细胞，并根据它是否被视为低质量并丢弃进行着色。"</span></span>
<span id="cb13-182"><a href="#cb13-182" aria-hidden="true" tabindex="-1"></a>sce.zeisel <span class="ot">&lt;-</span> <span class="fu">addPerCellQC</span>(</span>
<span id="cb13-183"><a href="#cb13-183" aria-hidden="true" tabindex="-1"></a>  sce.zeisel,<span class="at">subsets=</span><span class="fu">list</span>(<span class="at">Mt=</span><span class="fu">rowData</span>(sce.zeisel)<span class="sc">$</span>featureType<span class="sc">==</span><span class="st">"mito"</span>))</span>
<span id="cb13-184"><a href="#cb13-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-185"><a href="#cb13-185" aria-hidden="true" tabindex="-1"></a>qc <span class="ot">&lt;-</span> <span class="fu">quickPerCellQC</span>(<span class="fu">colData</span>(sce.zeisel), </span>
<span id="cb13-186"><a href="#cb13-186" aria-hidden="true" tabindex="-1"></a>    <span class="at">sub.fields=</span><span class="fu">c</span>(<span class="st">"altexps_ERCC_percent"</span>, <span class="st">"subsets_Mt_percent"</span>))</span>
<span id="cb13-187"><a href="#cb13-187" aria-hidden="true" tabindex="-1"></a>sce.zeisel<span class="sc">$</span>discard <span class="ot">&lt;-</span> qc<span class="sc">$</span>discard</span>
<span id="cb13-188"><a href="#cb13-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-189"><a href="#cb13-189" aria-hidden="true" tabindex="-1"></a><span class="fu">plotColData</span>(sce.zeisel, <span class="at">x=</span><span class="st">"sum"</span>, <span class="at">y=</span><span class="st">"subsets_Mt_percent"</span>, <span class="at">colour_by=</span><span class="st">"discard"</span>)</span>
<span id="cb13-190"><a href="#cb13-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-191"><a href="#cb13-191" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-192"><a href="#cb13-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-195"><a href="#cb13-195" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-196"><a href="#cb13-196" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-mito-ERCC</span></span>
<span id="cb13-197"><a href="#cb13-197" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Zeisel 大脑数据集中分配给线粒体转录本的 UMI 百分比,与分配给加标转录本的UMI百分比作图。每个点代表一个细胞，并根据它是否被视为低质量并丢弃进行着色。"</span></span>
<span id="cb13-198"><a href="#cb13-198" aria-hidden="true" tabindex="-1"></a><span class="fu">plotColData</span>(sce.zeisel, <span class="at">x=</span><span class="st">"altexps_ERCC_percent"</span>, <span class="at">y=</span><span class="st">"subsets_Mt_percent"</span>,</span>
<span id="cb13-199"><a href="#cb13-199" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour_by=</span><span class="st">"discard"</span>)</span>
<span id="cb13-200"><a href="#cb13-200" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-201"><a href="#cb13-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-202"><a href="#cb13-202" aria-hidden="true" tabindex="-1"></a><span class="fu">## 去除劣质细胞</span></span>
<span id="cb13-203"><a href="#cb13-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-204"><a href="#cb13-204" aria-hidden="true" tabindex="-1"></a>对于常规分析，删除</span>
<span id="cb13-205"><a href="#cb13-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-208"><a href="#cb13-208" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-209"><a href="#cb13-209" aria-hidden="true" tabindex="-1"></a>filtered <span class="ot">&lt;-</span> sce<span class="fl">.416</span>b[,<span class="sc">!</span>reasons<span class="sc">$</span>discard]</span>
<span id="cb13-210"><a href="#cb13-210" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-211"><a href="#cb13-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-212"><a href="#cb13-212" aria-hidden="true" tabindex="-1"></a>标记</span>
<span id="cb13-213"><a href="#cb13-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-216"><a href="#cb13-216" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-217"><a href="#cb13-217" aria-hidden="true" tabindex="-1"></a>marked <span class="ot">&lt;-</span> sce<span class="fl">.416</span>b</span>
<span id="cb13-218"><a href="#cb13-218" aria-hidden="true" tabindex="-1"></a>marked<span class="sc">$</span>discard <span class="ot">&lt;-</span> reasons<span class="sc">$</span>discard</span>
<span id="cb13-219"><a href="#cb13-219" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>